#!/usr/bin/env python
# coding: utf-8

# # Cluster by accessory gene expression
# 
# This notebook is visualizing the expression of accessory genes in the PAO1 and PA14 compendia

# In[1]:


import os
import pandas as pd
import plotnine as pn
from core_acc_modules import paths_explore, utils


# ## Load data
# 
# Raw data was processed in an external repository by a member of the Hogan lab: https://github.com/hoganlab-dartmouth/pa-seq-compendia
# 
# The basic processing steps to process the data were as follows:
# 1. _P. aeruginosa_ transcriptome data was downloaded from SRA (~4K samples)
# 2. Aligned and quantified samples using Salmon against PAO1 and PA14 references
# 3. Quantified results were validated by performing a differential expression analysis and comparing the DEGs against the original publication.
# 
# _Note:_
# * Not sure yet where this data will permanently be stored but there are plans to share it. Currently this is being housed locally to run this analysis

# In[2]:


# Expression data files 
pao1_expression_filename = paths_explore.PAO1_GE
pa14_expression_filename = paths_explore.PA14_GE

# File containing table to map sample id to strain name
sample_to_strain_filename = paths_explore.SAMPLE_TO_STRAIN


# In[3]:


# Load expression data
# Matrices will be sample x gene after taking the transpose
pao1_expression = pd.read_csv(
    pao1_expression_filename,
    index_col=0, 
    header=0, 
    usecols=lambda c:c != 'Unnamed: 0').T

pa14_expression = pd.read_csv(
    pa14_expression_filename, 
    index_col=0, 
    header=0, 
    usecols=lambda c:c != 'Unnamed: 0').T


# In[4]:


print(pao1_expression.shape)
pao1_expression.head()


# In[5]:


print(pa14_expression.shape)
pa14_expression.head()


# In[6]:


# Load metadata
# Set index to experiment id, which is what we will use to map to expression data
sample_to_strain_table_full = pd.read_csv(sample_to_strain_filename, index_col=2)
sample_to_strain_table_full.head() 


# ## Process data

# In[7]:


# Format expression data indices so that values can be mapped to `sample_to_strain_table`
pao1_index_processed = pao1_expression.index.str.split("/").str[0]
pa14_index_processed = pa14_expression.index.str.split("/").str[0]

pao1_expression.index = pao1_index_processed
pa14_expression.index = pa14_index_processed


# In[8]:


pao1_expression.head()


# In[9]:


# Aggregate boolean labels into a single strain label
aggregated_label = []
for exp_id in list(sample_to_strain_table_full.index):
    if sample_to_strain_table_full.loc[exp_id,"PAO1"].all() == True:
        aggregated_label.append("PAO1")
    elif sample_to_strain_table_full.loc[exp_id,"PA14"].all() == True:
        aggregated_label.append("PA14")
    elif sample_to_strain_table_full.loc[exp_id,"PAK"].all() == True:
        aggregated_label.append("PAK")
    elif sample_to_strain_table_full.loc[exp_id,"ClinicalIsolate"].all() == True:
        aggregated_label.append("Clinical Isolate")
    else:
        aggregated_label.append("NA")
        
sample_to_strain_table_full["Strain type"] = aggregated_label

sample_to_strain_table = sample_to_strain_table_full["Strain type"].to_frame()


# In[10]:


sample_to_strain_table.head()


# In[11]:


# What are the NA strains?
print(sample_to_strain_table.shape)
na_sample_ids = list(sample_to_strain_table[sample_to_strain_table["Strain type"]== "NA"].index)
print(len(na_sample_ids))


# In[12]:


# Most of these strains don't have strain annotation available
sample_to_strain_table_full.loc[na_sample_ids, "Strain"].isnull().sum()


# In[13]:


# Use to manually lookup these experiments in SRA
#sample_to_strain_table_full.loc[na_sample_ids, "Strain"]


# In[14]:


pao1_expression.head()


# ## Get core and accessory genes

# In[15]:


# Get mapping between PAO1 and PA14 genes using PAO1 reference
gene_annot_file = paths_explore.GENE_PAO1_ANNOT
gene_mapping_pao1 = utils.get_pao1_pa14_gene_map(gene_annot_file, 'pao1')


# In[16]:


# Get mapping between PAO1 and PA14 genes using PA14 reference
gene_annot_file = paths_explore.GENE_PA14_ANNOT
gene_mapping_pa14 = utils.get_pao1_pa14_gene_map(gene_annot_file, 'pa14')


# In[17]:


# Get core genes: genes that have a homolog between PAO1 and PA14
core_pao1_genes, core_pa14_genes = utils.get_core_genes(gene_mapping_pao1,
                                                        gene_mapping_pa14,
                                                        False)
print(f"Number of PAO1 core genes: {len(core_pao1_genes)}")
print(f"Number of PA14 core genes: {len(core_pa14_genes)}")


# In[18]:


# Select only core genes that are included in my dataset
pao1_ref_genes = pao1_expression.columns
my_core_pao1_genes = list(set(core_pao1_genes).intersection(pao1_ref_genes))

print(f"Number of PAO1 core genes in my dataset: {len(my_core_pao1_genes)}")


# In[19]:


# Select only core genes that are included in my dataset
pa14_ref_genes = pa14_expression.columns
my_core_pa14_genes = list(set(core_pa14_genes).intersection(pa14_ref_genes))

print(f"Number of PA14 core genes in my dataset: {len(my_core_pa14_genes)}")


# In[20]:


# Get PAO1-specific genes
pao1_acc = list(set(pao1_ref_genes) - set(my_core_pao1_genes))
print(f"Number of PAO1-specific genes: {len(pao1_acc)}")


# In[21]:


## TO DO: Get PA14-specific genes
pa14_acc = list(set(pa14_ref_genes) - set(my_core_pa14_genes))
print(f"Number of PA14-specific genes: {len(pa14_acc)}")


# ## Create df for plotting

# In[22]:


# Merge strain to expression matrix
pao1_expression_label = pao1_expression.merge(sample_to_strain_table, left_index=True, right_index=True)
pa14_expression_label = pa14_expression.merge(sample_to_strain_table, left_index=True, right_index=True)

pao1_expression_label.head()


# In[23]:


# Create dataframe with columns:
# accessory gene ids | median accessory expression | strain label

# PAO1
pao1_acc_expression_label = pao1_expression_label[pao1_acc + ["Strain type"]]
pao1_acc_expression = pao1_expression_label[pao1_acc]
pao1_acc_expression_label["median acc expression"] = pao1_acc_expression.median(axis=1)

# PA14
pa14_acc_expression_label = pa14_expression_label[pa14_acc + ["Strain type"]]
pa14_acc_expression = pa14_expression_label[pa14_acc]
pa14_acc_expression_label["median acc expression"] = pa14_acc_expression.median(axis=1)

pao1_acc_expression_label.head()


# In[24]:


pa14_acc_expression_label.head()


# In[25]:


# Merge PAO1 and PA14 dataframes
pao1_pa14_acc_expression_label = pao1_acc_expression_label.merge(
    pa14_acc_expression_label,
    left_index=True,
    right_index=True,
    suffixes=["_pao1", "_pa14"]
)

pao1_pa14_acc_expression_label.head()


# In[26]:


# Plot
fig = pn.ggplot(
    pao1_pa14_acc_expression_label, 
    pn.aes(x='median acc expression_pao1', y='median acc expression_pa14'))
fig += pn.geom_point(pn.aes(color='Strain type_pao1'), alpha=0.4)
fig += pn.labs(x ='median expression of PAO1-only genes (TPM)',
            y = 'median expression of PA14-only genes (TPM)',
            title = 'TPM of accessory genes')
fig += pn.theme_bw()
fig += pn.theme(
    legend_title_align = "center",
    plot_background=pn.element_rect(fill='white'),
    legend_key=pn.element_rect(fill='white', colour='white'), 
    legend_title=pn.element_text(family='sans-serif', size=15),
    legend_text=pn.element_text(family='sans-serif', size=12),
    plot_title=pn.element_text(family='sans-serif', size=15),
    axis_text=pn.element_text(family='sans-serif', size=10),
    axis_title=pn.element_text(family='sans-serif', size=12)
    )
fig += pn.guides(colour=pn.guide_legend(override_aes={'alpha': 1}))

print(fig)


# In[27]:


# Plot log-scaled
fig2 = pn.ggplot(
    pao1_pa14_acc_expression_label, 
    pn.aes(x='median acc expression_pao1', y='median acc expression_pa14'))
fig2 += pn.scales.scale_x_log10()
fig2 += pn.scales.scale_y_log10()
fig2 += pn.geom_point(pn.aes(color='Strain type_pao1'), alpha=0.4)
fig2 += pn.labs(x ='median expression of PAO1-only genes (log10 TPM)',
            y = 'median expression of PA14-only genes (log10 TPM)',
            title = 'TPM of accessory genes')
fig2 += pn.theme_bw()
fig2 += pn.theme(
    legend_title_align = "center",
    plot_background=pn.element_rect(fill='white'),
    legend_key=pn.element_rect(fill='white', colour='white'), 
    legend_title=pn.element_text(family='sans-serif', size=15),
    legend_text=pn.element_text(family='sans-serif', size=12),
    plot_title=pn.element_text(family='sans-serif', size=15),
    axis_text=pn.element_text(family='sans-serif', size=10),
    axis_title=pn.element_text(family='sans-serif', size=12)
    )
fig2 += pn.guides(colour=pn.guide_legend(override_aes={'alpha': 1}))


print(fig2)


# **Takeaway:** 
# * This is a positive control that PAO1 annotated samples have higher median expression of PAO1-only genes compared to PA14-only genes. And similarly, PA14 annotated samples have higher median expression of PA14-only genes compared to PAO1-only genes. In other words, we expect that PA14-only genes will have either 0 or very low values in PAO1 samples and vice versa.
# 
# * This result also shows that we can anticipate a very clear binning of our samples into PAO1 and PA14 if we use mapping rates. 
# 
# * Note: The NA strains are those where the strain information was not available in the metadata. By a quick manual spot check it looks like a bunch were clinical isolates (which is good since these NA seem to cluster with other clinical isolates).
