{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8815744c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['path:pae00010\\tGlycolysis / Gluconeogenesis - Pseudomonas aeruginosa PAO1',\n",
       " 'path:pae00020\\tCitrate cycle (TCA cycle) - Pseudomonas aeruginosa PAO1',\n",
       " 'path:pae00030\\tPentose phosphate pathway - Pseudomonas aeruginosa PAO1',\n",
       " 'path:pae00040\\tPentose and glucuronate interconversions - Pseudomonas aeruginosa PAO1',\n",
       " 'path:pae00051\\tFructose and mannose metabolism - Pseudomonas aeruginosa PAO1']"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio.KEGG import REST\n",
    "\n",
    "pao1_pathways = REST.kegg_list(\"pathway\", \"pae\").read()\n",
    "\n",
    "pao1_pathways.split(\"\\n\")[0:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d270f704",
   "metadata": {},
   "outputs": [],
   "source": [
    "# This python package or the R getGeneSet both return only the KEGG pathways but not the modules\n",
    "# Not sure how to get modules but don't worry about this right now\n",
    "# Python we will need to parse each line\n",
    "# OR R need to figure out how to map nonsense ids to gene ids\n",
    "# Need to make sure to have pathway name and id too\n",
    "# https://biopython-tutorial.readthedocs.io/en/latest/notebooks/18%20-%20KEGG.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "dd82d6d1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "123\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['path:pae00010',\n",
       " 'path:pae00020',\n",
       " 'path:pae00030',\n",
       " 'path:pae00040',\n",
       " 'path:pae00051',\n",
       " 'path:pae00052',\n",
       " 'path:pae00053',\n",
       " 'path:pae00061',\n",
       " 'path:pae00071',\n",
       " 'path:pae00130',\n",
       " 'path:pae00190',\n",
       " 'path:pae00220',\n",
       " 'path:pae00230',\n",
       " 'path:pae00240',\n",
       " 'path:pae00250',\n",
       " 'path:pae00260',\n",
       " 'path:pae00261',\n",
       " 'path:pae00270',\n",
       " 'path:pae00280',\n",
       " 'path:pae00281',\n",
       " 'path:pae00290',\n",
       " 'path:pae00300',\n",
       " 'path:pae00310',\n",
       " 'path:pae00330',\n",
       " 'path:pae00332',\n",
       " 'path:pae00340',\n",
       " 'path:pae00350',\n",
       " 'path:pae00360',\n",
       " 'path:pae00361',\n",
       " 'path:pae00362',\n",
       " 'path:pae00364',\n",
       " 'path:pae00380',\n",
       " 'path:pae00400',\n",
       " 'path:pae00401',\n",
       " 'path:pae00405',\n",
       " 'path:pae00410',\n",
       " 'path:pae00430',\n",
       " 'path:pae00440',\n",
       " 'path:pae00450',\n",
       " 'path:pae00460',\n",
       " 'path:pae00470',\n",
       " 'path:pae00480',\n",
       " 'path:pae00500',\n",
       " 'path:pae00520',\n",
       " 'path:pae00521',\n",
       " 'path:pae00523',\n",
       " 'path:pae00525',\n",
       " 'path:pae00540',\n",
       " 'path:pae00541',\n",
       " 'path:pae00550',\n",
       " 'path:pae00561',\n",
       " 'path:pae00562',\n",
       " 'path:pae00564',\n",
       " 'path:pae00565',\n",
       " 'path:pae00590',\n",
       " 'path:pae00592',\n",
       " 'path:pae00600',\n",
       " 'path:pae00620',\n",
       " 'path:pae00622',\n",
       " 'path:pae00623',\n",
       " 'path:pae00625',\n",
       " 'path:pae00626',\n",
       " 'path:pae00627',\n",
       " 'path:pae00630',\n",
       " 'path:pae00633',\n",
       " 'path:pae00640',\n",
       " 'path:pae00643',\n",
       " 'path:pae00650',\n",
       " 'path:pae00660',\n",
       " 'path:pae00670',\n",
       " 'path:pae00680',\n",
       " 'path:pae00730',\n",
       " 'path:pae00740',\n",
       " 'path:pae00750',\n",
       " 'path:pae00760',\n",
       " 'path:pae00770',\n",
       " 'path:pae00780',\n",
       " 'path:pae00785',\n",
       " 'path:pae00790',\n",
       " 'path:pae00791',\n",
       " 'path:pae00860',\n",
       " 'path:pae00900',\n",
       " 'path:pae00903',\n",
       " 'path:pae00910',\n",
       " 'path:pae00920',\n",
       " 'path:pae00930',\n",
       " 'path:pae00970',\n",
       " 'path:pae00997',\n",
       " 'path:pae00999',\n",
       " 'path:pae01040',\n",
       " 'path:pae01053',\n",
       " 'path:pae01100',\n",
       " 'path:pae01110',\n",
       " 'path:pae01120',\n",
       " 'path:pae01200',\n",
       " 'path:pae01210',\n",
       " 'path:pae01212',\n",
       " 'path:pae01220',\n",
       " 'path:pae01230',\n",
       " 'path:pae01240',\n",
       " 'path:pae01250',\n",
       " 'path:pae01501',\n",
       " 'path:pae01502',\n",
       " 'path:pae01503',\n",
       " 'path:pae02010',\n",
       " 'path:pae02020',\n",
       " 'path:pae02024',\n",
       " 'path:pae02025',\n",
       " 'path:pae02030',\n",
       " 'path:pae02040',\n",
       " 'path:pae02060',\n",
       " 'path:pae03010',\n",
       " 'path:pae03018',\n",
       " 'path:pae03020',\n",
       " 'path:pae03030',\n",
       " 'path:pae03060',\n",
       " 'path:pae03070',\n",
       " 'path:pae03410',\n",
       " 'path:pae03420',\n",
       " 'path:pae03430',\n",
       " 'path:pae03440',\n",
       " 'path:pae03450',\n",
       " 'path:pae04122']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Clean up pathway names\n",
    "pao1_pathways_clean = []\n",
    "for line in pao1_pathways.rstrip().split(\"\\n\"):\n",
    "    entry, description = line.split(\"\\t\")\n",
    "    pao1_pathways_clean.append(entry)\n",
    "\n",
    "print(len(pao1_pathways_clean))\n",
    "pao1_pathways_clean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "402baee5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# List of all pae pathways in KEGG\n",
    "# as a check there are 123\n",
    "# https://www.kegg.jp/kegg-bin/search_pathway_text?map=pae&keyword=&mode=1&viewImage=true"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "895c61f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Example of a KEGG pathway page that is being parsed\n",
    "# https://www.genome.jp/dbget-bin/www_bget?pathway:pae00020\n",
    "# For a given pathway there are multiple modules and multiple gene ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "702470cb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "path:pae00010\n",
      "ENTRY       pae00010                    Pathway\n",
      "NAME        Glycolysis / Gluconeogenesis - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Glycolysis is the process of converting glucose into pyruvate and generating small amounts of ATP (energy) and NADH (reducing power). It is a central pathway that produces important precursor metabolites: six-carbon compounds of glucose-6P and fructose-6P and three-carbon compounds of glycerone-P, glyceraldehyde-3P, glycerate-3P, phosphoenolpyruvate, and pyruvate [MD:M00001]. Acetyl-CoA, another important precursor metabolite, is produced by oxidative decarboxylation of pyruvate [MD:M00307]. When the enzyme genes of this pathway are examined in completely sequenced genomes, the reaction steps of three-carbon compounds from glycerone-P to pyruvate form a conserved core module [MD:M00002], which is found in almost all organisms and which sometimes contains operon structures in bacterial genomes. Gluconeogenesis is a synthesis pathway of glucose from noncarbohydrate precursors. It is essentially a reversal of glycolysis with minor variations of alternative paths [MD:M00003].\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00010  Glycolysis / Gluconeogenesis\n",
      "MODULE      pae_M00002  Glycolysis, core module involving three-carbon compounds [PATH:pae00010]\n",
      "            pae_M00003  Gluconeogenesis, oxaloacetate => fructose-6P [PATH:pae00010]\n",
      "            pae_M00307  Pyruvate oxidation, pyruvate => acetyl-CoA [PATH:pae00010]\n",
      "DBLINKS     GO: 0006096 0006094\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3193  glk; glucokinase [KO:K00845] [EC:2.7.1.2]\n",
      "line in gene PA3193  glk; glucokinase [KO:K00845] [EC:2.7.1.2]\n",
      "            PA4732  pgi; glucose-6-phosphate isomerase [KO:K01810] [EC:5.3.1.9]\n",
      "line in gene PA4732  pgi; glucose-6-phosphate isomerase [KO:K01810] [EC:5.3.1.9]\n",
      "            PA5110  fbp; fructose-1,6-bisphosphatase [KO:K03841] [EC:3.1.3.11]\n",
      "line in gene PA5110  fbp; fructose-1,6-bisphosphatase [KO:K03841] [EC:3.1.3.11]\n",
      "            PA0555  fda; fructose-1,6-bisphosphate aldolase [KO:K01624] [EC:4.1.2.13]\n",
      "line in gene PA0555  fda; fructose-1,6-bisphosphate aldolase [KO:K01624] [EC:4.1.2.13]\n",
      "            PA4748  tpiA; triosephosphate isomerase [KO:K01803] [EC:5.3.1.1]\n",
      "line in gene PA4748  tpiA; triosephosphate isomerase [KO:K01803] [EC:5.3.1.1]\n",
      "            PA3195  gapA; glyceraldehyde 3-phosphate dehydrogenase [KO:K00134] [EC:1.2.1.12]\n",
      "line in gene PA3195  gapA; glyceraldehyde 3-phosphate dehydrogenase [KO:K00134] [EC:1.2.1.12]\n",
      "            PA3001  glyceraldehyde-3-phosphate dehydrogenase [KO:K00134] [EC:1.2.1.12]\n",
      "line in gene PA3001  glyceraldehyde-3-phosphate dehydrogenase [KO:K00134] [EC:1.2.1.12]\n",
      "            PA0552  pgk; phosphoglycerate kinase [KO:K00927] [EC:2.7.2.3]\n",
      "line in gene PA0552  pgk; phosphoglycerate kinase [KO:K00927] [EC:2.7.2.3]\n",
      "            PA5131  pgm; 2,3-bisphosphoglycerate-independent phosphoglycerate mutase [KO:K15633] [EC:5.4.2.12]\n",
      "line in gene PA5131  pgm; 2,3-bisphosphoglycerate-independent phosphoglycerate mutase [KO:K15633] [EC:5.4.2.12]\n",
      "            PA3635  eno; enolase [KO:K01689] [EC:4.2.1.11]\n",
      "line in gene PA3635  eno; enolase [KO:K01689] [EC:4.2.1.11]\n",
      "            PA1498  pykF; pyruvate kinase [KO:K00873] [EC:2.7.1.40]\n",
      "line in gene PA1498  pykF; pyruvate kinase [KO:K00873] [EC:2.7.1.40]\n",
      "            PA4329  pykA; pyruvate kinase [KO:K00873] [EC:2.7.1.40]\n",
      "line in gene PA4329  pykA; pyruvate kinase [KO:K00873] [EC:2.7.1.40]\n",
      "            PA1770  ppsA; phosphoenolpyruvate synthase [KO:K01007] [EC:2.7.9.2]\n",
      "line in gene PA1770  ppsA; phosphoenolpyruvate synthase [KO:K01007] [EC:2.7.9.2]\n",
      "            PA5015  aceE; pyruvate dehydrogenase subunit E1 [KO:K00163] [EC:1.2.4.1]\n",
      "line in gene PA5015  aceE; pyruvate dehydrogenase subunit E1 [KO:K00163] [EC:1.2.4.1]\n",
      "            PA5016  aceF; dihydrolipoamide acetyltransferase [KO:K00627] [EC:2.3.1.12]\n",
      "line in gene PA5016  aceF; dihydrolipoamide acetyltransferase [KO:K00627] [EC:2.3.1.12]\n",
      "            PA4152  branched-chain alpha-keto acid dehydrogenase subunit E2 [KO:K00627] [EC:2.3.1.12]\n",
      "line in gene PA4152  branched-chain alpha-keto acid dehydrogenase subunit E2 [KO:K00627] [EC:2.3.1.12]\n",
      "            PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "line in gene PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "            PA5427  adhA; alcohol dehydrogenase [KO:K13953] [EC:1.1.1.1]\n",
      "line in gene PA5427  adhA; alcohol dehydrogenase [KO:K13953] [EC:1.1.1.1]\n",
      "            PA2275  alcohol dehydrogenase [KO:K13979] [EC:1.1.1.2]\n",
      "line in gene PA2275  alcohol dehydrogenase [KO:K13979] [EC:1.1.1.2]\n",
      "            PA1982  exaA; quinoprotein ethanol dehydrogenase [KO:K00114] [EC:1.1.2.8]\n",
      "line in gene PA1982  exaA; quinoprotein ethanol dehydrogenase [KO:K00114] [EC:1.1.2.8]\n",
      "            PA2266  cytochrome C [KO:K22474] [EC:1.1.5.5]\n",
      "line in gene PA2266  cytochrome C [KO:K22474] [EC:1.1.5.5]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA1984  exaC; NAD+ dependent aldehyde dehydrogenase ExaC [KO:K00138] [EC:1.2.1.-]\n",
      "line in gene PA1984  exaC; NAD+ dependent aldehyde dehydrogenase ExaC [KO:K00138] [EC:1.2.1.-]\n",
      "            PA4022  aldehyde dehydrogenase [KO:K00138] [EC:1.2.1.-]\n",
      "line in gene PA4022  aldehyde dehydrogenase [KO:K00138] [EC:1.2.1.-]\n",
      "            PA0887  acsA; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA0887  acsA; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "            PA4733  acsB; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA4733  acsB; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "            PA2555  AMP-binding protein [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA2555  AMP-binding protein [KO:K01895] [EC:6.2.1.1]\n",
      "            PA1017  pauA; pimeloyl-CoA synthetase [KO:K24012] [EC:6.2.1.13]\n",
      "line in gene PA1017  pauA; pimeloyl-CoA synthetase [KO:K24012] [EC:6.2.1.13]\n",
      "            PA4978  hypothetical protein [KO:K24012] [EC:6.2.1.13]\n",
      "line in gene PA4978  hypothetical protein [KO:K24012] [EC:6.2.1.13]\n",
      "            PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "line in gene PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "            PA5422  hypothetical protein [KO:K01792] [EC:5.1.3.15]\n",
      "line in gene PA5422  hypothetical protein [KO:K01792] [EC:5.1.3.15]\n",
      "            PA2323  glyceraldehyde-3-phosphate dehydrogenase [KO:K00131] [EC:1.2.1.9]\n",
      "line in gene PA2323  glyceraldehyde-3-phosphate dehydrogenase [KO:K00131] [EC:1.2.1.9]\n",
      "            PA5192  pckA; phosphoenolpyruvate carboxykinase [KO:K01610] [EC:4.1.1.49]\n",
      "line in gene PA5192  pckA; phosphoenolpyruvate carboxykinase [KO:K01610] [EC:4.1.1.49]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00031  D-Glucose\n",
      "            C00033  Acetate\n",
      "            C00036  Oxaloacetate\n",
      "            C00068  Thiamin diphosphate\n",
      "            C00074  Phosphoenolpyruvate\n",
      "            C00084  Acetaldehyde\n",
      "            C00103  D-Glucose 1-phosphate\n",
      "            C00111  Glycerone phosphate\n",
      "            C00118  D-Glyceraldehyde 3-phosphate\n",
      "            C00186  (S)-Lactate\n",
      "            C00197  3-Phospho-D-glycerate\n",
      "            C00221  beta-D-Glucose\n",
      "            C00236  3-Phospho-D-glyceroyl phosphate\n",
      "            C00267  alpha-D-Glucose\n",
      "            C00469  Ethanol\n",
      "            C00631  2-Phospho-D-glycerate\n",
      "            C00668  alpha-D-Glucose 6-phosphate\n",
      "            C01159  2,3-Bisphospho-D-glycerate\n",
      "            C01172  beta-D-Glucose 6-phosphate\n",
      "            C01451  Salicin\n",
      "            C05125  2-(alpha-Hydroxyethyl)thiamine diphosphate\n",
      "            C05345  beta-D-Fructose 6-phosphate\n",
      "            C05378  beta-D-Fructose 1,6-bisphosphate\n",
      "            C06186  Arbutin\n",
      "            C06187  Arbutin 6-phosphate\n",
      "            C06188  Salicin 6-phosphate\n",
      "            C15972  Enzyme N6-(lipoyl)lysine\n",
      "            C15973  Enzyme N6-(dihydrolipoyl)lysine\n",
      "            C16255  [Dihydrolipoyllysine-residue acetyltransferase] S-acetyldihydrolipoyllysine\n",
      "REFERENCE   \n",
      "  AUTHORS   Nishizuka Y (ed).\n",
      "  TITLE     [Metabolic Maps] (In Japanese)\n",
      "  JOURNAL   Tokyo Kagaku Dojin (1980)\n",
      "REFERENCE   \n",
      "  AUTHORS   Nishizuka Y, Seyama Y, Ikai A, Ishimura Y, Kawaguchi A (eds).\n",
      "  TITLE     [Cellular Functions and Metabolic Maps] (In Japanese)\n",
      "  JOURNAL   Tokyo Kagaku Dojin (1997)\n",
      "REFERENCE   \n",
      "  AUTHORS   Michal G.\n",
      "  TITLE     Biochemical Pathways\n",
      "  JOURNAL   Wiley (1999)\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00030  Pentose phosphate pathway\n",
      "            pae00500  Starch and sucrose metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00640  Propanoate metabolism\n",
      "KO_PATHWAY  ko00010\n",
      "///\n",
      "path:pae00020\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00020                    Pathway\n",
      "NAME        Citrate cycle (TCA cycle) - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION The citrate cycle (TCA cycle, Krebs cycle) is an important aerobic pathway for the final steps of the oxidation of carbohydrates and fatty acids. The cycle starts with acetyl-CoA, the activated form of acetate, derived from glycolysis and pyruvate oxidation for carbohydrates and from beta oxidation of fatty acids. The two-carbon acetyl group in acetyl-CoA is transferred to the four-carbon compound of oxaloacetate to form the six-carbon compound of citrate. In a series of reactions two carbons in citrate are oxidized to CO2 and the reaction pathway supplies NADH for use in the oxidative phosphorylation and other metabolic processes. The pathway also supplies important precursor metabolites including 2-oxoglutarate. At the end of the cycle the remaining four-carbon part is transformed back to oxaloacetate. According to the genome sequence data, many organisms seem to lack genes for the full cycle [MD:M00009], but contain genes for specific segments [MD:M00010 M00011].\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00020  Citrate cycle (TCA cycle)\n",
      "MODULE      pae_M00003  Gluconeogenesis, oxaloacetate => fructose-6P [PATH:pae00020]\n",
      "            pae_M00009  Citrate cycle (TCA cycle, Krebs cycle) [PATH:pae00020]\n",
      "            pae_M00010  Citrate cycle, first carbon oxidation, oxaloacetate => 2-oxoglutarate [PATH:pae00020]\n",
      "            pae_M00011  Citrate cycle, second carbon oxidation, 2-oxoglutarate => oxaloacetate [PATH:pae00020]\n",
      "            pae_M00307  Pyruvate oxidation, pyruvate => acetyl-CoA [PATH:pae00020]\n",
      "DBLINKS     GO: 0006099\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1580  gltA; citrate synthase [KO:K01647] [EC:2.3.3.1]\n",
      "line in gene PA1580  gltA; citrate synthase [KO:K01647] [EC:2.3.3.1]\n",
      "            PA1562  acnA; aconitate hydratase [KO:K01681] [EC:4.2.1.3]\n",
      "line in gene PA1562  acnA; aconitate hydratase [KO:K01681] [EC:4.2.1.3]\n",
      "            PA1787  acnB; aconitate hydratase B [KO:K01682] [EC:4.2.1.3 4.2.1.99]\n",
      "line in gene PA1787  acnB; aconitate hydratase B [KO:K01682] [EC:4.2.1.3 4.2.1.99]\n",
      "            PA2623  icd; isocitrate dehydrogenase [KO:K00031] [EC:1.1.1.42]\n",
      "line in gene PA2623  icd; isocitrate dehydrogenase [KO:K00031] [EC:1.1.1.42]\n",
      "            PA2624  idh; isocitrate dehydrogenase [KO:K00031] [EC:1.1.1.42]\n",
      "line in gene PA2624  idh; isocitrate dehydrogenase [KO:K00031] [EC:1.1.1.42]\n",
      "            PA1585  sucA; 2-oxoglutarate dehydrogenase subunit E1 [KO:K00164] [EC:1.2.4.2]\n",
      "line in gene PA1585  sucA; 2-oxoglutarate dehydrogenase subunit E1 [KO:K00164] [EC:1.2.4.2]\n",
      "            PA1586  sucB; 2-oxoglutarate dehydrogenase complex dihydrolipoyllysine-residue succinyltransferase [KO:K00658] [EC:2.3.1.61]\n",
      "line in gene PA1586  sucB; 2-oxoglutarate dehydrogenase complex dihydrolipoyllysine-residue succinyltransferase [KO:K00658] [EC:2.3.1.61]\n",
      "            PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA1589  sucD; succinyl-CoA ligase subunit alpha [KO:K01902] [EC:6.2.1.5]\n",
      "line in gene PA1589  sucD; succinyl-CoA ligase subunit alpha [KO:K01902] [EC:6.2.1.5]\n",
      "            PA1588  sucC; succinyl-CoA ligase subunit beta [KO:K01903] [EC:6.2.1.5]\n",
      "line in gene PA1588  sucC; succinyl-CoA ligase subunit beta [KO:K01903] [EC:6.2.1.5]\n",
      "            PA5445  coenzyme A transferase [KO:K18118] [EC:2.8.3.18]\n",
      "line in gene PA5445  coenzyme A transferase [KO:K18118] [EC:2.8.3.18]\n",
      "            PA1583  sdhA; succinate dehydrogenase flavoprotein subunit [KO:K00239] [EC:1.3.5.1 1.3.5.4]\n",
      "line in gene PA1583  sdhA; succinate dehydrogenase flavoprotein subunit [KO:K00239] [EC:1.3.5.1 1.3.5.4]\n",
      "            PA1584  sdhB; succinate dehydrogenase iron-sulfur subunit [KO:K00240] [EC:1.3.5.1 1.3.5.4]\n",
      "line in gene PA1584  sdhB; succinate dehydrogenase iron-sulfur subunit [KO:K00240] [EC:1.3.5.1 1.3.5.4]\n",
      "            PA1581  sdhC; succinate dehydrogenase subunit C [KO:K00241]\n",
      "line in gene PA1581  sdhC; succinate dehydrogenase subunit C [KO:K00241]\n",
      "            PA1582  sdhD; succinate dehydrogenase subunit D [KO:K00242]\n",
      "line in gene PA1582  sdhD; succinate dehydrogenase subunit D [KO:K00242]\n",
      "            PA4333  fumarase [KO:K01676] [EC:4.2.1.2]\n",
      "line in gene PA4333  fumarase [KO:K01676] [EC:4.2.1.2]\n",
      "            PA0854  fumC2; fumarate hydratase [KO:K01679] [EC:4.2.1.2]\n",
      "line in gene PA0854  fumC2; fumarate hydratase [KO:K01679] [EC:4.2.1.2]\n",
      "            PA4470  fumC1; fumarate hydratase [KO:K01679] [EC:4.2.1.2]\n",
      "line in gene PA4470  fumC1; fumarate hydratase [KO:K01679] [EC:4.2.1.2]\n",
      "            PA3452  mqoA; malate:quinone oxidoreductase [KO:K00116] [EC:1.1.5.4]\n",
      "line in gene PA3452  mqoA; malate:quinone oxidoreductase [KO:K00116] [EC:1.1.5.4]\n",
      "            PA4640  mqoB; malate:quinone oxidoreductase [KO:K00116] [EC:1.1.5.4]\n",
      "line in gene PA4640  mqoB; malate:quinone oxidoreductase [KO:K00116] [EC:1.1.5.4]\n",
      "            PA5436  acetyl-CoA carboxylase subunit alpha [KO:K01959] [EC:6.4.1.1]\n",
      "line in gene PA5436  acetyl-CoA carboxylase subunit alpha [KO:K01959] [EC:6.4.1.1]\n",
      "            PA5435  pyruvate carboxylase subunit B [KO:K01960] [EC:6.4.1.1]\n",
      "line in gene PA5435  pyruvate carboxylase subunit B [KO:K01960] [EC:6.4.1.1]\n",
      "            PA5192  pckA; phosphoenolpyruvate carboxykinase [KO:K01610] [EC:4.1.1.49]\n",
      "line in gene PA5192  pckA; phosphoenolpyruvate carboxykinase [KO:K01610] [EC:4.1.1.49]\n",
      "            PA5015  aceE; pyruvate dehydrogenase subunit E1 [KO:K00163] [EC:1.2.4.1]\n",
      "line in gene PA5015  aceE; pyruvate dehydrogenase subunit E1 [KO:K00163] [EC:1.2.4.1]\n",
      "            PA5016  aceF; dihydrolipoamide acetyltransferase [KO:K00627] [EC:2.3.1.12]\n",
      "line in gene PA5016  aceF; dihydrolipoamide acetyltransferase [KO:K00627] [EC:2.3.1.12]\n",
      "            PA4152  branched-chain alpha-keto acid dehydrogenase subunit E2 [KO:K00627] [EC:2.3.1.12]\n",
      "line in gene PA4152  branched-chain alpha-keto acid dehydrogenase subunit E2 [KO:K00627] [EC:2.3.1.12]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00036  Oxaloacetate\n",
      "            C00042  Succinate\n",
      "            C00068  Thiamin diphosphate\n",
      "            C00074  Phosphoenolpyruvate\n",
      "            C00091  Succinyl-CoA\n",
      "            C00122  Fumarate\n",
      "            C00149  (S)-Malate\n",
      "            C00158  Citrate\n",
      "            C00311  Isocitrate\n",
      "            C00417  cis-Aconitate\n",
      "            C05125  2-(alpha-Hydroxyethyl)thiamine diphosphate\n",
      "            C05379  Oxalosuccinate\n",
      "            C05381  3-Carboxy-1-hydroxypropyl-ThPP\n",
      "            C15972  Enzyme N6-(lipoyl)lysine\n",
      "            C15973  Enzyme N6-(dihydrolipoyl)lysine\n",
      "            C16254  [Dihydrolipoyllysine-residue succinyltransferase] S-succinyldihydrolipoyllysine\n",
      "            C16255  [Dihydrolipoyllysine-residue acetyltransferase] S-acetyldihydrolipoyllysine\n",
      "REFERENCE   \n",
      "  AUTHORS   Nishizuka Y (ed).\n",
      "  TITLE     [Metabolic Maps] (In Japanese)\n",
      "  JOURNAL   Tokyo Kagaku Dojin (1980)\n",
      "REFERENCE   \n",
      "  AUTHORS   Nishizuka Y, Seyama Y, Ikai A, Ishimura Y, Kawaguchi A (eds).\n",
      "  TITLE     [Cellular Functions and Metabolic Maps] (In Japanese)\n",
      "  JOURNAL   Tokyo Kagaku Dojin (1997)\n",
      "REFERENCE   \n",
      "  AUTHORS   Michal G.\n",
      "  TITLE     Biochemical Pathways\n",
      "  JOURNAL   Wiley (1999)\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00053  Ascorbate and aldarate metabolism\n",
      "            pae00061  Fatty acid biosynthesis\n",
      "            pae00071  Fatty acid degradation\n",
      "            pae00190  Oxidative phosphorylation\n",
      "            pae00220  Arginine biosynthesis\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00280  Valine, leucine and isoleucine degradation\n",
      "            pae00350  Tyrosine metabolism\n",
      "            pae00470  D-Amino acid metabolism\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "KO_PATHWAY  ko00020\n",
      "///\n",
      "path:pae00030\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00030                    Pathway\n",
      "NAME        Pentose phosphate pathway - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION The pentose phosphate pathway is a process of glucose turnover that produces NADPH as reducing equivalents and pentoses as essential parts of nucleotides. There are two different phases in the pathway. One is irreversible oxidative phase in which glucose-6P is converted to ribulose-5P by oxidative decarboxylation, and NADPH is generated [MD:M00006]. The other is reversible non-oxidative phase in which phosphorylated sugars are interconverted to generate xylulose-5P, ribulose-5P, and ribose-5P [MD:M00007]. Phosphoribosyl pyrophosphate (PRPP) formed from ribose-5P [MD:M00005] is an activated compound used in the biosynthesis of histidine and purine/pyrimidine nucleotides. This pathway map also shows the Entner-Doudoroff pathway where 6-P-gluconate is dehydrated and then cleaved into pyruvate and glyceraldehyde-3P [MD:M00008].\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00030  Pentose phosphate pathway\n",
      "MODULE      pae_M00005  PRPP biosynthesis, ribose 5P => PRPP [PATH:pae00030]\n",
      "            pae_M00007  Pentose phosphate pathway, non-oxidative phase, fructose 6P => ribose 5P [PATH:pae00030]\n",
      "            pae_M00008  Entner-Doudoroff pathway, glucose-6P => glyceraldehyde-3P + pyruvate [PATH:pae00030]\n",
      "DBLINKS     GO: 0006098\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4732  pgi; glucose-6-phosphate isomerase [KO:K01810] [EC:5.3.1.9]\n",
      "line in gene PA4732  pgi; glucose-6-phosphate isomerase [KO:K01810] [EC:5.3.1.9]\n",
      "            PA5439  glucose-6-phosphate 1-dehydrogenase [KO:K00036] [EC:1.1.1.49 1.1.1.363]\n",
      "line in gene PA5439  glucose-6-phosphate 1-dehydrogenase [KO:K00036] [EC:1.1.1.49 1.1.1.363]\n",
      "            PA3183  zwf; glucose-6-phosphate 1-dehydrogenase [KO:K00036] [EC:1.1.1.49 1.1.1.363]\n",
      "line in gene PA3183  zwf; glucose-6-phosphate 1-dehydrogenase [KO:K00036] [EC:1.1.1.49 1.1.1.363]\n",
      "            PA3182  pgl; 6-phosphogluconolactonase [KO:K01057] [EC:3.1.1.31]\n",
      "line in gene PA3182  pgl; 6-phosphogluconolactonase [KO:K01057] [EC:3.1.1.31]\n",
      "            PA4204  ppgL; gluconolactonase PpgL [KO:K07404] [EC:3.1.1.31]\n",
      "line in gene PA4204  ppgL; gluconolactonase PpgL [KO:K07404] [EC:3.1.1.31]\n",
      "            PA0607  rpe; ribulose-phosphate 3-epimerase [KO:K01783] [EC:5.1.3.1]\n",
      "line in gene PA0607  rpe; ribulose-phosphate 3-epimerase [KO:K01783] [EC:5.1.3.1]\n",
      "            PA0548  tktA; transketolase [KO:K00615] [EC:2.2.1.1]\n",
      "line in gene PA0548  tktA; transketolase [KO:K00615] [EC:2.2.1.1]\n",
      "            PA2796  tal; transaldolase B [KO:K00616] [EC:2.2.1.2]\n",
      "line in gene PA2796  tal; transaldolase B [KO:K00616] [EC:2.2.1.2]\n",
      "            PA0330  rpiA; ribose-5-phosphate isomerase A [KO:K01807] [EC:5.3.1.6]\n",
      "line in gene PA0330  rpiA; ribose-5-phosphate isomerase A [KO:K01807] [EC:5.3.1.6]\n",
      "            PA1950  rbsK; ribokinase [KO:K00852] [EC:2.7.1.15]\n",
      "line in gene PA1950  rbsK; ribokinase [KO:K00852] [EC:2.7.1.15]\n",
      "            PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "line in gene PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "            PA3373  hypothetical protein [KO:K05774] [EC:2.7.4.23]\n",
      "line in gene PA3373  hypothetical protein [KO:K05774] [EC:2.7.4.23]\n",
      "            PA4670  prs; ribose-phosphate pyrophosphokinase [KO:K00948] [EC:2.7.6.1]\n",
      "line in gene PA4670  prs; ribose-phosphate pyrophosphokinase [KO:K00948] [EC:2.7.6.1]\n",
      "            PA3194  edd; phosphogluconate dehydratase [KO:K01690] [EC:4.2.1.12]\n",
      "line in gene PA3194  edd; phosphogluconate dehydratase [KO:K01690] [EC:4.2.1.12]\n",
      "            PA3181  2-dehydro-3-deoxy-phosphogluconate aldolase [KO:K01625] [EC:4.1.2.14 4.1.3.42]\n",
      "line in gene PA3181  2-dehydro-3-deoxy-phosphogluconate aldolase [KO:K01625] [EC:4.1.2.14 4.1.3.42]\n",
      "            PA3131  aldolase [KO:K01625] [EC:4.1.2.14 4.1.3.42]\n",
      "line in gene PA3131  aldolase [KO:K01625] [EC:4.1.2.14 4.1.3.42]\n",
      "            PA2323  glyceraldehyde-3-phosphate dehydrogenase [KO:K00131] [EC:1.2.1.9]\n",
      "line in gene PA2323  glyceraldehyde-3-phosphate dehydrogenase [KO:K00131] [EC:1.2.1.9]\n",
      "            PA2290  gcd; glucose dehydrogenase [KO:K00117] [EC:1.1.5.2]\n",
      "line in gene PA2290  gcd; glucose dehydrogenase [KO:K00117] [EC:1.1.5.2]\n",
      "            PA2265  gluconate dehydrogenase [KO:K06151] [EC:1.1.99.3]\n",
      "line in gene PA2265  gluconate dehydrogenase [KO:K06151] [EC:1.1.99.3]\n",
      "            PA2264  hypothetical protein [KO:K06152] [EC:1.1.99.3]\n",
      "line in gene PA2264  hypothetical protein [KO:K06152] [EC:1.1.99.3]\n",
      "            PA3896  2-hydroxyacid dehydrogenase [KO:K00090] [EC:1.1.1.79 1.1.1.81 1.1.1.215]\n",
      "line in gene PA3896  2-hydroxyacid dehydrogenase [KO:K00090] [EC:1.1.1.79 1.1.1.81 1.1.1.215]\n",
      "            PA2261  2-ketogluconate kinase [KO:K11441] [EC:2.7.1.13]\n",
      "line in gene PA2261  2-ketogluconate kinase [KO:K11441] [EC:2.7.1.13]\n",
      "            PA2263  2-hydroxyacid dehydrogenase [KO:K00032] [EC:1.1.1.43]\n",
      "line in gene PA2263  2-hydroxyacid dehydrogenase [KO:K00032] [EC:1.1.1.43]\n",
      "            PA2321  gluconokinase [KO:K00851] [EC:2.7.1.12]\n",
      "line in gene PA2321  gluconokinase [KO:K00851] [EC:2.7.1.12]\n",
      "            PA1499  hypothetical protein [KO:K11529] [EC:2.7.1.165]\n",
      "line in gene PA1499  hypothetical protein [KO:K11529] [EC:2.7.1.165]\n",
      "            PA0849  trxB2; thioredoxin reductase [KO:K22345] [EC:4.3.1.9]\n",
      "line in gene PA0849  trxB2; thioredoxin reductase [KO:K22345] [EC:4.3.1.9]\n",
      "            PA0555  fda; fructose-1,6-bisphosphate aldolase [KO:K01624] [EC:4.1.2.13]\n",
      "line in gene PA0555  fda; fructose-1,6-bisphosphate aldolase [KO:K01624] [EC:4.1.2.13]\n",
      "            PA5110  fbp; fructose-1,6-bisphosphatase [KO:K03841] [EC:3.1.3.11]\n",
      "line in gene PA5110  fbp; fructose-1,6-bisphosphatase [KO:K03841] [EC:3.1.3.11]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00031  D-Glucose\n",
      "            C00117  D-Ribose 5-phosphate\n",
      "            C00118  D-Glyceraldehyde 3-phosphate\n",
      "            C00119  5-Phospho-alpha-D-ribose 1-diphosphate\n",
      "            C00121  D-Ribose\n",
      "            C00197  3-Phospho-D-glycerate\n",
      "            C00198  D-Glucono-1,5-lactone\n",
      "            C00199  D-Ribulose 5-phosphate\n",
      "            C00204  2-Dehydro-3-deoxy-D-gluconate\n",
      "            C00221  beta-D-Glucose\n",
      "            C00231  D-Xylulose 5-phosphate\n",
      "            C00257  D-Gluconic acid\n",
      "            C00258  D-Glycerate\n",
      "            C00279  D-Erythrose 4-phosphate\n",
      "            C00345  6-Phospho-D-gluconate\n",
      "            C00577  D-Glyceraldehyde\n",
      "            C00620  alpha-D-Ribose 1-phosphate\n",
      "            C00631  2-Phospho-D-glycerate\n",
      "            C00668  alpha-D-Glucose 6-phosphate\n",
      "            C00672  2-Deoxy-D-ribose 1-phosphate\n",
      "            C00673  2-Deoxy-D-ribose 5-phosphate\n",
      "            C01151  D-Ribose 1,5-bisphosphate\n",
      "            C01172  beta-D-Glucose 6-phosphate\n",
      "            C01218  6-Phospho-2-dehydro-D-gluconate\n",
      "            C01236  D-Glucono-1,5-lactone 6-phosphate\n",
      "            C01801  Deoxyribose\n",
      "            C03752  2-Amino-2-deoxy-D-gluconate\n",
      "            C04442  2-Dehydro-3-deoxy-6-phospho-D-gluconate\n",
      "            C05345  beta-D-Fructose 6-phosphate\n",
      "            C05378  beta-D-Fructose 1,6-bisphosphate\n",
      "            C05382  Sedoheptulose 7-phosphate\n",
      "            C06019  D-arabino-Hex-3-ulose 6-phosphate\n",
      "            C06473  2-Keto-D-gluconic acid\n",
      "            C20589  D-Glucosaminate-6-phosphate\n",
      "REFERENCE   \n",
      "  AUTHORS   Nishizuka Y (ed).\n",
      "  TITLE     [Metabolic Maps] (In Japanese)\n",
      "  JOURNAL   Tokyo Kagaku Dojin (1980)\n",
      "REFERENCE   \n",
      "  AUTHORS   Nishizuka Y, Seyama Y, Ikai A, Ishimura Y, Kawaguchi A (eds).\n",
      "  TITLE     [Cellular Functions and Metabolic Maps] (In Japanese)\n",
      "  JOURNAL   Tokyo Kagaku Dojin (1997)\n",
      "REFERENCE   \n",
      "  AUTHORS   Michal G.\n",
      "  TITLE     Biochemical Pathways\n",
      "  JOURNAL   Wiley (1999)\n",
      "REFERENCE   PMID:12700258\n",
      "  AUTHORS   Hove-Jensen B, Rosenkrantz TJ, Haldimann A, Wanner BL.\n",
      "  TITLE     Escherichia coli phnN, encoding ribose 1,5-bisphosphokinase activity (phosphoribosyl diphosphate forming): dual role in phosphonate degradation and NAD biosynthesis pathways.\n",
      "  JOURNAL   J Bacteriol 185:2793-801 (2003)\n",
      "            DOI:10.1128/JB.185.9.2793-2801.2003\n",
      "REFERENCE   PMID:16788179\n",
      "  AUTHORS   Orita I, Sato T, Yurimoto H, Kato N, Atomi H, Imanaka T, Sakai Y\n",
      "  TITLE     The ribulose monophosphate pathway substitutes for the missing pentose phosphate pathway in the archaeon Thermococcus kodakaraensis.\n",
      "  JOURNAL   J Bacteriol 188:4698-704 (2006)\n",
      "            DOI:10.1128/JB.00492-06\n",
      "REFERENCE   PMID:16428816\n",
      "  AUTHORS   Kato N, Yurimoto H, Thauer RK\n",
      "  TITLE     The physiological role of the ribulose monophosphate pathway in bacteria and archaea.\n",
      "  JOURNAL   Biosci Biotechnol Biochem 70:10-21 (2006)\n",
      "            DOI:10.1271/bbb.70.10\n",
      "REFERENCE   PMID:23279921\n",
      "  AUTHORS   Kouril T, Wieloch P, Reimann J, Wagner M, Zaparty M, Albers SV, Schomburg D, Ruoff P, Siebers B\n",
      "  TITLE     Unraveling the function of the two Entner-Doudoroff branches in the thermoacidophilic Crenarchaeon Sulfolobus solfataricus P2.\n",
      "  JOURNAL   FEBS J 280:1126-38 (2013)\n",
      "            DOI:10.1111/febs.12106\n",
      "REFERENCE   PMID:16458304\n",
      "  AUTHORS   Reher M, Schonheit P\n",
      "  TITLE     Glyceraldehyde dehydrogenases from the thermoacidophilic euryarchaeota Picrophilus torridus and Thermoplasma acidophilum, key enzymes of the non-phosphorylative Entner-Doudoroff pathway, constitute a novel enzyme family within the aldehyde dehydrogenase superfamily.\n",
      "  JOURNAL   FEBS Lett 580:1198-204 (2006)\n",
      "            DOI:10.1016/j.febslet.2006.01.029\n",
      "REFERENCE   PMID:20023024\n",
      "  AUTHORS   Reher M, Fuhrer T, Bott M, Schonheit P\n",
      "  TITLE     The nonphosphorylative Entner-Doudoroff pathway in the thermoacidophilic euryarchaeon Picrophilus torridus involves a novel 2-keto-3-deoxygluconate- specific aldolase.\n",
      "  JOURNAL   J Bacteriol 192:964-74 (2010)\n",
      "            DOI:10.1128/JB.01281-09\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00040  Pentose and glucuronate interconversions\n",
      "            pae00052  Galactose metabolism\n",
      "            pae00230  Purine metabolism\n",
      "            pae00240  Pyrimidine metabolism\n",
      "            pae00340  Histidine metabolism\n",
      "            pae00750  Vitamin B6 metabolism\n",
      "KO_PATHWAY  ko00030\n",
      "///\n",
      "path:pae00040\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00040                    Pathway\n",
      "NAME        Pentose and glucuronate interconversions - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00040  Pentose and glucuronate interconversions\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2022  UDP-glucose 6-dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "line in gene PA2022  UDP-glucose 6-dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "            PA3559  nucleotide sugar dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "line in gene PA3559  nucleotide sugar dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "            PA2023  galU; UTP-glucose-1-phosphate uridylyltransferase [KO:K00963] [EC:2.7.7.9]\n",
      "line in gene PA2023  galU; UTP-glucose-1-phosphate uridylyltransferase [KO:K00963] [EC:2.7.7.9]\n",
      "            PA0607  rpe; ribulose-phosphate 3-epimerase [KO:K01783] [EC:5.1.3.1]\n",
      "line in gene PA0607  rpe; ribulose-phosphate 3-epimerase [KO:K01783] [EC:5.1.3.1]\n",
      "            PA2343  mtlY; xylulose kinase [KO:K00854] [EC:2.7.1.17]\n",
      "line in gene PA2343  mtlY; xylulose kinase [KO:K00854] [EC:2.7.1.17]\n",
      "            PA1253  semialdehyde dehydrogenase [KO:K13877] [EC:1.2.1.26]\n",
      "line in gene PA1253  semialdehyde dehydrogenase [KO:K13877] [EC:1.2.1.26]\n",
      "            PA2217  aldehyde dehydrogenase [KO:K13877] [EC:1.2.1.26]\n",
      "line in gene PA2217  aldehyde dehydrogenase [KO:K13877] [EC:1.2.1.26]\n",
      "            PA0224  class II aldolase/adducin domain-containing protein [KO:K01628] [EC:4.1.2.17]\n",
      "line in gene PA0224  class II aldolase/adducin domain-containing protein [KO:K01628] [EC:4.1.2.17]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00029  UDP-glucose\n",
      "            C00103  D-Glucose 1-phosphate\n",
      "            C00111  Glycerone phosphate\n",
      "            C00116  Glycerol\n",
      "            C00167  UDP-glucuronate\n",
      "            C00181  D-Xylose\n",
      "            C00191  D-Glucuronate\n",
      "            C00199  D-Ribulose 5-phosphate\n",
      "            C00204  2-Dehydro-3-deoxy-D-gluconate\n",
      "            C00216  D-Arabinose\n",
      "            C00231  D-Xylulose 5-phosphate\n",
      "            C00259  L-Arabinose\n",
      "            C00266  Glycolaldehyde\n",
      "            C00309  D-Ribulose\n",
      "            C00310  D-Xylulose\n",
      "            C00312  L-Xylulose\n",
      "            C00333  D-Galacturonate\n",
      "            C00379  Xylitol\n",
      "            C00433  2,5-Dioxopentanoate\n",
      "            C00470  Pectate\n",
      "            C00474  Ribitol\n",
      "            C00476  D-Lyxose\n",
      "            C00502  D-Xylonate\n",
      "            C00508  L-Ribulose\n",
      "            C00514  D-Mannonate\n",
      "            C00532  L-Arabitol\n",
      "            C00558  D-Tagaturonate\n",
      "            C00618  3-Dehydro-L-gulonate\n",
      "            C00714  Pectin\n",
      "            C00789  CDP-ribitol\n",
      "            C00800  L-Gulonate\n",
      "            C00817  D-Altronate\n",
      "            C00905  D-Fructuronate\n",
      "            C01068  D-Ribitol 5-phosphate\n",
      "            C01101  L-Ribulose 5-phosphate\n",
      "            C01508  L-Lyxose\n",
      "            C01904  D-Arabitol\n",
      "            C02266  D-Xylonolactone\n",
      "            C02273  Digalacturonate\n",
      "            C02426  L-Glyceraldehyde\n",
      "            C02753  D-Xylono-1,4-lactone\n",
      "            C03033  beta-D-Glucuronoside\n",
      "            C03291  L-Xylulose 5-phosphate\n",
      "            C03826  2-Dehydro-3-deoxy-D-xylonate\n",
      "            C04053  5-Dehydro-4-deoxy-D-glucuronate\n",
      "            C04349  (4S)-4,6-Dihydroxy-2,5-dioxohexanoate\n",
      "            C04575  (4R,5S)-4,5,6-Trihydroxy-2,3-dioxohexanoate\n",
      "            C05385  D-Glucuronate 1-phosphate\n",
      "            C05411  L-Xylonate\n",
      "            C05412  L-Lyxonate\n",
      "            C06118  4-(4-Deoxy-alpha-D-gluc-4-enuronosyl)-D-galacturonate\n",
      "            C06441  L-Xylulose 1-phosphate\n",
      "            C14899  3-Dehydro-L-gulonate 6-phosphate\n",
      "            C15930  L-Galactonate\n",
      "            C20680  2-Dehydro-3-deoxy-L-galactonate\n",
      "            C22337  D-Ribulose 1-phosphate\n",
      "REFERENCE   PMID:11741871\n",
      "  AUTHORS   Yew WS, Gerlt JA.\n",
      "  TITLE     Utilization of L-ascorbate by Escherichia coli K-12: assignments of functions to products of the yjf-sga and yia-sgb operons.\n",
      "  JOURNAL   J Bacteriol 184:302-6 (2002)\n",
      "            DOI:10.1128/JB.184.1.302-306.2002\n",
      "REFERENCE   PMID:15697206\n",
      "  AUTHORS   Yew WS, Akana J, Wise EL, Rayment I, Gerlt JA\n",
      "  TITLE     Evolution of enzymatic activities in the orotidine 5'-monophosphate decarboxylase suprafamily: enhancing the promiscuous D-arabino-hex-3-ulose 6-phosphate synthase reaction catalyzed by 3-keto-L-gulonate 6-phosphate decarboxylase.\n",
      "  JOURNAL   Biochemistry 44:1807-15 (2005)\n",
      "            DOI:10.1021/bi047815v\n",
      "REFERENCE   PMID:10572115\n",
      "  AUTHORS   Yasueda H, Kawahara Y, Sugimoto S.\n",
      "  TITLE     Bacillus subtilis yckG and yckF encode two key enzymes of the ribulose monophosphate pathway used by methylotrophs, and yckH is required for their expression.\n",
      "  JOURNAL   J Bacteriol 181:7154-60 (1999)\n",
      "            DOI:10.1128/JB.181.23.7154-7160.1999\n",
      "REFERENCE   PMID:10913097\n",
      "  AUTHORS   Ibanez E, Gimenez R, Pedraza T, Baldoma L, Aguilar J, Badia J\n",
      "  TITLE     Role of the yiaR and yiaS genes of Escherichia coli in metabolism of endogenously formed L-xylulose.\n",
      "  JOURNAL   J Bacteriol 182:4625-7 (2000)\n",
      "            DOI:10.1128/JB.182.16.4625-4627.2000\n",
      "REFERENCE   PMID:15901685\n",
      "  AUTHORS   Orita I, Yurimoto H, Hirai R, Kawarabayasi Y, Sakai Y, Kato N\n",
      "  TITLE     The archaeon Pyrococcus horikoshii possesses a bifunctional enzyme for formaldehyde fixation via the ribulose monophosphate pathway.\n",
      "  JOURNAL   J Bacteriol 187:3636-42 (2005)\n",
      "            DOI:10.1128/JB.187.11.3636-3642.2005\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00030  Pentose phosphate pathway\n",
      "            pae00052  Galactose metabolism\n",
      "            pae00053  Ascorbate and aldarate metabolism\n",
      "            pae00500  Starch and sucrose metabolism\n",
      "            pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "            pae00561  Glycerolipid metabolism\n",
      "            pae00562  Inositol phosphate metabolism\n",
      "            pae00740  Riboflavin metabolism\n",
      "KO_PATHWAY  ko00040\n",
      "///\n",
      "path:pae00051\n",
      "ENTRY       pae00051                    Pathway\n",
      "NAME        Fructose and mannose metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00051  Fructose and mannose metabolism\n",
      "DBLINKS     GO: 0006000 0006013\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2344  mtlZ; fructokinase [KO:K00847] [EC:2.7.1.4]\n",
      "line in gene PA2344  mtlZ; fructokinase [KO:K00847] [EC:2.7.1.4]\n",
      "            PA3551  algA; bifunctional mannose-1-phosphate guanylyltransferase/mannose-6-phosphate isomerase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "line in gene PA3551  algA; bifunctional mannose-1-phosphate guanylyltransferase/mannose-6-phosphate isomerase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "            PA2232  pslB; biofilm formation protein PslB [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "line in gene PA2232  pslB; biofilm formation protein PslB [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "            PA5452  wbpW; phosphomannose isomerase/mannose-1-phosphate guanylyl transferase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "line in gene PA5452  wbpW; phosphomannose isomerase/mannose-1-phosphate guanylyl transferase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "            PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "line in gene PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "            PA5453  gmd; GDP-mannose 4,6-dehydratase [KO:K01711] [EC:4.2.1.47]\n",
      "line in gene PA5453  gmd; GDP-mannose 4,6-dehydratase [KO:K01711] [EC:4.2.1.47]\n",
      "            PA0224  class II aldolase/adducin domain-containing protein [KO:K01628] [EC:4.1.2.17]\n",
      "line in gene PA0224  class II aldolase/adducin domain-containing protein [KO:K01628] [EC:4.1.2.17]\n",
      "            PA5110  fbp; fructose-1,6-bisphosphatase [KO:K03841] [EC:3.1.3.11]\n",
      "line in gene PA5110  fbp; fructose-1,6-bisphosphatase [KO:K03841] [EC:3.1.3.11]\n",
      "            PA3561  fruK; 1-phosphofructokinase [KO:K00882] [EC:2.7.1.56]\n",
      "line in gene PA3561  fruK; 1-phosphofructokinase [KO:K00882] [EC:2.7.1.56]\n",
      "            PA2342  mtlD; mannitol dehydrogenase [KO:K00045] [EC:1.1.1.67]\n",
      "line in gene PA2342  mtlD; mannitol dehydrogenase [KO:K00045] [EC:1.1.1.67]\n",
      "            PA3540  algD; GDP-mannose 6-dehydrogenase AlgD [KO:K00066] [EC:1.1.1.132]\n",
      "line in gene PA3540  algD; GDP-mannose 6-dehydrogenase AlgD [KO:K00066] [EC:1.1.1.132]\n",
      "            PA3541  alg8; glycosyltransferase alg8 [KO:K19290] [EC:2.4.1.33]\n",
      "line in gene PA3541  alg8; glycosyltransferase alg8 [KO:K19290] [EC:2.4.1.33]\n",
      "            PA3542  alg44; alginate biosynthesis protein Alg44 [KO:K19291] [EC:2.4.1.33]\n",
      "line in gene PA3542  alg44; alginate biosynthesis protein Alg44 [KO:K19291] [EC:2.4.1.33]\n",
      "            PA3545  algG; alginate-c5-mannuronan-epimerase AlgG [KO:K01795] [EC:5.1.3.37]\n",
      "line in gene PA3545  algG; alginate-c5-mannuronan-epimerase AlgG [KO:K01795] [EC:5.1.3.37]\n",
      "            PA3547  algL; alginate lyase [KO:K01729] [EC:4.2.2.3]\n",
      "line in gene PA3547  algL; alginate lyase [KO:K01729] [EC:4.2.2.3]\n",
      "            PA0555  fda; fructose-1,6-bisphosphate aldolase [KO:K01624] [EC:4.1.2.13]\n",
      "line in gene PA0555  fda; fructose-1,6-bisphosphate aldolase [KO:K01624] [EC:4.1.2.13]\n",
      "            PA4748  tpiA; triosephosphate isomerase [KO:K01803] [EC:5.3.1.1]\n",
      "line in gene PA4748  tpiA; triosephosphate isomerase [KO:K01803] [EC:5.3.1.1]\n",
      "            PA5454  rmd; oxidoreductase Rmd [KO:K15856] [EC:1.1.1.281]\n",
      "line in gene PA5454  rmd; oxidoreductase Rmd [KO:K15856] [EC:1.1.1.281]\n",
      "            PA3560  fruA; PTS system fructose-specific transporter subunit IIBC [KO:K02770] [EC:2.7.1.202]\n",
      "line in gene PA3560  fruA; PTS system fructose-specific transporter subunit IIBC [KO:K02770] [EC:2.7.1.202]\n",
      "COMPOUND    C00095  D-Fructose\n",
      "            C00096  GDP-mannose\n",
      "            C00111  Glycerone phosphate\n",
      "            C00118  D-Glyceraldehyde 3-phosphate\n",
      "            C00159  D-Mannose\n",
      "            C00186  (S)-Lactate\n",
      "            C00247  L-Sorbose\n",
      "            C00267  alpha-D-Glucose\n",
      "            C00275  D-Mannose 6-phosphate\n",
      "            C00325  GDP-L-fucose\n",
      "            C00392  Mannitol\n",
      "            C00424  (S)-Lactaldehyde\n",
      "            C00464  Mannan\n",
      "            C00507  L-Rhamnose\n",
      "            C00577  D-Glyceraldehyde\n",
      "            C00636  D-Mannose 1-phosphate\n",
      "            C00644  D-Mannitol 1-phosphate\n",
      "            C00665  beta-D-Fructose 2,6-bisphosphate\n",
      "            C00794  D-Sorbitol\n",
      "            C00861  L-Rhamnulose\n",
      "            C00976  GDP-D-mannuronate\n",
      "            C01019  6-Deoxy-L-galactose\n",
      "            C01094  D-Fructose 1-phosphate\n",
      "            C01096  Sorbitol 6-phosphate\n",
      "            C01099  L-Fuculose 1-phosphate\n",
      "            C01131  L-Rhamnulose 1-phosphate\n",
      "            C01222  GDP-4-dehydro-6-deoxy-D-mannose\n",
      "            C01355  Fructan\n",
      "            C01487  D-Allose\n",
      "            C01720  L-Fuconate\n",
      "            C01721  L-Fuculose\n",
      "            C01768  (Alginate)n\n",
      "            C01934  L-Rhamnonate\n",
      "            C02431  L-Rhamnofuranose\n",
      "            C02492  1,4-beta-D-Mannan\n",
      "            C02888  Sorbose 1-phosphate\n",
      "            C02962  D-Allose 6-phosphate\n",
      "            C02977  GDP-6-deoxy-D-talose\n",
      "            C02985  L-Fucose 1-phosphate\n",
      "            C02991  L-Rhamnono-1,4-lactone\n",
      "            C03117  GDP-6-deoxy-D-mannose\n",
      "            C03267  beta-D-Fructose 2-phosphate\n",
      "            C03827  2-Dehydro-3-deoxy-L-fuconate\n",
      "            C03979  2-Dehydro-3-deoxy-L-rhamnonate\n",
      "            C05144  (beta-D-Mannuronate)n\n",
      "            C05345  beta-D-Fructose 6-phosphate\n",
      "            C05378  beta-D-Fructose 1,6-bisphosphate\n",
      "            C05392  Oligouronide with 4-deoxy-alpha-L-erythro-hex-4-enopyranuronosyl group\n",
      "            C06192  ADP-mannose\n",
      "            C11516  2-(alpha-D-Mannosyl)-3-phosphoglycerate\n",
      "            C11544  2-O-(alpha-D-Mannosyl)-D-glycerate\n",
      "            C18028  L-Fucono-1,5-lactone\n",
      "            C18096  D-Allulose 6-phosphate\n",
      "            C20781  2,4-Diketo-3-deoxy-L-fuconate\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00052  Galactose metabolism\n",
      "            pae00053  Ascorbate and aldarate metabolism\n",
      "            pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "KO_PATHWAY  ko00051\n",
      "///\n",
      "path:pae00052\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00052                    Pathway\n",
      "NAME        Galactose metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00052  Galactose metabolism\n",
      "DBLINKS     GO: 0006012\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4068  epimerase [KO:K01784] [EC:5.1.3.2]\n",
      "line in gene PA4068  epimerase [KO:K01784] [EC:5.1.3.2]\n",
      "            PA1384  galE; UDP-glucose 4-epimerase [KO:K01784] [EC:5.1.3.2]\n",
      "line in gene PA1384  galE; UDP-glucose 4-epimerase [KO:K01784] [EC:5.1.3.2]\n",
      "            PA2023  galU; UTP-glucose-1-phosphate uridylyltransferase [KO:K00963] [EC:2.7.7.9]\n",
      "line in gene PA2023  galU; UTP-glucose-1-phosphate uridylyltransferase [KO:K00963] [EC:2.7.7.9]\n",
      "            PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "line in gene PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "            PA3193  glk; glucokinase [KO:K00845] [EC:2.7.1.2]\n",
      "line in gene PA3193  glk; glucokinase [KO:K00845] [EC:2.7.1.2]\n",
      "COMPOUND    C00029  UDP-glucose\n",
      "            C00031  D-Glucose\n",
      "            C00052  UDP-alpha-D-galactose\n",
      "            C00085  D-Fructose 6-phosphate\n",
      "            C00089  Sucrose\n",
      "            C00095  D-Fructose\n",
      "            C00103  D-Glucose 1-phosphate\n",
      "            C00111  Glycerone phosphate\n",
      "            C00116  Glycerol\n",
      "            C00118  D-Glyceraldehyde 3-phosphate\n",
      "            C00124  D-Galactose\n",
      "            C00137  myo-Inositol\n",
      "            C00159  D-Mannose\n",
      "            C00243  Lactose\n",
      "            C00267  alpha-D-Glucose\n",
      "            C00446  alpha-D-Galactose 1-phosphate\n",
      "            C00492  Raffinose\n",
      "            C00577  D-Glyceraldehyde\n",
      "            C00668  alpha-D-Glucose 6-phosphate\n",
      "            C00794  D-Sorbitol\n",
      "            C00795  D-Tagatose\n",
      "            C00880  D-Galactonate\n",
      "            C00984  alpha-D-Galactose\n",
      "            C01097  D-Tagatose 6-phosphate\n",
      "            C01113  D-Galactose 6-phosphate\n",
      "            C01132  N-Acetyl-D-galactosamine\n",
      "            C01216  2-Dehydro-3-deoxy-D-galactonate\n",
      "            C01235  alpha-D-Galactosyl-(1->3)-1D-myo-inositol\n",
      "            C01286  2-Dehydro-3-deoxy-6-phospho-D-galactonate\n",
      "            C01613  Stachyose\n",
      "            C01697  Galactitol\n",
      "            C02262  D-Galactosamine\n",
      "            C02669  D-Galactono-1,5-lactone\n",
      "            C03383  D-Galactono-1,4-lactone\n",
      "            C03733  UDP-alpha-D-galactofuranose\n",
      "            C03785  D-Tagatose 1,6-bisphosphate\n",
      "            C05396  Lactose 6'-phosphate\n",
      "            C05399  Melibiitol\n",
      "            C05400  Epimelibiose\n",
      "            C05401  3-beta-D-Galactosyl-sn-glycerol\n",
      "            C05402  Melibiose\n",
      "            C05404  D-Gal alpha 1->6D-Gal alpha 1->6D-Glucose\n",
      "            C05796  Galactan\n",
      "            C06311  Galactitol 1-phosphate\n",
      "            C06376  N-Acetyl-D-galactosamine 6-phosphate\n",
      "            C06377  D-Galactosamine 6-phosphate\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00030  Pentose phosphate pathway\n",
      "            pae00040  Pentose and glucuronate interconversions\n",
      "            pae00051  Fructose and mannose metabolism\n",
      "            pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "KO_PATHWAY  ko00052\n",
      "///\n",
      "path:pae00053\n",
      "ENTRY       pae00053                    Pathway\n",
      "NAME        Ascorbate and aldarate metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00053  Ascorbate and aldarate metabolism\n",
      "DBLINKS     GO: 0019852 0019577 0019572\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2022  UDP-glucose 6-dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "line in gene PA2022  UDP-glucose 6-dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "            PA3559  nucleotide sugar dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "line in gene PA3559  nucleotide sugar dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA2215  hypothetical protein [KO:K25260] [EC:4.2.1.176]\n",
      "line in gene PA2215  hypothetical protein [KO:K25260] [EC:4.2.1.176]\n",
      "            PA1253  semialdehyde dehydrogenase [KO:K13877] [EC:1.2.1.26]\n",
      "line in gene PA1253  semialdehyde dehydrogenase [KO:K13877] [EC:1.2.1.26]\n",
      "            PA2217  aldehyde dehydrogenase [KO:K13877] [EC:1.2.1.26]\n",
      "line in gene PA2217  aldehyde dehydrogenase [KO:K13877] [EC:1.2.1.26]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00029  UDP-glucose\n",
      "            C00072  Ascorbate\n",
      "            C00096  GDP-mannose\n",
      "            C00137  myo-Inositol\n",
      "            C00167  UDP-glucuronate\n",
      "            C00191  D-Glucuronate\n",
      "            C00216  D-Arabinose\n",
      "            C00231  D-Xylulose 5-phosphate\n",
      "            C00259  L-Arabinose\n",
      "            C00333  D-Galacturonate\n",
      "            C00433  2,5-Dioxopentanoate\n",
      "            C00545  L-Arabinonate\n",
      "            C00558  D-Tagaturonate\n",
      "            C00617  UDP-D-galacturonate\n",
      "            C00618  3-Dehydro-L-gulonate\n",
      "            C00652  D-Arabinono-1,4-lactone\n",
      "            C00679  5-Dehydro-4-deoxy-D-glucarate\n",
      "            C00684  2-Dehydro-3-deoxy-L-arabinonate\n",
      "            C00800  L-Gulonate\n",
      "            C00818  D-Glucarate\n",
      "            C00878  D-Arabinonate\n",
      "            C00879  D-Galactarate\n",
      "            C01040  L-Gulono-1,4-lactone\n",
      "            C01041  Monodehydroascorbate\n",
      "            C01101  L-Ribulose 5-phosphate\n",
      "            C01114  L-Arabinono-1,4-lactone\n",
      "            C01115  L-Galactono-1,4-lactone\n",
      "            C01146  2-Hydroxy-3-oxopropanoate\n",
      "            C01620  Threonate\n",
      "            C01825  L-Galactose\n",
      "            C02280  GDP-L-galactose\n",
      "            C02670  D-Glucuronolactone\n",
      "            C03033  beta-D-Glucuronoside\n",
      "            C03064  3-Dehydro-L-threonate\n",
      "            C03289  L-xylo-Hexulonolactone\n",
      "            C03291  L-Xylulose 5-phosphate\n",
      "            C03826  2-Dehydro-3-deoxy-D-xylonate\n",
      "            C03921  2-Dehydro-3-deoxy-D-glucarate\n",
      "            C04037  1-Phospho-alpha-D-galacturonate\n",
      "            C04575  (4R,5S)-4,5,6-Trihydroxy-2,3-dioxohexanoate\n",
      "            C05385  D-Glucuronate 1-phosphate\n",
      "            C05411  L-Xylonate\n",
      "            C05412  L-Lyxonate\n",
      "            C05422  Dehydroascorbate\n",
      "            C06316  Dehydro-D-arabinono-1,4-lactone\n",
      "            C14899  3-Dehydro-L-gulonate 6-phosphate\n",
      "            C15923  L-Gulose\n",
      "            C15924  L-Gulose 1-phosphate\n",
      "            C15925  GDP-L-gulose\n",
      "            C15926  beta-L-Galactose 1-phosphate\n",
      "            C15930  L-Galactonate\n",
      "            C16186  L-Ascorbate 6-phosphate\n",
      "            C20889  D-Galactaro-1,5-lactone\n",
      "            C20896  D-Galactaro-1,4-lactone\n",
      "            C21955  L-Galactono-1,5-lactone\n",
      "REFERENCE   PMID:17222174\n",
      "  AUTHORS   Linster CL, Van Schaftingen E\n",
      "  TITLE     Vitamin C. Biosynthesis, recycling and degradation in mammals.\n",
      "  JOURNAL   FEBS J 274:1-22 (2007)\n",
      "            DOI:10.1111/j.1742-4658.2006.05607.x\n",
      "REFERENCE   PMID:15564123\n",
      "  AUTHORS   Valpuesta V, Botella MA.\n",
      "  TITLE     Biosynthesis of L-ascorbic acid in plants: new pathways for an old antioxidant.\n",
      "  JOURNAL   Trends Plant Sci 9:573-7 (2004)\n",
      "            DOI:10.1016/j.tplants.2004.10.002\n",
      "REFERENCE   PMID:10517845\n",
      "  AUTHORS   Davey MW, Gilot C, Persiau G, Ostergaard J, Han Y, Bauw GC, Van Montagu MC.\n",
      "  TITLE     Ascorbate biosynthesis in Arabidopsis cell suspension culture.\n",
      "  JOURNAL   Plant Physiol 121:535-43 (1999)\n",
      "            DOI:10.1104/pp.121.2.535\n",
      "REFERENCE   PMID:16595667\n",
      "  AUTHORS   Conklin PL, Gatzek S, Wheeler GL, Dowdle J, Raymond MJ, Rolinski S, Isupov M, Littlechild JA, Smirnoff N.\n",
      "  TITLE     Arabidopsis thaliana VTC4 encodes L-galactose-1-P phosphatase, a plant ascorbic acid biosynthetic enzyme.\n",
      "  JOURNAL   J Biol Chem 281:15662-70 (2006)\n",
      "            DOI:10.1074/jbc.M601409200\n",
      "REFERENCE   PMID:17462988\n",
      "  AUTHORS   Linster CL, Gomez TA, Christensen KC, Adler LN, Young BD, Brenner C, Clarke SG.\n",
      "  TITLE     Arabidopsis VTC2 Encodes a GDP-L-Galactose Phosphorylase, the Last Unknown Enzyme in the Smirnoff-Wheeler Pathway to Ascorbic Acid in Plants.\n",
      "  JOURNAL   J Biol Chem 282:18879-85 (2007)\n",
      "            DOI:10.1074/jbc.M702094200\n",
      "REFERENCE   PMID:14996803\n",
      "  AUTHORS   Campos E, Baldoma L, Aguilar J, Badia J.\n",
      "  TITLE     Regulation of expression of the divergent ulaG and ulaABCDEF operons involved in LaAscorbate dissimilation in Escherichia coli.\n",
      "  JOURNAL   J Bacteriol 186:1720-8 (2004)\n",
      "            DOI:10.1128/JB.186.6.1720-1728.2004\n",
      "REFERENCE   PMID:12644495\n",
      "  AUTHORS   Zhang Z, Aboulwafa M, Smith MH, Saier MH Jr.\n",
      "  TITLE     The ascorbate transporter of Escherichia coli.\n",
      "  JOURNAL   J Bacteriol 185:2243-50 (2003)\n",
      "            DOI:10.1128/JB.185.7.2243-2250.2003\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00040  Pentose and glucuronate interconversions\n",
      "            pae00051  Fructose and mannose metabolism\n",
      "            pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "KO_PATHWAY  ko00053\n",
      "///\n",
      "path:pae00061\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00061                    Pathway\n",
      "NAME        Fatty acid biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Lipid metabolism\n",
      "PATHWAY_MAP pae00061  Fatty acid biosynthesis\n",
      "MODULE      pae_M00082  Fatty acid biosynthesis, initiation [PATH:pae00061]\n",
      "            pae_M00083  Fatty acid biosynthesis, elongation [PATH:pae00061]\n",
      "            pae_M00086  beta-Oxidation, acyl-CoA synthesis [PATH:pae00061]\n",
      "DBLINKS     GO: 0006633\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3639  accA; acetyl-CoA carboxylase carboxyltransferase subunit alpha [KO:K01962] [EC:6.4.1.2 2.1.3.15]\n",
      "line in gene PA3639  accA; acetyl-CoA carboxylase carboxyltransferase subunit alpha [KO:K01962] [EC:6.4.1.2 2.1.3.15]\n",
      "            PA4847  accB; acetyl-CoA carboxylase biotin carboxyl carrier protein subunit [KO:K02160]\n",
      "line in gene PA4847  accB; acetyl-CoA carboxylase biotin carboxyl carrier protein subunit [KO:K02160]\n",
      "            PA0493  acetyl-CoA carboxylase biotin carboxyl carrier protein subunit [KO:K02160]\n",
      "line in gene PA0493  acetyl-CoA carboxylase biotin carboxyl carrier protein subunit [KO:K02160]\n",
      "            PA4848  accC; acetyl-CoA carboxylase biotin carboxylase subunit [KO:K01961] [EC:6.4.1.2 6.3.4.14]\n",
      "line in gene PA4848  accC; acetyl-CoA carboxylase biotin carboxylase subunit [KO:K01961] [EC:6.4.1.2 6.3.4.14]\n",
      "            PA0494  acetyl-CoA carboxylase biotin carboxylase subunit [KO:K01961] [EC:6.4.1.2 6.3.4.14]\n",
      "line in gene PA0494  acetyl-CoA carboxylase biotin carboxylase subunit [KO:K01961] [EC:6.4.1.2 6.3.4.14]\n",
      "            PA3112  accD; acetyl-CoA carboxylase carboxyltransferase subunit beta [KO:K01963] [EC:6.4.1.2 2.1.3.15]\n",
      "line in gene PA3112  accD; acetyl-CoA carboxylase carboxyltransferase subunit beta [KO:K01963] [EC:6.4.1.2 2.1.3.15]\n",
      "            PA2968  fabD; malonyl CoA-ACP transacylase [KO:K00645] [EC:2.3.1.39]\n",
      "line in gene PA2968  fabD; malonyl CoA-ACP transacylase [KO:K00645] [EC:2.3.1.39]\n",
      "            PA3333  fabH2; 3-oxoacyl-ACP synthase III [KO:K00648] [EC:2.3.1.180]\n",
      "line in gene PA3333  fabH2; 3-oxoacyl-ACP synthase III [KO:K00648] [EC:2.3.1.180]\n",
      "            PA5174  beta-ketoacyl synthase [KO:K18473] [EC:2.3.1.180]\n",
      "line in gene PA5174  beta-ketoacyl synthase [KO:K18473] [EC:2.3.1.180]\n",
      "            PA1609  fabB; 3-oxoacyl-ACP synthase [KO:K00647] [EC:2.3.1.41]\n",
      "line in gene PA1609  fabB; 3-oxoacyl-ACP synthase [KO:K00647] [EC:2.3.1.41]\n",
      "            PA0098  3-oxoacyl-ACP synthase [KO:K00647] [EC:2.3.1.41]\n",
      "line in gene PA0098  3-oxoacyl-ACP synthase [KO:K00647] [EC:2.3.1.41]\n",
      "            PA2965  fabF1; 3-oxoacyl-ACP synthase [KO:K09458] [EC:2.3.1.179]\n",
      "line in gene PA2965  fabF1; 3-oxoacyl-ACP synthase [KO:K09458] [EC:2.3.1.179]\n",
      "            PA1373  fabF2; 3-oxoacyl-ACP synthase [KO:K09458] [EC:2.3.1.179]\n",
      "line in gene PA1373  fabF2; 3-oxoacyl-ACP synthase [KO:K09458] [EC:2.3.1.179]\n",
      "            PA5524  short-chain dehydrogenase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA5524  short-chain dehydrogenase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA1470  short-chain dehydrogenase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA1470  short-chain dehydrogenase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA4389  speA; 3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA4389  speA; 3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA2967  fabG; 3-oxoacyl-[acyl-carrier-protein] reductase FabG [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA2967  fabG; 3-oxoacyl-[acyl-carrier-protein] reductase FabG [KO:K00059] [EC:1.1.1.100]\n",
      "            PA4786  3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA4786  3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA4089  3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA4089  3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA0182  3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA0182  3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA1827  short-chain dehydrogenase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA1827  short-chain dehydrogenase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA1610  fabA; 3-hydroxydecanoyl-ACP dehydratase [KO:K01716] [EC:4.2.1.59 5.3.3.14]\n",
      "line in gene PA1610  fabA; 3-hydroxydecanoyl-ACP dehydratase [KO:K01716] [EC:4.2.1.59 5.3.3.14]\n",
      "            PA3645  fabZ; 3-hydroxyacyl-[acyl-carrier-protein] dehydratase FabZ [KO:K02372] [EC:4.2.1.59]\n",
      "line in gene PA3645  fabZ; 3-hydroxyacyl-[acyl-carrier-protein] dehydratase FabZ [KO:K02372] [EC:4.2.1.59]\n",
      "            PA1806  fabI; NADH-dependent enoyl-ACP reductase [KO:K00208] [EC:1.3.1.9 1.3.1.10]\n",
      "line in gene PA1806  fabI; NADH-dependent enoyl-ACP reductase [KO:K00208] [EC:1.3.1.9 1.3.1.10]\n",
      "            PA2950  reductase [KO:K00209] [EC:1.3.1.9 1.3.1.44]\n",
      "line in gene PA2950  reductase [KO:K00209] [EC:1.3.1.9 1.3.1.44]\n",
      "            PA3299  fadD1; long-chain-fatty-acid--CoA ligase [KO:K01897] [EC:6.2.1.3]\n",
      "line in gene PA3299  fadD1; long-chain-fatty-acid--CoA ligase [KO:K01897] [EC:6.2.1.3]\n",
      "            PA3300  fadD2; long-chain-fatty-acid--CoA ligase [KO:K01897] [EC:6.2.1.3]\n",
      "line in gene PA3300  fadD2; long-chain-fatty-acid--CoA ligase [KO:K01897] [EC:6.2.1.3]\n",
      "            PA1617  AMP-binding protein [KO:K01897] [EC:6.2.1.3]\n",
      "line in gene PA1617  AMP-binding protein [KO:K01897] [EC:6.2.1.3]\n",
      "COMPOUND    C00024  Acetyl-CoA\n",
      "            C00083  Malonyl-CoA\n",
      "            C00154  Palmitoyl-CoA\n",
      "            C00173  Acyl-[acyl-carrier protein]\n",
      "            C00229  Acyl-carrier protein\n",
      "            C00249  Hexadecanoic acid\n",
      "            C00383  Malonate\n",
      "            C00638  Long-chain fatty acid\n",
      "            C00685  3-Oxoacyl-[acyl-carrier protein]\n",
      "            C00693  trans-2,3-Dehydroacyl-[acyl-carrier protein]\n",
      "            C00712  (9Z)-Octadecenoic acid\n",
      "            C01203  Oleoyl-[acyl-carrier protein]\n",
      "            C01209  Malonyl-[acyl-carrier protein]\n",
      "            C01271  (3R)-3-Hydroxyacyl-[acyl-carrier protein]\n",
      "            C01530  Octadecanoic acid\n",
      "            C01571  Decanoic acid\n",
      "            C02679  Dodecanoic acid\n",
      "            C02843  Long-chain acyl-CoA\n",
      "            C03939  Acetyl-[acyl-carrier protein]\n",
      "            C04088  Octadecanoyl-[acyl-carrier protein]\n",
      "            C04180  cis-Dec-3-enoyl-[acp]\n",
      "            C04246  But-2-enoyl-[acyl-carrier protein]\n",
      "            C04618  (3R)-3-Hydroxybutanoyl-[acyl-carrier protein]\n",
      "            C04619  (3R)-3-Hydroxydecanoyl-[acyl-carrier protein]\n",
      "            C04620  (3R)-3-Hydroxyoctanoyl-[acyl-carrier protein]\n",
      "            C04633  (3R)-3-Hydroxypalmitoyl-[acyl-carrier protein]\n",
      "            C04688  (3R)-3-Hydroxytetradecanoyl-[acyl-carrier protein]\n",
      "            C05223  Dodecanoyl-[acyl-carrier protein]\n",
      "            C05744  Acetoacetyl-[acp]\n",
      "            C05745  Butyryl-[acp]\n",
      "            C05746  3-Oxohexanoyl-[acp]\n",
      "            C05747  (R)-3-Hydroxyhexanoyl-[acp]\n",
      "            C05748  trans-Hex-2-enoyl-[acp]\n",
      "            C05749  Hexanoyl-[acp]\n",
      "            C05750  3-Oxooctanoyl-[acp]\n",
      "            C05751  trans-Oct-2-enoyl-[acp]\n",
      "            C05752  Octanoyl-[acp]\n",
      "            C05753  3-Oxodecanoyl-[acp]\n",
      "            C05754  trans-Dec-2-enoyl-[acp]\n",
      "            C05755  Decanoyl-[acp]\n",
      "            C05756  3-Oxododecanoyl-[acp]\n",
      "            C05757  (R)-3-Hydroxydodecanoyl-[acp]\n",
      "            C05758  trans-Dodec-2-enoyl-[acp]\n",
      "            C05759  3-Oxotetradecanoyl-[acp]\n",
      "            C05760  trans-Tetradec-2-enoyl-[acp]\n",
      "            C05761  Tetradecanoyl-[acp]\n",
      "            C05762  3-Oxohexadecanoyl-[acp]\n",
      "            C05763  trans-Hexadec-2-enoyl-[acp]\n",
      "            C05764  Hexadecanoyl-[acp]\n",
      "            C06423  Octanoic acid\n",
      "            C06424  Tetradecanoic acid\n",
      "            C08362  (9Z)-Hexadecenoic acid\n",
      "            C16219  3-Oxostearoyl-[acp]\n",
      "            C16220  (R)-3-Hydroxyoctadecanoyl-[acp]\n",
      "            C16221  (2E)-Octadecenoyl-[acp]\n",
      "            C16520  Hexadecenoyl-[acyl-carrier protein]\n",
      "            C20683  Long-chain acyl-[acyl-carrier protein]\n",
      "            C20794  n-7 Unsaturated acyl-[acyl-carrier protein]\n",
      "REFERENCE   PMID:12061798\n",
      "  AUTHORS   Salas JJ, Ohlrogge JB.\n",
      "  TITLE     Characterization of substrate specificity of plant FatA and FatB acyl-ACP thioesterases.\n",
      "  JOURNAL   Arch Biochem Biophys 403:25-34 (2002)\n",
      "            DOI:10.1016/S0003-9861(02)00017-6\n",
      "REFERENCE   PMID:12518017\n",
      "  AUTHORS   Zhang YM, Marrakchi H, White SW, Rock CO.\n",
      "  TITLE     The application of computational methods to explore the diversity and structure of bacterial fatty acid synthase.\n",
      "  JOURNAL   J Lipid Res 44:1-10 (2003)\n",
      "            DOI:10.1194/jlr.R200016-JLR200\n",
      "REFERENCE   PMID:11337402\n",
      "  AUTHORS   Voelker T, Kinney AJ.\n",
      "  TITLE     VARIATIONS IN THE BIOSYNTHESIS OF SEED-STORAGE LIPIDS.\n",
      "  JOURNAL   Annu Rev Plant Physiol Plant Mol Biol 52:335-361 (2001)\n",
      "            DOI:10.1146/annurev.arplant.52.1.335\n",
      "REFERENCE   PMID:17573542\n",
      "  AUTHORS   Barker GC, Larson TR, Graham IA, Lynn JR, King GJ.\n",
      "  TITLE     Novel insights into seed fatty acid synthesis and modification pathways from genetic diversity and quantitative trait Loci analysis of the Brassica C genome.\n",
      "  JOURNAL   Plant Physiol 144:1827-42 (2007)\n",
      "            DOI:10.1104/pp.107.096172\n",
      "REFERENCE   PMID:30201289\n",
      "  AUTHORS   Bowman CE, Wolfgang MJ\n",
      "  TITLE     Role of the malonyl-CoA synthetase ACSF3 in mitochondrial metabolism.\n",
      "  JOURNAL   Adv Biol Regul 71:34-40 (2019)\n",
      "            DOI:10.1016/j.jbior.2018.09.002\n",
      "REFERENCE   PMID:25203508\n",
      "  AUTHORS   Venkatesan R, Sah-Teli SK, Awoniyi LO, Jiang G, Prus P, Kastaniotis AJ, Hiltunen JK, Wierenga RK, Chen Z\n",
      "  TITLE     Insights into mitochondrial fatty acid synthesis from the structure of heterotetrameric 3-ketoacyl-ACP reductase/3R-hydroxyacyl-CoA dehydrogenase.\n",
      "  JOURNAL   Nat Commun 5:4805 (2014)\n",
      "            DOI:10.1038/ncomms5805\n",
      "REFERENCE   PMID:19686777\n",
      "  AUTHORS   Hiltunen JK, Chen Z, Haapalainen AM, Wierenga RK, Kastaniotis AJ\n",
      "  TITLE     Mitochondrial fatty acid synthesis--an adopted set of enzymes making a pathway of major importance for the cellular metabolism.\n",
      "  JOURNAL   Prog Lipid Res 49:27-45 (2010)\n",
      "            DOI:10.1016/j.plipres.2009.08.001\n",
      "REFERENCE   PMID:27553474\n",
      "  AUTHORS   Kastaniotis AJ, Autio KJ, Keratar JM, Monteuuis G, Makela AM, Nair RR, Pietikainen LP, Shvetsova A, Chen Z, Hiltunen JK\n",
      "  TITLE     Mitochondrial fatty acid synthesis, fatty acids and mitochondrial physiology.\n",
      "  JOURNAL   Biochim Biophys Acta Mol Cell Biol Lipids 1862:39-48 (2017)\n",
      "            DOI:10.1016/j.bbalip.2016.08.011\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00071  Fatty acid degradation\n",
      "            pae00410  beta-Alanine metabolism\n",
      "            pae00561  Glycerolipid metabolism\n",
      "            pae00564  Glycerophospholipid metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00785  Lipoic acid metabolism\n",
      "KO_PATHWAY  ko00061\n",
      "///\n",
      "path:pae00071\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00071                    Pathway\n",
      "NAME        Fatty acid degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Lipid metabolism\n",
      "PATHWAY_MAP pae00071  Fatty acid degradation\n",
      "MODULE      pae_M00086  beta-Oxidation, acyl-CoA synthesis [PATH:pae00071]\n",
      "            pae_M00087  beta-Oxidation [PATH:pae00071]\n",
      "DBLINKS     GO: 0009062\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA2940  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA2940  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA3013  foaB; 3-ketoacyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA3013  foaB; 3-ketoacyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA3454  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA3454  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "line in gene PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "            PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "line in gene PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA2552  acyl-CoA dehydrogenase [KO:K00248] [EC:1.3.8.1]\n",
      "line in gene PA2552  acyl-CoA dehydrogenase [KO:K00248] [EC:1.3.8.1]\n",
      "            PA2550  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "line in gene PA2550  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "            PA0879  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "line in gene PA0879  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "            PA1535  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "line in gene PA1535  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "            PA4994  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "line in gene PA4994  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "            PA4979  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "line in gene PA4979  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "            PA3593  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "line in gene PA3593  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "            PA2815  acyl-CoA dehydrogenase [KO:K06445] [EC:1.3.99.-]\n",
      "line in gene PA2815  acyl-CoA dehydrogenase [KO:K06445] [EC:1.3.99.-]\n",
      "            PA0447  gcdH; glutaryl-CoA dehydrogenase [KO:K00252] [EC:1.3.8.6]\n",
      "line in gene PA0447  gcdH; glutaryl-CoA dehydrogenase [KO:K00252] [EC:1.3.8.6]\n",
      "            PA3299  fadD1; long-chain-fatty-acid--CoA ligase [KO:K01897] [EC:6.2.1.3]\n",
      "line in gene PA3299  fadD1; long-chain-fatty-acid--CoA ligase [KO:K01897] [EC:6.2.1.3]\n",
      "            PA3300  fadD2; long-chain-fatty-acid--CoA ligase [KO:K01897] [EC:6.2.1.3]\n",
      "line in gene PA3300  fadD2; long-chain-fatty-acid--CoA ligase [KO:K01897] [EC:6.2.1.3]\n",
      "            PA1617  AMP-binding protein [KO:K01897] [EC:6.2.1.3]\n",
      "line in gene PA1617  AMP-binding protein [KO:K01897] [EC:6.2.1.3]\n",
      "            PA1525  alkB2; alkane-1 monooxygenase [KO:K00496] [EC:1.14.15.3]\n",
      "line in gene PA1525  alkB2; alkane-1 monooxygenase [KO:K00496] [EC:1.14.15.3]\n",
      "            PA2574  alkB1; alkane-1 monooxygenase [KO:K00496] [EC:1.14.15.3]\n",
      "line in gene PA2574  alkB1; alkane-1 monooxygenase [KO:K00496] [EC:1.14.15.3]\n",
      "            PA5349  rubredoxin reductase [KO:K05297] [EC:1.18.1.1]\n",
      "line in gene PA5349  rubredoxin reductase [KO:K05297] [EC:1.18.1.1]\n",
      "            PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "line in gene PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "            PA5427  adhA; alcohol dehydrogenase [KO:K13953] [EC:1.1.1.1]\n",
      "line in gene PA5427  adhA; alcohol dehydrogenase [KO:K13953] [EC:1.1.1.1]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "COMPOUND    C00010  CoA\n",
      "            C00024  Acetyl-CoA\n",
      "            C00071  Aldehyde\n",
      "            C00136  Butanoyl-CoA\n",
      "            C00154  Palmitoyl-CoA\n",
      "            C00162  Fatty acid\n",
      "            C00226  Primary alcohol\n",
      "            C00229  Acyl-carrier protein\n",
      "            C00249  Hexadecanoic acid\n",
      "            C00332  Acetoacetyl-CoA\n",
      "            C00340  Reduced rubredoxin\n",
      "            C00435  Oxidized rubredoxin\n",
      "            C00489  Glutarate\n",
      "            C00517  Hexadecanal\n",
      "            C00527  Glutaryl-CoA\n",
      "            C00638  Long-chain fatty acid\n",
      "            C00823  1-Hexadecanol\n",
      "            C00877  Crotonoyl-CoA\n",
      "            C01144  (S)-3-Hydroxybutanoyl-CoA\n",
      "            C01371  Alkane\n",
      "            C01832  Lauroyl-CoA\n",
      "            C01944  Octanoyl-CoA\n",
      "            C02593  Tetradecanoyl-CoA\n",
      "            C02990  L-Palmitoylcarnitine\n",
      "            C03221  2-trans-Dodecenoyl-CoA\n",
      "            C03547  omega-Hydroxy fatty acid\n",
      "            C03561  (R)-3-Hydroxybutanoyl-CoA\n",
      "            C05102  alpha-Hydroxy fatty acid\n",
      "            C05258  (S)-3-Hydroxyhexadecanoyl-CoA\n",
      "            C05259  3-Oxopalmitoyl-CoA\n",
      "            C05260  (S)-3-Hydroxytetradecanoyl-CoA\n",
      "            C05261  3-Oxotetradecanoyl-CoA\n",
      "            C05262  (S)-3-Hydroxydodecanoyl-CoA\n",
      "            C05263  3-Oxododecanoyl-CoA\n",
      "            C05264  (S)-Hydroxydecanoyl-CoA\n",
      "            C05265  3-Oxodecanoyl-CoA\n",
      "            C05266  (S)-3-Hydroxyoctanoyl-CoA\n",
      "            C05267  3-Oxooctanoyl-CoA\n",
      "            C05268  (S)-Hydroxyhexanoyl-CoA\n",
      "            C05269  3-Oxohexanoyl-CoA\n",
      "            C05270  Hexanoyl-CoA\n",
      "            C05271  trans-Hex-2-enoyl-CoA\n",
      "            C05272  trans-Hexadec-2-enoyl-CoA\n",
      "            C05273  trans-Tetradec-2-enoyl-CoA\n",
      "            C05274  Decanoyl-CoA\n",
      "            C05275  trans-Dec-2-enoyl-CoA\n",
      "            C05276  trans-Oct-2-enoyl-CoA\n",
      "            C05279  trans,cis-Lauro-2,6-dienoyl-CoA\n",
      "            C05280  cis,cis-3,6-Dodecadienoyl-CoA\n",
      "            C20683  Long-chain acyl-[acyl-carrier protein]\n",
      "REFERENCE   PMID:869535\n",
      "  AUTHORS   Parekh VR, Traxler RW, Sobek JM\n",
      "  TITLE     N-Alkane oxidation enzymes of a pseudomonad.\n",
      "  JOURNAL   Appl Environ Microbiol 33:881-4 (1977)\n",
      "            DOI:10.1128/AEM.33.4.881-884.1977\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00061  Fatty acid biosynthesis\n",
      "            pae00561  Glycerolipid metabolism\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00650  Butanoate metabolism\n",
      "KO_PATHWAY  ko00071\n",
      "///\n",
      "path:pae00130\n",
      "ENTRY       pae00130                    Pathway\n",
      "NAME        Ubiquinone and other terpenoid-quinone biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Ubiquinone (UQ), also called coenzyme Q, and plastoquinone (PQ) are electron carriers in oxidative phosphorylation and photosynthesis, respectively. The quinoid nucleus of ubiquinone is derived from the shikimate pathway; 4-hydroxybenzoate is directly formed from chorismate in bacteria, while it can be formed from either chorismate or tyrosine in yeast. The following biosynthesis of terpenoid moiety involves reactions of prenylation, decarboxylation, and three hydroxylations alternating with three methylations. The order of these reactions are somewhat different between bacteria and yeast. Phylloquinone (vitamin K1), menaquinone (vitamin K2), and tocopherol (vitamin E) are fat-soluble vitamins. Phylloquinone is a compound present in all photosynthetic plants serving as a cofactor for photosystem I-mediated electron transport. Menaquinone is an obligatory component of the electron-transfer pathway in bacteria.\n",
      "CLASS       Metabolism; Metabolism of cofactors and vitamins\n",
      "PATHWAY_MAP pae00130  Ubiquinone and other terpenoid-quinone biosynthesis\n",
      "DBLINKS     GO: 0006744 0009234 0042372\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0968  hypothetical protein [KO:K01075] [EC:3.1.2.23]\n",
      "line in gene PA0968  hypothetical protein [KO:K01075] [EC:3.1.2.23]\n",
      "            PA5357  hypothetical protein [KO:K03181] [EC:4.1.3.40]\n",
      "line in gene PA5357  hypothetical protein [KO:K03181] [EC:4.1.3.40]\n",
      "            PA5358  ubiA; 4-hydroxybenzoate octaprenyltransferase [KO:K03179] [EC:2.5.1.39]\n",
      "line in gene PA5358  ubiA; 4-hydroxybenzoate octaprenyltransferase [KO:K03179] [EC:2.5.1.39]\n",
      "            PA5237  3-octaprenyl-4-hydroxybenzoate carboxy-lyase [KO:K03182] [EC:4.1.1.98]\n",
      "line in gene PA5237  3-octaprenyl-4-hydroxybenzoate carboxy-lyase [KO:K03182] [EC:4.1.1.98]\n",
      "            PA4019  aromatic acid decarboxylase [KO:K03186] [EC:2.5.1.129]\n",
      "line in gene PA4019  aromatic acid decarboxylase [KO:K03186] [EC:2.5.1.129]\n",
      "            PA5221  2-octaprenyl-3-methyl-6-methoxy-1,4-benzoquinol hydroxylase [KO:K18800] [EC:1.14.13.240]\n",
      "line in gene PA5221  2-octaprenyl-3-methyl-6-methoxy-1,4-benzoquinol hydroxylase [KO:K18800] [EC:1.14.13.240]\n",
      "            PA3171  ubiG; ubiquinone biosynthesis O-methyltransferase [KO:K00568] [EC:2.1.1.222 2.1.1.64]\n",
      "line in gene PA3171  ubiG; ubiquinone biosynthesis O-methyltransferase [KO:K00568] [EC:2.1.1.222 2.1.1.64]\n",
      "            PA5223  ubiH; 2-octaprenyl-6-methoxyphenyl hydroxylase [KO:K03185] [EC:1.14.13.-]\n",
      "line in gene PA5223  ubiH; 2-octaprenyl-6-methoxyphenyl hydroxylase [KO:K03185] [EC:1.14.13.-]\n",
      "            PA5063  ubiE; ubiquinone/menaquinone biosynthesis methyltransferase [KO:K03183] [EC:2.1.1.163 2.1.1.201]\n",
      "line in gene PA5063  ubiE; ubiquinone/menaquinone biosynthesis methyltransferase [KO:K03183] [EC:2.1.1.163 2.1.1.201]\n",
      "            PA0655  2-nonaprenyl-3-methyl-6-methoxy-1,4-benzoquinol hydroxylase [KO:K06134] [EC:1.14.99.60]\n",
      "line in gene PA0655  2-nonaprenyl-3-methyl-6-methoxy-1,4-benzoquinol hydroxylase [KO:K06134] [EC:1.14.99.60]\n",
      "            PA4231  pchA; salicylate biosynthesis isochorismate synthase [KO:K01851] [EC:5.4.4.2]\n",
      "line in gene PA4231  pchA; salicylate biosynthesis isochorismate synthase [KO:K01851] [EC:5.4.4.2]\n",
      "            PA1618  esterase [KO:K19222] [EC:3.1.2.28]\n",
      "line in gene PA1618  esterase [KO:K19222] [EC:3.1.2.28]\n",
      "            PA1225  NAD(P)H dehydrogenase [KO:K00355] [EC:1.6.5.2]\n",
      "line in gene PA1225  NAD(P)H dehydrogenase [KO:K00355] [EC:1.6.5.2]\n",
      "            PA4975  NAD(P)H quinone oxidoreductase [KO:K00355] [EC:1.6.5.2]\n",
      "line in gene PA4975  NAD(P)H quinone oxidoreductase [KO:K00355] [EC:1.6.5.2]\n",
      "            PA1224  NAD(P)H dehydrogenase [KO:K00355] [EC:1.6.5.2]\n",
      "line in gene PA1224  NAD(P)H dehydrogenase [KO:K00355] [EC:1.6.5.2]\n",
      "            PA0853  oxidoreductase [KO:K00355] [EC:1.6.5.2]\n",
      "line in gene PA0853  oxidoreductase [KO:K00355] [EC:1.6.5.2]\n",
      "            PA0949  wrbA; NAD(P)H dehydrogenase [KO:K03809] [EC:1.6.5.2]\n",
      "line in gene PA0949  wrbA; NAD(P)H dehydrogenase [KO:K03809] [EC:1.6.5.2]\n",
      "            PA0865  hpd; 4-hydroxyphenylpyruvate dioxygenase [KO:K00457] [EC:1.13.11.27]\n",
      "line in gene PA0865  hpd; 4-hydroxyphenylpyruvate dioxygenase [KO:K00457] [EC:1.13.11.27]\n",
      "            PA0242  hypothetical protein [KO:K00457] [EC:1.13.11.27]\n",
      "line in gene PA0242  hypothetical protein [KO:K00457] [EC:1.13.11.27]\n",
      "COMPOUND    C00082  L-Tyrosine\n",
      "            C00156  4-Hydroxybenzoate\n",
      "            C00223  p-Coumaroyl-CoA\n",
      "            C00251  Chorismate\n",
      "            C00341  Geranyl diphosphate\n",
      "            C00353  Geranylgeranyl diphosphate\n",
      "            C00390  Ubiquinol\n",
      "            C00423  trans-Cinnamate\n",
      "            C00544  Homogentisate\n",
      "            C00811  4-Coumarate\n",
      "            C00828  Menaquinone\n",
      "            C00885  Isochorismate\n",
      "            C01179  3-(4-Hydroxyphenyl)pyruvate\n",
      "            C02059  Phylloquinone\n",
      "            C02477  alpha-Tocopherol\n",
      "            C02483  gamma-Tocopherol\n",
      "            C02730  2-Succinylbenzoate\n",
      "            C02949  4-Hydroxybenzoyl-CoA\n",
      "            C03160  2-Succinylbenzoyl-CoA\n",
      "            C03313  Phylloquinol\n",
      "            C03657  1,4-Dihydroxy-2-naphthoate\n",
      "            C03964  (R)-3-(4-Hydroxyphenyl)lactate\n",
      "            C03993  4-(beta-D-Glucosyloxy)benzoate\n",
      "            C04145  all-trans-Nonaprenyl diphosphate\n",
      "            C05427  Phytyl diphosphate\n",
      "            C05807  2-Polyprenylphenol\n",
      "            C05817  (1R,6R)-6-Hydroxy-2-succinylcyclohexa-2,4-diene-1-carboxylate\n",
      "            C05819  Menaquinol\n",
      "            C05847  all-trans-Polyprenyl diphosphate\n",
      "            C05848  4-Hydroxy-3-polyprenylbenzoate\n",
      "            C05849  Vitamin K1 epoxide\n",
      "            C06986  Spirodilactone\n",
      "            C10793  Geranylhydroquinone\n",
      "            C13309  2-Phytyl-1,4-naphthoquinone\n",
      "            C14151  delta-Tocopherol\n",
      "            C14152  beta-Tocopherol\n",
      "            C14153  alpha-Tocotrienol\n",
      "            C14154  beta-Tocotrienol\n",
      "            C14155  gamma-Tocotrienol\n",
      "            C14156  delta-Tocotrienol\n",
      "            C15547  1,4-Dihydroxy-2-naphthoyl-CoA\n",
      "            C15882  2-Methyl-6-phytylquinol\n",
      "            C15883  2,3-Dimethyl-5-phytylquinol\n",
      "            C16519  2-Succinyl-5-enolpyruvyl-6-hydroxy-3-cyclohexene-1-carboxylate\n",
      "            C16695  Plastoquinol-9\n",
      "            C16999  Futalosine\n",
      "            C17010  Dehypoxanthine futalosine\n",
      "            C17017  Cyclic dehypoxanthine futalosine\n",
      "            C17018  1,4-Dihydroxy-6-naphthoate\n",
      "            C17412  Shikonin\n",
      "            C17551  2-Polyprenyl-6-hydroxyphenol\n",
      "            C17552  2-Polyprenyl-6-methoxyphenol\n",
      "            C17554  3,4-Dihydroxy-5-polyprenylbenzoate\n",
      "            C17559  3-Methoxy-4-hydroxy-5-polyprenylbenzoate\n",
      "            C17570  2-Methyl-6-solanyl-1,4-benzoquinol\n",
      "            C18131  Geranyl-hydroxybenzoate\n",
      "            C18132  3''-Hydroxy-geranylhydroquinone\n",
      "            C18133  Deoxyshikonin\n",
      "            C18134  Dihydroechinofuran\n",
      "            C18135  Dihydroshikonofuran\n",
      "            C19847  Demethylmenaquinol\n",
      "            C19858  2-Methoxy-6-all-trans-polyprenyl-1,4-benzoquinol\n",
      "            C19859  6-Methoxy-3-methyl-2-all-trans-polyprenyl-1,4-benzoquinol\n",
      "            C20265  2,3-Epoxymenaquinone\n",
      "            C20737  6-Geranylgeranyl-2-methylbenzene-1,4-diol\n",
      "            C20738  6-Geranylgeranyl-2,3-dimethylbenzene-1,4-diol\n",
      "            C20772  3-[(1-Carboxyvinyl)oxy]benzoate\n",
      "            C20773  6-Amino-6-deoxyfutalosine\n",
      "            C21084  Demethylphylloquinol\n",
      "            C21860  3-Demethylubiquinol\n",
      "            C22039  2-Carboxy-1,4-naphthoquinone\n",
      "REFERENCE   PMID:12908086\n",
      "  AUTHORS   Ward MJ, Fu QS, Rhoads KR, Yeung CH, Spormann AM, Criddle CS.\n",
      "  TITLE     A derivative of the menaquinone precursor 1,4-dihydroxy-2-naphthoate is involved in the reductive transformation of carbon tetrachloride by aerobically grown Shewanella oneidensis MR-1.\n",
      "  JOURNAL   Appl Microbiol Biotechnol 63:571-7 (2004)\n",
      "            DOI:10.1007/s00253-003-1407-3\n",
      "REFERENCE   PMID:14499605\n",
      "  AUTHORS   Monzingo AF, Gao J, Qiu J, Georgiou G, Robertus JD.\n",
      "  TITLE     The X-ray structure of Escherichia coli RraA (MenG), A protein inhibitor of RNA processing.\n",
      "  JOURNAL   J Mol Biol 332:1015-24 (2003)\n",
      "            DOI:10.1016/S0022-2836(03)00970-7\n",
      "REFERENCE   PMID:11470786\n",
      "  AUTHORS   Johnson TW, Zybailov B, Jones AD, Bittl R, Zech S, Stehlik D, Golbeck JH, Chitnis PR.\n",
      "  TITLE     Recruitment of a foreign quinone into the A1 site of photosystem I. In vivo replacement of plastoquinone-9 by media-supplemented naphthoquinones in phylloquinone biosynthetic pathway mutants of Synechocystis sp. PCC 6803.\n",
      "  JOURNAL   J Biol Chem 276:39512-21 (2001)\n",
      "            DOI:10.1074/jbc.M104040200\n",
      "REFERENCE   PMID:12615349\n",
      "  AUTHORS   Wade Johnson T, Naithani S, Stewart C Jr, Zybailov B, Daniel Jones A, Golbeck JH, Chitnis PR.\n",
      "  TITLE     The menD and menE homologs code for 2-succinyl-6-hydroxyl-2,4-cyclohexadiene-1-carboxylate synthase and O-succinylbenzoic acid-CoA synthase in the phylloquinone biosynthetic pathway of Synechocystis sp. PCC 6803.\n",
      "  JOURNAL   Biochim Biophys Acta 1557:67-76 (2003)\n",
      "            DOI:10.1016/S0005-2728(02)00396-1\n",
      "REFERENCE   PMID:12721307\n",
      "  AUTHORS   Gin P, Hsu AY, Rothman SC, Jonassen T, Lee PT, Tzagoloff A, Clarke CF.\n",
      "  TITLE     The Saccharomyces cerevisiae COQ6 gene encodes a mitochondrial flavin-dependent monooxygenase required for coenzyme Q biosynthesis.\n",
      "  JOURNAL   J Biol Chem 278:25308-16 (2003)\n",
      "            DOI:10.1074/jbc.M303234200\n",
      "REFERENCE   PMID:18284213\n",
      "  AUTHORS   Jiang M, Chen X, Guo ZF, Cao Y, Chen M, Guo Z.\n",
      "  TITLE     Identification and Characterization of (1R,6R)-2-Succinyl-6-hydroxy-2,4-cyclohexadiene-1-carboxylate Synthase in the Menaquinone Biosynthesis of Escherichia coli.\n",
      "  JOURNAL   Biochemistry 47:3426-34 (2008)\n",
      "            DOI:10.1021/bi7023755\n",
      "REFERENCE   PMID:17760421\n",
      "  AUTHORS   Jiang M, Cao Y, Guo ZF, Chen M, Chen X, Guo Z.\n",
      "  TITLE     Menaquinone biosynthesis in Escherichia coli: identification of 2-succinyl-5-enolpyruvyl-6-hydroxy-3-cyclohexene-1-carboxylate as a novel intermediate and re-evaluation of MenD activity.\n",
      "  JOURNAL   Biochemistry 46:10979-89 (2007)\n",
      "            DOI:10.1021/bi700810x\n",
      "REFERENCE   PMID:17956107\n",
      "  AUTHORS   Jiang M, Chen M, Cao Y, Yang Y, Sze KH, Chen X, Guo Z.\n",
      "  TITLE     Determination of the stereochemistry of 2-succinyl-5-enolpyruvyl-6-hydroxy-3- cyclohexene-1-carboxylate, a key intermediate in menaquinone biosynthesis.\n",
      "  JOURNAL   Org Lett 9:4765-7 (2007)\n",
      "            DOI:10.1021/ol702126m\n",
      "REFERENCE   PMID:18801996\n",
      "  AUTHORS   Hiratsuka T, Furihata K, Ishikawa J, Yamashita H, Itoh N, Seto H, Dairi T\n",
      "  TITLE     An alternative menaquinone biosynthetic pathway operating in microorganisms.\n",
      "  JOURNAL   Science 321:1670-3 (2008)\n",
      "            DOI:10.1126/science.1160446\n",
      "REFERENCE   PMID:12232327\n",
      "  AUTHORS   Loscher R, Heide L\n",
      "  TITLE     Biosynthesis of p-Hydroxybenzoate from p-Coumarate and p-Coumaroyl-Coenzyme A in Cell-Free Extracts of Lithospermum erythrorhizon Cell Cultures.\n",
      "  JOURNAL   Plant Physiol 106:271-279 (1994)\n",
      "            DOI:10.1104/pp.106.1.271\n",
      "REFERENCE   PMID:11583838\n",
      "  AUTHORS   Meganathan R\n",
      "  TITLE     Ubiquinone biosynthesis in microorganisms.\n",
      "  JOURNAL   FEMS Microbiol Lett 203:131-9 (2001)\n",
      "            DOI:10.1111/j.1574-6968.2001.tb10831.x\n",
      "REFERENCE   PMID:14508009\n",
      "  AUTHORS   Cheng Z, Sattler S, Maeda H, Sakuragi Y, Bryant DA, DellaPenna D\n",
      "  TITLE     Highly divergent methyltransferases catalyze a conserved reaction in tocopherol and plastoquinone synthesis in cyanobacteria and photosynthetic eukaryotes.\n",
      "  JOURNAL   Plant Cell 15:2343-56 (2003)\n",
      "            DOI:10.1105/tpc.013656\n",
      "REFERENCE   PMID:16989822\n",
      "  AUTHORS   Sadre R, Gruber J, Frentzen M\n",
      "  TITLE     Characterization of homogentisate prenyltransferases involved in plastoquinone-9 and tocochromanol biosynthesis.\n",
      "  JOURNAL   FEBS Lett 580:5357-62 (2006)\n",
      "            DOI:10.1016/j.febslet.2006.09.002\n",
      "REFERENCE   PMID:12523993\n",
      "  AUTHORS   Hofius D, Sonnewald U.\n",
      "  TITLE     Vitamin E biosynthesis: biochemistry meets cell biology.\n",
      "  JOURNAL   Trends Plant Sci 8:6-8 (2003)\n",
      "            DOI:10.1016/S1360-1385(02)00002-X\n",
      "REFERENCE   PMID:16233343\n",
      "  AUTHORS   Kawamukai M\n",
      "  TITLE     Biosynthesis, bioproduction and novel roles of ubiquinone.\n",
      "  JOURNAL   J Biosci Bioeng 94:511-7 (2002)\n",
      "            DOI:10.1016/S1389-1723(02)80188-8\n",
      "REFERENCE   PMID:21239697\n",
      "  AUTHORS   Tie JK, Jin DY, Straight DL, Stafford DW\n",
      "  TITLE     Functional study of the vitamin K cycle in mammalian cells.\n",
      "  JOURNAL   Blood 117:2967-74 (2011)\n",
      "            DOI:10.1182/blood-2010-08-304303\n",
      "REFERENCE   PMID:23113660\n",
      "  AUTHORS   Zhou L, Wang JY, Wu J, Wang J, Poplawsky A, Lin S, Zhu B, Chang C, Zhou T, Zhang LH, He YW\n",
      "  TITLE     The diffusible factor synthase XanB2 is a bifunctional chorismatase that links the shikimate pathway to ubiquinone and xanthomonadins biosynthetic pathways.\n",
      "  JOURNAL   Mol Microbiol 87:80-93 (2013)\n",
      "            DOI:10.1111/mmi.12084\n",
      "REL_PATHWAY pae00190  Oxidative phosphorylation\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "            pae00900  Terpenoid backbone biosynthesis\n",
      "KO_PATHWAY  ko00130\n",
      "///\n",
      "path:pae00190\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00190                    Pathway\n",
      "NAME        Oxidative phosphorylation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Energy metabolism\n",
      "PATHWAY_MAP pae00190  Oxidative phosphorylation\n",
      "MODULE      pae_M00144  NADH:quinone oxidoreductase, prokaryotes [PATH:pae00190]\n",
      "            pae_M00149  Succinate dehydrogenase, prokaryotes [PATH:pae00190]\n",
      "            pae_M00151  Cytochrome bc1 complex respiratory unit [PATH:pae00190]\n",
      "            pae_M00155  Cytochrome c oxidase, prokaryotes [PATH:pae00190]\n",
      "            pae_M00156  Cytochrome c oxidase, cbb3-type [PATH:pae00190]\n",
      "            pae_M00157  F-type ATPase, prokaryotes and chloroplasts [PATH:pae00190]\n",
      "            pae_M00417  Cytochrome o ubiquinol oxidase [PATH:pae00190]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1883  NADH-quinone oxidoreductase subunit A [KO:K00330] [EC:7.1.1.2]\n",
      "line in gene PA1883  NADH-quinone oxidoreductase subunit A [KO:K00330] [EC:7.1.1.2]\n",
      "            PA2637  nuoA; NADH-quinone oxidoreductase subunit A [KO:K00330] [EC:7.1.1.2]\n",
      "line in gene PA2637  nuoA; NADH-quinone oxidoreductase subunit A [KO:K00330] [EC:7.1.1.2]\n",
      "            PA2638  nuoB; NADH-quinone oxidoreductase subunit B [KO:K00331] [EC:7.1.1.2]\n",
      "line in gene PA2638  nuoB; NADH-quinone oxidoreductase subunit B [KO:K00331] [EC:7.1.1.2]\n",
      "            PA2639  nuoD; NADH:-quinone oxidoreductase subunit C/D [KO:K13378] [EC:7.1.1.2]\n",
      "line in gene PA2639  nuoD; NADH:-quinone oxidoreductase subunit C/D [KO:K13378] [EC:7.1.1.2]\n",
      "            PA2640  nuoE; NADH-quinone oxidoreductase subunit E [KO:K00334] [EC:7.1.1.2]\n",
      "line in gene PA2640  nuoE; NADH-quinone oxidoreductase subunit E [KO:K00334] [EC:7.1.1.2]\n",
      "            PA2641  nuoF; NADH dehydrogenase I subunit F [KO:K00335] [EC:7.1.1.2]\n",
      "line in gene PA2641  nuoF; NADH dehydrogenase I subunit F [KO:K00335] [EC:7.1.1.2]\n",
      "            PA2642  nuoG; NADH-quinone oxidoreductase subunit G [KO:K00336] [EC:7.1.1.2]\n",
      "line in gene PA2642  nuoG; NADH-quinone oxidoreductase subunit G [KO:K00336] [EC:7.1.1.2]\n",
      "            PA2643  nuoH; NADH-quinone oxidoreductase subunit H [KO:K00337] [EC:7.1.1.2]\n",
      "line in gene PA2643  nuoH; NADH-quinone oxidoreductase subunit H [KO:K00337] [EC:7.1.1.2]\n",
      "            PA2644  nuoI; NADH-quinone oxidoreductase subunit I [KO:K00338] [EC:7.1.1.2]\n",
      "line in gene PA2644  nuoI; NADH-quinone oxidoreductase subunit I [KO:K00338] [EC:7.1.1.2]\n",
      "            PA2645  nuoJ; NADH-quinone oxidoreductase subunit J [KO:K00339] [EC:7.1.1.2]\n",
      "line in gene PA2645  nuoJ; NADH-quinone oxidoreductase subunit J [KO:K00339] [EC:7.1.1.2]\n",
      "            PA2646  nuoK; NADH-quinone oxidoreductase subunit K [KO:K00340] [EC:7.1.1.2]\n",
      "line in gene PA2646  nuoK; NADH-quinone oxidoreductase subunit K [KO:K00340] [EC:7.1.1.2]\n",
      "            PA2647  nuoL; NADH-quinone oxidoreductase subunit L [KO:K00341] [EC:7.1.1.2]\n",
      "line in gene PA2647  nuoL; NADH-quinone oxidoreductase subunit L [KO:K00341] [EC:7.1.1.2]\n",
      "            PA2648  nuoM; NADH-quinone oxidoreductase subunit M [KO:K00342] [EC:7.1.1.2]\n",
      "line in gene PA2648  nuoM; NADH-quinone oxidoreductase subunit M [KO:K00342] [EC:7.1.1.2]\n",
      "            PA2649  nuoN; NADH-quinone oxidoreductase subunit N [KO:K00343] [EC:7.1.1.2]\n",
      "line in gene PA2649  nuoN; NADH-quinone oxidoreductase subunit N [KO:K00343] [EC:7.1.1.2]\n",
      "            PA2691  hypothetical protein [KO:K03885] [EC:7.1.1.2]\n",
      "line in gene PA2691  hypothetical protein [KO:K03885] [EC:7.1.1.2]\n",
      "            PA4538  ndh; NADH dehydrogenase [KO:K03885] [EC:7.1.1.2]\n",
      "line in gene PA4538  ndh; NADH dehydrogenase [KO:K03885] [EC:7.1.1.2]\n",
      "            PA1583  sdhA; succinate dehydrogenase flavoprotein subunit [KO:K00239] [EC:1.3.5.1 1.3.5.4]\n",
      "line in gene PA1583  sdhA; succinate dehydrogenase flavoprotein subunit [KO:K00239] [EC:1.3.5.1 1.3.5.4]\n",
      "            PA1584  sdhB; succinate dehydrogenase iron-sulfur subunit [KO:K00240] [EC:1.3.5.1 1.3.5.4]\n",
      "line in gene PA1584  sdhB; succinate dehydrogenase iron-sulfur subunit [KO:K00240] [EC:1.3.5.1 1.3.5.4]\n",
      "            PA1581  sdhC; succinate dehydrogenase subunit C [KO:K00241]\n",
      "line in gene PA1581  sdhC; succinate dehydrogenase subunit C [KO:K00241]\n",
      "            PA1582  sdhD; succinate dehydrogenase subunit D [KO:K00242]\n",
      "line in gene PA1582  sdhD; succinate dehydrogenase subunit D [KO:K00242]\n",
      "            PA4431  iron-sulfur protein [KO:K00411] [EC:7.1.1.8]\n",
      "line in gene PA4431  iron-sulfur protein [KO:K00411] [EC:7.1.1.8]\n",
      "            PA4430  cytochrome b [KO:K00412]\n",
      "line in gene PA4430  cytochrome b [KO:K00412]\n",
      "            PA4429  cytochrome C1 [KO:K00413]\n",
      "line in gene PA4429  cytochrome C1 [KO:K00413]\n",
      "            PA0113  protoheme IX farnesyltransferase [KO:K02257] [EC:2.5.1.141]\n",
      "line in gene PA0113  protoheme IX farnesyltransferase [KO:K02257] [EC:2.5.1.141]\n",
      "            PA1321  cyoE; protoheme IX farnesyltransferase [KO:K02257] [EC:2.5.1.141]\n",
      "line in gene PA1321  cyoE; protoheme IX farnesyltransferase [KO:K02257] [EC:2.5.1.141]\n",
      "            PA1320  cyoD; cytochrome o ubiquinol oxidase subunit IV [KO:K02300]\n",
      "line in gene PA1320  cyoD; cytochrome o ubiquinol oxidase subunit IV [KO:K02300]\n",
      "            PA0108  coIII; cytochrome C oxidase subunit III [KO:K02276] [EC:7.1.1.9]\n",
      "line in gene PA0108  coIII; cytochrome C oxidase subunit III [KO:K02276] [EC:7.1.1.9]\n",
      "            PA0106  coxA; cytochrome C oxidase subunit I [KO:K02274] [EC:7.1.1.9]\n",
      "line in gene PA0106  coxA; cytochrome C oxidase subunit I [KO:K02274] [EC:7.1.1.9]\n",
      "            PA0105  coxB; cytochrome C oxidase subunit II [KO:K02275] [EC:7.1.1.9]\n",
      "line in gene PA0105  coxB; cytochrome C oxidase subunit II [KO:K02275] [EC:7.1.1.9]\n",
      "            PA0107  cytochrome C oxidase assembly protein [KO:K02258]\n",
      "line in gene PA0107  cytochrome C oxidase assembly protein [KO:K02258]\n",
      "            PA0112  hypothetical protein [KO:K02259] [EC:1.17.99.9]\n",
      "line in gene PA0112  hypothetical protein [KO:K02259] [EC:1.17.99.9]\n",
      "            PA4133  cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "line in gene PA4133  cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "            PA1554  ccoN1; cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "line in gene PA1554  ccoN1; cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "            PA1557  ccoN2; cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "line in gene PA1557  ccoN2; cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "            PA1856  cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "line in gene PA1856  cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "            PA1553  ccoO1; cbb3-type cytochrome C oxidase subunit II [KO:K00405]\n",
      "line in gene PA1553  ccoO1; cbb3-type cytochrome C oxidase subunit II [KO:K00405]\n",
      "            PA1556  ccoO2; cbb3-type cytochrome C oxidase subunit II [KO:K00405]\n",
      "line in gene PA1556  ccoO2; cbb3-type cytochrome C oxidase subunit II [KO:K00405]\n",
      "            PA1552.1  ccoQ1; cytochrome C oxidase cbb3-type subunit CcoQ [KO:K00407]\n",
      "line in gene PA1552.1  ccoQ1; cytochrome C oxidase cbb3-type subunit CcoQ [KO:K00407]\n",
      "            PA1555.1  ccoQ2; cytochrome C oxidase cbb3-type subunit CcoQ [KO:K00407]\n",
      "line in gene PA1555.1  ccoQ2; cytochrome C oxidase cbb3-type subunit CcoQ [KO:K00407]\n",
      "            PA1552  ccoP1; cytochrome C oxidase cbb3-type subunit CcoP [KO:K00406]\n",
      "line in gene PA1552  ccoP1; cytochrome C oxidase cbb3-type subunit CcoP [KO:K00406]\n",
      "            PA1555  ccoP2; cytochrome C oxidase cbb3-type subunit CcoP [KO:K00406]\n",
      "line in gene PA1555  ccoP2; cytochrome C oxidase cbb3-type subunit CcoP [KO:K00406]\n",
      "            PA1319  cyoC; cytochrome o ubiquinol oxidase subunit III [KO:K02299]\n",
      "line in gene PA1319  cyoC; cytochrome o ubiquinol oxidase subunit III [KO:K02299]\n",
      "            PA1318  cyoB; cytochrome o ubiquinol oxidase subunit I [KO:K02298] [EC:7.1.1.3]\n",
      "line in gene PA1318  cyoB; cytochrome o ubiquinol oxidase subunit I [KO:K02298] [EC:7.1.1.3]\n",
      "            PA1317  cyoA; cytochrome o ubiquinol oxidase subunit II [KO:K02297] [EC:7.1.1.3]\n",
      "line in gene PA1317  cyoA; cytochrome o ubiquinol oxidase subunit II [KO:K02297] [EC:7.1.1.3]\n",
      "            PA3930  cioA; cyanide insensitive terminal oxidase [KO:K00425] [EC:7.1.1.7]\n",
      "line in gene PA3930  cioA; cyanide insensitive terminal oxidase [KO:K00425] [EC:7.1.1.7]\n",
      "            PA3929  cioB; cyanide insensitive terminal oxidase [KO:K00426] [EC:7.1.1.7]\n",
      "line in gene PA3929  cioB; cyanide insensitive terminal oxidase [KO:K00426] [EC:7.1.1.7]\n",
      "            PA0518  nirM; cytochrome C-551 [KO:K08738]\n",
      "line in gene PA0518  nirM; cytochrome C-551 [KO:K08738]\n",
      "            PA5556  atpA; ATP synthase subunit alpha [KO:K02111] [EC:7.1.2.2 7.2.2.1]\n",
      "line in gene PA5556  atpA; ATP synthase subunit alpha [KO:K02111] [EC:7.1.2.2 7.2.2.1]\n",
      "            PA5554  atpD; ATP synthase subunit beta [KO:K02112] [EC:7.1.2.2 7.2.2.1]\n",
      "line in gene PA5554  atpD; ATP synthase subunit beta [KO:K02112] [EC:7.1.2.2 7.2.2.1]\n",
      "            PA5555  atpG; ATP synthase subunit gamma [KO:K02115]\n",
      "line in gene PA5555  atpG; ATP synthase subunit gamma [KO:K02115]\n",
      "            PA5557  atpH; ATP synthase subunit delta [KO:K02113]\n",
      "line in gene PA5557  atpH; ATP synthase subunit delta [KO:K02113]\n",
      "            PA5553  atpC; ATP synthase subunit epsilon [KO:K02114]\n",
      "line in gene PA5553  atpC; ATP synthase subunit epsilon [KO:K02114]\n",
      "            PA5560  atpB; ATP synthase subunit A [KO:K02108]\n",
      "line in gene PA5560  atpB; ATP synthase subunit A [KO:K02108]\n",
      "            PA5558  atpF; ATP synthase subunit B [KO:K02109]\n",
      "line in gene PA5558  atpF; ATP synthase subunit B [KO:K02109]\n",
      "            PA5559  atpE; ATP synthase subunit C [KO:K02110]\n",
      "line in gene PA5559  atpE; ATP synthase subunit C [KO:K02110]\n",
      "            PA4031  ppa; inorganic pyrophosphatase [KO:K01507] [EC:3.6.1.1]\n",
      "line in gene PA4031  ppa; inorganic pyrophosphatase [KO:K01507] [EC:3.6.1.1]\n",
      "            PA5242  ppk; polyphosphate kinase [KO:K00937] [EC:2.7.4.1]\n",
      "line in gene PA5242  ppk; polyphosphate kinase [KO:K00937] [EC:2.7.4.1]\n",
      "            PA2428  hypothetical protein [KO:K22468] [EC:2.7.4.34]\n",
      "line in gene PA2428  hypothetical protein [KO:K22468] [EC:2.7.4.34]\n",
      "            PA0141  hypothetical protein [KO:K22468] [EC:2.7.4.34]\n",
      "line in gene PA0141  hypothetical protein [KO:K22468] [EC:2.7.4.34]\n",
      "COMPOUND    C00001  H2O\n",
      "            C00002  ATP\n",
      "            C00003  NAD+\n",
      "            C00004  NADH\n",
      "            C00007  Oxygen\n",
      "            C00008  ADP\n",
      "            C00009  Orthophosphate\n",
      "            C00013  Diphosphate\n",
      "            C00042  Succinate\n",
      "            C00061  FMN\n",
      "            C00080  H+\n",
      "            C00122  Fumarate\n",
      "            C00390  Ubiquinol\n",
      "            C00399  Ubiquinone\n",
      "            C00524  Cytochrome c\n",
      "            C00536  Triphosphate\n",
      "REFERENCE   PMID:16469879\n",
      "  AUTHORS   Sazanov LA, Hinchliffe P.\n",
      "  TITLE     Structure of the hydrophilic domain of respiratory complex I from Thermus thermophilus.\n",
      "  JOURNAL   Science 311:1430-6 (2006)\n",
      "            DOI:10.1126/science.1123809\n",
      "REFERENCE   PMID:16584177\n",
      "  AUTHORS   Hinchliffe P, Carroll J, Sazanov LA.\n",
      "  TITLE     Identification of a novel subunit of respiratory complex I from Thermus thermophilus.\n",
      "  JOURNAL   Biochemistry 45:4413-20 (2006)\n",
      "            DOI:10.1021/bi0600998\n",
      "KO_PATHWAY  ko00190\n",
      "///\n",
      "path:pae00220\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00220                    Pathway\n",
      "NAME        Arginine biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00220  Arginine biosynthesis\n",
      "MODULE      pae_M00028  Ornithine biosynthesis, glutamate => ornithine [PATH:pae00220]\n",
      "            pae_M00844  Arginine biosynthesis, ornithine => arginine [PATH:pae00220]\n",
      "DBLINKS     GO: 0006526\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3537  argF; ornithine carbamoyltransferase [KO:K00611] [EC:2.1.3.3]\n",
      "line in gene PA3537  argF; ornithine carbamoyltransferase [KO:K00611] [EC:2.1.3.3]\n",
      "            PA5172  arcB; ornithine carbamoyltransferase [KO:K00611] [EC:2.1.3.3]\n",
      "line in gene PA5172  arcB; ornithine carbamoyltransferase [KO:K00611] [EC:2.1.3.3]\n",
      "            PA3525  argG; argininosuccinate synthase [KO:K01940] [EC:6.3.4.5]\n",
      "line in gene PA3525  argG; argininosuccinate synthase [KO:K01940] [EC:6.3.4.5]\n",
      "            PA5263  argH; argininosuccinate lyase [KO:K01755] [EC:4.3.2.1]\n",
      "line in gene PA5263  argH; argininosuccinate lyase [KO:K01755] [EC:4.3.2.1]\n",
      "            PA0495  hypothetical protein [KO:K01941] [EC:6.3.4.6]\n",
      "line in gene PA0495  hypothetical protein [KO:K01941] [EC:6.3.4.6]\n",
      "            PA4868  ureC; urease subunit alpha [KO:K01428] [EC:3.5.1.5]\n",
      "line in gene PA4868  ureC; urease subunit alpha [KO:K01428] [EC:3.5.1.5]\n",
      "            PA4867  ureB; urease subunit beta [KO:K01429] [EC:3.5.1.5]\n",
      "line in gene PA4867  ureB; urease subunit beta [KO:K01429] [EC:3.5.1.5]\n",
      "            PA4865  ureA; urease subunit gamma [KO:K01430] [EC:3.5.1.5]\n",
      "line in gene PA4865  ureA; urease subunit gamma [KO:K01430] [EC:3.5.1.5]\n",
      "            PA5171  arcA; arginine deiminase [KO:K01478] [EC:3.5.3.6]\n",
      "line in gene PA5171  arcA; arginine deiminase [KO:K01478] [EC:3.5.3.6]\n",
      "            PA1638  glutaminase [KO:K01425] [EC:3.5.1.2]\n",
      "line in gene PA1638  glutaminase [KO:K01425] [EC:3.5.1.2]\n",
      "            PA1337  ansB; glutaminase-asparaginase [KO:K05597] [EC:3.5.1.38]\n",
      "line in gene PA1337  ansB; glutaminase-asparaginase [KO:K05597] [EC:3.5.1.38]\n",
      "            PA0296  spuI; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA0296  spuI; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA0298  spuB; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA0298  spuB; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA2040  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA2040  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5119  glnA; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5119  glnA; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5522  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5522  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA1566  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA1566  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "            PA3356  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA3356  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5508  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5508  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA3068  gdhB; NAD-specific glutamate dehydrogenase [KO:K15371] [EC:1.4.1.2]\n",
      "line in gene PA3068  gdhB; NAD-specific glutamate dehydrogenase [KO:K15371] [EC:1.4.1.2]\n",
      "            PA4588  gdhA; glutamate dehydrogenase [KO:K00262] [EC:1.4.1.4]\n",
      "line in gene PA4588  gdhA; glutamate dehydrogenase [KO:K00262] [EC:1.4.1.4]\n",
      "            PA5173  arcC; carbamate kinase [KO:K00926] [EC:2.7.2.2]\n",
      "line in gene PA5173  arcC; carbamate kinase [KO:K00926] [EC:2.7.2.2]\n",
      "            PA2828  aminotransferase [KO:K14260] [EC:2.6.1.66 2.6.1.2]\n",
      "line in gene PA2828  aminotransferase [KO:K14260] [EC:2.6.1.66 2.6.1.2]\n",
      "            PA5204  argA; N-acetylglutamate synthase [KO:K14682] [EC:2.3.1.1]\n",
      "line in gene PA5204  argA; N-acetylglutamate synthase [KO:K14682] [EC:2.3.1.1]\n",
      "            PA4402  argJ; bifunctional ornithine acetyltransferase/N-acetylglutamate synthase [KO:K00620] [EC:2.3.1.35 2.3.1.1]\n",
      "line in gene PA4402  argJ; bifunctional ornithine acetyltransferase/N-acetylglutamate synthase [KO:K00620] [EC:2.3.1.35 2.3.1.1]\n",
      "            PA4534  hypothetical protein [KO:K22477] [EC:2.3.1.1]\n",
      "line in gene PA4534  hypothetical protein [KO:K22477] [EC:2.3.1.1]\n",
      "            PA5323  argB; acetylglutamate kinase [KO:K00930] [EC:2.7.2.8]\n",
      "line in gene PA5323  argB; acetylglutamate kinase [KO:K00930] [EC:2.7.2.8]\n",
      "            PA0662  argC; N-acetyl-gamma-glutamyl-phosphate reductase [KO:K00145] [EC:1.2.1.38]\n",
      "line in gene PA0662  argC; N-acetyl-gamma-glutamyl-phosphate reductase [KO:K00145] [EC:1.2.1.38]\n",
      "            PA0530  class III pyridoxal phosphate-dependent aminotransferase [KO:K00821] [EC:2.6.1.11 2.6.1.17]\n",
      "line in gene PA0530  class III pyridoxal phosphate-dependent aminotransferase [KO:K00821] [EC:2.6.1.11 2.6.1.17]\n",
      "            PA5206  argE; acetylornithine deacetylase [KO:K01438] [EC:3.5.1.16]\n",
      "line in gene PA5206  argE; acetylornithine deacetylase [KO:K01438] [EC:3.5.1.16]\n",
      "            PA5390  acetylornithine deacetylase [KO:K01438] [EC:3.5.1.16]\n",
      "line in gene PA5390  acetylornithine deacetylase [KO:K01438] [EC:3.5.1.16]\n",
      "COMPOUND    C00011  CO2\n",
      "            C00014  Ammonia\n",
      "            C00025  L-Glutamate\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00049  L-Aspartate\n",
      "            C00062  L-Arginine\n",
      "            C00064  L-Glutamine\n",
      "            C00077  L-Ornithine\n",
      "            C00086  Urea\n",
      "            C00122  Fumarate\n",
      "            C00169  Carbamoyl phosphate\n",
      "            C00327  L-Citrulline\n",
      "            C00437  N-Acetylornithine\n",
      "            C00624  N-Acetyl-L-glutamate\n",
      "            C01010  Urea-1-carboxylate\n",
      "            C01250  N-Acetyl-L-glutamate 5-semialdehyde\n",
      "            C03406  N-(L-Arginino)succinate\n",
      "            C04133  N-Acetyl-L-glutamate 5-phosphate\n",
      "            C15532  N-Acetyl-L-citrulline\n",
      "            C20948  LysW-L-glutamate\n",
      "            C20949  LysW-L-glutamyl 5-phosphate\n",
      "            C20950  LysW-L-glutamate 5-semialdehyde\n",
      "            C20951  LysW-L-ornithine\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00240  Pyrimidine metabolism\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00330  Arginine and proline metabolism\n",
      "            pae00470  D-Amino acid metabolism\n",
      "            pae00910  Nitrogen metabolism\n",
      "KO_PATHWAY  ko00220\n",
      "///\n",
      "path:pae00230\n",
      "ENTRY       pae00230                    Pathway\n",
      "NAME        Purine metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Nucleotide metabolism\n",
      "PATHWAY_MAP pae00230  Purine metabolism\n",
      "MODULE      pae_M00005  PRPP biosynthesis, ribose 5P => PRPP [PATH:pae00230]\n",
      "            pae_M00048  Inosine monophosphate biosynthesis, PRPP + glutamine => IMP [PATH:pae00230]\n",
      "            pae_M00049  Adenine ribonucleotide biosynthesis, IMP => ADP,ATP [PATH:pae00230]\n",
      "            pae_M00050  Guanine ribonucleotide biosynthesis, IMP => GDP,GTP [PATH:pae00230]\n",
      "DBLINKS     GO: 0006163 0042278\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4971  aspP; adenosine diphosphate sugar pyrophosphatase [KO:K01515] [EC:3.6.1.13 3.6.1.-]\n",
      "line in gene PA4971  aspP; adenosine diphosphate sugar pyrophosphatase [KO:K01515] [EC:3.6.1.13 3.6.1.-]\n",
      "            PA5176  ADP-ribose diphosphatase NudE [KO:K08312] [EC:3.6.1.-]\n",
      "line in gene PA5176  ADP-ribose diphosphatase NudE [KO:K08312] [EC:3.6.1.-]\n",
      "            PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "line in gene PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "            PA4670  prs; ribose-phosphate pyrophosphokinase [KO:K00948] [EC:2.7.6.1]\n",
      "line in gene PA4670  prs; ribose-phosphate pyrophosphokinase [KO:K00948] [EC:2.7.6.1]\n",
      "            PA3108  purF; amidophosphoribosyltransferase [KO:K00764] [EC:2.4.2.14]\n",
      "line in gene PA3108  purF; amidophosphoribosyltransferase [KO:K00764] [EC:2.4.2.14]\n",
      "            PA4855  purD; phosphoribosylamine--glycine ligase [KO:K01945] [EC:6.3.4.13]\n",
      "line in gene PA4855  purD; phosphoribosylamine--glycine ligase [KO:K01945] [EC:6.3.4.13]\n",
      "            PA0944  purN; phosphoribosylglycinamide formyltransferase [KO:K11175] [EC:2.1.2.2]\n",
      "line in gene PA0944  purN; phosphoribosylglycinamide formyltransferase [KO:K11175] [EC:2.1.2.2]\n",
      "            PA3751  purT; phosphoribosylglycinamide formyltransferase [KO:K08289] [EC:6.3.1.21]\n",
      "line in gene PA3751  purT; phosphoribosylglycinamide formyltransferase [KO:K08289] [EC:6.3.1.21]\n",
      "            PA3763  purL; phosphoribosylformylglycinamidine synthase [KO:K01952] [EC:6.3.5.3]\n",
      "line in gene PA3763  purL; phosphoribosylformylglycinamidine synthase [KO:K01952] [EC:6.3.5.3]\n",
      "            PA0945  purM; phosphoribosylformylglycinamidine cyclo-ligase [KO:K01933] [EC:6.3.3.1]\n",
      "line in gene PA0945  purM; phosphoribosylformylglycinamidine cyclo-ligase [KO:K01933] [EC:6.3.3.1]\n",
      "            PA5425  purK; 5-(carboxyamino)imidazole ribonucleotide synthase [KO:K01589] [EC:6.3.4.18]\n",
      "line in gene PA5425  purK; 5-(carboxyamino)imidazole ribonucleotide synthase [KO:K01589] [EC:6.3.4.18]\n",
      "            PA5426  purE; 5-(carboxyamino)imidazole ribonucleotide mutase [KO:K01588] [EC:5.4.99.18]\n",
      "line in gene PA5426  purE; 5-(carboxyamino)imidazole ribonucleotide mutase [KO:K01588] [EC:5.4.99.18]\n",
      "            PA1013  purC; phosphoribosylaminoimidazole-succinocarboxamide synthase [KO:K01923] [EC:6.3.2.6]\n",
      "line in gene PA1013  purC; phosphoribosylaminoimidazole-succinocarboxamide synthase [KO:K01923] [EC:6.3.2.6]\n",
      "            PA3516  adenylosuccinate lyase [KO:K01756] [EC:4.3.2.2]\n",
      "line in gene PA3516  adenylosuccinate lyase [KO:K01756] [EC:4.3.2.2]\n",
      "            PA3517  adenylosuccinate lyase [KO:K01756] [EC:4.3.2.2]\n",
      "line in gene PA3517  adenylosuccinate lyase [KO:K01756] [EC:4.3.2.2]\n",
      "            PA2629  purB; adenylosuccinate lyase [KO:K01756] [EC:4.3.2.2]\n",
      "line in gene PA2629  purB; adenylosuccinate lyase [KO:K01756] [EC:4.3.2.2]\n",
      "            PA4854  purH; bifunctional phosphoribosylaminoimidazolecarboxamide formyltransferase/IMP cyclohydrolase [KO:K00602] [EC:2.1.2.3 3.5.4.10]\n",
      "line in gene PA4854  purH; bifunctional phosphoribosylaminoimidazolecarboxamide formyltransferase/IMP cyclohydrolase [KO:K00602] [EC:2.1.2.3 3.5.4.10]\n",
      "            PA1543  apt; adenine phosphoribosyltransferase [KO:K00759] [EC:2.4.2.7]\n",
      "line in gene PA1543  apt; adenine phosphoribosyltransferase [KO:K00759] [EC:2.4.2.7]\n",
      "            PA3625  surE; 5'-nucleotidase SurE [KO:K03787] [EC:3.1.3.5 3.1.3.6]\n",
      "line in gene PA3625  surE; 5'-nucleotidase SurE [KO:K03787] [EC:3.1.3.5 3.1.3.6]\n",
      "            PA5177  hydrolase [KO:K20881] [EC:3.1.3.-]\n",
      "line in gene PA5177  hydrolase [KO:K20881] [EC:3.1.3.-]\n",
      "            PA1574  hypothetical protein [KO:K09913] [EC:2.4.2.1 2.4.2.2]\n",
      "line in gene PA1574  hypothetical protein [KO:K09913] [EC:2.4.2.1 2.4.2.2]\n",
      "            PA1469  hypothetical protein [KO:K05810] [EC:2.4.2.1 2.4.2.28 3.5.4.4]\n",
      "line in gene PA1469  hypothetical protein [KO:K05810] [EC:2.4.2.1 2.4.2.28 3.5.4.4]\n",
      "            PA4543  hypothetical protein [KO:K05810] [EC:2.4.2.1 2.4.2.28 3.5.4.4]\n",
      "line in gene PA4543  hypothetical protein [KO:K05810] [EC:2.4.2.1 2.4.2.28 3.5.4.4]\n",
      "            PA4645  hypoxanthine-guanine phosphoribosyltransferase [KO:K00760] [EC:2.4.2.8]\n",
      "line in gene PA4645  hypoxanthine-guanine phosphoribosyltransferase [KO:K00760] [EC:2.4.2.8]\n",
      "            PA3770  guaB; inosine 5'-monophosphate dehydrogenase [KO:K00088] [EC:1.1.1.205]\n",
      "line in gene PA3770  guaB; inosine 5'-monophosphate dehydrogenase [KO:K00088] [EC:1.1.1.205]\n",
      "            PA3807  ndk; nucleoside diphosphate kinase [KO:K00940] [EC:2.7.4.6]\n",
      "line in gene PA3807  ndk; nucleoside diphosphate kinase [KO:K00940] [EC:2.7.4.6]\n",
      "            PA0387  non-canonical purine NTP pyrophosphatase [KO:K01519] [EC:3.6.1.66]\n",
      "line in gene PA0387  non-canonical purine NTP pyrophosphatase [KO:K01519] [EC:3.6.1.66]\n",
      "            PA5298  xanthine phosphoribosyltransferase [KO:K03816] [EC:2.4.2.22]\n",
      "line in gene PA5298  xanthine phosphoribosyltransferase [KO:K03816] [EC:2.4.2.22]\n",
      "            PA1524  xdhA; xanthine dehydrogenase [KO:K13481] [EC:1.17.1.4]\n",
      "line in gene PA1524  xdhA; xanthine dehydrogenase [KO:K13481] [EC:1.17.1.4]\n",
      "            PA1523  xdhB; xanthine dehydrogenase [KO:K13482] [EC:1.17.1.4]\n",
      "line in gene PA1523  xdhB; xanthine dehydrogenase [KO:K13482] [EC:1.17.1.4]\n",
      "            PA1933  hydroxylase large subunit [KO:K11177] [EC:1.17.1.4]\n",
      "line in gene PA1933  hydroxylase large subunit [KO:K11177] [EC:1.17.1.4]\n",
      "            PA1932  hydroxylase molybdopterin-containing subunit [KO:K11178] [EC:1.17.1.4]\n",
      "line in gene PA1932  hydroxylase molybdopterin-containing subunit [KO:K11178] [EC:1.17.1.4]\n",
      "            PA1931  ferredoxin [KO:K13483]\n",
      "line in gene PA1931  ferredoxin [KO:K13483]\n",
      "            PA3769  guaA; GMP synthase [KO:K01951] [EC:6.3.5.2]\n",
      "line in gene PA3769  guaA; GMP synthase [KO:K01951] [EC:6.3.5.2]\n",
      "            PA1521  guanine deaminase [KO:K01487] [EC:3.5.4.3]\n",
      "line in gene PA1521  guanine deaminase [KO:K01487] [EC:3.5.4.3]\n",
      "            PA0134  guanine deaminase [KO:K01487] [EC:3.5.4.3]\n",
      "line in gene PA0134  guanine deaminase [KO:K01487] [EC:3.5.4.3]\n",
      "            PA2499  deaminase [KO:K01487] [EC:3.5.4.3]\n",
      "line in gene PA2499  deaminase [KO:K01487] [EC:3.5.4.3]\n",
      "            PA5336  gmk; guanylate kinase [KO:K00942] [EC:2.7.4.8]\n",
      "line in gene PA5336  gmk; guanylate kinase [KO:K00942] [EC:2.7.4.8]\n",
      "            PA5497  nrdJa; ribonucleoside-diphosphate reductase [KO:K00525] [EC:1.17.4.1]\n",
      "line in gene PA5497  nrdJa; ribonucleoside-diphosphate reductase [KO:K00525] [EC:1.17.4.1]\n",
      "            PA1156  nrdA; ribonucleotide-diphosphate reductase subunit alpha [KO:K00525] [EC:1.17.4.1]\n",
      "line in gene PA1156  nrdA; ribonucleotide-diphosphate reductase subunit alpha [KO:K00525] [EC:1.17.4.1]\n",
      "            PA1155  nrdB; ribonucleotide-diphosphate reductase subunit beta [KO:K00526] [EC:1.17.4.1]\n",
      "line in gene PA1155  nrdB; ribonucleotide-diphosphate reductase subunit beta [KO:K00526] [EC:1.17.4.1]\n",
      "            PA1920  nrdD; anaerobic ribonucleoside triphosphate reductase [KO:K21636] [EC:1.1.98.6]\n",
      "line in gene PA1920  nrdD; anaerobic ribonucleoside triphosphate reductase [KO:K21636] [EC:1.1.98.6]\n",
      "            PA1124  dgt; deoxyguanosinetriphosphate triphosphohydrolase [KO:K01129] [EC:3.1.5.1]\n",
      "line in gene PA1124  dgt; deoxyguanosinetriphosphate triphosphohydrolase [KO:K01129] [EC:3.1.5.1]\n",
      "            PA3043  deoxyguanosinetriphosphate triphosphohydrolase [KO:K01129] [EC:3.1.5.1]\n",
      "line in gene PA3043  deoxyguanosinetriphosphate triphosphohydrolase [KO:K01129] [EC:3.1.5.1]\n",
      "            PA5338  spoT; guanosine-3',5'-bis(diphosphate) 3'-pyrophosphohydrolase [KO:K01139] [EC:2.7.6.5 3.1.7.2]\n",
      "line in gene PA5338  spoT; guanosine-3',5'-bis(diphosphate) 3'-pyrophosphohydrolase [KO:K01139] [EC:2.7.6.5 3.1.7.2]\n",
      "            PA5241  ppx; exopolyphosphatase [KO:K01524] [EC:3.6.1.11 3.6.1.40]\n",
      "line in gene PA5241  ppx; exopolyphosphatase [KO:K01524] [EC:3.6.1.11 3.6.1.40]\n",
      "            PA0934  relA; GTP pyrophosphokinase [KO:K00951] [EC:2.7.6.5]\n",
      "line in gene PA0934  relA; GTP pyrophosphokinase [KO:K00951] [EC:2.7.6.5]\n",
      "            PA3217  cyaB; protein CyaB [KO:K01768] [EC:4.6.1.1]\n",
      "line in gene PA3217  cyaB; protein CyaB [KO:K01768] [EC:4.6.1.1]\n",
      "            PA5272  cyaA; adenylate cyclase [KO:K05851] [EC:4.6.1.1]\n",
      "line in gene PA5272  cyaA; adenylate cyclase [KO:K05851] [EC:4.6.1.1]\n",
      "            PA4938  purA; adenylosuccinate synthetase [KO:K01939] [EC:6.3.4.4]\n",
      "line in gene PA4938  purA; adenylosuccinate synthetase [KO:K01939] [EC:6.3.4.4]\n",
      "            PA3970  amn; AMP nucleosidase [KO:K01241] [EC:3.2.2.4]\n",
      "line in gene PA3970  amn; AMP nucleosidase [KO:K01241] [EC:3.2.2.4]\n",
      "            PA4115  hypothetical protein [KO:K06966] [EC:3.2.2.10 3.2.2.-]\n",
      "line in gene PA4115  hypothetical protein [KO:K06966] [EC:3.2.2.10 3.2.2.-]\n",
      "            PA0148  adenosine deaminase [KO:K21053] [EC:3.5.4.2]\n",
      "line in gene PA0148  adenosine deaminase [KO:K21053] [EC:3.5.4.2]\n",
      "            PA3686  adk; adenylate kinase [KO:K00939] [EC:2.7.4.3]\n",
      "line in gene PA3686  adk; adenylate kinase [KO:K00939] [EC:2.7.4.3]\n",
      "            PA4969  cpdA; cAMP phosphodiesterase [KO:K03651] [EC:3.1.4.53]\n",
      "line in gene PA4969  cpdA; cAMP phosphodiesterase [KO:K03651] [EC:3.1.4.53]\n",
      "            PA0590  apaH; bis(5'-nucleosyl)-tetraphosphatase [KO:K01525] [EC:3.6.1.41]\n",
      "line in gene PA0590  apaH; bis(5'-nucleosyl)-tetraphosphatase [KO:K01525] [EC:3.6.1.41]\n",
      "            PA4442  cysN; bifunctional sulfate adenylyltransferase subunit 1/adenylylsulfate kinase [KO:K00955] [EC:2.7.7.4 2.7.1.25]\n",
      "line in gene PA4442  cysN; bifunctional sulfate adenylyltransferase subunit 1/adenylylsulfate kinase [KO:K00955] [EC:2.7.7.4 2.7.1.25]\n",
      "            PA1393  cysC; adenylyl-sulfate kinase [KO:K00860] [EC:2.7.1.25]\n",
      "line in gene PA1393  cysC; adenylyl-sulfate kinase [KO:K00860] [EC:2.7.1.25]\n",
      "            PA0935  nucleoside triphosphate pyrophosphohydrolase [KO:K04765] [EC:3.6.1.9]\n",
      "line in gene PA0935  nucleoside triphosphate pyrophosphohydrolase [KO:K04765] [EC:3.6.1.9]\n",
      "            PA4443  cysD; sulfate adenylyltransferase subunit 2 [KO:K00957] [EC:2.7.7.4]\n",
      "line in gene PA4443  cysD; sulfate adenylyltransferase subunit 2 [KO:K00957] [EC:2.7.7.4]\n",
      "            PA1518  5-hydroxyisourate hydrolase [KO:K07127] [EC:3.5.2.17]\n",
      "line in gene PA1518  5-hydroxyisourate hydrolase [KO:K07127] [EC:3.5.2.17]\n",
      "            PA1516  hypothetical protein [KO:K13485] [EC:4.1.1.97]\n",
      "line in gene PA1516  hypothetical protein [KO:K13485] [EC:4.1.1.97]\n",
      "            PA1515  alc; allantoicase [KO:K01477] [EC:3.5.3.4]\n",
      "line in gene PA1515  alc; allantoicase [KO:K01477] [EC:3.5.3.4]\n",
      "            PA1140  hypothetical protein [KO:K14977] [EC:3.5.3.26]\n",
      "line in gene PA1140  hypothetical protein [KO:K14977] [EC:3.5.3.26]\n",
      "            PA1514  ureidoglycolate hydrolase [KO:K01483] [EC:4.3.2.3]\n",
      "line in gene PA1514  ureidoglycolate hydrolase [KO:K01483] [EC:4.3.2.3]\n",
      "            PA4868  ureC; urease subunit alpha [KO:K01428] [EC:3.5.1.5]\n",
      "line in gene PA4868  ureC; urease subunit alpha [KO:K01428] [EC:3.5.1.5]\n",
      "            PA4867  ureB; urease subunit beta [KO:K01429] [EC:3.5.1.5]\n",
      "line in gene PA4867  ureB; urease subunit beta [KO:K01429] [EC:3.5.1.5]\n",
      "            PA4865  ureA; urease subunit gamma [KO:K01430] [EC:3.5.1.5]\n",
      "line in gene PA4865  ureA; urease subunit gamma [KO:K01430] [EC:3.5.1.5]\n",
      "            PA5173  arcC; carbamate kinase [KO:K00926] [EC:2.7.2.2]\n",
      "line in gene PA5173  arcC; carbamate kinase [KO:K00926] [EC:2.7.2.2]\n",
      "COMPOUND    C00002  ATP\n",
      "            C00008  ADP\n",
      "            C00011  CO2\n",
      "            C00014  Ammonia\n",
      "            C00020  AMP\n",
      "            C00035  GDP\n",
      "            C00037  Glycine\n",
      "            C00039  DNA\n",
      "            C00044  GTP\n",
      "            C00046  RNA\n",
      "            C00048  Glyoxylate\n",
      "            C00053  3'-Phosphoadenylyl sulfate\n",
      "            C00054  Adenosine 3',5'-bisphosphate\n",
      "            C00059  Sulfate\n",
      "            C00064  L-Glutamine\n",
      "            C00081  ITP\n",
      "            C00086  Urea\n",
      "            C00104  IDP\n",
      "            C00117  D-Ribose 5-phosphate\n",
      "            C00119  5-Phospho-alpha-D-ribose 1-diphosphate\n",
      "            C00130  IMP\n",
      "            C00131  dATP\n",
      "            C00144  GMP\n",
      "            C00147  Adenine\n",
      "            C00169  Carbamoyl phosphate\n",
      "            C00206  dADP\n",
      "            C00209  Oxalate\n",
      "            C00212  Adenosine\n",
      "            C00224  Adenylyl sulfate\n",
      "            C00242  Guanine\n",
      "            C00262  Hypoxanthine\n",
      "            C00286  dGTP\n",
      "            C00288  HCO3-\n",
      "            C00294  Inosine\n",
      "            C00301  ADP-ribose\n",
      "            C00330  Deoxyguanosine\n",
      "            C00360  dAMP\n",
      "            C00361  dGDP\n",
      "            C00362  dGMP\n",
      "            C00366  Urate\n",
      "            C00385  Xanthine\n",
      "            C00387  Guanosine\n",
      "            C00499  Allantoate\n",
      "            C00559  Deoxyadenosine\n",
      "            C00575  3',5'-Cyclic AMP\n",
      "            C00603  (S)-Ureidoglycolate\n",
      "            C00620  alpha-D-Ribose 1-phosphate\n",
      "            C00655  Xanthosine 5'-phosphate\n",
      "            C00700  XTP\n",
      "            C00802  Oxalureate\n",
      "            C00942  3',5'-Cyclic GMP\n",
      "            C01228  Guanosine 3',5'-bis(diphosphate)\n",
      "            C01260  P1,P4-Bis(5'-adenosyl)tetraphosphate\n",
      "            C01261  P1,P4-Bis(5'-guanosyl) tetraphosphate\n",
      "            C01344  dIDP\n",
      "            C01345  dITP\n",
      "            C01367  3'-AMP\n",
      "            C01444  Oxamate\n",
      "            C01762  Xanthosine\n",
      "            C02091  (S)-Ureidoglycine\n",
      "            C02348  (R)(-)-Allantoin\n",
      "            C02350  (S)-Allantoin\n",
      "            C02353  2',3'-Cyclic AMP\n",
      "            C02718  N-Formiminoglycine\n",
      "            C03090  5-Phosphoribosylamine\n",
      "            C03373  Aminoimidazole ribotide\n",
      "            C03483  Adenosine tetraphosphate\n",
      "            C03614  Inosine 5'-tetraphosphate\n",
      "            C03794  N6-(1,2-Dicarboxyethyl)-AMP\n",
      "            C03838  5'-Phosphoribosylglycinamide\n",
      "            C04051  5-Amino-4-imidazolecarboxyamide\n",
      "            C04376  5'-Phosphoribosyl-N-formylglycinamide\n",
      "            C04392  P1,P4-Bis(5'-xanthosyl) tetraphosphate\n",
      "            C04494  Guanosine 3'-diphosphate 5'-triphosphate\n",
      "            C04640  2-(Formamido)-N1-(5'-phosphoribosyl)acetamidine\n",
      "            C04677  1-(5'-Phosphoribosyl)-5-amino-4-imidazolecarboxamide\n",
      "            C04734  1-(5'-Phosphoribosyl)-5-formamido-4-imidazolecarboxamide\n",
      "            C04751  1-(5-Phospho-D-ribosyl)-5-amino-4-imidazolecarboxylate\n",
      "            C04823  1-(5'-Phosphoribosyl)-5-amino-4-(N-succinocarboxamide)-imidazole\n",
      "            C05239  5-Aminoimidazole\n",
      "            C05512  Deoxyinosine\n",
      "            C05513  Urate-3-ribonucleoside\n",
      "            C05515  5-Ureido-4-imidazole carboxylate\n",
      "            C05516  5-Amino-4-imidazole carboxylate\n",
      "            C05993  Acetyl adenylate\n",
      "            C06193  Guanosine 3'-phosphate\n",
      "            C06194  2',3'-Cyclic GMP\n",
      "            C06195  Imidazolone\n",
      "            C06196  2'-Deoxyinosine 5'-phosphate\n",
      "            C06197  P1,P3-Bis(5'-adenosyl) triphosphate\n",
      "            C06433  5'-Benzoylphosphoadenosine\n",
      "            C06435  5'-Butyrylphosphoinosine\n",
      "            C11821  5-Hydroxyisourate\n",
      "            C12248  5-Hydroxy-2-oxo-4-ureido-2,5-dihydro-1H-imidazole-5-carboxylate\n",
      "            C15667  5-Carboxyamino-1-(5-phospho-D-ribosyl)imidazole\n",
      "            C22395  N6-Succino-2-amino-2'-deoxyadenylate\n",
      "            C22441  dZMP\n",
      "            C22442  dZDP\n",
      "            C22443  dZTP\n",
      "REFERENCE   PMID:10601204\n",
      "  AUTHORS   Cusa E, Obradors N, Baldoma L, Badia J, Aguilar J.\n",
      "  TITLE     Genetic analysis of a chromosomal region containing genes required for assimilation of allantoin nitrogen and linked glyoxylate metabolism in Escherichia coli.\n",
      "  JOURNAL   J Bacteriol 181:7479-84 (1999)\n",
      "            DOI:10.1128/JB.181.24.7479-7484.1999\n",
      "REFERENCE   PMID:10986234\n",
      "  AUTHORS   Xi H, Schneider BL, Reitzer L.\n",
      "  TITLE     Purine catabolism in Escherichia coli and function of xanthine dehydrogenase in purine salvage.\n",
      "  JOURNAL   J Bacteriol 182:5332-41 (2000)\n",
      "            DOI:10.1128/JB.182.19.5332-5341.2000\n",
      "REL_PATHWAY pae00030  Pentose phosphate pathway\n",
      "            pae00220  Arginine biosynthesis\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00340  Histidine metabolism\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00730  Thiamine metabolism\n",
      "            pae00740  Riboflavin metabolism\n",
      "            pae00790  Folate biosynthesis\n",
      "KO_PATHWAY  ko00230\n",
      "///\n",
      "path:pae00240\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00240                    Pathway\n",
      "NAME        Pyrimidine metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Nucleotide metabolism\n",
      "PATHWAY_MAP pae00240  Pyrimidine metabolism\n",
      "MODULE      pae_M00046  Pyrimidine degradation, uracil => beta-alanine, thymine => 3-aminoisobutanoate [PATH:pae00240]\n",
      "            pae_M00052  Pyrimidine ribonucleotide biosynthesis, UMP => UDP/UTP,CDP/CTP [PATH:pae00240]\n",
      "            pae_M00053  Pyrimidine deoxyribonuleotide biosynthesis, CDP => dCTP [PATH:pae00240]\n",
      "            pae_M00938  Pyrimidine deoxyribonuleotide biosynthesis, UDP => dTTP [PATH:pae00240]\n",
      "DBLINKS     GO: 0006220 0006213\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4756  carB; carbamoyl phosphate synthase large subunit [KO:K01955] [EC:6.3.5.5]\n",
      "line in gene PA4756  carB; carbamoyl phosphate synthase large subunit [KO:K01955] [EC:6.3.5.5]\n",
      "            PA4758  carA; carbamoyl phosphate synthase small subunit [KO:K01956] [EC:6.3.5.5]\n",
      "line in gene PA4758  carA; carbamoyl phosphate synthase small subunit [KO:K01956] [EC:6.3.5.5]\n",
      "            PA0402  pyrB; aspartate carbamoyltransferase [KO:K00609] [EC:2.1.3.2]\n",
      "line in gene PA0402  pyrB; aspartate carbamoyltransferase [KO:K00609] [EC:2.1.3.2]\n",
      "            PA0401  dihydroorotase [KO:K01465] [EC:3.5.2.3]\n",
      "line in gene PA0401  dihydroorotase [KO:K01465] [EC:3.5.2.3]\n",
      "            PA3527  pyrC; dihydroorotase [KO:K01465] [EC:3.5.2.3]\n",
      "line in gene PA3527  pyrC; dihydroorotase [KO:K01465] [EC:3.5.2.3]\n",
      "            PA5541  pyrQ; dihydroorotase [KO:K01465] [EC:3.5.2.3]\n",
      "line in gene PA5541  pyrQ; dihydroorotase [KO:K01465] [EC:3.5.2.3]\n",
      "            PA3050  pyrD; dihydroorotate dehydrogenase [KO:K00254] [EC:1.3.5.2]\n",
      "line in gene PA3050  pyrD; dihydroorotate dehydrogenase [KO:K00254] [EC:1.3.5.2]\n",
      "            PA5331  pyrE; orotate phosphoribosyltransferase [KO:K00762] [EC:2.4.2.10]\n",
      "line in gene PA5331  pyrE; orotate phosphoribosyltransferase [KO:K00762] [EC:2.4.2.10]\n",
      "            PA2876  pyrF; orotidine 5'-phosphate decarboxylase [KO:K01591] [EC:4.1.1.23]\n",
      "line in gene PA2876  pyrF; orotidine 5'-phosphate decarboxylase [KO:K01591] [EC:4.1.1.23]\n",
      "            PA3654  pyrH; uridylate kinase [KO:K09903] [EC:2.7.4.22]\n",
      "line in gene PA3654  pyrH; uridylate kinase [KO:K09903] [EC:2.7.4.22]\n",
      "            PA3807  ndk; nucleoside diphosphate kinase [KO:K00940] [EC:2.7.4.6]\n",
      "line in gene PA3807  ndk; nucleoside diphosphate kinase [KO:K00940] [EC:2.7.4.6]\n",
      "            PA0935  nucleoside triphosphate pyrophosphohydrolase [KO:K04765] [EC:3.6.1.9]\n",
      "line in gene PA0935  nucleoside triphosphate pyrophosphohydrolase [KO:K04765] [EC:3.6.1.9]\n",
      "            PA4478  hypothetical protein [KO:K06287] [EC:3.6.1.-]\n",
      "line in gene PA4478  hypothetical protein [KO:K06287] [EC:3.6.1.-]\n",
      "            PA0560  hypothetical protein [KO:K07043] [EC:3.6.1.-]\n",
      "line in gene PA0560  hypothetical protein [KO:K07043] [EC:3.6.1.-]\n",
      "            PA3637  pyrG; CTP synthetase [KO:K01937] [EC:6.3.4.2]\n",
      "line in gene PA3637  pyrG; CTP synthetase [KO:K01937] [EC:6.3.4.2]\n",
      "            PA3480  deoxycytidine triphosphate deaminase [KO:K01494] [EC:3.5.4.13]\n",
      "line in gene PA3480  deoxycytidine triphosphate deaminase [KO:K01494] [EC:3.5.4.13]\n",
      "            PA3163  cmk; cytidylate kinase [KO:K00945] [EC:2.7.4.25]\n",
      "line in gene PA3163  cmk; cytidylate kinase [KO:K00945] [EC:2.7.4.25]\n",
      "            PA3625  surE; 5'-nucleotidase SurE [KO:K03787] [EC:3.1.3.5 3.1.3.6]\n",
      "line in gene PA3625  surE; 5'-nucleotidase SurE [KO:K03787] [EC:3.1.3.5 3.1.3.6]\n",
      "            PA1574  hypothetical protein [KO:K09913] [EC:2.4.2.1 2.4.2.2]\n",
      "line in gene PA1574  hypothetical protein [KO:K09913] [EC:2.4.2.1 2.4.2.2]\n",
      "            PA4646  upp; uracil phosphoribosyltransferase [KO:K00761] [EC:2.4.2.9]\n",
      "line in gene PA4646  upp; uracil phosphoribosyltransferase [KO:K00761] [EC:2.4.2.9]\n",
      "            PA0403  pyrR; bifunctional pyrimidine regulatory protein PyrR/uracil phosphoribosyltransferase [KO:K02825] [EC:2.4.2.9]\n",
      "line in gene PA0403  pyrR; bifunctional pyrimidine regulatory protein PyrR/uracil phosphoribosyltransferase [KO:K02825] [EC:2.4.2.9]\n",
      "            PA0440  oxidoreductase [KO:K17722] [EC:1.3.1.1]\n",
      "line in gene PA0440  oxidoreductase [KO:K17722] [EC:1.3.1.1]\n",
      "            PA0439  dihydropyrimidine dehydrogenase [KO:K17723] [EC:1.3.1.1]\n",
      "line in gene PA0439  dihydropyrimidine dehydrogenase [KO:K17723] [EC:1.3.1.1]\n",
      "            PA0441  dht; D-hydantoinase/dihydropyrimidinase [KO:K01464] [EC:3.5.2.2]\n",
      "line in gene PA0441  dht; D-hydantoinase/dihydropyrimidinase [KO:K01464] [EC:3.5.2.2]\n",
      "            PA0444  allantoate amidohydrolase [KO:K06016] [EC:3.5.1.6 3.5.1.87]\n",
      "line in gene PA0444  allantoate amidohydrolase [KO:K06016] [EC:3.5.1.6 3.5.1.87]\n",
      "            PA4115  hypothetical protein [KO:K06966] [EC:3.2.2.10 3.2.2.-]\n",
      "line in gene PA4115  hypothetical protein [KO:K06966] [EC:3.2.2.10 3.2.2.-]\n",
      "            PA0437  codA; cytosine deaminase [KO:K01485] [EC:3.5.4.1 3.5.4.21]\n",
      "line in gene PA0437  codA; cytosine deaminase [KO:K01485] [EC:3.5.4.1 3.5.4.21]\n",
      "            PA1920  nrdD; anaerobic ribonucleoside triphosphate reductase [KO:K21636] [EC:1.1.98.6]\n",
      "line in gene PA1920  nrdD; anaerobic ribonucleoside triphosphate reductase [KO:K21636] [EC:1.1.98.6]\n",
      "            PA5497  nrdJa; ribonucleoside-diphosphate reductase [KO:K00525] [EC:1.17.4.1]\n",
      "line in gene PA5497  nrdJa; ribonucleoside-diphosphate reductase [KO:K00525] [EC:1.17.4.1]\n",
      "            PA1156  nrdA; ribonucleotide-diphosphate reductase subunit alpha [KO:K00525] [EC:1.17.4.1]\n",
      "line in gene PA1156  nrdA; ribonucleotide-diphosphate reductase subunit alpha [KO:K00525] [EC:1.17.4.1]\n",
      "            PA1155  nrdB; ribonucleotide-diphosphate reductase subunit beta [KO:K00526] [EC:1.17.4.1]\n",
      "line in gene PA1155  nrdB; ribonucleotide-diphosphate reductase subunit beta [KO:K00526] [EC:1.17.4.1]\n",
      "            PA2962  tmk; thymidylate kinase [KO:K00943] [EC:2.7.4.9]\n",
      "line in gene PA2962  tmk; thymidylate kinase [KO:K00943] [EC:2.7.4.9]\n",
      "            PA5321  dut; deoxyuridine 5'-triphosphate nucleotidohydrolase [KO:K01520] [EC:3.6.1.23]\n",
      "line in gene PA5321  dut; deoxyuridine 5'-triphosphate nucleotidohydrolase [KO:K01520] [EC:3.6.1.23]\n",
      "            PA0342  thyA; thymidylate synthase [KO:K00560] [EC:2.1.1.45]\n",
      "line in gene PA0342  thyA; thymidylate synthase [KO:K00560] [EC:2.1.1.45]\n",
      "COMPOUND    C00015  UDP\n",
      "            C00039  DNA\n",
      "            C00046  RNA\n",
      "            C00055  CMP\n",
      "            C00063  CTP\n",
      "            C00064  L-Glutamine\n",
      "            C00075  UTP\n",
      "            C00086  Urea\n",
      "            C00099  beta-Alanine\n",
      "            C00105  UMP\n",
      "            C00106  Uracil\n",
      "            C00112  CDP\n",
      "            C00119  5-Phospho-alpha-D-ribose 1-diphosphate\n",
      "            C00169  Carbamoyl phosphate\n",
      "            C00178  Thymine\n",
      "            C00214  Thymidine\n",
      "            C00222  3-Oxopropanoate\n",
      "            C00239  dCMP\n",
      "            C00295  Orotate\n",
      "            C00299  Uridine\n",
      "            C00337  (S)-Dihydroorotate\n",
      "            C00363  dTDP\n",
      "            C00364  dTMP\n",
      "            C00365  dUMP\n",
      "            C00380  Cytosine\n",
      "            C00383  Malonate\n",
      "            C00429  5,6-Dihydrouracil\n",
      "            C00438  N-Carbamoyl-L-aspartate\n",
      "            C00458  dCTP\n",
      "            C00459  dTTP\n",
      "            C00460  dUTP\n",
      "            C00475  Cytidine\n",
      "            C00526  Deoxyuridine\n",
      "            C00672  2-Deoxy-D-ribose 1-phosphate\n",
      "            C00705  dCDP\n",
      "            C00813  Barbiturate\n",
      "            C00881  Deoxycytidine\n",
      "            C01013  3-Hydroxypropanoate\n",
      "            C01103  Orotidine 5'-phosphate\n",
      "            C01168  Pseudouridine 5'-phosphate\n",
      "            C01205  (R)-3-Amino-2-methylpropanoate\n",
      "            C01346  dUDP\n",
      "            C01368  3'-UMP\n",
      "            C01563  Carbamate\n",
      "            C02067  Pseudouridine\n",
      "            C02170  Methylmalonate\n",
      "            C02354  2',3'-Cyclic CMP\n",
      "            C02355  2',3'-Cyclic UMP\n",
      "            C02376  5-Methylcytosine\n",
      "            C02642  3-Ureidopropionate\n",
      "            C03997  5-Hydroxymethyldeoxycytidylate\n",
      "            C05281  5-Methylbarbiturate\n",
      "            C05822  3'-CMP\n",
      "            C06198  P1,P4-Bis(5'-uridyl) tetraphosphate\n",
      "            C11038  2'-Deoxy-5-hydroxymethylcytidine-5'-diphosphate\n",
      "            C11039  2'-Deoxy-5-hydroxymethylcytidine-5'-triphosphate\n",
      "            C15607  3-Oxo-3-ureidopropanoate\n",
      "            C20231  (Z)-3-Ureidoacrylate peracid\n",
      "            C20249  (Z)-3-Peroxyaminoacrylate\n",
      "            C20253  Aminoacrylate\n",
      "            C20254  Ureidoacrylate\n",
      "            C20846  3-Oxo-3-ureidoisobutyrate\n",
      "            C21028  (R)-5,6-Dihydrothymine\n",
      "            C21029  (R)-3-Ureidoisobutyrate\n",
      "REL_PATHWAY pae00030  Pentose phosphate pathway\n",
      "            pae00220  Arginine biosynthesis\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00280  Valine, leucine and isoleucine degradation\n",
      "            pae00410  beta-Alanine metabolism\n",
      "KO_PATHWAY  ko00240\n",
      "///\n",
      "path:pae00250\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00250                    Pathway\n",
      "NAME        Alanine, aspartate and glutamate metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00250  Alanine, aspartate and glutamate metabolism\n",
      "DBLINKS     GO: 0006522 0006531 0006536\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0761  nadB; L-aspartate oxidase [KO:K00278] [EC:1.4.3.16]\n",
      "line in gene PA0761  nadB; L-aspartate oxidase [KO:K00278] [EC:1.4.3.16]\n",
      "            PA2253  ansA; L-asparaginase I [KO:K01424] [EC:3.5.1.1]\n",
      "line in gene PA2253  ansA; L-asparaginase I [KO:K01424] [EC:3.5.1.1]\n",
      "            PA1337  ansB; glutaminase-asparaginase [KO:K05597] [EC:3.5.1.38]\n",
      "line in gene PA1337  ansB; glutaminase-asparaginase [KO:K05597] [EC:3.5.1.38]\n",
      "            PA2084  asparagine synthetase [KO:K01953] [EC:6.3.5.4]\n",
      "line in gene PA2084  asparagine synthetase [KO:K01953] [EC:6.3.5.4]\n",
      "            PA3459  glutamine amidotransferase [KO:K01953] [EC:6.3.5.4]\n",
      "line in gene PA3459  glutamine amidotransferase [KO:K01953] [EC:6.3.5.4]\n",
      "            PA0051  phzH; phenazine-modifying protein [KO:K01953] [EC:6.3.5.4]\n",
      "line in gene PA0051  phzH; phenazine-modifying protein [KO:K01953] [EC:6.3.5.4]\n",
      "            PA3797  hypothetical protein [KO:K13566] [EC:3.5.1.3]\n",
      "line in gene PA3797  hypothetical protein [KO:K13566] [EC:3.5.1.3]\n",
      "            PA2828  aminotransferase [KO:K14260] [EC:2.6.1.66 2.6.1.2]\n",
      "line in gene PA2828  aminotransferase [KO:K14260] [EC:2.6.1.66 2.6.1.2]\n",
      "            PA5429  aspA; aspartate ammonia-lyase [KO:K01744] [EC:4.3.1.1]\n",
      "line in gene PA5429  aspA; aspartate ammonia-lyase [KO:K01744] [EC:4.3.1.1]\n",
      "            PA3525  argG; argininosuccinate synthase [KO:K01940] [EC:6.3.4.5]\n",
      "line in gene PA3525  argG; argininosuccinate synthase [KO:K01940] [EC:6.3.4.5]\n",
      "            PA5263  argH; argininosuccinate lyase [KO:K01755] [EC:4.3.2.1]\n",
      "line in gene PA5263  argH; argininosuccinate lyase [KO:K01755] [EC:4.3.2.1]\n",
      "            PA4938  purA; adenylosuccinate synthetase [KO:K01939] [EC:6.3.4.4]\n",
      "line in gene PA4938  purA; adenylosuccinate synthetase [KO:K01939] [EC:6.3.4.4]\n",
      "            PA3516  adenylosuccinate lyase [KO:K01756] [EC:4.3.2.2]\n",
      "line in gene PA3516  adenylosuccinate lyase [KO:K01756] [EC:4.3.2.2]\n",
      "            PA3517  adenylosuccinate lyase [KO:K01756] [EC:4.3.2.2]\n",
      "line in gene PA3517  adenylosuccinate lyase [KO:K01756] [EC:4.3.2.2]\n",
      "            PA2629  purB; adenylosuccinate lyase [KO:K01756] [EC:4.3.2.2]\n",
      "line in gene PA2629  purB; adenylosuccinate lyase [KO:K01756] [EC:4.3.2.2]\n",
      "            PA0402  pyrB; aspartate carbamoyltransferase [KO:K00609] [EC:2.1.3.2]\n",
      "line in gene PA0402  pyrB; aspartate carbamoyltransferase [KO:K00609] [EC:2.1.3.2]\n",
      "            PA0266  5-aminovalerate aminotransferase DavT [KO:K14268] [EC:2.6.1.48 2.6.1.19]\n",
      "line in gene PA0266  5-aminovalerate aminotransferase DavT [KO:K14268] [EC:2.6.1.48 2.6.1.19]\n",
      "            PA0265  gabD; glutarate-semialdehyde dehydrogenase DavD [KO:K00135] [EC:1.2.1.16 1.2.1.79 1.2.1.20]\n",
      "line in gene PA0265  gabD; glutarate-semialdehyde dehydrogenase DavD [KO:K00135] [EC:1.2.1.16 1.2.1.79 1.2.1.20]\n",
      "            PA5036  gltB; glutamate synthase subunit alpha [KO:K00265] [EC:1.4.1.13]\n",
      "line in gene PA5036  gltB; glutamate synthase subunit alpha [KO:K00265] [EC:1.4.1.13]\n",
      "            PA5035  gltD; glutamate synthase subunit beta [KO:K00266] [EC:1.4.1.13]\n",
      "line in gene PA5035  gltD; glutamate synthase subunit beta [KO:K00266] [EC:1.4.1.13]\n",
      "            PA3068  gdhB; NAD-specific glutamate dehydrogenase [KO:K15371] [EC:1.4.1.2]\n",
      "line in gene PA3068  gdhB; NAD-specific glutamate dehydrogenase [KO:K15371] [EC:1.4.1.2]\n",
      "            PA4588  gdhA; glutamate dehydrogenase [KO:K00262] [EC:1.4.1.4]\n",
      "line in gene PA4588  gdhA; glutamate dehydrogenase [KO:K00262] [EC:1.4.1.4]\n",
      "            PA0782  putA; bifunctional proline dehydrogenase/pyrroline-5-carboxylate dehydrogenase [KO:K13821] [EC:1.5.5.2 1.2.1.88]\n",
      "line in gene PA0782  putA; bifunctional proline dehydrogenase/pyrroline-5-carboxylate dehydrogenase [KO:K13821] [EC:1.5.5.2 1.2.1.88]\n",
      "            PA0296  spuI; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA0296  spuI; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA0298  spuB; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA0298  spuB; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA2040  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA2040  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5119  glnA; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5119  glnA; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5522  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5522  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA1566  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA1566  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "            PA3356  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA3356  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5508  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5508  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA4756  carB; carbamoyl phosphate synthase large subunit [KO:K01955] [EC:6.3.5.5]\n",
      "line in gene PA4756  carB; carbamoyl phosphate synthase large subunit [KO:K01955] [EC:6.3.5.5]\n",
      "            PA4758  carA; carbamoyl phosphate synthase small subunit [KO:K01956] [EC:6.3.5.5]\n",
      "line in gene PA4758  carA; carbamoyl phosphate synthase small subunit [KO:K01956] [EC:6.3.5.5]\n",
      "            PA1638  glutaminase [KO:K01425] [EC:3.5.1.2]\n",
      "line in gene PA1638  glutaminase [KO:K01425] [EC:3.5.1.2]\n",
      "            PA5549  glmS; glucosamine--fructose-6-phosphate aminotransferase [KO:K00820] [EC:2.6.1.16]\n",
      "line in gene PA5549  glmS; glucosamine--fructose-6-phosphate aminotransferase [KO:K00820] [EC:2.6.1.16]\n",
      "            PA3759  aminotransferase [KO:K00820] [EC:2.6.1.16]\n",
      "line in gene PA3759  aminotransferase [KO:K00820] [EC:2.6.1.16]\n",
      "            PA3108  purF; amidophosphoribosyltransferase [KO:K00764] [EC:2.4.2.14]\n",
      "line in gene PA3108  purF; amidophosphoribosyltransferase [KO:K00764] [EC:2.4.2.14]\n",
      "COMPOUND    C00014  Ammonia\n",
      "            C00022  Pyruvate\n",
      "            C00025  L-Glutamate\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00036  Oxaloacetate\n",
      "            C00041  L-Alanine\n",
      "            C00042  Succinate\n",
      "            C00049  L-Aspartate\n",
      "            C00064  L-Glutamine\n",
      "            C00122  Fumarate\n",
      "            C00152  L-Asparagine\n",
      "            C00158  Citrate\n",
      "            C00169  Carbamoyl phosphate\n",
      "            C00232  Succinate semialdehyde\n",
      "            C00334  4-Aminobutanoate\n",
      "            C00352  D-Glucosamine 6-phosphate\n",
      "            C00402  D-Aspartate\n",
      "            C00438  N-Carbamoyl-L-aspartate\n",
      "            C00940  2-Oxoglutaramate\n",
      "            C01042  N-Acetyl-L-aspartate\n",
      "            C02362  2-Oxosuccinamate\n",
      "            C03090  5-Phosphoribosylamine\n",
      "            C03406  N-(L-Arginino)succinate\n",
      "            C03794  N6-(1,2-Dicarboxyethyl)-AMP\n",
      "            C03912  (S)-1-Pyrroline-5-carboxylate\n",
      "            C12270  N-Acetylaspartylglutamate\n",
      "            C20775  beta-Citryl-L-glutamate\n",
      "            C20776  N-Acetylaspartylglutamylglutamate\n",
      "REFERENCE   \n",
      "  AUTHORS   Nishizuka Y, Seyama Y, Ikai A, Ishimura Y, Kawaguchi A (eds).\n",
      "  TITLE     [Cellular Functions and Metabolic Maps] (In Japanese)\n",
      "  JOURNAL   Tokyo Kagaku Dojin (1997)\n",
      "REFERENCE   PMID:9687539\n",
      "  AUTHORS   Wu G\n",
      "  TITLE     Intestinal mucosal amino acid catabolism.\n",
      "  JOURNAL   J Nutr 128:1249-52 (1998)\n",
      "            DOI:10.1093/jn/128.8.1249\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00220  Arginine biosynthesis\n",
      "            pae00230  Purine metabolism\n",
      "            pae00240  Pyrimidine metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00261  Monobactam biosynthesis\n",
      "            pae00300  Lysine biosynthesis\n",
      "            pae00330  Arginine and proline metabolism\n",
      "            pae00340  Histidine metabolism\n",
      "            pae00410  beta-Alanine metabolism\n",
      "            pae00460  Cyanoamino acid metabolism\n",
      "            pae00470  D-Amino acid metabolism\n",
      "            pae00480  Glutathione metabolism\n",
      "            pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00650  Butanoate metabolism\n",
      "            pae00660  C5-Branched dibasic acid metabolism\n",
      "            pae00760  Nicotinate and nicotinamide metabolism\n",
      "            pae00770  Pantothenate and CoA biosynthesis\n",
      "            pae00860  Porphyrin metabolism\n",
      "            pae00910  Nitrogen metabolism\n",
      "KO_PATHWAY  ko00250\n",
      "///\n",
      "path:pae00260\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00260                    Pathway\n",
      "NAME        Glycine, serine and threonine metabolism - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Serine is derived from 3-phospho-D-glycerate, an intermediate of glycolysis [MD:M00020], and glycine is derived from serine. Threonine is an essential amino acid, which animals cannot synthesize. In bacteria and plants, threonine is derived from aspartate [MD:M00018].\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00260  Glycine, serine and threonine metabolism\n",
      "MODULE      pae_M00018  Threonine biosynthesis, aspartate => homoserine => threonine [PATH:pae00260]\n",
      "            pae_M00020  Serine biosynthesis, glycerate-3P => serine [PATH:pae00260]\n",
      "            pae_M00338  Cysteine biosynthesis, homocysteine + serine => cysteine [PATH:pae00260]\n",
      "            pae_M00555  Betaine biosynthesis, choline => betaine [PATH:pae00260]\n",
      "DBLINKS     GO: 0006544 0006563 0070178 0006566\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0904  lysC; aspartokinase [KO:K00928] [EC:2.7.2.4]\n",
      "line in gene PA0904  lysC; aspartokinase [KO:K00928] [EC:2.7.2.4]\n",
      "            PA3117  asd; aspartate-semialdehyde dehydrogenase [KO:K00133] [EC:1.2.1.11]\n",
      "line in gene PA3117  asd; aspartate-semialdehyde dehydrogenase [KO:K00133] [EC:1.2.1.11]\n",
      "            PA3736  hom; homoserine dehydrogenase [KO:K00003] [EC:1.1.1.3]\n",
      "line in gene PA3736  hom; homoserine dehydrogenase [KO:K00003] [EC:1.1.1.3]\n",
      "            PA5495  thrB; homoserine kinase [KO:K02204] [EC:2.7.1.39]\n",
      "line in gene PA5495  thrB; homoserine kinase [KO:K02204] [EC:2.7.1.39]\n",
      "            PA1757  thrH; phosphoserine phosphatase [KO:K02203] [EC:3.1.3.3 2.7.1.39]\n",
      "line in gene PA1757  thrH; phosphoserine phosphatase [KO:K02203] [EC:3.1.3.3 2.7.1.39]\n",
      "            PA3735  thrC; threonine synthase [KO:K01733] [EC:4.2.3.1]\n",
      "line in gene PA3735  thrC; threonine synthase [KO:K01733] [EC:4.2.3.1]\n",
      "            PA5413  ltaA; low specificity l-threonine aldolase [KO:K01620] [EC:4.1.2.48]\n",
      "line in gene PA5413  ltaA; low specificity l-threonine aldolase [KO:K01620] [EC:4.1.2.48]\n",
      "            PA0902  hypothetical protein [KO:K01620] [EC:4.1.2.48]\n",
      "line in gene PA0902  hypothetical protein [KO:K01620] [EC:4.1.2.48]\n",
      "            PA2444  glyA2; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA2444  glyA2; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA5415  glyA1; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA5415  glyA1; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA4602  glyA3; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA4602  glyA3; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA4626  hprA; glycerate dehydrogenase [KO:K00018] [EC:1.1.1.29]\n",
      "line in gene PA4626  hprA; glycerate dehydrogenase [KO:K00018] [EC:1.1.1.29]\n",
      "            PA3896  2-hydroxyacid dehydrogenase [KO:K00090] [EC:1.1.1.79 1.1.1.81 1.1.1.215]\n",
      "line in gene PA3896  2-hydroxyacid dehydrogenase [KO:K00090] [EC:1.1.1.79 1.1.1.81 1.1.1.215]\n",
      "            PA1052  hypothetical protein [KO:K00865] [EC:2.7.1.165]\n",
      "line in gene PA1052  hypothetical protein [KO:K00865] [EC:2.7.1.165]\n",
      "            PA1499  hypothetical protein [KO:K11529] [EC:2.7.1.165]\n",
      "line in gene PA1499  hypothetical protein [KO:K11529] [EC:2.7.1.165]\n",
      "            PA5131  pgm; 2,3-bisphosphoglycerate-independent phosphoglycerate mutase [KO:K15633] [EC:5.4.2.12]\n",
      "line in gene PA5131  pgm; 2,3-bisphosphoglycerate-independent phosphoglycerate mutase [KO:K15633] [EC:5.4.2.12]\n",
      "            PA0316  serA; D-3-phosphoglycerate dehydrogenase [KO:K00058] [EC:1.1.1.95 1.1.1.399]\n",
      "line in gene PA0316  serA; D-3-phosphoglycerate dehydrogenase [KO:K00058] [EC:1.1.1.95 1.1.1.399]\n",
      "            PA3167  serC; 3-phosphoserine/phosphohydroxythreonine aminotransferase [KO:K00831] [EC:2.6.1.52]\n",
      "line in gene PA3167  serC; 3-phosphoserine/phosphohydroxythreonine aminotransferase [KO:K00831] [EC:2.6.1.52]\n",
      "            PA4960  phosphoserine phosphatase [KO:K01079] [EC:3.1.3.3]\n",
      "line in gene PA4960  phosphoserine phosphatase [KO:K01079] [EC:3.1.3.3]\n",
      "            PA0421  hypothetical protein [KO:K00274] [EC:1.4.3.4]\n",
      "line in gene PA0421  hypothetical protein [KO:K00274] [EC:1.4.3.4]\n",
      "            PA2445  gcvP2; glycine dehydrogenase [KO:K00281] [EC:1.4.4.2]\n",
      "line in gene PA2445  gcvP2; glycine dehydrogenase [KO:K00281] [EC:1.4.4.2]\n",
      "            PA5213  gcvP1; glycine dehydrogenase [KO:K00281] [EC:1.4.4.2]\n",
      "line in gene PA5213  gcvP1; glycine dehydrogenase [KO:K00281] [EC:1.4.4.2]\n",
      "            PA2442  gcvT2; glycine cleavage system protein T2 [KO:K00605] [EC:2.1.2.10]\n",
      "line in gene PA2442  gcvT2; glycine cleavage system protein T2 [KO:K00605] [EC:2.1.2.10]\n",
      "            PA5215  gcvT1; glycine cleavage system aminomethyltransferase T [KO:K00605] [EC:2.1.2.10]\n",
      "line in gene PA5215  gcvT1; glycine cleavage system aminomethyltransferase T [KO:K00605] [EC:2.1.2.10]\n",
      "            PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA2446  gcvH2; glycine cleavage system protein H [KO:K02437]\n",
      "line in gene PA2446  gcvH2; glycine cleavage system protein H [KO:K02437]\n",
      "            PA5214  gcvH1; glycine cleavage system protein H [KO:K02437]\n",
      "line in gene PA5214  gcvH1; glycine cleavage system protein H [KO:K02437]\n",
      "            PA4796  hypothetical protein [KO:K02437]\n",
      "line in gene PA4796  hypothetical protein [KO:K02437]\n",
      "            PA4693  pssA; phosphatidylserine synthase [KO:K17103] [EC:2.7.8.8]\n",
      "line in gene PA4693  pssA; phosphatidylserine synthase [KO:K17103] [EC:2.7.8.8]\n",
      "            PA5372  betA; choline dehydrogenase [KO:K00108] [EC:1.1.99.1]\n",
      "line in gene PA5372  betA; choline dehydrogenase [KO:K00108] [EC:1.1.99.1]\n",
      "            PA3710  GMC-type oxidoreductase [KO:K00108] [EC:1.1.99.1]\n",
      "line in gene PA3710  GMC-type oxidoreductase [KO:K00108] [EC:1.1.99.1]\n",
      "            PA2124  dehydrogenase [KO:K00108] [EC:1.1.99.1]\n",
      "line in gene PA2124  dehydrogenase [KO:K00108] [EC:1.1.99.1]\n",
      "            PA3504  aldehyde dehydrogenase [KO:K00130] [EC:1.2.1.8]\n",
      "line in gene PA3504  aldehyde dehydrogenase [KO:K00130] [EC:1.2.1.8]\n",
      "            PA5373  betB; betaine aldehyde dehydrogenase [KO:K00130] [EC:1.2.1.8]\n",
      "line in gene PA5373  betB; betaine aldehyde dehydrogenase [KO:K00130] [EC:1.2.1.8]\n",
      "            PA5418  soxA; sarcosine oxidase subunit alpha [KO:K00302] [EC:1.5.3.1]\n",
      "line in gene PA5418  soxA; sarcosine oxidase subunit alpha [KO:K00302] [EC:1.5.3.1]\n",
      "            PA5416  soxB; sarcosine oxidase subunit beta [KO:K00303] [EC:1.5.3.1]\n",
      "line in gene PA5416  soxB; sarcosine oxidase subunit beta [KO:K00303] [EC:1.5.3.1]\n",
      "            PA5419  soxG; sarcosine oxidase subunit gamma [KO:K00305] [EC:1.5.3.1]\n",
      "line in gene PA5419  soxG; sarcosine oxidase subunit gamma [KO:K00305] [EC:1.5.3.1]\n",
      "            PA5417  soxD; sarcosine oxidase subunit delta [KO:K00304] [EC:1.5.3.1]\n",
      "line in gene PA5417  soxD; sarcosine oxidase subunit delta [KO:K00304] [EC:1.5.3.1]\n",
      "            PA0399  cystathionine beta-synthase [KO:K01697] [EC:4.2.1.22]\n",
      "line in gene PA0399  cystathionine beta-synthase [KO:K01697] [EC:4.2.1.22]\n",
      "            PA0400  cystathionine gamma-lyase [KO:K01758] [EC:4.4.1.1]\n",
      "line in gene PA0400  cystathionine gamma-lyase [KO:K01758] [EC:4.4.1.1]\n",
      "            PA2443  sdaA; L-serine dehydratase [KO:K01752] [EC:4.3.1.17]\n",
      "line in gene PA2443  sdaA; L-serine dehydratase [KO:K01752] [EC:4.3.1.17]\n",
      "            PA5379  sdaB; L-serine dehydratase [KO:K01752] [EC:4.3.1.17]\n",
      "line in gene PA5379  sdaB; L-serine dehydratase [KO:K01752] [EC:4.3.1.17]\n",
      "            PA0331  ilvA1; threonine dehydratase [KO:K01754] [EC:4.3.1.19]\n",
      "line in gene PA0331  ilvA1; threonine dehydratase [KO:K01754] [EC:4.3.1.19]\n",
      "            PA1326  ilvA2; threonine dehydratase [KO:K01754] [EC:4.3.1.19]\n",
      "line in gene PA1326  ilvA2; threonine dehydratase [KO:K01754] [EC:4.3.1.19]\n",
      "            PA0851  hypothetical protein [KO:K01754] [EC:4.3.1.19]\n",
      "line in gene PA0851  hypothetical protein [KO:K01754] [EC:4.3.1.19]\n",
      "            PA3357  dsdA; D-serine dehydratase [KO:K01753] [EC:4.3.1.18]\n",
      "line in gene PA3357  dsdA; D-serine dehydratase [KO:K01753] [EC:4.3.1.18]\n",
      "            PA0035  trpA; tryptophan synthase subunit alpha [KO:K01695] [EC:4.2.1.20]\n",
      "line in gene PA0035  trpA; tryptophan synthase subunit alpha [KO:K01695] [EC:4.2.1.20]\n",
      "            PA0036  trpB; tryptophan synthase subunit beta [KO:K01696] [EC:4.2.1.20]\n",
      "line in gene PA0036  trpB; tryptophan synthase subunit beta [KO:K01696] [EC:4.2.1.20]\n",
      "            PA2413  pvdH; diaminobutyrate--2-oxoglutarate aminotransferase [KO:K00836] [EC:2.6.1.76]\n",
      "line in gene PA2413  pvdH; diaminobutyrate--2-oxoglutarate aminotransferase [KO:K00836] [EC:2.6.1.76]\n",
      "COMPOUND    C00011  CO2\n",
      "            C00014  Ammonia\n",
      "            C00022  Pyruvate\n",
      "            C00037  Glycine\n",
      "            C00048  Glyoxylate\n",
      "            C00049  L-Aspartate\n",
      "            C00065  L-Serine\n",
      "            C00078  L-Tryptophan\n",
      "            C00097  L-Cysteine\n",
      "            C00101  Tetrahydrofolate\n",
      "            C00109  2-Oxobutanoate\n",
      "            C00114  Choline\n",
      "            C00143  5,10-Methylenetetrahydrofolate\n",
      "            C00168  Hydroxypyruvate\n",
      "            C00188  L-Threonine\n",
      "            C00197  3-Phospho-D-glycerate\n",
      "            C00213  Sarcosine\n",
      "            C00258  D-Glycerate\n",
      "            C00263  L-Homoserine\n",
      "            C00300  Creatine\n",
      "            C00430  5-Aminolevulinate\n",
      "            C00441  L-Aspartate 4-semialdehyde\n",
      "            C00546  Methylglyoxal\n",
      "            C00576  Betaine aldehyde\n",
      "            C00581  Guanidinoacetate\n",
      "            C00631  2-Phospho-D-glycerate\n",
      "            C00719  Betaine\n",
      "            C00740  D-Serine\n",
      "            C00986  1,3-Diaminopropane\n",
      "            C01005  O-Phospho-L-serine\n",
      "            C01026  N,N-Dimethylglycine\n",
      "            C01102  O-Phospho-L-homoserine\n",
      "            C01242  [Protein]-S8-aminomethyldihydrolipoyllysine\n",
      "            C01888  Aminoacetone\n",
      "            C02051  Lipoylprotein\n",
      "            C02291  L-Cystathionine\n",
      "            C02737  Phosphatidylserine\n",
      "            C02972  Dihydrolipoylprotein\n",
      "            C03082  4-Phospho-L-aspartate\n",
      "            C03194  (R)-1-Aminopropan-2-ol\n",
      "            C03232  3-Phosphonooxypyruvate\n",
      "            C03283  L-2,4-Diaminobutanoate\n",
      "            C03508  L-2-Amino-3-oxobutanoic acid\n",
      "            C05519  L-Allothreonine\n",
      "            C06231  Ectoine\n",
      "            C06442  N(gamma)-Acetyldiaminobutyrate\n",
      "            C16432  5-Hydroxyectoine\n",
      "            C19929  N(alpha)-Acetyl-L-2,4-diaminobutyrate\n",
      "REFERENCE   \n",
      "  AUTHORS   Nishizuka Y, Seyama Y, Ikai A, Ishimura Y, Kawaguchi A (eds).\n",
      "  TITLE     [Cellular Functions and Metabolic Maps] (In Japanese)\n",
      "  JOURNAL   Tokyo Kagaku Dojin (1997)\n",
      "REFERENCE   PMID:11823218\n",
      "  AUTHORS   Kuhlmann AU, Bremer E.\n",
      "  TITLE     Osmotically regulated synthesis of the compatible solute ectoine in Bacillus pasteurii and related Bacillus spp.\n",
      "  JOURNAL   Appl Environ Microbiol 68:772-83 (2002)\n",
      "            DOI:10.1128/AEM.68.2.772-783.2002\n",
      "REFERENCE   PMID:10473374\n",
      "  AUTHORS   Canovas D, Borges N, Vargas C, Ventosa A, Nieto JJ, Santos H.\n",
      "  TITLE     Role of Ngamma-acetyldiaminobutyrate as an enzyme stabilizer and an intermediate in the biosynthesis of hydroxyectoine.\n",
      "  JOURNAL   Appl Environ Microbiol 65:3774-9 (1999)\n",
      "            DOI:10.1128/AEM.65.9.3774-3779.1999\n",
      "REFERENCE   PMID:17636255\n",
      "  AUTHORS   Bursy J, Pierik AJ, Pica N, Bremer E.\n",
      "  TITLE     Osmotically induced synthesis of the compatible solute hydroxyectoine is mediated by an evolutionarily conserved ectoine hydroxylase.\n",
      "  JOURNAL   J Biol Chem 282:31147-55 (2007)\n",
      "            DOI:10.1074/jbc.M704023200\n",
      "REFERENCE   PMID:16707670\n",
      "  AUTHORS   Garcia-Estepa R, Argandona M, Reina-Bueno M, Capote N, Iglesias-Guerra F, Nieto JJ, Vargas C.\n",
      "  TITLE     The ectD gene, which is involved in the synthesis of the compatible solute hydroxyectoine, is essential for thermoprotection of the halophilic bacterium Chromohalobacter salexigens.\n",
      "  JOURNAL   J Bacteriol 188:3774-84 (2006)\n",
      "            DOI:10.1128/JB.00136-06\n",
      "REFERENCE   PMID:15470257\n",
      "  AUTHORS   Arevalo-Rodriguez M, Pan X, Boeke JD, Heitman J\n",
      "  TITLE     FKBP12 controls aspartate pathway flux in Saccharomyces cerevisiae to prevent toxic intermediate accumulation.\n",
      "  JOURNAL   Eukaryot Cell 3:1287-96 (2004)\n",
      "            DOI:10.1128/EC.3.5.1287-1296.2004\n",
      "REFERENCE   PMID:7639721\n",
      "  AUTHORS   Rees WD, Hay SM\n",
      "  TITLE     The biosynthesis of threonine by mammalian cells: expression of a complete bacterial biosynthetic pathway in an animal cell.\n",
      "  JOURNAL   Biochem J 309 ( Pt 3):999-1007 (1995)\n",
      "            DOI:10.1042/bj3090999\n",
      "REFERENCE   PMID:11341914\n",
      "  AUTHORS   Jacques SL, Nieman C, Bareich D, Broadhead G, Kinach R, Honek JF, Wright GD\n",
      "  TITLE     Characterization of yeast homoserine dehydrogenase, an antifungal target: the invariant histidine 309 is important for enzyme integrity.\n",
      "  JOURNAL   Biochim Biophys Acta 1544:28-41 (2001)\n",
      "            DOI:10.1016/S0167-4838(00)00203-X\n",
      "REFERENCE   PMID:15520012\n",
      "  AUTHORS   Chen X, Jhee KH, Kruger WD\n",
      "  TITLE     Production of the neuromodulator H2S by cystathionine beta-synthase via the condensation of cysteine and homocysteine.\n",
      "  JOURNAL   J Biol Chem 279:52082-6 (2004)\n",
      "            DOI:10.1074/jbc.C400481200\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00230  Purine metabolism\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00290  Valine, leucine and isoleucine biosynthesis\n",
      "            pae00300  Lysine biosynthesis\n",
      "            pae00330  Arginine and proline metabolism\n",
      "            pae00460  Cyanoamino acid metabolism\n",
      "            pae00470  D-Amino acid metabolism\n",
      "            pae00564  Glycerophospholipid metabolism\n",
      "            pae00600  Sphingolipid metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00640  Propanoate metabolism\n",
      "            pae00680  Methane metabolism\n",
      "            pae00860  Porphyrin metabolism\n",
      "            pae00920  Sulfur metabolism\n",
      "KO_PATHWAY  ko00260\n",
      "///\n",
      "path:pae00261\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00261                    Pathway\n",
      "NAME        Monobactam biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Monobactams are beta-lactam antibiotics containing a monocyclic beta-lactam nucleus, which is structurally different from penicillin and cephalosporin core structures with another fused ring. This diagram shows biosynthesis of nocardicin A, a naturally occurring monobactam, via the pentapeptide formed by condensation of L-4-hydroxyphenylglycine (L-pHPG), L-arginine and L-serine [MD:M00736]. Other naturally occurring monobactams are also shown but the biosynthetic pathway is not yet fully characterized. Sulfazecin and other 3-aminomonobactamic acid derivatives are derived from serine or threonine, and tabtoxinine-beta-lactam is a phytotoxin. Aztreonam, a synthetic monobactam originally isolated as SQ 26,180 from Chromobacterium violaceum, is the first clinically used monobactam.\n",
      "CLASS       Metabolism; Biosynthesis of other secondary metabolites\n",
      "PATHWAY_MAP pae00261  Monobactam biosynthesis\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2412  hypothetical protein [KO:K05375]\n",
      "line in gene PA2412  hypothetical protein [KO:K05375]\n",
      "            PA4442  cysN; bifunctional sulfate adenylyltransferase subunit 1/adenylylsulfate kinase [KO:K00955] [EC:2.7.7.4 2.7.1.25]\n",
      "line in gene PA4442  cysN; bifunctional sulfate adenylyltransferase subunit 1/adenylylsulfate kinase [KO:K00955] [EC:2.7.7.4 2.7.1.25]\n",
      "            PA4443  cysD; sulfate adenylyltransferase subunit 2 [KO:K00957] [EC:2.7.7.4]\n",
      "line in gene PA4443  cysD; sulfate adenylyltransferase subunit 2 [KO:K00957] [EC:2.7.7.4]\n",
      "            PA0904  lysC; aspartokinase [KO:K00928] [EC:2.7.2.4]\n",
      "line in gene PA0904  lysC; aspartokinase [KO:K00928] [EC:2.7.2.4]\n",
      "            PA3117  asd; aspartate-semialdehyde dehydrogenase [KO:K00133] [EC:1.2.1.11]\n",
      "line in gene PA3117  asd; aspartate-semialdehyde dehydrogenase [KO:K00133] [EC:1.2.1.11]\n",
      "            PA0223  dihydrodipicolinate synthetase [KO:K01714] [EC:4.3.3.7]\n",
      "line in gene PA0223  dihydrodipicolinate synthetase [KO:K01714] [EC:4.3.3.7]\n",
      "            PA1010  dapA; 4-hydroxy-tetrahydrodipicolinate synthase [KO:K01714] [EC:4.3.3.7]\n",
      "line in gene PA1010  dapA; 4-hydroxy-tetrahydrodipicolinate synthase [KO:K01714] [EC:4.3.3.7]\n",
      "            PA4188  hypothetical protein [KO:K01714] [EC:4.3.3.7]\n",
      "line in gene PA4188  hypothetical protein [KO:K01714] [EC:4.3.3.7]\n",
      "            PA4759  dapB; 4-hydroxy-tetrahydrodipicolinate reductase [KO:K00215] [EC:1.17.1.8]\n",
      "line in gene PA4759  dapB; 4-hydroxy-tetrahydrodipicolinate reductase [KO:K00215] [EC:1.17.1.8]\n",
      "COMPOUND    C00019  S-Adenosyl-L-methionine\n",
      "            C00022  Pyruvate\n",
      "            C00049  L-Aspartate\n",
      "            C00059  Sulfate\n",
      "            C00062  L-Arginine\n",
      "            C00065  L-Serine\n",
      "            C00082  L-Tyrosine\n",
      "            C00188  L-Threonine\n",
      "            C00224  Adenylyl sulfate\n",
      "            C00441  L-Aspartate 4-semialdehyde\n",
      "            C00927  Isonocardicin A\n",
      "            C01179  3-(4-Hydroxyphenyl)pyruvate\n",
      "            C01739  Nocardicin E\n",
      "            C01941  Nocardicin A\n",
      "            C03082  4-Phospho-L-aspartate\n",
      "            C03198  (S)-4-Hydroxymandelate\n",
      "            C03590  4-Hydroxyphenylglyoxylate\n",
      "            C03972  2,3,4,5-Tetrahydrodipicolinate\n",
      "            C05539  N-Acetyl-L-2-amino-6-oxopimelate\n",
      "            C12323  L-4-Hydroxyphenylglycine\n",
      "            C16842  SQ 26180\n",
      "            C17351  Nocardicin C\n",
      "            C17352  Isonocardicin C\n",
      "            C17355  Nocardicin G\n",
      "            C20258  (2S,4S)-4-Hydroxy-2,3,4,5-tetrahydrodipicolinate\n",
      "            C20911  C20911\n",
      "            C20912  C20912\n",
      "            C20913  C20913\n",
      "            C20914  C20914\n",
      "            C20915  C20915\n",
      "            C20916  C20916\n",
      "            C20917  Tabtoxin\n",
      "            C20918  Tabtoxinine-beta-lactam\n",
      "            C20919  Isotabtoxin\n",
      "            C20920  Tabtoxinine-delta-lactam\n",
      "            C20921  N1-Acetyl-tabtoxinine-beta-lactam\n",
      "            C20927  Sulfazecin\n",
      "            C20928  MM 42842\n",
      "            C20929  3alpha-Hydroxy-3-aminoacylmonobactamic acid\n",
      "REFERENCE   PMID:23135477\n",
      "  AUTHORS   Hamed RB, Gomez-Castellanos JR, Henry L, Ducho C, McDonough MA, Schofield CJ\n",
      "  TITLE     The enzymes of beta-lactam biosynthesis.\n",
      "  JOURNAL   Nat Prod Rep 30:21-107 (2013)\n",
      "            DOI:10.1039/c2np20065a\n",
      "REFERENCE   PMID:15271351\n",
      "  AUTHORS   Gunsior M, Breazeale SD, Lind AJ, Ravel J, Janc JW, Townsend CA\n",
      "  TITLE     The biosynthetic gene cluster for a monocyclic beta-lactam antibiotic, nocardicin A.\n",
      "  JOURNAL   Chem Biol 11:927-38 (2004)\n",
      "            DOI:10.1016/j.chembiol.2004.04.012\n",
      "REFERENCE   PMID:15252031\n",
      "  AUTHORS   Kelly WL, Townsend CA\n",
      "  TITLE     Mutational analysis and characterization of nocardicin C-9' epimerase.\n",
      "  JOURNAL   J Biol Chem 279:38220-7 (2004)\n",
      "            DOI:10.1074/jbc.M405450200\n",
      "REFERENCE   PMID:15629944\n",
      "  AUTHORS   Kelly WL, Townsend CA\n",
      "  TITLE     Mutational analysis of nocK and nocL in the nocardicin a producer Nocardia uniformis.\n",
      "  JOURNAL   J Bacteriol 187:739-46 (2005)\n",
      "            DOI:10.1128/JB.187.2.739-746.2005\n",
      "REFERENCE   PMID:9804844\n",
      "  AUTHORS   Reeve AM, Breazeale SD, Townsend CA\n",
      "  TITLE     Purification, characterization, and cloning of an S-adenosylmethionine-dependent 3-amino-3-carboxypropyltransferase in nocardicin biosynthesis.\n",
      "  JOURNAL   J Biol Chem 273:30695-703 (1998)\n",
      "            DOI:10.1074/jbc.273.46.30695\n",
      "REFERENCE   PMID:12105888\n",
      "  AUTHORS   Kelly WL, Townsend CA\n",
      "  TITLE     Role of the cytochrome P450 NocL in nocardicin A biosynthesis.\n",
      "  JOURNAL   J Am Chem Soc 124:8186-7 (2002)\n",
      "            DOI:10.1021/ja025926g\n",
      "REFERENCE   PMID:11137816\n",
      "  AUTHORS   Hubbard BK, Thomas MG, Walsh CT\n",
      "  TITLE     Biosynthesis of L-p-hydroxyphenylglycine, a non-proteinogenic amino acid constituent of peptide antibiotics.\n",
      "  JOURNAL   Chem Biol 7:931-42 (2000)\n",
      "            DOI:10.1016/S1074-5521(00)00043-0\n",
      "REFERENCE   PMID:25624104\n",
      "  AUTHORS   Gaudelli NM, Long DH, Townsend CA\n",
      "  TITLE     beta-Lactam formation by a non-ribosomal peptide synthetase during antibiotic biosynthesis.\n",
      "  JOURNAL   Nature 520:383-7 (2015)\n",
      "            DOI:10.1038/nature14100\n",
      "REFERENCE   PMID:24531841\n",
      "  AUTHORS   Gaudelli NM, Townsend CA\n",
      "  TITLE     Epimerization and substrate gating by a TE domain in beta-lactam antibiotic biosynthesis.\n",
      "  JOURNAL   Nat Chem Biol 10:251-8 (2014)\n",
      "            DOI:10.1038/nchembio.1456\n",
      "REFERENCE   PMID:22365611\n",
      "  AUTHORS   Davidsen JM, Townsend CA\n",
      "  TITLE     In vivo characterization of nonribosomal peptide synthetases NocA and NocB in the biosynthesis of nocardicin A.\n",
      "  JOURNAL   Chem Biol 19:297-306 (2012)\n",
      "            DOI:10.1016/j.chembiol.2011.10.020\n",
      "REFERENCE   PMID:23330869\n",
      "  AUTHORS   Davidsen JM, Bartley DM, Townsend CA\n",
      "  TITLE     Non-ribosomal propeptide precursor in nocardicin A biosynthesis predicted from adenylation domain specificity dependent on the MbtH family protein NocI.\n",
      "  JOURNAL   J Am Chem Soc 135:1749-59 (2013)\n",
      "            DOI:10.1021/ja307710d\n",
      "REFERENCE   PMID:6805424\n",
      "  AUTHORS   O'Sullivan J, Gillum AM, Aklonis CA, Souser ML, Sykes RB\n",
      "  TITLE     Biosynthesis of monobactam compounds: origin of the carbon atoms in the beta-lactam ring.\n",
      "  JOURNAL   Antimicrob Agents Chemother 21:558-64 (1982)\n",
      "            DOI:10.1128/AAC.21.4.558\n",
      "REFERENCE   PMID:6859838\n",
      "  AUTHORS   O'Sullivan J, Souser ML, Kao CC, Aklonis CA\n",
      "  TITLE     Sulfur metabolism in the biosynthesis of monobactams.\n",
      "  JOURNAL   Antimicrob Agents Chemother 23:598-602 (1983)\n",
      "            DOI:10.1128/AAC.23.4.598\n",
      "REFERENCE   PMID:16506699\n",
      "  AUTHORS   Kinscherf TG, Willis DK\n",
      "  TITLE     The biosynthetic gene cluster for the beta-lactam antibiotic tabtoxin in Pseudomonas syringae.\n",
      "  JOURNAL   J Antibiot (Tokyo) 58:817-21 (2005)\n",
      "            DOI:10.1038/ja.2005.109\n",
      "REFERENCE   PMID:22994681\n",
      "  AUTHORS   Wencewicz TA, Walsh CT\n",
      "  TITLE     Pseudomonas syringae self-protection from tabtoxinine-beta-lactam by ligase TblF and acetylase Ttr.\n",
      "  JOURNAL   Biochemistry 51:7712-25 (2012)\n",
      "            DOI:10.1021/bi3011384\n",
      "REFERENCE   PMID:23770908\n",
      "  AUTHORS   Arai T, Arimura Y, Ishikura S, Kino K\n",
      "  TITLE     L-amino acid ligase from Pseudomonas syringae producing tabtoxin can be used for enzymatic synthesis of various functional peptides.\n",
      "  JOURNAL   Appl Environ Microbiol 79:5023-9 (2013)\n",
      "            DOI:10.1128/AEM.01003-13\n",
      "REFERENCE   PMID:1515668\n",
      "  AUTHORS   Engst K, Shaw PD\n",
      "  TITLE     Identification of a lysA-like gene required for tabtoxin biosynthesis and pathogenicity in Pseudomonas syringae pv. tabaci strain PTBR2.024.\n",
      "  JOURNAL   Mol Plant Microbe Interact 5:322-9 (1992)\n",
      "            DOI:10.1094/mpmi-5-322\n",
      "REFERENCE   PMID:8990304\n",
      "  AUTHORS   Liu L, Shaw PD\n",
      "  TITLE     Characterization of dapB, a gene required by Pseudomonas syringae pv. tabaci BR2.024 for lysine and tabtoxinine-beta-lactam biosynthesis.\n",
      "  JOURNAL   J Bacteriol 179:507-13 (1997)\n",
      "            DOI:10.1128/JB.179.2.507-513.1997\n",
      "REFERENCE   PMID:9294453\n",
      "  AUTHORS   Liu L, Shaw PD\n",
      "  TITLE     A possible role for acetylated intermediates in diaminopimelate and tabtoxinine-beta-lactam biosynthesis in Pseudomonas syringae pv. tabaci BR2.024.\n",
      "  JOURNAL   J Bacteriol 179:5922-7 (1997)\n",
      "            DOI:10.1128/JB.179.18.5922-5927.1997\n",
      "REFERENCE   PMID:2881927\n",
      "  AUTHORS   Unkefer CJ, London RE, Durbin RD, Uchytil TF, Langston-Unkefer PJ\n",
      "  TITLE     The biosynthesis of tabtoxinine-beta-lactam. Use of specifically 13C-labeled glucose and 13C NMR spectroscopy to identify its biosynthetic precursors.\n",
      "  JOURNAL   J Biol Chem 262:4994-9 (1987)\n",
      "REL_PATHWAY pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00300  Lysine biosynthesis\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "KO_PATHWAY  ko00261\n",
      "///\n",
      "path:pae00270\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00270                    Pathway\n",
      "NAME        Cysteine and methionine metabolism - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Cysteine and methionine are sulfur-containing amino acids. Cysteine is synthesized from serine through different pathways in different organism groups. In bacteria and plants, cysteine is converted from serine (via acetylserine) by transfer of hydrogen sulfide [MD:M00021]. In animals, methionine-derived homocysteine is used as sulfur source and its condensation product with serine (cystathionine) is converted to cysteine [MD:M00338]. Cysteine is metabolized to pyruvate in multiple routes. Methionine is an essential amino acid, which animals cannot synthesize. In bacteria and plants, methionine is synthesized from aspartate [MD:M00017]. S-Adenosylmethionine (SAM), synthesized from methionine and ATP, is a methyl group donor in many important transfer reactions including DNA methylation for regulation of gene expression. SAM may also be used to regenerate methionine in the methionine salvage pathway [MD:M00034].\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00270  Cysteine and methionine metabolism\n",
      "MODULE      pae_M00021  Cysteine biosynthesis, serine => cysteine [PATH:pae00270]\n",
      "            pae_M00034  Methionine salvage pathway [PATH:pae00270]\n",
      "            pae_M00338  Cysteine biosynthesis, homocysteine + serine => cysteine [PATH:pae00270]\n",
      "DBLINKS     GO: 0006534 0006555\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3816  cysE; O-acetylserine synthase [KO:K00640] [EC:2.3.1.30]\n",
      "line in gene PA3816  cysE; O-acetylserine synthase [KO:K00640] [EC:2.3.1.30]\n",
      "            PA2105  acetyltransferase [KO:K00640] [EC:2.3.1.30]\n",
      "line in gene PA2105  acetyltransferase [KO:K00640] [EC:2.3.1.30]\n",
      "            PA1061  hypothetical protein [KO:K01738] [EC:2.5.1.47]\n",
      "line in gene PA1061  hypothetical protein [KO:K01738] [EC:2.5.1.47]\n",
      "            PA2104  cysteine synthase [KO:K01738] [EC:2.5.1.47]\n",
      "line in gene PA2104  cysteine synthase [KO:K01738] [EC:2.5.1.47]\n",
      "            PA2709  cysK; cysteine synthase A [KO:K01738] [EC:2.5.1.47]\n",
      "line in gene PA2709  cysK; cysteine synthase A [KO:K01738] [EC:2.5.1.47]\n",
      "            PA0400  cystathionine gamma-lyase [KO:K01758] [EC:4.4.1.1]\n",
      "line in gene PA0400  cystathionine gamma-lyase [KO:K01758] [EC:4.4.1.1]\n",
      "            PA0399  cystathionine beta-synthase [KO:K01697] [EC:4.2.1.22]\n",
      "line in gene PA0399  cystathionine beta-synthase [KO:K01697] [EC:4.2.1.22]\n",
      "            PA1843  metH; B12-dependent methionine synthase [KO:K00548] [EC:2.1.1.13]\n",
      "line in gene PA1843  metH; B12-dependent methionine synthase [KO:K00548] [EC:2.1.1.13]\n",
      "            PA1927  metE; 5-methyltetrahydropteroyltriglutamate--homocysteine methyltransferase [KO:K00549] [EC:2.1.1.14]\n",
      "line in gene PA1927  metE; 5-methyltetrahydropteroyltriglutamate--homocysteine methyltransferase [KO:K00549] [EC:2.1.1.14]\n",
      "            PA0546  metK; S-adenosylmethionine synthetase [KO:K00789] [EC:2.5.1.6]\n",
      "line in gene PA0546  metK; S-adenosylmethionine synthetase [KO:K00789] [EC:2.5.1.6]\n",
      "            PA0654  speD; S-adenosylmethionine decarboxylase [KO:K01611] [EC:4.1.1.50]\n",
      "line in gene PA0654  speD; S-adenosylmethionine decarboxylase [KO:K01611] [EC:4.1.1.50]\n",
      "            PA4773  hypothetical protein [KO:K01611] [EC:4.1.1.50]\n",
      "line in gene PA4773  hypothetical protein [KO:K01611] [EC:4.1.1.50]\n",
      "            PA1687  speE; polyamine aminopropyltransferase [KO:K00797] [EC:2.5.1.16]\n",
      "line in gene PA1687  speE; polyamine aminopropyltransferase [KO:K00797] [EC:2.5.1.16]\n",
      "            PA4774  hypothetical protein [KO:K00797] [EC:2.5.1.16]\n",
      "line in gene PA4774  hypothetical protein [KO:K00797] [EC:2.5.1.16]\n",
      "            PA1432  lasI; acyl-homoserine-lactone synthase [KO:K13060] [EC:2.3.1.184]\n",
      "line in gene PA1432  lasI; acyl-homoserine-lactone synthase [KO:K13060] [EC:2.3.1.184]\n",
      "            PA3476  rhlI; acyl-homoserine-lactone synthase [KO:K13061] [EC:2.3.1.184]\n",
      "line in gene PA3476  rhlI; acyl-homoserine-lactone synthase [KO:K13061] [EC:2.3.1.184]\n",
      "            PA1469  hypothetical protein [KO:K05810] [EC:2.4.2.1 2.4.2.28 3.5.4.4]\n",
      "line in gene PA1469  hypothetical protein [KO:K05810] [EC:2.4.2.1 2.4.2.28 3.5.4.4]\n",
      "            PA4543  hypothetical protein [KO:K05810] [EC:2.4.2.1 2.4.2.28 3.5.4.4]\n",
      "line in gene PA4543  hypothetical protein [KO:K05810] [EC:2.4.2.1 2.4.2.28 3.5.4.4]\n",
      "            PA3170  N-ethylammeline chlorohydrolase [KO:K12960] [EC:3.5.4.31 3.5.4.28]\n",
      "line in gene PA3170  N-ethylammeline chlorohydrolase [KO:K12960] [EC:3.5.4.31 3.5.4.28]\n",
      "            PA3004  S-methyl-5'-thioinosine phosphorylase [KO:K19696] [EC:2.4.2.44]\n",
      "line in gene PA3004  S-methyl-5'-thioinosine phosphorylase [KO:K19696] [EC:2.4.2.44]\n",
      "            PA3169  methylthioribose-1-phosphate isomerase [KO:K08963] [EC:5.3.1.23]\n",
      "line in gene PA3169  methylthioribose-1-phosphate isomerase [KO:K08963] [EC:5.3.1.23]\n",
      "            PA1683  methylthioribulose-1-phosphate dehydratase [KO:K08964] [EC:4.2.1.109]\n",
      "line in gene PA1683  methylthioribulose-1-phosphate dehydratase [KO:K08964] [EC:4.2.1.109]\n",
      "            PA1685  masA; enolase-phosphatase [KO:K09880] [EC:3.1.3.77]\n",
      "line in gene PA1685  masA; enolase-phosphatase [KO:K09880] [EC:3.1.3.77]\n",
      "            PA1684  acireductone dioxygenase [KO:K08967] [EC:1.13.11.53 1.13.11.54]\n",
      "line in gene PA1684  acireductone dioxygenase [KO:K08967] [EC:1.13.11.53 1.13.11.54]\n",
      "            PA3139  aspartate aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "line in gene PA3139  aspartate aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "            PA0870  phhC; aromatic amino acid aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "line in gene PA0870  phhC; aromatic amino acid aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "            PA2822  hypothetical protein [KO:K08968] [EC:1.8.4.14]\n",
      "line in gene PA2822  hypothetical protein [KO:K08968] [EC:1.8.4.14]\n",
      "            PA0432  sahH; adenosylhomocysteinase [KO:K01251] [EC:3.3.1.1]\n",
      "line in gene PA0432  sahH; adenosylhomocysteinase [KO:K01251] [EC:3.3.1.1]\n",
      "            PA3089a  1-aminocyclopropane-1-carboxylate deaminase [KO:K01505] [EC:3.5.99.7]\n",
      "line in gene PA3089a  1-aminocyclopropane-1-carboxylate deaminase [KO:K01505] [EC:3.5.99.7]\n",
      "            PA0904  lysC; aspartokinase [KO:K00928] [EC:2.7.2.4]\n",
      "line in gene PA0904  lysC; aspartokinase [KO:K00928] [EC:2.7.2.4]\n",
      "            PA3117  asd; aspartate-semialdehyde dehydrogenase [KO:K00133] [EC:1.2.1.11]\n",
      "line in gene PA3117  asd; aspartate-semialdehyde dehydrogenase [KO:K00133] [EC:1.2.1.11]\n",
      "            PA3736  hom; homoserine dehydrogenase [KO:K00003] [EC:1.1.1.3]\n",
      "line in gene PA3736  hom; homoserine dehydrogenase [KO:K00003] [EC:1.1.1.3]\n",
      "            PA0390  metX; homoserine O-acetyltransferase [KO:K00641] [EC:2.3.1.31 2.3.1.46]\n",
      "line in gene PA0390  metX; homoserine O-acetyltransferase [KO:K00641] [EC:2.3.1.31 2.3.1.46]\n",
      "            PA5025  metY; O-acetylhomoserine aminocarboxypropyltransferase [KO:K01740] [EC:2.5.1.49]\n",
      "line in gene PA5025  metY; O-acetylhomoserine aminocarboxypropyltransferase [KO:K01740] [EC:2.5.1.49]\n",
      "            PA3107  metZ; O-succinylhomoserine sulfhydrylase [KO:K10764] [EC:2.5.1.-]\n",
      "line in gene PA3107  metZ; O-succinylhomoserine sulfhydrylase [KO:K10764] [EC:2.5.1.-]\n",
      "            PA5013  ilvE; branched-chain amino acid aminotransferase [KO:K00826] [EC:2.6.1.42]\n",
      "line in gene PA5013  ilvE; branched-chain amino acid aminotransferase [KO:K00826] [EC:2.6.1.42]\n",
      "            PA5203  gshA; glutamate--cysteine ligase [KO:K01919] [EC:6.3.2.2]\n",
      "line in gene PA5203  gshA; glutamate--cysteine ligase [KO:K01919] [EC:6.3.2.2]\n",
      "            PA0407  gshB; glutathione synthetase [KO:K01920] [EC:6.3.2.3]\n",
      "line in gene PA0407  gshB; glutathione synthetase [KO:K01920] [EC:6.3.2.3]\n",
      "            PA1292  3-mercaptopyruvate sulfurtransferase [KO:K01011] [EC:2.8.1.1 2.8.1.2]\n",
      "line in gene PA1292  3-mercaptopyruvate sulfurtransferase [KO:K01011] [EC:2.8.1.1 2.8.1.2]\n",
      "            PA4956  rhdA; thiosulfate:cyanide sulfurtransferase [KO:K01011] [EC:2.8.1.1 2.8.1.2]\n",
      "line in gene PA4956  rhdA; thiosulfate:cyanide sulfurtransferase [KO:K01011] [EC:2.8.1.1 2.8.1.2]\n",
      "            PA2443  sdaA; L-serine dehydratase [KO:K01752] [EC:4.3.1.17]\n",
      "line in gene PA2443  sdaA; L-serine dehydratase [KO:K01752] [EC:4.3.1.17]\n",
      "            PA5379  sdaB; L-serine dehydratase [KO:K01752] [EC:4.3.1.17]\n",
      "line in gene PA5379  sdaB; L-serine dehydratase [KO:K01752] [EC:4.3.1.17]\n",
      "            PA0932  cysM; cysteine synthase B [KO:K12339] [EC:2.5.1.144]\n",
      "line in gene PA0932  cysM; cysteine synthase B [KO:K12339] [EC:2.5.1.144]\n",
      "            PA0316  serA; D-3-phosphoglycerate dehydrogenase [KO:K00058] [EC:1.1.1.95 1.1.1.399]\n",
      "line in gene PA0316  serA; D-3-phosphoglycerate dehydrogenase [KO:K00058] [EC:1.1.1.95 1.1.1.399]\n",
      "            PA3167  serC; 3-phosphoserine/phosphohydroxythreonine aminotransferase [KO:K00831] [EC:2.6.1.52]\n",
      "line in gene PA3167  serC; 3-phosphoserine/phosphohydroxythreonine aminotransferase [KO:K00831] [EC:2.6.1.52]\n",
      "COMPOUND    C00019  S-Adenosyl-L-methionine\n",
      "            C00021  S-Adenosyl-L-homocysteine\n",
      "            C00022  Pyruvate\n",
      "            C00041  L-Alanine\n",
      "            C00049  L-Aspartate\n",
      "            C00051  Glutathione\n",
      "            C00059  Sulfate\n",
      "            C00065  L-Serine\n",
      "            C00073  L-Methionine\n",
      "            C00094  Sulfite\n",
      "            C00097  L-Cysteine\n",
      "            C00109  2-Oxobutanoate\n",
      "            C00155  L-Homocysteine\n",
      "            C00170  5'-Methylthioadenosine\n",
      "            C00197  3-Phospho-D-glycerate\n",
      "            C00263  L-Homoserine\n",
      "            C00283  Hydrogen sulfide\n",
      "            C00409  Methanethiol\n",
      "            C00441  L-Aspartate 4-semialdehyde\n",
      "            C00491  L-Cystine\n",
      "            C00506  L-Cysteate\n",
      "            C00606  3-Sulfino-L-alanine\n",
      "            C00793  D-Cysteine\n",
      "            C00957  Mercaptopyruvate\n",
      "            C00979  O-Acetyl-L-serine\n",
      "            C01005  O-Phospho-L-serine\n",
      "            C01077  O-Acetyl-L-homoserine\n",
      "            C01118  O-Succinyl-L-homoserine\n",
      "            C01137  S-Adenosylmethioninamine\n",
      "            C01180  4-Methylthio-2-oxobutanoic acid\n",
      "            C01234  1-Aminocyclopropane-1-carboxylate\n",
      "            C01817  L-Homocystine\n",
      "            C01962  Thiocysteine\n",
      "            C02218  Dehydroalanine\n",
      "            C02291  L-Cystathionine\n",
      "            C02356  (S)-2-Aminobutanoate\n",
      "            C02989  L-Methionine S-oxide\n",
      "            C03082  4-Phospho-L-aspartate\n",
      "            C03089  5-Methylthio-D-ribose\n",
      "            C03145  N-Formylmethionine\n",
      "            C03232  3-Phosphonooxypyruvate\n",
      "            C03539  S-Ribosyl-L-homocysteine\n",
      "            C04188  S-Methyl-5-thio-D-ribose 1-phosphate\n",
      "            C04582  S-Methyl-5-thio-D-ribulose 1-phosphate\n",
      "            C05324  Nicotianamine\n",
      "            C05524  Aminoacyl-L-methionine\n",
      "            C05526  S-Glutathionyl-L-cysteine\n",
      "            C05527  3-Sulfinylpyruvate\n",
      "            C05528  3-Sulfopyruvate\n",
      "            C05823  3-Mercaptolactate\n",
      "            C05824  S-Sulfo-L-cysteine\n",
      "            C06547  Ethylene\n",
      "            C08276  3-(Methylthio)propanoate\n",
      "            C09306  Sulfur dioxide\n",
      "            C11437  1-Deoxy-D-xylulose 5-phosphate\n",
      "            C11481  HSO3-\n",
      "            C11499  (S)-3-Sulfolactate\n",
      "            C11537  (2R)-3-Sulfolactate\n",
      "            C15606  1,2-Dihydroxy-5-(methylthio)pent-1-en-3-one\n",
      "            C15650  2,3-Diketo-5-methylthiopentyl-1-phosphate\n",
      "            C15651  2-Hydroxy-3-keto-5-methylthiopentenyl-1-phosphate\n",
      "            C18049  N-Acyl-L-homoserine lactone\n",
      "            C19787  5'-S-Methyl-5'-thioinosine\n",
      "            C21015  gamma-L-Glutamyl-L-2-aminobutyrate\n",
      "            C21016  Ophthalmate\n",
      "            C22359  S-Methyl-1-thio-D-xylulose 5-phosphate\n",
      "REFERENCE   PMID:15102328\n",
      "  AUTHORS   Sekowska A, Denervaud V, Ashida H, Michoud K, Haas D, Yokota A, Danchin A\n",
      "  TITLE     Bacterial variations on the methionine salvage pathway.\n",
      "  JOURNAL   BMC Microbiol 4:9 (2004)\n",
      "            DOI:10.1186/1471-2180-4-9\n",
      "REFERENCE   PMID:14551435\n",
      "  AUTHORS   Ashida H, Saito Y, Kojima C, Kobayashi K, Ogasawara N, Yokota A.\n",
      "  TITLE     A functional link between RuBisCO-like protein of Bacillus and photosynthetic RuBisCO.\n",
      "  JOURNAL   Science 302:286-90 (2003)\n",
      "            DOI:10.1126/science.1086997\n",
      "REFERENCE   PMID:12022921\n",
      "  AUTHORS   Sekowska A, Danchin A.\n",
      "  TITLE     The methionine salvage pathway in Bacillus subtilis.\n",
      "  JOURNAL   BMC Microbiol 2:8 (2002)\n",
      "            DOI:10.1186/1471-2180-2-8\n",
      "REFERENCE   PMID:12670965\n",
      "  AUTHORS   Berger BJ, English S, Chan G, Knodel MH.\n",
      "  TITLE     Methionine regeneration and aminotransferases in Bacillus subtilis, Bacillus cereus, and Bacillus anthracis.\n",
      "  JOURNAL   J Bacteriol 185:2418-31 (2003)\n",
      "            DOI:10.1128/JB.185.8.2418-2431.2003\n",
      "REFERENCE   PMID:17169919\n",
      "  AUTHORS   Goyer A, Collakova E, Shachar-Hill Y, Hanson AD\n",
      "  TITLE     Functional characterization of a methionine gamma-lyase in Arabidopsis and its implication in an alternative to the reverse trans-sulfuration pathway.\n",
      "  JOURNAL   Plant Cell Physiol 48:232-42 (2007)\n",
      "            DOI:10.1093/pcp/pcl055\n",
      "REFERENCE   PMID:17030798\n",
      "  AUTHORS   Rebeille F, Jabrin S, Bligny R, Loizeau K, Gambonnet B, Van Wilder V, Douce R, Ravanel S\n",
      "  TITLE     Methionine catabolism in Arabidopsis cells is initiated by a gamma-cleavage process and leads to S-methylcysteine and isoleucine syntheses.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 103:15687-92 (2006)\n",
      "            DOI:10.1073/pnas.0606195103\n",
      "REFERENCE   PMID:18625006\n",
      "  AUTHORS   Pirkov I, Norbeck J, Gustafsson L, Albers E\n",
      "  TITLE     A complete inventory of all enzymes in the eukaryotic methionine salvage pathway.\n",
      "  JOURNAL   FEBS J 275:4111-20 (2008)\n",
      "            DOI:10.1111/j.1742-4658.2008.06552.x\n",
      "REFERENCE   PMID:18391471\n",
      "  AUTHORS   Ashida H, Saito Y, Kojima C, Yokota A\n",
      "  TITLE     Enzymatic characterization of 5-methylthioribulose-1-phosphate dehydratase of the methionine salvage pathway in Bacillus subtilis.\n",
      "  JOURNAL   Biosci Biotechnol Biochem 72:959-67 (2008)\n",
      "            DOI:10.1271/bbb.70651\n",
      "REFERENCE   PMID:10613873\n",
      "  AUTHORS   Kitabatake M, So MW, Tumbula DL, Soll D\n",
      "  TITLE     Cysteine biosynthesis pathway in the archaeon Methanosarcina barkeri encoded by acquired bacterial genes?\n",
      "  JOURNAL   J Bacteriol 182:143-5 (2000)\n",
      "            DOI:10.1128/JB.182.1.143-145.2000\n",
      "REFERENCE   PMID:12644499\n",
      "  AUTHORS   Mino K, Ishikawa K\n",
      "  TITLE     Characterization of a novel thermostable O-acetylserine sulfhydrylase from Aeropyrum pernix K1.\n",
      "  JOURNAL   J Bacteriol 185:2277-84 (2003)\n",
      "            DOI:10.1128/JB.185.7.2277-2284.2003\n",
      "REFERENCE   PMID:18520135\n",
      "  AUTHORS   Tanabe S\n",
      "  TITLE     Development of assay methods for endogenous inorganic sulfur compounds and sulfurtransferases and evaluation of the physiological functions of bound sulfur.\n",
      "  JOURNAL   Yakugaku Zasshi 128:881-900 (2008)\n",
      "            DOI:10.1248/yakushi.128.881\n",
      "REFERENCE   \n",
      "  AUTHORS   Nishizuka Y, Seyama Y, Ikai A, Ishimura Y, Kawaguchi A (eds).\n",
      "  TITLE     [Cellular Functions and Metabolic Maps] (In Japanese)\n",
      "  JOURNAL   Tokyo Kagaku Dojin (1997)\n",
      "REFERENCE   PMID:19270684\n",
      "  AUTHORS   Gutierrez JA, Crowder T, Rinaldo-Matthis A, Ho MC, Almo SC, Schramm VL\n",
      "  TITLE     Transition state analogs of 5'-methylthioadenosine nucleosidase disrupt quorum sensing.\n",
      "  JOURNAL   Nat Chem Biol 5:251-7 (2009)\n",
      "            DOI:10.1038/nchembio.153\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00290  Valine, leucine and isoleucine biosynthesis\n",
      "            pae00430  Taurine and hypotaurine metabolism\n",
      "            pae00480  Glutathione metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00640  Propanoate metabolism\n",
      "            pae00770  Pantothenate and CoA biosynthesis\n",
      "            pae00900  Terpenoid backbone biosynthesis\n",
      "            pae00920  Sulfur metabolism\n",
      "KO_PATHWAY  ko00270\n",
      "///\n",
      "path:pae00280\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00280                    Pathway\n",
      "NAME        Valine, leucine and isoleucine degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00280  Valine, leucine and isoleucine degradation\n",
      "MODULE      pae_M00036  Leucine degradation, leucine => acetoacetate + acetyl-CoA [PATH:pae00280]\n",
      "DBLINKS     GO: 0006550 0006552 0006574\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA5013  ilvE; branched-chain amino acid aminotransferase [KO:K00826] [EC:2.6.1.42]\n",
      "line in gene PA5013  ilvE; branched-chain amino acid aminotransferase [KO:K00826] [EC:2.6.1.42]\n",
      "            PA3418  ldh; leucine dehydrogenase [KO:K00263] [EC:1.4.1.9]\n",
      "line in gene PA3418  ldh; leucine dehydrogenase [KO:K00263] [EC:1.4.1.9]\n",
      "            PA2247  bkdA1; 2-oxoisovalerate dehydrogenase subunit alpha [KO:K00166] [EC:1.2.4.4]\n",
      "line in gene PA2247  bkdA1; 2-oxoisovalerate dehydrogenase subunit alpha [KO:K00166] [EC:1.2.4.4]\n",
      "            PA3417  pyruvate dehydrogenase E1 component subunit alpha [KO:K00166] [EC:1.2.4.4]\n",
      "line in gene PA3417  pyruvate dehydrogenase E1 component subunit alpha [KO:K00166] [EC:1.2.4.4]\n",
      "            PA2248  bkdA2; 2-oxoisovalerate dehydrogenase subunit beta [KO:K00167] [EC:1.2.4.4]\n",
      "line in gene PA2248  bkdA2; 2-oxoisovalerate dehydrogenase subunit beta [KO:K00167] [EC:1.2.4.4]\n",
      "            PA3416  pyruvate dehydrogenase E1 component subunit beta [KO:K00167] [EC:1.2.4.4]\n",
      "line in gene PA3416  pyruvate dehydrogenase E1 component subunit beta [KO:K00167] [EC:1.2.4.4]\n",
      "            PA2249  bkdB; branched-chain alpha-keto acid dehydrogenase complex lipoamide acyltransferase [KO:K09699] [EC:2.3.1.168]\n",
      "line in gene PA2249  bkdB; branched-chain alpha-keto acid dehydrogenase complex lipoamide acyltransferase [KO:K09699] [EC:2.3.1.168]\n",
      "            PA3415  branched-chain alpha-keto acid dehydrogenase subunit E2 [KO:K09699] [EC:2.3.1.168]\n",
      "line in gene PA3415  branched-chain alpha-keto acid dehydrogenase subunit E2 [KO:K09699] [EC:2.3.1.168]\n",
      "            PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA2552  acyl-CoA dehydrogenase [KO:K00248] [EC:1.3.8.1]\n",
      "line in gene PA2552  acyl-CoA dehydrogenase [KO:K00248] [EC:1.3.8.1]\n",
      "            PA2550  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "line in gene PA2550  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "            PA0879  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "line in gene PA0879  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "            PA1535  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "line in gene PA1535  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "            PA4994  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "line in gene PA4994  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "            PA4979  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "line in gene PA4979  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "            PA3593  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "line in gene PA3593  acyl-CoA dehydrogenase [KO:K00249] [EC:1.3.8.7]\n",
      "            PA2015  liuA; isovaleryl-CoA dehydrogenase [KO:K00253] [EC:1.3.8.4]\n",
      "line in gene PA2015  liuA; isovaleryl-CoA dehydrogenase [KO:K00253] [EC:1.3.8.4]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "line in gene PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "            PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "line in gene PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "            PA2940  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA2940  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA3013  foaB; 3-ketoacyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA3013  foaB; 3-ketoacyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA3454  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA3454  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA3569  mmsB; 3-hydroxyisobutyrate dehydrogenase [KO:K00020] [EC:1.1.1.31]\n",
      "line in gene PA3569  mmsB; 3-hydroxyisobutyrate dehydrogenase [KO:K00020] [EC:1.1.1.31]\n",
      "            PA3312  3-hydroxyisobutyrate dehydrogenase [KO:K00020] [EC:1.1.1.31]\n",
      "line in gene PA3312  3-hydroxyisobutyrate dehydrogenase [KO:K00020] [EC:1.1.1.31]\n",
      "            PA1576  3-hydroxyisobutyrate dehydrogenase [KO:K00020] [EC:1.1.1.31]\n",
      "line in gene PA1576  3-hydroxyisobutyrate dehydrogenase [KO:K00020] [EC:1.1.1.31]\n",
      "            PA3570  mmsA; methylmalonate-semialdehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "line in gene PA3570  mmsA; methylmalonate-semialdehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "            PA0747  aldehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "line in gene PA0747  aldehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "            PA0130  3-oxopropanoate dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "line in gene PA0130  3-oxopropanoate dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA0132  beta alanine--pyruvate transaminase [KO:K00822] [EC:2.6.1.18]\n",
      "line in gene PA0132  beta alanine--pyruvate transaminase [KO:K00822] [EC:2.6.1.18]\n",
      "            PA5313  omega amino acid--pyruvate transaminase [KO:K00822] [EC:2.6.1.18]\n",
      "line in gene PA5313  omega amino acid--pyruvate transaminase [KO:K00822] [EC:2.6.1.18]\n",
      "            PA2012  liuD; methylcrotonyl-CoA carboxylase subunit alpha [KO:K01968] [EC:6.4.1.4]\n",
      "line in gene PA2012  liuD; methylcrotonyl-CoA carboxylase subunit alpha [KO:K01968] [EC:6.4.1.4]\n",
      "            PA2014  liuB; methylcrotonyl-CoA carboxylase subunit beta [KO:K01969] [EC:6.4.1.4]\n",
      "line in gene PA2014  liuB; methylcrotonyl-CoA carboxylase subunit beta [KO:K01969] [EC:6.4.1.4]\n",
      "            PA2013  liuC; gamma-carboxygeranoyl-CoA hydratase [KO:K13766] [EC:4.2.1.18]\n",
      "line in gene PA2013  liuC; gamma-carboxygeranoyl-CoA hydratase [KO:K13766] [EC:4.2.1.18]\n",
      "            PA2011  liuE; 3-hydroxy-3-isohexenylglutaryl-CoA/hydroxy-methylglutaryl-CoA lyase [KO:K01640] [EC:4.1.3.4]\n",
      "line in gene PA2011  liuE; 3-hydroxy-3-isohexenylglutaryl-CoA/hydroxy-methylglutaryl-CoA lyase [KO:K01640] [EC:4.1.3.4]\n",
      "            PA1999  dhcA; dehydrocarnitine CoA transferase subunit A [KO:K01028] [EC:2.8.3.5]\n",
      "line in gene PA1999  dhcA; dehydrocarnitine CoA transferase subunit A [KO:K01028] [EC:2.8.3.5]\n",
      "            PA2000  dhcB; dehydrocarnitine CoA transferase subunit B [KO:K01029] [EC:2.8.3.5]\n",
      "line in gene PA2000  dhcB; dehydrocarnitine CoA transferase subunit B [KO:K01029] [EC:2.8.3.5]\n",
      "            PA1997  acetoacetyl-CoA synthetase [KO:K01907] [EC:6.2.1.16]\n",
      "line in gene PA1997  acetoacetyl-CoA synthetase [KO:K01907] [EC:6.2.1.16]\n",
      "            PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "COMPOUND    C00024  Acetyl-CoA\n",
      "            C00068  Thiamin diphosphate\n",
      "            C00091  Succinyl-CoA\n",
      "            C00100  Propanoyl-CoA\n",
      "            C00123  L-Leucine\n",
      "            C00141  3-Methyl-2-oxobutanoic acid\n",
      "            C00164  Acetoacetate\n",
      "            C00183  L-Valine\n",
      "            C00233  4-Methyl-2-oxopentanoate\n",
      "            C00332  Acetoacetyl-CoA\n",
      "            C00349  2-Methyl-3-oxopropanoate\n",
      "            C00356  (S)-3-Hydroxy-3-methylglutaryl-CoA\n",
      "            C00407  L-Isoleucine\n",
      "            C00630  2-Methylpropanoyl-CoA\n",
      "            C00671  (S)-3-Methyl-2-oxopentanoic acid\n",
      "            C00683  (S)-Methylmalonyl-CoA\n",
      "            C01205  (R)-3-Amino-2-methylpropanoate\n",
      "            C01213  (R)-Methylmalonyl-CoA\n",
      "            C02170  Methylmalonate\n",
      "            C02939  3-Methylbutanoyl-CoA\n",
      "            C03069  3-Methylcrotonyl-CoA\n",
      "            C03231  3-Methylglutaconyl-CoA\n",
      "            C03284  L-3-Aminoisobutanoate\n",
      "            C03344  2-Methylacetoacetyl-CoA\n",
      "            C03345  2-Methylbut-2-enoyl-CoA\n",
      "            C03460  2-Methylprop-2-enoyl-CoA\n",
      "            C04405  (2S,3S)-3-Hydroxy-2-methylbutanoyl-CoA\n",
      "            C05996  Branched chain fatty acid\n",
      "            C05998  3-Hydroxyisovaleryl-CoA\n",
      "            C06000  (S)-3-Hydroxyisobutyryl-CoA\n",
      "            C06001  (S)-3-Hydroxyisobutyrate\n",
      "            C06002  (S)-Methylmalonate semialdehyde\n",
      "            C15972  Enzyme N6-(lipoyl)lysine\n",
      "            C15973  Enzyme N6-(dihydrolipoyl)lysine\n",
      "            C15974  3-Methyl-1-hydroxybutyl-ThPP\n",
      "            C15975  [Dihydrolipoyllysine-residue (2-methylpropanoyl)transferase] S-(3-methylbutanoyl)dihydrolipoyllysine\n",
      "            C15976  2-Methyl-1-hydroxypropyl-ThPP\n",
      "            C15977  [Dihydrolipoyllysine-residue (2-methylpropanoyl)transferase] S-(2-methylpropanoyl)dihydrolipoyllysine\n",
      "            C15978  2-Methyl-1-hydroxybutyl-ThPP\n",
      "            C15979  [Dihydrolipoyllysine-residue (2-methylpropanoyl)transferase] S-(2-methylbutanoyl)dihydrolipoyllysine\n",
      "            C15980  (S)-2-Methylbutanoyl-CoA\n",
      "            C20827  3-Hydroxyisovalerate\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00240  Pyrimidine metabolism\n",
      "            pae00290  Valine, leucine and isoleucine biosynthesis\n",
      "            pae00640  Propanoate metabolism\n",
      "            pae00900  Terpenoid backbone biosynthesis\n",
      "KO_PATHWAY  ko00280\n",
      "///\n",
      "path:pae00281\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00281                    Pathway\n",
      "NAME        Geraniol degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of terpenoids and polyketides\n",
      "PATHWAY_MAP pae00281  Geraniol degradation\n",
      "DBLINKS     UMBBD: ger cit\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2887  atuB; citronellol catabolism dehydrogenase [KO:K13774]\n",
      "line in gene PA2887  atuB; citronellol catabolism dehydrogenase [KO:K13774]\n",
      "            PA2892  atuG; short-chain dehydrogenase [KO:K13775]\n",
      "line in gene PA2892  atuG; short-chain dehydrogenase [KO:K13775]\n",
      "            PA2893  atuH; long-chain-acyl-CoA synthetase [KO:K13776] [EC:6.2.1.-]\n",
      "line in gene PA2893  atuH; long-chain-acyl-CoA synthetase [KO:K13776] [EC:6.2.1.-]\n",
      "            PA2889  atuD; citronellyl-CoA dehydrogenase [KO:K11731] [EC:1.3.99.-]\n",
      "line in gene PA2889  atuD; citronellyl-CoA dehydrogenase [KO:K11731] [EC:1.3.99.-]\n",
      "            PA2891  atuF; geranyl-CoA carboxylase subunit alpha [KO:K13777] [EC:6.4.1.5]\n",
      "line in gene PA2891  atuF; geranyl-CoA carboxylase subunit alpha [KO:K13777] [EC:6.4.1.5]\n",
      "            PA2888  atuC; geranyl-CoA carboxylase subunit beta [KO:K13778] [EC:6.4.1.5]\n",
      "line in gene PA2888  atuC; geranyl-CoA carboxylase subunit beta [KO:K13778] [EC:6.4.1.5]\n",
      "            PA2890  atuE; isohexenylglutaconyl-CoA hydratase [KO:K13779] [EC:4.2.1.57]\n",
      "line in gene PA2890  atuE; isohexenylglutaconyl-CoA hydratase [KO:K13779] [EC:4.2.1.57]\n",
      "            PA2011  liuE; 3-hydroxy-3-isohexenylglutaryl-CoA/hydroxy-methylglutaryl-CoA lyase [KO:K01640] [EC:4.1.3.4]\n",
      "line in gene PA2011  liuE; 3-hydroxy-3-isohexenylglutaryl-CoA/hydroxy-methylglutaryl-CoA lyase [KO:K01640] [EC:4.1.3.4]\n",
      "            PA2940  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA2940  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA3013  foaB; 3-ketoacyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA3013  foaB; 3-ketoacyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA3454  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA3454  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "line in gene PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "            PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "line in gene PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "COMPOUND    C01291  3-(4-Methylpent-3-en-1-yl)pent-2-enedioyl-CoA\n",
      "            C01499  Geranial\n",
      "            C01500  Geraniol\n",
      "            C01920  Geranoyl-CoA\n",
      "            C03069  3-Methylcrotonyl-CoA\n",
      "            C04675  3-Hydroxy-3-(4-methylpent-3-en-1-yl)glutaryl-CoA\n",
      "            C09847  cis-Citral\n",
      "            C09848  (R)-(+)-Citronellal\n",
      "            C09849  beta-Citronellol\n",
      "            C09871  Nerol\n",
      "            C11384  (S)-(-)-Citronellal\n",
      "            C11386  (-)-Citronellol\n",
      "            C16461  Geranic acid\n",
      "            C16462  Citronellate\n",
      "            C16464  Citronellyl-CoA\n",
      "            C16465  trans-Geranyl-CoA\n",
      "            C16466  7-Methyl-3-oxo-6-octenoyl-CoA\n",
      "            C16468  (2E)-5-Methylhexa-2,4-dienoyl-CoA\n",
      "            C16469  3-Hydroxy-5-methylhex-4-enoyl-CoA\n",
      "            C16470  5-Methylhex-4-enoyl-CoA\n",
      "            C16471  5-Methyl-3-oxo-4-hexenoyl-CoA\n",
      "REFERENCE   PMID:11778886\n",
      "  AUTHORS   Wolken WA, van der Werf MJ.\n",
      "  TITLE     Geraniol biotransformation-pathway in spores of Penicillium digitatum.\n",
      "  JOURNAL   Appl Microbiol Biotechnol 57:731-7 (2001)\n",
      "            DOI:10.1007/s002530100821\n",
      "REFERENCE   PMID:16272386\n",
      "  AUTHORS   Hoschle B, Gnau V, Jendrossek D.\n",
      "  TITLE     Methylcrotonyl-CoA and geranyl-CoA carboxylases are involved in leucine/isovalerate utilization (Liu) and acyclic terpene utilization (Atu), and are encoded by liuB/liuD and atuC/atuF, in Pseudomonas aeruginosa.\n",
      "  JOURNAL   Microbiology 151:3649-56 (2005)\n",
      "            DOI:10.1099/mic.0.28260-0\n",
      "REFERENCE   PMID:16820476\n",
      "  AUTHORS   Forster-Fromme K, Hoschle B, Mack C, Bott M, Armbruster W, Jendrossek D\n",
      "  TITLE     Identification of genes and proteins necessary for catabolism of acyclic terpenes and leucine/isovalerate in Pseudomonas aeruginosa.\n",
      "  JOURNAL   Appl Environ Microbiol 72:4819-28 (2006)\n",
      "            DOI:10.1128/AEM.00853-06\n",
      "REFERENCE   PMID:16000717\n",
      "  AUTHORS   Hoschle B, Jendrossek D.\n",
      "  TITLE     Utilization of geraniol is dependent on molybdenum in Pseudomonas aeruginosa: evidence for different metabolic routes for oxidation of geraniol and citronellol.\n",
      "  JOURNAL   Microbiology 151:2277-83 (2005)\n",
      "            DOI:10.1099/mic.0.27957-0\n",
      "REFERENCE   PMID:681275\n",
      "  AUTHORS   Cantwell SG, Lau EP, Watt DS, Fall RR.\n",
      "  TITLE     Biodegradation of acyclic isoprenoids by Pseudomonas species.\n",
      "  JOURNAL   J Bacteriol 135:324-33 (1978)\n",
      "            DOI:10.1128/JB.135.2.324-333.1978\n",
      "REFERENCE   PMID:19597963\n",
      "  AUTHORS   Chavez-Aviles M, Diaz-Perez AL, Campos-Garcia J\n",
      "  TITLE     The bifunctional role of LiuE from Pseudomonas aeruginosa, displays additionally HIHG-CoA lyase enzymatic activity.\n",
      "  JOURNAL   Mol Biol Rep 37:1787-91 (2010)\n",
      "            DOI:10.1007/s11033-009-9611-6\n",
      "REL_PATHWAY pae00280  Valine, leucine and isoleucine degradation\n",
      "KO_PATHWAY  ko00281\n",
      "///\n",
      "path:pae00290\n",
      "ENTRY       pae00290                    Pathway\n",
      "NAME        Valine, leucine and isoleucine biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00290  Valine, leucine and isoleucine biosynthesis\n",
      "MODULE      pae_M00019  Valine/isoleucine biosynthesis, pyruvate => valine / 2-oxobutanoate => isoleucine [PATH:pae00290]\n",
      "            pae_M00432  Leucine biosynthesis, 2-oxoisovalerate => 2-oxoisocaproate [PATH:pae00290]\n",
      "            pae_M00570  Isoleucine biosynthesis, threonine => 2-oxobutanoate => isoleucine [PATH:pae00290]\n",
      "DBLINKS     GO: 0009097 0009098 0009099\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0331  ilvA1; threonine dehydratase [KO:K01754] [EC:4.3.1.19]\n",
      "line in gene PA0331  ilvA1; threonine dehydratase [KO:K01754] [EC:4.3.1.19]\n",
      "            PA1326  ilvA2; threonine dehydratase [KO:K01754] [EC:4.3.1.19]\n",
      "line in gene PA1326  ilvA2; threonine dehydratase [KO:K01754] [EC:4.3.1.19]\n",
      "            PA0851  hypothetical protein [KO:K01754] [EC:4.3.1.19]\n",
      "line in gene PA0851  hypothetical protein [KO:K01754] [EC:4.3.1.19]\n",
      "            PA3121  leuC; 3-isopropylmalate dehydratase large subunit [KO:K01703] [EC:4.2.1.33 4.2.1.35]\n",
      "line in gene PA3121  leuC; 3-isopropylmalate dehydratase large subunit [KO:K01703] [EC:4.2.1.33 4.2.1.35]\n",
      "            PA3120  leuD; isopropylmalate isomerase small subunit [KO:K01704] [EC:4.2.1.33 4.2.1.35]\n",
      "line in gene PA3120  leuD; isopropylmalate isomerase small subunit [KO:K01704] [EC:4.2.1.33 4.2.1.35]\n",
      "            PA3118  leuB; 3-isopropylmalate dehydrogenase [KO:K00052] [EC:1.1.1.85]\n",
      "line in gene PA3118  leuB; 3-isopropylmalate dehydrogenase [KO:K00052] [EC:1.1.1.85]\n",
      "            PA2035  thiamine pyrophosphate protein [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA2035  thiamine pyrophosphate protein [KO:K01652] [EC:2.2.1.6]\n",
      "            PA4180  acetolactate synthase [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA4180  acetolactate synthase [KO:K01652] [EC:2.2.1.6]\n",
      "            PA4696  ilvI; acetolactate synthase 3 catalytic subunit [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA4696  ilvI; acetolactate synthase 3 catalytic subunit [KO:K01652] [EC:2.2.1.6]\n",
      "            PA1417  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA1417  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "            PA3506  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA3506  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "            PA4695  ilvH; acetolactate synthase small subunit [KO:K01653] [EC:2.2.1.6]\n",
      "line in gene PA4695  ilvH; acetolactate synthase small subunit [KO:K01653] [EC:2.2.1.6]\n",
      "            PA4694  ilvC; ketol-acid reductoisomerase [KO:K00053] [EC:1.1.1.86]\n",
      "line in gene PA4694  ilvC; ketol-acid reductoisomerase [KO:K00053] [EC:1.1.1.86]\n",
      "            PA0353  ilvD; dihydroxy-acid dehydratase [KO:K01687] [EC:4.2.1.9]\n",
      "line in gene PA0353  ilvD; dihydroxy-acid dehydratase [KO:K01687] [EC:4.2.1.9]\n",
      "            PA5013  ilvE; branched-chain amino acid aminotransferase [KO:K00826] [EC:2.6.1.42]\n",
      "line in gene PA5013  ilvE; branched-chain amino acid aminotransferase [KO:K00826] [EC:2.6.1.42]\n",
      "            PA3418  ldh; leucine dehydrogenase [KO:K00263] [EC:1.4.1.9]\n",
      "line in gene PA3418  ldh; leucine dehydrogenase [KO:K00263] [EC:1.4.1.9]\n",
      "            PA2828  aminotransferase [KO:K14260] [EC:2.6.1.66 2.6.1.2]\n",
      "line in gene PA2828  aminotransferase [KO:K14260] [EC:2.6.1.66 2.6.1.2]\n",
      "            PA1217  2-isopropylmalate synthase [KO:K01649] [EC:2.3.3.13]\n",
      "line in gene PA1217  2-isopropylmalate synthase [KO:K01649] [EC:2.3.3.13]\n",
      "            PA3792  leuA; 2-isopropylmalate synthase [KO:K01649] [EC:2.3.3.13]\n",
      "line in gene PA3792  leuA; 2-isopropylmalate synthase [KO:K01649] [EC:2.3.3.13]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00109  2-Oxobutanoate\n",
      "            C00123  L-Leucine\n",
      "            C00141  3-Methyl-2-oxobutanoic acid\n",
      "            C00183  L-Valine\n",
      "            C00188  L-Threonine\n",
      "            C00233  4-Methyl-2-oxopentanoate\n",
      "            C00407  L-Isoleucine\n",
      "            C00671  (S)-3-Methyl-2-oxopentanoic acid\n",
      "            C02226  2-Methylmaleate\n",
      "            C02504  alpha-Isopropylmalate\n",
      "            C02612  (R)-2-Methylmalate\n",
      "            C02631  2-Isopropylmaleate\n",
      "            C04181  3-Hydroxy-3-methyl-2-oxobutanoic acid\n",
      "            C04236  (2S)-2-Isopropyl-3-oxosuccinate\n",
      "            C04272  (R)-2,3-Dihydroxy-3-methylbutanoate\n",
      "            C04411  (2R,3S)-3-Isopropylmalate\n",
      "            C06006  (S)-2-Aceto-2-hydroxybutanoate\n",
      "            C06007  (R)-2,3-Dihydroxy-3-methylpentanoate\n",
      "            C06010  (S)-2-Acetolactate\n",
      "            C06032  D-erythro-3-Methylmalate\n",
      "            C14463  (R)-3-Hydroxy-3-methyl-2-oxopentanoate\n",
      "REFERENCE   PMID:15292141\n",
      "  AUTHORS   Xu H, Zhang Y, Guo X, Ren S, Staempfli AA, Chiao J, Jiang W, Zhao G.\n",
      "  TITLE     Isoleucine biosynthesis in Leptospira interrogans serotype lai strain 56601 proceeds via a threonine-independent pathway.\n",
      "  JOURNAL   J Bacteriol 186:5400-9 (2004)\n",
      "            DOI:10.1128/JB.186.16.5400-5409.2004\n",
      "REL_PATHWAY pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00280  Valine, leucine and isoleucine degradation\n",
      "            pae00620  Pyruvate metabolism\n",
      "KO_PATHWAY  ko00290\n",
      "///\n",
      "path:pae00300\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00300                    Pathway\n",
      "NAME        Lysine biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00300  Lysine biosynthesis\n",
      "MODULE      pae_M00016  Lysine biosynthesis, succinyl-DAP pathway, aspartate => lysine [PATH:pae00300]\n",
      "DBLINKS     GO: 0009085\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3736  hom; homoserine dehydrogenase [KO:K00003] [EC:1.1.1.3]\n",
      "line in gene PA3736  hom; homoserine dehydrogenase [KO:K00003] [EC:1.1.1.3]\n",
      "            PA0904  lysC; aspartokinase [KO:K00928] [EC:2.7.2.4]\n",
      "line in gene PA0904  lysC; aspartokinase [KO:K00928] [EC:2.7.2.4]\n",
      "            PA3117  asd; aspartate-semialdehyde dehydrogenase [KO:K00133] [EC:1.2.1.11]\n",
      "line in gene PA3117  asd; aspartate-semialdehyde dehydrogenase [KO:K00133] [EC:1.2.1.11]\n",
      "            PA0223  dihydrodipicolinate synthetase [KO:K01714] [EC:4.3.3.7]\n",
      "line in gene PA0223  dihydrodipicolinate synthetase [KO:K01714] [EC:4.3.3.7]\n",
      "            PA1010  dapA; 4-hydroxy-tetrahydrodipicolinate synthase [KO:K01714] [EC:4.3.3.7]\n",
      "line in gene PA1010  dapA; 4-hydroxy-tetrahydrodipicolinate synthase [KO:K01714] [EC:4.3.3.7]\n",
      "            PA4188  hypothetical protein [KO:K01714] [EC:4.3.3.7]\n",
      "line in gene PA4188  hypothetical protein [KO:K01714] [EC:4.3.3.7]\n",
      "            PA4759  dapB; 4-hydroxy-tetrahydrodipicolinate reductase [KO:K00215] [EC:1.17.1.8]\n",
      "line in gene PA4759  dapB; 4-hydroxy-tetrahydrodipicolinate reductase [KO:K00215] [EC:1.17.1.8]\n",
      "            PA3666  dapD; 2,3,4,5-tetrahydropyridine-2,6-dicarboxylate N-succinyltransferase [KO:K00674] [EC:2.3.1.117]\n",
      "line in gene PA3666  dapD; 2,3,4,5-tetrahydropyridine-2,6-dicarboxylate N-succinyltransferase [KO:K00674] [EC:2.3.1.117]\n",
      "            PA0530  class III pyridoxal phosphate-dependent aminotransferase [KO:K00821] [EC:2.6.1.11 2.6.1.17]\n",
      "line in gene PA0530  class III pyridoxal phosphate-dependent aminotransferase [KO:K00821] [EC:2.6.1.11 2.6.1.17]\n",
      "            PA3659  succinyldiaminopimelate transaminase [KO:K14267] [EC:2.6.1.17]\n",
      "line in gene PA3659  succinyldiaminopimelate transaminase [KO:K14267] [EC:2.6.1.17]\n",
      "            PA1162  dapE; succinyl-diaminopimelate desuccinylase [KO:K01439] [EC:3.5.1.18]\n",
      "line in gene PA1162  dapE; succinyl-diaminopimelate desuccinylase [KO:K01439] [EC:3.5.1.18]\n",
      "            PA5278  dapF; diaminopimelate epimerase [KO:K01778] [EC:5.1.1.7]\n",
      "line in gene PA5278  dapF; diaminopimelate epimerase [KO:K01778] [EC:5.1.1.7]\n",
      "            PA5277  lysA; diaminopimelate decarboxylase [KO:K01586] [EC:4.1.1.20]\n",
      "line in gene PA5277  lysA; diaminopimelate decarboxylase [KO:K01586] [EC:4.1.1.20]\n",
      "            PA1654  aminotransferase [KO:K05825] [EC:2.6.1.-]\n",
      "line in gene PA1654  aminotransferase [KO:K05825] [EC:2.6.1.-]\n",
      "            PA4417  murE; UDP-N-acetylmuramoylalanyl-D-glutamate--2,6-diaminopimelate ligase [KO:K01928] [EC:6.3.2.13]\n",
      "line in gene PA4417  murE; UDP-N-acetylmuramoylalanyl-D-glutamate--2,6-diaminopimelate ligase [KO:K01928] [EC:6.3.2.13]\n",
      "            PA4416  murF; UDP-N-acetylmuramoyl-tripeptide--D-alanyl-D-alanine ligase [KO:K01929] [EC:6.3.2.10]\n",
      "line in gene PA4416  murF; UDP-N-acetylmuramoyl-tripeptide--D-alanyl-D-alanine ligase [KO:K01929] [EC:6.3.2.10]\n",
      "COMPOUND    C00024  Acetyl-CoA\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00047  L-Lysine\n",
      "            C00049  L-Aspartate\n",
      "            C00263  L-Homoserine\n",
      "            C00322  2-Oxoadipate\n",
      "            C00441  L-Aspartate 4-semialdehyde\n",
      "            C00449  N6-(L-1,3-Dicarboxypropyl)-L-lysine\n",
      "            C00666  LL-2,6-Diaminoheptanedioate\n",
      "            C00680  meso-2,6-Diaminoheptanedioate\n",
      "            C00956  L-2-Aminoadipate\n",
      "            C01251  (R)-2-Hydroxybutane-1,2,4-tricarboxylate\n",
      "            C03082  4-Phospho-L-aspartate\n",
      "            C03871  L-2-Amino-6-oxoheptanedioate\n",
      "            C03972  2,3,4,5-Tetrahydrodipicolinate\n",
      "            C04002  (Z)-But-1-ene-1,2,4-tricarboxylate\n",
      "            C04076  L-2-Aminoadipate 6-semialdehyde\n",
      "            C04390  N6-Acetyl-LL-2,6-diaminoheptanedioate\n",
      "            C04421  N-Succinyl-LL-2,6-diaminoheptanedioate\n",
      "            C04462  N-Succinyl-2-L-amino-6-oxoheptanedioate\n",
      "            C04877  UDP-N-acetylmuramoyl-L-alanyl-gamma-D-glutamyl-meso-2,6-diaminopimelate\n",
      "            C04882  UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-6-carboxy-L-lysyl-D-alanyl-D-alanine\n",
      "            C05535  alpha-Aminoadipoyl-S-acyl enzyme\n",
      "            C05539  N-Acetyl-L-2-amino-6-oxopimelate\n",
      "            C05560  L-2-Aminoadipate adenylate\n",
      "            C05662  Homoisocitrate\n",
      "            C16138  L-Pyrrolysine\n",
      "            C19886  LysW-gamma-L-alpha-aminoadipate\n",
      "            C19887  LysW-gamma-L-alpha-aminoadipyl 6-phosphate\n",
      "            C19888  LysW-gamma-L-alpha-aminoadipate 6-semialdehyde\n",
      "            C19889  LysW-gamma-L-lysine\n",
      "            C20258  (2S,4S)-4-Hydroxy-2,3,4,5-tetrahydrodipicolinate\n",
      "            C20277  (2R,3R)-3-Methylornithine\n",
      "            C20278  (2R,3R)-3-Methylornithinyl-N6-lysine\n",
      "            C20279  (2R,3R)-3-Methylglutamyl-5-semialdehyde-N6-lysine\n",
      "REFERENCE   PMID:10074354\n",
      "  AUTHORS   Ledwidge R, Blanchard JS.\n",
      "  TITLE     The dual biosynthetic capability of N-acetylornithine aminotransferase in arginine and lysine biosynthesis.\n",
      "  JOURNAL   Biochemistry 38:3019-3024 (1999)\n",
      "            DOI:10.1021/bi982574a\n",
      "REFERENCE   PMID:10850974\n",
      "  AUTHORS   Fuchs TM, Schneider B, Krumbach K, Eggeling L, Gross R.\n",
      "  TITLE     Characterization of a bordetella pertussis diaminopimelate (DAP) biosynthesis locus identifies dapC, a novel gene coding for an N-succinyl-L,L-DAP aminotransferase.\n",
      "  JOURNAL   J Bacteriol 182:3626-3631 (2000)\n",
      "            DOI:10.1128/JB.182.13.3626-3631.2000\n",
      "REFERENCE   PMID:10714900\n",
      "  AUTHORS   Zabriskie TM, Jackson MD.\n",
      "  TITLE     Lysine biosynthesis and metabolism in fungi.\n",
      "  JOURNAL   Nat Prod Rep 17:85-97 (2000)\n",
      "            DOI:10.1039/A801345D\n",
      "REFERENCE   PMID:14627808\n",
      "  AUTHORS   Rodionov DA, Vitreschak AG, Mironov AA, Gelfand MS.\n",
      "  TITLE     Regulation of lysine biosynthesis and transport genes in bacteria: yet another RNA riboswitch?\n",
      "  JOURNAL   Nucleic Acids Res 31:6748-57 (2003)\n",
      "            DOI:10.1093/nar/gkg900\n",
      "REFERENCE   PMID:12355264\n",
      "  AUTHORS   Velasco AM, Leguina JI, Lazcano A.\n",
      "  TITLE     Molecular evolution of the lysine biosynthetic pathways.\n",
      "  JOURNAL   J Mol Evol 55:445-59 (2002)\n",
      "            DOI:10.1007/s00239-002-2340-2\n",
      "REFERENCE   PMID:15256574\n",
      "  AUTHORS   Miyazaki T, Miyazaki J, Yamane H, Nishiyama M.\n",
      "  TITLE     alpha-Aminoadipate aminotransferase from an extremely thermophilic bacterium, Thermus thermophilus.\n",
      "  JOURNAL   Microbiology 150:2327-34 (2004)\n",
      "            DOI:10.1099/mic.0.27037-0\n",
      "REFERENCE   PMID:12126930\n",
      "  AUTHORS   Goh DL, Patel A, Thomas GH, Salomons GS, Schor DS, Jakobs C, Geraghty MT.\n",
      "  TITLE     Characterization of the human gene encoding alpha-aminoadipate aminotransferase (AADAT).\n",
      "  JOURNAL   Mol Genet Metab 76:172-80 (2002)\n",
      "            DOI:10.1016/S1096-7192(02)00037-9\n",
      "REFERENCE   PMID:15160885\n",
      "  AUTHORS   Nishiyama M.\n",
      "  TITLE     [Novel lysine biosynthesis: a key to elucidate evolution of metabolic and biosynthetic pathways]\n",
      "  JOURNAL   Tanpakushitsu Kakusan Koso 49:758-63 (2004)\n",
      "REFERENCE   PMID:10613839\n",
      "  AUTHORS   Nishida H, Nishiyama M, Kobashi N, Kosuge T, Hoshino T, Yamane H.\n",
      "  TITLE     A prokaryotic gene cluster involved in synthesis of lysine through the amino adipate pathway: a key to the evolution of amino acid biosynthesis.\n",
      "  JOURNAL   Genome Res 9:1175-83 (1999)\n",
      "            DOI:10.1101/gr.9.12.1175\n",
      "REFERENCE   PMID:12963379\n",
      "  AUTHORS   Sakai H, Vassylyeva MN, Matsuura T, Sekine S, Gotoh K, Nishiyama M, Terada T, Shirouzu M, Kuramitsu S, Vassylyev DG, Yokoyama S.\n",
      "  TITLE     Crystal structure of a lysine biosynthesis enzyme, LysX, from Thermus thermophilus HB8.\n",
      "  JOURNAL   J Mol Biol 332:729-40 (2003)\n",
      "            DOI:10.1016/S0022-2836(03)00946-X\n",
      "REFERENCE   PMID:16361515\n",
      "  AUTHORS   Hudson AO, Singh BK, Leustek T, Gilvarg C.\n",
      "  TITLE     An LL-diaminopimelate aminotransferase defines a novel variant of the lysine biosynthesis pathway in plants.\n",
      "  JOURNAL   Plant Physiol 140:292-301 (2006)\n",
      "            DOI:10.1104/pp.105.072629\n",
      "REFERENCE   PMID:19620981\n",
      "  AUTHORS   Horie A, Tomita T, Saiki A, Kono H, Taka H, Mineki R, Fujimura T, Nishiyama C, Kuzuyama T, Nishiyama M\n",
      "  TITLE     Discovery of proteinaceous N-modification in lysine biosynthesis of Thermus thermophilus.\n",
      "  JOURNAL   Nat Chem Biol 5:673-9 (2009)\n",
      "            DOI:10.1038/nchembio.198\n",
      "REFERENCE   PMID:21547300\n",
      "  AUTHORS   Dairi T, Kuzuyama T, Nishiyama M, Fujii I\n",
      "  TITLE     Convergent strategies in biosynthesis.\n",
      "  JOURNAL   Nat Prod Rep 28:1054-86 (2011)\n",
      "            DOI:10.1039/c0np00047g\n",
      "REFERENCE   PMID:21550296\n",
      "  AUTHORS   Gaston MA, Jiang R, Krzycki JA\n",
      "  TITLE     Functional context, biosynthesis, and genetic encoding of pyrrolysine.\n",
      "  JOURNAL   Curr Opin Microbiol 14:342-9 (2011)\n",
      "            DOI:10.1016/j.mib.2011.04.001\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00310  Lysine degradation\n",
      "            pae00550  Peptidoglycan biosynthesis\n",
      "            pae00620  Pyruvate metabolism\n",
      "KO_PATHWAY  ko00300\n",
      "///\n",
      "path:pae00310\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00310                    Pathway\n",
      "NAME        Lysine degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00310  Lysine degradation\n",
      "DBLINKS     GO: 0006554\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1586  sucB; 2-oxoglutarate dehydrogenase complex dihydrolipoyllysine-residue succinyltransferase [KO:K00658] [EC:2.3.1.61]\n",
      "line in gene PA1586  sucB; 2-oxoglutarate dehydrogenase complex dihydrolipoyllysine-residue succinyltransferase [KO:K00658] [EC:2.3.1.61]\n",
      "            PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA0447  gcdH; glutaryl-CoA dehydrogenase [KO:K00252] [EC:1.3.8.6]\n",
      "line in gene PA0447  gcdH; glutaryl-CoA dehydrogenase [KO:K00252] [EC:1.3.8.6]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "line in gene PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "            PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "line in gene PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "            PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA2074  hypothetical protein [KO:K01506] [EC:3.5.1.30]\n",
      "line in gene PA2074  hypothetical protein [KO:K01506] [EC:3.5.1.30]\n",
      "            PA0266  5-aminovalerate aminotransferase DavT [KO:K14268] [EC:2.6.1.48 2.6.1.19]\n",
      "line in gene PA0266  5-aminovalerate aminotransferase DavT [KO:K14268] [EC:2.6.1.48 2.6.1.19]\n",
      "            PA0265  gabD; glutarate-semialdehyde dehydrogenase DavD [KO:K00135] [EC:1.2.1.16 1.2.1.79 1.2.1.20]\n",
      "line in gene PA0265  gabD; glutarate-semialdehyde dehydrogenase DavD [KO:K00135] [EC:1.2.1.16 1.2.1.79 1.2.1.20]\n",
      "            PA1252  bifunctional Delta(1)-pyrroline-2-carboxylate/Delta(1)-piperideine-2-carboxylate reductase [KO:K13609] [EC:1.5.1.21]\n",
      "line in gene PA1252  bifunctional Delta(1)-pyrroline-2-carboxylate/Delta(1)-piperideine-2-carboxylate reductase [KO:K13609] [EC:1.5.1.21]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "COMPOUND    C00024  Acetyl-CoA\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00037  Glycine\n",
      "            C00042  Succinate\n",
      "            C00047  L-Lysine\n",
      "            C00136  Butanoyl-CoA\n",
      "            C00164  Acetoacetate\n",
      "            C00322  2-Oxoadipate\n",
      "            C00332  Acetoacetyl-CoA\n",
      "            C00408  L-Pipecolate\n",
      "            C00431  5-Aminopentanoate\n",
      "            C00449  N6-(L-1,3-Dicarboxypropyl)-L-lysine\n",
      "            C00450  (S)-2,3,4,5-Tetrahydropyridine-2-carboxylate\n",
      "            C00487  Carnitine\n",
      "            C00489  Glutarate\n",
      "            C00527  Glutaryl-CoA\n",
      "            C00739  D-Lysine\n",
      "            C00877  Crotonoyl-CoA\n",
      "            C00956  L-2-Aminoadipate\n",
      "            C00990  5-Aminopentanamide\n",
      "            C01142  (3S)-3,6-Diaminohexanoate\n",
      "            C01144  (S)-3-Hydroxybutanoyl-CoA\n",
      "            C01149  4-Trimethylammoniobutanal\n",
      "            C01181  4-Trimethylammoniobutanoate\n",
      "            C01186  (3S,5S)-3,5-Diaminohexanoate\n",
      "            C01211  Procollagen 5-hydroxy-L-lysine\n",
      "            C01259  (3S)-3-Hydroxy-N6,N6,N6-trimethyl-L-lysine\n",
      "            C01672  Cadaverine\n",
      "            C02188  Protein lysine\n",
      "            C02727  N6-Acetyl-L-lysine\n",
      "            C03087  5-Acetamidopentanoate\n",
      "            C03196  (S)-2-Hydroxyglutarate\n",
      "            C03239  6-Amino-2-oxohexanoate\n",
      "            C03273  5-Oxopentanoate\n",
      "            C03366  5-Phosphooxy-L-lysine\n",
      "            C03656  (S)-5-Amino-3-oxohexanoic acid\n",
      "            C03793  N6,N6,N6-Trimethyl-L-lysine\n",
      "            C04020  D-Lysopine\n",
      "            C04076  L-2-Aminoadipate 6-semialdehyde\n",
      "            C04092  Delta1-Piperideine-2-carboxylate\n",
      "            C04487  5-(D-Galactosyloxy)-L-lysine-procollagen\n",
      "            C05161  (2R,5S)-2,5-Diaminohexanoate\n",
      "            C05231  L-3-Aminobutyryl-CoA\n",
      "            C05544  Protein N6-methyl-L-lysine\n",
      "            C05545  Protein N6,N6-dimethyl-L-lysine\n",
      "            C05546  Protein N6,N6,N6-trimethyl-L-lysine\n",
      "            C05548  6-Acetamido-2-oxohexanoate\n",
      "            C05825  2-Amino-5-oxohexanoate\n",
      "            C12455  5-Aminopentanal\n",
      "            C16741  L-Hydroxylysine\n",
      "REFERENCE   PMID:11802770\n",
      "  AUTHORS   Vaz FM, Wanders RJ.\n",
      "  TITLE     Carnitine biosynthesis in mammals.\n",
      "  JOURNAL   Biochem J 361:417-29 (2002)\n",
      "            DOI:10.1042/bj3610417\n",
      "REFERENCE   PMID:1682209\n",
      "  AUTHORS   Large PJ, Robertson A.\n",
      "  TITLE     The route of lysine breakdown in Candida tropicalis.\n",
      "  JOURNAL   FEMS Microbiol Lett 66:209-13 (1991)\n",
      "            DOI:10.1111/j.1574-6968.1990.tb03998.x\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00300  Lysine biosynthesis\n",
      "            pae00780  Biotin metabolism\n",
      "KO_PATHWAY  ko00310\n",
      "///\n",
      "path:pae00330\n",
      "ENTRY       pae00330                    Pathway\n",
      "NAME        Arginine and proline metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00330  Arginine and proline metabolism\n",
      "MODULE      pae_M00015  Proline biosynthesis, glutamate => proline [PATH:pae00330]\n",
      "            pae_M00135  GABA biosynthesis, eukaryotes, putrescine => GABA [PATH:pae00330]\n",
      "            pae_M00879  Arginine succinyltransferase pathway, arginine => glutamate [PATH:pae00330]\n",
      "DBLINKS     GO: 0006525 0006560\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0437  codA; cytosine deaminase [KO:K01485] [EC:3.5.4.1 3.5.4.21]\n",
      "line in gene PA0437  codA; cytosine deaminase [KO:K01485] [EC:3.5.4.1 3.5.4.21]\n",
      "            PA1818  ldcA; lysine-specific pyridoxal 5'-phosphate-dependent carboxylase LdcA [KO:K01584] [EC:4.1.1.19]\n",
      "line in gene PA1818  ldcA; lysine-specific pyridoxal 5'-phosphate-dependent carboxylase LdcA [KO:K01584] [EC:4.1.1.19]\n",
      "            PA4839  speA; arginine decarboxylase [KO:K01585] [EC:4.1.1.19]\n",
      "line in gene PA4839  speA; arginine decarboxylase [KO:K01585] [EC:4.1.1.19]\n",
      "            PA0292  aguA; agmatine deiminase [KO:K10536] [EC:3.5.3.12]\n",
      "line in gene PA0292  aguA; agmatine deiminase [KO:K10536] [EC:3.5.3.12]\n",
      "            PA0293  aguB; N-carbamoylputrescine amidohydrolase [KO:K12251] [EC:3.5.1.53]\n",
      "line in gene PA0293  aguB; N-carbamoylputrescine amidohydrolase [KO:K12251] [EC:3.5.1.53]\n",
      "            PA4519  speC; ornithine decarboxylase [KO:K01581] [EC:4.1.1.17]\n",
      "line in gene PA4519  speC; ornithine decarboxylase [KO:K01581] [EC:4.1.1.17]\n",
      "            PA1687  speE; polyamine aminopropyltransferase [KO:K00797] [EC:2.5.1.16]\n",
      "line in gene PA1687  speE; polyamine aminopropyltransferase [KO:K00797] [EC:2.5.1.16]\n",
      "            PA4774  hypothetical protein [KO:K00797] [EC:2.5.1.16]\n",
      "line in gene PA4774  hypothetical protein [KO:K00797] [EC:2.5.1.16]\n",
      "            PA0654  speD; S-adenosylmethionine decarboxylase [KO:K01611] [EC:4.1.1.50]\n",
      "line in gene PA0654  speD; S-adenosylmethionine decarboxylase [KO:K01611] [EC:4.1.1.50]\n",
      "            PA4773  hypothetical protein [KO:K01611] [EC:4.1.1.50]\n",
      "line in gene PA4773  hypothetical protein [KO:K01611] [EC:4.1.1.50]\n",
      "            PA0299  spuC; aminotransferase [KO:K12256] [EC:2.6.1.113]\n",
      "line in gene PA0299  spuC; aminotransferase [KO:K12256] [EC:2.6.1.113]\n",
      "            PA0221  aminotransferase [KO:K12256] [EC:2.6.1.113]\n",
      "line in gene PA0221  aminotransferase [KO:K12256] [EC:2.6.1.113]\n",
      "            PA3713  spdH; spermidine dehydrogenase SpdH [KO:K00316] [EC:1.5.99.6]\n",
      "line in gene PA3713  spdH; spermidine dehydrogenase SpdH [KO:K00316] [EC:1.5.99.6]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA5312  aldehyde dehydrogenase [KO:K12254] [EC:1.2.1.54 1.2.1.-]\n",
      "line in gene PA5312  aldehyde dehydrogenase [KO:K12254] [EC:1.2.1.54 1.2.1.-]\n",
      "            PA4114  spermidine acetyltransferase [KO:K00657] [EC:2.3.1.57]\n",
      "line in gene PA4114  spermidine acetyltransferase [KO:K00657] [EC:2.3.1.57]\n",
      "            PA0421  hypothetical protein [KO:K00274] [EC:1.4.3.4]\n",
      "line in gene PA0421  hypothetical protein [KO:K00274] [EC:1.4.3.4]\n",
      "            PA4976  aruH; arginine:pyruvate transaminase AruH [KO:K12252] [EC:2.6.1.84]\n",
      "line in gene PA4976  aruH; arginine:pyruvate transaminase AruH [KO:K12252] [EC:2.6.1.84]\n",
      "            PA4977  aruI; hypothetical protein [KO:K12253] [EC:4.1.1.75]\n",
      "line in gene PA4977  aruI; hypothetical protein [KO:K12253] [EC:4.1.1.75]\n",
      "            PA1421  gbuA; guanidinobutyrase [KO:K12255] [EC:3.5.3.7]\n",
      "line in gene PA1421  gbuA; guanidinobutyrase [KO:K12255] [EC:3.5.3.7]\n",
      "            PA0704  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA0704  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA0202  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA0202  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA3366  amiE; acylamide amidohydrolase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA3366  amiE; acylamide amidohydrolase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA4342  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA4342  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA4163  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA4163  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA0896  aruF; arginine N-succinyltransferase subunit alpha [KO:K00673] [EC:2.3.1.109]\n",
      "line in gene PA0896  aruF; arginine N-succinyltransferase subunit alpha [KO:K00673] [EC:2.3.1.109]\n",
      "            PA0897  aruG; arginine N-succinyltransferase subunit beta [KO:K00673] [EC:2.3.1.109]\n",
      "line in gene PA0897  aruG; arginine N-succinyltransferase subunit beta [KO:K00673] [EC:2.3.1.109]\n",
      "            PA0899  aruB; N-succinylarginine dihydrolase [KO:K01484] [EC:3.5.3.23]\n",
      "line in gene PA0899  aruB; N-succinylarginine dihydrolase [KO:K01484] [EC:3.5.3.23]\n",
      "            PA0895  aruC; acetylornithine aminotransferase [KO:K00840] [EC:2.6.1.81]\n",
      "line in gene PA0895  aruC; acetylornithine aminotransferase [KO:K00840] [EC:2.6.1.81]\n",
      "            PA0898  aruD; N-succinylglutamate 5-semialdehyde dehydrogenase [KO:K06447] [EC:1.2.1.71]\n",
      "line in gene PA0898  aruD; N-succinylglutamate 5-semialdehyde dehydrogenase [KO:K06447] [EC:1.2.1.71]\n",
      "            PA0901  aruE; succinylglutamate desuccinylase [KO:K05526] [EC:3.5.1.96]\n",
      "line in gene PA0901  aruE; succinylglutamate desuccinylase [KO:K05526] [EC:3.5.1.96]\n",
      "            PA2776  hypothetical protein [KO:K09471] [EC:1.4.3.-]\n",
      "line in gene PA2776  hypothetical protein [KO:K09471] [EC:1.4.3.-]\n",
      "            PA2317  oxidoreductase [KO:K09471] [EC:1.4.3.-]\n",
      "line in gene PA2317  oxidoreductase [KO:K09471] [EC:1.4.3.-]\n",
      "            PA1565  oxidoreductase [KO:K09471] [EC:1.4.3.-]\n",
      "line in gene PA1565  oxidoreductase [KO:K09471] [EC:1.4.3.-]\n",
      "            PA5309  oxidoreductase [KO:K09471] [EC:1.4.3.-]\n",
      "line in gene PA5309  oxidoreductase [KO:K09471] [EC:1.4.3.-]\n",
      "            PA4189  aldehyde dehydrogenase [KO:K09472] [EC:1.2.1.99]\n",
      "line in gene PA4189  aldehyde dehydrogenase [KO:K09472] [EC:1.2.1.99]\n",
      "            PA0219  aldehyde dehydrogenase [KO:K09472] [EC:1.2.1.99]\n",
      "line in gene PA0219  aldehyde dehydrogenase [KO:K09472] [EC:1.2.1.99]\n",
      "            PA0393  proC; pyrroline-5-carboxylate reductase [KO:K00286] [EC:1.5.1.2]\n",
      "line in gene PA0393  proC; pyrroline-5-carboxylate reductase [KO:K00286] [EC:1.5.1.2]\n",
      "            PA0782  putA; bifunctional proline dehydrogenase/pyrroline-5-carboxylate dehydrogenase [KO:K13821] [EC:1.5.5.2 1.2.1.88]\n",
      "line in gene PA0782  putA; bifunctional proline dehydrogenase/pyrroline-5-carboxylate dehydrogenase [KO:K13821] [EC:1.5.5.2 1.2.1.88]\n",
      "            PA4565  proB; gamma-glutamyl kinase [KO:K00931] [EC:2.7.2.11]\n",
      "line in gene PA4565  proB; gamma-glutamyl kinase [KO:K00931] [EC:2.7.2.11]\n",
      "            PA4007  proA; gamma-glutamyl phosphate reductase [KO:K00147] [EC:1.2.1.41]\n",
      "line in gene PA4007  proA; gamma-glutamyl phosphate reductase [KO:K00147] [EC:1.2.1.41]\n",
      "            PA5080  prolyl aminopeptidase [KO:K01259] [EC:3.4.11.5]\n",
      "line in gene PA5080  prolyl aminopeptidase [KO:K01259] [EC:3.4.11.5]\n",
      "            PA1255  trans-3-hydroxy-L-proline dehydratase [KO:K22548] [EC:4.2.1.77]\n",
      "line in gene PA1255  trans-3-hydroxy-L-proline dehydratase [KO:K22548] [EC:4.2.1.77]\n",
      "            PA1259  hypothetical protein [KO:K09123] [EC:4.2.1.171]\n",
      "line in gene PA1259  hypothetical protein [KO:K09123] [EC:4.2.1.171]\n",
      "COMPOUND    C00012  Peptide\n",
      "            C00019  S-Adenosyl-L-methionine\n",
      "            C00022  Pyruvate\n",
      "            C00025  L-Glutamate\n",
      "            C00048  Glyoxylate\n",
      "            C00062  L-Arginine\n",
      "            C00077  L-Ornithine\n",
      "            C00086  Urea\n",
      "            C00134  Putrescine\n",
      "            C00148  L-Proline\n",
      "            C00179  Agmatine\n",
      "            C00213  Sarcosine\n",
      "            C00223  p-Coumaroyl-CoA\n",
      "            C00300  Creatine\n",
      "            C00315  Spermidine\n",
      "            C00334  4-Aminobutanoate\n",
      "            C00406  Feruloyl-CoA\n",
      "            C00431  5-Aminopentanoate\n",
      "            C00436  N-Carbamoylputrescine\n",
      "            C00441  L-Aspartate 4-semialdehyde\n",
      "            C00533  Nitric oxide\n",
      "            C00555  4-Aminobutyraldehyde\n",
      "            C00581  Guanidinoacetate\n",
      "            C00750  Spermine\n",
      "            C00763  D-Proline\n",
      "            C00791  Creatinine\n",
      "            C00884  Homocarnosine\n",
      "            C00986  1,3-Diaminopropane\n",
      "            C01035  4-Guanidinobutanoate\n",
      "            C01043  N-Carbamoylsarcosine\n",
      "            C01110  5-Amino-2-oxopentanoic acid\n",
      "            C01137  S-Adenosylmethioninamine\n",
      "            C01157  Hydroxyproline\n",
      "            C01165  L-Glutamate 5-semialdehyde\n",
      "            C01250  N-Acetyl-L-glutamate 5-semialdehyde\n",
      "            C01682  Nopaline\n",
      "            C01877  4-Oxoproline\n",
      "            C02305  Phosphocreatine\n",
      "            C02565  N-Methylhydantoin\n",
      "            C02647  4-Guanidinobutanal\n",
      "            C02714  N-Acetylputrescine\n",
      "            C02946  4-Acetamidobutanoate\n",
      "            C03078  4-Guanidinobutanamide\n",
      "            C03166  Phosphoguanidinoacetate\n",
      "            C03287  L-Glutamyl 5-phosphate\n",
      "            C03296  N2-Succinyl-L-arginine\n",
      "            C03375  Norspermidine\n",
      "            C03415  N2-Succinyl-L-ornithine\n",
      "            C03564  1-Pyrroline-2-carboxylate\n",
      "            C03771  5-Guanidino-2-oxopentanoate\n",
      "            C03912  (S)-1-Pyrroline-5-carboxylate\n",
      "            C04137  Octopine\n",
      "            C04281  L-1-Pyrroline-3-hydroxy-5-carboxylate\n",
      "            C04498  p-Coumaroylagmatine\n",
      "            C05147  trans-3-Hydroxy-L-proline\n",
      "            C05931  N-Succinyl-L-glutamate\n",
      "            C05932  N-Succinyl-L-glutamate 5-semialdehyde\n",
      "            C05933  N(omega)-Hydroxyarginine\n",
      "            C05936  N4-Acetylaminobutanal\n",
      "            C05938  L-4-Hydroxyglutamate semialdehyde\n",
      "            C05945  L-Arginine phosphate\n",
      "            C05946  (4R)-4-Hydroxy-2-oxoglutarate\n",
      "            C05947  L-erythro-4-Hydroxyglutamate\n",
      "            C10497  Feruloylputrescine\n",
      "            C15699  gamma-L-Glutamylputrescine\n",
      "            C15700  gamma-Glutamyl-gamma-aminobutyraldehyde\n",
      "            C15767  4-(L-gamma-Glutamylamino)butanoate\n",
      "            C18172  Carboxyspermidine\n",
      "            C18174  Carboxynorspermidine\n",
      "            C18325  Feruloylagmatine\n",
      "            C18326  p-Coumaroylputrescine\n",
      "            C19706  cis-3-Hydroxy-L-proline\n",
      "REFERENCE   PMID:5288244\n",
      "  AUTHORS   Haake P, Allen GW.\n",
      "  TITLE     Studies on phosphorylation by phosphoroguanidinates. The mechanism of action of creatine: ATP transphosphorylase (creatine kinase).\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 68:2691-3 (1971)\n",
      "            DOI:10.1073/pnas.68.11.2691\n",
      "REFERENCE   PMID:16499623\n",
      "  AUTHORS   Kurihara S, Oda S, Kumagai H, Suzuki H.\n",
      "  TITLE     Gamma-glutamyl-gamma-aminobutyrate hydrolase in the putrescine utilization pathway of Escherichia coli K-12.\n",
      "  JOURNAL   FEMS Microbiol Lett 256:318-23 (2006)\n",
      "            DOI:10.1111/j.1574-6968.2006.00137.x\n",
      "REFERENCE   PMID:15590624\n",
      "  AUTHORS   Kurihara S, Oda S, Kato K, Kim HG, Koyanagi T, Kumagai H, Suzuki H.\n",
      "  TITLE     A novel putrescine utilization pathway involves gamma-glutamylated intermediates of Escherichia coli K-12.\n",
      "  JOURNAL   J Biol Chem 280:4602-8 (2005)\n",
      "            DOI:10.1074/jbc.M411114200\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00220  Arginine biosynthesis\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00310  Lysine degradation\n",
      "            pae00410  beta-Alanine metabolism\n",
      "            pae00470  D-Amino acid metabolism\n",
      "            pae00480  Glutathione metabolism\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00650  Butanoate metabolism\n",
      "            pae00770  Pantothenate and CoA biosynthesis\n",
      "KO_PATHWAY  ko00330\n",
      "///\n",
      "path:pae00332\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00332                    Pathway\n",
      "NAME        Carbapenem biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Carbapenems are broad-spectrum beta-lactam antibiotics, which are often considered as the antibiotics of last resort. A naturally occurring carbapenem, thienamycin, was first discovered in Streptomyces cattleya. This diagram shows how a simple carbapenem, carbapenem-3-carboxylate, is synthesized from malonyl-CoA and pyrroline-5-carboxylate [MD:M00675]. For structurally complex carbapenems, such as thienamycin, olivanic acid, epi-thienamycin and carbapenems of the OA-6129 group, uncertainty remains about the mechanism and timing for the inversion of the C-5 carbapenam bridgehead, the desaturation of the C-2/C-3 bond and the attachment of the C-2 and C-6 side chains.\n",
      "CLASS       Metabolism; Biosynthesis of other secondary metabolites\n",
      "PATHWAY_MAP pae00332  Carbapenem biosynthesis\n",
      "DBLINKS     GO: 1901769\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4565  proB; gamma-glutamyl kinase [KO:K00931] [EC:2.7.2.11]\n",
      "line in gene PA4565  proB; gamma-glutamyl kinase [KO:K00931] [EC:2.7.2.11]\n",
      "            PA4007  proA; gamma-glutamyl phosphate reductase [KO:K00147] [EC:1.2.1.41]\n",
      "line in gene PA4007  proA; gamma-glutamyl phosphate reductase [KO:K00147] [EC:1.2.1.41]\n",
      "COMPOUND    C00010  CoA\n",
      "            C00025  L-Glutamate\n",
      "            C00083  Malonyl-CoA\n",
      "            C00097  L-Cysteine\n",
      "            C00148  L-Proline\n",
      "            C00831  Pantetheine\n",
      "            C01134  Pantetheine 4'-phosphate\n",
      "            C01165  L-Glutamate 5-semialdehyde\n",
      "            C03287  L-Glutamyl 5-phosphate\n",
      "            C03912  (S)-1-Pyrroline-5-carboxylate\n",
      "            C06664  Thienamycin\n",
      "            C06669  (5R)-Carbapen-2-em-3-carboxylate\n",
      "            C11755  Epithienamycin E\n",
      "            C17366  (2S,5S)-trans-Carboxymethylproline\n",
      "            C17367  (3S,5S)-Carbapenam-3-carboxylic acid\n",
      "            C17368  8-Dehydroxythienamycin\n",
      "            C17369  Northienamycin\n",
      "            C17373  OA-6129 B1\n",
      "            C17374  OA-6129 B2\n",
      "            C17377  N-Acetylthienamycin\n",
      "            C17399  Epithienamycin F\n",
      "            C20813  OA-6129 A\n",
      "            C20814  OA-6129 C\n",
      "            C20815  MM 4550\n",
      "            C20816  Deacetylepithienamycin F\n",
      "            C20817  C20817\n",
      "            C20818  C20818\n",
      "            C20819  C20819\n",
      "            C20820  2,3-Dihydrothienamycin\n",
      "            C20821  C20821\n",
      "            C20822  C20822\n",
      "            C20823  C20823\n",
      "REFERENCE   PMID:12848554\n",
      "  AUTHORS   Stapon A, Li R, Townsend CA\n",
      "  TITLE     Carbapenem biosynthesis: confirmation of stereochemical assignments and the role of CarC in the ring stereoinversion process from L-proline.\n",
      "  JOURNAL   J Am Chem Soc 125:8486-93 (2003)\n",
      "            DOI:10.1021/ja034248a\n",
      "REFERENCE   PMID:15174175\n",
      "  AUTHORS   Sleeman MC, Smith P, Kellam B, Chhabra SR, Bycroft BW, Schofield CJ\n",
      "  TITLE     Biosynthesis of carbapenem antibiotics: new carbapenam substrates for carbapenem synthase (CarC).\n",
      "  JOURNAL   Chembiochem 5:879-82 (2004)\n",
      "            DOI:10.1002/cbic.200300908\n",
      "REFERENCE   PMID:14625287\n",
      "  AUTHORS   Sleeman MC, Schofield CJ\n",
      "  TITLE     Carboxymethylproline synthase (CarB), an unusual carbon-carbon bond-forming enzyme of the crotonase superfamily involved in carbapenem biosynthesis.\n",
      "  JOURNAL   J Biol Chem 279:6730-6 (2004)\n",
      "            DOI:10.1074/jbc.M311824200\n",
      "REFERENCE   PMID:19090510\n",
      "  AUTHORS   Hamed RB, Batchelar ET, Mecinovic J, Claridge TD, Schofield CJ\n",
      "  TITLE     Evidence that thienamycin biosynthesis proceeds via C-5 epimerization: ThnE catalyzes the formation of (2S,5S)-trans-carboxymethylproline.\n",
      "  JOURNAL   Chembiochem 10:246-50 (2009)\n",
      "            DOI:10.1002/cbic.200800652\n",
      "REFERENCE   PMID:21913298\n",
      "  AUTHORS   Bodner MJ, Li R, Phelan RM, Freeman MF, Moshos KA, Lloyd EP, Townsend CA\n",
      "  TITLE     Definition of the common and divergent steps in carbapenem beta-lactam antibiotic biosynthesis.\n",
      "  JOURNAL   Chembiochem 12:2159-65 (2011)\n",
      "            DOI:10.1002/cbic.201100366\n",
      "REFERENCE   PMID:12147472\n",
      "  AUTHORS   Derzelle S, Duchaud E, Kunst F, Danchin A, Bertin P\n",
      "  TITLE     Identification, characterization, and regulation of a cluster of genes involved in carbapenem biosynthesis in Photorhabdus luminescens.\n",
      "  JOURNAL   Appl Environ Microbiol 68:3780-9 (2002)\n",
      "            DOI:10.1128/AEM.68.8.3780-3789.2002\n",
      "REFERENCE   PMID:21263049\n",
      "  AUTHORS   Rodriguez M, Nunez LE, Brana AF, Mendez C, Salas JA, Blanco G\n",
      "  TITLE     Mutational analysis of the thienamycin biosynthetic gene cluster from Streptomyces cattleya.\n",
      "  JOURNAL   Antimicrob Agents Chemother 55:1638-49 (2011)\n",
      "            DOI:10.1128/AAC.01366-10\n",
      "REFERENCE   PMID:3988730\n",
      "  AUTHORS   Williamson JM, Inamine E, Wilson KE, Douglas AW, Liesch JM, Albers-Schonberg G\n",
      "  TITLE     Biosynthesis of the beta-lactam antibiotic, thienamycin, by Streptomyces cattleya.\n",
      "  JOURNAL   J Biol Chem 260:4637-47 (1985)\n",
      "REFERENCE   PMID:12725858\n",
      "  AUTHORS   Nunez LE, Mendez C, Brana AF, Blanco G, Salas JA\n",
      "  TITLE     The biosynthetic gene cluster for the beta-lactam carbapenem thienamycin in Streptomyces cattleya.\n",
      "  JOURNAL   Chem Biol 10:301-11 (2003)\n",
      "            DOI:10.1016/S1074-5521(03)00069-3\n",
      "REFERENCE   PMID:20017478\n",
      "  AUTHORS   Bodner MJ, Phelan RM, Freeman MF, Li R, Townsend CA\n",
      "  TITLE     Non-heme iron oxygenases generate natural structural diversity in carbapenem antibiotics.\n",
      "  JOURNAL   J Am Chem Soc 132:12-3 (2010)\n",
      "            DOI:10.1021/ja907320n\n",
      "REFERENCE   PMID:18678912\n",
      "  AUTHORS   Freeman MF, Moshos KA, Bodner MJ, Li R, Townsend CA\n",
      "  TITLE     Four enzymes define the incorporation of coenzyme A in thienamycin biosynthesis.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 105:11128-33 (2008)\n",
      "            DOI:10.1073/pnas.0804500105\n",
      "REFERENCE   PMID:22706025\n",
      "  AUTHORS   Buller AR, Labonte JW, Freeman MF, Wright NT, Schildbach JF, Townsend CA\n",
      "  TITLE     Autoproteolytic activation of ThnT results in structural reorganization necessary for substrate binding and catalysis.\n",
      "  JOURNAL   J Mol Biol 422:508-18 (2012)\n",
      "            DOI:10.1016/j.jmb.2012.06.012\n",
      "REFERENCE   PMID:24420617\n",
      "  AUTHORS   Li R, Lloyd EP, Moshos KA, Townsend CA\n",
      "  TITLE     Identification and characterization of the carbapenem MM 4550 and its gene cluster in Streptomyces argenteolus ATCC 11009.\n",
      "  JOURNAL   Chembiochem 15:320-31 (2014)\n",
      "            DOI:10.1002/cbic.201300319\n",
      "REL_PATHWAY pae00330  Arginine and proline metabolism\n",
      "KO_PATHWAY  ko00332\n",
      "///\n",
      "path:pae00340\n",
      "ENTRY       pae00340                    Pathway\n",
      "NAME        Histidine metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00340  Histidine metabolism\n",
      "MODULE      pae_M00045  Histidine degradation, histidine => N-formiminoglutamate => glutamate [PATH:pae00340]\n",
      "DBLINKS     GO: 0006547\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4449  hisG; ATP phosphoribosyltransferase [KO:K00765] [EC:2.4.2.17]\n",
      "line in gene PA4449  hisG; ATP phosphoribosyltransferase [KO:K00765] [EC:2.4.2.17]\n",
      "            PA4939  ATP phosphoribosyltransferase [KO:K02502]\n",
      "line in gene PA4939  ATP phosphoribosyltransferase [KO:K02502]\n",
      "            PA5067  hisE; phosphoribosyl-ATP pyrophosphatase [KO:K01523] [EC:3.6.1.31]\n",
      "line in gene PA5067  hisE; phosphoribosyl-ATP pyrophosphatase [KO:K01523] [EC:3.6.1.31]\n",
      "            PA5066  hisI; phosphoribosyl-AMP cyclohydrolase [KO:K01496] [EC:3.5.4.19]\n",
      "line in gene PA5066  hisI; phosphoribosyl-AMP cyclohydrolase [KO:K01496] [EC:3.5.4.19]\n",
      "            PA5141  hisA; 1-(5-phosphoribosyl)-5-[(5-phosphoribosylamino)methylideneamino] imidazole-4-carboxamide isomerase [KO:K01814] [EC:5.3.1.16]\n",
      "line in gene PA5141  hisA; 1-(5-phosphoribosyl)-5-[(5-phosphoribosylamino)methylideneamino] imidazole-4-carboxamide isomerase [KO:K01814] [EC:5.3.1.16]\n",
      "            PA5142  hisH1; imidazole glycerol phosphate synthase subunit HisH [KO:K02501] [EC:4.3.2.10]\n",
      "line in gene PA5142  hisH1; imidazole glycerol phosphate synthase subunit HisH [KO:K02501] [EC:4.3.2.10]\n",
      "            PA3152  hisH2; imidazole glycerol phosphate synthase subunit HisH [KO:K02501] [EC:4.3.2.10]\n",
      "line in gene PA3152  hisH2; imidazole glycerol phosphate synthase subunit HisH [KO:K02501] [EC:4.3.2.10]\n",
      "            PA5140  hisF1; imidazole glycerol phosphate synthase subunit HisF [KO:K02500] [EC:4.3.2.10]\n",
      "line in gene PA5140  hisF1; imidazole glycerol phosphate synthase subunit HisF [KO:K02500] [EC:4.3.2.10]\n",
      "            PA3151  hisF2; imidazole glycerol phosphate synthase subunit HisF [KO:K02500] [EC:4.3.2.10]\n",
      "line in gene PA3151  hisF2; imidazole glycerol phosphate synthase subunit HisF [KO:K02500] [EC:4.3.2.10]\n",
      "            PA5143  hisB; imidazoleglycerol-phosphate dehydratase [KO:K01693] [EC:4.2.1.19]\n",
      "line in gene PA5143  hisB; imidazoleglycerol-phosphate dehydratase [KO:K01693] [EC:4.2.1.19]\n",
      "            PA2531  aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA2531  aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA4447  hisC1; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA4447  hisC1; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA3165  hisC2; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA3165  hisC2; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA4448  hisD; histidinol dehydrogenase [KO:K00013] [EC:1.1.1.23]\n",
      "line in gene PA4448  hisD; histidinol dehydrogenase [KO:K00013] [EC:1.1.1.23]\n",
      "            PA5093  histidine/phenylalanine ammonia-lyase [KO:K01745] [EC:4.3.1.3]\n",
      "line in gene PA5093  histidine/phenylalanine ammonia-lyase [KO:K01745] [EC:4.3.1.3]\n",
      "            PA5098  hutH; histidine ammonia-lyase [KO:K01745] [EC:4.3.1.3]\n",
      "line in gene PA5098  hutH; histidine ammonia-lyase [KO:K01745] [EC:4.3.1.3]\n",
      "            PA5100  hutU; urocanate hydratase [KO:K01712] [EC:4.2.1.49]\n",
      "line in gene PA5100  hutU; urocanate hydratase [KO:K01712] [EC:4.2.1.49]\n",
      "            PA5092  hutI; imidazolonepropionase [KO:K01468] [EC:3.5.2.7]\n",
      "line in gene PA5092  hutI; imidazolonepropionase [KO:K01468] [EC:3.5.2.7]\n",
      "            PA3175  formimidoylglutamase [KO:K01479] [EC:3.5.3.8]\n",
      "line in gene PA3175  formimidoylglutamase [KO:K01479] [EC:3.5.3.8]\n",
      "            PA5106  N-formimino-L-glutamate deiminase [KO:K05603] [EC:3.5.3.13]\n",
      "line in gene PA5106  N-formimino-L-glutamate deiminase [KO:K05603] [EC:3.5.3.13]\n",
      "            PA5091  hutG; N-formylglutamate amidohydrolase [KO:K01458] [EC:3.5.1.68]\n",
      "line in gene PA5091  hutG; N-formylglutamate amidohydrolase [KO:K01458] [EC:3.5.1.68]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA0421  hypothetical protein [KO:K00274] [EC:1.4.3.4]\n",
      "line in gene PA0421  hypothetical protein [KO:K00274] [EC:1.4.3.4]\n",
      "COMPOUND    C00025  L-Glutamate\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00049  L-Aspartate\n",
      "            C00119  5-Phospho-alpha-D-ribose 1-diphosphate\n",
      "            C00135  L-Histidine\n",
      "            C00386  Carnosine\n",
      "            C00388  Histamine\n",
      "            C00439  N-Formimino-L-glutamate\n",
      "            C00785  Urocanate\n",
      "            C00860  L-Histidinol\n",
      "            C01044  N-Formyl-L-aspartate\n",
      "            C01045  N-Formyl-L-glutamate\n",
      "            C01100  L-Histidinol phosphate\n",
      "            C01152  N(pi)-Methyl-L-histidine\n",
      "            C01262  beta-Alanyl-N(pi)-methyl-L-histidine\n",
      "            C01267  3-(Imidazol-4-yl)-2-oxopropyl phosphate\n",
      "            C01929  L-Histidinal\n",
      "            C02739  1-(5-Phospho-D-ribosyl)-ATP\n",
      "            C02741  Phosphoribosyl-AMP\n",
      "            C02835  Imidazole-4-acetate\n",
      "            C03277  Imidazol-5-yl-pyruvate\n",
      "            C03409  N-Formimino-L-aspartate\n",
      "            C03680  4-Imidazolone-5-propanoate\n",
      "            C04437  1-(5-Phosphoribosyl)imidazole-4-acetate\n",
      "            C04666  D-erythro-1-(Imidazol-4-yl)glycerol 3-phosphate\n",
      "            C04677  1-(5'-Phosphoribosyl)-5-amino-4-imidazolecarboxamide\n",
      "            C04896  5-(5-Phospho-D-ribosylaminoformimino)-1-(5-phosphoribosyl)-imidazole-4-carboxamide\n",
      "            C04916  N-(5'-Phospho-D-1'-ribulosylformimino)-5-amino-1-(5''-phospho-D-ribosyl)-4-imidazolecarboxamide\n",
      "            C05127  N-Methylhistamine\n",
      "            C05130  Imidazole-4-acetaldehyde\n",
      "            C05131  (1-Ribosylimidazole)-4-acetate\n",
      "            C05133  4-Imidazolone-5-acetate\n",
      "            C05135  4-(beta-Acetylaminoethyl)imidazole\n",
      "            C05565  Hydantoin-5-propionate\n",
      "            C05568  Imidazole lactate\n",
      "            C05570  Ergothioneine\n",
      "            C05571  Thiourocanic acid\n",
      "            C05572  4-Oxoglutaramate\n",
      "            C05575  Hercynine\n",
      "            C05827  Methylimidazole acetaldehyde\n",
      "            C05828  Methylimidazoleacetic acid\n",
      "            C05829  N-Carbamyl-L-glutamate\n",
      "            C16673  Isoglutamine\n",
      "            C16674  Formylisoglutamine\n",
      "            C20522  Dihydrourocanate\n",
      "            C20994  S-(Hercyn-2-yl)-L-cysteine S-oxide\n",
      "            C20995  gamma-L-Glutamyl-S-(hercyn-2-yl)-L-cysteine S-oxide\n",
      "REL_PATHWAY pae00030  Pentose phosphate pathway\n",
      "            pae00230  Purine metabolism\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "KO_PATHWAY  ko00340\n",
      "///\n",
      "path:pae00350\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00350                    Pathway\n",
      "NAME        Tyrosine metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00350  Tyrosine metabolism\n",
      "MODULE      pae_M00044  Tyrosine degradation, tyrosine => homogentisate [PATH:pae00350]\n",
      "            pae_M00533  Homoprotocatechuate degradation, homoprotocatechuate => 2-oxohept-3-enedioate [PATH:pae00350]\n",
      "DBLINKS     GO: 0006570\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2531  aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA2531  aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA4447  hisC1; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA4447  hisC1; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA3165  hisC2; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA3165  hisC2; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA3139  aspartate aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "line in gene PA3139  aspartate aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "            PA0870  phhC; aromatic amino acid aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "line in gene PA0870  phhC; aromatic amino acid aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "            PA0865  hpd; 4-hydroxyphenylpyruvate dioxygenase [KO:K00457] [EC:1.13.11.27]\n",
      "line in gene PA0865  hpd; 4-hydroxyphenylpyruvate dioxygenase [KO:K00457] [EC:1.13.11.27]\n",
      "            PA0242  hypothetical protein [KO:K00457] [EC:1.13.11.27]\n",
      "line in gene PA0242  hypothetical protein [KO:K00457] [EC:1.13.11.27]\n",
      "            PA2009  hmgA; homogentisate 1,2-dioxygenase [KO:K00451] [EC:1.13.11.5]\n",
      "line in gene PA2009  hmgA; homogentisate 1,2-dioxygenase [KO:K00451] [EC:1.13.11.5]\n",
      "            PA2473  glutathione S-transferase [KO:K01800] [EC:5.2.1.2]\n",
      "line in gene PA2473  glutathione S-transferase [KO:K01800] [EC:5.2.1.2]\n",
      "            PA2007  maiA; maleylacetoacetate isomerase [KO:K01800] [EC:5.2.1.2]\n",
      "line in gene PA2007  maiA; maleylacetoacetate isomerase [KO:K01800] [EC:5.2.1.2]\n",
      "            PA2008  fahA; fumarylacetoacetase [KO:K01555] [EC:3.7.1.2]\n",
      "line in gene PA2008  fahA; fumarylacetoacetase [KO:K01555] [EC:3.7.1.2]\n",
      "            PA0421  hypothetical protein [KO:K00274] [EC:1.4.3.4]\n",
      "line in gene PA0421  hypothetical protein [KO:K00274] [EC:1.4.3.4]\n",
      "            PA4073  aldehyde dehydrogenase [KO:K00146] [EC:1.2.1.39]\n",
      "line in gene PA4073  aldehyde dehydrogenase [KO:K00146] [EC:1.2.1.39]\n",
      "            PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "line in gene PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "            PA5427  adhA; alcohol dehydrogenase [KO:K13953] [EC:1.1.1.1]\n",
      "line in gene PA5427  adhA; alcohol dehydrogenase [KO:K13953] [EC:1.1.1.1]\n",
      "            PA4091  hpaA; 4-hydroxyphenylacetate 3-monooxygenase large subunit [KO:K00483] [EC:1.14.14.9]\n",
      "line in gene PA4091  hpaA; 4-hydroxyphenylacetate 3-monooxygenase large subunit [KO:K00483] [EC:1.14.14.9]\n",
      "            PA4092  hpaC; 4-hydroxyphenylacetate 3-monooxygenase small subunit [KO:K00484] [EC:1.5.1.36]\n",
      "line in gene PA4092  hpaC; 4-hydroxyphenylacetate 3-monooxygenase small subunit [KO:K00484] [EC:1.5.1.36]\n",
      "            PA4124  hpcB; homoprotocatechuate 2,3-dioxygenase [KO:K00455] [EC:1.13.11.15]\n",
      "line in gene PA4124  hpcB; homoprotocatechuate 2,3-dioxygenase [KO:K00455] [EC:1.13.11.15]\n",
      "            PA4123  hpcC; 5-carboxy-2-hydroxymuconate semialdehyde dehydrogenase [KO:K00151] [EC:1.2.1.60]\n",
      "line in gene PA4123  hpcC; 5-carboxy-2-hydroxymuconate semialdehyde dehydrogenase [KO:K00151] [EC:1.2.1.60]\n",
      "            PA4125  hpcD; 5-carboxymethyl-2-hydroxymuconate isomerase [KO:K01826] [EC:5.3.3.10]\n",
      "line in gene PA4125  hpcD; 5-carboxymethyl-2-hydroxymuconate isomerase [KO:K01826] [EC:5.3.3.10]\n",
      "            PA1966  hypothetical protein [KO:K01826] [EC:5.3.3.10]\n",
      "line in gene PA1966  hypothetical protein [KO:K01826] [EC:5.3.3.10]\n",
      "            PA4121  hypothetical protein [KO:K05921] [EC:4.1.1.68 5.3.3.-]\n",
      "line in gene PA4121  hypothetical protein [KO:K05921] [EC:4.1.1.68 5.3.3.-]\n",
      "            PA4122  hypothetical protein [KO:K05921] [EC:4.1.1.68 5.3.3.-]\n",
      "line in gene PA4122  hypothetical protein [KO:K05921] [EC:4.1.1.68 5.3.3.-]\n",
      "            PA4127  hpcG; 2-oxo-hepta-3-ene-1,7-dioic acid hydratase [KO:K02509] [EC:4.2.1.-]\n",
      "line in gene PA4127  hpcG; 2-oxo-hepta-3-ene-1,7-dioic acid hydratase [KO:K02509] [EC:4.2.1.-]\n",
      "            PA4128  2,4-dihydroxyhept-2-ene-1,7-dioic acid aldolase [KO:K02510] [EC:4.1.2.52]\n",
      "line in gene PA4128  2,4-dihydroxyhept-2-ene-1,7-dioic acid aldolase [KO:K02510] [EC:4.1.2.52]\n",
      "            PA0265  gabD; glutarate-semialdehyde dehydrogenase DavD [KO:K00135] [EC:1.2.1.16 1.2.1.79 1.2.1.20]\n",
      "line in gene PA0265  gabD; glutarate-semialdehyde dehydrogenase DavD [KO:K00135] [EC:1.2.1.16 1.2.1.79 1.2.1.20]\n",
      "            PA2470  gtdA; gentisate 1,2-dioxygenase [KO:K00450] [EC:1.13.11.4]\n",
      "line in gene PA2470  gtdA; gentisate 1,2-dioxygenase [KO:K00450] [EC:1.13.11.4]\n",
      "            PA2471  hypothetical protein [KO:K16165] [EC:3.7.1.20]\n",
      "line in gene PA2471  hypothetical protein [KO:K16165] [EC:3.7.1.20]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00042  Succinate\n",
      "            C00082  L-Tyrosine\n",
      "            C00122  Fumarate\n",
      "            C00146  Phenol\n",
      "            C00164  Acetoacetate\n",
      "            C00232  Succinate semialdehyde\n",
      "            C00355  3,4-Dihydroxy-L-phenylalanine\n",
      "            C00483  Tyramine\n",
      "            C00530  Hydroquinone\n",
      "            C00544  Homogentisate\n",
      "            C00547  L-Noradrenaline\n",
      "            C00628  2,5-Dihydroxybenzoate\n",
      "            C00642  4-Hydroxyphenylacetate\n",
      "            C00788  L-Adrenaline\n",
      "            C00811  4-Coumarate\n",
      "            C00822  Dopaquinone\n",
      "            C01036  4-Maleylacetoacetate\n",
      "            C01060  3,5-Diiodo-L-tyrosine\n",
      "            C01061  4-Fumarylacetoacetate\n",
      "            C01161  3,4-Dihydroxyphenylacetate\n",
      "            C01179  3-(4-Hydroxyphenyl)pyruvate\n",
      "            C01384  Maleic acid\n",
      "            C01693  L-Dopachrome\n",
      "            C01829  Thyroxine\n",
      "            C01850  Rosmarinate\n",
      "            C02167  Maleylpyruvate\n",
      "            C02442  N-Methyltyramine\n",
      "            C02465  Triiodothyronine\n",
      "            C02514  3-Fumarylpyruvate\n",
      "            C02515  3-Iodo-L-tyrosine\n",
      "            C03063  2-Oxohept-3-enedioate\n",
      "            C03077  4-Chlorophenylacetate\n",
      "            C03758  Dopamine\n",
      "            C03765  4-Hydroxyphenylacetaldehyde\n",
      "            C03964  (R)-3-(4-Hydroxyphenyl)lactate\n",
      "            C04043  3,4-Dihydroxyphenylacetaldehyde\n",
      "            C04045  3-(3,4-Dihydroxyphenyl)pyruvate\n",
      "            C04052  5-Carboxy-2-oxohept-3-enedioate\n",
      "            C04185  5,6-Dihydroxyindole-2-carboxylate\n",
      "            C04186  5-Carboxymethyl-2-hydroxymuconate\n",
      "            C04368  3-Amino-3-(4-hydroxyphenyl)propanoate\n",
      "            C04642  2-Hydroxy-5-carboxymethylmuconate semialdehyde\n",
      "            C04796  4-(L-Alanin-3-yl)-2-hydroxy-cis,cis-muconate 6-semialdehyde\n",
      "            C04797  5-(L-Alanin-3-yl)-2-hydroxy-cis,cis-muconate 6-semialdehyde\n",
      "            C05338  4-Hydroxyphenylacetyl-CoA\n",
      "            C05350  2-Hydroxy-3-(4-hydroxyphenyl)propenoate\n",
      "            C05576  3,4-Dihydroxyphenylethyleneglycol\n",
      "            C05577  3,4-Dihydroxymandelaldehyde\n",
      "            C05578  5,6-Dihydroxyindole\n",
      "            C05579  Indole-5,6-quinone\n",
      "            C05580  3,4-Dihydroxymandelate\n",
      "            C05581  3-Methoxy-4-hydroxyphenylacetaldehyde\n",
      "            C05582  Homovanillate\n",
      "            C05583  3-Methoxy-4-hydroxyphenylglycolaldehyde\n",
      "            C05584  3-Methoxy-4-hydroxymandelate\n",
      "            C05585  Gentisate aldehyde\n",
      "            C05587  3-Methoxytyramine\n",
      "            C05588  L-Metanephrine\n",
      "            C05589  L-Normetanephrine\n",
      "            C05593  3-Hydroxyphenylacetate\n",
      "            C05594  3-Methoxy-4-hydroxyphenylethyleneglycol\n",
      "            C05595  4-Hydroxyphenylacetylglutamic acid\n",
      "            C05596  p-Hydroxyphenylacetylglycine\n",
      "            C05600  2-Hydroxyhepta-2,4-dienedioate\n",
      "            C05604  2-Carboxy-2,3-dihydro-5,6-dihydroxyindole\n",
      "            C06044  4-Hydroxyphenylethanol\n",
      "            C06046  Salidroside\n",
      "            C06047  Stizolobate\n",
      "            C06048  Stizolobinate\n",
      "            C06199  Hordenine\n",
      "            C06201  2,4-Dihydroxyhept-2-enedioate\n",
      "            C10447  3,4-Dihydroxyphenylpropanoate\n",
      "            C17935  Cysteinyldopa\n",
      "            C17936  Phaeomelanin\n",
      "            C17937  Eumelanin\n",
      "            C17938  5,6-Indolequinone-2-carboxylic acid\n",
      "            C22038  (R)-3-(3,4-Dihydroxyphenyl)lactate\n",
      "REFERENCE   PMID:8550403\n",
      "  AUTHORS   Prieto MA, Diaz E, Garcia JL.\n",
      "  TITLE     Molecular characterization of the 4-hydroxyphenylacetate catabolic pathway of Escherichia coli W: engineering a mobile aromatic degradative cluster.\n",
      "  JOURNAL   J Bacteriol 178:111-20 (1996)\n",
      "            DOI:10.1128/JB.178.1.111-120.1996\n",
      "REFERENCE   PMID:1194238\n",
      "  AUTHORS   Sparnins VL, Dagley S.\n",
      "  TITLE     Alternative routes of aromatic catabolism in Pseudomonas acidovorans and Pseudomonas putida: gallic acid as a substrate and inhibitor of dioxygenases.\n",
      "  JOURNAL   J Bacteriol 124:1374-81 (1975)\n",
      "            DOI:10.1128/JB.124.3.1374-1381.1975\n",
      "REFERENCE   PMID:18435614\n",
      "  AUTHORS   Ito S, Wakamatsu K\n",
      "  TITLE     Chemistry of mixed melanogenesis--pivotal roles of dopaquinone.\n",
      "  JOURNAL   Photochem Photobiol 84:582-92 (2008)\n",
      "            DOI:10.1111/j.1751-1097.2007.00238.x\n",
      "REFERENCE   PMID:11171088\n",
      "  AUTHORS   Olivares C, Jimenez-Cervantes C, Lozano JA, Solano F, Garcia-Borron JC\n",
      "  TITLE     The 5,6-dihydroxyindole-2-carboxylic acid (DHICA) oxidase activity of human tyrosinase.\n",
      "  JOURNAL   Biochem J 354:131-9 (2001)\n",
      "            DOI:10.1042/bj3540131\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00360  Phenylalanine metabolism\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00650  Butanoate metabolism\n",
      "KO_PATHWAY  ko00350\n",
      "///\n",
      "path:pae00360\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00360                    Pathway\n",
      "NAME        Phenylalanine metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00360  Phenylalanine metabolism\n",
      "DBLINKS     GO: 0006558\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0872  phhA; phenylalanine 4-monooxygenase [KO:K00500] [EC:1.14.16.1]\n",
      "line in gene PA0872  phhA; phenylalanine 4-monooxygenase [KO:K00500] [EC:1.14.16.1]\n",
      "            PA0421  hypothetical protein [KO:K00274] [EC:1.4.3.4]\n",
      "line in gene PA0421  hypothetical protein [KO:K00274] [EC:1.4.3.4]\n",
      "            PA4073  aldehyde dehydrogenase [KO:K00146] [EC:1.2.1.39]\n",
      "line in gene PA4073  aldehyde dehydrogenase [KO:K00146] [EC:1.2.1.39]\n",
      "            PA4980  enoyl-CoA hydratase [KO:K15866] [EC:5.3.3.18]\n",
      "line in gene PA4980  enoyl-CoA hydratase [KO:K15866] [EC:5.3.3.18]\n",
      "            PA1021  enoyl-CoA hydratase [KO:K15866] [EC:5.3.3.18]\n",
      "line in gene PA1021  enoyl-CoA hydratase [KO:K15866] [EC:5.3.3.18]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3590  3-hydroxyacyl-CoA dehydrogenase [KO:K00074] [EC:1.1.1.157]\n",
      "line in gene PA3590  3-hydroxyacyl-CoA dehydrogenase [KO:K00074] [EC:1.1.1.157]\n",
      "            PA1628  3-hydroxyacyl-CoA dehydrogenase [KO:K00074] [EC:1.1.1.157]\n",
      "line in gene PA1628  3-hydroxyacyl-CoA dehydrogenase [KO:K00074] [EC:1.1.1.157]\n",
      "            PA2531  aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA2531  aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA4447  hisC1; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA4447  hisC1; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA3165  hisC2; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA3165  hisC2; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA3139  aspartate aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "line in gene PA3139  aspartate aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "            PA0870  phhC; aromatic amino acid aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "line in gene PA0870  phhC; aromatic amino acid aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "            PA1377  hypothetical protein [KO:K03825] [EC:2.3.1.53 2.3.1.-]\n",
      "line in gene PA1377  hypothetical protein [KO:K03825] [EC:2.3.1.53 2.3.1.-]\n",
      "            PA0865  hpd; 4-hydroxyphenylpyruvate dioxygenase [KO:K00457] [EC:1.13.11.27]\n",
      "line in gene PA0865  hpd; 4-hydroxyphenylpyruvate dioxygenase [KO:K00457] [EC:1.13.11.27]\n",
      "            PA0242  hypothetical protein [KO:K00457] [EC:1.13.11.27]\n",
      "line in gene PA0242  hypothetical protein [KO:K00457] [EC:1.13.11.27]\n",
      "            PA5304  dadA; D-amino acid dehydrogenase small subunit [KO:K00285] [EC:1.4.5.1]\n",
      "line in gene PA5304  dadA; D-amino acid dehydrogenase small subunit [KO:K00285] [EC:1.4.5.1]\n",
      "            PA5084  oxidoreductase [KO:K00285] [EC:1.4.5.1]\n",
      "line in gene PA5084  oxidoreductase [KO:K00285] [EC:1.4.5.1]\n",
      "            PA0704  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA0704  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA0202  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA0202  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA3366  amiE; acylamide amidohydrolase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA3366  amiE; acylamide amidohydrolase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA4342  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA4342  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA4163  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA4163  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00042  Succinate\n",
      "            C00079  L-Phenylalanine\n",
      "            C00082  L-Tyrosine\n",
      "            C00084  Acetaldehyde\n",
      "            C00091  Succinyl-CoA\n",
      "            C00122  Fumarate\n",
      "            C00166  Phenylpyruvate\n",
      "            C00423  trans-Cinnamate\n",
      "            C00512  Benzoyl-CoA\n",
      "            C00582  Phenylacetyl-CoA\n",
      "            C00596  2-Hydroxy-2,4-pentadienoate\n",
      "            C00601  Phenylacetaldehyde\n",
      "            C00642  4-Hydroxyphenylacetate\n",
      "            C01198  3-(2-Hydroxyphenyl)propanoate\n",
      "            C01772  trans-2-Hydroxycinnamate\n",
      "            C02137  alpha-Oxo-benzeneacetic acid\n",
      "            C02232  3-Oxoadipyl-CoA\n",
      "            C02265  D-Phenylalanine\n",
      "            C02505  2-Phenylacetamide\n",
      "            C02763  2-Hydroxy-3-phenylpropenoate\n",
      "            C03519  N-Acetyl-L-phenylalanine\n",
      "            C03589  4-Hydroxy-2-oxopentanoate\n",
      "            C04044  3-(2,3-Dihydroxyphenyl)propanoate\n",
      "            C04148  Phenylacetylglutamine\n",
      "            C04479  2-Hydroxy-6-oxonona-2,4-diene-1,9-dioate\n",
      "            C05332  Phenethylamine\n",
      "            C05593  3-Hydroxyphenylacetate\n",
      "            C05598  Phenylacetylglycine\n",
      "            C05607  Phenyllactate\n",
      "            C05629  Phenylpropanoate\n",
      "            C05852  2-Hydroxyphenylacetate\n",
      "            C05853  Phenylethyl alcohol\n",
      "            C06207  2,6-Dihydroxyphenylacetate\n",
      "            C07086  Phenylacetic acid\n",
      "            C11457  3-(3-Hydroxyphenyl)propanoic acid\n",
      "            C11588  cis-3-(Carboxy-ethyl)-3,5-cyclo-hexadiene-1,2-diol\n",
      "            C12621  trans-3-Hydroxycinnamate\n",
      "            C12622  cis-3-(3-Carboxyethenyl)-3,5-cyclohexadiene-1,2-diol\n",
      "            C12623  trans-2,3-Dihydroxycinnamate\n",
      "            C12624  2-Hydroxy-6-ketononatrienedioate\n",
      "            C14144  5-Carboxy-2-pentenoyl-CoA\n",
      "            C14145  (3S)-3-Hydroxyadipyl-CoA\n",
      "            C15524  Phenylglyoxylyl-CoA\n",
      "            C19945  3-Oxo-5,6-dehydrosuberyl-CoA\n",
      "            C19946  3-Oxo-5,6-dehydrosuberyl-CoA semialdehyde\n",
      "            C19975  2-Oxepin-2(3H)-ylideneacetyl-CoA\n",
      "            C20062  2-(1,2-Epoxy-1,2-dihydrophenyl)acetyl-CoA\n",
      "REFERENCE   \n",
      "  AUTHORS   Nishizuka Y, Seyama Y, Ikai A, Ishimura Y, Kawaguchi A (eds).\n",
      "  TITLE     [Cellular Functions and Metabolic Maps] (In Japanese)\n",
      "  JOURNAL   Tokyo Kagaku Dojin (1997)\n",
      "REFERENCE   PMID:10537203\n",
      "  AUTHORS   Arai H, Yamamoto T, Ohishi T, Shimizu T, Nakata T, Kudo T.\n",
      "  TITLE     Genetic organization and characteristics of the 3-(3-hydroxyphenyl)propionic acid degradation pathway of Comamonas testosteroni TA441.\n",
      "  JOURNAL   Microbiology 145 ( Pt 10):2813-20 (1999)\n",
      "            DOI:10.1099/00221287-145-10-2813\n",
      "REFERENCE   PMID:9603882\n",
      "  AUTHORS   Diaz E, Ferrandez A, Garcia JL.\n",
      "  TITLE     Characterization of the hca cluster encoding the dioxygenolytic pathway for initial catabolism of 3-phenylpropionic acid in Escherichia coli K-12.\n",
      "  JOURNAL   J Bacteriol 180:2915-23 (1998)\n",
      "            DOI:10.1128/JB.180.11.2915-2923.1998\n",
      "REFERENCE   PMID:9098055\n",
      "  AUTHORS   Ferrandez A, Garcia JL, Diaz E.\n",
      "  TITLE     Genetic characterization and expression in heterologous hosts of the 3-(3-hydroxyphenyl)propionate catabolic pathway of Escherichia coli K-12.\n",
      "  JOURNAL   J Bacteriol 179:2573-81 (1997)\n",
      "            DOI:10.1128/JB.179.8.2573-2581.1997\n",
      "REFERENCE   PMID:19016485\n",
      "  AUTHORS   Hwang JY, Park J, Seo JH, Cha M, Cho BK, Kim J, Kim BG\n",
      "  TITLE     Simultaneous synthesis of 2-phenylethanol and L-homophenylalanine using aromatic transaminase with yeast Ehrlich pathway.\n",
      "  JOURNAL   Biotechnol Bioeng 102:1323-9 (2009)\n",
      "            DOI:10.1002/bit.22178\n",
      "REFERENCE   PMID:20660314\n",
      "  AUTHORS   Teufel R, Mascaraque V, Ismail W, Voss M, Perera J, Eisenreich W, Haehnel W, Fuchs G\n",
      "  TITLE     Bacterial phenylalanine and phenylacetate catabolic pathway revealed.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 107:14390-5 (2010)\n",
      "            DOI:10.1073/pnas.1005399107\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00350  Tyrosine metabolism\n",
      "            pae00362  Benzoate degradation\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "            pae00643  Styrene degradation\n",
      "KO_PATHWAY  ko00360\n",
      "///\n",
      "path:pae00361\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00361                    Pathway\n",
      "NAME        Chlorocyclohexane and chlorobenzene degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Xenobiotics biodegradation and metabolism\n",
      "PATHWAY_MAP pae00361  Chlorocyclohexane and chlorobenzene degradation\n",
      "DBLINKS     UMBBD: hch ghch dcz 2,4-d pcp 2,4,5-t tcb\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2507  catA; catechol 1,2-dioxygenase [KO:K03381] [EC:1.13.11.1]\n",
      "line in gene PA2507  catA; catechol 1,2-dioxygenase [KO:K03381] [EC:1.13.11.1]\n",
      "            PA2682  hypothetical protein [KO:K01061] [EC:3.1.1.45]\n",
      "line in gene PA2682  hypothetical protein [KO:K01061] [EC:3.1.1.45]\n",
      "            PA3498  oxidoreductase [KO:K21607] [EC:1.1.1.404]\n",
      "line in gene PA3498  oxidoreductase [KO:K21607] [EC:1.1.1.404]\n",
      "            PA2509  catB; muconate cycloisomerase I [KO:K01856] [EC:5.5.1.1]\n",
      "line in gene PA2509  catB; muconate cycloisomerase I [KO:K01856] [EC:5.5.1.1]\n",
      "            PA0810  haloacid dehalogenase [KO:K01560] [EC:3.8.1.2]\n",
      "line in gene PA0810  haloacid dehalogenase [KO:K01560] [EC:3.8.1.2]\n",
      "            PA2086  epoxide hydrolase [KO:K01561] [EC:3.8.1.3]\n",
      "line in gene PA2086  epoxide hydrolase [KO:K01561] [EC:3.8.1.3]\n",
      "COMPOUND    C00011  CO2\n",
      "            C00042  Succinate\n",
      "            C00090  Catechol\n",
      "            C00146  Phenol\n",
      "            C00160  Glycolate\n",
      "            C00530  Hydroquinone\n",
      "            C00682  2-Hydroxymuconate semialdehyde\n",
      "            C01407  Benzene\n",
      "            C02124  4-Chlorophenol\n",
      "            C02222  2-Maleylacetate\n",
      "            C02375  4-Chlorocatechol\n",
      "            C02575  Pentachlorophenol\n",
      "            C02625  2,4-Dichlorophenol\n",
      "            C02814  Benzene-1,2,4-triol\n",
      "            C02933  3,5-Dichlorocatechol\n",
      "            C03434  Tetrachlorohydroquinone\n",
      "            C03572  2-Chloro-cis,cis-muconate\n",
      "            C03585  3-Chloro-cis,cis-muconate\n",
      "            C03664  2,4-Dichlorophenoxyacetate\n",
      "            C03918  2,4-Dichloro-cis,cis-muconate\n",
      "            C04091  cis-1,2-Dihydrobenzene-1,2-diol\n",
      "            C04431  cis-4-Carboxymethylenebut-2-en-4-olide\n",
      "            C04522  2-Chloro-2,5-dihydro-5-oxofuran-2-acetate\n",
      "            C04706  cis-2-Chloro-4-carboxymethylenebut-2-en-1,4-olide\n",
      "            C04729  trans-2-Chloro-4-carboxymethylenebut-2-en-1,4-olide\n",
      "            C05618  3-Chlorocatechol\n",
      "            C06328  6-Chlorobenzene-1,2,4-triol\n",
      "            C06329  2-Chloromaleylacetate\n",
      "            C06594  1,2,4-Trichlorobenzene\n",
      "            C06596  gamma-Pentachlorocyclohexene\n",
      "            C06597  1,3,4,6-Tetrachloro-1,4-cyclohexadiene\n",
      "            C06598  2,4,5-Trichloro-2,5-cyclohexadiene-1-ol\n",
      "            C06599  2,5-Dichloro-2,5-cyclohexadiene-1,4-diol\n",
      "            C06600  2,5-Dichlorohydroquinone\n",
      "            C06601  Chlorohydroquinone\n",
      "            C06602  2,5-Dichlorophenol\n",
      "            C06603  cis,cis-4-Hydroxymuconic semialdehyde\n",
      "            C06755  Chloroacetic acid\n",
      "            C06988  beta-1,2,3,4,5,6-Hexachlorocyclohexane\n",
      "            C06989  delta-3,4,5,6-Tetrachlorocyclohexene\n",
      "            C06990  Chlorobenzene\n",
      "            C07075  Lindane\n",
      "            C07088  4-Chlorophenoxyacetate\n",
      "            C07089  5-Chloro-2-hydroxymuconic semialdehyde\n",
      "            C07090  Protoanemonin\n",
      "            C07091  cis-Acetylacrylate\n",
      "            C07092  1,4-Dichlorobenzene\n",
      "            C07093  3,6-Dichloro-cis-1,2-dihydroxycyclohexa-3,5-diene\n",
      "            C07094  3,6-Dichlorocatechol\n",
      "            C07095  2,5-Dichloro-cis,cis-muconate\n",
      "            C07096  2,6-Dichlorophenol\n",
      "            C07097  2,6-Dichlorohydroquinone\n",
      "            C07098  2,4,6-Trichlorophenol\n",
      "            C07099  2,3,6-Trichlorohydroquinone\n",
      "            C07100  2,4,5-Trichlorophenoxyacetic acid\n",
      "            C07101  2,4,5-Trichlorophenol\n",
      "            C07102  5-Chloro-1,2,4-trihydroxybenzene\n",
      "            C07103  2-Hydroxy-1,4-benzoquinone\n",
      "            C11352  5,6-Dichloro-1,3-cyclohexadiene\n",
      "            C12831  3,4,6-Trichlorocatechol\n",
      "            C12832  3,4,6-Trichloro-cis-1,2-dihydroxycyclohexa-3,5-diene\n",
      "            C12833  2,3,5-Trichloro-cis,cis-muconate\n",
      "            C12834  2,5-Dichloro-carboxymethylenebut-2-en-4-olide\n",
      "            C12835  2,5-Dichloro-4-oxohex-2-enedioate\n",
      "            C12836  2-Chloro-3-oxoadipate\n",
      "            C12837  3-Chloro-cis-1,2-dihydroxycyclohexa-3,5-diene\n",
      "            C12838  trans-4-Carboxymethylenebut-2-en-4-olide\n",
      "            C14419  2,4-Dichloroaniline\n",
      "            C14450  4-Chloroaniline\n",
      "            C16181  beta-2,3,4,5,6-Pentachlorocyclohexanol\n",
      "            C16182  beta-2,3,5,6-Tetrachloro-1,4-cyclohexanediol\n",
      "            C16266  3-Chloro-2-hydroxymuconic semialdehyde\n",
      "            C18236  1,2,3,4-Tetrachlorobenzene\n",
      "            C18238  cis-Chlorobenzene dihydrodiol\n",
      "            C18240  Tetrachlorocatechol\n",
      "            C18241  Tetrachloro-cis,cis-muconate\n",
      "            C18242  2,3,5-Trichlorodienelactone\n",
      "            C18243  2,3,5-Trichloromaleylacetate\n",
      "            C18244  2,4-Dichloro-3-oxoadipate\n",
      "            C18933  Chloranil\n",
      "            C21103  2,6-Dichloro-p-benzoquinone\n",
      "            C21104  2,5-Dichloro-p-benzoquinone\n",
      "            C21105  5-Chloro-2-hydroxy-p-benzoquinone\n",
      "REFERENCE   PMID:10600501\n",
      "  AUTHORS   Xun L, Bohuslavek J, Cai M.\n",
      "  TITLE     Characterization of 2,6-dichloro-p-hydroquinone 1,2-dioxygenase (PcpA) of Sphingomonas chlorophenolica ATCC 39723.\n",
      "  JOURNAL   Biochem Biophys Res Commun 266:322-5 (1999)\n",
      "            DOI:10.1006/bbrc.1999.1805\n",
      "REFERENCE   PMID:1987135\n",
      "  AUTHORS   van der Meer JR, van Neerven AR, de Vries EJ, de Vos WM, Zehnder AJ.\n",
      "  TITLE     Cloning and characterization of plasmid-encoded genes for the degradation of 1,2-dichloro-, 1,4-dichloro-, and 1,2,4-trichlorobenzene of Pseudomonas sp. strain P51.\n",
      "  JOURNAL   J Bacteriol 173:6-15 (1991)\n",
      "            DOI:10.1128/jb.173.1.6-15.1991\n",
      "REFERENCE   PMID:10464214\n",
      "  AUTHORS   Nagata Y, Futamura A, Miyauchi K, Takagi M.\n",
      "  TITLE     Two different types of dehalogenases, LinA and LinB, involved in gamma-hexachlorocyclohexane degradation in Sphingomonas paucimobilis UT26 are localized in the periplasmic space without molecular processing.\n",
      "  JOURNAL   J Bacteriol 181:5409-13 (1999)\n",
      "            DOI:10.1128/JB.181.17.5409-5413.1999\n",
      "REFERENCE   PMID:11423959\n",
      "  AUTHORS   Nagata Y, Miyauchi K, Takagi M.\n",
      "  TITLE     Complete analysis of genes and enzymes for gamma-hexachlorocyclohexane degradation in Sphingomonas paucimobilis UT26.\n",
      "  JOURNAL   J Ind Microbiol Biotechnol 23:380-390 (1999)\n",
      "            DOI:10.1038/sj/jim/2900736\n",
      "REFERENCE   PMID:15060023\n",
      "  AUTHORS   Dogra C, Raina V, Pal R, Suar M, Lal S, Gartemann KH, Holliger C, van der Meer JR, Lal R.\n",
      "  TITLE     Organization of lin genes and IS6100 among different strains of hexachlorocyclohexane-degrading Sphingomonas paucimobilis: evidence for horizontal gene transfer.\n",
      "  JOURNAL   J Bacteriol 186:2225-35 (2004)\n",
      "            DOI:10.1128/JB.186.8.2225-2235.2004\n",
      "REFERENCE   PMID:12676719\n",
      "  AUTHORS   Nagata Y, Prokop Z, Marvanova S, Sykorova J, Monincova M, Tsuda M, Damborsky J.\n",
      "  TITLE     Reconstruction of mycobacterial dehalogenase Rv2579 by cumulative mutagenesis of haloalkane dehalogenase LinB.\n",
      "  JOURNAL   Appl Environ Microbiol 69:2349-55 (2003)\n",
      "            DOI:10.1128/AEM.69.4.2349-2355.2003\n",
      "REFERENCE   PMID:15489427\n",
      "  AUTHORS   Vedler E, Vahter M, Heinaru A.\n",
      "  TITLE     The completely sequenced plasmid pEST4011 contains a novel IncP1 backbone and a catabolic transposon harboring tfd genes for 2,4-dichlorophenoxyacetic acid degradation.\n",
      "  JOURNAL   J Bacteriol 186:7161-74 (2004)\n",
      "            DOI:10.1128/JB.186.21.7161-7174.2004\n",
      "REFERENCE   PMID:9148781\n",
      "  AUTHORS   Sommer C, Gorisch H.\n",
      "  TITLE     Enzymology of the degradation of (di)chlorobenzenes by Xanthobacter flavus 14p1.\n",
      "  JOURNAL   Arch Microbiol 167:384-91 (1997)\n",
      "            DOI:10.1007/s002030050459\n",
      "REFERENCE   PMID:12070765\n",
      "  AUTHORS   Laemmli CM, Schonenberger R, Suter M, Zehnder AJ, van der Meer JR.\n",
      "  TITLE     TfdD(II), one of the two chloromuconate cycloisomerases of Ralstonia eutropha JMP134 (pJP4), cannot efficiently convert 2-chloro- cis, cis-muconate to trans-dienelactone to allow growth on 3-chlorobenzoate.\n",
      "  JOURNAL   Arch Microbiol 178:13-25 (2002)\n",
      "            DOI:10.1007/s00203-002-0417-3\n",
      "REFERENCE   PMID:16977465\n",
      "  AUTHORS   Wu J, Hong Q, Han P, He J, Li S.\n",
      "  TITLE     A gene linB2 responsible for the conversion of beta-HCH and 2,3,4,5,6-pentachlorocyclohexanol in Sphingomonas sp. BHC-A.\n",
      "  JOURNAL   Appl Microbiol Biotechnol 73:1097-105 (2007)\n",
      "            DOI:10.1007/s00253-006-0579-z\n",
      "REFERENCE   PMID:9758802\n",
      "  AUTHORS   Potrawfke T, Timmis KN, Wittich RM\n",
      "  TITLE     Degradation of 1,2,3,4-tetrachlorobenzene by pseudomonas chlororaphis RW71\n",
      "  JOURNAL   Appl Environ Microbiol 64:3798-806 (1998)\n",
      "            DOI:10.1128/AEM.64.10.3798-3806.1998\n",
      "REFERENCE   PMID:11208799\n",
      "  AUTHORS   Potrawfke T, Armengaud J, Wittich RM\n",
      "  TITLE     Chlorocatechols substituted at positions 4 and 5 are substrates of the broad-spectrum chlorocatechol 1,2-dioxygenase of Pseudomonas chlororaphis RW71.\n",
      "  JOURNAL   J Bacteriol 183:997-1011 (2001)\n",
      "            DOI:10.1128/JB.183.3.997-1011.2001\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00362  Benzoate degradation\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "KO_PATHWAY  ko00361\n",
      "///\n",
      "path:pae00362\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00362                    Pathway\n",
      "NAME        Benzoate degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Xenobiotics biodegradation and metabolism\n",
      "PATHWAY_MAP pae00362  Benzoate degradation\n",
      "MODULE      pae_M00551  Benzoate degradation, benzoate => catechol / methylbenzoate => methylcatechol [PATH:pae00362]\n",
      "            pae_M00568  Catechol ortho-cleavage, catechol => 3-oxoadipate [PATH:pae00362]\n",
      "DBLINKS     UMBBD: benz2 benz abs\n",
      "            GO: 0043639\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2518  xylX; toluate 1,2-dioxygenase subunit alpha [KO:K05549] [EC:1.14.12.10 1.14.12.-]\n",
      "line in gene PA2518  xylX; toluate 1,2-dioxygenase subunit alpha [KO:K05549] [EC:1.14.12.10 1.14.12.-]\n",
      "            PA2517  xylY; toluate 1,2-dioxygenase subunit beta [KO:K05550] [EC:1.14.12.10 1.14.12.-]\n",
      "line in gene PA2517  xylY; toluate 1,2-dioxygenase subunit beta [KO:K05550] [EC:1.14.12.10 1.14.12.-]\n",
      "            PA2516  xylZ; toluate 1,2-dioxygenase electron transfer subunit [KO:K05784] [EC:1.18.1.-]\n",
      "line in gene PA2516  xylZ; toluate 1,2-dioxygenase electron transfer subunit [KO:K05784] [EC:1.18.1.-]\n",
      "            PA2515  xylL; 1,6-dihydroxycyclohexa-2,4-diene-1-carboxylate dehydrogenase [KO:K05783] [EC:1.3.1.25 1.3.1.-]\n",
      "line in gene PA2515  xylL; 1,6-dihydroxycyclohexa-2,4-diene-1-carboxylate dehydrogenase [KO:K05783] [EC:1.3.1.25 1.3.1.-]\n",
      "            PA2507  catA; catechol 1,2-dioxygenase [KO:K03381] [EC:1.13.11.1]\n",
      "line in gene PA2507  catA; catechol 1,2-dioxygenase [KO:K03381] [EC:1.13.11.1]\n",
      "            PA2509  catB; muconate cycloisomerase I [KO:K01856] [EC:5.5.1.1]\n",
      "line in gene PA2509  catB; muconate cycloisomerase I [KO:K01856] [EC:5.5.1.1]\n",
      "            PA2508  catC; muconolactone delta-isomerase [KO:K03464] [EC:5.3.3.4]\n",
      "line in gene PA2508  catC; muconolactone delta-isomerase [KO:K03464] [EC:5.3.3.4]\n",
      "            PA0480  3-oxoadipate enol-lactonase [KO:K01055] [EC:3.1.1.24]\n",
      "line in gene PA0480  3-oxoadipate enol-lactonase [KO:K01055] [EC:3.1.1.24]\n",
      "            PA0231  pcaD; 3-oxoadipate enol-lactonase [KO:K01055] [EC:3.1.1.24]\n",
      "line in gene PA0231  pcaD; 3-oxoadipate enol-lactonase [KO:K01055] [EC:3.1.1.24]\n",
      "            PA3226  hydrolase [KO:K01055] [EC:3.1.1.24]\n",
      "line in gene PA3226  hydrolase [KO:K01055] [EC:3.1.1.24]\n",
      "            PA2940  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA2940  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA3013  foaB; 3-ketoacyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA3013  foaB; 3-ketoacyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA3454  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA3454  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA0228  pcaF; beta-ketoadipyl CoA thiolase [KO:K07823] [EC:2.3.1.174]\n",
      "line in gene PA0228  pcaF; beta-ketoadipyl CoA thiolase [KO:K07823] [EC:2.3.1.174]\n",
      "            PA0154  pcaG; protocatechuate 3,4-dioxygenase subunit alpha [KO:K00448] [EC:1.13.11.3]\n",
      "line in gene PA0154  pcaG; protocatechuate 3,4-dioxygenase subunit alpha [KO:K00448] [EC:1.13.11.3]\n",
      "            PA0153  pcaH; protocatechuate 3,4-dioxygenase subunit beta [KO:K00449] [EC:1.13.11.3]\n",
      "line in gene PA0153  pcaH; protocatechuate 3,4-dioxygenase subunit beta [KO:K00449] [EC:1.13.11.3]\n",
      "            PA0230  pcaB; 3-carboxy-cis,cis-muconate cycloisomerase [KO:K01857] [EC:5.5.1.2]\n",
      "line in gene PA0230  pcaB; 3-carboxy-cis,cis-muconate cycloisomerase [KO:K01857] [EC:5.5.1.2]\n",
      "            PA0232  pcaC; 4-carboxymuconolactone decarboxylase [KO:K01607] [EC:4.1.1.44]\n",
      "line in gene PA0232  pcaC; 4-carboxymuconolactone decarboxylase [KO:K01607] [EC:4.1.1.44]\n",
      "            PA4486  hypothetical protein [KO:K01607] [EC:4.1.1.44]\n",
      "line in gene PA4486  hypothetical protein [KO:K01607] [EC:4.1.1.44]\n",
      "            PA0247  pobA; 4-hydroxybenzoate 3-monooxygenase [KO:K00481] [EC:1.14.13.2]\n",
      "line in gene PA0247  pobA; 4-hydroxybenzoate 3-monooxygenase [KO:K00481] [EC:1.14.13.2]\n",
      "            PA0968  hypothetical protein [KO:K01075] [EC:3.1.2.23]\n",
      "line in gene PA0968  hypothetical protein [KO:K01075] [EC:3.1.2.23]\n",
      "            PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "line in gene PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "            PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "line in gene PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "            PA0447  gcdH; glutaryl-CoA dehydrogenase [KO:K00252] [EC:1.3.8.6]\n",
      "line in gene PA0447  gcdH; glutaryl-CoA dehydrogenase [KO:K00252] [EC:1.3.8.6]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3590  3-hydroxyacyl-CoA dehydrogenase [KO:K00074] [EC:1.1.1.157]\n",
      "line in gene PA3590  3-hydroxyacyl-CoA dehydrogenase [KO:K00074] [EC:1.1.1.157]\n",
      "            PA1628  3-hydroxyacyl-CoA dehydrogenase [KO:K00074] [EC:1.1.1.157]\n",
      "line in gene PA1628  3-hydroxyacyl-CoA dehydrogenase [KO:K00074] [EC:1.1.1.157]\n",
      "            PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00036  Oxaloacetate\n",
      "            C00084  Acetaldehyde\n",
      "            C00090  Catechol\n",
      "            C00091  Succinyl-CoA\n",
      "            C00146  Phenol\n",
      "            C00156  4-Hydroxybenzoate\n",
      "            C00180  Benzoate\n",
      "            C00196  2,3-Dihydroxybenzoate\n",
      "            C00230  3,4-Dihydroxybenzoate\n",
      "            C00332  Acetoacetyl-CoA\n",
      "            C00512  Benzoyl-CoA\n",
      "            C00527  Glutaryl-CoA\n",
      "            C00530  Hydroquinone\n",
      "            C00587  3-Hydroxybenzoate\n",
      "            C00596  2-Hydroxy-2,4-pentadienoate\n",
      "            C00628  2,5-Dihydroxybenzoate\n",
      "            C00682  2-Hydroxymuconate semialdehyde\n",
      "            C00846  3-Oxoadipate\n",
      "            C00877  Crotonoyl-CoA\n",
      "            C01063  6-Carboxyhexanoyl-CoA\n",
      "            C01144  (S)-3-Hydroxybutanoyl-CoA\n",
      "            C01163  3-Carboxy-cis,cis-muconate\n",
      "            C01278  2-Carboxy-2,5-dihydro-5-oxofuran-2-acetate\n",
      "            C01407  Benzene\n",
      "            C01751  Resorcinol\n",
      "            C02222  2-Maleylacetate\n",
      "            C02232  3-Oxoadipyl-CoA\n",
      "            C02411  Glutaconyl-1-CoA\n",
      "            C02480  cis,cis-Muconate\n",
      "            C02501  2-Hydroxymuconate\n",
      "            C02519  4-Methoxybenzoate\n",
      "            C02814  Benzene-1,2,4-triol\n",
      "            C02949  4-Hydroxybenzoyl-CoA\n",
      "            C03453  gamma-Oxalocrotonate\n",
      "            C03586  2-Oxo-2,3-dihydrofuran-5-acetate\n",
      "            C03589  4-Hydroxy-2-oxopentanoate\n",
      "            C03671  2-Pyrone-4,6-dicarboxylate\n",
      "            C03676  3-Hydroxy-cis,cis-muconate\n",
      "            C04091  cis-1,2-Dihydrobenzene-1,2-diol\n",
      "            C04115  4-Carboxy-4-hydroxy-2-oxoadipate\n",
      "            C04324  (1E,3E)-4-Hydroxybuta-1,3-diene-1,2,4-tricarboxylate\n",
      "            C04434  (1E)-4-Oxobut-1-ene-1,2,4-tricarboxylate\n",
      "            C04480  3-Carboxy-2-hydroxymuconate semialdehyde\n",
      "            C04484  4-Carboxy-2-hydroxymuconate semialdehyde\n",
      "            C04553  3-Carboxy-2,5-dihydro-5-oxofuran-2-acetate\n",
      "            C05375  2-Hydroxy-2-hydropyrone-4,6-dicarboxylate\n",
      "            C06033  Parapyruvate\n",
      "            C06035  4-Methylene-2-oxoglutarate\n",
      "            C06321  (1R,6S)-1,6-Dihydroxycyclohexa-2,4-diene-1-carboxylate\n",
      "            C06322  Cyclohexa-1,5-diene-1-carbonyl-CoA\n",
      "            C06333  2-Aminobenzenesulfonate\n",
      "            C06336  3-Sulfocatechol\n",
      "            C06714  3-Hydroxypimeloyl-CoA\n",
      "            C06715  3-Oxopimeloyl-CoA\n",
      "            C06723  6-Carboxyhex-2-enoyl-CoA\n",
      "            C06749  6-Hydroxycyclohex-1-ene-1-carbonyl-CoA\n",
      "            C09809  Cyclohex-2,5-diene-1-carboxyl-CoA\n",
      "            C09810  Cyclohex-1,4-diene-1-carboxyl-CoA\n",
      "            C09811  Cyclohex-1-ene-1-carboxyl-CoA\n",
      "            C09812  2-Hydroxycyclohexane-1-carboxyl-CoA\n",
      "            C09813  2-Ketocyclohexane-1-carboxyl-CoA\n",
      "            C09821  6-Oxocyclohex-1-ene-1-carbonyl-CoA\n",
      "            C09822  Cyclohexane-1-carboxylate\n",
      "            C09823  Cyclohexane-1-carboxyl-CoA\n",
      "            C14144  5-Carboxy-2-pentenoyl-CoA\n",
      "            C14610  (S)-5-Oxo-2,5-dihydrofuran-2-acetate\n",
      "            C19682  3,4-Didehydroadipyl-CoA\n",
      "            C19683  3,4-Didehydroadipyl-CoA semialdehyde\n",
      "            C19684  2,3-Epoxy-2,3-dihydrobenzoyl-CoA\n",
      "            C21298  2,6-Dihydroxybenzoate\n",
      "REFERENCE   PMID:19258534\n",
      "  AUTHORS   Carmona M, Zamarro MT, Blazquez B, Durante-Rodriguez G, Juarez JF, Valderrama JA, Barragan MJ, Garcia JL, Diaz E\n",
      "  TITLE     Anaerobic catabolism of aromatic compounds: a genetic and genomic view.\n",
      "  JOURNAL   Microbiol Mol Biol Rev 73:71-133 (2009)\n",
      "            DOI:10.1128/MMBR.00021-08\n",
      "REFERENCE   PMID:12486039\n",
      "  AUTHORS   Hara H, Masai E, Miyauchi K, Katayama Y, Fukuda M.\n",
      "  TITLE     Characterization of the 4-carboxy-4-hydroxy-2-oxoadipate aldolase gene and operon structure of the protocatechuate 4,5-cleavage pathway genes in Sphingomonas paucimobilis SYK-6.\n",
      "  JOURNAL   J Bacteriol 185:41-50 (2003)\n",
      "            DOI:10.1128/JB.185.1.41-50.2003\n",
      "REFERENCE   PMID:15277747\n",
      "  AUTHORS   Maruyama K, Shibayama T, Ichikawa A, Sakou Y, Yamada S, Sugisaki H.\n",
      "  TITLE     Cloning and characterization of the genes encoding enzymes for the protocatechuate meta-degradation pathway of Pseudomonas ochraceae NGJ1.\n",
      "  JOURNAL   Biosci Biotechnol Biochem 68:1434-41 (2004)\n",
      "            DOI:10.1271/bbb.68.1434\n",
      "REFERENCE   PMID:10589735\n",
      "  AUTHORS   Mampel J, Ruff J, Junker F, Cook AM.\n",
      "  TITLE     The oxygenase component of the 2-aminobenzenesulfonate dioxygenase system from Alcaligenes sp. strain O-1.\n",
      "  JOURNAL   Microbiology 145 ( Pt 11):3255-64 (1999)\n",
      "            DOI:10.1099/00221287-145-11-3255\n",
      "REFERENCE   PMID:8002948\n",
      "  AUTHORS   Junker F, Field JA, Bangerter F, Ramsteiner K, Kohler HP, Joannou CL, Mason JR, Leisinger T, Cook AM.\n",
      "  TITLE     Oxygenation and spontaneous deamination of 2-aminobenzenesulphonic acid in Alcaligenes sp. strain O-1 with subsequent meta ring cleavage and spontaneous desulphonation to 2-hydroxymuconic acid.\n",
      "  JOURNAL   Biochem J 300 ( Pt 2):429-36 (1994)\n",
      "            DOI:10.1042/bj3000429\n",
      "REFERENCE   PMID:8990279\n",
      "  AUTHORS   Harwood CS, Gibson J.\n",
      "  TITLE     Shedding light on anaerobic benzene ring degradation: a process unique to prokaryotes?\n",
      "  JOURNAL   J Bacteriol 179:301-9 (1997)\n",
      "            DOI:10.1128/JB.179.2.301-309.1997\n",
      "REFERENCE   PMID:17122342\n",
      "  AUTHORS   Peters F, Shinoda Y, McInerney MJ, Boll M.\n",
      "  TITLE     Cyclohexa-1,5-diene-1-carbonyl-coenzyme A (CoA) hydratases of Geobacter metallireducens and Syntrophus aciditrophicus: Evidence for a common benzoyl-CoA degradation pathway in facultative and strict anaerobes.\n",
      "  JOURNAL   J Bacteriol 189:1055-60 (2007)\n",
      "            DOI:10.1128/JB.01467-06\n",
      "REFERENCE   PMID:11053377\n",
      "  AUTHORS   Cowles CE, Nichols NN, Harwood CS.\n",
      "  TITLE     BenR, a XylS homologue, regulates three different pathways of aromatic acid degradation in Pseudomonas putida.\n",
      "  JOURNAL   J Bacteriol 182:6339-46 (2000)\n",
      "            DOI:10.1128/JB.182.22.6339-6346.2000\n",
      "REFERENCE   PMID:12514037\n",
      "  AUTHORS   Smith MA, Weaver VB, Young DM, Ornston LN.\n",
      "  TITLE     Genes for chlorogenate and hydroxycinnamate catabolism (hca) are linked to functionally related genes in the dca-pca-qui-pob-hca chromosomal cluster of Acinetobacter sp. strain ADP1.\n",
      "  JOURNAL   Appl Environ Microbiol 69:524-32 (2003)\n",
      "            DOI:10.1128/AEM.69.1.524-532.2003\n",
      "REFERENCE   PMID:18312395\n",
      "  AUTHORS   Kuntze K, Shinoda Y, Moutakki H, McInerney MJ, Vogt C, Richnow HH, Boll M\n",
      "  TITLE     6-Oxocyclohex-1-ene-1-carbonyl-coenzyme A hydrolases from obligately anaerobic bacteria: characterization and identification of its gene as a functional marker for aromatic compounds degrading anaerobes.\n",
      "  JOURNAL   Environ Microbiol 10:1547-56 (2008)\n",
      "            DOI:10.1111/j.1462-2920.2008.01570.x\n",
      "REFERENCE   PMID:10781543\n",
      "  AUTHORS   Pelletier DA, Harwood CS\n",
      "  TITLE     2-Hydroxycyclohexanecarboxyl coenzyme A dehydrogenase, an enzyme characteristic of the anaerobic benzoate degradation pathway used by Rhodopseudomonas palustris.\n",
      "  JOURNAL   J Bacteriol 182:2753-60 (2000)\n",
      "            DOI:10.1128/JB.182.10.2753-2760.2000\n",
      "REFERENCE   PMID:11282627\n",
      "  AUTHORS   Elshahed MS, Bhupathiraju VK, Wofford NQ, Nanny MA, McInerney MJ\n",
      "  TITLE     Metabolism of benzoate, cyclohex-1-ene carboxylate, and cyclohexane carboxylate by \"Syntrophus aciditrophicus\" strain SB in syntrophic association with H(2)-using microorganisms.\n",
      "  JOURNAL   Appl Environ Microbiol 67:1728-38 (2001)\n",
      "            DOI:10.1128/AEM.67.4.1728-1738.2001\n",
      "REFERENCE   PMID:21219457\n",
      "  AUTHORS   Nogales J, Canales A, Jimenez-Barbero J, Serra B, Pingarron JM, Garcia JL, Diaz E\n",
      "  TITLE     Unravelling the gallic acid degradation pathway in bacteria: the gal cluster from Pseudomonas putida.\n",
      "  JOURNAL   Mol Microbiol 79:359-74 (2011)\n",
      "            DOI:10.1111/j.1365-2958.2010.07448.x\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00350  Tyrosine metabolism\n",
      "            pae00360  Phenylalanine metabolism\n",
      "            pae00361  Chlorocyclohexane and chlorobenzene degradation\n",
      "            pae00627  Aminobenzoate degradation\n",
      "            pae00790  Folate biosynthesis\n",
      "KO_PATHWAY  ko00362\n",
      "///\n",
      "path:pae00364\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00364                    Pathway\n",
      "NAME        Fluorobenzoate degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Xenobiotics biodegradation and metabolism\n",
      "PATHWAY_MAP pae00364  Fluorobenzoate degradation\n",
      "DBLINKS     UMBBD: 3fb 4fb box dcb\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2518  xylX; toluate 1,2-dioxygenase subunit alpha [KO:K05549] [EC:1.14.12.10 1.14.12.-]\n",
      "line in gene PA2518  xylX; toluate 1,2-dioxygenase subunit alpha [KO:K05549] [EC:1.14.12.10 1.14.12.-]\n",
      "            PA2517  xylY; toluate 1,2-dioxygenase subunit beta [KO:K05550] [EC:1.14.12.10 1.14.12.-]\n",
      "line in gene PA2517  xylY; toluate 1,2-dioxygenase subunit beta [KO:K05550] [EC:1.14.12.10 1.14.12.-]\n",
      "            PA2516  xylZ; toluate 1,2-dioxygenase electron transfer subunit [KO:K05784] [EC:1.18.1.-]\n",
      "line in gene PA2516  xylZ; toluate 1,2-dioxygenase electron transfer subunit [KO:K05784] [EC:1.18.1.-]\n",
      "            PA2515  xylL; 1,6-dihydroxycyclohexa-2,4-diene-1-carboxylate dehydrogenase [KO:K05783] [EC:1.3.1.25 1.3.1.-]\n",
      "line in gene PA2515  xylL; 1,6-dihydroxycyclohexa-2,4-diene-1-carboxylate dehydrogenase [KO:K05783] [EC:1.3.1.25 1.3.1.-]\n",
      "            PA2507  catA; catechol 1,2-dioxygenase [KO:K03381] [EC:1.13.11.1]\n",
      "line in gene PA2507  catA; catechol 1,2-dioxygenase [KO:K03381] [EC:1.13.11.1]\n",
      "            PA2682  hypothetical protein [KO:K01061] [EC:3.1.1.45]\n",
      "line in gene PA2682  hypothetical protein [KO:K01061] [EC:3.1.1.45]\n",
      "            PA2509  catB; muconate cycloisomerase I [KO:K01856] [EC:5.5.1.1]\n",
      "line in gene PA2509  catB; muconate cycloisomerase I [KO:K01856] [EC:5.5.1.1]\n",
      "COMPOUND    C00090  Catechol\n",
      "            C02222  2-Maleylacetate\n",
      "            C02357  2-Chlorobenzoate\n",
      "            C02359  2-Fluorobenzoate\n",
      "            C02364  3-Fluorobenzoate\n",
      "            C02370  4-Chlorobenzoate\n",
      "            C02371  4-Fluorobenzoate\n",
      "            C02802  4-Fluorobenzoyl-CoA\n",
      "            C02949  4-Hydroxybenzoyl-CoA\n",
      "            C03925  3,5-Dibromo-4-hydroxybenzoate\n",
      "            C04178  Bromoxynil\n",
      "            C06387  4-Chlorobenzoyl-CoA\n",
      "            C06670  2,4-Dichlorobenzoate\n",
      "            C06671  2,4-Dichlorobenzoyl-CoA\n",
      "            C11272  Fluorobenzene\n",
      "            C16246  3,5-Dibromo-4-hydroxybenzamide\n",
      "            C16247  2,6-Dibromophenol\n",
      "            C16248  2,6-Dibromohydroquinone\n",
      "            C16249  2-Bromomaleylacetate\n",
      "            C16472  3-Fluorocatechol\n",
      "            C16473  4-Fluorocatechol\n",
      "            C16474  3-Fluoro-cis,cis-muconate\n",
      "            C16475  2-Fluoro-cis,cis-muconate\n",
      "            C16476  4-Fluoromuconolactone\n",
      "            C16477  5-Fluoromuconolactone\n",
      "            C16478  3-Fluorocyclohexadiene-cis,cis-1,2-diol-1-carboxylate\n",
      "            C16479  5-Fluorocyclohexadiene-cis,cis-1,2-diol-1-carboxylate\n",
      "            C16480  4-Fluorocyclohexadiene-cis,cis-1,2-diol-1-carboxylate\n",
      "            C16481  6-Fluorocyclohexadiene-cis,cis-1,2-diol-1-carboxylate\n",
      "            C16482  2-Fluorocyclohexadiene-cis,cis-1,2-diol-1-carboxylate\n",
      "            C16483  4-Fluorocyclohexadiene-cis,cis-1,2-diol\n",
      "            C16484  1-Fluorocyclohexadiene-cis,cis-1,2-diol\n",
      "REFERENCE   PMID:16345496\n",
      "  AUTHORS   Schreiber A, Hellwig M, Dorn E, Reineke W, Knackmuss HJ.\n",
      "  TITLE     Critical Reactions in Fluorobenzoic Acid Degradation by Pseudomonas sp. B13.\n",
      "  JOURNAL   Appl Environ Microbiol 39:58-67 (1980)\n",
      "            DOI:10.1128/AEM.39.1.58-67.1980\n",
      "REFERENCE   PMID:12957954\n",
      "  AUTHORS   Solyanikova IP, Moiseeva OV, Boeren S, Boersma MG, Kolomytseva MP, Vervoort J, Rietjens IM, Golovleva LA, van Berkel WJ.\n",
      "  TITLE     Conversion of 2-fluoromuconate to cis-dienelactone by purified enzymes of Rhodococcus opacus 1cp.\n",
      "  JOURNAL   Appl Environ Microbiol 69:5636-42 (2003)\n",
      "            DOI:10.1128/AEM.69.9.5636-5642.2003\n",
      "REFERENCE   PMID:16980423\n",
      "  AUTHORS   Carvalho MF, Ferreira MI, Moreira IS, Castro PM, Janssen DB.\n",
      "  TITLE     Degradation of fluorobenzene by Rhizobiales strain F11 via ortho cleavage of 4-fluorocatechol and catechol.\n",
      "  JOURNAL   Appl Environ Microbiol 72:7413-7 (2006)\n",
      "            DOI:10.1128/AEM.01162-06\n",
      "REFERENCE   PMID:18355280\n",
      "  AUTHORS   Chae JC, Song B, Zylstra GJ\n",
      "  TITLE     Identification of genes coding for hydrolytic dehalogenation in the metagenome derived from a denitrifying 4-chlorobenzoate degrading consortium.\n",
      "  JOURNAL   FEMS Microbiol Lett 281:203-9 (2008)\n",
      "            DOI:10.1111/j.1574-6968.2008.01106.x\n",
      "REFERENCE   \n",
      "  AUTHORS   Rosenbrock P, Munch JC, Scheunert I, Dorfler U.\n",
      "  TITLE     Biodegradation of the herbicide bromoxynil and its plant cell wall bound residues in an agricultural soil.\n",
      "  JOURNAL   Pestic Biochem Physiol 78:49-57 (2004)\n",
      "            DOI:10.1016/j.pestbp.2003.09.002\n",
      "REFERENCE   PMID:1610174\n",
      "  AUTHORS   Topp E, Xun LY, Orser CS.\n",
      "  TITLE     Biodegradation of the herbicide bromoxynil (3,5-dibromo-4-hydroxybenzonitrile) by purified pentachlorophenol hydroxylase and whole cells of Flavobacterium sp. strain ATCC 39723 is accompanied by cyanogenesis.\n",
      "  JOURNAL   Appl Environ Microbiol 58:502-6 (1992)\n",
      "            DOI:10.1128/AEM.58.2.502-506.1992\n",
      "REL_PATHWAY pae00362  Benzoate degradation\n",
      "KO_PATHWAY  ko00364\n",
      "///\n",
      "path:pae00380\n",
      "ENTRY       pae00380                    Pathway\n",
      "NAME        Tryptophan metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00380  Tryptophan metabolism\n",
      "DBLINKS     GO: 0006568 0009851\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2579  kynA; tryptophan 2,3-dioxygenase [KO:K00453] [EC:1.13.11.11]\n",
      "line in gene PA2579  kynA; tryptophan 2,3-dioxygenase [KO:K00453] [EC:1.13.11.11]\n",
      "            PA2081  kynB; kynurenine formamidase KynB [KO:K07130] [EC:3.5.1.9]\n",
      "line in gene PA2081  kynB; kynurenine formamidase KynB [KO:K07130] [EC:3.5.1.9]\n",
      "            PA2080  kynU; kynureninase KynU [KO:K01556] [EC:3.7.1.3]\n",
      "line in gene PA2080  kynU; kynureninase KynU [KO:K01556] [EC:3.7.1.3]\n",
      "            PA1586  sucB; 2-oxoglutarate dehydrogenase complex dihydrolipoyllysine-residue succinyltransferase [KO:K00658] [EC:2.3.1.61]\n",
      "line in gene PA1586  sucB; 2-oxoglutarate dehydrogenase complex dihydrolipoyllysine-residue succinyltransferase [KO:K00658] [EC:2.3.1.61]\n",
      "            PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA0447  gcdH; glutaryl-CoA dehydrogenase [KO:K00252] [EC:1.3.8.6]\n",
      "line in gene PA0447  gcdH; glutaryl-CoA dehydrogenase [KO:K00252] [EC:1.3.8.6]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "line in gene PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "            PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "line in gene PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "            PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA0421  hypothetical protein [KO:K00274] [EC:1.4.3.4]\n",
      "line in gene PA0421  hypothetical protein [KO:K00274] [EC:1.4.3.4]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA0704  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA0704  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA0202  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA0202  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA3366  amiE; acylamide amidohydrolase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA3366  amiE; acylamide amidohydrolase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA4342  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA4342  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA4163  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA4163  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA2147  katE; catalase HPII [KO:K03781] [EC:1.11.1.6]\n",
      "line in gene PA2147  katE; catalase HPII [KO:K03781] [EC:1.11.1.6]\n",
      "            PA4613  katB; catalase [KO:K03781] [EC:1.11.1.6]\n",
      "line in gene PA4613  katB; catalase [KO:K03781] [EC:1.11.1.6]\n",
      "            PA4236  katA; catalase [KO:K03781] [EC:1.11.1.6]\n",
      "line in gene PA4236  katA; catalase [KO:K03781] [EC:1.11.1.6]\n",
      "COMPOUND    C00024  Acetyl-CoA\n",
      "            C00078  L-Tryptophan\n",
      "            C00108  Anthranilate\n",
      "            C00322  2-Oxoadipate\n",
      "            C00328  L-Kynurenine\n",
      "            C00331  Indolepyruvate\n",
      "            C00332  Acetoacetyl-CoA\n",
      "            C00398  Tryptamine\n",
      "            C00463  Indole\n",
      "            C00527  Glutaryl-CoA\n",
      "            C00632  3-Hydroxyanthranilate\n",
      "            C00637  Indole-3-acetaldehyde\n",
      "            C00643  5-Hydroxy-L-tryptophan\n",
      "            C00780  Serotonin\n",
      "            C00877  Crotonoyl-CoA\n",
      "            C00954  Indole-3-acetate\n",
      "            C00955  Indole-3-ethanol\n",
      "            C00978  N-Acetylserotonin\n",
      "            C01111  7,8-Dihydroxykynurenate\n",
      "            C01144  (S)-3-Hydroxybutanoyl-CoA\n",
      "            C01249  7,8-Dihydro-7,8-dihydroxykynurenate\n",
      "            C01252  4-(2-Aminophenyl)-2,4-dioxobutanoate\n",
      "            C01598  Melatonin\n",
      "            C01717  4-Hydroxy-2-quinolinecarboxylic acid\n",
      "            C01987  2-Aminophenol\n",
      "            C02161  2-Aminophenoxazin-3-one\n",
      "            C02172  N-Acetylisatin\n",
      "            C02220  2-Aminomuconate\n",
      "            C02298  N-Acetylindoxyl\n",
      "            C02470  Xanthurenic acid\n",
      "            C02693  (Indol-3-yl)acetamide\n",
      "            C02700  L-Formylkynurenine\n",
      "            C02775  2,3-Dihydroxyindole\n",
      "            C02937  Indole-3-acetaldehyde oxime\n",
      "            C02938  3-Indoleacetonitrile\n",
      "            C03227  3-Hydroxy-L-kynurenine\n",
      "            C03230  (Indol-3-yl)glycolaldehyde\n",
      "            C03453  gamma-Oxalocrotonate\n",
      "            C03574  2-Formylaminobenzaldehyde\n",
      "            C03722  Quinolinate\n",
      "            C03824  2-Aminomuconate semialdehyde\n",
      "            C04409  2-Amino-3-carboxymuconate semialdehyde\n",
      "            C05634  5-Hydroxyindoleacetaldehyde\n",
      "            C05635  5-Hydroxyindoleacetate\n",
      "            C05636  3-Hydroxykynurenamine\n",
      "            C05637  4,8-Dihydroxyquinoline\n",
      "            C05638  5-Hydroxykynurenamine\n",
      "            C05639  4,6-Dihydroxyquinoline\n",
      "            C05640  Cinnavalininate\n",
      "            C05641  5-(3'-Carboxy-3'-oxopropenyl)-4,6-dihydroxypicolinate\n",
      "            C05642  Formyl-N-acetyl-5-methoxykynurenamine\n",
      "            C05643  6-Hydroxymelatonin\n",
      "            C05644  3-Methylindolepyruvate\n",
      "            C05645  4-(2-Amino-3-hydroxyphenyl)-2,4-dioxobutanoate\n",
      "            C05646  5-Hydroxyindolepyruvate\n",
      "            C05647  Formyl-5-hydroxykynurenamine\n",
      "            C05648  5-Hydroxy-N-formylkynurenine\n",
      "            C05651  5-Hydroxykynurenine\n",
      "            C05652  4-(2-Amino-5-hydroxyphenyl)-2,4-dioxobutanoate\n",
      "            C05653  Formylanthranilate\n",
      "            C05654  5-(2'-Formylethyl)-4,6-dihydroxypicolinate\n",
      "            C05655  5-(2'-Carboxyethyl)-4,6-dihydroxypicolinate\n",
      "            C05656  5-(3'-Carboxy-3'-oxopropyl)-4,6-dihydroxypicolinate\n",
      "            C05658  Indoxyl\n",
      "            C05659  5-Methoxytryptamine\n",
      "            C05660  5-Methoxyindoleacetate\n",
      "            C05663  6-Hydroxykynurenate\n",
      "            C05830  8-Methoxykynurenate\n",
      "            C05831  3-Methoxyanthranilate\n",
      "            C05832  5-Hydroxyindoleacetylglycine\n",
      "            C05834  3-Methyldioxyindole\n",
      "            C05835  2-Formaminobenzoylacetate\n",
      "            C05837  Glucobrassicin\n",
      "            C06212  N-Methylserotonin\n",
      "            C06213  N-Methyltryptamine\n",
      "            C08313  Skatole\n",
      "            C10164  Picolinic acid\n",
      "            C16516  Indolylmethylthiohydroximate\n",
      "            C16517  Indolylmethyl-desulfoglucosinolate\n",
      "            C16518  S-(Indolylmethylthiohydroximoyl)-L-cysteine\n",
      "            C17203  N-Hydroxyl-tryptamine\n",
      "            C22006  (R)-(Indol-3-yl)lactate\n",
      "            C22202  2-Oxoindole-3-acetate\n",
      "REFERENCE   PMID:15749753\n",
      "  AUTHORS   Woodward AW, Bartel B\n",
      "  TITLE     Auxin: regulation, action, and interaction.\n",
      "  JOURNAL   Ann Bot 95:707-35 (2005)\n",
      "            DOI:10.1093/aob/mci083\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00362  Benzoate degradation\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "            pae00627  Aminobenzoate degradation\n",
      "            pae00760  Nicotinate and nicotinamide metabolism\n",
      "KO_PATHWAY  ko00380\n",
      "///\n",
      "path:pae00400\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00400                    Pathway\n",
      "NAME        Phenylalanine, tyrosine and tryptophan biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Amino acid metabolism\n",
      "PATHWAY_MAP pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "MODULE      pae_M00023  Tryptophan biosynthesis, chorismate => tryptophan [PATH:pae00400]\n",
      "            pae_M00024  Phenylalanine biosynthesis, chorismate => phenylpyruvate => phenylalanine [PATH:pae00400]\n",
      "DBLINKS     GO: 0009094 0006571 0000162\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1750  phospho-2-dehydro-3-deoxyheptonate aldolase [KO:K01626] [EC:2.5.1.54]\n",
      "line in gene PA1750  phospho-2-dehydro-3-deoxyheptonate aldolase [KO:K01626] [EC:2.5.1.54]\n",
      "            PA1901  phzC2; phenazine biosynthesis protein PhzC [KO:K01626] [EC:2.5.1.54]\n",
      "line in gene PA1901  phzC2; phenazine biosynthesis protein PhzC [KO:K01626] [EC:2.5.1.54]\n",
      "            PA2843  aldolase [KO:K01626] [EC:2.5.1.54]\n",
      "line in gene PA2843  aldolase [KO:K01626] [EC:2.5.1.54]\n",
      "            PA2943  phospho-2-dehydro-3-deoxyheptonate aldolase [KO:K01626] [EC:2.5.1.54]\n",
      "line in gene PA2943  phospho-2-dehydro-3-deoxyheptonate aldolase [KO:K01626] [EC:2.5.1.54]\n",
      "            PA4212  phzC1; phenazine biosynthesis protein PhzC [KO:K01626] [EC:2.5.1.54]\n",
      "line in gene PA4212  phzC1; phenazine biosynthesis protein PhzC [KO:K01626] [EC:2.5.1.54]\n",
      "            PA5038  aroB; 3-dehydroquinate synthase [KO:K01735] [EC:4.2.3.4]\n",
      "line in gene PA5038  aroB; 3-dehydroquinate synthase [KO:K01735] [EC:4.2.3.4]\n",
      "            PA4846  aroQ1; 3-dehydroquinate dehydratase [KO:K03786] [EC:4.2.1.10]\n",
      "line in gene PA4846  aroQ1; 3-dehydroquinate dehydratase [KO:K03786] [EC:4.2.1.10]\n",
      "            PA0245  aroQ2; 3-dehydroquinate dehydratase [KO:K03786] [EC:4.2.1.10]\n",
      "line in gene PA0245  aroQ2; 3-dehydroquinate dehydratase [KO:K03786] [EC:4.2.1.10]\n",
      "            PA0025  aroE; shikimate dehydrogenase [KO:K00014] [EC:1.1.1.25]\n",
      "line in gene PA0025  aroE; shikimate dehydrogenase [KO:K00014] [EC:1.1.1.25]\n",
      "            PA0244  shikimate 5-dehydrogenase [KO:K00014] [EC:1.1.1.25]\n",
      "line in gene PA0244  shikimate 5-dehydrogenase [KO:K00014] [EC:1.1.1.25]\n",
      "            PA5039  aroK; shikimate kinase [KO:K00891] [EC:2.7.1.71]\n",
      "line in gene PA5039  aroK; shikimate kinase [KO:K00891] [EC:2.7.1.71]\n",
      "            PA1681  aroC; chorismate synthase [KO:K01736] [EC:4.2.3.5]\n",
      "line in gene PA1681  aroC; chorismate synthase [KO:K01736] [EC:4.2.3.5]\n",
      "            PA0609  trpE; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "line in gene PA0609  trpE; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "            PA1001  phnA; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "line in gene PA1001  phnA; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "            PA0649  trpG; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "line in gene PA0649  trpG; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "            PA1002  phnB; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "line in gene PA1002  phnB; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "            PA0650  trpD; anthranilate phosphoribosyltransferase [KO:K00766] [EC:2.4.2.18]\n",
      "line in gene PA0650  trpD; anthranilate phosphoribosyltransferase [KO:K00766] [EC:2.4.2.18]\n",
      "            PA3113  trpF; N-(5'-phosphoribosyl)anthranilate isomerase [KO:K01817] [EC:5.3.1.24]\n",
      "line in gene PA3113  trpF; N-(5'-phosphoribosyl)anthranilate isomerase [KO:K01817] [EC:5.3.1.24]\n",
      "            PA0651  trpC; indole-3-glycerol phosphate synthase [KO:K01609] [EC:4.1.1.48]\n",
      "line in gene PA0651  trpC; indole-3-glycerol phosphate synthase [KO:K01609] [EC:4.1.1.48]\n",
      "            PA0035  trpA; tryptophan synthase subunit alpha [KO:K01695] [EC:4.2.1.20]\n",
      "line in gene PA0035  trpA; tryptophan synthase subunit alpha [KO:K01695] [EC:4.2.1.20]\n",
      "            PA0036  trpB; tryptophan synthase subunit beta [KO:K01696] [EC:4.2.1.20]\n",
      "line in gene PA0036  trpB; tryptophan synthase subunit beta [KO:K01696] [EC:4.2.1.20]\n",
      "            PA5184  chorismate mutase [KO:K04093] [EC:5.4.99.5]\n",
      "line in gene PA5184  chorismate mutase [KO:K04093] [EC:5.4.99.5]\n",
      "            PA3166  pheA; bifunctional chorismate mutase/prephenate dehydratase [KO:K14170] [EC:5.4.99.5 4.2.1.51]\n",
      "line in gene PA3166  pheA; bifunctional chorismate mutase/prephenate dehydratase [KO:K14170] [EC:5.4.99.5 4.2.1.51]\n",
      "            PA3475  pheC; cyclohexadienyl dehydratase [KO:K01713] [EC:4.2.1.51 4.2.1.91]\n",
      "line in gene PA3475  pheC; cyclohexadienyl dehydratase [KO:K01713] [EC:4.2.1.51 4.2.1.91]\n",
      "            PA2531  aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA2531  aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA4447  hisC1; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA4447  hisC1; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA3165  hisC2; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA3165  hisC2; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA3139  aspartate aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "line in gene PA3139  aspartate aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "            PA0870  phhC; aromatic amino acid aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "line in gene PA0870  phhC; aromatic amino acid aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "            PA0872  phhA; phenylalanine 4-monooxygenase [KO:K00500] [EC:1.14.16.1]\n",
      "line in gene PA0872  phhA; phenylalanine 4-monooxygenase [KO:K00500] [EC:1.14.16.1]\n",
      "COMPOUND    C00074  Phosphoenolpyruvate\n",
      "            C00078  L-Tryptophan\n",
      "            C00079  L-Phenylalanine\n",
      "            C00082  L-Tyrosine\n",
      "            C00108  Anthranilate\n",
      "            C00119  5-Phospho-alpha-D-ribose 1-diphosphate\n",
      "            C00166  Phenylpyruvate\n",
      "            C00251  Chorismate\n",
      "            C00254  Prephenate\n",
      "            C00279  D-Erythrose 4-phosphate\n",
      "            C00296  Quinate\n",
      "            C00354  D-Fructose 1,6-bisphosphate\n",
      "            C00441  L-Aspartate 4-semialdehyde\n",
      "            C00463  Indole\n",
      "            C00493  Shikimate\n",
      "            C00587  3-Hydroxybenzoate\n",
      "            C00826  L-Arogenate\n",
      "            C00944  3-Dehydroquinate\n",
      "            C01094  D-Fructose 1-phosphate\n",
      "            C01179  3-(4-Hydroxyphenyl)pyruvate\n",
      "            C01269  5-O-(1-Carboxyvinyl)-3-phosphoshikimate\n",
      "            C01302  1-(2-Carboxyphenylamino)-1-deoxy-D-ribulose 5-phosphate\n",
      "            C02637  3-Dehydroshikimate\n",
      "            C03175  Shikimate 3-phosphate\n",
      "            C03506  Indoleglycerol phosphate\n",
      "            C04302  N-(5-Phospho-D-ribosyl)anthranilate\n",
      "            C04691  2-Dehydro-3-deoxy-D-arabino-heptonate 7-phosphate\n",
      "            C16848  6-Deoxy-5-ketofructose 1-phosphate\n",
      "            C16850  2-Amino-3,7-dideoxy-D-threo-hept-6-ulosonic acid\n",
      "            C17235  L-Homophenylalanine\n",
      "            C20327  2-Oxo-4-phenylbutyric acid\n",
      "            C20653  2-Benzylmalic acid\n",
      "            C20654  3-Benzylmalic acid\n",
      "            C20710  (4R,5R)-4,5-Dihydroxycyclohexa-1(6),2-diene-1-carboxylate\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00030  Pentose phosphate pathway\n",
      "            pae00130  Ubiquinone and other terpenoid-quinone biosynthesis\n",
      "            pae00350  Tyrosine metabolism\n",
      "            pae00360  Phenylalanine metabolism\n",
      "            pae00380  Tryptophan metabolism\n",
      "            pae00790  Folate biosynthesis\n",
      "            pae01053  Biosynthesis of siderophore group nonribosomal peptides\n",
      "KO_PATHWAY  ko00400\n",
      "///\n",
      "path:pae00401\n",
      "ENTRY       pae00401                    Pathway\n",
      "NAME        Novobiocin biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Biosynthesis of other secondary metabolites\n",
      "PATHWAY_MAP pae00401  Novobiocin biosynthesis\n",
      "DBLINKS     GO: 0043642\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2531  aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA2531  aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA4447  hisC1; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA4447  hisC1; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA3165  hisC2; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "line in gene PA3165  hisC2; histidinol-phosphate aminotransferase [KO:K00817] [EC:2.6.1.9]\n",
      "            PA3139  aspartate aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "line in gene PA3139  aspartate aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "            PA0870  phhC; aromatic amino acid aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "line in gene PA0870  phhC; aromatic amino acid aminotransferase [KO:K00832] [EC:2.6.1.57]\n",
      "COMPOUND    C00082  L-Tyrosine\n",
      "            C00148  L-Proline\n",
      "            C00254  Prephenate\n",
      "            C01179  3-(4-Hydroxyphenyl)pyruvate\n",
      "            C05073  Coumermycin A1\n",
      "            C05080  Novobiocin\n",
      "            C12032  Clorobiocin\n",
      "            C12456  3-Dimethylallyl-4-hydroxyphenylpyruvate\n",
      "            C12457  3-Dimethylallyl-4-hydroxymandelic acid\n",
      "            C12458  3-Dimethylallyl-4-hydroxybenzoate\n",
      "            C12459  3-Dimethylallyl-4-hydroxybenzaldehyde\n",
      "            C12460  3-Dimethylallyl-beta-hydroxy-L-tyrosyl-[pcp]\n",
      "            C12461  L-Tyrosyl-[pcp]\n",
      "            C12462  beta-Hydroxy-L-tyrosyl-[pcp]\n",
      "            C12463  beta-Keto-L-tyrosyl-[pcp]\n",
      "            C12467  2-Hydroxy-beta-keto-L-tyrosyl-[pcp]\n",
      "            C12468  3-Amino-4,7-dihydroxycoumarin\n",
      "            C12469  3-Amino-4,7-dihydroxy-8-chlorocoumarin\n",
      "            C12470  L-Prolyl-[pcp]\n",
      "            C12471  Chlorobiocic acid\n",
      "            C12472  Novclobiocin 105\n",
      "            C12473  Novclobiocin 104\n",
      "            C12474  Novobiocic acid\n",
      "            C12475  Demethyldecarbamoylnovobiocin\n",
      "            C12476  Decarbamoylnovobiocin\n",
      "            C12477  3-Methylpyrrole-2,4-dicarboxylic acid\n",
      "            C12479  Coumermic acid\n",
      "            C12480  Coumermycin D\n",
      "            C12481  dTDP-5-dimethyl-L-lyxose\n",
      "            C12483  Pyrrole-2-carbonyl-[pcp]\n",
      "            C20629  8-Demethylnovobiocic acid\n",
      "            C20637  5-[[(4,7-Dihydroxy-2-oxo-2H-1-benzopyran-3-yl)amino]carbonyl]-4-methyl-1H-pyrrole-3-carboxylate\n",
      "REFERENCE   PMID:12480894\n",
      "  AUTHORS   Pojer F, Li SM, Heide L.\n",
      "  TITLE     Molecular cloning and sequence analysis of the clorobiocin biosynthetic gene cluster: new insights into the biosynthesis of aminocoumarin antibiotics.\n",
      "  JOURNAL   Microbiology 148:3901-11 (2002)\n",
      "            DOI:10.1099/00221287-148-12-3901\n",
      "REFERENCE   PMID:11325587\n",
      "  AUTHORS   Chen H, Walsh CT.\n",
      "  TITLE     Coumarin formation in novobiocin biosynthesis: beta-hydroxylation of the aminoacyl enzyme tyrosyl-S-NovH by a cytochrome P450 NovI.\n",
      "  JOURNAL   Chem Biol 8:301-12 (2001)\n",
      "            DOI:10.1016/S1074-5521(01)00009-6\n",
      "REFERENCE   PMID:14622269\n",
      "  AUTHORS   Schmutz E, Steffensky M, Schmidt J, Porzel A, Li SM, Heide L.\n",
      "  TITLE     An unusual amide synthetase (CouL) from the coumermycin A1 biosynthetic gene cluster from Streptomyces rishiriensis DSM 40489.\n",
      "  JOURNAL   Eur J Biochem 270:4413-9 (2003)\n",
      "            DOI:10.1046/j.1432-1033.2003.03830.x\n",
      "REFERENCE   PMID:12904558\n",
      "  AUTHORS   Xu H, Kahlich R, Kammerer B, Heide L, Li SM.\n",
      "  TITLE     CloN2, a novel acyltransferase involved in the attachment of the pyrrole-2-carboxyl moiety to the deoxysugar of clorobiocin.\n",
      "  JOURNAL   Microbiology 149:2183-91 (2003)\n",
      "            DOI:10.1099/mic.0.26314-0\n",
      "REFERENCE   PMID:12436260\n",
      "  AUTHORS   Xu H, Wang ZX, Schmidt J, Heide L, Li SM.\n",
      "  TITLE     Genetic analysis of the biosynthesis of the pyrrole and carbamoyl moieties of coumermycin A1 and novobiocin.\n",
      "  JOURNAL   Mol Genet Genomics 268:387-96 (2002)\n",
      "            DOI:10.1007/s00438-002-0759-1\n",
      "REFERENCE   PMID:14694473\n",
      "  AUTHORS   Freel Meyers CL, Oberthur M, Xu H, Heide L, Kahne D, Walsh CT\n",
      "  TITLE     Characterization of NovP and NovN: completion of novobiocin biosynthesis by sequential tailoring of the noviosyl ring.\n",
      "  JOURNAL   Angew Chem Int Ed Engl 43:67-70 (2004)\n",
      "            DOI:10.1002/anie.200352626\n",
      "REFERENCE   PMID:12777382\n",
      "  AUTHORS   Pojer F, Kahlich R, Kammerer B, Li SM, Heide L.\n",
      "  TITLE     CloR, a bifunctional non-heme iron oxygenase involved in clorobiocin biosynthesis.\n",
      "  JOURNAL   J Biol Chem 278:30661-8 (2003)\n",
      "            DOI:10.1074/jbc.M303190200\n",
      "REFERENCE   PMID:12368465\n",
      "  AUTHORS   Li SM, Westrich L, Schmidt J, Kuhnt C, Heide L.\n",
      "  TITLE     Methyltransferase genes in Streptomyces rishiriensis: new coumermycin derivatives from gene-inactivation experiments.\n",
      "  JOURNAL   Microbiology 148:3317-26 (2002)\n",
      "            DOI:10.1099/00221287-148-10-3317\n",
      "REFERENCE   PMID:12618544\n",
      "  AUTHORS   Pojer F, Wemakor E, Kammerer B, Chen H, Walsh CT, Li SM, Heide L.\n",
      "  TITLE     CloQ, a prenyltransferase involved in clorobiocin biosynthesis.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 100:2316-21 (2003)\n",
      "            DOI:10.1073/pnas.0337708100\n",
      "REL_PATHWAY pae00330  Arginine and proline metabolism\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "            pae00523  Polyketide sugar unit biosynthesis\n",
      "KO_PATHWAY  ko00401\n",
      "///\n",
      "path:pae00405\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00405                    Pathway\n",
      "NAME        Phenazine biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Biosynthesis of other secondary metabolites\n",
      "PATHWAY_MAP pae00405  Phenazine biosynthesis\n",
      "MODULE      pae_M00835  Pyocyanine biosynthesis, chorismate => pyocyanine [PATH:pae00405]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1903  phzE2; phenazine biosynthesis protein PhzE [KO:K13063] [EC:2.6.1.86]\n",
      "line in gene PA1903  phzE2; phenazine biosynthesis protein PhzE [KO:K13063] [EC:2.6.1.86]\n",
      "            PA4214  phzE1; phenazine biosynthesis protein PhzE [KO:K13063] [EC:2.6.1.86]\n",
      "line in gene PA4214  phzE1; phenazine biosynthesis protein PhzE [KO:K13063] [EC:2.6.1.86]\n",
      "            PA1902  phzD2; phenazine biosynthesis protein PhzD [KO:K20261] [EC:3.3.2.15]\n",
      "line in gene PA1902  phzD2; phenazine biosynthesis protein PhzD [KO:K20261] [EC:3.3.2.15]\n",
      "            PA4213  phzD1; phenazine biosynthesis protein PhzD [KO:K20261] [EC:3.3.2.15]\n",
      "line in gene PA4213  phzD1; phenazine biosynthesis protein PhzD [KO:K20261] [EC:3.3.2.15]\n",
      "            PA1904  phzF2; trans-2,3-dihydro-3-hydroxyanthranilate isomerase [KO:K06998] [EC:5.3.3.17]\n",
      "line in gene PA1904  phzF2; trans-2,3-dihydro-3-hydroxyanthranilate isomerase [KO:K06998] [EC:5.3.3.17]\n",
      "            PA4215  phzF1; trans-2,3-dihydro-3-hydroxyanthranilate isomerase [KO:K06998] [EC:5.3.3.17]\n",
      "line in gene PA4215  phzF1; trans-2,3-dihydro-3-hydroxyanthranilate isomerase [KO:K06998] [EC:5.3.3.17]\n",
      "            PA4210  phzA1; phenazine biosynthesis protein [KO:K20260]\n",
      "line in gene PA4210  phzA1; phenazine biosynthesis protein [KO:K20260]\n",
      "            PA1899  phzA2; phenazine biosynthesis protein PhzA [KO:K20260]\n",
      "line in gene PA1899  phzA2; phenazine biosynthesis protein PhzA [KO:K20260]\n",
      "            PA4211  phzB1; phenazine biosynthesis protein [KO:K20260]\n",
      "line in gene PA4211  phzB1; phenazine biosynthesis protein [KO:K20260]\n",
      "            PA1900  phzB2; phenazine biosynthesis protein PhzB [KO:K20260]\n",
      "line in gene PA1900  phzB2; phenazine biosynthesis protein PhzB [KO:K20260]\n",
      "            PA1905  phzG2; pyridoxamine 5'-phosphate oxidase [KO:K20262] [EC:1.10.3.16]\n",
      "line in gene PA1905  phzG2; pyridoxamine 5'-phosphate oxidase [KO:K20262] [EC:1.10.3.16]\n",
      "            PA4216  phzG1; pyridoxamine 5'-phosphate oxidase [KO:K20262] [EC:1.10.3.16]\n",
      "line in gene PA4216  phzG1; pyridoxamine 5'-phosphate oxidase [KO:K20262] [EC:1.10.3.16]\n",
      "            PA4209  phzM; phenazine-specific methyltransferase [KO:K21103] [EC:2.1.1.327]\n",
      "line in gene PA4209  phzM; phenazine-specific methyltransferase [KO:K21103] [EC:2.1.1.327]\n",
      "            PA4217  phzS; hypothetical protein [KO:K20940] [EC:1.14.13.218]\n",
      "line in gene PA4217  phzS; hypothetical protein [KO:K20940] [EC:1.14.13.218]\n",
      "            PA0609  trpE; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "line in gene PA0609  trpE; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "            PA1001  phnA; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "line in gene PA1001  phnA; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "            PA0649  trpG; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "line in gene PA0649  trpG; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "            PA1002  phnB; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "line in gene PA1002  phnB; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "            PA0996  pqsA; anthranilate--CoA ligase [KO:K18000] [EC:6.2.1.32]\n",
      "line in gene PA0996  pqsA; anthranilate--CoA ligase [KO:K18000] [EC:6.2.1.32]\n",
      "            PA0999  pqsD; 3-oxoacyl-ACP synthase [KO:K18003] [EC:2.3.1.262]\n",
      "line in gene PA0999  pqsD; 3-oxoacyl-ACP synthase [KO:K18003] [EC:2.3.1.262]\n",
      "            PA1000  pqsE; thioesterase PqsE [KO:K20257] [EC:3.1.2.32]\n",
      "line in gene PA1000  pqsE; thioesterase PqsE [KO:K20257] [EC:3.1.2.32]\n",
      "            PA0997  pqsB; hypothetical protein [KO:K18001] [EC:2.3.1.230]\n",
      "line in gene PA0997  pqsB; hypothetical protein [KO:K18001] [EC:2.3.1.230]\n",
      "            PA0998  pqsC; hypothetical protein [KO:K18002] [EC:2.3.1.230]\n",
      "line in gene PA0998  pqsC; hypothetical protein [KO:K18002] [EC:2.3.1.230]\n",
      "            PA2587  pqsH; 2-heptyl-3-hydroxy-4(1H)-quinolone synthase [KO:K17940] [EC:1.14.13.182]\n",
      "line in gene PA2587  pqsH; 2-heptyl-3-hydroxy-4(1H)-quinolone synthase [KO:K17940] [EC:1.14.13.182]\n",
      "COMPOUND    C00108  Anthranilate\n",
      "            C00251  Chorismate\n",
      "            C01748  Pyocyanine\n",
      "            C01944  Octanoyl-CoA\n",
      "            C02247  Anthraniloyl-CoA\n",
      "            C11848  2-Heptyl-3-hydroxy-quinolone\n",
      "            C12119  Phenazine-1,6-dicarboxylic acid\n",
      "            C18054  2-Amino-2-deoxyisochorismate\n",
      "            C19830  (5S,6S)-6-Amino-5-hydroxycyclohexa-1,3-diene-1-carboxylate\n",
      "            C19831  (1R,6S)-6-Amino-5-oxocyclohex-2-ene-1-carboxylate\n",
      "            C20643  2-Heptyl-4(1H)-quinolone\n",
      "            C20981  5,10-Dihydrophenazine-1-carboxylate\n",
      "            C20982  9-(Dimethylallyl)-5,10-dihydrophenazine-1-carboxylate\n",
      "            C21295  5-Methylphenazine-1-carboxylate\n",
      "            C21408  (1R,6R)-1,4,5,5a,6,9-Hexahydrophenazine-1,6-dicarboxylate\n",
      "            C21409  (1R,10aS)-1,4,10,10a-Tetrahydrophenazine-1,6-dicarboxylate\n",
      "            C21410  (5aS)-5,5a-Dihydrophenazine-1,6-dicarboxylate\n",
      "            C21411  (1R,10aS)-1,4,10,10a-Tetrahydrophenazine-1-carboxylate\n",
      "            C21412  (10aS)-10,10a-Dihydrophenazine-1-carboxylate\n",
      "            C21413  (1R)-1,4,5,10-Tetrahydrophenazine-1-carboxylate\n",
      "            C21442  Phenazine-1-carboxylate\n",
      "            C21452  (2-Aminobenzoyl)acetyl-CoA\n",
      "            C21453  (2-Aminobenzoyl)acetate\n",
      "            C21473  (1R,5aS,6R)-4a-Hydroxy-1,4,4a,5,5a,6,9,10a-octahydrophenazine-1,6-dicarboxylate\n",
      "            C21474  5,10-Dihydrophenazine-1,6-dicarboxylate\n",
      "            C21475  5,10-Dihydrophenazine\n",
      "            C21476  Phenazine\n",
      "            C21477  1-Hydroxyphenazine\n",
      "            C21488  S-Anthraniloyl-L-cysteinyl-protein\n",
      "            C21489  S-Octanoyl-L-cysteinyl-protein\n",
      "            C21490  1-(2-Aminophenyl)decane-1,3-dione\n",
      "REFERENCE   PMID:23897464\n",
      "  AUTHORS   Xu N, Ahuja EG, Janning P, Mavrodi DV, Thomashow LS, Blankenfeldt W\n",
      "  TITLE     Trapped intermediates in crystals of the FMN-dependent oxidase PhzG provide insight into the final steps of phenazine biosynthesis.\n",
      "  JOURNAL   Acta Crystallogr D Biol Crystallogr 69:1403-13 (2013)\n",
      "            DOI:10.1107/S0907444913008354\n",
      "REFERENCE   PMID:20352425\n",
      "  AUTHORS   Pierson LS 3rd, Pierson EA\n",
      "  TITLE     Metabolism and function of phenazines in bacteria: impacts on the behavior of bacteria in the environment and biotechnological processes.\n",
      "  JOURNAL   Appl Microbiol Biotechnol 86:1659-70 (2010)\n",
      "            DOI:10.1007/s00253-010-2509-3\n",
      "REFERENCE   PMID:19053436\n",
      "  AUTHORS   Ahuja EG, Janning P, Mentel M, Graebsch A, Breinbauer R, Hiller W, Costisella B, Thomashow LS, Mavrodi DV, Blankenfeldt W\n",
      "  TITLE     PhzA/B catalyzes the formation of the tricycle in phenazine biosynthesis.\n",
      "  JOURNAL   J Am Chem Soc 130:17053-61 (2008)\n",
      "            DOI:10.1021/ja806325k\n",
      "REFERENCE   PMID:19658148\n",
      "  AUTHORS   Mentel M, Ahuja EG, Mavrodi DV, Breinbauer R, Thomashow LS, Blankenfeldt W\n",
      "  TITLE     Of two make one: the biosynthesis of phenazines.\n",
      "  JOURNAL   Chembiochem 10:2295-304 (2009)\n",
      "            DOI:10.1002/cbic.200900323\n",
      "REFERENCE   PMID:15502343\n",
      "  AUTHORS   Parsons JF, Calabrese K, Eisenstein E, Ladner JE\n",
      "  TITLE     Structure of the phenazine biosynthesis enzyme PhzG.\n",
      "  JOURNAL   Acta Crystallogr D Biol Crystallogr 60:2110-3 (2004)\n",
      "            DOI:10.1107/S0907444904022474\n",
      "REFERENCE   PMID:11591691\n",
      "  AUTHORS   Mavrodi DV, Bonsall RF, Delaney SM, Soule MJ, Phillips G, Thomashow LS\n",
      "  TITLE     Functional analysis of genes for biosynthesis of pyocyanin and phenazine-1-carboxamide from Pseudomonas aeruginosa PAO1.\n",
      "  JOURNAL   J Bacteriol 183:6454-65 (2001)\n",
      "            DOI:10.1128/JB.183.21.6454-6465.2001\n",
      "REL_PATHWAY pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "KO_PATHWAY  ko00405\n",
      "///\n",
      "path:pae00410\n",
      "ENTRY       pae00410                    Pathway\n",
      "NAME        beta-Alanine metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of other amino acids\n",
      "PATHWAY_MAP pae00410  beta-Alanine metabolism\n",
      "DBLINKS     GO: 0019482\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4731  panD; aspartate alpha-decarboxylase [KO:K01579] [EC:4.1.1.11]\n",
      "line in gene PA4731  panD; aspartate alpha-decarboxylase [KO:K01579] [EC:4.1.1.11]\n",
      "            PA3713  spdH; spermidine dehydrogenase SpdH [KO:K00316] [EC:1.5.99.6]\n",
      "line in gene PA3713  spdH; spermidine dehydrogenase SpdH [KO:K00316] [EC:1.5.99.6]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4730  panC; pantoate--beta-alanine ligase [KO:K01918] [EC:6.3.2.1]\n",
      "line in gene PA4730  panC; pantoate--beta-alanine ligase [KO:K01918] [EC:6.3.2.1]\n",
      "            PA0440  oxidoreductase [KO:K17722] [EC:1.3.1.1]\n",
      "line in gene PA0440  oxidoreductase [KO:K17722] [EC:1.3.1.1]\n",
      "            PA0439  dihydropyrimidine dehydrogenase [KO:K17723] [EC:1.3.1.1]\n",
      "line in gene PA0439  dihydropyrimidine dehydrogenase [KO:K17723] [EC:1.3.1.1]\n",
      "            PA0441  dht; D-hydantoinase/dihydropyrimidinase [KO:K01464] [EC:3.5.2.2]\n",
      "line in gene PA0441  dht; D-hydantoinase/dihydropyrimidinase [KO:K01464] [EC:3.5.2.2]\n",
      "            PA0132  beta alanine--pyruvate transaminase [KO:K00822] [EC:2.6.1.18]\n",
      "line in gene PA0132  beta alanine--pyruvate transaminase [KO:K00822] [EC:2.6.1.18]\n",
      "            PA5313  omega amino acid--pyruvate transaminase [KO:K00822] [EC:2.6.1.18]\n",
      "line in gene PA5313  omega amino acid--pyruvate transaminase [KO:K00822] [EC:2.6.1.18]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "line in gene PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "            PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "line in gene PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "            PA2552  acyl-CoA dehydrogenase [KO:K00248] [EC:1.3.8.1]\n",
      "line in gene PA2552  acyl-CoA dehydrogenase [KO:K00248] [EC:1.3.8.1]\n",
      "            PA3570  mmsA; methylmalonate-semialdehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "line in gene PA3570  mmsA; methylmalonate-semialdehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "            PA0747  aldehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "line in gene PA0747  aldehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "            PA0130  3-oxopropanoate dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "line in gene PA0130  3-oxopropanoate dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "COMPOUND    C00024  Acetyl-CoA\n",
      "            C00049  L-Aspartate\n",
      "            C00083  Malonyl-CoA\n",
      "            C00099  beta-Alanine\n",
      "            C00100  Propanoyl-CoA\n",
      "            C00106  Uracil\n",
      "            C00135  L-Histidine\n",
      "            C00222  3-Oxopropanoate\n",
      "            C00315  Spermidine\n",
      "            C00334  4-Aminobutanoate\n",
      "            C00383  Malonate\n",
      "            C00386  Carnosine\n",
      "            C00429  5,6-Dihydrouracil\n",
      "            C00555  4-Aminobutyraldehyde\n",
      "            C00750  Spermine\n",
      "            C00804  Propynoate\n",
      "            C00864  Pantothenate\n",
      "            C00894  Propenoyl-CoA\n",
      "            C00986  1,3-Diaminopropane\n",
      "            C01013  3-Hydroxypropanoate\n",
      "            C01073  N-Acetyl-beta-alanine\n",
      "            C01262  beta-Alanyl-N(pi)-methyl-L-histidine\n",
      "            C02335  beta-Alanyl-CoA\n",
      "            C02642  3-Ureidopropionate\n",
      "            C03492  D-4'-Phosphopantothenate\n",
      "            C03722  Quinolinate\n",
      "            C05340  beta-Alanyl-L-arginine\n",
      "            C05341  beta-Alanyl-L-lysine\n",
      "            C05665  3-Aminopropanal\n",
      "            C05668  3-Hydroxypropionyl-CoA\n",
      "            C05669  beta-Nitropropanoate\n",
      "            C05670  3-Aminopropiononitrile\n",
      "REL_PATHWAY pae00061  Fatty acid biosynthesis\n",
      "            pae00240  Pyrimidine metabolism\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00330  Arginine and proline metabolism\n",
      "            pae00460  Cyanoamino acid metabolism\n",
      "            pae00640  Propanoate metabolism\n",
      "            pae00770  Pantothenate and CoA biosynthesis\n",
      "KO_PATHWAY  ko00410\n",
      "///\n",
      "path:pae00430\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00430                    Pathway\n",
      "NAME        Taurine and hypotaurine metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of other amino acids\n",
      "PATHWAY_MAP pae00430  Taurine and hypotaurine metabolism\n",
      "MODULE      pae_M00579  Phosphate acetyltransferase-acetate kinase pathway, acetyl-CoA => acetate [PATH:pae00430]\n",
      "DBLINKS     GO: 0019530\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0361  gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "line in gene PA0361  gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "            PA0164  gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "line in gene PA0164  gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "            PA1338  ggt; gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "line in gene PA1338  ggt; gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "            PA2310  hypothetical protein [KO:K03119] [EC:1.14.11.17]\n",
      "line in gene PA2310  hypothetical protein [KO:K03119] [EC:1.14.11.17]\n",
      "            PA3935  tauD; taurine dioxygenase [KO:K03119] [EC:1.14.11.17]\n",
      "line in gene PA3935  tauD; taurine dioxygenase [KO:K03119] [EC:1.14.11.17]\n",
      "            PA3068  gdhB; NAD-specific glutamate dehydrogenase [KO:K15371] [EC:1.4.1.2]\n",
      "line in gene PA3068  gdhB; NAD-specific glutamate dehydrogenase [KO:K15371] [EC:1.4.1.2]\n",
      "            PA0835  pta; phosphate acetyltransferase [KO:K13788] [EC:2.3.1.8]\n",
      "line in gene PA0835  pta; phosphate acetyltransferase [KO:K13788] [EC:2.3.1.8]\n",
      "            PA0836  ackA; acetate kinase [KO:K00925] [EC:2.7.2.1]\n",
      "line in gene PA0836  ackA; acetate kinase [KO:K00925] [EC:2.7.2.1]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00025  L-Glutamate\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00033  Acetate\n",
      "            C00041  L-Alanine\n",
      "            C00084  Acetaldehyde\n",
      "            C00094  Sulfite\n",
      "            C00097  L-Cysteine\n",
      "            C00227  Acetyl phosphate\n",
      "            C00245  Taurine\n",
      "            C00506  L-Cysteate\n",
      "            C00519  Hypotaurine\n",
      "            C00593  Sulfoacetaldehyde\n",
      "            C00606  3-Sulfino-L-alanine\n",
      "            C01678  Cysteamine\n",
      "            C01959  Taurocyamine\n",
      "            C03149  N-Phosphotaurocyamine\n",
      "            C05122  Taurocholate\n",
      "            C05123  2-Hydroxyethanesulfonate\n",
      "            C05844  5-L-Glutamyl-taurine\n",
      "            C06735  Aminoacetaldehyde\n",
      "            C14179  Sulfoacetate\n",
      "REFERENCE   PMID:12471498\n",
      "  AUTHORS   Cook AM, Denger K.\n",
      "  TITLE     Dissimilation of the C2 sulfonates.\n",
      "  JOURNAL   Arch Microbiol 179:1-6 (2002)\n",
      "            DOI:10.1007/s00203-002-0497-0\n",
      "REFERENCE   PMID:11082195\n",
      "  AUTHORS   Laue H, Cook AM.\n",
      "  TITLE     Biochemical and molecular characterization of taurine:pyruvate aminotransferase from the anaerobe Bilophila wadsworthia.\n",
      "  JOURNAL   Eur J Biochem 267:6841-8 (2000)\n",
      "            DOI:10.1046/j.1432-1033.2000.01782.x\n",
      "REFERENCE   PMID:12358600\n",
      "  AUTHORS   Ruff J, Denger K, Cook AM.\n",
      "  TITLE     Sulphoacetaldehyde acetyltransferase yields acetyl phosphate: purification from Alcaligenes defragrans and gene clusters in taurine degradation.\n",
      "  JOURNAL   Biochem J 369:275-85 (2003)\n",
      "            DOI:10.1042/BJ20021455\n",
      "REFERENCE   PMID:15073291\n",
      "  AUTHORS   Bruggemann C, Denger K, Cook AM, Ruff J.\n",
      "  TITLE     Enzymes and genes of taurine and isethionate dissimilation in Paracoccus denitrificans.\n",
      "  JOURNAL   Microbiology 150:805-16 (2004)\n",
      "            DOI:10.1099/mic.0.26795-0\n",
      "REFERENCE   PMID:15340795\n",
      "  AUTHORS   Denger K, Weinitschke S, Hollemeyer K, Cook AM.\n",
      "  TITLE     Sulfoacetate generated by Rhodopseudomonas palustris from taurine.\n",
      "  JOURNAL   Arch Microbiol 182:254-8 (2004)\n",
      "            DOI:10.1007/s00203-004-0678-0\n",
      "REFERENCE   PMID:15184572\n",
      "  AUTHORS   Denger K, Ruff J, Schleheck D, Cook AM.\n",
      "  TITLE     Rhodococcus opacus expresses the xsc gene to utilize taurine as a carbon source or as a nitrogen source but not as a sulfur source.\n",
      "  JOURNAL   Microbiology 150:1859-67 (2004)\n",
      "            DOI:10.1099/mic.0.27077-0\n",
      "REFERENCE   PMID:15883781\n",
      "  AUTHORS   Styp von Rekowski K, Denger K, Cook AM.\n",
      "  TITLE     Isethionate as a product from taurine during nitrogen-limited growth of Klebsiella oxytoca TauN1.\n",
      "  JOURNAL   Arch Microbiol 183:325-30 (2005)\n",
      "            DOI:10.1007/s00203-005-0776-7\n",
      "REFERENCE   PMID:4297098\n",
      "  AUTHORS   Jacobsen JG, Smith LH\n",
      "  TITLE     Biochemistry and physiology of taurine and taurine derivatives.\n",
      "  JOURNAL   Physiol Rev 48:424-511 (1968)\n",
      "            DOI:10.1152/physrev.1968.48.2.424\n",
      "REL_PATHWAY pae00270  Cysteine and methionine metabolism\n",
      "            pae00460  Cyanoamino acid metabolism\n",
      "            pae00480  Glutathione metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00920  Sulfur metabolism\n",
      "KO_PATHWAY  ko00430\n",
      "///\n",
      "path:pae00440\n",
      "ENTRY       pae00440                    Pathway\n",
      "NAME        Phosphonate and phosphinate metabolism - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Natural products containing carbon-phosphorous bonds, so-called C-P compounds, are derivatives of phosphonate and phosphinate with substitution of alkyl group for hydrogen of phosphorus-hydrogen bonds. C-P compounds have been found in many organisms, but only protists and bacteria, mostly Actinobacteria, have biosynthetic capacity. A common reaction in the biosynthetic pathway is C-P bond forming reaction from phosphoenolpyruvate (PEP) to phosphonopyruvate (PnPy) catalyzed by PEP phosphomutase. 2-Aminoethylphosphonate (AEP) is the most abundant C-P compound in the natural world. AEP derivatives include phosphonoprotein, phosphonoglycan, and phosphonolipid. Other known C-P compounds are bioactive substances used in medicine (antibiotics) and agriculture (herbicide) such as fosfomycin, FR-33289, rhizocticin, and bialaphos.\n",
      "CLASS       Metabolism; Metabolism of other amino acids\n",
      "PATHWAY_MAP pae00440  Phosphonate and phosphinate metabolism\n",
      "DBLINKS     GO: 0019634\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1311  phnX; phosphonoacetaldehyde hydrolase [KO:K05306] [EC:3.11.1.1]\n",
      "line in gene PA1311  phnX; phosphonoacetaldehyde hydrolase [KO:K05306] [EC:3.11.1.1]\n",
      "            PA1310  phnW; 2-aminoethylphosphonate--pyruvate transaminase [KO:K03430] [EC:2.6.1.37]\n",
      "line in gene PA1310  phnW; 2-aminoethylphosphonate--pyruvate transaminase [KO:K03430] [EC:2.6.1.37]\n",
      "            PA3378  hypothetical protein [KO:K06164] [EC:2.7.8.37]\n",
      "line in gene PA3378  hypothetical protein [KO:K06164] [EC:2.7.8.37]\n",
      "            PA3380  hypothetical protein [KO:K06166] [EC:2.7.8.37]\n",
      "line in gene PA3380  hypothetical protein [KO:K06166] [EC:2.7.8.37]\n",
      "            PA3379  carbon-phosphorus lyase complex subunit [KO:K06165] [EC:2.7.8.37]\n",
      "line in gene PA3379  carbon-phosphorus lyase complex subunit [KO:K06165] [EC:2.7.8.37]\n",
      "            PA3375  ABC transporter ATP-binding protein [KO:K05780] [EC:2.7.8.37]\n",
      "line in gene PA3375  ABC transporter ATP-binding protein [KO:K05780] [EC:2.7.8.37]\n",
      "            PA3374  hypothetical protein [KO:K06162] [EC:3.6.1.63]\n",
      "line in gene PA3374  hypothetical protein [KO:K06162] [EC:3.6.1.63]\n",
      "            PA3377  hypothetical protein [KO:K06163] [EC:4.7.1.1]\n",
      "line in gene PA3377  hypothetical protein [KO:K06163] [EC:4.7.1.1]\n",
      "            PA3372  carbon-phosphorus lyase complex accessory protein [KO:K06167] [EC:3.1.4.55]\n",
      "line in gene PA3372  carbon-phosphorus lyase complex accessory protein [KO:K06167] [EC:3.1.4.55]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00033  Acetate\n",
      "            C00037  Glycine\n",
      "            C00058  Formate\n",
      "            C00074  Phosphoenolpyruvate\n",
      "            C00084  Acetaldehyde\n",
      "            C00117  D-Ribose 5-phosphate\n",
      "            C01151  D-Ribose 1,5-bisphosphate\n",
      "            C01438  Methane\n",
      "            C02798  3-Phosphonopyruvate\n",
      "            C03167  Phosphonoacetaldehyde\n",
      "            C03557  2-Aminoethylphosphonate\n",
      "            C04650  L-Phosphinothricin\n",
      "            C05672  2-Amino-3-phosphonopropanoate\n",
      "            C05673  CMP-2-aminoethylphosphonate\n",
      "            C05674  CMP-N-trimethyl-2-aminoethylphosphonate\n",
      "            C05675  Diacylglyceryl-2-aminoethylphosphonate\n",
      "            C05676  Diacylglyceryl-N-trimethyl-2-aminoethylphosphonate\n",
      "            C05677  Lipophosphonoglycan\n",
      "            C05678  (2-Amino-1-hydroxyethyl)phosphonate\n",
      "            C05679  N-Monomethyl-2-aminoethylphosphonate\n",
      "            C05680  N-Dimethyl-2-aminoethylphosphonate\n",
      "            C05681  Ceramide 2-aminoethylphosphonate\n",
      "            C05682  Phosphonoacetate\n",
      "            C05683  Ciliatocholate\n",
      "            C06367  1-Carboxyvinyl carboxyphosphonate\n",
      "            C06368  3-(Hydrohydroxyphosphoryl)pyruvate\n",
      "            C06451  2-Hydroxyethylphosphonate\n",
      "            C06455  Hydroxymethylphosphonate\n",
      "            C06456  Phosphonoformate\n",
      "            C06457  Bialaphos\n",
      "            C06459  N-Trimethyl-2-aminoethylphosphonate\n",
      "            C17940  Rhizocticin B\n",
      "            C17941  2-Oxo-4-phosphonobutanoate\n",
      "            C17942  FR 900098\n",
      "            C17943  Formylphosphonate\n",
      "            C17944  Rhizocticin A\n",
      "            C17945  Phosphonoformyl-CMP\n",
      "            C17947  2-Phosphinomethylmalate\n",
      "            C17948  Deamino-alpha-keto-demethylphosphinothricin\n",
      "            C17949  N-Acetyldemethylphosphinothricin\n",
      "            C17950  N-Acetyldemethylphosphinothricin tripeptide\n",
      "            C17951  N-Acetylbialaphos\n",
      "            C17952  N-Acetylphosphinothricin\n",
      "            C17960  Rhizocticin C\n",
      "            C17961  Rhizocticin D\n",
      "            C17962  Demethylphosphinothricin\n",
      "            C20396  Methylphosphonate\n",
      "            C20422  alpha-D-Ribose 1-methylphosphonate 5-triphosphate\n",
      "            C20423  alpha-D-Ribose 1-methylphosphonate 5-phosphate\n",
      "            C20440  alpha-D-Ribose 1,2-cyclic phosphate 5-phosphate\n",
      "            C20986  D-Ribose 2,5-bisphosphate\n",
      "            C21372  3-Phosphinomethylmalate\n",
      "            C21403  2-Acetamidoethylphosphonate\n",
      "            C21404  N-Ethylacetamide\n",
      "            C21613  (R)-2-(Phosphonomethyl)malate\n",
      "REFERENCE   PMID:19489722\n",
      "  AUTHORS   Metcalf WW, van der Donk WA\n",
      "  TITLE     Biosynthesis of phosphonic and phosphinic acid natural products.\n",
      "  JOURNAL   Annu Rev Biochem 78:65-94 (2009)\n",
      "            DOI:10.1146/annurev.biochem.78.091707.100215\n",
      "REFERENCE   PMID:19053721\n",
      "  AUTHORS   Xiao Y, Lee K, Liu P\n",
      "  TITLE     Syntheses of the P-methylase substrates of the bialaphos biosynthetic pathway.\n",
      "  JOURNAL   Org Lett 10:5521-4 (2008)\n",
      "            DOI:10.1021/ol802269x\n",
      "REFERENCE   PMID:17168571\n",
      "  AUTHORS   van der Donk WA\n",
      "  TITLE     Rings, radicals, and regeneration: the early years of a bioorganic laboratory.\n",
      "  JOURNAL   J Org Chem 71:9561-71 (2006)\n",
      "            DOI:10.1021/jo0614240\n",
      "REFERENCE   PMID:15616300\n",
      "  AUTHORS   Blodgett JA, Zhang JK, Metcalf WW\n",
      "  TITLE     Molecular cloning, sequence analysis, and heterologous expression of the phosphinothricin tripeptide biosynthetic gene cluster from Streptomyces viridochromogenes DSM 40736.\n",
      "  JOURNAL   Antimicrob Agents Chemother 49:230-40 (2005)\n",
      "            DOI:10.1128/AAC.49.1.230-240.2005\n",
      "REFERENCE   PMID:15574905\n",
      "  AUTHORS   Schwartz D, Berger S, Heinzelmann E, Muschko K, Welzel K, Wohlleben W\n",
      "  TITLE     Biosynthetic gene cluster of the herbicide phosphinothricin tripeptide from Streptomyces viridochromogenes Tu494.\n",
      "  JOURNAL   Appl Environ Microbiol 70:7093-102 (2004)\n",
      "            DOI:10.1128/AEM.70.12.7093-7102.2004\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00030  Pentose phosphate pathway\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "KO_PATHWAY  ko00440\n",
      "///\n",
      "path:pae00450\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00450                    Pathway\n",
      "NAME        Selenocompound metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of other amino acids\n",
      "PATHWAY_MAP pae00450  Selenocompound metabolism\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1843  metH; B12-dependent methionine synthase [KO:K00548] [EC:2.1.1.13]\n",
      "line in gene PA1843  metH; B12-dependent methionine synthase [KO:K00548] [EC:2.1.1.13]\n",
      "            PA1927  metE; 5-methyltetrahydropteroyltriglutamate--homocysteine methyltransferase [KO:K00549] [EC:2.1.1.14]\n",
      "line in gene PA1927  metE; 5-methyltetrahydropteroyltriglutamate--homocysteine methyltransferase [KO:K00549] [EC:2.1.1.14]\n",
      "            PA0400  cystathionine gamma-lyase [KO:K01758] [EC:4.4.1.1]\n",
      "line in gene PA0400  cystathionine gamma-lyase [KO:K01758] [EC:4.4.1.1]\n",
      "            PA2616  trxB1; thioredoxin reductase [KO:K00384] [EC:1.8.1.9]\n",
      "line in gene PA2616  trxB1; thioredoxin reductase [KO:K00384] [EC:1.8.1.9]\n",
      "            PA4442  cysN; bifunctional sulfate adenylyltransferase subunit 1/adenylylsulfate kinase [KO:K00955] [EC:2.7.7.4 2.7.1.25]\n",
      "line in gene PA4442  cysN; bifunctional sulfate adenylyltransferase subunit 1/adenylylsulfate kinase [KO:K00955] [EC:2.7.7.4 2.7.1.25]\n",
      "            PA4443  cysD; sulfate adenylyltransferase subunit 2 [KO:K00957] [EC:2.7.7.4]\n",
      "line in gene PA4443  cysD; sulfate adenylyltransferase subunit 2 [KO:K00957] [EC:2.7.7.4]\n",
      "            PA1642  selD; selenide, water dikinase [KO:K01008] [EC:2.7.9.3]\n",
      "line in gene PA1642  selD; selenide, water dikinase [KO:K01008] [EC:2.7.9.3]\n",
      "            PA4808  selA; selenocysteine synthase [KO:K01042] [EC:2.9.1.1]\n",
      "line in gene PA4808  selA; selenocysteine synthase [KO:K01042] [EC:2.9.1.1]\n",
      "            PA3482  metG; methionine--tRNA ligase [KO:K01874] [EC:6.1.1.10]\n",
      "line in gene PA3482  metG; methionine--tRNA ligase [KO:K01874] [EC:6.1.1.10]\n",
      "COMPOUND    C00017  Protein\n",
      "            C00041  L-Alanine\n",
      "            C01528  Hydrogen selenide\n",
      "            C02535  Dimethyl selenide\n",
      "            C05172  Selenophosphoric acid\n",
      "            C05335  L-Selenomethionine\n",
      "            C05336  Selenomethionyl-tRNA(Met)\n",
      "            C05684  Selenite\n",
      "            C05686  Adenylylselenate\n",
      "            C05688  L-Selenocysteine\n",
      "            C05689  Se-Methyl-L-selenocysteine\n",
      "            C05690  Se-Methylselenomethionine\n",
      "            C05695  gamma-Glutamyl-Se-methylselenocysteine\n",
      "            C05697  Selenate\n",
      "            C05698  Selenohomocysteine\n",
      "            C05699  L-Selenocystathionine\n",
      "            C05703  Methaneselenol\n",
      "            C06481  L-Seryl-tRNA(Sec)\n",
      "            C06482  L-Selenocysteinyl-tRNA(Sec)\n",
      "            C16638  O-Phosphoseryl-tRNA(Sec)\n",
      "            C18870  Selenodiglutathione\n",
      "            C18871  Glutathioselenol\n",
      "            C18872  Trimethylselenonium\n",
      "            C18893  1-Methylseleno-N-acetyl-D-galactosamine\n",
      "            C18902  Methylselenic acid\n",
      "            C18904  Methylselenopyruvate\n",
      "            C18905  Methylselenocysteine Se-oxide\n",
      "REFERENCE   PMID:21190829\n",
      "  AUTHORS   Hoefig CS, Renko K, Kohrle J, Birringer M, Schomburg L\n",
      "  TITLE     Comparison of different selenocompounds with respect to nutritional value vs. toxicity using liver cells in culture.\n",
      "  JOURNAL   J Nutr Biochem 22:945-55 (2011)\n",
      "            DOI:10.1016/j.jnutbio.2010.08.006\n",
      "REFERENCE   PMID:19051051\n",
      "  AUTHORS   McKenzie MJ, Hunter DA, Pathirana R, Watson LM, Joyce NI, Matich AJ, Rowan DD, Brummell DA\n",
      "  TITLE     Accumulation of an organic anticancer selenium compound in a transgenic Solanaceous species shows wider applicability of the selenocysteine methyltransferase transgene from selenium hyperaccumulators.\n",
      "  JOURNAL   Transgenic Res 18:407-24 (2009)\n",
      "            DOI:10.1007/s11248-008-9233-0\n",
      "REFERENCE   PMID:20383543\n",
      "  AUTHORS   Pinto JT, Lee JI, Sinha R, MacEwan ME, Cooper AJ\n",
      "  TITLE     Chemopreventive mechanisms of alpha-keto acid metabolites of naturally occurring organoselenium compounds.\n",
      "  JOURNAL   Amino Acids 41:29-41 (2011)\n",
      "            DOI:10.1007/s00726-010-0578-3\n",
      "REFERENCE   PMID:17451884\n",
      "  AUTHORS   Suzuki KT, Kurasaki K, Suzuki N\n",
      "  TITLE     Selenocysteine beta-lyase and methylselenol demethylase in the metabolism of Se-methylated selenocompounds into selenide.\n",
      "  JOURNAL   Biochim Biophys Acta 1770:1053-61 (2007)\n",
      "            DOI:10.1016/j.bbagen.2007.03.007\n",
      "REFERENCE   PMID:11782468\n",
      "  AUTHORS   Gromer S, Gross JH\n",
      "  TITLE     Methylseleninate is a substrate rather than an inhibitor of mammalian thioredoxin reductase. Implications for the antitumor effects of selenium.\n",
      "  JOURNAL   J Biol Chem 277:9701-6 (2002)\n",
      "            DOI:10.1074/jbc.M109234200\n",
      "KO_PATHWAY  ko00450\n",
      "///\n",
      "path:pae00460\n",
      "ENTRY       pae00460                    Pathway\n",
      "NAME        Cyanoamino acid metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of other amino acids\n",
      "PATHWAY_MAP pae00460  Cyanoamino acid metabolism\n",
      "DBLINKS     GO: 0033052 0019756\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1726  bglX; beta-glucosidase [KO:K05349] [EC:3.2.1.21]\n",
      "line in gene PA1726  bglX; beta-glucosidase [KO:K05349] [EC:3.2.1.21]\n",
      "            PA0361  gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "line in gene PA0361  gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "            PA0164  gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "line in gene PA0164  gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "            PA1338  ggt; gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "line in gene PA1338  ggt; gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "            PA2253  ansA; L-asparaginase I [KO:K01424] [EC:3.5.1.1]\n",
      "line in gene PA2253  ansA; L-asparaginase I [KO:K01424] [EC:3.5.1.1]\n",
      "            PA2444  glyA2; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA2444  glyA2; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA5415  glyA1; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA5415  glyA1; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA4602  glyA3; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA4602  glyA3; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA2193  hcnA; hydrogen cyanide synthase subunit HcnA [KO:K10814] [EC:1.4.99.5]\n",
      "line in gene PA2193  hcnA; hydrogen cyanide synthase subunit HcnA [KO:K10814] [EC:1.4.99.5]\n",
      "            PA2194  hcnB; hydrogen cyanide synthase subunit HcnB [KO:K10815] [EC:1.4.99.5]\n",
      "line in gene PA2194  hcnB; hydrogen cyanide synthase subunit HcnB [KO:K10815] [EC:1.4.99.5]\n",
      "            PA2195  hcnC; hydrogen cyanide synthase subunit HcnC [KO:K10816] [EC:1.4.99.5]\n",
      "line in gene PA2195  hcnC; hydrogen cyanide synthase subunit HcnC [KO:K10816] [EC:1.4.99.5]\n",
      "COMPOUND    C00014  Ammonia\n",
      "            C00037  Glycine\n",
      "            C00049  L-Aspartate\n",
      "            C00065  L-Serine\n",
      "            C00079  L-Phenylalanine\n",
      "            C00082  L-Tyrosine\n",
      "            C00097  L-Cysteine\n",
      "            C00152  L-Asparagine\n",
      "            C00183  L-Valine\n",
      "            C00302  Glutamate\n",
      "            C00407  L-Isoleucine\n",
      "            C00488  Formamide\n",
      "            C00561  Mandelonitrile\n",
      "            C00844  Prunasin\n",
      "            C01326  Hydrogen cyanide\n",
      "            C01401  Alanine\n",
      "            C01594  Linamarin\n",
      "            C02512  3-Cyano-L-alanine\n",
      "            C02659  Acetone cyanohydrin\n",
      "            C03004  N-Hydroxy-L-tyrosine\n",
      "            C03219  (E)-2-Methylpropanal oxime\n",
      "            C03742  (S)-4-Hydroxymandelonitrile\n",
      "            C04350  (E)-4-Hydroxyphenylacetaldehyde oxime\n",
      "            C05143  Dhurrin\n",
      "            C05670  3-Aminopropiononitrile\n",
      "            C05711  gamma-Glutamyl-beta-cyanoalanine\n",
      "            C05714  alpha-Aminopropiononitrile\n",
      "            C05715  gamma-Amino-gamma-cyanobutanoate\n",
      "            C06114  gamma-Glutamyl-beta-aminopropiononitrile\n",
      "            C08325  Amygdalin\n",
      "            C08334  Lotaustralin\n",
      "            C15503  N,N-Dihydroxy-L-tyrosine\n",
      "            C16074  Phenylacetonitrile\n",
      "            C16075  (Z)-Phenylacetaldehyde oxime\n",
      "            C18796  (2R)-2-Hydroxy-2-methylbutanenitrile\n",
      "            C19491  (E)-2-Methylbutanal oxime\n",
      "            C19712  N-Hydroxy-L-phenylalanine\n",
      "            C19714  (E)-Phenylacetaldoxime\n",
      "            C19715  N,N-Dihydroxy-L-phenylalanine\n",
      "            C20310  N-Hydroxy-L-isoleucine\n",
      "            C20311  N,N-Dihydroxy-L-isoleucine\n",
      "            C20312  (Z)-2-Methylbutanal oxime\n",
      "            C20313  N-Hydroxy-L-valine\n",
      "            C20314  N,N-Dihydroxy-L-valine\n",
      "            C20315  (Z)-2-Methylpropanal oxime\n",
      "REFERENCE   PMID:11575729\n",
      "  AUTHORS   Maruyama A, Saito K, Ishizawa K\n",
      "  TITLE     Beta-cyanoalanine synthase and cysteine synthase from potato: molecular cloning, biochemical characterization, and spatial and hormonal regulation.\n",
      "  JOURNAL   Plant Mol Biol 46:749-60 (2001)\n",
      "            DOI:10.1023/A:1011629703784\n",
      "REFERENCE   PMID:10636899\n",
      "  AUTHORS   Andersen MD, Busk PK, Svendsen I, Moller BL\n",
      "  TITLE     Cytochromes P-450 from cassava (Manihot esculenta Crantz) catalyzing the first steps in the biosynthesis of the cyanogenic glucosides linamarin and lotaustralin. Cloning, functional expression in Pichia pastoris, and substrate specificity of the isolated recombinant enzymes.\n",
      "  JOURNAL   J Biol Chem 275:1966-75 (2000)\n",
      "            DOI:10.1074/jbc.275.3.1966\n",
      "REFERENCE   \n",
      "  AUTHORS   Hickel A, Hasslacher M, Griengl H.\n",
      "  TITLE     Hydroxynitrile lyases: Functions and properties\n",
      "  JOURNAL   Physiol Plant 98:891-898 (1996)\n",
      "            DOI:10.1111/j.1399-3054.1996.tb06700.x\n",
      "REL_PATHWAY pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00290  Valine, leucine and isoleucine biosynthesis\n",
      "            pae00350  Tyrosine metabolism\n",
      "            pae00360  Phenylalanine metabolism\n",
      "            pae00410  beta-Alanine metabolism\n",
      "            pae00450  Selenocompound metabolism\n",
      "            pae00470  D-Amino acid metabolism\n",
      "            pae00480  Glutathione metabolism\n",
      "            pae00910  Nitrogen metabolism\n",
      "KO_PATHWAY  ko00460\n",
      "///\n",
      "path:pae00470\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00470                    Pathway\n",
      "NAME        D-Amino acid metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of other amino acids\n",
      "PATHWAY_MAP pae00470  D-Amino acid metabolism\n",
      "MODULE      pae_M00947  D-Arginine racemization, D-arginine => L-arginine [PATH:pae00470]\n",
      "DBLINKS     GO: 0046416\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA5302  dadX; alanine racemase [KO:K01775] [EC:5.1.1.1]\n",
      "line in gene PA5302  dadX; alanine racemase [KO:K01775] [EC:5.1.1.1]\n",
      "            PA4930  alr; biosynthetic alanine racemase [KO:K01775] [EC:5.1.1.1]\n",
      "line in gene PA4930  alr; biosynthetic alanine racemase [KO:K01775] [EC:5.1.1.1]\n",
      "            PA4201  ddlA; D-alanine--D-alanine ligase [KO:K01921] [EC:6.3.2.4]\n",
      "line in gene PA4201  ddlA; D-alanine--D-alanine ligase [KO:K01921] [EC:6.3.2.4]\n",
      "            PA4410  ddlB; D-alanine--D-alanine ligase [KO:K01921] [EC:6.3.2.4]\n",
      "line in gene PA4410  ddlB; D-alanine--D-alanine ligase [KO:K01921] [EC:6.3.2.4]\n",
      "            PA1638  glutaminase [KO:K01425] [EC:3.5.1.2]\n",
      "line in gene PA1638  glutaminase [KO:K01425] [EC:3.5.1.2]\n",
      "            PA1337  ansB; glutaminase-asparaginase [KO:K05597] [EC:3.5.1.38]\n",
      "line in gene PA1337  ansB; glutaminase-asparaginase [KO:K05597] [EC:3.5.1.38]\n",
      "            PA4662  murI; glutamate racemase [KO:K01776] [EC:5.1.1.3]\n",
      "line in gene PA4662  murI; glutamate racemase [KO:K01776] [EC:5.1.1.3]\n",
      "            PA4414  murD; UDP-N-acetylmuramoyl-L-alanyl-D-glutamate synthetase [KO:K01925] [EC:6.3.2.9]\n",
      "line in gene PA4414  murD; UDP-N-acetylmuramoyl-L-alanyl-D-glutamate synthetase [KO:K01925] [EC:6.3.2.9]\n",
      "            PA3357  dsdA; D-serine dehydratase [KO:K01753] [EC:4.3.1.18]\n",
      "line in gene PA3357  dsdA; D-serine dehydratase [KO:K01753] [EC:4.3.1.18]\n",
      "            PA5278  dapF; diaminopimelate epimerase [KO:K01778] [EC:5.1.1.7]\n",
      "line in gene PA5278  dapF; diaminopimelate epimerase [KO:K01778] [EC:5.1.1.7]\n",
      "            PA5277  lysA; diaminopimelate decarboxylase [KO:K01586] [EC:4.1.1.20]\n",
      "line in gene PA5277  lysA; diaminopimelate decarboxylase [KO:K01586] [EC:4.1.1.20]\n",
      "            PA3863  dauA; FAD-dependent catabolic D-arginine dehydrogenase DauA [KO:K19746] [EC:1.4.99.6]\n",
      "line in gene PA3863  dauA; FAD-dependent catabolic D-arginine dehydrogenase DauA [KO:K19746] [EC:1.4.99.6]\n",
      "            PA3862  dauB; NAD(P)H-dependent anabolic L-arginine dehydrogenase DauB [KO:K19744] [EC:1.4.1.25]\n",
      "line in gene PA3862  dauB; NAD(P)H-dependent anabolic L-arginine dehydrogenase DauB [KO:K19744] [EC:1.4.1.25]\n",
      "            PA1268  4-hydroxyproline 2-epimerase [KO:K12658] [EC:5.1.1.8]\n",
      "line in gene PA1268  4-hydroxyproline 2-epimerase [KO:K12658] [EC:5.1.1.8]\n",
      "            PA1266  oxidoreductase [KO:K22549] [EC:1.5.99.-]\n",
      "line in gene PA1266  oxidoreductase [KO:K22549] [EC:1.5.99.-]\n",
      "            PA1267  hypothetical protein [KO:K21061] [EC:1.5.99.-]\n",
      "line in gene PA1267  hypothetical protein [KO:K21061] [EC:1.5.99.-]\n",
      "            PA1254  dihydrodipicolinate synthetase [KO:K21062] [EC:3.5.4.22]\n",
      "line in gene PA1254  dihydrodipicolinate synthetase [KO:K21062] [EC:3.5.4.22]\n",
      "            PA1253  semialdehyde dehydrogenase [KO:K13877] [EC:1.2.1.26]\n",
      "line in gene PA1253  semialdehyde dehydrogenase [KO:K13877] [EC:1.2.1.26]\n",
      "            PA2217  aldehyde dehydrogenase [KO:K13877] [EC:1.2.1.26]\n",
      "line in gene PA2217  aldehyde dehydrogenase [KO:K13877] [EC:1.2.1.26]\n",
      "            PA4836  hypothetical protein [KO:K23120] [EC:2.5.1.152 2.5.1.-]\n",
      "line in gene PA4836  hypothetical protein [KO:K23120] [EC:2.5.1.152 2.5.1.-]\n",
      "            PA4835  hypothetical protein [KO:K23122] [EC:1.5.1.52 1.5.1.-]\n",
      "line in gene PA4835  hypothetical protein [KO:K23122] [EC:1.5.1.52 1.5.1.-]\n",
      "            PA5304  dadA; D-amino acid dehydrogenase small subunit [KO:K00285] [EC:1.4.5.1]\n",
      "line in gene PA5304  dadA; D-amino acid dehydrogenase small subunit [KO:K00285] [EC:1.4.5.1]\n",
      "            PA5084  oxidoreductase [KO:K00285] [EC:1.4.5.1]\n",
      "line in gene PA5084  oxidoreductase [KO:K00285] [EC:1.4.5.1]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00025  L-Glutamate\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00036  Oxaloacetate\n",
      "            C00037  Glycine\n",
      "            C00041  L-Alanine\n",
      "            C00047  L-Lysine\n",
      "            C00049  L-Aspartate\n",
      "            C00062  L-Arginine\n",
      "            C00064  L-Glutamine\n",
      "            C00065  L-Serine\n",
      "            C00073  L-Methionine\n",
      "            C00077  L-Ornithine\n",
      "            C00079  L-Phenylalanine\n",
      "            C00084  Acetaldehyde\n",
      "            C00097  L-Cysteine\n",
      "            C00133  D-Alanine\n",
      "            C00134  Putrescine\n",
      "            C00135  L-Histidine\n",
      "            C00148  L-Proline\n",
      "            C00166  Phenylpyruvate\n",
      "            C00188  L-Threonine\n",
      "            C00217  D-Glutamate\n",
      "            C00402  D-Aspartate\n",
      "            C00431  5-Aminopentanoate\n",
      "            C00433  2,5-Dioxopentanoate\n",
      "            C00515  D-Ornithine\n",
      "            C00666  LL-2,6-Diaminoheptanedioate\n",
      "            C00680  meso-2,6-Diaminoheptanedioate\n",
      "            C00692  UDP-N-acetylmuramoyl-L-alanyl-D-glutamate\n",
      "            C00739  D-Lysine\n",
      "            C00740  D-Serine\n",
      "            C00763  D-Proline\n",
      "            C00792  D-Arginine\n",
      "            C00793  D-Cysteine\n",
      "            C00819  D-Glutamine\n",
      "            C00820  D-Threonine\n",
      "            C00855  D-Methionine\n",
      "            C00993  D-Alanyl-D-alanine\n",
      "            C01110  5-Amino-2-oxopentanoic acid\n",
      "            C01157  Hydroxyproline\n",
      "            C01180  4-Methylthio-2-oxobutanoic acid\n",
      "            C01667  Bacitracin\n",
      "            C01672  Cadaverine\n",
      "            C01726  D-Lombricine\n",
      "            C02237  5-Oxo-D-proline\n",
      "            C02265  D-Phenylalanine\n",
      "            C02855  N-Phospho-D-lombricine\n",
      "            C03239  6-Amino-2-oxohexanoate\n",
      "            C03341  2-Amino-4-oxopentanoic acid\n",
      "            C03440  cis-4-Hydroxy-D-proline\n",
      "            C03564  1-Pyrroline-2-carboxylate\n",
      "            C03771  5-Guanidino-2-oxopentanoate\n",
      "            C03933  5-D-Glutamyl-D-glutamyl-peptide\n",
      "            C03943  (2R,4S)-2,4-Diaminopentanoate\n",
      "            C04260  O-D-Alanyl-poly(ribitol phosphate)\n",
      "            C04282  1-Pyrroline-4-hydroxy-2-carboxylate\n",
      "            C04457  D-Alanyl-alanyl-poly(glycerolphosphate)\n",
      "            C05161  (2R,5S)-2,5-Diaminohexanoate\n",
      "            C05620  N-Acetyl-D-phenylalanine\n",
      "            C05825  2-Amino-5-oxohexanoate\n",
      "            C05939  Linatine\n",
      "            C05941  2-Oxo-4-hydroxy-5-aminovalerate\n",
      "            C05942  Pyrrole-2-carboxylate\n",
      "            C06419  D-Histidine\n",
      "            C22024  Staphylopine\n",
      "            C22025  (2S)-2-Amino-4-{[(1R)-1-carboxy-2-(1H-imidazol-4-yl)ethyl]amino}butanoate\n",
      "REFERENCE   PMID:33788945\n",
      "  AUTHORS   Miyamoto T, Homma H\n",
      "  TITLE     D-Amino acid metabolism in bacteria.\n",
      "  JOURNAL   J Biochem 170:5-13 (2021)\n",
      "            DOI:10.1093/jb/mvab043\n",
      "REFERENCE   PMID:27230378\n",
      "  AUTHORS   Ghssein G, Brutesco C, Ouerdane L, Fojcik C, Izaute A, Wang S, Hajjar C, Lobinski R, Lemaire D, Richaud P, Voulhoux R, Espaillat A, Cava F, Pignol D, Borezee-Durant E, Arnoux P\n",
      "  TITLE     Biosynthesis of a broad-spectrum nicotianamine-like metallophore in Staphylococcus aureus.\n",
      "  JOURNAL   Science 352:1105-9 (2016)\n",
      "            DOI:10.1126/science.aaf1018\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00300  Lysine biosynthesis\n",
      "            pae00310  Lysine degradation\n",
      "            pae00330  Arginine and proline metabolism\n",
      "            pae00340  Histidine metabolism\n",
      "            pae00360  Phenylalanine metabolism\n",
      "            pae00550  Peptidoglycan biosynthesis\n",
      "            pae00620  Pyruvate metabolism\n",
      "KO_PATHWAY  ko00470\n",
      "///\n",
      "path:pae00480\n",
      "ENTRY       pae00480                    Pathway\n",
      "NAME        Glutathione metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of other amino acids\n",
      "PATHWAY_MAP pae00480  Glutathione metabolism\n",
      "MODULE      pae_M00118  Glutathione biosynthesis, glutamate => glutathione [PATH:pae00480]\n",
      "DBLINKS     GO: 0006749\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0361  gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "line in gene PA0361  gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "            PA0164  gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "line in gene PA0164  gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "            PA1338  ggt; gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "line in gene PA1338  ggt; gamma-glutamyltranspeptidase [KO:K00681] [EC:2.3.2.2 3.4.19.13]\n",
      "            PA0492  hypothetical protein [KO:K07160] [EC:3.5.2.9]\n",
      "line in gene PA0492  hypothetical protein [KO:K07160] [EC:3.5.2.9]\n",
      "            PA2112  hypothetical protein [KO:K07160] [EC:3.5.2.9]\n",
      "line in gene PA2112  hypothetical protein [KO:K07160] [EC:3.5.2.9]\n",
      "            PA4511  hypothetical protein [KO:K07160] [EC:3.5.2.9]\n",
      "line in gene PA4511  hypothetical protein [KO:K07160] [EC:3.5.2.9]\n",
      "            PA2110  hypothetical protein [KO:K23124] [EC:3.5.2.9]\n",
      "line in gene PA2110  hypothetical protein [KO:K23124] [EC:3.5.2.9]\n",
      "            PA4509  hypothetical protein [KO:K23124] [EC:3.5.2.9]\n",
      "line in gene PA4509  hypothetical protein [KO:K23124] [EC:3.5.2.9]\n",
      "            PA2181  glutamate--cysteine ligase [KO:K06048] [EC:6.3.2.2 6.3.-.-]\n",
      "line in gene PA2181  glutamate--cysteine ligase [KO:K06048] [EC:6.3.2.2 6.3.-.-]\n",
      "            PA5203  gshA; glutamate--cysteine ligase [KO:K01919] [EC:6.3.2.2]\n",
      "line in gene PA5203  gshA; glutamate--cysteine ligase [KO:K01919] [EC:6.3.2.2]\n",
      "            PA0407  gshB; glutathione synthetase [KO:K01920] [EC:6.3.2.3]\n",
      "line in gene PA0407  gshB; glutathione synthetase [KO:K01920] [EC:6.3.2.3]\n",
      "            PA3831  pepA; leucyl aminopeptidase [KO:K01255] [EC:3.4.11.1]\n",
      "line in gene PA3831  pepA; leucyl aminopeptidase [KO:K01255] [EC:3.4.11.1]\n",
      "            PA3083  pepN; aminopeptidase [KO:K01256] [EC:3.4.11.2]\n",
      "line in gene PA3083  pepN; aminopeptidase [KO:K01256] [EC:3.4.11.2]\n",
      "            PA1655  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "line in gene PA1655  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "            PA3035  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "line in gene PA3035  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "            PA1466  hypothetical protein [KO:K00799] [EC:2.5.1.18]\n",
      "line in gene PA1466  hypothetical protein [KO:K00799] [EC:2.5.1.18]\n",
      "            PA1890  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "line in gene PA1890  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "            PA2813  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "line in gene PA2813  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "            PA4401  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "line in gene PA4401  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "            PA0473  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "line in gene PA0473  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "            PA2821  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "line in gene PA2821  glutathione S-transferase [KO:K00799] [EC:2.5.1.18]\n",
      "            PA2025  gor; glutathione reductase [KO:K00383] [EC:1.8.1.7]\n",
      "line in gene PA2025  gor; glutathione reductase [KO:K00383] [EC:1.8.1.7]\n",
      "            PA2623  icd; isocitrate dehydrogenase [KO:K00031] [EC:1.1.1.42]\n",
      "line in gene PA2623  icd; isocitrate dehydrogenase [KO:K00031] [EC:1.1.1.42]\n",
      "            PA2624  idh; isocitrate dehydrogenase [KO:K00031] [EC:1.1.1.42]\n",
      "line in gene PA2624  idh; isocitrate dehydrogenase [KO:K00031] [EC:1.1.1.42]\n",
      "            PA2263  2-hydroxyacid dehydrogenase [KO:K00032] [EC:1.1.1.43]\n",
      "line in gene PA2263  2-hydroxyacid dehydrogenase [KO:K00032] [EC:1.1.1.43]\n",
      "            PA5439  glucose-6-phosphate 1-dehydrogenase [KO:K00036] [EC:1.1.1.49 1.1.1.363]\n",
      "line in gene PA5439  glucose-6-phosphate 1-dehydrogenase [KO:K00036] [EC:1.1.1.49 1.1.1.363]\n",
      "            PA3183  zwf; glucose-6-phosphate 1-dehydrogenase [KO:K00036] [EC:1.1.1.49 1.1.1.363]\n",
      "line in gene PA3183  zwf; glucose-6-phosphate 1-dehydrogenase [KO:K00036] [EC:1.1.1.49 1.1.1.363]\n",
      "            PA0838  glutathione peroxidase [KO:K00432] [EC:1.11.1.9]\n",
      "line in gene PA0838  glutathione peroxidase [KO:K00432] [EC:1.11.1.9]\n",
      "            PA1287  glutathione peroxidase [KO:K00432] [EC:1.11.1.9]\n",
      "line in gene PA1287  glutathione peroxidase [KO:K00432] [EC:1.11.1.9]\n",
      "            PA2826  glutathione peroxidase [KO:K00432] [EC:1.11.1.9]\n",
      "line in gene PA2826  glutathione peroxidase [KO:K00432] [EC:1.11.1.9]\n",
      "            PA4519  speC; ornithine decarboxylase [KO:K01581] [EC:4.1.1.17]\n",
      "line in gene PA4519  speC; ornithine decarboxylase [KO:K01581] [EC:4.1.1.17]\n",
      "            PA1687  speE; polyamine aminopropyltransferase [KO:K00797] [EC:2.5.1.16]\n",
      "line in gene PA1687  speE; polyamine aminopropyltransferase [KO:K00797] [EC:2.5.1.16]\n",
      "            PA4774  hypothetical protein [KO:K00797] [EC:2.5.1.16]\n",
      "line in gene PA4774  hypothetical protein [KO:K00797] [EC:2.5.1.16]\n",
      "COMPOUND    C00005  NADPH\n",
      "            C00006  NADP+\n",
      "            C00024  Acetyl-CoA\n",
      "            C00025  L-Glutamate\n",
      "            C00037  Glycine\n",
      "            C00051  Glutathione\n",
      "            C00072  Ascorbate\n",
      "            C00077  L-Ornithine\n",
      "            C00097  L-Cysteine\n",
      "            C00127  Glutathione disulfide\n",
      "            C00134  Putrescine\n",
      "            C00151  L-Amino acid\n",
      "            C00315  Spermidine\n",
      "            C00669  gamma-L-Glutamyl-L-cysteine\n",
      "            C00750  Spermine\n",
      "            C01322  RX\n",
      "            C01419  Cys-Gly\n",
      "            C01672  Cadaverine\n",
      "            C01879  5-Oxoproline\n",
      "            C02090  Trypanothione\n",
      "            C02320  R-S-Glutathione\n",
      "            C03170  Trypanothione disulfide\n",
      "            C03646  Bis-gamma-glutamylcystine\n",
      "            C03740  (5-L-Glutamyl)-L-amino acid\n",
      "            C05422  Dehydroascorbate\n",
      "            C05726  S-Substituted L-cysteine\n",
      "            C05727  S-Substituted N-acetyl-L-cysteine\n",
      "            C05729  R-S-Cysteinylglycine\n",
      "            C05730  Glutathionylspermidine\n",
      "            C16562  Glutathionylspermine\n",
      "            C16563  Bis(glutathionyl)spermine\n",
      "            C16564  Bis(glutathionyl)spermine disulfide\n",
      "            C16565  Aminopropylcadaverine\n",
      "            C16566  Glutathionylaminopropylcadaverine\n",
      "            C16567  Homotrypanothione\n",
      "            C16568  Homotrypanothione disulfide\n",
      "            C16663  Tryparedoxin\n",
      "            C16664  Tryparedoxin disulfide\n",
      "REFERENCE   PMID:12675513\n",
      "  AUTHORS   Josch C, Klotz LO, Sies H.\n",
      "  TITLE     Identification of cytosolic leucyl aminopeptidase (EC 3.4.11.1) as the major cysteinylglycine-hydrolysing activity in rat liver.\n",
      "  JOURNAL   Biol Chem 384:213-8 (2003)\n",
      "            DOI:10.1515/BC.2003.023\n",
      "REFERENCE   PMID:18482986\n",
      "  AUTHORS   Chu L, Lai Y, Xu X, Eddy S, Yang S, Song L, Kolodrubetz D.\n",
      "  TITLE     A 52-kDa leucyl aminopeptidase from treponema denticola is a cysteinylglycinase that mediates the second step of glutathione metabolism.\n",
      "  JOURNAL   J Biol Chem 283:19351-8 (2008)\n",
      "            DOI:10.1074/jbc.M801034200\n",
      "REFERENCE   PMID:14583094\n",
      "  AUTHORS   Cappiello M, Lazzarotti A, Buono F, Scaloni A, D'Ambrosio C, Amodeo P, Mendez BL, Pelosi P, Del Corso A, Mura U.\n",
      "  TITLE     New role for leucyl aminopeptidase in glutathione turnover.\n",
      "  JOURNAL   Biochem J 378:35-44 (2004)\n",
      "            DOI:10.1042/BJ20031336\n",
      "REFERENCE   PMID:11157967\n",
      "  AUTHORS   Suzuki H, Kamatani S, Kim ES, Kumagai H.\n",
      "  TITLE     Aminopeptidases A, B, and N and dipeptidase D are the four cysteinylglycinases of Escherichia coli K-12.\n",
      "  JOURNAL   J Bacteriol 183:1489-90 (2001)\n",
      "            DOI:10.1128/JB.183.4.1489-1490.2001\n",
      "REFERENCE   PMID:15610825\n",
      "  AUTHORS   Oza SL, Shaw MP, Wyllie S, Fairlamb AH.\n",
      "  TITLE     Trypanothione biosynthesis in Leishmania major.\n",
      "  JOURNAL   Mol Biochem Parasitol 139:107-16 (2005)\n",
      "            DOI:10.1016/j.molbiopara.2004.10.004\n",
      "REFERENCE   PMID:12049631\n",
      "  AUTHORS   Oza SL, Ariyanayagam MR, Fairlamb AH.\n",
      "  TITLE     Characterization of recombinant glutathionylspermidine synthetase/amidase from Crithidia fasciculata.\n",
      "  JOURNAL   Biochem J 364:679-86 (2002)\n",
      "            DOI:10.1042/BJ20011370\n",
      "REFERENCE   PMID:9677355\n",
      "  AUTHORS   Tetaud E, Manai F, Barrett MP, Nadeau K, Walsh CT, Fairlamb AH.\n",
      "  TITLE     Cloning and characterization of the two enzymes responsible for trypanothione biosynthesis in Crithidia fasciculata.\n",
      "  JOURNAL   J Biol Chem 273:19383-90 (1998)\n",
      "            DOI:10.1074/jbc.273.31.19383\n",
      "REFERENCE   PMID:12750367\n",
      "  AUTHORS   Ariyanayagam MR, Oza SL, Mehlert A, Fairlamb AH.\n",
      "  TITLE     Bis(glutathionyl)spermine and other novel trypanothione analogues in Trypanosoma cruzi.\n",
      "  JOURNAL   J Biol Chem 278:27612-9 (2003)\n",
      "            DOI:10.1074/jbc.M302750200\n",
      "REFERENCE   PMID:12967709\n",
      "  AUTHORS   Oza SL, Ariyanayagam MR, Aitcheson N, Fairlamb AH.\n",
      "  TITLE     Properties of trypanothione synthetase from Trypanosoma brucei.\n",
      "  JOURNAL   Mol Biochem Parasitol 131:25-33 (2003)\n",
      "            DOI:10.1016/S0166-6851(03)00176-2\n",
      "REFERENCE   PMID:12121990\n",
      "  AUTHORS   Oza SL, Tetaud E, Ariyanayagam MR, Warnon SS, Fairlamb AH.\n",
      "  TITLE     A single enzyme catalyses formation of Trypanothione from glutathione and spermidine in Trypanosoma cruzi.\n",
      "  JOURNAL   J Biol Chem 277:35853-61 (2002)\n",
      "            DOI:10.1074/jbc.M204403200\n",
      "REFERENCE   PMID:15537651\n",
      "  AUTHORS   Comini M, Menge U, Wissing J, Flohe L.\n",
      "  TITLE     Trypanothione synthesis in crithidia revisited.\n",
      "  JOURNAL   J Biol Chem 280:6850-60 (2005)\n",
      "            DOI:10.1074/jbc.M404486200\n",
      "REFERENCE   PMID:7813456\n",
      "  AUTHORS   Hunter KJ, Le Quesne SA, Fairlamb AH.\n",
      "  TITLE     Identification and biosynthesis of N1,N9-bis(glutathionyl)aminopropylcadaverine (homotrypanothione) in Trypanosoma cruzi.\n",
      "  JOURNAL   Eur J Biochem 226:1019-27 (1994)\n",
      "            DOI:10.1111/j.1432-1033.1994.t01-1-01019.x\n",
      "REFERENCE   PMID:12751784\n",
      "  AUTHORS   Krauth-Siegel RL, Meiering SK, Schmidt H.\n",
      "  TITLE     The parasite-specific trypanothione metabolism of trypanosoma and leishmania.\n",
      "  JOURNAL   Biol Chem 384:539-49 (2003)\n",
      "            DOI:10.1515/BC.2003.062\n",
      "REFERENCE   PMID:18395526\n",
      "  AUTHORS   Krauth-Siegel RL, Comini MA.\n",
      "  TITLE     Redox control in trypanosomatids, parasitic protozoa with trypanothione-based thiol metabolism.\n",
      "  JOURNAL   Biochim Biophys Acta 1780:1236-48 (2008)\n",
      "            DOI:10.1016/j.bbagen.2008.03.006\n",
      "REFERENCE   PMID:8892297\n",
      "  AUTHORS   Krauth-Siegel RL, Ludemann H.\n",
      "  TITLE     Reduction of dehydroascorbate by trypanothione.\n",
      "  JOURNAL   Mol Biochem Parasitol 80:203-8 (1996)\n",
      "            DOI:10.1016/0166-6851(96)02689-8\n",
      "REFERENCE   PMID:11150302\n",
      "  AUTHORS   Dormeyer M, Reckenfelderbaumer N, Ludemann H, Krauth-Siegel RL.\n",
      "  TITLE     Trypanothione-dependent synthesis of deoxyribonucleotides by Trypanosoma brucei ribonucleotide reductase.\n",
      "  JOURNAL   J Biol Chem 276:10602-6 (2001)\n",
      "            DOI:10.1074/jbc.M010352200\n",
      "REFERENCE   PMID:12949079\n",
      "  AUTHORS   Schmidt H, Krauth-Siegel RL.\n",
      "  TITLE     Functional and physicochemical characterization of the thioredoxin system in Trypanosoma brucei.\n",
      "  JOURNAL   J Biol Chem 278:46329-36 (2003)\n",
      "            DOI:10.1074/jbc.M305338200\n",
      "REFERENCE   PMID:9851611\n",
      "  AUTHORS   Tetaud E, Fairlamb AH.\n",
      "  TITLE     Cloning, expression and reconstitution of the trypanothione-dependent peroxidase system of Crithidia fasciculata.\n",
      "  JOURNAL   Mol Biochem Parasitol 96:111-23 (1998)\n",
      "            DOI:10.1016/S0166-6851(98)00120-0\n",
      "REFERENCE   PMID:12446213\n",
      "  AUTHORS   Castro H, Sousa C, Santos M, Cordeiro-da-Silva A, Flohe L, Tomas AM.\n",
      "  TITLE     Complementary antioxidant defense by cytoplasmic and mitochondrial peroxiredoxins in Leishmania infantum.\n",
      "  JOURNAL   Free Radic Biol Med 33:1552-62 (2002)\n",
      "            DOI:10.1016/S0891-5849(02)01089-4\n",
      "REFERENCE   PMID:10713147\n",
      "  AUTHORS   Wilkinson SR, Temperton NJ, Mondragon A, Kelly JM.\n",
      "  TITLE     Distinct mitochondrial and cytosolic enzymes mediate trypanothione-dependent peroxide metabolism in Trypanosoma cruzi.\n",
      "  JOURNAL   J Biol Chem 275:8220-5 (2000)\n",
      "            DOI:10.1074/jbc.275.11.8220\n",
      "REFERENCE   PMID:17922848\n",
      "  AUTHORS   Konig J, Fairlamb AH.\n",
      "  TITLE     A comparative study of type I and type II tryparedoxin peroxidases in Leishmania major.\n",
      "  JOURNAL   FEBS J 274:5643-58 (2007)\n",
      "            DOI:10.1111/j.1742-4658.2007.06087.x\n",
      "REFERENCE   PMID:12466271\n",
      "  AUTHORS   Hillebrand H, Schmidt A, Krauth-Siegel RL.\n",
      "  TITLE     A second class of peroxidases linked to the trypanothione metabolism.\n",
      "  JOURNAL   J Biol Chem 278:6809-15 (2003)\n",
      "            DOI:10.1074/jbc.M210392200\n",
      "REFERENCE   PMID:14982633\n",
      "  AUTHORS   Soksawatmaekhin W, Kuraishi A, Sakata K, Kashiwagi K, Igarashi K.\n",
      "  TITLE     Excretion and uptake of cadaverine by CadB and its physiological functions in Escherichia coli.\n",
      "  JOURNAL   Mol Microbiol 51:1401-12 (2004)\n",
      "            DOI:10.1046/j.1365-2958.2003.03913.x\n",
      "REFERENCE   PMID:6798961\n",
      "  AUTHORS   Pegg AE, Shuttleworth K, Hibasami H.\n",
      "  TITLE     Specificity of mammalian spermidine synthase and spermine synthase.\n",
      "  JOURNAL   Biochem J 197:315-20 (1981)\n",
      "            DOI:10.1042/bj1970315\n",
      "REL_PATHWAY pae00220  Arginine biosynthesis\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00430  Taurine and hypotaurine metabolism\n",
      "            pae00460  Cyanoamino acid metabolism\n",
      "KO_PATHWAY  ko00480\n",
      "///\n",
      "path:pae00500\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00500                    Pathway\n",
      "NAME        Starch and sucrose metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00500  Starch and sucrose metabolism\n",
      "DBLINKS     GO: 0005982 0005985\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2023  galU; UTP-glucose-1-phosphate uridylyltransferase [KO:K00963] [EC:2.7.7.9]\n",
      "line in gene PA2023  galU; UTP-glucose-1-phosphate uridylyltransferase [KO:K00963] [EC:2.7.7.9]\n",
      "            PA1726  bglX; beta-glucosidase [KO:K05349] [EC:3.2.1.21]\n",
      "line in gene PA1726  bglX; beta-glucosidase [KO:K05349] [EC:3.2.1.21]\n",
      "            PA2165  glycogen synthase [KO:K00703] [EC:2.4.1.21]\n",
      "line in gene PA2165  glycogen synthase [KO:K00703] [EC:2.4.1.21]\n",
      "            PA2153  glgB; 1,4-alpha-glucan branching protein GlgB [KO:K00700] [EC:2.4.1.18]\n",
      "line in gene PA2153  glgB; 1,4-alpha-glucan branching protein GlgB [KO:K00700] [EC:2.4.1.18]\n",
      "            PA2144  glgP; glycogen phosphorylase [KO:K00688] [EC:2.4.1.1]\n",
      "line in gene PA2144  glgP; glycogen phosphorylase [KO:K00688] [EC:2.4.1.1]\n",
      "            PA2152  trehalose synthase [KO:K05343] [EC:5.4.99.16 3.2.1.1]\n",
      "line in gene PA2152  trehalose synthase [KO:K05343] [EC:5.4.99.16 3.2.1.1]\n",
      "            PA2163  4-alpha-glucanotransferase [KO:K00705] [EC:2.4.1.25]\n",
      "line in gene PA2163  4-alpha-glucanotransferase [KO:K00705] [EC:2.4.1.25]\n",
      "            PA2151  alpha-1,4-glucan:maltose-1-phosphate maltosyltransferase [KO:K16147] [EC:2.4.99.16]\n",
      "line in gene PA2151  alpha-1,4-glucan:maltose-1-phosphate maltosyltransferase [KO:K16147] [EC:2.4.99.16]\n",
      "            PA2160  glycosyl hydrolase [KO:K01214] [EC:3.2.1.68]\n",
      "line in gene PA2160  glycosyl hydrolase [KO:K01214] [EC:3.2.1.68]\n",
      "            PA2162  malto-oligosyltrehalose synthase [KO:K06044] [EC:5.4.99.15]\n",
      "line in gene PA2162  malto-oligosyltrehalose synthase [KO:K06044] [EC:5.4.99.15]\n",
      "            PA2164  glycosyl hydrolase [KO:K01236] [EC:3.2.1.141]\n",
      "line in gene PA2164  glycosyl hydrolase [KO:K01236] [EC:3.2.1.141]\n",
      "            PA2416  treA; trehalase [KO:K01194] [EC:3.2.1.28]\n",
      "line in gene PA2416  treA; trehalase [KO:K01194] [EC:3.2.1.28]\n",
      "            PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "line in gene PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "            PA3193  glk; glucokinase [KO:K00845] [EC:2.7.1.2]\n",
      "line in gene PA3193  glk; glucokinase [KO:K00845] [EC:2.7.1.2]\n",
      "            PA4732  pgi; glucose-6-phosphate isomerase [KO:K01810] [EC:5.3.1.9]\n",
      "line in gene PA4732  pgi; glucose-6-phosphate isomerase [KO:K01810] [EC:5.3.1.9]\n",
      "            PA2344  mtlZ; fructokinase [KO:K00847] [EC:2.7.1.4]\n",
      "line in gene PA2344  mtlZ; fructokinase [KO:K00847] [EC:2.7.1.4]\n",
      "COMPOUND    C00029  UDP-glucose\n",
      "            C00031  D-Glucose\n",
      "            C00085  D-Fructose 6-phosphate\n",
      "            C00089  Sucrose\n",
      "            C00092  D-Glucose 6-phosphate\n",
      "            C00095  D-Fructose\n",
      "            C00103  D-Glucose 1-phosphate\n",
      "            C00185  Cellobiose\n",
      "            C00208  Maltose\n",
      "            C00252  Isomaltose\n",
      "            C00369  Starch\n",
      "            C00372  Dextran\n",
      "            C00394  GDP-glucose\n",
      "            C00498  ADP-glucose\n",
      "            C00501  CDP-glucose\n",
      "            C00663  beta-D-Glucose 1-phosphate\n",
      "            C00689  alpha,alpha'-Trehalose 6-phosphate\n",
      "            C00718  Amylose\n",
      "            C00721  Dextrin\n",
      "            C00760  Cellulose\n",
      "            C00963  beta-D-Glucoside\n",
      "            C00965  1,3-beta-D-Glucan\n",
      "            C00973  Cyclomaltodextrin\n",
      "            C01083  alpha,alpha-Trehalose\n",
      "            C01231  alpha-D-Glucose 1,6-bisphosphate\n",
      "            C01725  Levanbiose\n",
      "            C01898  Cellodextrin\n",
      "            C01935  Maltodextrin\n",
      "            C02591  Sucrose 6'-phosphate\n",
      "            C02995  Maltose 6'-phosphate\n",
      "            C03323  (2,1-beta-D-Fructosyl)n\n",
      "            C04534  6-Phospho-beta-D-glucosyl-(1,4)-D-glucose\n",
      "            C05731  3-Ketosucrose\n",
      "            C06215  Levan\n",
      "            C06400  1-alpha-D-[(1->4)-alpha-D-Glucosyl](n-1)-alpha-D-glucopyranoside\n",
      "            C16688  Sucrose 6-phosphate\n",
      "            C20237  alpha-Maltose 1-phosphate\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "KO_PATHWAY  ko00500\n",
      "///\n",
      "path:pae00520\n",
      "ENTRY       pae00520                    Pathway\n",
      "NAME        Amino sugar and nucleotide sugar metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "MODULE      pae_M00549  Nucleotide sugar biosynthesis, glucose => UDP-glucose [PATH:pae00520]\n",
      "            pae_M00761  Undecaprenylphosphate alpha-L-Ara4N biosynthesis, UDP-GlcA => undecaprenyl phosphate alpha-L-Ara4N [PATH:pae00520]\n",
      "            pae_M00909  UDP-N-acetyl-D-glucosamine biosynthesis, prokaryotes, glucose => UDP-GlcNAc [PATH:pae00520]\n",
      "DBLINKS     GO: 0006040 0009225\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2300  chiC; chitinase [KO:K01183] [EC:3.2.1.14]\n",
      "line in gene PA2300  chiC; chitinase [KO:K01183] [EC:3.2.1.14]\n",
      "            PA3005  nagZ; beta-hexosaminidase [KO:K01207] [EC:3.2.1.52]\n",
      "line in gene PA3005  nagZ; beta-hexosaminidase [KO:K01207] [EC:3.2.1.52]\n",
      "            PA3761  N-acetyl-D-glucosamine phosphotransferase system transporter [KO:K02804] [EC:2.7.1.193]\n",
      "line in gene PA3761  N-acetyl-D-glucosamine phosphotransferase system transporter [KO:K02804] [EC:2.7.1.193]\n",
      "            PA3758  N-acetylglucosamine-6-phosphate deacetylase [KO:K01443] [EC:3.5.1.25]\n",
      "line in gene PA3758  N-acetylglucosamine-6-phosphate deacetylase [KO:K01443] [EC:3.5.1.25]\n",
      "            PA2344  mtlZ; fructokinase [KO:K00847] [EC:2.7.1.4]\n",
      "line in gene PA2344  mtlZ; fructokinase [KO:K00847] [EC:2.7.1.4]\n",
      "            PA5549  glmS; glucosamine--fructose-6-phosphate aminotransferase [KO:K00820] [EC:2.6.1.16]\n",
      "line in gene PA5549  glmS; glucosamine--fructose-6-phosphate aminotransferase [KO:K00820] [EC:2.6.1.16]\n",
      "            PA3759  aminotransferase [KO:K00820] [EC:2.6.1.16]\n",
      "line in gene PA3759  aminotransferase [KO:K00820] [EC:2.6.1.16]\n",
      "            PA4749  glmM; phosphoglucosamine mutase [KO:K03431] [EC:5.4.2.10]\n",
      "line in gene PA4749  glmM; phosphoglucosamine mutase [KO:K03431] [EC:5.4.2.10]\n",
      "            PA5552  glmU; bifunctional glucosamine-1-phosphate acetyltransferase/N-acetylglucosamine-1-phosphate uridyltransferase [KO:K04042] [EC:2.7.7.23 2.3.1.157]\n",
      "line in gene PA5552  glmU; bifunctional glucosamine-1-phosphate acetyltransferase/N-acetylglucosamine-1-phosphate uridyltransferase [KO:K04042] [EC:2.7.7.23 2.3.1.157]\n",
      "            PA0666  anhydro-N-acetylmuramic acid kinase [KO:K09001] [EC:2.7.1.170]\n",
      "line in gene PA0666  anhydro-N-acetylmuramic acid kinase [KO:K09001] [EC:2.7.1.170]\n",
      "            PA3172  phosphoglycolate phosphatase [KO:K22292] [EC:3.1.3.105]\n",
      "line in gene PA3172  phosphoglycolate phosphatase [KO:K22292] [EC:3.1.3.105]\n",
      "            PA0596  hypothetical protein [KO:K07102] [EC:2.7.1.221]\n",
      "line in gene PA0596  hypothetical protein [KO:K07102] [EC:2.7.1.221]\n",
      "            PA0597  nucleotidyl transferase [KO:K00992] [EC:2.7.7.99]\n",
      "line in gene PA0597  nucleotidyl transferase [KO:K00992] [EC:2.7.7.99]\n",
      "            PA4450  murA; UDP-N-acetylglucosamine 1-carboxyvinyltransferase [KO:K00790] [EC:2.5.1.7]\n",
      "line in gene PA4450  murA; UDP-N-acetylglucosamine 1-carboxyvinyltransferase [KO:K00790] [EC:2.5.1.7]\n",
      "            PA2977  murB; UDP-N-acetylenolpyruvoylglucosamine reductase [KO:K00075] [EC:1.3.1.98]\n",
      "line in gene PA2977  murB; UDP-N-acetylenolpyruvoylglucosamine reductase [KO:K00075] [EC:1.3.1.98]\n",
      "            PA3159  wbpA; UDP-N-acetyl-d-glucosamine 6-dehydrogenase [KO:K13015] [EC:1.1.1.136]\n",
      "line in gene PA3159  wbpA; UDP-N-acetyl-d-glucosamine 6-dehydrogenase [KO:K13015] [EC:1.1.1.136]\n",
      "            PA3158  wbpB; UDP-N-acetyl-2-amino-2-deoxy-D-glucuronate oxidase [KO:K13016] [EC:1.1.1.335]\n",
      "line in gene PA3158  wbpB; UDP-N-acetyl-2-amino-2-deoxy-D-glucuronate oxidase [KO:K13016] [EC:1.1.1.335]\n",
      "            PA3155  wbpE; UDP-2-acetamido-2-deoxy-3-oxo-D-glucuronate aminotransferase [KO:K13017] [EC:2.6.1.98]\n",
      "line in gene PA3155  wbpE; UDP-2-acetamido-2-deoxy-3-oxo-D-glucuronate aminotransferase [KO:K13017] [EC:2.6.1.98]\n",
      "            PA3156  wbpD; UDP-2-acetamido-3-amino-2,3-dideoxy-D-glucuronate N-acetyltransferase [KO:K13018] [EC:2.3.1.201]\n",
      "line in gene PA3156  wbpD; UDP-2-acetamido-3-amino-2,3-dideoxy-D-glucuronate N-acetyltransferase [KO:K13018] [EC:2.3.1.201]\n",
      "            PA3148  wbpI; UDP-2,3-diacetamido-2,3-dideoxy-D-glucuronate 2-epimeras [KO:K13019] [EC:5.1.3.23]\n",
      "line in gene PA3148  wbpI; UDP-2,3-diacetamido-2,3-dideoxy-D-glucuronate 2-epimeras [KO:K13019] [EC:5.1.3.23]\n",
      "            PA3554  arnA; bifunctional UDP-glucuronic acid decarboxylase/UDP-4-amino-4-deoxy-L-arabinose formyltransferase [KO:K10011] [EC:2.1.2.13 1.1.1.305]\n",
      "line in gene PA3554  arnA; bifunctional UDP-glucuronic acid decarboxylase/UDP-4-amino-4-deoxy-L-arabinose formyltransferase [KO:K10011] [EC:2.1.2.13 1.1.1.305]\n",
      "            PA3552  arnB; UDP-4-amino-4-deoxy-L-arabinose--oxoglutarate aminotransferase [KO:K07806] [EC:2.6.1.87]\n",
      "line in gene PA3552  arnB; UDP-4-amino-4-deoxy-L-arabinose--oxoglutarate aminotransferase [KO:K07806] [EC:2.6.1.87]\n",
      "            PA3553  arnC; undecaprenyl-phosphate 4-deoxy-4-formamido-L-arabinose transferase [KO:K10012] [EC:2.4.2.53]\n",
      "line in gene PA3553  arnC; undecaprenyl-phosphate 4-deoxy-4-formamido-L-arabinose transferase [KO:K10012] [EC:2.4.2.53]\n",
      "            PA3555  arnD; 4-deoxy-4-formamido-L-arabinose-phosphoundecaprenol deformylase ArnD [KO:K13014] [EC:3.5.1.-]\n",
      "line in gene PA3555  arnD; 4-deoxy-4-formamido-L-arabinose-phosphoundecaprenol deformylase ArnD [KO:K13014] [EC:3.5.1.-]\n",
      "            PA3193  glk; glucokinase [KO:K00845] [EC:2.7.1.2]\n",
      "line in gene PA3193  glk; glucokinase [KO:K00845] [EC:2.7.1.2]\n",
      "            PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "line in gene PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "            PA2023  galU; UTP-glucose-1-phosphate uridylyltransferase [KO:K00963] [EC:2.7.7.9]\n",
      "line in gene PA2023  galU; UTP-glucose-1-phosphate uridylyltransferase [KO:K00963] [EC:2.7.7.9]\n",
      "            PA2022  UDP-glucose 6-dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "line in gene PA2022  UDP-glucose 6-dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "            PA3559  nucleotide sugar dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "line in gene PA3559  nucleotide sugar dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "            PA4068  epimerase [KO:K01784] [EC:5.1.3.2]\n",
      "line in gene PA4068  epimerase [KO:K01784] [EC:5.1.3.2]\n",
      "            PA1384  galE; UDP-glucose 4-epimerase [KO:K01784] [EC:5.1.3.2]\n",
      "line in gene PA1384  galE; UDP-glucose 4-epimerase [KO:K01784] [EC:5.1.3.2]\n",
      "            PA4732  pgi; glucose-6-phosphate isomerase [KO:K01810] [EC:5.3.1.9]\n",
      "line in gene PA4732  pgi; glucose-6-phosphate isomerase [KO:K01810] [EC:5.3.1.9]\n",
      "            PA3551  algA; bifunctional mannose-1-phosphate guanylyltransferase/mannose-6-phosphate isomerase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "line in gene PA3551  algA; bifunctional mannose-1-phosphate guanylyltransferase/mannose-6-phosphate isomerase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "            PA2232  pslB; biofilm formation protein PslB [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "line in gene PA2232  pslB; biofilm formation protein PslB [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "            PA5452  wbpW; phosphomannose isomerase/mannose-1-phosphate guanylyl transferase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "line in gene PA5452  wbpW; phosphomannose isomerase/mannose-1-phosphate guanylyl transferase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "            PA5453  gmd; GDP-mannose 4,6-dehydratase [KO:K01711] [EC:4.2.1.47]\n",
      "line in gene PA5453  gmd; GDP-mannose 4,6-dehydratase [KO:K01711] [EC:4.2.1.47]\n",
      "            PA5454  rmd; oxidoreductase Rmd [KO:K15856] [EC:1.1.1.281]\n",
      "line in gene PA5454  rmd; oxidoreductase Rmd [KO:K15856] [EC:1.1.1.281]\n",
      "            PA3540  algD; GDP-mannose 6-dehydrogenase AlgD [KO:K00066] [EC:1.1.1.132]\n",
      "line in gene PA3540  algD; GDP-mannose 6-dehydrogenase AlgD [KO:K00066] [EC:1.1.1.132]\n",
      "            PA5236  CDP-6-deoxy-delta-3,4-glucoseen reductase [KO:K00523] [EC:1.17.1.1]\n",
      "line in gene PA5236  CDP-6-deoxy-delta-3,4-glucoseen reductase [KO:K00523] [EC:1.17.1.1]\n",
      "COMPOUND    C00029  UDP-glucose\n",
      "            C00031  D-Glucose\n",
      "            C00043  UDP-N-acetyl-alpha-D-glucosamine\n",
      "            C00052  UDP-alpha-D-galactose\n",
      "            C00096  GDP-mannose\n",
      "            C00103  D-Glucose 1-phosphate\n",
      "            C00128  CMP-N-acetylneuraminate\n",
      "            C00140  N-Acetyl-D-glucosamine\n",
      "            C00159  D-Mannose\n",
      "            C00167  UDP-glucuronate\n",
      "            C00181  D-Xylose\n",
      "            C00190  UDP-D-xylose\n",
      "            C00191  D-Glucuronate\n",
      "            C00203  UDP-N-acetyl-D-galactosamine\n",
      "            C00216  D-Arabinose\n",
      "            C00259  L-Arabinose\n",
      "            C00267  alpha-D-Glucose\n",
      "            C00270  N-Acetylneuraminate\n",
      "            C00275  D-Mannose 6-phosphate\n",
      "            C00325  GDP-L-fucose\n",
      "            C00329  D-Glucosamine\n",
      "            C00333  D-Galacturonate\n",
      "            C00352  D-Glucosamine 6-phosphate\n",
      "            C00357  N-Acetyl-D-glucosamine 6-phosphate\n",
      "            C00394  GDP-glucose\n",
      "            C00446  alpha-D-Galactose 1-phosphate\n",
      "            C00461  Chitin\n",
      "            C00498  ADP-glucose\n",
      "            C00501  CDP-glucose\n",
      "            C00617  UDP-D-galacturonate\n",
      "            C00636  D-Mannose 1-phosphate\n",
      "            C00645  N-Acetyl-D-mannosamine\n",
      "            C00668  alpha-D-Glucose 6-phosphate\n",
      "            C00714  Pectin\n",
      "            C00734  Chitosan\n",
      "            C00935  UDP-L-arabinose\n",
      "            C00976  GDP-D-mannuronate\n",
      "            C00984  alpha-D-Galactose\n",
      "            C00996  Ferricytochrome b5\n",
      "            C00999  Ferrocytochrome b5\n",
      "            C01019  6-Deoxy-L-galactose\n",
      "            C01050  UDP-N-acetylmuramate\n",
      "            C01132  N-Acetyl-D-galactosamine\n",
      "            C01170  UDP-N-acetyl-D-mannosamine\n",
      "            C01219  CDP-4-dehydro-6-deoxy-D-glucose\n",
      "            C01222  GDP-4-dehydro-6-deoxy-D-mannose\n",
      "            C01623  UDP-apiose\n",
      "            C01674  Chitobiose\n",
      "            C01788  CDP-abequose\n",
      "            C02199  UDP-L-rhamnose\n",
      "            C02280  GDP-L-galactose\n",
      "            C02336  beta-D-Fructose\n",
      "            C02352  1,4-beta-D-Xylan\n",
      "            C02474  Arabinan\n",
      "            C02713  N-Acetylmuramate\n",
      "            C02977  GDP-6-deoxy-D-talose\n",
      "            C02985  L-Fucose 1-phosphate\n",
      "            C03117  GDP-6-deoxy-D-mannose\n",
      "            C03410  N-Glycoloyl-neuraminate\n",
      "            C03598  CDP-3,6-dideoxy-D-glucose\n",
      "            C03599  CDP-3,6-dideoxy-D-mannose\n",
      "            C03691  CMP-N-glycoloylneuraminate\n",
      "            C03733  UDP-alpha-D-galactofuranose\n",
      "            C03906  beta-L-Arabinose 1-phosphate\n",
      "            C04037  1-Phospho-alpha-D-galacturonate\n",
      "            C04089  UDP-4-dehydro-6-deoxy-D-glucose\n",
      "            C04257  N-Acetyl-D-mannosamine 6-phosphate\n",
      "            C04297  CDP-4-dehydro-3,6-dideoxy-D-glucose\n",
      "            C04501  N-Acetyl-alpha-D-glucosamine 1-phosphate\n",
      "            C04573  UDP-N-acetyl-2-amino-2-deoxy-D-glucuronate\n",
      "            C04613  UDP-2-acetamido-4-dehydro-2,6-dideoxyglucose\n",
      "            C04630  UDP-2-acetamido-4-amino-2,4,6-trideoxy-alpha-D-glucose\n",
      "            C04631  UDP-N-acetyl-3-(1-carboxyvinyl)-D-glucosamine\n",
      "            C05345  beta-D-Fructose 6-phosphate\n",
      "            C05385  D-Glucuronate 1-phosphate\n",
      "            C06023  D-Glucosaminide\n",
      "            C06156  alpha-D-Glucosamine 1-phosphate\n",
      "            C06240  UDP-N-acetyl-D-mannosaminouronate\n",
      "            C06241  N-Acetylneuraminate 9-phosphate\n",
      "            C06377  D-Galactosamine 6-phosphate\n",
      "            C11521  UDP-6-sulfoquinovose\n",
      "            C13952  UDP-N-acetyl-D-galactosaminuronic acid\n",
      "            C15925  GDP-L-gulose\n",
      "            C16153  UDP-L-Ara4N\n",
      "            C16154  UDP-L-Ara4FN\n",
      "            C16155  UDP-L-Ara4O\n",
      "            C16156  Undecaprenyl phosphate alpha-L-Ara4FN\n",
      "            C16157  Undecaprenyl phosphate alpha-L-Ara4N\n",
      "            C16183  alpha-D-Arabinopyranose 1-phosphate\n",
      "            C16698  N-Acetylmuramic acid 6-phosphate\n",
      "            C17269  GDP-D-arabinopyranose\n",
      "            C17326  CDP-4-dehydro-3,6-dideoxy-D-glucose epimer\n",
      "            C17327  CDP-ascarylose\n",
      "            C17328  UDP-4-keto-rhamnose\n",
      "            C18060  N-Acetyl-alpha-D-galactosamine 1-phosphate\n",
      "            C18094  UDP-L-arabinofuranose\n",
      "            C19725  UDP-2,3-diacetamido-2,3-dideoxy-alpha-D-glucuronate\n",
      "            C19745  UDP-2,3-diacetamido-2,3-dideoxy-alpha-D-mannuronate\n",
      "            C19769  1,6-Anhydro-N-acetyl-beta-muramate\n",
      "            C19823  UDP-2-acetamido-2,6-dideoxy-beta-L-arabino-hexos-4-ulose\n",
      "            C19961  UDP-4-amino-4,6-dideoxy-N-acetyl-beta-L-altrosamine\n",
      "            C19971  UDP-2,4-bis(acetamido)-2,4,6-trideoxy-beta-L-altropyranose\n",
      "            C19972  2,4-Bis(acetamido)-2,4,6-trideoxy-beta-L-altropyranose\n",
      "            C20082  Pseudaminic acid\n",
      "            C20083  CMP-pseudaminic acid\n",
      "            C20357  UDP-N,N'-diacetylbacillosamine\n",
      "            C20359  UDP-2-acetamido-3-amino-2,3-dideoxy-alpha-D-glucuronate\n",
      "            C20395  UDP-2-acetamido-2-deoxy-alpha-D-ribo-hex-3-uluronate\n",
      "            C20418  N,N'-Diacetyllegionaminate\n",
      "            C20419  CMP-N,N'-diacetyllegionaminate\n",
      "            C20424  2,4-Diacetamido-2,4,6-trideoxy-D-mannopyranose\n",
      "            C20638  GDP-4-amino-4,6-dideoxy-alpha-D-mannose\n",
      "            C20672  GDP-4-acetamido-4,6-dideoxy-alpha-D-mannose\n",
      "            C20934  3-Deoxy-D-glycero-D-galacto-non-2-ulopyranosonate\n",
      "            C21027  N-Acetylmuramic acid alpha-1-phosphate\n",
      "            C21383  3-Deoxy-D-glycero-D-galacto-non-2-ulopyranosonate 9-phosphate\n",
      "            C21384  CMP-3-deoxy-D-glycero-D-galacto-non-2-ulopyranosonate\n",
      "            C22029  alpha-D-Galactosamine 1-phosphate\n",
      "REFERENCE   PMID:2793832\n",
      "  AUTHORS   Wyk P, Reeves P\n",
      "  TITLE     Identification and sequence of the gene for abequose synthase, which confers antigenic specificity on group B salmonellae: homology with galactose epimerase.\n",
      "  JOURNAL   J Bacteriol 171:5687-93 (1989)\n",
      "            DOI:10.1128/JB.171.10.5687-5693.1989\n",
      "REFERENCE   PMID:8071227\n",
      "  AUTHORS   Thorson JS, Lo SF, Ploux O, He X, Liu HW\n",
      "  TITLE     Studies of the biosynthesis of 3,6-dideoxyhexoses: molecular cloning and characterization of the asc (ascarylose) region from Yersinia pseudotuberculosis serogroup VA.\n",
      "  JOURNAL   J Bacteriol 176:5483-93 (1994)\n",
      "            DOI:10.1128/JB.176.17.5483-5493.1994\n",
      "REFERENCE   PMID:8444803\n",
      "  AUTHORS   Kessler AC, Haase A, Reeves PR\n",
      "  TITLE     Molecular analysis of the 3,6-dideoxyhexose pathway genes of Yersinia pseudotuberculosis serogroup IIA.\n",
      "  JOURNAL   J Bacteriol 175:1412-22 (1993)\n",
      "            DOI:10.1128/JB.175.5.1412-1422.1993\n",
      "REFERENCE   PMID:9811644\n",
      "  AUTHORS   Danese PN, Oliver GR, Barr K, Bowman GD, Rick PD, Silhavy TJ\n",
      "  TITLE     Accumulation of the enterobacterial common antigen lipid II biosynthetic intermediate stimulates degP transcription in Escherichia coli.\n",
      "  JOURNAL   J Bacteriol 180:5875-84 (1998)\n",
      "            DOI:10.1128/JB.180.22.5875-5884.1998\n",
      "REFERENCE   PMID:10931327\n",
      "  AUTHORS   Skurnik M, Peippo A, Ervela E\n",
      "  TITLE     Characterization of the O-antigen gene clusters of Yersinia pseudotuberculosis and the cryptic O-antigen gene cluster of Yersinia pestis shows that the plague bacillus is most closely related to and has evolved from Y. pseudotuberculosis serotype O:1b.\n",
      "  JOURNAL   Mol Microbiol 37:316-30 (2000)\n",
      "            DOI:10.1046/j.1365-2958.2000.01993.x\n",
      "REFERENCE   PMID:10984043\n",
      "  AUTHORS   Stover CK, Pham XQ, Erwin AL, Mizoguchi SD, Warrener P, Hickey MJ, Brinkman FS, Hufnagle WO, Kowalik DJ, Lagrou M, Garber RL, Goltry L, Tolentino E, Westbrock-Wadman S, Yuan Y, Brody LL, Coulter SN, Folger KR, Kas A, Larbig K, Lim R, Smith K, Spencer D, Wong GK, Wu Z, Paulsen IT, Reizer J, Saier MH, Hancock RE, Lory S, Olson MV\n",
      "  TITLE     Complete genome sequence of Pseudomonas aeruginosa PAO1, an opportunistic pathogen.\n",
      "  JOURNAL   Nature 406:959-64 (2000)\n",
      "            DOI:10.1038/35023079\n",
      "REFERENCE   PMID:11447132\n",
      "  AUTHORS   Ringenberg M, Lichtensteiger C, Vimr E\n",
      "  TITLE     Redirection of sialic acid metabolism in genetically engineered Escherichia coli.\n",
      "  JOURNAL   Glycobiology 11:533-9 (2001)\n",
      "            DOI:10.1093/glycob/11.7.533\n",
      "REFERENCE   PMID:12562791\n",
      "  AUTHORS   McCoy AJ, Sandlin RC, Maurelli AT\n",
      "  TITLE     In vitro and in vivo functional activity of Chlamydia MurA, a UDP-N-acetylglucosamine enolpyruvyl transferase involved in peptidoglycan synthesis and fosfomycin resistance.\n",
      "  JOURNAL   J Bacteriol 185:1218-28 (2003)\n",
      "            DOI:10.1128/JB.185.4.1218-1228.2003\n",
      "REFERENCE   PMID:14450717\n",
      "  AUTHORS   JACOBSON B, DAVIDSON EA\n",
      "  TITLE     Biosynthesis of uronic acids by skin enzymes. I. Uridine diphosphate-D-glucuronic acid-5-epimerase.\n",
      "  JOURNAL   J Biol Chem 237:638-42 (1962)\n",
      "REFERENCE   PMID:15134748\n",
      "  AUTHORS   Seifert GJ\n",
      "  TITLE     Nucleotide sugar interconversions and cell wall biosynthesis: how to bring the inside to the outside.\n",
      "  JOURNAL   Curr Opin Plant Biol 7:277-84 (2004)\n",
      "            DOI:10.1016/j.pbi.2004.03.004\n",
      "REFERENCE   PMID:15489439\n",
      "  AUTHORS   Uehara T, Park JT\n",
      "  TITLE     The N-acetyl-D-glucosamine kinase of Escherichia coli and its role in murein recycling.\n",
      "  JOURNAL   J Bacteriol 186:7273-9 (2004)\n",
      "            DOI:10.1128/JB.186.21.7273-7279.2004\n",
      "REFERENCE   PMID:15695810\n",
      "  AUTHORS   Breazeale SD, Ribeiro AA, McClerren AL, Raetz CR.\n",
      "  TITLE     A formyltransferase required for polymyxin resistance in Escherichia coli and the modification of lipid A with 4-Amino-4-deoxy-L-arabinose. Identification and function oF UDP-4-deoxy-4-formamido-L-arabinose.\n",
      "  JOURNAL   J Biol Chem 280:14154-67 (2005)\n",
      "            DOI:10.1074/jbc.M414265200\n",
      "REFERENCE   PMID:15809294\n",
      "  AUTHORS   Williams GJ, Breazeale SD, Raetz CR, Naismith JH.\n",
      "  TITLE     Structure and function of both domains of ArnA, a dual function decarboxylase and a formyltransferase, involved in 4-amino-4-deoxy-L-arabinose biosynthesis.\n",
      "  JOURNAL   J Biol Chem 280:23000-8 (2005)\n",
      "            DOI:10.1074/jbc.M501534200\n",
      "REFERENCE   PMID:16237198\n",
      "  AUTHORS   Maliekal P, Vertommen D, Delpierre G, Van Schaftingen E\n",
      "  TITLE     Identification of the sequence encoding N-acetylneuraminate-9-phosphate phosphatase.\n",
      "  JOURNAL   Glycobiology 16:165-72 (2006)\n",
      "            DOI:10.1093/glycob/cwj050\n",
      "REFERENCE   PMID:16277754\n",
      "  AUTHORS   Campbell CT, Yarema KJ\n",
      "  TITLE     Large-scale approaches for glycobiology.\n",
      "  JOURNAL   Genome Biol 6:236 (2005)\n",
      "            DOI:10.1186/gb-2005-6-11-236\n",
      "REFERENCE   PMID:16452451\n",
      "  AUTHORS   Uehara T, Suefuji K, Jaeger T, Mayer C, Park JT\n",
      "  TITLE     MurQ Etherase is required by Escherichia coli in order to metabolize anhydro-N-acetylmuramic acid obtained either from the environment or from its own cell wall.\n",
      "  JOURNAL   J Bacteriol 188:1660-2 (2006)\n",
      "            DOI:10.1128/JB.188.4.1660-1662.2006\n",
      "REFERENCE   PMID:16995900\n",
      "  AUTHORS   Lanot A, Hodge D, Jackson RG, George GL, Elias L, Lim EK, Vaistij FE, Bowles DJ\n",
      "  TITLE     The glucosyltransferase UGT72E2 is responsible for monolignol 4-O-glucoside production in Arabidopsis thaliana.\n",
      "  JOURNAL   Plant J 48:286-95 (2006)\n",
      "            DOI:10.1111/j.1365-313X.2006.02872.x\n",
      "REFERENCE   PMID:17085508\n",
      "  AUTHORS   Muller R, Morant M, Jarmer H, Nilsson L, Nielsen TH\n",
      "  TITLE     Genome-wide analysis of the Arabidopsis leaf transcriptome reveals interaction of phosphate and sugar metabolism.\n",
      "  JOURNAL   Plant Physiol 143:156-71 (2007)\n",
      "            DOI:10.1104/pp.106.090167\n",
      "REFERENCE   PMID:17190829\n",
      "  AUTHORS   Oka T, Nemoto T, Jigami Y\n",
      "  TITLE     Functional analysis of Arabidopsis thaliana RHM2/MUM4, a multidomain protein involved in UDP-D-glucose to UDP-L-rhamnose conversion.\n",
      "  JOURNAL   J Biol Chem 282:5389-403 (2007)\n",
      "            DOI:10.1074/jbc.M610196200\n",
      "REFERENCE   PMID:17827295\n",
      "  AUTHORS   Greenberg DE, Porcella SF, Zelazny AM, Virtaneva K, Sturdevant DE, Kupko JJ 3rd, Barbian KD, Babar A, Dorward DW, Holland SM.\n",
      "  TITLE     Genome sequence analysis of the emerging human pathogenic acetic acid bacterium Granulibacter bethesdensis.\n",
      "  JOURNAL   J Bacteriol 189:8727-36 (2007)\n",
      "            DOI:10.1128/JB.00793-07\n",
      "REFERENCE   PMID:18199744\n",
      "  AUTHORS   Kotake T, Hojo S, Tajima N, Matsuoka K, Koyama T, Tsumuraya Y\n",
      "  TITLE     A bifunctional enzyme with L-fucokinase and GDP-L-fucose pyrophosphorylase activities salvages free L-fucose in Arabidopsis.\n",
      "  JOURNAL   J Biol Chem 283:8125-35 (2008)\n",
      "            DOI:10.1074/jbc.M710078200\n",
      "REFERENCE   PMID:18304669\n",
      "  AUTHORS   Vorholter FJ, Schneiker S, Goesmann A, Krause L, Bekel T, Kaiser O, Linke B, Patschkowski T, Ruckert C, Schmid J, Sidhu VK, Sieber V, Tauch A, Watt SA, Weisshaar B, Becker A, Niehaus K, Puhler A.\n",
      "  TITLE     The genome of Xanthomonas campestris pv. campestris B100 and its use for the reconstruction of metabolic pathways involved in xanthan biosynthesis.\n",
      "  JOURNAL   J Biotechnol 134:33-45 (2008)\n",
      "            DOI:10.1016/j.jbiotec.2007.12.013\n",
      "REFERENCE   PMID:11479276\n",
      "  AUTHORS   Albermann C, Piepersberg W\n",
      "  TITLE     Expression and identification of the RfbE protein from Vibrio cholerae O1 and its use for the enzymatic synthesis of GDP-D-perosamine.\n",
      "  JOURNAL   Glycobiology 11:655-61 (2001)\n",
      "            DOI:10.1093/glycob/11.8.655\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00040  Pentose and glucuronate interconversions\n",
      "            pae00051  Fructose and mannose metabolism\n",
      "            pae00052  Galactose metabolism\n",
      "            pae00053  Ascorbate and aldarate metabolism\n",
      "            pae00500  Starch and sucrose metabolism\n",
      "            pae00523  Polyketide sugar unit biosynthesis\n",
      "            pae00525  Acarbose and validamycin biosynthesis\n",
      "            pae00540  Lipopolysaccharide biosynthesis\n",
      "            pae00550  Peptidoglycan biosynthesis\n",
      "KO_PATHWAY  ko00520\n",
      "///\n",
      "path:pae00521\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00521                    Pathway\n",
      "NAME        Streptomycin biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Streptomycin is an aminocyclitol-aminoglycoside antibiotic produced by Streptomyces griseus. Streptomycin consists of aminocyclitol (streptidine), 6-deoxyhexose (streptose), and N-methyl-L-glucosamine moieties, which are formed by independent biosynthetic pathways. All of the moieties are derived from D-glucose. The streptidine moiety is synthesized via myo-inositol, which is then oxidized at C-1 and transaminated to give scyllo-inosamine. After phosphorylation, the compound is transamidinated by arginine. The same procedure is repeated at the C-3 position. The streptose moiety is made from D-glucose via a dTDP-glucose pathway. The exact biosynthetic route for the N-methyl-L-glucosamine moiety is unknown, though the biosynthetic gene cluster have been proposed.\n",
      "CLASS       Metabolism; Biosynthesis of other secondary metabolites\n",
      "PATHWAY_MAP pae00521  Streptomycin biosynthesis\n",
      "MODULE      pae_M00793  dTDP-L-rhamnose biosynthesis [PATH:pae00521]\n",
      "DBLINKS     GO: 0019872\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3193  glk; glucokinase [KO:K00845] [EC:2.7.1.2]\n",
      "line in gene PA3193  glk; glucokinase [KO:K00845] [EC:2.7.1.2]\n",
      "            PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "line in gene PA5322  algC; phosphomannomutase [KO:K15778] [EC:5.4.2.8 5.4.2.2]\n",
      "            PA3818  type III secretion system regulator SuhB [KO:K01092] [EC:3.1.3.25]\n",
      "line in gene PA3818  type III secretion system regulator SuhB [KO:K01092] [EC:3.1.3.25]\n",
      "            PA5163  rmlA; glucose-1-phosphate thymidylyltransferase [KO:K00973] [EC:2.7.7.24]\n",
      "line in gene PA5163  rmlA; glucose-1-phosphate thymidylyltransferase [KO:K00973] [EC:2.7.7.24]\n",
      "            PA5161  rmlB; dTDP-D-glucose 4,6-dehydratase [KO:K01710] [EC:4.2.1.46]\n",
      "line in gene PA5161  rmlB; dTDP-D-glucose 4,6-dehydratase [KO:K01710] [EC:4.2.1.46]\n",
      "            PA5164  rmlC; dTDP-4-dehydrorhamnose 3,5-epimerase [KO:K01790] [EC:5.1.3.13]\n",
      "line in gene PA5164  rmlC; dTDP-4-dehydrorhamnose 3,5-epimerase [KO:K01790] [EC:5.1.3.13]\n",
      "            PA5162  rmlD; dTDP-4-dehydrorhamnose reductase [KO:K00067] [EC:1.1.1.133]\n",
      "line in gene PA5162  rmlD; dTDP-4-dehydrorhamnose reductase [KO:K00067] [EC:1.1.1.133]\n",
      "            PA4069  hypothetical protein [KO:K00067] [EC:1.1.1.133]\n",
      "line in gene PA4069  hypothetical protein [KO:K00067] [EC:1.1.1.133]\n",
      "COMPOUND    C00031  D-Glucose\n",
      "            C00092  D-Glucose 6-phosphate\n",
      "            C00103  D-Glucose 1-phosphate\n",
      "            C00137  myo-Inositol\n",
      "            C00413  Streptomycin\n",
      "            C00688  dTDP-4-dehydro-beta-L-rhamnose\n",
      "            C00691  2,4,6/3,5-Pentahydroxycyclohexanone\n",
      "            C00842  dTDP-glucose\n",
      "            C01121  Streptidine 6-phosphate\n",
      "            C01138  Streptomycin 6-phosphate\n",
      "            C01214  1-Amino-1-deoxy-scyllo-inositol\n",
      "            C01221  Dihydrostreptomycin 6-phosphate\n",
      "            C01283  1-Amino-1-deoxy-scyllo-inositol 4-phosphate\n",
      "            C01294  1-Guanidino-1-deoxy-scyllo-inositol 4-phosphate\n",
      "            C01298  1D-1-Guanidino-3-amino-1,3-dideoxy-scyllo-inositol\n",
      "            C03319  dTDP-L-rhamnose\n",
      "            C03442  dTDP-L-dihydrostreptose\n",
      "            C04006  1D-myo-Inositol 3-phosphate\n",
      "            C04280  1-Guanidino-1-deoxy-scyllo-inositol\n",
      "            C04673  1D-1-Guanidino-1-deoxy-3-dehydro-scyllo-inositol\n",
      "            C04767  O-(1->4)-alpha-L-Dihydrostreptosyl-streptidine 6-phosphate\n",
      "            C06592  NDP-N-methyl-L-glucosamine\n",
      "            C11351  N1-Amidinostreptamine 6-phosphate\n",
      "            C11907  dTDP-4-oxo-6-deoxy-D-glucose\n",
      "REFERENCE   PMID:19362651\n",
      "  AUTHORS   Wehmeier UF, Piepersberg W\n",
      "  TITLE     Enzymology of aminoglycoside biosynthesis-deduction from gene clusters.\n",
      "  JOURNAL   Methods Enzymol 459:459-91 (2009)\n",
      "            DOI:10.1016/S0076-6879(09)04619-9\n",
      "REFERENCE   PMID:17390001\n",
      "  AUTHORS   Flatt PM, Mahmud T\n",
      "  TITLE     Biosynthesis of aminocyclitol-aminoglycoside antibiotics and related compounds.\n",
      "  JOURNAL   Nat Prod Rep 24:358-92 (2007)\n",
      "            DOI:10.1039/b603816f\n",
      "REFERENCE   PMID:1698764\n",
      "  AUTHORS   Walker JB\n",
      "  TITLE     Possible evolutionary relationships between streptomycin and bluensomycin biosynthetic pathways: detection of novel inositol kinase and O-carbamoyltransferase activities.\n",
      "  JOURNAL   J Bacteriol 172:5844-51 (1990)\n",
      "            DOI:10.1128/JB.172.10.5844-5851.1990\n",
      "REFERENCE   PMID:2995326\n",
      "  AUTHORS   Ohnuki T, Imanaka T, Aiba S\n",
      "  TITLE     Self-cloning in Streptomyces griseus of an str gene cluster for streptomycin biosynthesis and streptomycin resistance.\n",
      "  JOURNAL   J Bacteriol 164:85-94 (1985)\n",
      "            DOI:10.1128/JB.164.1.85-94.1985\n",
      "REFERENCE   PMID:9238101\n",
      "  AUTHORS   Ahlert J, Distler J, Mansouri K, Piepersberg W\n",
      "  TITLE     Identification of stsC, the gene encoding the L-glutamine:scyllo-inosose aminotransferase from streptomycin-producing Streptomycetes.\n",
      "  JOURNAL   Arch Microbiol 168:102-13 (1997)\n",
      "            DOI:10.1007/s002030050475\n",
      "REFERENCE   PMID:1377151\n",
      "  AUTHORS   Distler J, Mansouri K, Mayer G, Stockmann M, Piepersberg W\n",
      "  TITLE     Streptomycin biosynthesis and its regulation in Streptomycetes.\n",
      "  JOURNAL   Gene 115:105-11 (1992)\n",
      "            DOI:10.1016/0378-1119(92)90547-3\n",
      "REFERENCE   PMID:1654502\n",
      "  AUTHORS   Mansouri K, Piepersberg W.\n",
      "  TITLE     Genetics of streptomycin production in Streptomyces griseus: nucleotide sequence of five genes, strFGHIK, including a phosphatase gene.\n",
      "  JOURNAL   Mol Gen Genet 228:459-69 (1991)\n",
      "            DOI:10.1007/BF00260640\n",
      "REFERENCE   PMID:9922132\n",
      "  AUTHORS   Fritsche E, Bergner A, Humm A, Piepersberg W, Huber R\n",
      "  TITLE     Crystal structure of L-arginine:inosamine-phosphate amidinotransferase StrB1 from Streptomyces griseus: an enzyme involved in streptomycin biosynthesis.\n",
      "  JOURNAL   Biochemistry 37:17664-72 (1998)\n",
      "            DOI:10.1021/bi981949p\n",
      "REL_PATHWAY pae00523  Polyketide sugar unit biosynthesis\n",
      "KO_PATHWAY  ko00521\n",
      "///\n",
      "path:pae00523\n",
      "ENTRY       pae00523                    Pathway\n",
      "NAME        Polyketide sugar unit biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of terpenoids and polyketides\n",
      "PATHWAY_MAP pae00523  Polyketide sugar unit biosynthesis\n",
      "MODULE      pae_M00793  dTDP-L-rhamnose biosynthesis [PATH:pae00523]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA5163  rmlA; glucose-1-phosphate thymidylyltransferase [KO:K00973] [EC:2.7.7.24]\n",
      "line in gene PA5163  rmlA; glucose-1-phosphate thymidylyltransferase [KO:K00973] [EC:2.7.7.24]\n",
      "            PA5161  rmlB; dTDP-D-glucose 4,6-dehydratase [KO:K01710] [EC:4.2.1.46]\n",
      "line in gene PA5161  rmlB; dTDP-D-glucose 4,6-dehydratase [KO:K01710] [EC:4.2.1.46]\n",
      "            PA5164  rmlC; dTDP-4-dehydrorhamnose 3,5-epimerase [KO:K01790] [EC:5.1.3.13]\n",
      "line in gene PA5164  rmlC; dTDP-4-dehydrorhamnose 3,5-epimerase [KO:K01790] [EC:5.1.3.13]\n",
      "            PA5162  rmlD; dTDP-4-dehydrorhamnose reductase [KO:K00067] [EC:1.1.1.133]\n",
      "line in gene PA5162  rmlD; dTDP-4-dehydrorhamnose reductase [KO:K00067] [EC:1.1.1.133]\n",
      "            PA4069  hypothetical protein [KO:K00067] [EC:1.1.1.133]\n",
      "line in gene PA4069  hypothetical protein [KO:K00067] [EC:1.1.1.133]\n",
      "COMPOUND    C00103  D-Glucose 1-phosphate\n",
      "            C00688  dTDP-4-dehydro-beta-L-rhamnose\n",
      "            C00842  dTDP-glucose\n",
      "            C03187  dTDP-6-deoxy-beta-L-talose\n",
      "            C03319  dTDP-L-rhamnose\n",
      "            C04268  dTDP-4-amino-4,6-dideoxy-D-glucose\n",
      "            C06017  dTDP-D-glucuronate\n",
      "            C06620  dTDP-3,4-dioxo-2,6-dideoxy-D-glucose\n",
      "            C07277  dTDP-D-fucose\n",
      "            C11447  dTDP-4-dimethylamino-4,6-dideoxy-5-C-methyl-D-allose\n",
      "            C11460  dTDP-4-oxo-5-C-methyl-L-rhamnose\n",
      "            C11461  dTDP-4-amino-4,6-dideoxy-5-C-methyl-D-allose\n",
      "            C11907  dTDP-4-oxo-6-deoxy-D-glucose\n",
      "            C11908  3,6-Dideoxy-3-oxo-dTDP-D-glucose\n",
      "            C11909  dTDP-3-oxo-4,6-dideoxy-D-glucose\n",
      "            C11910  dTDP-3-amino-3,4,6-trideoxy-D-glucose\n",
      "            C11911  dTDP-D-desosamine\n",
      "            C11912  dTDP-6-deoxy-D-allose\n",
      "            C11915  dTDP-3-methyl-4-oxo-2,6-dideoxy-L-allose\n",
      "            C11916  dTDP-L-mycarose\n",
      "            C11918  Maltol\n",
      "            C11919  dTDP-L-megosamine\n",
      "            C11920  dTDP-L-olivose\n",
      "            C11921  dTDP-L-oleandrose\n",
      "            C11922  dTDP-4-oxo-2,6-dideoxy-D-glucose\n",
      "            C11925  dTDP-3-amino-3,6-dideoxy-D-glucose\n",
      "            C11926  dTDP-4-oxo-6-deoxy-D-allose\n",
      "            C11927  dTDP-4-oxo-2,6-dideoxy-L-mannose\n",
      "            C11928  dTDP-4-oxo-2,6-dideoxy-D-allose\n",
      "            C11933  dTDP-D-mycaminose\n",
      "            C12318  dTDP-3-amino-2,3,6-trideoxy-4-keto-D-glucose\n",
      "            C12481  dTDP-5-dimethyl-L-lyxose\n",
      "            C18029  dTDP-3-oxo-2,6-dideoxy-D-glucose\n",
      "            C18030  dTDP-2,6-dideoxy-D-kanosamine\n",
      "            C18031  dTDP-D-angolosamine\n",
      "            C18032  dTDP-4-oxo-2,3,6-trideoxy-D-glucose\n",
      "            C18033  dTDP-4-amino-2,3,4,6-tetradeoxy-D-glucose\n",
      "            C18034  dTDP-D-forosamine\n",
      "            C18035  dTDP-3-methyl-4-oxo-2,6-dideoxy-D-glucose\n",
      "            C19947  dTDP-3-amino-3,6-dideoxy-alpha-D-galactopyranose\n",
      "            C19948  dTDP-3-dimethylamino-3,6-dideoxy-alpha-D-galactopyranose\n",
      "            C19960  dTDP-3-dehydro-6-deoxy-alpha-D-galactopyranose\n",
      "            C20156  dTDP-3-acetamido-3,6-dideoxy-alpha-D-galactopyranose\n",
      "            C20682  dTDP-alpha-D-fucofuranose\n",
      "            C21071  dTDP-3-O-methyl-4-oxo-2,6-dideoxy-L-mannose\n",
      "            C21073  dTDP-3-N,N-dimethylamino-2,3,6-trideoxy-4-keto-D-glucose\n",
      "            C21074  dTDP-3-N,N-dimethylamino-4-oxo-2,3,6-trideoxy-L-allose\n",
      "            C21348  dTDP-4-hydroxyamino-4,6-dideoxy-alpha-D-glucose\n",
      "            C21349  dTDP-4-deoxy-4-thio-alpha-D-digitoxose\n",
      "            C21350  dTDP-3-O-methyl-beta-L-rhamnose\n",
      "            C21351  dTDP-4-oxo-alpha-D-xylose\n",
      "            C21352  dTDP-4-oxo-2-deoxy-alpha-D-pentos-2-ene\n",
      "            C21353  dTDP-4-oxo-2-deoxy-beta-L-xylose\n",
      "            C21354  dTDP-4-amino-2,4-dideoxy-beta-L-xylose\n",
      "            C21355  dTDP-4-methylamino-2,4-dideoxy-beta-L-xylose\n",
      "            C21356  dTDP-4-ethylamino-3-O-methyl-2,4-dideoxy-L-threo-pentopyranose\n",
      "            C21362  dTDP-3-methyl-4-oxo-alpha-D-ribopyranose\n",
      "            C21363  dTDP-madurosamine\n",
      "REFERENCE   PMID:11336289\n",
      "  AUTHORS   Rawlings BJ.\n",
      "  TITLE     Type I polyketide biosynthesis in bacteria (Part A--erythromycin biosynthesis).\n",
      "  JOURNAL   Nat Prod Rep 18:190-227 (2001)\n",
      "            DOI:10.1039/b009329g\n",
      "REFERENCE   PMID:11476481\n",
      "  AUTHORS   Rawlings BJ.\n",
      "  TITLE     Type I polyketide biosynthesis in bacteria (part B).\n",
      "  JOURNAL   Nat Prod Rep 18:231-81 (2001)\n",
      "            DOI:10.1039/B100191O\n",
      "REFERENCE   PMID:11451669\n",
      "  AUTHORS   Wohlert S, Lomovskaya N, Kulowski K, Fonstein L, Occi JL, Gewain KM, MacNeil DJ, Hutchinson CR\n",
      "  TITLE     Insights about the biosynthesis of the avermectin deoxysugar L-oleandrose through heterologous expression of Streptomyces avermitilis deoxysugar genes in Streptomyces lividans.\n",
      "  JOURNAL   Chem Biol 8:681-700 (2001)\n",
      "            DOI:10.1016/S1074-5521(01)00043-6\n",
      "REFERENCE   PMID:10658660\n",
      "  AUTHORS   Bate N, Butler AR, Smith IP, Cundliffe E\n",
      "  TITLE     The mycarose-biosynthetic genes of Streptomyces fradiae, producer of tylosin.\n",
      "  JOURNAL   Microbiology 146 ( Pt 1):139-46 (2000)\n",
      "            DOI:10.1099/00221287-146-1-139\n",
      "REFERENCE   PMID:11553351\n",
      "  AUTHORS   Amann S, Drager G, Rupprath C, Kirschning A, Elling L\n",
      "  TITLE     (Chemo)enzymatic synthesis of dTDP-activated 2,6-dideoxysugars as building blocks of polyketide antibiotics.\n",
      "  JOURNAL   Carbohydr Res 335:23-32 (2001)\n",
      "            DOI:10.1016/S0008-6215(01)00195-1\n",
      "REFERENCE   PMID:17460661\n",
      "  AUTHORS   Thibodeaux CJ, Melancon CE, Liu HW\n",
      "  TITLE     Unusual sugar biosynthesis and natural product glycodiversification.\n",
      "  JOURNAL   Nature 446:1008-16 (2007)\n",
      "            DOI:10.1038/nature05814\n",
      "REFERENCE   PMID:10972798\n",
      "  AUTHORS   Volchegursky Y, Hu Z, Katz L, McDaniel R\n",
      "  TITLE     Biosynthesis of the anti-parasitic agent megalomicin: transformation of erythromycin to megalomicin in Saccharopolyspora erythraea.\n",
      "  JOURNAL   Mol Microbiol 37:752-62 (2000)\n",
      "            DOI:10.1046/j.1365-2958.2000.02059.x\n",
      "REFERENCE   \n",
      "  AUTHORS   Mironov VA, Sergienko OV, Nastasiak IN, Danilenko VN.\n",
      "  TITLE     Biogenesis and regulation of biosynthesis of erythromycins in Saccharopolyspora.\n",
      "  JOURNAL   Appl Biochem Microbiol 40:531-541 (2004)\n",
      "            DOI:10.1023/B:ABIM.0000046985.66328.7a\n",
      "REFERENCE   PMID:16514445\n",
      "  AUTHORS   Tello M, Jakimowicz P, Errey JC, Freel Meyers CL, Walsh CT, Buttner MJ, Lawson DM, Field RA\n",
      "  TITLE     Characterisation of Streptomyces spheroides NovW and revision of its functional assignment to a dTDP-6-deoxy-D-xylo-4-hexulose 3-epimerase.\n",
      "  JOURNAL   Chem Commun (Camb) 1079-81 (2006)\n",
      "            DOI:10.1039/b515763c\n",
      "REFERENCE   PMID:15984840\n",
      "  AUTHORS   Takahashi H, Liu YN, Chen H, Liu HW\n",
      "  TITLE     Biosynthesis of TDP-l-mycarose: the specificity of a single enzyme governs the outcome of the pathway.\n",
      "  JOURNAL   J Am Chem Soc 127:9340-1 (2005)\n",
      "            DOI:10.1021/ja051409x\n",
      "REFERENCE   PMID:10770761\n",
      "  AUTHORS   Aguirrezabalaga I, Olano C, Allende N, Rodriguez L, Brana AF, Mendez C, Salas JA\n",
      "  TITLE     Identification and expression of genes involved in biosynthesis of L-oleandrose and its intermediate L-olivose in the oleandomycin producer Streptomyces antibioticus.\n",
      "  JOURNAL   Antimicrob Agents Chemother 44:1266-75 (2000)\n",
      "            DOI:10.1128/AAC.44.5.1266-1275.2000\n",
      "REFERENCE   PMID:11514520\n",
      "  AUTHORS   Rodriguez L, Rodriguez D, Olano C, Brana AF, Mendez C, Salas JA\n",
      "  TITLE     Functional analysis of OleY L-oleandrosyl 3-O-methyltransferase of the oleandomycin biosynthetic pathway in Streptomyces antibioticus.\n",
      "  JOURNAL   J Bacteriol 183:5358-63 (2001)\n",
      "            DOI:10.1128/JB.183.18.5358-5363.2001\n",
      "REFERENCE   PMID:20418422\n",
      "  AUTHORS   Useglio M, Peiru S, Rodriguez E, Labadie GR, Carney JR, Gramajo H\n",
      "  TITLE     TDP-L-megosamine biosynthesis pathway elucidation and megalomicin a production in Escherichia coli.\n",
      "  JOURNAL   Appl Environ Microbiol 76:3869-77 (2010)\n",
      "            DOI:10.1128/AEM.03083-09\n",
      "REFERENCE   PMID:12183628\n",
      "  AUTHORS   Liu W, Christenson SD, Standage S, Shen B\n",
      "  TITLE     Biosynthesis of the enediyne antitumor antibiotic C-1027.\n",
      "  JOURNAL   Science 297:1170-3 (2002)\n",
      "            DOI:10.1126/science.1072110\n",
      "REFERENCE   PMID:17918933\n",
      "  AUTHORS   Van Lanen SG, Oh TJ, Liu W, Wendt-Pienkowski E, Shen B\n",
      "  TITLE     Characterization of the maduropeptin biosynthetic gene cluster from Actinomadura madurae ATCC 39144 supporting a unifying paradigm for enediyne biosynthesis.\n",
      "  JOURNAL   J Am Chem Soc 129:13082-94 (2007)\n",
      "            DOI:10.1021/ja073275o\n",
      "REFERENCE   PMID:16873021\n",
      "  AUTHORS   Gao Q, Zhang C, Blanchard S, Thorson JS\n",
      "  TITLE     Deciphering indolocarbazole and enediyne aminodideoxypentose biosynthesis through comparative genomics: insights from the AT2433 biosynthetic locus.\n",
      "  JOURNAL   Chem Biol 13:733-43 (2006)\n",
      "            DOI:10.1016/j.chembiol.2006.05.009\n",
      "REFERENCE   PMID:18807197\n",
      "  AUTHORS   Simkhada D, Oh TJ, Kim EM, Yoo JC, Sohng JK\n",
      "  TITLE     Cloning and characterization of CalS7 from Micromonospora echinospora sp. calichensis as a glucose-1-phosphate nucleotidyltransferase.\n",
      "  JOURNAL   Biotechnol Lett 31:147-53 (2009)\n",
      "            DOI:10.1007/s10529-008-9844-9\n",
      "REFERENCE   PMID:24911465\n",
      "  AUTHORS   Singh S, Peltier-Pain P, Tonelli M, Thorson JS\n",
      "  TITLE     A general NMR-based strategy for the in situ characterization of sugar-nucleotide-dependent biosynthetic pathways.\n",
      "  JOURNAL   Org Lett 16:3220-3 (2014)\n",
      "            DOI:10.1021/ol501241a\n",
      "REFERENCE   PMID:23662776\n",
      "  AUTHORS   Singh S, Chang A, Helmich KE, Bingman CA, Wrobel RL, Beebe ET, Makino S, Aceti DJ, Dyer K, Hura GL, Sunkara M, Morris AJ, Phillips GN Jr, Thorson JS\n",
      "  TITLE     Structural and functional characterization of CalS11, a TDP-rhamnose 3'-O-methyltransferase involved in calicheamicin biosynthesis.\n",
      "  JOURNAL   ACS Chem Biol 8:1632-9 (2013)\n",
      "            DOI:10.1021/cb400068k\n",
      "REFERENCE   PMID:20336235\n",
      "  AUTHORS   Liang ZX\n",
      "  TITLE     Complexity and simplicity in the biosynthesis of enediyne natural products.\n",
      "  JOURNAL   Nat Prod Rep 27:499-528 (2010)\n",
      "            DOI:10.1039/b908165h\n",
      "REFERENCE   PMID:26240141\n",
      "  AUTHORS   Singh S, Michalska K, Bigelow L, Endres M, Kharel MK, Babnigg G, Yennamalli RM, Bingman CA, Joachimiak A, Thorson JS, Phillips GN Jr\n",
      "  TITLE     Structural Characterization of CalS8, a TDP-alpha-D-Glucose Dehydrogenase Involved in Calicheamicin Aminodideoxypentose Biosynthesis.\n",
      "  JOURNAL   J Biol Chem 290:26249-58 (2015)\n",
      "            DOI:10.1074/jbc.M115.673459\n",
      "REFERENCE   PMID:26289554\n",
      "  AUTHORS   Peltier-Pain P, Singh S, Thorson JS\n",
      "  TITLE     Characterization of Early Enzymes Involved in TDP-Aminodideoxypentose Biosynthesis en Route to Indolocarbazole AT2433.\n",
      "  JOURNAL   Chembiochem 16:2141-6 (2015)\n",
      "            DOI:10.1002/cbic.201500365\n",
      "REFERENCE   PMID:26061967\n",
      "  AUTHORS   Singh S, Kim Y, Wang F, Bigelow L, Endres M, Kharel MK, Babnigg G, Bingman CA, Joachimiak A, Thorson JS, Phillips GN Jr\n",
      "  TITLE     Structural characterization of AtmS13, a putative sugar aminotransferase involved in indolocarbazole AT2433 aminopentose biosynthesis.\n",
      "  JOURNAL   Proteins 83:1547-54 (2015)\n",
      "            DOI:10.1002/prot.24844\n",
      "REL_PATHWAY pae00401  Novobiocin biosynthesis\n",
      "            pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "            pae00521  Streptomycin biosynthesis\n",
      "KO_PATHWAY  ko00523\n",
      "///\n",
      "path:pae00525\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00525                    Pathway\n",
      "NAME        Acarbose and validamycin biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Biosynthesis of other secondary metabolites\n",
      "PATHWAY_MAP pae00525  Acarbose and validamycin biosynthesis\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA5163  rmlA; glucose-1-phosphate thymidylyltransferase [KO:K00973] [EC:2.7.7.24]\n",
      "line in gene PA5163  rmlA; glucose-1-phosphate thymidylyltransferase [KO:K00973] [EC:2.7.7.24]\n",
      "            PA5161  rmlB; dTDP-D-glucose 4,6-dehydratase [KO:K01710] [EC:4.2.1.46]\n",
      "line in gene PA5161  rmlB; dTDP-D-glucose 4,6-dehydratase [KO:K01710] [EC:4.2.1.46]\n",
      "COMPOUND    C00103  D-Glucose 1-phosphate\n",
      "            C00842  dTDP-glucose\n",
      "            C04268  dTDP-4-amino-4,6-dideoxy-D-glucose\n",
      "            C06802  Acarbose\n",
      "            C11907  dTDP-4-oxo-6-deoxy-D-glucose\n",
      "            C12112  Validamycin A\n",
      "            C17691  2-epi-5-epi-Valiolone\n",
      "            C17692  2-epi-5-epi-Valiolone 7-phosphate\n",
      "            C17693  5-epi-Valiolone 7-phosphate\n",
      "            C17694  5-epi-Valiolone\n",
      "            C17695  2-Keto-5-epi-valiolone\n",
      "            C17696  Valienone\n",
      "            C17697  Valienone 7-phosphate\n",
      "            C17698  Salbostatin\n",
      "            C17699  Cetoniacytone A\n",
      "            C17700  Validoxylamine A\n",
      "            C20956  alpha-D-Sedoheptulopyranose 7-phosphate\n",
      "            C21156  Acarbose 7IV-phosphate\n",
      "            C21204  5-epi-Valiolol 7-phosphate\n",
      "            C21205  1-epi-Valienol 7-phosphate\n",
      "            C21206  1,7-Diphospho-1-epi-valienol\n",
      "            C21207  NDP-1-epi-valienol 7-phosphate\n",
      "            C21208  dTDP-acarviose-7-phosphate\n",
      "            C21209  1-epi-Valienol 1-phosphate\n",
      "            C21210  NDP-1-epi-valienol\n",
      "            C21211  Salbostatin 6'-phosphate\n",
      "            C21216  Validamine 7-phosphate\n",
      "            C21218  Validone 7-phosphate\n",
      "            C21219  Validoxylamine A 7'-phosphate\n",
      "            C21247  2-Amino-5-epi-valiolone\n",
      "            C21248  2-Aminovalienone\n",
      "            C21249  Cetoniacytone B\n",
      "            C21293  Validamycin B\n",
      "            C21367  Valienol 1-phosphate\n",
      "            C21368  GDP-valienol\n",
      "            C21415  Valienol 7-phosphate\n",
      "REFERENCE   PMID:17390001\n",
      "  AUTHORS   Flatt PM, Mahmud T\n",
      "  TITLE     Biosynthesis of aminocyclitol-aminoglycoside antibiotics and related compounds.\n",
      "  JOURNAL   Nat Prod Rep 24:358-92 (2007)\n",
      "            DOI:10.1039/b603816f\n",
      "REFERENCE   PMID:17661520\n",
      "  AUTHORS   Mahmud T, Flatt PM, Wu X\n",
      "  TITLE     Biosynthesis of unusual aminocyclitol-containing natural products.\n",
      "  JOURNAL   J Nat Prod 70:1384-91 (2007)\n",
      "            DOI:10.1021/np070210q\n",
      "REFERENCE   PMID:22944206\n",
      "  AUTHORS   Wendler S, Hurtgen D, Kalinowski J, Klein A, Niehaus K, Schulte F, Schwientek P, Wehlmann H, Wehmeier UF, Puhler A\n",
      "  TITLE     The cytosolic and extracellular proteomes of Actinoplanes sp. SE50/110 led to the identification of gene products involved in acarbose metabolism.\n",
      "  JOURNAL   J Biotechnol 167:178-89 (2013)\n",
      "            DOI:10.1016/j.jbiotec.2012.08.011\n",
      "REFERENCE   PMID:22443545\n",
      "  AUTHORS   Schwientek P, Szczepanowski R, Ruckert C, Kalinowski J, Klein A, Selber K, Wehmeier UF, Stoye J, Puhler A\n",
      "  TITLE     The complete genome sequence of the acarbose producer Actinoplanes sp. SE50/110.\n",
      "  JOURNAL   BMC Genomics 13:112 (2012)\n",
      "            DOI:10.1186/1471-2164-13-112\n",
      "REFERENCE   PMID:17003911\n",
      "  AUTHORS   Deng Z, Bai L\n",
      "  TITLE     Antibiotic biosynthetic pathways and pathway engineering--a growing research field in China.\n",
      "  JOURNAL   Nat Prod Rep 23:811-27 (2006)\n",
      "            DOI:10.1039/b611140h\n",
      "REFERENCE   PMID:21766819\n",
      "  AUTHORS   Asamizu S, Yang J, Almabruk KH, Mahmud T\n",
      "  TITLE     Pseudoglycosyltransferase catalyzes nonglycosidic C-N coupling in validamycin a biosynthesis.\n",
      "  JOURNAL   J Am Chem Soc 133:12124-35 (2011)\n",
      "            DOI:10.1021/ja203574u\n",
      "REFERENCE   PMID:18648803\n",
      "  AUTHORS   Choi WS, Wu X, Choeng YH, Mahmud T, Jeong BC, Lee SH, Chang YK, Kim CJ, Hong SK\n",
      "  TITLE     Genetic organization of the putative salbostatin biosynthetic gene cluster including the 2-epi-5-epi-valiolone synthase gene in Streptomyces albus ATCC 21838.\n",
      "  JOURNAL   Appl Microbiol Biotechnol 80:637-45 (2008)\n",
      "            DOI:10.1007/s00253-008-1591-2\n",
      "REFERENCE   PMID:19101977\n",
      "  AUTHORS   Wu X, Flatt PM, Xu H, Mahmud T\n",
      "  TITLE     Biosynthetic gene cluster of cetoniacytone A, an unusual aminocyclitol from the endosymbiotic Bacterium Actinomyces sp. Lu 9419.\n",
      "  JOURNAL   Chembiochem 10:304-14 (2009)\n",
      "            DOI:10.1002/cbic.200800527\n",
      "REL_PATHWAY pae00030  Pentose phosphate pathway\n",
      "            pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "KO_PATHWAY  ko00525\n",
      "///\n",
      "path:pae00540\n",
      "ENTRY       pae00540                    Pathway\n",
      "NAME        Lipopolysaccharide biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Lipopolysaccharide (LPS) is the major component of the outer membrane of Gram-negative bacteria consisting of three domains: a hydrophobic anchor called lipid A, a non-repeating core oligosaccharide (OS) and a repetitive glycan polymer called O-antigen or O-polysaccharide (O-PS). The enzymes for the biosynthesis of lipid A are well conserved among Gram-negative bacterial species [MD:M00060 M00866]. The core OS connecting lipid A with O-antigen is divided into inner and outer parts. Five core types, K-12 and R1-R4, are identified in Escherichia coli, with the common inner part composed of Kdo (ketodeoxyoctulosonic acid) and Hep (heptose) residues and the outer part made of various sugar residues. O-antigen is the most variable part used for serotyping of pathogenic Escherichia coli, where about 170 types are known. All three domains are involved in toxicity, pathogenicity, antimicrobial resistance and other activities.\n",
      "CLASS       Metabolism; Glycan biosynthesis and metabolism\n",
      "PATHWAY_MAP pae00540  Lipopolysaccharide biosynthesis\n",
      "MODULE      pae_M00063  CMP-KDO biosynthesis [PATH:pae00540]\n",
      "            pae_M00064  ADP-L-glycero-D-manno-heptose biosynthesis [PATH:pae00540]\n",
      "DBLINKS     GO: 0009103\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3644  lpxA; acyl-[acyl-carrier-protein]--UDP-N-acetylglucosamine O-acyltransferase [KO:K00677] [EC:2.3.1.129]\n",
      "line in gene PA3644  lpxA; acyl-[acyl-carrier-protein]--UDP-N-acetylglucosamine O-acyltransferase [KO:K00677] [EC:2.3.1.129]\n",
      "            PA4406  lpxC; UDP-3-O-[3-hydroxymyristoyl] N-acetylglucosamine deacetylase [KO:K02535] [EC:3.5.1.108]\n",
      "line in gene PA4406  lpxC; UDP-3-O-[3-hydroxymyristoyl] N-acetylglucosamine deacetylase [KO:K02535] [EC:3.5.1.108]\n",
      "            PA3646  lpxD; UDP-3-O-acylglucosamine N-acyltransferase [KO:K02536] [EC:2.3.1.191]\n",
      "line in gene PA3646  lpxD; UDP-3-O-acylglucosamine N-acyltransferase [KO:K02536] [EC:2.3.1.191]\n",
      "            PA1792  UDP-2,3-diacylglucosamine hydrolase [KO:K03269] [EC:3.6.1.54]\n",
      "line in gene PA1792  UDP-2,3-diacylglucosamine hydrolase [KO:K03269] [EC:3.6.1.54]\n",
      "            PA3643  lpxB; lipid-A-disaccharide synthase [KO:K00748] [EC:2.4.1.182]\n",
      "line in gene PA3643  lpxB; lipid-A-disaccharide synthase [KO:K00748] [EC:2.4.1.182]\n",
      "            PA2981  lpxK; tetraacyldisaccharide 4'-kinase [KO:K00912] [EC:2.7.1.130]\n",
      "line in gene PA2981  lpxK; tetraacyldisaccharide 4'-kinase [KO:K00912] [EC:2.7.1.130]\n",
      "            PA4457  arabinose-5-phosphate isomerase KdsD [KO:K06041] [EC:5.3.1.13]\n",
      "line in gene PA4457  arabinose-5-phosphate isomerase KdsD [KO:K06041] [EC:5.3.1.13]\n",
      "            PA3636  kdsA; 2-dehydro-3-deoxyphosphooctonate aldolase [KO:K01627] [EC:2.5.1.55]\n",
      "line in gene PA3636  kdsA; 2-dehydro-3-deoxyphosphooctonate aldolase [KO:K01627] [EC:2.5.1.55]\n",
      "            PA4458  hypothetical protein [KO:K03270] [EC:3.1.3.45]\n",
      "line in gene PA4458  hypothetical protein [KO:K03270] [EC:3.1.3.45]\n",
      "            PA2979  kdsB; 3-deoxy-manno-octulosonate cytidylyltransferase [KO:K00979] [EC:2.7.7.38]\n",
      "line in gene PA2979  kdsB; 3-deoxy-manno-octulosonate cytidylyltransferase [KO:K00979] [EC:2.7.7.38]\n",
      "            PA4988  waaA; 3-deoxy-D-manno-octulosonic acid transferase [KO:K02527] [EC:2.4.99.12 2.4.99.13 2.4.99.14 2.4.99.15]\n",
      "line in gene PA4988  waaA; 3-deoxy-D-manno-octulosonic acid transferase [KO:K02527] [EC:2.4.99.12 2.4.99.13 2.4.99.14 2.4.99.15]\n",
      "            PA0011  lipid A biosynthesis lauroyl acyltransferase [KO:K02517] [EC:2.3.1.241 2.3.1.-]\n",
      "line in gene PA0011  lipid A biosynthesis lauroyl acyltransferase [KO:K02517] [EC:2.3.1.241 2.3.1.-]\n",
      "            PA3242  lipid A biosynthesis lauroyl acyltransferase [KO:K02517] [EC:2.3.1.241 2.3.1.-]\n",
      "line in gene PA3242  lipid A biosynthesis lauroyl acyltransferase [KO:K02517] [EC:2.3.1.241 2.3.1.-]\n",
      "            PA4512  lpxO1; lipopolysaccharide biosynthetic protein LpxO [KO:K12979] [EC:1.14.11.-]\n",
      "line in gene PA4512  lpxO1; lipopolysaccharide biosynthetic protein LpxO [KO:K12979] [EC:1.14.11.-]\n",
      "            PA0936  lpxO2; lipopolysaccharide biosynthetic protein LpxO [KO:K12979] [EC:1.14.11.-]\n",
      "line in gene PA0936  lpxO2; lipopolysaccharide biosynthetic protein LpxO [KO:K12979] [EC:1.14.11.-]\n",
      "            PA1972  hypothetical protein [KO:K03760] [EC:2.7.8.43]\n",
      "line in gene PA1972  hypothetical protein [KO:K03760] [EC:2.7.8.43]\n",
      "            PA3310  hypothetical protein [KO:K03760] [EC:2.7.8.43]\n",
      "line in gene PA3310  hypothetical protein [KO:K03760] [EC:2.7.8.43]\n",
      "            PA3556  arnT; 4-amino-4-deoxy-L-arabinose lipid A transferase [KO:K07264] [EC:2.4.2.43]\n",
      "line in gene PA3556  arnT; 4-amino-4-deoxy-L-arabinose lipid A transferase [KO:K07264] [EC:2.4.2.43]\n",
      "            PA4661  pagL; lipid A 3-O-deacylase [KO:K12976] [EC:3.1.1.-]\n",
      "line in gene PA4661  pagL; lipid A 3-O-deacylase [KO:K12976] [EC:3.1.1.-]\n",
      "            PA4425  phosphoheptose isomerase [KO:K03271] [EC:5.3.1.28]\n",
      "line in gene PA4425  phosphoheptose isomerase [KO:K03271] [EC:5.3.1.28]\n",
      "            PA4996  rfaE; bifunctional heptose 7-phosphate kinase/heptose 1-phosphate adenyltransferase [KO:K03272] [EC:2.7.1.167 2.7.7.70]\n",
      "line in gene PA4996  rfaE; bifunctional heptose 7-phosphate kinase/heptose 1-phosphate adenyltransferase [KO:K03272] [EC:2.7.1.167 2.7.7.70]\n",
      "            PA0006  D-glycero-beta-D-manno-heptose-1,7-bisphosphate 7-phosphatase [KO:K03273] [EC:3.1.3.82 3.1.3.83]\n",
      "line in gene PA0006  D-glycero-beta-D-manno-heptose-1,7-bisphosphate 7-phosphatase [KO:K03273] [EC:3.1.3.82 3.1.3.83]\n",
      "            PA3337  rfaD; ADP-L-glycero-D-mannoheptose-6-epimerase [KO:K03274] [EC:5.1.3.20]\n",
      "line in gene PA3337  rfaD; ADP-L-glycero-D-mannoheptose-6-epimerase [KO:K03274] [EC:5.1.3.20]\n",
      "            PA5011  waaC; heptosyltransferase I [KO:K02841] [EC:2.4.-.-]\n",
      "line in gene PA5011  waaC; heptosyltransferase I [KO:K02841] [EC:2.4.-.-]\n",
      "            PA5012  waaF; heptosyltransferase II [KO:K02843] [EC:2.4.-.-]\n",
      "line in gene PA5012  waaF; heptosyltransferase II [KO:K02843] [EC:2.4.-.-]\n",
      "            PA5009  waaP; lipopolysaccharide kinase WaaP [KO:K02848] [EC:2.7.1.235]\n",
      "line in gene PA5009  waaP; lipopolysaccharide kinase WaaP [KO:K02848] [EC:2.7.1.235]\n",
      "            PA4517  hypothetical protein [KO:K19353] [EC:2.7.8.-]\n",
      "line in gene PA4517  hypothetical protein [KO:K19353] [EC:2.7.8.-]\n",
      "            PA5010  waaG; UDP-glucose:(heptosyl) LPS alpha 1,3-glucosyltransferase WaaG [KO:K02844] [EC:2.4.1.-]\n",
      "line in gene PA5010  waaG; UDP-glucose:(heptosyl) LPS alpha 1,3-glucosyltransferase WaaG [KO:K02844] [EC:2.4.1.-]\n",
      "COMPOUND    C00043  UDP-N-acetyl-alpha-D-glucosamine\n",
      "            C00199  D-Ribulose 5-phosphate\n",
      "            C00229  Acyl-carrier protein\n",
      "            C01112  D-Arabinose 5-phosphate\n",
      "            C01187  3-Deoxy-D-manno-octulosonate\n",
      "            C04088  Octadecanoyl-[acyl-carrier protein]\n",
      "            C04121  CMP-3-deoxy-D-manno-octulosonate\n",
      "            C04478  3-Deoxy-D-manno-octulosonate 8-phosphate\n",
      "            C04652  UDP-2,3-bis(3-hydroxytetradecanoyl)glucosamine\n",
      "            C04738  UDP-3-O-(3-hydroxytetradecanoyl)-N-acetylglucosamine\n",
      "            C04824  Lipid X\n",
      "            C04919  2,3,2'3'-Tetrakis(3-hydroxytetradecanoyl)-D-glucosaminyl-1,6-beta-D-glucosamine 1,4'-bisphosphate\n",
      "            C04932  Lipid A disaccharide\n",
      "            C05223  Dodecanoyl-[acyl-carrier protein]\n",
      "            C05382  Sedoheptulose 7-phosphate\n",
      "            C05761  Tetradecanoyl-[acp]\n",
      "            C06022  UDP-3-O-(3-hydroxytetradecanoyl)-D-glucosamine\n",
      "            C06024  3-Deoxy-D-manno-octulosonyl-lipid IV(A)\n",
      "            C06025  KDO2-lipid IVA\n",
      "            C06026  KDO2-lipid A\n",
      "            C06251  Lauroyl-KDO2-lipid IV(A)\n",
      "            C06397  ADP-D-glycero-beta-D-manno-heptose\n",
      "            C06398  ADP-L-glycero-beta-D-manno-heptose\n",
      "            C07836  D-glycero-beta-D-manno-Heptose 7-phosphate\n",
      "            C07838  D-glycero-beta-D-manno-Heptose 1-phosphate\n",
      "            C11472  D-glycero-beta-D-manno-Heptose 1,7-bisphosphate\n",
      "            C19877  4-O-Phospho-alpha-Kdo-(2->6)-lipid IVA\n",
      "            C19878  D-glycero-alpha-D-manno-Heptose 7-phosphate\n",
      "            C19879  D-glycero-alpha-D-manno-Heptose 1,7-bisphosphate\n",
      "            C19880  D-glycero-alpha-D-manno-Heptose 1-phosphate\n",
      "            C19881  GDP-D-glycero-alpha-D-manno-heptose\n",
      "            C19884  Lipid IIA\n",
      "            C19890  alpha-Kdo-(2->4)-alpha-Kdo-(2->6)-[4-P-L-Ara4N]-lipid A\n",
      "            C20933  (9Z)-Hexadec-9-enoyl-KDO2-lipid IV(A)\n",
      "            C21157  KDO2-lipid A 1-diphosphate\n",
      "            C21173  PEtN-KDO2-lipid A\n",
      "            C21461  Kdo2-lipid A 1-(2-aminoethyl diphosphate)\n",
      "            C21463  Kdo2-lipid A 1,4'-bis(2-aminoethyl diphosphate)\n",
      "            C21982  Lipid IVB\n",
      "            C21983  Deacyl-lipid IVB\n",
      "            C21984  Deacyl-lipid IVA\n",
      "            C21988  H. pylori KDO2-lipid A\n",
      "            C21991  Palmitoleoyl-myristoyl-KDO2-lipid A\n",
      "            C21992  3'-O-Deacylated KDO2-lipid A\n",
      "            C21993  C16-KDO2-lipid A\n",
      "            C21994  1-Dephospho-KDO2-lipid A\n",
      "            C21995  1-PEtN-KDO2-lipid A\n",
      "            C21996  1-PEtN-KDO-lipid A\n",
      "            C21997  4'-Dephosphorylated 1-PEtN-KDO-lipid A\n",
      "            C21998  H. pylori KDO-lipid A\n",
      "            C21999  Stearoyl-KDO2-lipid IVA\n",
      "            C22002  Lipid IIB\n",
      "            C22003  (S)-2-Hydroxymyristate-modified lipid A\n",
      "            C22068  H. pylori KDO2-lipid IVA\n",
      "REFERENCE   PMID:9791168\n",
      "  AUTHORS   Heinrichs DE, Yethon JA, Whitfield C.\n",
      "  TITLE     Molecular basis for structural diversity in the core regions of the lipopolysaccharides of Escherichia coli and Salmonella enterica.\n",
      "  JOURNAL   Mol Microbiol 30:221-32 (1998)\n",
      "            DOI:10.1046/j.1365-2958.1998.01063.x\n",
      "REFERENCE   PMID:11751812\n",
      "  AUTHORS   Kneidinger B, Marolda C, Graninger M, Zamyatina A, McArthur F, Kosma P, Valvano MA, Messner P.\n",
      "  TITLE     Biosynthesis pathway of ADP-L-glycero-beta-D-manno-heptose in Escherichia coli.\n",
      "  JOURNAL   J Bacteriol 184:363-9 (2002)\n",
      "            DOI:10.1128/JB.184.2.363-369.2002\n",
      "REFERENCE   PMID:9587193\n",
      "  AUTHORS   Wyckoff TJ, Raetz CR, Jackman JE.\n",
      "  TITLE     Antibacterial and anti-inflammatory agents that target endotoxin.\n",
      "  JOURNAL   Trends Microbiol 6:154-9 (1998)\n",
      "            DOI:10.1016/S0966-842X(98)01230-X\n",
      "REFERENCE   PMID:9765561\n",
      "  AUTHORS   Shibayama K, Ohsuka S, Tanaka T, Arakawa Y, Ohta M.\n",
      "  TITLE     Conserved structural regions involved in the catalytic mechanism of Escherichia coli K-12 WaaO (RfaI).\n",
      "  JOURNAL   J Bacteriol 180:5313-8 (1998)\n",
      "            DOI:10.1128/JB.180.20.5313-5318.1998\n",
      "REFERENCE   PMID:12045108\n",
      "  AUTHORS   Raetz CR, Whitfield C\n",
      "  TITLE     Lipopolysaccharide endotoxins.\n",
      "  JOURNAL   Annu Rev Biochem 71:635-700 (2002)\n",
      "            DOI:10.1146/annurev.biochem.71.110601.135414\n",
      "REFERENCE   PMID:24372821\n",
      "  AUTHORS   Rubin EJ, O'Brien JP, Ivanov PL, Brodbelt JS, Trent MS\n",
      "  TITLE     Identification of a broad family of lipid A late acyltransferases with non-canonical substrate specificity.\n",
      "  JOURNAL   Mol Microbiol 91:887-99 (2014)\n",
      "            DOI:10.1111/mmi.12501\n",
      "REFERENCE   PMID:20659292\n",
      "  AUTHORS   Stead CM, Zhao J, Raetz CR, Trent MS\n",
      "  TITLE     Removal of the outer Kdo from Helicobacter pylori lipopolysaccharide and its impact on the bacterial surface.\n",
      "  JOURNAL   Mol Microbiol 78:837-52 (2010)\n",
      "            DOI:10.1111/j.1365-2958.2010.07304.x\n",
      "REFERENCE   PMID:30245684\n",
      "  AUTHORS   Ebbensgaard A, Mordhorst H, Aarestrup FM, Hansen EB\n",
      "  TITLE     The Role of Outer Membrane Proteins and Lipopolysaccharides for the Sensitivity of Escherichia coli to Antimicrobial Peptides.\n",
      "  JOURNAL   Front Microbiol 9:2153 (2018)\n",
      "            DOI:10.3389/fmicb.2018.02153\n",
      "REFERENCE   PMID:28630161\n",
      "  AUTHORS   Botos I, Noinaj N, Buchanan SK\n",
      "  TITLE     Insertion of proteins and lipopolysaccharide into the bacterial outer membrane.\n",
      "  JOURNAL   Philos Trans R Soc Lond B Biol Sci 372:20160224 (2017)\n",
      "            DOI:10.1098/rstb.2016.0224\n",
      "REL_PATHWAY pae00030  Pentose phosphate pathway\n",
      "            pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "KO_PATHWAY  ko00540\n",
      "///\n",
      "path:pae00541\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00541                    Pathway\n",
      "NAME        O-Antigen nucleotide sugar biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION The O-antigen is a repetitive glycan polymer with a repeat unit of two to six sugar residues, and comprises the outermost and immunogenic domain of the lipopolysaccharide (map00540). Most of the sugar residues are derived from nucleotide sugars and the genes for their biosynthesis seem to be well conserved among Gram-negative bacterial species. They include UDP-sugars synthesized from Glc-1P via UDP-Glc or from GlcNAc-1P via UDP-GlcNAc, GDP-sugars synthesized from Fru-6P via GDP-Man, and dTDP-sugars and CDP-sugars synthesized from Glc-1P via dTDP-Glc and CDP-Glc, respectively. CMP-sugars are also synthesized from UDP-GlcNAc. The O-antigen nucleotide sugar biosynthesis genes often form gene clusters with glycosyltransferase genes and other processing genes such as for translocation and polymerization.\n",
      "CLASS       Metabolism; Glycan biosynthesis and metabolism\n",
      "PATHWAY_MAP pae00541  O-Antigen nucleotide sugar biosynthesis\n",
      "MODULE      pae_M00793  dTDP-L-rhamnose biosynthesis [PATH:pae00541]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2023  galU; UTP-glucose-1-phosphate uridylyltransferase [KO:K00963] [EC:2.7.7.9]\n",
      "line in gene PA2023  galU; UTP-glucose-1-phosphate uridylyltransferase [KO:K00963] [EC:2.7.7.9]\n",
      "            PA4068  epimerase [KO:K01784] [EC:5.1.3.2]\n",
      "line in gene PA4068  epimerase [KO:K01784] [EC:5.1.3.2]\n",
      "            PA1384  galE; UDP-glucose 4-epimerase [KO:K01784] [EC:5.1.3.2]\n",
      "line in gene PA1384  galE; UDP-glucose 4-epimerase [KO:K01784] [EC:5.1.3.2]\n",
      "            PA2022  UDP-glucose 6-dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "line in gene PA2022  UDP-glucose 6-dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "            PA3559  nucleotide sugar dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "line in gene PA3559  nucleotide sugar dehydrogenase [KO:K00012] [EC:1.1.1.22]\n",
      "            PA5163  rmlA; glucose-1-phosphate thymidylyltransferase [KO:K00973] [EC:2.7.7.24]\n",
      "line in gene PA5163  rmlA; glucose-1-phosphate thymidylyltransferase [KO:K00973] [EC:2.7.7.24]\n",
      "            PA5161  rmlB; dTDP-D-glucose 4,6-dehydratase [KO:K01710] [EC:4.2.1.46]\n",
      "line in gene PA5161  rmlB; dTDP-D-glucose 4,6-dehydratase [KO:K01710] [EC:4.2.1.46]\n",
      "            PA5164  rmlC; dTDP-4-dehydrorhamnose 3,5-epimerase [KO:K01790] [EC:5.1.3.13]\n",
      "line in gene PA5164  rmlC; dTDP-4-dehydrorhamnose 3,5-epimerase [KO:K01790] [EC:5.1.3.13]\n",
      "            PA5162  rmlD; dTDP-4-dehydrorhamnose reductase [KO:K00067] [EC:1.1.1.133]\n",
      "line in gene PA5162  rmlD; dTDP-4-dehydrorhamnose reductase [KO:K00067] [EC:1.1.1.133]\n",
      "            PA4069  hypothetical protein [KO:K00067] [EC:1.1.1.133]\n",
      "line in gene PA4069  hypothetical protein [KO:K00067] [EC:1.1.1.133]\n",
      "            PA5236  CDP-6-deoxy-delta-3,4-glucoseen reductase [KO:K00523] [EC:1.17.1.1]\n",
      "line in gene PA5236  CDP-6-deoxy-delta-3,4-glucoseen reductase [KO:K00523] [EC:1.17.1.1]\n",
      "            PA3551  algA; bifunctional mannose-1-phosphate guanylyltransferase/mannose-6-phosphate isomerase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "line in gene PA3551  algA; bifunctional mannose-1-phosphate guanylyltransferase/mannose-6-phosphate isomerase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "            PA2232  pslB; biofilm formation protein PslB [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "line in gene PA2232  pslB; biofilm formation protein PslB [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "            PA5452  wbpW; phosphomannose isomerase/mannose-1-phosphate guanylyl transferase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "line in gene PA5452  wbpW; phosphomannose isomerase/mannose-1-phosphate guanylyl transferase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "            PA5453  gmd; GDP-mannose 4,6-dehydratase [KO:K01711] [EC:4.2.1.47]\n",
      "line in gene PA5453  gmd; GDP-mannose 4,6-dehydratase [KO:K01711] [EC:4.2.1.47]\n",
      "            PA5454  rmd; oxidoreductase Rmd [KO:K15856] [EC:1.1.1.281]\n",
      "line in gene PA5454  rmd; oxidoreductase Rmd [KO:K15856] [EC:1.1.1.281]\n",
      "            PA5552  glmU; bifunctional glucosamine-1-phosphate acetyltransferase/N-acetylglucosamine-1-phosphate uridyltransferase [KO:K04042] [EC:2.7.7.23 2.3.1.157]\n",
      "line in gene PA5552  glmU; bifunctional glucosamine-1-phosphate acetyltransferase/N-acetylglucosamine-1-phosphate uridyltransferase [KO:K04042] [EC:2.7.7.23 2.3.1.157]\n",
      "            PA3141  wbpM; nucleotide sugar epimerase/dehydratase WbpM [KO:K24300] [EC:4.2.1.135]\n",
      "line in gene PA3141  wbpM; nucleotide sugar epimerase/dehydratase WbpM [KO:K24300] [EC:4.2.1.135]\n",
      "            PA3146  wbpK; NAD-dependent epimerase/dehydratase [KO:K24312]\n",
      "line in gene PA3146  wbpK; NAD-dependent epimerase/dehydratase [KO:K24312]\n",
      "            PA3159  wbpA; UDP-N-acetyl-d-glucosamine 6-dehydrogenase [KO:K13015] [EC:1.1.1.136]\n",
      "line in gene PA3159  wbpA; UDP-N-acetyl-d-glucosamine 6-dehydrogenase [KO:K13015] [EC:1.1.1.136]\n",
      "            PA3158  wbpB; UDP-N-acetyl-2-amino-2-deoxy-D-glucuronate oxidase [KO:K13016] [EC:1.1.1.335]\n",
      "line in gene PA3158  wbpB; UDP-N-acetyl-2-amino-2-deoxy-D-glucuronate oxidase [KO:K13016] [EC:1.1.1.335]\n",
      "            PA3155  wbpE; UDP-2-acetamido-2-deoxy-3-oxo-D-glucuronate aminotransferase [KO:K13017] [EC:2.6.1.98]\n",
      "line in gene PA3155  wbpE; UDP-2-acetamido-2-deoxy-3-oxo-D-glucuronate aminotransferase [KO:K13017] [EC:2.6.1.98]\n",
      "            PA3156  wbpD; UDP-2-acetamido-3-amino-2,3-dideoxy-D-glucuronate N-acetyltransferase [KO:K13018] [EC:2.3.1.201]\n",
      "line in gene PA3156  wbpD; UDP-2-acetamido-3-amino-2,3-dideoxy-D-glucuronate N-acetyltransferase [KO:K13018] [EC:2.3.1.201]\n",
      "            PA3148  wbpI; UDP-2,3-diacetamido-2,3-dideoxy-D-glucuronate 2-epimeras [KO:K13019] [EC:5.1.3.23]\n",
      "line in gene PA3148  wbpI; UDP-2,3-diacetamido-2,3-dideoxy-D-glucuronate 2-epimeras [KO:K13019] [EC:5.1.3.23]\n",
      "            PA3150  wbpG; LPS biosynthesis protein WbpG [KO:K24326]\n",
      "line in gene PA3150  wbpG; LPS biosynthesis protein WbpG [KO:K24326]\n",
      "COMPOUND    C00029  UDP-glucose\n",
      "            C00043  UDP-N-acetyl-alpha-D-glucosamine\n",
      "            C00052  UDP-alpha-D-galactose\n",
      "            C00096  GDP-mannose\n",
      "            C00103  D-Glucose 1-phosphate\n",
      "            C00128  CMP-N-acetylneuraminate\n",
      "            C00167  UDP-glucuronate\n",
      "            C00203  UDP-N-acetyl-D-galactosamine\n",
      "            C00270  N-Acetylneuraminate\n",
      "            C00275  D-Mannose 6-phosphate\n",
      "            C00325  GDP-L-fucose\n",
      "            C00501  CDP-glucose\n",
      "            C00617  UDP-D-galacturonate\n",
      "            C00636  D-Mannose 1-phosphate\n",
      "            C00645  N-Acetyl-D-mannosamine\n",
      "            C00688  dTDP-4-dehydro-beta-L-rhamnose\n",
      "            C00842  dTDP-glucose\n",
      "            C01170  UDP-N-acetyl-D-mannosamine\n",
      "            C01219  CDP-4-dehydro-6-deoxy-D-glucose\n",
      "            C01222  GDP-4-dehydro-6-deoxy-D-mannose\n",
      "            C01788  CDP-abequose\n",
      "            C02330  UDP-L-iduronate\n",
      "            C02977  GDP-6-deoxy-D-talose\n",
      "            C03117  GDP-6-deoxy-D-mannose\n",
      "            C03187  dTDP-6-deoxy-beta-L-talose\n",
      "            C03319  dTDP-L-rhamnose\n",
      "            C03598  CDP-3,6-dideoxy-D-glucose\n",
      "            C03599  CDP-3,6-dideoxy-D-mannose\n",
      "            C03733  UDP-alpha-D-galactofuranose\n",
      "            C04268  dTDP-4-amino-4,6-dideoxy-D-glucose\n",
      "            C04297  CDP-4-dehydro-3,6-dideoxy-D-glucose\n",
      "            C04346  dTDP-4-amino-4,6-dideoxy-D-galactose\n",
      "            C04501  N-Acetyl-alpha-D-glucosamine 1-phosphate\n",
      "            C04573  UDP-N-acetyl-2-amino-2-deoxy-D-glucuronate\n",
      "            C04613  UDP-2-acetamido-4-dehydro-2,6-dideoxyglucose\n",
      "            C04630  UDP-2-acetamido-4-amino-2,4,6-trideoxy-alpha-D-glucose\n",
      "            C05345  beta-D-Fructose 6-phosphate\n",
      "            C06018  dTDP-4-acetamido-4,6-dideoxy-D-glucose\n",
      "            C06240  UDP-N-acetyl-D-mannosaminouronate\n",
      "            C07277  dTDP-D-fucose\n",
      "            C11907  dTDP-4-oxo-6-deoxy-D-glucose\n",
      "            C11908  3,6-Dideoxy-3-oxo-dTDP-D-glucose\n",
      "            C11925  dTDP-3-amino-3,6-dideoxy-D-glucose\n",
      "            C13952  UDP-N-acetyl-D-galactosaminuronic acid\n",
      "            C17326  CDP-4-dehydro-3,6-dideoxy-D-glucose epimer\n",
      "            C17327  CDP-ascarylose\n",
      "            C19725  UDP-2,3-diacetamido-2,3-dideoxy-alpha-D-glucuronate\n",
      "            C19745  UDP-2,3-diacetamido-2,3-dideoxy-alpha-D-mannuronate\n",
      "            C19823  UDP-2-acetamido-2,6-dideoxy-beta-L-arabino-hexos-4-ulose\n",
      "            C19947  dTDP-3-amino-3,6-dideoxy-alpha-D-galactopyranose\n",
      "            C19960  dTDP-3-dehydro-6-deoxy-alpha-D-galactopyranose\n",
      "            C19961  UDP-4-amino-4,6-dideoxy-N-acetyl-beta-L-altrosamine\n",
      "            C19971  UDP-2,4-bis(acetamido)-2,4,6-trideoxy-beta-L-altropyranose\n",
      "            C19972  2,4-Bis(acetamido)-2,4,6-trideoxy-beta-L-altropyranose\n",
      "            C20082  Pseudaminic acid\n",
      "            C20083  CMP-pseudaminic acid\n",
      "            C20156  dTDP-3-acetamido-3,6-dideoxy-alpha-D-galactopyranose\n",
      "            C20359  UDP-2-acetamido-3-amino-2,3-dideoxy-alpha-D-glucuronate\n",
      "            C20395  UDP-2-acetamido-2-deoxy-alpha-D-ribo-hex-3-uluronate\n",
      "            C20415  dTDP-4-acetamido-4,6-dideoxy-alpha-D-galactose\n",
      "            C20612  GDP-4-dehydro-3,6-dideoxy-alpha-D-mannose\n",
      "            C20613  GDP-beta-L-colitose\n",
      "            C20638  GDP-4-amino-4,6-dideoxy-alpha-D-mannose\n",
      "            C20672  GDP-4-acetamido-4,6-dideoxy-alpha-D-mannose\n",
      "            C20682  dTDP-alpha-D-fucofuranose\n",
      "            C20768  UDP-2-acetamido-2,6-dideoxy-beta-L-talose\n",
      "            C20769  UDP-2-acetamido-2,6-dideoxy-beta-L-lyxo-4-hexulose\n",
      "            C20774  UDP-N-acetyl-beta-L-fucosamine\n",
      "            C22245  UDP-2-acetamido-2-deoxy-alpha-D-galacturonamide\n",
      "            C22246  UDP-2,3-dideoxy-2-acetamido-3-acetamidino-alpha-D-mannuronic acid\n",
      "            C22247  UDP-2-acetamido-4-(D-alanylamino)-2,4,6-trideoxy-alpha-D-glucopyranose\n",
      "            C22248  UDP-2-acetamido-4-(D-alanylamino)-2,4,6-trideoxy-alpha-D-mannopyranose\n",
      "            C22249  2-Acetamido-4-(D-alanylamino)-2,4,6-trideoxy-D-mannopyranose\n",
      "            C22250  5-N-Acetyl-7-N-(D-alanyl)-legionaminic acid\n",
      "            C22251  CMP-5-N-acetyl-7-N-(D-alanyl)-legionaminic acid\n",
      "            C22252  UDP-2,4-diacetamido-2,4,6-trideoxy-beta-L-idose\n",
      "            C22253  UDP-2,4-diacetamido-2,4,6-trideoxy-beta-L-gulose\n",
      "            C22254  2,4-Diacetamido-2,4,6-trideoxy-beta-L-gulose\n",
      "            C22255  5,7-Diacetamido-3,5,7,9-tetradeoxy-L-glycero-D-galacto-non-2-ulosonic acid\n",
      "            C22256  CMP-5,7-diacetamido-3,5,7,9-tetradeoxy-L-glycero-D-galacto-non-2-ulosonic acid\n",
      "            C22257  dTDP-3-acetamido-3,6-dideoxy-alpha-D-glucose\n",
      "            C22259  UDP-2-acetamido-2,6-dideoxy-beta-L-mannose\n",
      "            C22260  UDP-2-acetamido-2,6-dideoxy-beta-L-glucose\n",
      "            C22261  UDP-2-acetamido-2,6-dideoxy-alpha-D-galactose\n",
      "            C22262  UDP-2-acetamido-2,6-dideoxy-alpha-D-glucose\n",
      "            C22270  UDP-2,6-dideoxy-2-acetamidino-beta-L-galactose\n",
      "            C22276  UDP-4-amino-4,6-dideoxy-N-acetyl-beta-L-idosamine\n",
      "            C22397  GDP-N-formyl-alpha-D-perosamine\n",
      "REFERENCE   PMID:14670712\n",
      "  AUTHORS   Samuel G, Reeves P\n",
      "  TITLE     Biosynthesis of O-antigens: genes and pathways involved in nucleotide sugar precursor synthesis and O-antigen assembly.\n",
      "  JOURNAL   Carbohydr Res 338:2503-19 (2003)\n",
      "            DOI:10.1016/j.carres.2003.07.009\n",
      "REFERENCE   PMID:8939432\n",
      "  AUTHORS   Burrows LL, Charter DF, Lam JS\n",
      "  TITLE     Molecular characterization of the Pseudomonas aeruginosa serotype O5 (PAO1) B-band lipopolysaccharide gene cluster.\n",
      "  JOURNAL   Mol Microbiol 22:481-95 (1996)\n",
      "            DOI:10.1046/j.1365-2958.1996.1351503.x\n",
      "REFERENCE   PMID:10627048\n",
      "  AUTHORS   Belanger M, Burrows LL, Lam JS\n",
      "  TITLE     Functional analysis of genes responsible for the synthesis of the B-band O antigen of Pseudomonas aeruginosa serotype O6 lipopolysaccharide.\n",
      "  JOURNAL   Microbiology 145 ( Pt 12):3505-21 (1999)\n",
      "            DOI:10.1099/00221287-145-12-3505\n",
      "REFERENCE   PMID:10828401\n",
      "  AUTHORS   Dean CR, Goldberg JB\n",
      "  TITLE     The wbpM gene in Pseudomonas aeruginosa serogroup O17 resides on a cryptic copy of the serogroup O11 O antigen gene locus.\n",
      "  JOURNAL   FEMS Microbiol Lett 187:59-63 (2000)\n",
      "            DOI:10.1111/j.1574-6968.2000.tb09137.x\n",
      "REFERENCE   PMID:15629947\n",
      "  AUTHORS   Feng L, Senchenkova SN, Tao J, Shashkov AS, Liu B, Shevelev SD, Reeves PR, Xu J, Knirel YA, Wang L\n",
      "  TITLE     Structural and genetic characterization of enterohemorrhagic Escherichia coli O145 O antigen and development of an O145 serogroup-specific PCR assay.\n",
      "  JOURNAL   J Bacteriol 187:758-64 (2005)\n",
      "            DOI:10.1128/JB.187.2.758-764.2005\n",
      "REFERENCE   PMID:18621892\n",
      "  AUTHORS   Westman EL, Preston A, Field RA, Lam JS\n",
      "  TITLE     Biosynthesis of a rare di-N-acetylated sugar in the lipopolysaccharides of both Pseudomonas aeruginosa and Bordetella pertussis occurs via an identical scheme despite different gene clusters.\n",
      "  JOURNAL   J Bacteriol 190:6060-9 (2008)\n",
      "            DOI:10.1128/JB.00579-08\n",
      "REFERENCE   PMID:19710102\n",
      "  AUTHORS   King JD, Kocincova D, Westman EL, Lam JS\n",
      "  TITLE     Review: Lipopolysaccharide biosynthesis in Pseudomonas aeruginosa.\n",
      "  JOURNAL   Innate Immun 15:261-312 (2009)\n",
      "            DOI:10.1177/1753425909106436\n",
      "REFERENCE   PMID:20331420\n",
      "  AUTHORS   Perepelov AV, Liu B, Senchenkova SN, Shashkov AS, Shevelev SD, Feng L, Wang L, Knirel YA\n",
      "  TITLE     Structure of the O-antigen and characterization of the O-antigen gene cluster of Escherichia coli O108 containing 5,7-diacetamido-3,5,7,9-tetradeoxy-L-glycero-D-galacto-non-2-ulosonic (8-epilegionaminic) acid.\n",
      "  JOURNAL   Biochemistry (Mosc) 75:19-24 (2010)\n",
      "            DOI:10.1134/s0006297910010037\n",
      "REFERENCE   PMID:26207987\n",
      "  AUTHORS   Hwang S, Aronov A, Bar-Peled M\n",
      "  TITLE     The Biosynthesis of UDP-D-QuiNAc in Bacillus cereus ATCC 14579.\n",
      "  JOURNAL   PLoS One 10:e0133790 (2015)\n",
      "            DOI:10.1371/journal.pone.0133790\n",
      "REFERENCE   PMID:30315110\n",
      "  AUTHORS   McDonald ND, DeMeester KE, Lewis AL, Grimes CL, Boyd EF\n",
      "  TITLE     Structural and functional characterization of a modified legionaminic acid involved in glycosylation of a bacterial lipopolysaccharide.\n",
      "  JOURNAL   J Biol Chem 293:19113-19126 (2018)\n",
      "            DOI:10.1074/jbc.RA118.004966\n",
      "REL_PATHWAY pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "KO_PATHWAY  ko00541\n",
      "///\n",
      "path:pae00550\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00550                    Pathway\n",
      "NAME        Peptidoglycan biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Peptidoglycan is a macromolecule made of long aminosugar strands cross-linked by short peptides. It forms the cell wall in bacteria surrounding the cytoplasmic membrane. The glycan strands are typically comprised of repeating N-acetylglucosamine (GlcNAc) and N-acetylmuramic acid (MurNAc) disaccharides. Each MurNAc is linked to a peptide of three to five amino acid residues. Disaccharide subunits are first assembled on the cytoplasmic side of the bacterial membrane on a polyisoprenoid anchor (lipid I and II). Polymerization of disaccharide subunits by transglycosylases and cross-linking of glycan strands by transpeptidases occur on the other side of the membrane. Bacterial cell wall biosynthesis inhibitors form a major class of antibiotics.\n",
      "CLASS       Metabolism; Glycan biosynthesis and metabolism\n",
      "PATHWAY_MAP pae00550  Peptidoglycan biosynthesis\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4450  murA; UDP-N-acetylglucosamine 1-carboxyvinyltransferase [KO:K00790] [EC:2.5.1.7]\n",
      "line in gene PA4450  murA; UDP-N-acetylglucosamine 1-carboxyvinyltransferase [KO:K00790] [EC:2.5.1.7]\n",
      "            PA2977  murB; UDP-N-acetylenolpyruvoylglucosamine reductase [KO:K00075] [EC:1.3.1.98]\n",
      "line in gene PA2977  murB; UDP-N-acetylenolpyruvoylglucosamine reductase [KO:K00075] [EC:1.3.1.98]\n",
      "            PA4411  murC; UDP-N-acetylmuramate--L-alanine ligase [KO:K01924] [EC:6.3.2.8]\n",
      "line in gene PA4411  murC; UDP-N-acetylmuramate--L-alanine ligase [KO:K01924] [EC:6.3.2.8]\n",
      "            PA4414  murD; UDP-N-acetylmuramoyl-L-alanyl-D-glutamate synthetase [KO:K01925] [EC:6.3.2.9]\n",
      "line in gene PA4414  murD; UDP-N-acetylmuramoyl-L-alanyl-D-glutamate synthetase [KO:K01925] [EC:6.3.2.9]\n",
      "            PA4417  murE; UDP-N-acetylmuramoylalanyl-D-glutamate--2,6-diaminopimelate ligase [KO:K01928] [EC:6.3.2.13]\n",
      "line in gene PA4417  murE; UDP-N-acetylmuramoylalanyl-D-glutamate--2,6-diaminopimelate ligase [KO:K01928] [EC:6.3.2.13]\n",
      "            PA4201  ddlA; D-alanine--D-alanine ligase [KO:K01921] [EC:6.3.2.4]\n",
      "line in gene PA4201  ddlA; D-alanine--D-alanine ligase [KO:K01921] [EC:6.3.2.4]\n",
      "            PA4410  ddlB; D-alanine--D-alanine ligase [KO:K01921] [EC:6.3.2.4]\n",
      "line in gene PA4410  ddlB; D-alanine--D-alanine ligase [KO:K01921] [EC:6.3.2.4]\n",
      "            PA4416  murF; UDP-N-acetylmuramoyl-tripeptide--D-alanyl-D-alanine ligase [KO:K01929] [EC:6.3.2.10]\n",
      "line in gene PA4416  murF; UDP-N-acetylmuramoyl-tripeptide--D-alanyl-D-alanine ligase [KO:K01929] [EC:6.3.2.10]\n",
      "            PA3652  uppS; ditrans,polycis-undecaprenyl-diphosphate synthase [KO:K00806] [EC:2.5.1.31]\n",
      "line in gene PA3652  uppS; ditrans,polycis-undecaprenyl-diphosphate synthase [KO:K00806] [EC:2.5.1.31]\n",
      "            PA1959  bacA; undecaprenyl-diphosphatase [KO:K06153] [EC:3.6.1.27]\n",
      "line in gene PA1959  bacA; undecaprenyl-diphosphatase [KO:K06153] [EC:3.6.1.27]\n",
      "            PA4415  mraY; phospho-N-acetylmuramoyl-pentapeptide-transferase [KO:K01000] [EC:2.7.8.13]\n",
      "line in gene PA4415  mraY; phospho-N-acetylmuramoyl-pentapeptide-transferase [KO:K01000] [EC:2.7.8.13]\n",
      "            PA4412  murG; undecaprenyldiphospho-muramoylpentapeptide beta-N-acetylglucosaminyltransferase [KO:K02563] [EC:2.4.1.227]\n",
      "line in gene PA4412  murG; undecaprenyldiphospho-muramoylpentapeptide beta-N-acetylglucosaminyltransferase [KO:K02563] [EC:2.4.1.227]\n",
      "            PA0378  monofunctional biosynthetic peptidoglycan transglycosylase [KO:K03814] [EC:2.4.1.129]\n",
      "line in gene PA0378  monofunctional biosynthetic peptidoglycan transglycosylase [KO:K03814] [EC:2.4.1.129]\n",
      "            PA5045  ponA; penicillin-binding protein 1A [KO:K05366] [EC:2.4.1.129 3.4.16.4]\n",
      "line in gene PA5045  ponA; penicillin-binding protein 1A [KO:K05366] [EC:2.4.1.129 3.4.16.4]\n",
      "            PA4700  mrcB; penicillin-binding protein 1B [KO:K05365] [EC:2.4.1.129 3.4.16.4]\n",
      "line in gene PA4700  mrcB; penicillin-binding protein 1B [KO:K05365] [EC:2.4.1.129 3.4.16.4]\n",
      "            PA4003  pbpA; penicillin-binding protein 2 [KO:K05515] [EC:3.4.16.4]\n",
      "line in gene PA4003  pbpA; penicillin-binding protein 2 [KO:K05515] [EC:3.4.16.4]\n",
      "            PA4418  ftsI; penicillin-binding protein 3 [KO:K03587] [EC:3.4.16.4]\n",
      "line in gene PA4418  ftsI; penicillin-binding protein 3 [KO:K03587] [EC:3.4.16.4]\n",
      "            PA2272  pbpC; penicillin-binding protein 3A [KO:K03587] [EC:3.4.16.4]\n",
      "line in gene PA2272  pbpC; penicillin-binding protein 3A [KO:K03587] [EC:3.4.16.4]\n",
      "            PA3999  dacC; D-ala-D-ala-carboxypeptidase [KO:K07258] [EC:3.4.16.4]\n",
      "line in gene PA3999  dacC; D-ala-D-ala-carboxypeptidase [KO:K07258] [EC:3.4.16.4]\n",
      "            PA3047  D-alanyl-D-alanine carboxypeptidase [KO:K07259] [EC:3.4.16.4 3.4.21.-]\n",
      "line in gene PA3047  D-alanyl-D-alanine carboxypeptidase [KO:K07259] [EC:3.4.16.4 3.4.21.-]\n",
      "COMPOUND    C00043  UDP-N-acetyl-alpha-D-glucosamine\n",
      "            C00133  D-Alanine\n",
      "            C00448  trans,trans-Farnesyl diphosphate\n",
      "            C00692  UDP-N-acetylmuramoyl-L-alanyl-D-glutamate\n",
      "            C00993  D-Alanyl-D-alanine\n",
      "            C01050  UDP-N-acetylmuramate\n",
      "            C01212  UDP-N-acetylmuramoyl-L-alanine\n",
      "            C04574  di-trans,poly-cis-Undecaprenyl diphosphate\n",
      "            C04631  UDP-N-acetyl-3-(1-carboxyvinyl)-D-glucosamine\n",
      "            C04702  UDPMurNAc(oyl-L-Ala-D-gamma-Glu-L-Lys-D-Ala-D-Ala)\n",
      "            C04851  MurAc(oyl-L-Ala-D-gamma-Glu-L-Lys-D-Ala-D-Ala)-diphospho-undecaprenol\n",
      "            C04877  UDP-N-acetylmuramoyl-L-alanyl-gamma-D-glutamyl-meso-2,6-diaminopimelate\n",
      "            C04882  UDP-N-acetylmuramoyl-L-alanyl-D-glutamyl-6-carboxy-L-lysyl-D-alanyl-D-alanine\n",
      "            C05892  UDP-N-acetylmuramoyl-L-alanyl-gamma-D-glutamyl-L-lysine\n",
      "            C05893  Undecaprenyl-diphospho-N-acetylmuramoyl-(N-acetylglucosamine)-L-alanyl-gamma-D-glutamyl-L-lysyl-D-alanyl-D-alanine\n",
      "            C05894  Undecaprenyl-diphospho-N-acetylmuramoyl-(N-acetylglucosamine)-L-alanyl-D-isoglutaminyl-L-lysyl-D-alanyl-D-alanine\n",
      "            C05895  Undecaprenyl-diphospho-N-acetylmuramoyl-(N-acetylglucosamine)-L-alanyl-D-isoglutaminyl-L-lysyl-(glycyl)5-D-alanyl-D-alanine\n",
      "            C05896  Crosslinked peptidoglycan\n",
      "            C05897  Undecaprenyl-diphospho-N-acetylmuramoyl-L-alanyl-D-glutamyl-meso-2,6-diaminopimeloyl-D-alanyl-D-alanine\n",
      "            C05898  Undecaprenyl-diphospho-N-acetylmuramoyl-(N-acetylglucosamine)-L-alanyl-D-glutamyl-meso-2,6-diaminopimeloyl-D-alanyl-D-alanine\n",
      "            C17541  Undecaprenyl-diphospho-N-acetylmuramoyl-(N-acetylglucosamine)-L-alanyl-D-isoglutaminyl-L-lysyl-(glycyl)-D-alanyl-D-alanine\n",
      "            C17542  Undecaprenyl-diphospho-N-acetylmuramoyl-(N-acetylglucosamine)-L-alanyl-D-isoglutaminyl-L-lysyl-(glycyl)3-D-alanyl-D-alanine\n",
      "            C17549  Undecaprenyl-diphospho-N-acetylmuramoyl-(N-acetylglucosamine)-L-alanyl-gamma-D-glutamyl-L-lysyl-(L-alanyl-L-alanyl)-D-alanyl-D-alanine\n",
      "            C17550  Undecaprenyl-diphospho-N-acetylmuramoyl-(N-acetylglucosamine)-L-alanyl-gamma-D-glutamyl-L-lysyl-(L-alanyl)-D-alanyl-D-alanine\n",
      "            C17556  di-trans,poly-cis-Undecaprenyl phosphate\n",
      "            C17558  di-trans,poly-cis-Undecaprenol\n",
      "            C22053  UDP-N-acetyl-alpha-D-muramoyl-L-alanyl-L-glutamate\n",
      "REFERENCE   PMID:18081839\n",
      "  AUTHORS   Bouhss A, Trunkfield AE, Bugg TD, Mengin-Lecreulx D\n",
      "  TITLE     The biosynthesis of peptidoglycan lipid-linked intermediates.\n",
      "  JOURNAL   FEMS Microbiol Rev 32:208-33 (2008)\n",
      "            DOI:10.1111/j.1574-6976.2007.00089.x\n",
      "REFERENCE   PMID:18077448\n",
      "  AUTHORS   Lloyd AJ, Gilbey AM, Blewett AM, De Pascale G, El Zoeiby A, Levesque RC, Catherwood AC, Tomasz A, Bugg TD, Roper DI, Dowson CG\n",
      "  TITLE     Characterization of tRNA-dependent peptide bond formation by MurM in the synthesis of Streptococcus pneumoniae peptidoglycan.\n",
      "  JOURNAL   J Biol Chem 283:6402-17 (2008)\n",
      "            DOI:10.1074/jbc.M708105200\n",
      "REFERENCE   PMID:18266853\n",
      "  AUTHORS   Barreteau H, Kovac A, Boniface A, Sova M, Gobec S, Blanot D\n",
      "  TITLE     Cytoplasmic steps of peptidoglycan biosynthesis.\n",
      "  JOURNAL   FEMS Microbiol Rev 32:168-207 (2008)\n",
      "            DOI:10.1111/j.1574-6976.2008.00104.x\n",
      "REFERENCE   PMID:11699883\n",
      "  AUTHORS   van Heijenoort J\n",
      "  TITLE     Recent advances in the formation of the bacterial peptidoglycan monomer unit.\n",
      "  JOURNAL   Nat Prod Rep 18:503-19 (2001)\n",
      "            DOI:10.1039/a804532a\n",
      "REFERENCE   PMID:11320055\n",
      "  AUTHORS   van Heijenoort  J\n",
      "  TITLE     Formation of the glycan chains in the synthesis of bacterial peptidoglycan.\n",
      "  JOURNAL   Glycobiology 11:25R-36R (2001)\n",
      "            DOI:10.1093/glycob/11.3.25R\n",
      "REL_PATHWAY pae00470  D-Amino acid metabolism\n",
      "            pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "            pae00900  Terpenoid backbone biosynthesis\n",
      "KO_PATHWAY  ko00550\n",
      "///\n",
      "path:pae00561\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00561                    Pathway\n",
      "NAME        Glycerolipid metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Lipid metabolism\n",
      "PATHWAY_MAP pae00561  Glycerolipid metabolism\n",
      "DBLINKS     GO: 0046486\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1052  hypothetical protein [KO:K00865] [EC:2.7.1.165]\n",
      "line in gene PA1052  hypothetical protein [KO:K00865] [EC:2.7.1.165]\n",
      "            PA1499  hypothetical protein [KO:K11529] [EC:2.7.1.165]\n",
      "line in gene PA1499  hypothetical protein [KO:K11529] [EC:2.7.1.165]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA2275  alcohol dehydrogenase [KO:K13979] [EC:1.1.1.2]\n",
      "line in gene PA2275  alcohol dehydrogenase [KO:K13979] [EC:1.1.1.2]\n",
      "            PA3579  glycerol kinase [KO:K00864] [EC:2.7.1.30]\n",
      "line in gene PA3579  glycerol kinase [KO:K00864] [EC:2.7.1.30]\n",
      "            PA3582  glpK; glycerol kinase [KO:K00864] [EC:2.7.1.30]\n",
      "line in gene PA3582  glpK; glycerol kinase [KO:K00864] [EC:2.7.1.30]\n",
      "            PA1487  carbohydrate kinase [KO:K00864] [EC:2.7.1.30]\n",
      "line in gene PA1487  carbohydrate kinase [KO:K00864] [EC:2.7.1.30]\n",
      "            PA3673  plsB; glycerol-3-phosphate acyltransferase [KO:K00631] [EC:2.3.1.15]\n",
      "line in gene PA3673  plsB; glycerol-3-phosphate acyltransferase [KO:K00631] [EC:2.3.1.15]\n",
      "            PA2969  plsX; phosphate acyltransferase [KO:K03621] [EC:2.3.1.274]\n",
      "line in gene PA2969  plsX; phosphate acyltransferase [KO:K03621] [EC:2.3.1.274]\n",
      "            PA0581  glycerol-3-phosphate acyltransferase PlsY [KO:K08591] [EC:2.3.1.275]\n",
      "line in gene PA0581  glycerol-3-phosphate acyltransferase PlsY [KO:K08591] [EC:2.3.1.275]\n",
      "            PA0005  lptA; lysophosphatidic acid acyltransferase [KO:K00655] [EC:2.3.1.51]\n",
      "line in gene PA0005  lptA; lysophosphatidic acid acyltransferase [KO:K00655] [EC:2.3.1.51]\n",
      "            PA3603  dgkA; diacylglycerol kinase [KO:K00901] [EC:2.7.1.107]\n",
      "line in gene PA3603  dgkA; diacylglycerol kinase [KO:K00901] [EC:2.7.1.107]\n",
      "            PA2862  lipA; lactonizing lipase [KO:K01046] [EC:3.1.1.3]\n",
      "line in gene PA2862  lipA; lactonizing lipase [KO:K01046] [EC:3.1.1.3]\n",
      "            PA4813  lipC; lipase LipC [KO:K01046] [EC:3.1.1.3]\n",
      "line in gene PA4813  lipC; lipase LipC [KO:K01046] [EC:3.1.1.3]\n",
      "COMPOUND    C00029  UDP-glucose\n",
      "            C00040  Acyl-CoA\n",
      "            C00093  sn-Glycerol 3-phosphate\n",
      "            C00103  D-Glucose 1-phosphate\n",
      "            C00111  Glycerone phosphate\n",
      "            C00116  Glycerol\n",
      "            C00162  Fatty acid\n",
      "            C00173  Acyl-[acyl-carrier protein]\n",
      "            C00184  Glycerone\n",
      "            C00197  3-Phospho-D-glycerate\n",
      "            C00258  D-Glycerate\n",
      "            C00416  Phosphatidate\n",
      "            C00422  Triacylglycerol\n",
      "            C00577  D-Glyceraldehyde\n",
      "            C00631  2-Phospho-D-glycerate\n",
      "            C00641  1,2-Diacyl-sn-glycerol\n",
      "            C00681  1-Acyl-sn-glycerol 3-phosphate\n",
      "            C00969  3-Hydroxypropanal\n",
      "            C01885  1-Acylglycerol\n",
      "            C02133  Acyl phosphate\n",
      "            C02457  Propane-1,3-diol\n",
      "            C03692  1,2-Diacyl-3-beta-D-galactosyl-sn-glycerol\n",
      "            C04046  3-D-Glucosyl-1,2-diacylglycerol\n",
      "            C04315  2-Acyl-3-O-(beta-D-galactosyl)-sn-glycerol\n",
      "            C04872  1,2-Diacyl-3-[3-(alpha-D-N-acetylneuraminyl)-beta-D-galactosyl]-sn-glycerol\n",
      "            C05401  3-beta-D-Galactosyl-sn-glycerol\n",
      "            C06037  Digalactosyl-diacylglycerol\n",
      "            C06038  Acyl1-monogalactosyl-diacylglycerol\n",
      "            C06040  Diglucosyldiacylglycerol\n",
      "            C06041  Glycerophosphoglycoglycerolipid\n",
      "            C06042  Lipoteichoic acid\n",
      "            C06364  1,2-Diacyl-3-alpha-D-glucosyl-sn-glycerol\n",
      "            C06365  alpha-Kojibiosyldiacylglycerol\n",
      "            C11521  UDP-6-sulfoquinovose\n",
      "            C13508  Sulfoquinovosyldiacylglycerol\n",
      "            C20897  Glycerophosphoglycoglycerolipid\n",
      "            C20898  Lipoteichoic acid\n",
      "            C20991  1,2-Diacyl-3-O-[beta-D-galactosyl-(1->6)-beta-D-galactosyl]-sn-glycerol\n",
      "REFERENCE   PMID:8662716\n",
      "  AUTHORS   Norbeck J, Pahlman AK, Akhtar N, Blomberg A, Adler L.\n",
      "  TITLE     Purification and characterization of two isoenzymes of DL-glycerol-3-phosphatase from Saccharomyces cerevisiae. Identification of the corresponding GPP1 and GPP2 genes and evidence for osmotic regulation of Gpp2p expression by the osmosensing mitogen-activated protein kinase signal transduction pathway.\n",
      "  JOURNAL   J Biol Chem 271:13875-81 (1996)\n",
      "            DOI:10.1074/jbc.271.23.13875\n",
      "REFERENCE   PMID:8995384\n",
      "  AUTHORS   Karlsson OP, Dahlqvist A, Vikstrom S, Wieslander A.\n",
      "  TITLE     Lipid dependence and basic kinetics of the purified 1,2-diacylglycerol 3-glucosyltransferase from membranes of Acholeplasma laidlawii.\n",
      "  JOURNAL   J Biol Chem 272:929-36 (1997)\n",
      "            DOI:10.1074/jbc.272.2.929\n",
      "REFERENCE   PMID:11294844\n",
      "  AUTHORS   Berg S, Edman M, Li L, Wikstrom M, Wieslander A.\n",
      "  TITLE     Sequence properties of the 1,2-diacylglycerol 3-glucosyltransferase from Acholeplasma laidlawii membranes. Recognition of a large group of lipid glycosyltransferases in eubacteria and archaea.\n",
      "  JOURNAL   J Biol Chem 276:22056-63 (2001)\n",
      "            DOI:10.1074/jbc.M102576200\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00071  Fatty acid degradation\n",
      "            pae00564  Glycerophospholipid metabolism\n",
      "KO_PATHWAY  ko00561\n",
      "///\n",
      "path:pae00562\n",
      "ENTRY       pae00562                    Pathway\n",
      "NAME        Inositol phosphate metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00562  Inositol phosphate metabolism\n",
      "DBLINKS     GO: 0006020\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0844  plcH; hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "line in gene PA0844  plcH; hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "            PA3319  plcN; non-hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "line in gene PA3319  plcN; non-hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "            PA3818  type III secretion system regulator SuhB [KO:K01092] [EC:3.1.3.25]\n",
      "line in gene PA3818  type III secretion system regulator SuhB [KO:K01092] [EC:3.1.3.25]\n",
      "            PA3570  mmsA; methylmalonate-semialdehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "line in gene PA3570  mmsA; methylmalonate-semialdehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "            PA0747  aldehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "line in gene PA0747  aldehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "            PA0130  3-oxopropanoate dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "line in gene PA0130  3-oxopropanoate dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "            PA4748  tpiA; triosephosphate isomerase [KO:K01803] [EC:5.3.1.1]\n",
      "line in gene PA4748  tpiA; triosephosphate isomerase [KO:K01803] [EC:5.3.1.1]\n",
      "COMPOUND    C00024  Acetyl-CoA\n",
      "            C00092  D-Glucose 6-phosphate\n",
      "            C00111  Glycerone phosphate\n",
      "            C00118  D-Glyceraldehyde 3-phosphate\n",
      "            C00137  myo-Inositol\n",
      "            C00191  D-Glucuronate\n",
      "            C00222  3-Oxopropanoate\n",
      "            C00558  D-Tagaturonate\n",
      "            C00641  1,2-Diacyl-sn-glycerol\n",
      "            C00691  2,4,6/3,5-Pentahydroxycyclohexanone\n",
      "            C01177  Inositol 1-phosphate\n",
      "            C01194  1-Phosphatidyl-D-myo-inositol\n",
      "            C01204  Phytic acid\n",
      "            C01220  1D-myo-Inositol 1,4-bisphosphate\n",
      "            C01243  1D-myo-Inositol 1,3,4-trisphosphate\n",
      "            C01245  D-myo-Inositol 1,4,5-trisphosphate\n",
      "            C01272  1D-myo-Inositol 1,3,4,5-tetrakisphosphate\n",
      "            C01277  1-Phosphatidyl-1D-myo-inositol 4-phosphate\n",
      "            C01284  1D-myo-Inositol 1,3,4,5,6-pentakisphosphate\n",
      "            C03546  myo-Inositol 4-phosphate\n",
      "            C03659  D-Bornesitol\n",
      "            C03660  L-Bornesitol\n",
      "            C04006  1D-myo-Inositol 3-phosphate\n",
      "            C04062  D-myo-Inositol 1,3-bisphosphate\n",
      "            C04063  D-myo-Inositol 3,4-bisphosphate\n",
      "            C04287  3D-3,5/4-Trihydroxycyclohexane-1,2-dione\n",
      "            C04477  1D-myo-Inositol 1,3,4,6-tetrakisphosphate\n",
      "            C04520  1D-myo-Inositol 3,4,5,6-tetrakisphosphate\n",
      "            C04549  1-Phosphatidyl-1D-myo-inositol 3-phosphate\n",
      "            C04563  D-myo-Inositol 1,2,4,5,6-pentakisphosphate\n",
      "            C04579  Inositol 1,2,3,5,6-pentakisphosphate\n",
      "            C04637  1-Phosphatidyl-D-myo-inositol 4,5-bisphosphate\n",
      "            C05981  Phosphatidylinositol-3,4,5-trisphosphate\n",
      "            C06153  scyllo-Inositol\n",
      "            C06892  2-Deoxy-5-keto-D-gluconic acid\n",
      "            C06893  2-Deoxy-5-keto-D-gluconic acid 6-phosphate\n",
      "            C11554  1-Phosphatidyl-1D-myo-inositol 3,4-bisphosphate\n",
      "            C11555  1D-myo-Inositol 1,4,5,6-tetrakisphosphate\n",
      "            C11556  1-Phosphatidyl-1D-myo-inositol 3,5-bisphosphate\n",
      "            C11557  1-Phosphatidyl-1D-myo-inositol 5-phosphate\n",
      "            C16737  5-Deoxy-D-glucuronate\n",
      "            C19794  CDP-1L-myo-inositol\n",
      "            C19799  Bis(1L-myo-inositol)-3,1'-phosphate 1-phosphate\n",
      "            C19891  1D-chiro-Inositol\n",
      "            C20251  1-Keto-D-chiro-inositol\n",
      "            C21643  3-Dehydro-scyllo-inosose\n",
      "            C21644  5-Dehydro-L-gluconate\n",
      "REFERENCE   PMID:14993306\n",
      "  AUTHORS   Yoshida K, Yamaguchi M, Ikeda H, Omae K, Tsurusaki K, Fujita Y\n",
      "  TITLE     The fifth gene of the iol operon of Bacillus subtilis, iolE, encodes 2-keto-myo-inositol dehydratase.\n",
      "  JOURNAL   Microbiology 150:571-80 (2004)\n",
      "            DOI:10.1099/mic.0.26768-0\n",
      "REFERENCE   PMID:18310071\n",
      "  AUTHORS   Yoshida K, Yamaguchi M, Morinaga T, Kinehara M, Ikeuchi M, Ashida H, Fujita Y.\n",
      "  TITLE     myo-Inositol catabolism in Bacillus subtilis.\n",
      "  JOURNAL   J Biol Chem 283:10415-24 (2008)\n",
      "            DOI:10.1074/jbc.M708043200\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00040  Pentose and glucuronate interconversions\n",
      "            pae00564  Glycerophospholipid metabolism\n",
      "KO_PATHWAY  ko00562\n",
      "///\n",
      "path:pae00564\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00564                    Pathway\n",
      "NAME        Glycerophospholipid metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Lipid metabolism\n",
      "PATHWAY_MAP pae00564  Glycerophospholipid metabolism\n",
      "MODULE      pae_M00093  Phosphatidylethanolamine (PE) biosynthesis, PA => PS => PE [PATH:pae00564]\n",
      "DBLINKS     GO: 0006650\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1614  gpsA; glycerol-3-phosphate dehydrogenase [KO:K00057] [EC:1.1.1.94]\n",
      "line in gene PA1614  gpsA; glycerol-3-phosphate dehydrogenase [KO:K00057] [EC:1.1.1.94]\n",
      "            PA3584  glpD; glycerol-3-phosphate dehydrogenase [KO:K00111] [EC:1.1.5.3]\n",
      "line in gene PA3584  glpD; glycerol-3-phosphate dehydrogenase [KO:K00111] [EC:1.1.5.3]\n",
      "            PA3025  FAD-dependent glycerol-3-phosphate dehydrogenase [KO:K00111] [EC:1.1.5.3]\n",
      "line in gene PA3025  FAD-dependent glycerol-3-phosphate dehydrogenase [KO:K00111] [EC:1.1.5.3]\n",
      "            PA1489  hypothetical protein [KO:K00111] [EC:1.1.5.3]\n",
      "line in gene PA1489  hypothetical protein [KO:K00111] [EC:1.1.5.3]\n",
      "            PA3673  plsB; glycerol-3-phosphate acyltransferase [KO:K00631] [EC:2.3.1.15]\n",
      "line in gene PA3673  plsB; glycerol-3-phosphate acyltransferase [KO:K00631] [EC:2.3.1.15]\n",
      "            PA0581  glycerol-3-phosphate acyltransferase PlsY [KO:K08591] [EC:2.3.1.275]\n",
      "line in gene PA0581  glycerol-3-phosphate acyltransferase PlsY [KO:K08591] [EC:2.3.1.275]\n",
      "            PA0005  lptA; lysophosphatidic acid acyltransferase [KO:K00655] [EC:2.3.1.51]\n",
      "line in gene PA0005  lptA; lysophosphatidic acid acyltransferase [KO:K00655] [EC:2.3.1.51]\n",
      "            PA3603  dgkA; diacylglycerol kinase [KO:K00901] [EC:2.7.1.107]\n",
      "line in gene PA3603  dgkA; diacylglycerol kinase [KO:K00901] [EC:2.7.1.107]\n",
      "            PA0844  plcH; hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "line in gene PA0844  plcH; hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "            PA3319  plcN; non-hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "line in gene PA3319  plcN; non-hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "            PA0347  glpQ; glycerophosphoryl diester phosphodiesterase [KO:K01126] [EC:3.1.4.46]\n",
      "line in gene PA0347  glpQ; glycerophosphoryl diester phosphodiesterase [KO:K01126] [EC:3.1.4.46]\n",
      "            PA2352  glycerophosphoryl diester phosphodiesterase [KO:K01126] [EC:3.1.4.46]\n",
      "line in gene PA2352  glycerophosphoryl diester phosphodiesterase [KO:K01126] [EC:3.1.4.46]\n",
      "            PA2990  phosphodiesterase [KO:K01126] [EC:3.1.4.46]\n",
      "line in gene PA2990  phosphodiesterase [KO:K01126] [EC:3.1.4.46]\n",
      "            PA4792  hypothetical protein [KO:K01126] [EC:3.1.4.46]\n",
      "line in gene PA4792  hypothetical protein [KO:K01126] [EC:3.1.4.46]\n",
      "            PA4024  eutB; ethanolamine ammonia-lyase large subunit [KO:K03735] [EC:4.3.1.7]\n",
      "line in gene PA4024  eutB; ethanolamine ammonia-lyase large subunit [KO:K03735] [EC:4.3.1.7]\n",
      "            PA4025  ethanolamine ammonia-lyase small subunit [KO:K03736] [EC:4.3.1.7]\n",
      "line in gene PA4025  ethanolamine ammonia-lyase small subunit [KO:K03736] [EC:4.3.1.7]\n",
      "            PA2536  phosphatidate cytidylyltransferase [KO:K00981] [EC:2.7.7.41]\n",
      "line in gene PA2536  phosphatidate cytidylyltransferase [KO:K00981] [EC:2.7.7.41]\n",
      "            PA3651  cdsA; phosphatidate cytidylyltransferase [KO:K00981] [EC:2.7.7.41]\n",
      "line in gene PA3651  cdsA; phosphatidate cytidylyltransferase [KO:K00981] [EC:2.7.7.41]\n",
      "            PA3857  pcs; phosphatidylcholine synthase [KO:K01004] [EC:2.7.8.24]\n",
      "line in gene PA3857  pcs; phosphatidylcholine synthase [KO:K01004] [EC:2.7.8.24]\n",
      "            PA4693  pssA; phosphatidylserine synthase [KO:K17103] [EC:2.7.8.8]\n",
      "line in gene PA4693  pssA; phosphatidylserine synthase [KO:K17103] [EC:2.7.8.8]\n",
      "            PA4957  psd; phosphatidylserine decarboxylase [KO:K01613] [EC:4.1.1.65]\n",
      "line in gene PA4957  psd; phosphatidylserine decarboxylase [KO:K01613] [EC:4.1.1.65]\n",
      "            PA2541  CDP-alcohol phosphatidyltransferase [KO:K00995] [EC:2.7.8.5]\n",
      "line in gene PA2541  CDP-alcohol phosphatidyltransferase [KO:K00995] [EC:2.7.8.5]\n",
      "            PA2584  pgsA; CDP-diacylglycerol--glycerol-3-phosphate 3-phosphatidyltransferase [KO:K00995] [EC:2.7.8.5]\n",
      "line in gene PA2584  pgsA; CDP-diacylglycerol--glycerol-3-phosphate 3-phosphatidyltransferase [KO:K00995] [EC:2.7.8.5]\n",
      "            PA4050  pgpA; phosphatidylglycerophosphatase A [KO:K01095] [EC:3.1.3.27]\n",
      "line in gene PA4050  pgpA; phosphatidylglycerophosphatase A [KO:K01095] [EC:3.1.3.27]\n",
      "            PA5547  hypothetical protein [KO:K18697] [EC:3.1.3.27]\n",
      "line in gene PA5547  hypothetical protein [KO:K18697] [EC:3.1.3.27]\n",
      "            PA5394  cls; cardiolipin synthetase [KO:K06131] [EC:2.7.8.-]\n",
      "line in gene PA5394  cls; cardiolipin synthetase [KO:K06131] [EC:2.7.8.-]\n",
      "            PA2155  cardiolipin synthase 2 [KO:K06131] [EC:2.7.8.-]\n",
      "line in gene PA2155  cardiolipin synthase 2 [KO:K06131] [EC:2.7.8.-]\n",
      "            PA4339  phospholipase [KO:K06131] [EC:2.7.8.-]\n",
      "line in gene PA4339  phospholipase [KO:K06131] [EC:2.7.8.-]\n",
      "            PA5310  hypothetical protein [KO:K06132] [EC:2.7.8.-]\n",
      "line in gene PA5310  hypothetical protein [KO:K06132] [EC:2.7.8.-]\n",
      "COMPOUND    C00040  Acyl-CoA\n",
      "            C00065  L-Serine\n",
      "            C00084  Acetaldehyde\n",
      "            C00093  sn-Glycerol 3-phosphate\n",
      "            C00111  Glycerone phosphate\n",
      "            C00114  Choline\n",
      "            C00157  Phosphatidylcholine\n",
      "            C00189  Ethanolamine\n",
      "            C00269  CDP-diacylglycerol\n",
      "            C00307  CDP-choline\n",
      "            C00344  Phosphatidylglycerol\n",
      "            C00346  Ethanolamine phosphate\n",
      "            C00350  Phosphatidylethanolamine\n",
      "            C00416  Phosphatidate\n",
      "            C00513  CDP-glycerol\n",
      "            C00570  CDP-ethanolamine\n",
      "            C00588  Choline phosphate\n",
      "            C00623  sn-Glycerol 1-phosphate\n",
      "            C00641  1,2-Diacyl-sn-glycerol\n",
      "            C00670  sn-Glycero-3-phosphocholine\n",
      "            C00681  1-Acyl-sn-glycerol 3-phosphate\n",
      "            C01194  1-Phosphatidyl-D-myo-inositol\n",
      "            C01210  N-Methylethanolamine phosphate\n",
      "            C01233  sn-Glycero-3-phosphoethanolamine\n",
      "            C01241  Phosphatidyl-N-methylethanolamine\n",
      "            C01996  Acetylcholine\n",
      "            C02737  Phosphatidylserine\n",
      "            C03372  Acylglycerone phosphate\n",
      "            C03819  1-Acylglycerophosphoinositol\n",
      "            C03872  L-Serine-phosphoethanolamine\n",
      "            C03892  Phosphatidylglycerophosphate\n",
      "            C03974  2-Acyl-sn-glycerol 3-phosphate\n",
      "            C04230  1-Acyl-sn-glycero-3-phosphocholine\n",
      "            C04233  2-Acyl-sn-glycero-3-phosphocholine\n",
      "            C04308  Phosphatidyl-N-dimethylethanolamine\n",
      "            C04438  1-Acyl-sn-glycero-3-phosphoethanolamine\n",
      "            C04590  3-(O-Geranylgeranyl)-sn-glycerol 1-phosphate\n",
      "            C04638  2,3-Bis-O-(geranylgeranyl)glycerol 1-phosphate\n",
      "            C05973  2-Acyl-sn-glycero-3-phosphoethanolamine\n",
      "            C05974  2-Acyl-sn-glycero-3-phosphoserine\n",
      "            C05976  Aminoacyl-phosphatidylglycerol\n",
      "            C05980  Cardiolipin\n",
      "            C06771  Triethanolamine\n",
      "            C06772  Diethanolamine\n",
      "            C13482  Phosphodimethylethanolamine\n",
      "            C18059  CDP-2,3-bis-O-(geranylgeranyl)-sn-glycerol\n",
      "            C18125  1-Acyl-sn-glycero-3-phosphoserine\n",
      "            C18126  1-Acyl-sn-glycero-3-phosphoglycerol\n",
      "            C18129  Monolysocardiolipin\n",
      "            C18168  Diacylglycerylhomoserine\n",
      "            C18169  Diacylglyceryl-N,N,N-trimethylhomoserine\n",
      "            C20466  2,3-Bis-O-(geranylgeranyl)-sn-glycero-1-phospho-L-serine\n",
      "            C20518  2,3-Bis-(O-phytanyl)-sn-glycerol 1-phosphate\n",
      "            C22340  Heptaprenylglyceryl phosphate\n",
      "            C22341  Heptaprenylglycerol\n",
      "            C22342  Diacetylheptaprenylglycerol\n",
      "REFERENCE   PMID:1512203\n",
      "  AUTHORS   Racenis PV, Lai JL, Das AK, Mullick PC, Hajra AK, Greenberg ML.\n",
      "  TITLE     The acyl dihydroxyacetone phosphate pathway enzymes for glycerolipid biosynthesis are present in the yeast Saccharomyces cerevisiae.\n",
      "  JOURNAL   J Bacteriol 174:5702-10 (1992)\n",
      "            DOI:10.1128/JB.174.17.5702-5710.1992\n",
      "REFERENCE   PMID:10092508\n",
      "  AUTHORS   Taniyama Y, Shibata S, Kita S, Horikoshi K, Fuse H, Shirafuji H, Sumino Y, Fujino M.\n",
      "  TITLE     Cloning and expression of a novel lysophospholipase which structurally resembles lecithin cholesterol acyltransferase.\n",
      "  JOURNAL   Biochem Biophys Res Commun 257:50-6 (1999)\n",
      "            DOI:10.1006/bbrc.1999.0411\n",
      "REFERENCE   PMID:10542045\n",
      "  AUTHORS   Athenstaedt K, Daum G.\n",
      "  TITLE     Phosphatidic acid, a key intermediate in lipid metabolism.\n",
      "  JOURNAL   Eur J Biochem 266:1-16 (1999)\n",
      "            DOI:10.1046/j.1432-1327.1999.00822.x\n",
      "REFERENCE   PMID:12431977\n",
      "  AUTHORS   Shields DJ, Lehner R, Agellon LB, Vance DE.\n",
      "  TITLE     Membrane topography of human phosphatidylethanolamine N-methyltransferase.\n",
      "  JOURNAL   J Biol Chem 278:2956-62 (2003)\n",
      "            DOI:10.1074/jbc.M210904200\n",
      "REFERENCE   PMID:14663079\n",
      "  AUTHORS   Martinez-Morales F, Schobert M, Lopez-Lara IM, Geiger O.\n",
      "  TITLE     Pathways for phosphatidylcholine biosynthesis in bacteria.\n",
      "  JOURNAL   Microbiology 149:3461-71 (2003)\n",
      "            DOI:10.1099/mic.0.26522-0\n",
      "REFERENCE   PMID:15522832\n",
      "  AUTHORS   Burgermeister M, Birner-Grunberger R, Nebauer R, Daum G.\n",
      "  TITLE     Contribution of different pathways to the supply of phosphatidylethanolamine and phosphatidylcholine to mitochondrial membranes of the yeast Saccharomyces cerevisiae.\n",
      "  JOURNAL   Biochim Biophys Acta 1686:161-8 (2004)\n",
      "            DOI:10.1016/j.bbalip.2004.09.007\n",
      "REFERENCE   PMID:1649829\n",
      "  AUTHORS   Hsu L, Jackowski S, Rock CO\n",
      "  TITLE     Isolation and characterization of Escherichia coli K-12 mutants lacking both 2-acyl-glycerophosphoethanolamine acyltransferase and acyl-acyl carrier protein synthetase activity.\n",
      "  JOURNAL   J Biol Chem 266:13783-8 (1991)\n",
      "REFERENCE   PMID:10938271\n",
      "  AUTHORS   Stone SJ, Vance JE.\n",
      "  TITLE     Phosphatidylserine synthase-1 and -2 are localized to mitochondria-associated membranes.\n",
      "  JOURNAL   J Biol Chem 275:34534-40 (2000)\n",
      "            DOI:10.1074/jbc.M002865200\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00561  Glycerolipid metabolism\n",
      "            pae00562  Inositol phosphate metabolism\n",
      "            pae00565  Ether lipid metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "KO_PATHWAY  ko00564\n",
      "///\n",
      "path:pae00565\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00565                    Pathway\n",
      "NAME        Ether lipid metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Lipid metabolism\n",
      "PATHWAY_MAP pae00565  Ether lipid metabolism\n",
      "DBLINKS     GO: 0046485\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3026  hypothetical protein [KO:K00803] [EC:2.5.1.26]\n",
      "line in gene PA3026  hypothetical protein [KO:K00803] [EC:2.5.1.26]\n",
      "            PA0844  plcH; hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "line in gene PA0844  plcH; hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "            PA3319  plcN; non-hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "line in gene PA3319  plcN; non-hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "COMPOUND    C00670  sn-Glycero-3-phosphocholine\n",
      "            C00958  Plasmenylcholine\n",
      "            C01233  sn-Glycero-3-phosphoethanolamine\n",
      "            C01264  2-Acetyl-1-alkyl-sn-glycero-3-phosphate\n",
      "            C02773  1-Alkyl-sn-glycerol\n",
      "            C03201  1-Alkyl-2-acylglycerol\n",
      "            C03372  Acylglycerone phosphate\n",
      "            C03454  1-Alkenyl-2-acylglycerol\n",
      "            C03715  O-Alkylglycerone phosphate\n",
      "            C03820  1-Alkyl-2-acetyl-sn-glycerol\n",
      "            C03968  1-Alkyl-sn-glycero-3-phosphate\n",
      "            C04317  1-Organyl-2-lyso-sn-glycero-3-phosphocholine\n",
      "            C04361  1-O-Alkyl-2-acetyl-3-acyl-sn-glycerol\n",
      "            C04475  1-Alkyl-2-acylglycerophosphoethanolamine\n",
      "            C04517  1-(1-Alkenyl)-sn-glycero-3-phosphocholine\n",
      "            C04598  2-Acetyl-1-alkyl-sn-glycero-3-phosphocholine\n",
      "            C04635  1-(1-Alkenyl)-sn-glycero-3-phosphoethanolamine\n",
      "            C04756  O-1-Alk-1-enyl-2-acyl-sn-glycero-3-phosphoethanolamine\n",
      "            C05212  1-Radyl-2-acyl-sn-glycero-3-phosphocholine\n",
      "            C05977  2-Acyl-1-alkyl-sn-glycero-3-phosphate\n",
      "            C15645  1-(1-Alkenyl)-sn-glycerol\n",
      "            C15646  1-(1-Alkenyl)-sn-glycero-3-phosphate\n",
      "            C15647  2-Acyl-1-(1-alkenyl)-sn-glycero-3-phosphate\n",
      "            C20825  Seminolipid\n",
      "            C20859  Galactosyl-1-alkyl-2-acylglycerol\n",
      "REFERENCE   PMID:11275267\n",
      "  AUTHORS   Nagan N, Zoeller RA.\n",
      "  TITLE     Plasmalogens: biosynthesis and functions.\n",
      "  JOURNAL   Prog Lipid Res 40:199-229 (2001)\n",
      "            DOI:10.1016/S0163-7827(01)00003-0\n",
      "REFERENCE   PMID:11015622\n",
      "  AUTHORS   Montrucchio G, Alloatti G, Camussi G.\n",
      "  TITLE     Role of platelet-activating factor in cardiovascular pathophysiology.\n",
      "  JOURNAL   Physiol Rev 80:1669-99 (2000)\n",
      "            DOI:10.1152/physrev.2000.80.4.1669\n",
      "REFERENCE   PMID:15788404\n",
      "  AUTHORS   Sakagami H, Aoki J, Natori Y, Nishikawa K, Kakehi Y, Natori Y, Arai H.\n",
      "  TITLE     Biochemical and molecular characterization of a novel choline-specific glycerophosphodiester phosphodiesterase belonging to the nucleotide pyrophosphatase/phosphodiesterase family.\n",
      "  JOURNAL   J Biol Chem 280:23084-93 (2005)\n",
      "            DOI:10.1074/jbc.M413438200\n",
      "REL_PATHWAY pae00564  Glycerophospholipid metabolism\n",
      "KO_PATHWAY  ko00565\n",
      "///\n",
      "path:pae00590\n",
      "ENTRY       pae00590                    Pathway\n",
      "NAME        Arachidonic acid metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Lipid metabolism\n",
      "PATHWAY_MAP pae00590  Arachidonic acid metabolism\n",
      "DBLINKS     GO: 0019369\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0838  glutathione peroxidase [KO:K00432] [EC:1.11.1.9]\n",
      "line in gene PA0838  glutathione peroxidase [KO:K00432] [EC:1.11.1.9]\n",
      "            PA1287  glutathione peroxidase [KO:K00432] [EC:1.11.1.9]\n",
      "line in gene PA1287  glutathione peroxidase [KO:K00432] [EC:1.11.1.9]\n",
      "            PA2826  glutathione peroxidase [KO:K00432] [EC:1.11.1.9]\n",
      "line in gene PA2826  glutathione peroxidase [KO:K00432] [EC:1.11.1.9]\n",
      "            PA1169  arachidonate 15-lipoxygenase [KO:K19246] [EC:1.13.11.33]\n",
      "line in gene PA1169  arachidonate 15-lipoxygenase [KO:K19246] [EC:1.13.11.33]\n",
      "COMPOUND    C00157  Phosphatidylcholine\n",
      "            C00219  Arachidonate\n",
      "            C00427  Prostaglandin H2\n",
      "            C00584  Prostaglandin E2\n",
      "            C00639  Prostaglandin F2alpha\n",
      "            C00696  Prostaglandin D2\n",
      "            C00909  Leukotriene A4\n",
      "            C01312  Prostaglandin I2\n",
      "            C02165  Leukotriene B4\n",
      "            C02166  Leukotriene C4\n",
      "            C02198  Thromboxane A2\n",
      "            C03577  20-Hydroxyleukotriene E4\n",
      "            C04577  15-OxoETE\n",
      "            C04742  (15S)-15-Hydroxy-5,8,11-cis-13-trans-eicosatetraenoate\n",
      "            C04805  5(S)-HETE\n",
      "            C04822  8(R)-HPETE\n",
      "            C04853  20-OH-Leukotriene B4\n",
      "            C05356  5(S)-HPETE\n",
      "            C05949  12-Keto-leukotriene B4\n",
      "            C05950  20-COOH-Leukotriene B4\n",
      "            C05951  Leukotriene D4\n",
      "            C05952  Leukotriene E4\n",
      "            C05953  Prostaglandin A2\n",
      "            C05954  Prostaglandin B2\n",
      "            C05955  Prostaglandin C2\n",
      "            C05956  Prostaglandin G2\n",
      "            C05957  Prostaglandin J2\n",
      "            C05958  Delta-12-Prostaglandin J2\n",
      "            C05959  11-epi-Prostaglandin F2alpha\n",
      "            C05960  15-Keto-prostaglandin F2alpha\n",
      "            C05961  6-Keto-prostaglandin F1alpha\n",
      "            C05962  6-Keto-prostaglandin E1\n",
      "            C05963  Thromboxane B2\n",
      "            C05964  11-Dehydro-thromboxane B2\n",
      "            C05965  12(S)-HPETE\n",
      "            C05966  15(S)-HPETE\n",
      "            C06314  Lipoxin A4\n",
      "            C06315  Lipoxin B4\n",
      "            C06462  Leukotriene F4\n",
      "            C13809  9,11,15-Trihydroxy-prosta-5,13-dien-1-oic acid\n",
      "            C14717  15-Deoxy-Delta12,14-PGJ2\n",
      "            C14732  5-OxoETE\n",
      "            C14748  20-HETE\n",
      "            C14749  19(S)-HETE\n",
      "            C14768  5,6-EET\n",
      "            C14769  8,9-EET\n",
      "            C14770  11,12-EET\n",
      "            C14771  14,15-EET\n",
      "            C14772  5,6-DHET\n",
      "            C14773  8,9-DHET\n",
      "            C14774  11,12-DHET\n",
      "            C14775  14,15-DHET\n",
      "            C14776  8(S)-HETE\n",
      "            C14777  12(S)-HETE\n",
      "            C14778  16(R)-HETE\n",
      "            C14779  9(S)-HETE\n",
      "            C14780  11(R)-HETE\n",
      "            C14781  15H-11,12-EETA\n",
      "            C14782  11,12,15-THETA\n",
      "            C14794  2,3-Dinor-8-iso prostaglandin F2alpha\n",
      "            C14795  2,3-Dinor-8-iso prostaglandin F1alpha\n",
      "            C14807  12-OxoETE\n",
      "            C14808  Hepoxilin A3\n",
      "            C14809  Trioxilin A3\n",
      "            C14810  Hepoxilin B3\n",
      "            C14811  Trioxilin B3\n",
      "            C14812  12(R)-HPETE\n",
      "            C14813  11H-14,15-EETA\n",
      "            C14814  11,14,15-THETA\n",
      "            C14815  5,6-Epoxytetraene\n",
      "            C14820  11(R)-HPETE\n",
      "            C14821  9(S)-HPETE\n",
      "            C14822  12(R)-HETE\n",
      "            C14823  8(S)-HPETE\n",
      "            C14824  8(R)-HETE\n",
      "REFERENCE   PMID:14636671\n",
      "  AUTHORS   Spector AA, Fang X, Snyder GD, Weintraub NL.\n",
      "  TITLE     Epoxyeicosatrienoic acids (EETs): metabolism and biochemical function.\n",
      "  JOURNAL   Prog Lipid Res 43:55-90 (2004)\n",
      "            DOI:10.1016/S0163-7827(03)00049-3\n",
      "REFERENCE   PMID:15292194\n",
      "  AUTHORS   Harman CA, Rieke CJ, Garavito RM, Smith WL.\n",
      "  TITLE     Crystal structure of arachidonic acid bound to a mutant of prostaglandin endoperoxide H synthase-1 that forms predominantly 11-hydroperoxyeicosatetraenoic acid.\n",
      "  JOURNAL   J Biol Chem 279:42929-35 (2004)\n",
      "            DOI:10.1074/jbc.M403013200\n",
      "REFERENCE   PMID:15205388\n",
      "  AUTHORS   Poloyac SM, Tortorici MA, Przychodzin DI, Reynolds RB, Xie W, Frye RF, Zemaitis MA.\n",
      "  TITLE     The effect of isoniazid on CYP2E1- and CYP4A-mediated hydroxylation of arachidonic acid in the rat liver and kidney.\n",
      "  JOURNAL   Drug Metab Dispos 32:727-33 (2004)\n",
      "            DOI:10.1124/dmd.32.7.727\n",
      "REFERENCE   PMID:14660610\n",
      "  AUTHORS   Chuang SS, Helvig C, Taimi M, Ramshaw HA, Collop AH, Amad M, White JA, Petkovich M, Jones G, Korczak B.\n",
      "  TITLE     CYP2U1, a novel human thymus- and brain-specific cytochrome P450, catalyzes omega- and (omega-1)-hydroxylation of fatty acids.\n",
      "  JOURNAL   J Biol Chem 279:6305-14 (2004)\n",
      "            DOI:10.1074/jbc.M311830200\n",
      "REFERENCE   PMID:14626496\n",
      "  AUTHORS   Sacerdoti D, Gatta A, McGiff JC.\n",
      "  TITLE     Role of cytochrome P450-dependent arachidonic acid metabolites in liver physiology and pathophysiology.\n",
      "  JOURNAL   Prostaglandins Other Lipid Mediat 72:51-71 (2003)\n",
      "            DOI:10.1016/S1098-8823(03)00077-7\n",
      "REFERENCE   PMID:14622984\n",
      "  AUTHORS   Pfister SL, Spitzbarth N, Zeldin DC, Lafite P, Mansuy D, Campbell WB.\n",
      "  TITLE     Rabbit aorta converts 15-HPETE to trihydroxyeicosatrienoic acids: potential role of cytochrome P450.\n",
      "  JOURNAL   Arch Biochem Biophys 420:142-52 (2003)\n",
      "            DOI:10.1016/j.abb.2003.09.026\n",
      "REFERENCE   PMID:11679404\n",
      "  AUTHORS   Fleming I.\n",
      "  TITLE     Cytochrome p450 and vascular homeostasis.\n",
      "  JOURNAL   Circ Res 89:753-62 (2001)\n",
      "            DOI:10.1161/hh2101.099268\n",
      "REFERENCE   PMID:11328810\n",
      "  AUTHORS   Qu W, Bradbury JA, Tsao CC, Maronpot R, Harry GJ, Parker CE, Davis LS, Breyer MD, Waalkes MP, Falck JR, Chen J, Rosenberg RL, Zeldin DC.\n",
      "  TITLE     Cytochrome P450 CYP2J9, a new mouse arachidonic acid omega-1 hydroxylase predominantly expressed in brain.\n",
      "  JOURNAL   J Biol Chem 276:25467-79 (2001)\n",
      "            DOI:10.1074/jbc.M100545200\n",
      "REFERENCE   PMID:10908295\n",
      "  AUTHORS   Tsao CC, Foley J, Coulter SJ, Maronpot R, Zeldin DC, Goldstein JA.\n",
      "  TITLE     CYP2C40, a unique arachidonic acid 16-hydroxylase, is the major CYP2C in murine intestinal tract.\n",
      "  JOURNAL   Mol Pharmacol 58:279-87 (2000)\n",
      "            DOI:10.1124/mol.58.2.279\n",
      "REFERENCE   PMID:9812980\n",
      "  AUTHORS   Pfister SL, Spitzbarth N, Nithipatikom K, Edgemond WS, Falck JR, Campbell WB.\n",
      "  TITLE     Identification of the 11,14,15- and 11,12, 15-trihydroxyeicosatrienoic acids as endothelium-derived relaxing factors of rabbit aorta.\n",
      "  JOURNAL   J Biol Chem 273:30879-87 (1998)\n",
      "            DOI:10.1074/jbc.273.47.30879\n",
      "KO_PATHWAY  ko00590\n",
      "///\n",
      "path:pae00592\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00592                    Pathway\n",
      "NAME        alpha-Linolenic acid metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Lipid metabolism\n",
      "PATHWAY_MAP pae00592  alpha-Linolenic acid metabolism\n",
      "DBLINKS     GO: 0036109\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2940  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA2940  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA3013  foaB; 3-ketoacyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA3013  foaB; 3-ketoacyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "            PA3454  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "line in gene PA3454  acyl-CoA thiolase [KO:K00632] [EC:2.3.1.16]\n",
      "COMPOUND    C00157  Phosphatidylcholine\n",
      "            C01226  12-OPDA\n",
      "            C04672  (9Z,15Z)-(13S)-12,13-Epoxyoctadeca-9,11,15-trienoic acid\n",
      "            C04780  8-[(1R,2R)-3-Oxo-2-{(Z)-pent-2-enyl}cyclopentyl]octanoate\n",
      "            C04785  13(S)-HPOT\n",
      "            C06427  (9Z,12Z,15Z)-Octadecatrienoic acid\n",
      "            C08491  (-)-Jasmonic acid\n",
      "            C08492  3-Hexenol\n",
      "            C11512  Methyl jasmonate\n",
      "            C16300  Stearidonic acid\n",
      "            C16308  Traumatic acid\n",
      "            C16309  Traumatin\n",
      "            C16310  3-Hexenal\n",
      "            C16311  12-Oxo-9(Z)-dodecenoic acid\n",
      "            C16316  13(S)-HOT\n",
      "            C16317  (+)-7-Isojasmonic acid\n",
      "            C16318  (+)-7-Isomethyljasmonate\n",
      "            C16319  Etherolenic acid\n",
      "            C16320  Colnelenic acid\n",
      "            C16321  9(S)-HPOT\n",
      "            C16322  9-Oxononanoic acid\n",
      "            C16323  3,6-Nonadienal\n",
      "            C16324  9,10-EOT\n",
      "            C16325  10-OPDA\n",
      "            C16326  9(S)-HOT\n",
      "            C16327  OPC8-CoA\n",
      "            C16328  trans-2-Enoyl-OPC8-CoA\n",
      "            C16330  3-Oxo-OPC8-CoA\n",
      "            C16331  OPC6-CoA\n",
      "            C16332  trans-2-Enoyl-OPC6-CoA\n",
      "            C16334  3-Oxo-OPC6-CoA\n",
      "            C16335  OPC4-CoA\n",
      "            C16336  trans-2-Enoyl-OPC4-CoA\n",
      "            C16338  3-Oxo-OPC4-CoA\n",
      "            C16339  (+)-7-Isojasmonic acid CoA\n",
      "            C16341  2(R)-HPOT\n",
      "            C16342  2(R)-HOT\n",
      "            C16343  Heptadecatrienal\n",
      "            C16344  Norlinolenic acid\n",
      "            C16345  Volicitin\n",
      "            C16346  17-Hydroxylinolenic acid\n",
      "            C19757  (Z)-3-Hexen-1-ol acetate\n",
      "            C21923  9-Hydroxy-12-oxo-10(E),15(Z)-octadecadienoic acid\n",
      "            C21924  9-Hydroxy-12-oxo-15(Z)-octadecenoic acid\n",
      "REFERENCE   PMID:17172287\n",
      "  AUTHORS   Schilmiller AL, Koo AJ, Howe GA.\n",
      "  TITLE     Functional diversification of acyl-coenzyme A oxidases in jasmonic acid biosynthesis and action.\n",
      "  JOURNAL   Plant Physiol 143:812-24 (2007)\n",
      "            DOI:10.1104/pp.106.092916\n",
      "REFERENCE   PMID:15141068\n",
      "  AUTHORS   Cruz Castillo M, Martinez C, Buchala A, Metraux JP, Leon J.\n",
      "  TITLE     Gene-specific involvement of beta-oxidation in wound-activated responses in Arabidopsis.\n",
      "  JOURNAL   Plant Physiol 135:85-94 (2004)\n",
      "            DOI:10.1104/pp.104.039925\n",
      "REFERENCE   PMID:17215350\n",
      "  AUTHORS   Truman W, Bennett MH, Kubigsteltig I, Turnbull C, Grant M.\n",
      "  TITLE     Arabidopsis systemic immunity uses conserved defense signaling pathways and is mediated by jasmonates.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 104:1075-80 (2007)\n",
      "            DOI:10.1073/pnas.0605423104\n",
      "REFERENCE   PMID:16116284\n",
      "  AUTHORS   Kuroda H, Oshima T, Kaneda H, Takashio M.\n",
      "  TITLE     Identification and functional analyses of two cDNAs that encode fatty acid 9-/13-hydroperoxide lyase (CYP74C) in rice.\n",
      "  JOURNAL   Biosci Biotechnol Biochem 69:1545-54 (2005)\n",
      "            DOI:10.1271/bbb.69.1545\n",
      "REFERENCE   PMID:15670154\n",
      "  AUTHORS   Hamberg M.\n",
      "  TITLE     Hidden stereospecificity in the biosynthesis of divinyl ether fatty acids.\n",
      "  JOURNAL   FEBS J 272:736-43 (2005)\n",
      "            DOI:10.1111/j.1742-4658.2004.04510.x\n",
      "REFERENCE   PMID:14522973\n",
      "  AUTHORS   Hamberg M, Sanz A, Rodriguez MJ, Calvo AP, Castresana C.\n",
      "  TITLE     Activation of the fatty acid alpha-dioxygenase pathway during bacterial infection of tobacco leaves. Formation of oxylipins protecting against cell death.\n",
      "  JOURNAL   J Biol Chem 278:51796-805 (2003)\n",
      "            DOI:10.1074/jbc.M310514200\n",
      "REFERENCE   \n",
      "  AUTHORS   Creelman RA, Mulpuri R.\n",
      "  TITLE     The Oxylipin Pathway in Arabidopsis\n",
      "  JOURNAL   The Arabidopsis Book 1:e0012 (2002)\n",
      "            DOI:10.1199/tab.0012\n",
      "KO_PATHWAY  ko00592\n",
      "///\n",
      "path:pae00600\n",
      "ENTRY       pae00600                    Pathway\n",
      "NAME        Sphingolipid metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Lipid metabolism\n",
      "PATHWAY_MAP pae00600  Sphingolipid metabolism\n",
      "DBLINKS     GO: 0006665\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0845  neutral ceramidase [KO:K12349] [EC:3.5.1.23]\n",
      "line in gene PA0845  neutral ceramidase [KO:K12349] [EC:3.5.1.23]\n",
      "COMPOUND    C00065  L-Serine\n",
      "            C00154  Palmitoyl-CoA\n",
      "            C00195  N-Acylsphingosine\n",
      "            C00319  Sphingosine\n",
      "            C00346  Ethanolamine phosphate\n",
      "            C00550  Sphingomyelin\n",
      "            C00836  Sphinganine\n",
      "            C01120  Sphinganine 1-phosphate\n",
      "            C01190  Glucosylceramide\n",
      "            C01290  Lactosylceramide\n",
      "            C01747  Psychosine\n",
      "            C02686  Galactosylceramide\n",
      "            C02934  3-Dehydrosphinganine\n",
      "            C02960  Ceramide 1-phosphate\n",
      "            C03640  Sphingosyl-phosphocholine\n",
      "            C05681  Ceramide 2-aminoethylphosphonate\n",
      "            C06062  Ceramide phosphoethanolamine\n",
      "            C06124  Sphingosine 1-phosphate\n",
      "            C06125  Sulfatide\n",
      "            C06126  Digalactosylceramide\n",
      "            C06127  Digalactosylceramide sulfate\n",
      "            C06128  GM4\n",
      "            C12126  Dihydroceramide\n",
      "            C12144  Phytosphingosine\n",
      "            C12145  Phytoceramide\n",
      "REFERENCE   PMID:12011104\n",
      "  AUTHORS   Merrill AH Jr.\n",
      "  TITLE     De novo sphingolipid biosynthesis: a necessary, but dangerous, pathway.\n",
      "  JOURNAL   J Biol Chem 277:25843-6 (2002)\n",
      "            DOI:10.1074/jbc.R200009200\n",
      "REFERENCE   PMID:10652340\n",
      "  AUTHORS   Tani M, Okino N, Mitsutake S, Tanigawa T, Izu H, Ito M.\n",
      "  TITLE     Purification and characterization of a neutral ceramidase from mouse liver. A single protein catalyzes the reversible reaction in which ceramide is both hydrolyzed and synthesized.\n",
      "  JOURNAL   J Biol Chem 275:3462-8 (2000)\n",
      "            DOI:10.1074/jbc.275.5.3462\n",
      "REFERENCE   PMID:11305904\n",
      "  AUTHORS   Hannun YA, Luberto C, Argraves KM.\n",
      "  TITLE     Enzymes of sphingolipid metabolism: from modular to integrative signaling.\n",
      "  JOURNAL   Biochemistry 40:4893-903 (2001)\n",
      "            DOI:10.1021/bi002836k\n",
      "REFERENCE   PMID:1495423\n",
      "  AUTHORS   Merrill AH Jr, Wang E.\n",
      "  TITLE     Enzymes of ceramide biosynthesis.\n",
      "  JOURNAL   Methods Enzymol 209:427-37 (1992)\n",
      "            DOI:10.1016/0076-6879(92)09053-6\n",
      "REFERENCE   PMID:8457204\n",
      "  AUTHORS   Hirschberg K, Rodger J, Futerman AH.\n",
      "  TITLE     The long-chain sphingoid base of sphingolipids is acylated at the cytosolic surface of the endoplasmic reticulum in rat liver.\n",
      "  JOURNAL   Biochem J 290 ( Pt 3):751-7 (1993)\n",
      "            DOI:10.1042/bj2900751\n",
      "REFERENCE   PMID:10366774\n",
      "  AUTHORS   Dickson RC, Lester RL.\n",
      "  TITLE     Metabolism and selected functions of sphingolipids in the yeast Saccharomyces cerevisiae.\n",
      "  JOURNAL   Biochim Biophys Acta 1438:305-21 (1999)\n",
      "            DOI:10.1016/S1388-1981(99)00068-2\n",
      "REL_PATHWAY pae00260  Glycine, serine and threonine metabolism\n",
      "KO_PATHWAY  ko00600\n",
      "///\n",
      "path:pae00620\n",
      "ENTRY       pae00620                    Pathway\n",
      "NAME        Pyruvate metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00620  Pyruvate metabolism\n",
      "MODULE      pae_M00307  Pyruvate oxidation, pyruvate => acetyl-CoA [PATH:pae00620]\n",
      "            pae_M00579  Phosphate acetyltransferase-acetate kinase pathway, acetyl-CoA => acetate [PATH:pae00620]\n",
      "DBLINKS     GO: 0006090\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0887  acsA; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA0887  acsA; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "            PA4733  acsB; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA4733  acsB; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "            PA2555  AMP-binding protein [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA2555  AMP-binding protein [KO:K01895] [EC:6.2.1.1]\n",
      "            PA5015  aceE; pyruvate dehydrogenase subunit E1 [KO:K00163] [EC:1.2.4.1]\n",
      "line in gene PA5015  aceE; pyruvate dehydrogenase subunit E1 [KO:K00163] [EC:1.2.4.1]\n",
      "            PA5016  aceF; dihydrolipoamide acetyltransferase [KO:K00627] [EC:2.3.1.12]\n",
      "line in gene PA5016  aceF; dihydrolipoamide acetyltransferase [KO:K00627] [EC:2.3.1.12]\n",
      "            PA4152  branched-chain alpha-keto acid dehydrogenase subunit E2 [KO:K00627] [EC:2.3.1.12]\n",
      "line in gene PA4152  branched-chain alpha-keto acid dehydrogenase subunit E2 [KO:K00627] [EC:2.3.1.12]\n",
      "            PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA5427  adhA; alcohol dehydrogenase [KO:K13953] [EC:1.1.1.1]\n",
      "line in gene PA5427  adhA; alcohol dehydrogenase [KO:K13953] [EC:1.1.1.1]\n",
      "            PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "line in gene PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "            PA2275  alcohol dehydrogenase [KO:K13979] [EC:1.1.1.2]\n",
      "line in gene PA2275  alcohol dehydrogenase [KO:K13979] [EC:1.1.1.2]\n",
      "            PA1982  exaA; quinoprotein ethanol dehydrogenase [KO:K00114] [EC:1.1.2.8]\n",
      "line in gene PA1982  exaA; quinoprotein ethanol dehydrogenase [KO:K00114] [EC:1.1.2.8]\n",
      "            PA2266  cytochrome C [KO:K22474] [EC:1.1.5.5]\n",
      "line in gene PA2266  cytochrome C [KO:K22474] [EC:1.1.5.5]\n",
      "            PA0836  ackA; acetate kinase [KO:K00925] [EC:2.7.2.1]\n",
      "line in gene PA0836  ackA; acetate kinase [KO:K00925] [EC:2.7.2.1]\n",
      "            PA0835  pta; phosphate acetyltransferase [KO:K13788] [EC:2.3.1.8]\n",
      "line in gene PA0835  pta; phosphate acetyltransferase [KO:K13788] [EC:2.3.1.8]\n",
      "            PA1498  pykF; pyruvate kinase [KO:K00873] [EC:2.7.1.40]\n",
      "line in gene PA1498  pykF; pyruvate kinase [KO:K00873] [EC:2.7.1.40]\n",
      "            PA4329  pykA; pyruvate kinase [KO:K00873] [EC:2.7.1.40]\n",
      "line in gene PA4329  pykA; pyruvate kinase [KO:K00873] [EC:2.7.1.40]\n",
      "            PA3639  accA; acetyl-CoA carboxylase carboxyltransferase subunit alpha [KO:K01962] [EC:6.4.1.2 2.1.3.15]\n",
      "line in gene PA3639  accA; acetyl-CoA carboxylase carboxyltransferase subunit alpha [KO:K01962] [EC:6.4.1.2 2.1.3.15]\n",
      "            PA4847  accB; acetyl-CoA carboxylase biotin carboxyl carrier protein subunit [KO:K02160]\n",
      "line in gene PA4847  accB; acetyl-CoA carboxylase biotin carboxyl carrier protein subunit [KO:K02160]\n",
      "            PA0493  acetyl-CoA carboxylase biotin carboxyl carrier protein subunit [KO:K02160]\n",
      "line in gene PA0493  acetyl-CoA carboxylase biotin carboxyl carrier protein subunit [KO:K02160]\n",
      "            PA4848  accC; acetyl-CoA carboxylase biotin carboxylase subunit [KO:K01961] [EC:6.4.1.2 6.3.4.14]\n",
      "line in gene PA4848  accC; acetyl-CoA carboxylase biotin carboxylase subunit [KO:K01961] [EC:6.4.1.2 6.3.4.14]\n",
      "            PA0494  acetyl-CoA carboxylase biotin carboxylase subunit [KO:K01961] [EC:6.4.1.2 6.3.4.14]\n",
      "line in gene PA0494  acetyl-CoA carboxylase biotin carboxylase subunit [KO:K01961] [EC:6.4.1.2 6.3.4.14]\n",
      "            PA3112  accD; acetyl-CoA carboxylase carboxyltransferase subunit beta [KO:K01963] [EC:6.4.1.2 2.1.3.15]\n",
      "line in gene PA3112  accD; acetyl-CoA carboxylase carboxyltransferase subunit beta [KO:K01963] [EC:6.4.1.2 2.1.3.15]\n",
      "            PA0954  acylphosphatase [KO:K01512] [EC:3.6.1.7]\n",
      "line in gene PA0954  acylphosphatase [KO:K01512] [EC:3.6.1.7]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA1984  exaC; NAD+ dependent aldehyde dehydrogenase ExaC [KO:K00138] [EC:1.2.1.-]\n",
      "line in gene PA1984  exaC; NAD+ dependent aldehyde dehydrogenase ExaC [KO:K00138] [EC:1.2.1.-]\n",
      "            PA4022  aldehyde dehydrogenase [KO:K00138] [EC:1.2.1.-]\n",
      "line in gene PA4022  aldehyde dehydrogenase [KO:K00138] [EC:1.2.1.-]\n",
      "            PA5445  coenzyme A transferase [KO:K18118] [EC:2.8.3.18]\n",
      "line in gene PA5445  coenzyme A transferase [KO:K18118] [EC:2.8.3.18]\n",
      "            PA1017  pauA; pimeloyl-CoA synthetase [KO:K24012] [EC:6.2.1.13]\n",
      "line in gene PA1017  pauA; pimeloyl-CoA synthetase [KO:K24012] [EC:6.2.1.13]\n",
      "            PA4978  hypothetical protein [KO:K24012] [EC:6.2.1.13]\n",
      "line in gene PA4978  hypothetical protein [KO:K24012] [EC:6.2.1.13]\n",
      "            PA2382  lldA; L-lactate dehydrogenase [KO:K00101] [EC:1.1.2.3]\n",
      "line in gene PA2382  lldA; L-lactate dehydrogenase [KO:K00101] [EC:1.1.2.3]\n",
      "            PA4771  lldD; L-lactate dehydrogenase [KO:K00101] [EC:1.1.2.3]\n",
      "line in gene PA4771  lldD; L-lactate dehydrogenase [KO:K00101] [EC:1.1.2.3]\n",
      "            PA5297  poxB; pyruvate dehydrogenase [KO:K00156] [EC:1.2.5.1]\n",
      "line in gene PA5297  poxB; pyruvate dehydrogenase [KO:K00156] [EC:1.2.5.1]\n",
      "            PA2108  thiamine pyrophosphate protein [KO:K00156] [EC:1.2.5.1]\n",
      "line in gene PA2108  thiamine pyrophosphate protein [KO:K00156] [EC:1.2.5.1]\n",
      "            PA0927  ldhA; D-lactate dehydrogenase [KO:K03778] [EC:1.1.1.28]\n",
      "line in gene PA0927  ldhA; D-lactate dehydrogenase [KO:K03778] [EC:1.1.1.28]\n",
      "            PA4772  ferredoxin [KO:K18930]\n",
      "line in gene PA4772  ferredoxin [KO:K18930]\n",
      "            PA3524  gloA1; lactoylglutathione lyase [KO:K01759] [EC:4.4.1.5]\n",
      "line in gene PA3524  gloA1; lactoylglutathione lyase [KO:K01759] [EC:4.4.1.5]\n",
      "            PA0710  gloA2; lactoylglutathione lyase [KO:K01759] [EC:4.4.1.5]\n",
      "line in gene PA0710  gloA2; lactoylglutathione lyase [KO:K01759] [EC:4.4.1.5]\n",
      "            PA5111  gloA3; lactoylglutathione lyase [KO:K01759] [EC:4.4.1.5]\n",
      "line in gene PA5111  gloA3; lactoylglutathione lyase [KO:K01759] [EC:4.4.1.5]\n",
      "            PA1813  hydroxyacylglutathione hydrolase [KO:K01069] [EC:3.1.2.6]\n",
      "line in gene PA1813  hydroxyacylglutathione hydrolase [KO:K01069] [EC:3.1.2.6]\n",
      "            PA0832  hypothetical protein [KO:K01069] [EC:3.1.2.6]\n",
      "line in gene PA0832  hypothetical protein [KO:K01069] [EC:3.1.2.6]\n",
      "            PA1135  molecular chaperone Hsp31/glyoxalase [KO:K05523] [EC:4.2.1.130 3.5.1.124]\n",
      "line in gene PA1135  molecular chaperone Hsp31/glyoxalase [KO:K05523] [EC:4.2.1.130 3.5.1.124]\n",
      "            PA3896  2-hydroxyacid dehydrogenase [KO:K00090] [EC:1.1.1.79 1.1.1.81 1.1.1.215]\n",
      "line in gene PA3896  2-hydroxyacid dehydrogenase [KO:K00090] [EC:1.1.1.79 1.1.1.81 1.1.1.215]\n",
      "            PA3471  NAD-dependent malic enzyme [KO:K00027] [EC:1.1.1.38]\n",
      "line in gene PA3471  NAD-dependent malic enzyme [KO:K00027] [EC:1.1.1.38]\n",
      "            PA5046  malic enzyme [KO:K00029] [EC:1.1.1.40]\n",
      "line in gene PA5046  malic enzyme [KO:K00029] [EC:1.1.1.40]\n",
      "            PA4872  hypothetical protein [KO:K01003] [EC:4.1.1.112]\n",
      "line in gene PA4872  hypothetical protein [KO:K01003] [EC:4.1.1.112]\n",
      "            PA5436  acetyl-CoA carboxylase subunit alpha [KO:K01959] [EC:6.4.1.1]\n",
      "line in gene PA5436  acetyl-CoA carboxylase subunit alpha [KO:K01959] [EC:6.4.1.1]\n",
      "            PA5435  pyruvate carboxylase subunit B [KO:K01960] [EC:6.4.1.1]\n",
      "line in gene PA5435  pyruvate carboxylase subunit B [KO:K01960] [EC:6.4.1.1]\n",
      "            PA3452  mqoA; malate:quinone oxidoreductase [KO:K00116] [EC:1.1.5.4]\n",
      "line in gene PA3452  mqoA; malate:quinone oxidoreductase [KO:K00116] [EC:1.1.5.4]\n",
      "            PA4640  mqoB; malate:quinone oxidoreductase [KO:K00116] [EC:1.1.5.4]\n",
      "line in gene PA4640  mqoB; malate:quinone oxidoreductase [KO:K00116] [EC:1.1.5.4]\n",
      "            PA4333  fumarase [KO:K01676] [EC:4.2.1.2]\n",
      "line in gene PA4333  fumarase [KO:K01676] [EC:4.2.1.2]\n",
      "            PA0854  fumC2; fumarate hydratase [KO:K01679] [EC:4.2.1.2]\n",
      "line in gene PA0854  fumC2; fumarate hydratase [KO:K01679] [EC:4.2.1.2]\n",
      "            PA4470  fumC1; fumarate hydratase [KO:K01679] [EC:4.2.1.2]\n",
      "line in gene PA4470  fumC1; fumarate hydratase [KO:K01679] [EC:4.2.1.2]\n",
      "            PA3687  ppc; phosphoenolpyruvate carboxylase [KO:K01595] [EC:4.1.1.31]\n",
      "line in gene PA3687  ppc; phosphoenolpyruvate carboxylase [KO:K01595] [EC:4.1.1.31]\n",
      "            PA5192  pckA; phosphoenolpyruvate carboxykinase [KO:K01610] [EC:4.1.1.49]\n",
      "line in gene PA5192  pckA; phosphoenolpyruvate carboxykinase [KO:K01610] [EC:4.1.1.49]\n",
      "            PA1770  ppsA; phosphoenolpyruvate synthase [KO:K01007] [EC:2.7.9.2]\n",
      "line in gene PA1770  ppsA; phosphoenolpyruvate synthase [KO:K01007] [EC:2.7.9.2]\n",
      "            PA0482  glcB; malate synthase G [KO:K01638] [EC:2.3.3.9]\n",
      "line in gene PA0482  glcB; malate synthase G [KO:K01638] [EC:2.3.3.9]\n",
      "            PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA1217  2-isopropylmalate synthase [KO:K01649] [EC:2.3.3.13]\n",
      "line in gene PA1217  2-isopropylmalate synthase [KO:K01649] [EC:2.3.3.13]\n",
      "            PA3792  leuA; 2-isopropylmalate synthase [KO:K01649] [EC:2.3.3.13]\n",
      "line in gene PA3792  leuA; 2-isopropylmalate synthase [KO:K01649] [EC:2.3.3.13]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00033  Acetate\n",
      "            C00036  Oxaloacetate\n",
      "            C00042  Succinate\n",
      "            C00058  Formate\n",
      "            C00068  Thiamin diphosphate\n",
      "            C00074  Phosphoenolpyruvate\n",
      "            C00083  Malonyl-CoA\n",
      "            C00084  Acetaldehyde\n",
      "            C00122  Fumarate\n",
      "            C00149  (S)-Malate\n",
      "            C00186  (S)-Lactate\n",
      "            C00227  Acetyl phosphate\n",
      "            C00256  (R)-Lactate\n",
      "            C00332  Acetoacetyl-CoA\n",
      "            C00424  (S)-Lactaldehyde\n",
      "            C00469  Ethanol\n",
      "            C00546  Methylglyoxal\n",
      "            C00937  (R)-Lactaldehyde\n",
      "            C01251  (R)-2-Hydroxybutane-1,2,4-tricarboxylate\n",
      "            C02488  (R)-2-Ethylmalate\n",
      "            C02504  alpha-Isopropylmalate\n",
      "            C03248  Acetylenedicarboxylate\n",
      "            C03451  (R)-S-Lactoylglutathione\n",
      "            C03981  2-Hydroxyethylenedicarboxylate\n",
      "            C05125  2-(alpha-Hydroxyethyl)thiamine diphosphate\n",
      "            C05993  Acetyl adenylate\n",
      "            C05994  2-Propylmalate\n",
      "            C15972  Enzyme N6-(lipoyl)lysine\n",
      "            C15973  Enzyme N6-(dihydrolipoyl)lysine\n",
      "            C16255  [Dihydrolipoyllysine-residue acetyltransferase] S-acetyldihydrolipoyllysine\n",
      "REFERENCE   \n",
      "  AUTHORS   Nishizuka Y, Seyama Y, Ikai A, Ishimura Y, Kawaguchi A (eds).\n",
      "  TITLE     [Cellular Functions and Metabolic Maps] (In Japanese)\n",
      "  JOURNAL   Tokyo Kagaku Dojin (1997)\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00061  Fatty acid biosynthesis\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00290  Valine, leucine and isoleucine biosynthesis\n",
      "            pae00300  Lysine biosynthesis\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00640  Propanoate metabolism\n",
      "            pae00650  Butanoate metabolism\n",
      "            pae00760  Nicotinate and nicotinamide metabolism\n",
      "KO_PATHWAY  ko00620\n",
      "///\n",
      "path:pae00622\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00622                    Pathway\n",
      "NAME        Xylene degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Xenobiotics biodegradation and metabolism\n",
      "PATHWAY_MAP pae00622  Xylene degradation\n",
      "MODULE      pae_M00551  Benzoate degradation, benzoate => catechol / methylbenzoate => methylcatechol [PATH:pae00622]\n",
      "DBLINKS     GO: 0042184\n",
      "            UMBBD: mxy oxy pxy pcy tol\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2518  xylX; toluate 1,2-dioxygenase subunit alpha [KO:K05549] [EC:1.14.12.10 1.14.12.-]\n",
      "line in gene PA2518  xylX; toluate 1,2-dioxygenase subunit alpha [KO:K05549] [EC:1.14.12.10 1.14.12.-]\n",
      "            PA2517  xylY; toluate 1,2-dioxygenase subunit beta [KO:K05550] [EC:1.14.12.10 1.14.12.-]\n",
      "line in gene PA2517  xylY; toluate 1,2-dioxygenase subunit beta [KO:K05550] [EC:1.14.12.10 1.14.12.-]\n",
      "            PA2516  xylZ; toluate 1,2-dioxygenase electron transfer subunit [KO:K05784] [EC:1.18.1.-]\n",
      "line in gene PA2516  xylZ; toluate 1,2-dioxygenase electron transfer subunit [KO:K05784] [EC:1.18.1.-]\n",
      "            PA2515  xylL; 1,6-dihydroxycyclohexa-2,4-diene-1-carboxylate dehydrogenase [KO:K05783] [EC:1.3.1.25 1.3.1.-]\n",
      "line in gene PA2515  xylL; 1,6-dihydroxycyclohexa-2,4-diene-1-carboxylate dehydrogenase [KO:K05783] [EC:1.3.1.25 1.3.1.-]\n",
      "            PA2083  ring-hydroxylating dioxygenase subunit [KO:K10619] [EC:1.14.12.25]\n",
      "line in gene PA2083  ring-hydroxylating dioxygenase subunit [KO:K10619] [EC:1.14.12.25]\n",
      "            PA2085  ring-hydroxylating dioxygenase small subunit [KO:K16303] [EC:1.14.12.25]\n",
      "line in gene PA2085  ring-hydroxylating dioxygenase small subunit [KO:K16303] [EC:1.14.12.25]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00084  Acetaldehyde\n",
      "            C00100  Propanoyl-CoA\n",
      "            C00479  Propanal\n",
      "            C00596  2-Hydroxy-2,4-pentadienoate\n",
      "            C01454  Toluate\n",
      "            C02923  2,3-Dihydroxytoluene\n",
      "            C03589  4-Hydroxy-2-oxopentanoate\n",
      "            C06210  2-Hydroxy-6-keto-2,4-heptadienoate\n",
      "            C06575  p-Cymene\n",
      "            C06576  p-Cumic alcohol\n",
      "            C06577  4-Isopropylbenzaldehyde\n",
      "            C06578  p-Cumate\n",
      "            C06579  cis-2,3-Dihydroxy-2,3-dihydro-p-cumate\n",
      "            C06580  2,3-Dihydroxy-p-cumate\n",
      "            C06581  2-Hydroxy-3-carboxy-6-oxo-7-methylocta-2,4-dienoate\n",
      "            C06582  2-Hydroxy-6-oxo-7-methylocta-2,4-dienoate\n",
      "            C06720  1,6-Dihydroxy-5-methylcyclohexa-2,4-dienecarboxylate\n",
      "            C06729  cis-1,2-Dihydroxy-4-methylcyclohexa-3,5-diene-1-carboxylate\n",
      "            C06730  4-Methylcatechol\n",
      "            C06731  1,2-Dihydroxy-6-methylcyclohexa-3,5-dienecarboxylate\n",
      "            C06756  p-Xylene\n",
      "            C06757  4-Methylbenzyl alcohol\n",
      "            C06758  p-Tolualdehyde\n",
      "            C06760  2-Hydroxy-5-methyl-cis,cis-muconic semialdehyde\n",
      "            C06762  4-Hydroxy-2-oxohexanoic acid\n",
      "            C07208  m-Xylene\n",
      "            C07209  3-Methylbenzaldehyde\n",
      "            C07211  m-Methylbenzoate\n",
      "            C07212  o-Xylene\n",
      "            C07213  2-Methylbenzyl alcohol\n",
      "            C07214  2-Methylbenzaldehyde\n",
      "            C07215  o-Toluate\n",
      "            C07216  3-Methylbenzyl alcohol\n",
      "            C07478  2-Hydroxy-5-methyl-cis,cis-muconate\n",
      "            C07479  2-Oxo-5-methyl-cis-muconate\n",
      "            C11354  2-Hydroxy-cis-hex-2,4-dienoate\n",
      "REFERENCE   PMID:11375157\n",
      "  AUTHORS   Haddad S, Eby DM, Neidle EL.\n",
      "  TITLE     Cloning and expression of the benzoate dioxygenase genes from Rhodococcus sp. strain 19070.\n",
      "  JOURNAL   Appl Environ Microbiol 67:2507-14 (2001)\n",
      "            DOI:10.1128/AEM.67.6.2507-2514.2001\n",
      "REFERENCE   PMID:7768806\n",
      "  AUTHORS   Kim E, Zylstra GJ\n",
      "  TITLE     Molecular and biochemical characterization of two meta-cleavage dioxygenases involved in biphenyl and m-xylene degradation by Beijerinckia sp. strain B1.\n",
      "  JOURNAL   J Bacteriol 177:3095-103 (1995)\n",
      "            DOI:10.1128/JB.177.11.3095-3103.1995\n",
      "REFERENCE   PMID:15127183\n",
      "  AUTHORS   Morasch B, Schink B, Tebbe CC, Meckenstock RU\n",
      "  TITLE     Degradation of o-xylene and m-xylene by a novel sulfate-reducer belonging to the genus Desulfotomaculum.\n",
      "  JOURNAL   Arch Microbiol 181:407-17 (2004)\n",
      "            DOI:10.1007/s00203-004-0672-6\n",
      "REFERENCE   PMID:9150211\n",
      "  AUTHORS   Eaton RW.\n",
      "  TITLE     p-Cymene catabolic pathway in Pseudomonas putida F1: cloning and characterization of DNA encoding conversion of p-cymene to p-cumate.\n",
      "  JOURNAL   J Bacteriol 179:3171-80 (1997)\n",
      "            DOI:10.1128/JB.179.10.3171-3180.1997\n",
      "REFERENCE   PMID:8631713\n",
      "  AUTHORS   Eaton RW.\n",
      "  TITLE     p-Cumate catabolic pathway in Pseudomonas putida Fl: cloning and characterization of DNA carrying the cmt operon.\n",
      "  JOURNAL   J Bacteriol 178:1351-62 (1996)\n",
      "            DOI:10.1128/JB.178.5.1351-1362.1996\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00623  Toluene degradation\n",
      "            pae00640  Propanoate metabolism\n",
      "KO_PATHWAY  ko00622\n",
      "///\n",
      "path:pae00623\n",
      "ENTRY       pae00623                    Pathway\n",
      "NAME        Toluene degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Xenobiotics biodegradation and metabolism\n",
      "PATHWAY_MAP pae00623  Toluene degradation\n",
      "DBLINKS     GO: 0042203\n",
      "            UMBBD: tol tol2 tsa dct\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2507  catA; catechol 1,2-dioxygenase [KO:K03381] [EC:1.13.11.1]\n",
      "line in gene PA2507  catA; catechol 1,2-dioxygenase [KO:K03381] [EC:1.13.11.1]\n",
      "            PA2509  catB; muconate cycloisomerase I [KO:K01856] [EC:5.5.1.1]\n",
      "line in gene PA2509  catB; muconate cycloisomerase I [KO:K01856] [EC:5.5.1.1]\n",
      "            PA2682  hypothetical protein [KO:K01061] [EC:3.1.1.45]\n",
      "line in gene PA2682  hypothetical protein [KO:K01061] [EC:3.1.1.45]\n",
      "COMPOUND    C00091  Succinyl-CoA\n",
      "            C00156  4-Hydroxybenzoate\n",
      "            C00180  Benzoate\n",
      "            C00230  3,4-Dihydroxybenzoate\n",
      "            C00261  Benzaldehyde\n",
      "            C00512  Benzoyl-CoA\n",
      "            C00556  Benzyl alcohol\n",
      "            C00587  3-Hydroxybenzoate\n",
      "            C00633  4-Hydroxybenzaldehyde\n",
      "            C01455  Toluene\n",
      "            C01467  3-Cresol\n",
      "            C01468  4-Cresol\n",
      "            C01542  o-Cresol\n",
      "            C02236  4-Sulfobenzoate\n",
      "            C02923  2,3-Dihydroxytoluene\n",
      "            C03067  3-Hydroxybenzaldehyde\n",
      "            C03351  3-Hydroxybenzyl alcohol\n",
      "            C04112  3-Methyl-cis,cis-hexadienedioate\n",
      "            C04558  3-Methylmuconolactone\n",
      "            C04559  4-Methylmuconolactone\n",
      "            C04592  Toluene-cis-dihydrodiol\n",
      "            C06677  Toluene-4-sulfonate\n",
      "            C06678  4-Sulfobenzyl alcohol\n",
      "            C06679  4-Sulfobenzaldehyde\n",
      "            C06730  4-Methylcatechol\n",
      "            C09816  Benzylsuccinate\n",
      "            C09817  Benzylsuccinyl-CoA\n",
      "            C09818  E-Phenylitaconyl-CoA\n",
      "            C09819  (Hydroxymethylphenyl)succinyl-CoA\n",
      "            C09820  Benzoylsuccinyl-CoA\n",
      "            C14582  2,4-Dimethylphenol\n",
      "            C18300  2,4-Dichlorotoluene\n",
      "            C18301  4,6-Dichloro-3-methyl-cis-1,2-dihydroxycyclohexa-3,5-diene\n",
      "            C18302  4,6-Dichloro-3-methylcatechol\n",
      "            C18303  3,5-Dichloro-2-methylmuconate\n",
      "            C18304  2-Chloro-5-methyl-cis-dienelactone\n",
      "            C18305  2-Chloro-5-methylmaleylacetate\n",
      "            C18306  5-Methylmaleylacetate\n",
      "            C18307  2-Methyl-3-oxoadipate\n",
      "            C18308  3,5-Dichloro-2-methylmuconolactone\n",
      "            C18309  3-Chloro-2-methyldienelactone\n",
      "            C18310  3-Chloro-2-methylmaleylacetate\n",
      "            C18311  4-Methyl-3-oxoadipate-enol-lactone\n",
      "            C18312  4-Methyl-3-oxoadipate\n",
      "            C21166  4-Hydroxy-3-methylbenzaldehyde\n",
      "            C21167  4-Hydroxy-3-methylbenzoate\n",
      "            C21168  4-Hydroxyisophthalate\n",
      "REFERENCE   PMID:15184119\n",
      "  AUTHORS   Vardar G, Wood TK\n",
      "  TITLE     Protein engineering of toluene-o-xylene monooxygenase from Pseudomonas stutzeri OX1 for synthesizing 4-methylresorcinol, methylhydroquinone, and pyrogallol.\n",
      "  JOURNAL   Appl Environ Microbiol 70:3253-62 (2004)\n",
      "            DOI:10.1128/AEM.70.6.3253-3262.2004\n",
      "REFERENCE   PMID:10629170\n",
      "  AUTHORS   Leuthner B, Heider J\n",
      "  TITLE     Anaerobic toluene catabolism of Thauera aromatica: the bbs operon codes for enzymes of beta oxidation of the intermediate benzylsuccinate.\n",
      "  JOURNAL   J Bacteriol 182:272-7 (2000)\n",
      "            DOI:10.1128/JB.182.2.272-277.2000\n",
      "REFERENCE   PMID:11418570\n",
      "  AUTHORS   Leutwein C, Heider J\n",
      "  TITLE     Succinyl-CoA:(R)-benzylsuccinate CoA-transferase: an enzyme of the anaerobic toluene catabolic pathway in denitrifying bacteria.\n",
      "  JOURNAL   J Bacteriol 183:4288-95 (2001)\n",
      "            DOI:10.1128/JB.183.14.4288-4295.2001\n",
      "REFERENCE   PMID:15687214\n",
      "  AUTHORS   Kuhner S, Wohlbrand L, Fritz I, Wruck W, Hultschig C, Hufnagel P, Kube M, Reinhardt R, Rabus R\n",
      "  TITLE     Substrate-dependent regulation of anaerobic degradation pathways for toluene and ethylbenzene in a denitrifying bacterium, strain EbN1.\n",
      "  JOURNAL   J Bacteriol 187:1493-503 (2005)\n",
      "            DOI:10.1128/JB.187.4.1493-1503.2005\n",
      "REFERENCE   PMID:2050632\n",
      "  AUTHORS   Locher HH, Leisinger T, Cook AM\n",
      "  TITLE     4-Toluene sulfonate methyl-monooxygenase from Comamonas testosteroni T-2: purification and some properties of the oxygenase component.\n",
      "  JOURNAL   J Bacteriol 173:3741-8 (1991)\n",
      "            DOI:10.1128/JB.173.12.3741-3748.1991\n",
      "REFERENCE   PMID:8828208\n",
      "  AUTHORS   Junker F, Saller E, Schlafli Oppenberg HR, Kroneck PM, Leisinger T, Cook AM\n",
      "  TITLE     Degradative pathways for p-toluenecarboxylate and p-toluenesulfonate and their multicomponent oxygenases in Comamonas testosteroni strains PSB-4 and T-2.\n",
      "  JOURNAL   Microbiology 142 ( Pt 9):2419-27 (1996)\n",
      "            DOI:10.1099/00221287-142-9-2419\n",
      "REFERENCE   PMID:10589735\n",
      "  AUTHORS   Mampel J, Ruff J, Junker F, Cook AM.\n",
      "  TITLE     The oxygenase component of the 2-aminobenzenesulfonate dioxygenase system from Alcaligenes sp. strain O-1.\n",
      "  JOURNAL   Microbiology 145 ( Pt 11):3255-64 (1999)\n",
      "            DOI:10.1099/00221287-145-11-3255\n",
      "REFERENCE   PMID:9006050\n",
      "  AUTHORS   Junker F, Kiewitz R, Cook AM\n",
      "  TITLE     Characterization of the p-toluenesulfonate operon tsaMBCD and tsaR in Comamonas testosteroni T-2.\n",
      "  JOURNAL   J Bacteriol 179:919-27 (1997)\n",
      "            DOI:10.1128/JB.179.3.919-927.1997\n",
      "REL_PATHWAY pae00362  Benzoate degradation\n",
      "            pae00622  Xylene degradation\n",
      "KO_PATHWAY  ko00623\n",
      "///\n",
      "path:pae00625\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00625                    Pathway\n",
      "NAME        Chloroalkane and chloroalkene degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Xenobiotics biodegradation and metabolism\n",
      "PATHWAY_MAP pae00625  Chloroalkane and chloroalkene degradation\n",
      "DBLINKS     UMBBD: tce tce2 tca atl dcm dce cpr caa\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA5427  adhA; alcohol dehydrogenase [KO:K13953] [EC:1.1.1.1]\n",
      "line in gene PA5427  adhA; alcohol dehydrogenase [KO:K13953] [EC:1.1.1.1]\n",
      "            PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "line in gene PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA5421  fdhA; glutathione-independent formaldehyde dehydrogenase [KO:K00148] [EC:1.2.1.46]\n",
      "line in gene PA5421  fdhA; glutathione-independent formaldehyde dehydrogenase [KO:K00148] [EC:1.2.1.46]\n",
      "            PA1982  exaA; quinoprotein ethanol dehydrogenase [KO:K00114] [EC:1.1.2.8]\n",
      "line in gene PA1982  exaA; quinoprotein ethanol dehydrogenase [KO:K00114] [EC:1.1.2.8]\n",
      "            PA2086  epoxide hydrolase [KO:K01561] [EC:3.8.1.3]\n",
      "line in gene PA2086  epoxide hydrolase [KO:K01561] [EC:3.8.1.3]\n",
      "            PA0810  haloacid dehalogenase [KO:K01560] [EC:3.8.1.2]\n",
      "line in gene PA0810  haloacid dehalogenase [KO:K01560] [EC:3.8.1.2]\n",
      "COMPOUND    C00011  CO2\n",
      "            C00024  Acetyl-CoA\n",
      "            C00048  Glyoxylate\n",
      "            C00058  Formate\n",
      "            C00067  Formaldehyde\n",
      "            C00084  Acetaldehyde\n",
      "            C00160  Glycolate\n",
      "            C00209  Oxalate\n",
      "            C00222  3-Oxopropanoate\n",
      "            C00237  CO\n",
      "            C01380  Ethylene glycol\n",
      "            C01548  Acetylene\n",
      "            C02271  Dichloromethane\n",
      "            C06547  Ethylene\n",
      "            C06548  Ethylene oxide\n",
      "            C06609  trans-1,3-Dichloropropene\n",
      "            C06610  cis-1,3-Dichloropropene\n",
      "            C06611  trans-3-Chloro-2-propene-1-ol\n",
      "            C06612  cis-3-Chloro-2-propene-1-ol\n",
      "            C06613  trans-3-Chloroallyl aldehyde\n",
      "            C06614  trans-3-Chloroacrylic acid\n",
      "            C06615  cis-3-Chloroacrylic acid\n",
      "            C06752  1,2-Dichloroethane\n",
      "            C06753  2-Chloroethanol\n",
      "            C06754  Chloroacetaldehyde\n",
      "            C06755  Chloroacetic acid\n",
      "            C06789  Tetrachloroethene\n",
      "            C06790  Trichloroethene\n",
      "            C06791  trans-1,2-Dichloroethene\n",
      "            C06792  cis-1,2-Dichloroethene\n",
      "            C06793  Vinyl chloride\n",
      "            C07490  Trichloroethanol\n",
      "            C11148  TCE epoxide\n",
      "            C11149  Dichloroacetate\n",
      "            C11150  Trichloroacetate\n",
      "            C14866  Chloral\n",
      "            C16348  cis-3-Chloroallyl aldehyde\n",
      "            C18246  1,1,1-Trichloroethane\n",
      "            C18247  1,1-Dichloroethane\n",
      "            C18248  Chloroethane\n",
      "            C20303  Chloroethylene oxide\n",
      "            C21752  2-Hydroxyethyl-CoM\n",
      "            C21753  2-Chloro-2-hydroxyethyl-CoM\n",
      "REFERENCE   PMID:15294827\n",
      "  AUTHORS   Muller JA, Rosner BM, Von Abendroth G, Meshulam-Simon G, McCarty PL, Spormann AM.\n",
      "  TITLE     Molecular identification of the catabolic vinyl chloride reductase from Dehalococcoides sp. strain VS and its environmental distribution.\n",
      "  JOURNAL   Appl Environ Microbiol 70:4880-8 (2004)\n",
      "            DOI:10.1128/AEM.70.8.4880-4888.2004\n",
      "REFERENCE   PMID:16957242\n",
      "  AUTHORS   Lee PK, Johnson DR, Holmes VF, He J, Alvarez-Cohen L.\n",
      "  TITLE     Reductive dehalogenase gene expression as a biomarker for physiological activity of Dehalococcoides spp.\n",
      "  JOURNAL   Appl Environ Microbiol 72:6161-8 (2006)\n",
      "            DOI:10.1128/AEM.01070-06\n",
      "REFERENCE   PMID:15865156\n",
      "  AUTHORS   Torz M, Beschkov V\n",
      "  TITLE     Biodegradation of monochloroacetic acid used as a sole carbon and energy source by Xanthobacter autotrophicus GJ10 strain in batch and continuous culture.\n",
      "  JOURNAL   Biodegradation 16:423-33 (2005)\n",
      "            DOI:10.1007/s10532-004-3614-8\n",
      "REFERENCE   PMID:10347028\n",
      "  AUTHORS   Hage JC, Hartmans S\n",
      "  TITLE     Monooxygenase-mediated 1,2-dichloroethane degradation by Pseudomonas sp. strain DCA1.\n",
      "  JOURNAL   Appl Environ Microbiol 65:2466-70 (1999)\n",
      "            DOI:10.1128/AEM.65.6.2466-2470.1999\n",
      "REFERENCE   PMID:9687453\n",
      "  AUTHORS   Poelarends GJ, Wilkens M, Larkin MJ, van Elsas JD, Janssen DB.\n",
      "  TITLE     Degradation of 1,3-dichloropropene by pseudomonas cichorii 170.\n",
      "  JOURNAL   Appl Environ Microbiol 64:2931-6 (1998)\n",
      "            DOI:10.1128/AEM.64.8.2931-2936.1998\n",
      "REFERENCE   PMID:12411705\n",
      "  AUTHORS   Sun B, Griffin BM, Ayala-del-Rio HL, Hashsham SA, Tiedje JM\n",
      "  TITLE     Microbial dehalorespiration with 1,1,1-trichloroethane.\n",
      "  JOURNAL   Science 298:1023-5 (2002)\n",
      "            DOI:10.1126/science.1074675\n",
      "REFERENCE   PMID:11999413\n",
      "  AUTHORS   Hashimoto A, Iwasaki K, Nakasugi N, Nakajima M, Yagi O\n",
      "  TITLE     Degradation pathways of trichloroethylene and 1,1,1-trichloroethane by Mycobacterium sp. TA27.\n",
      "  JOURNAL   Biosci Biotechnol Biochem 66:385-90 (2002)\n",
      "            DOI:10.1271/bbb.66.385\n",
      "REFERENCE   PMID:1768109\n",
      "  AUTHORS   Newman LM, Wackett LP\n",
      "  TITLE     Fate of 2,2,2-trichloroacetaldehyde (chloral hydrate) produced during trichloroethylene oxidation by methanotrophs.\n",
      "  JOURNAL   Appl Environ Microbiol 57:2399-402 (1991)\n",
      "            DOI:10.1128/AEM.57.8.2399-2402.1991\n",
      "REFERENCE   PMID:8981984\n",
      "  AUTHORS   Newman LM, Wackett LP\n",
      "  TITLE     Trichloroethylene oxidation by purified toluene 2-monooxygenase: products, kinetics, and turnover-dependent inactivation.\n",
      "  JOURNAL   J Bacteriol 179:90-6 (1997)\n",
      "            DOI:10.1128/JB.179.1.90-96.1997\n",
      "REL_PATHWAY pae00620  Pyruvate metabolism\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00680  Methane metabolism\n",
      "KO_PATHWAY  ko00625\n",
      "///\n",
      "path:pae00626\n",
      "ENTRY       pae00626                    Pathway\n",
      "NAME        Naphthalene degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Xenobiotics biodegradation and metabolism\n",
      "PATHWAY_MAP pae00626  Naphthalene degradation\n",
      "DBLINKS     UMBBD: naph nphs 1mna npa 2mna 2mna2\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA5427  adhA; alcohol dehydrogenase [KO:K13953] [EC:1.1.1.1]\n",
      "line in gene PA5427  adhA; alcohol dehydrogenase [KO:K13953] [EC:1.1.1.1]\n",
      "            PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "line in gene PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "COMPOUND    C00090  Catechol\n",
      "            C00091  Succinyl-CoA\n",
      "            C00628  2,5-Dihydroxybenzoate\n",
      "            C00805  Salicylate\n",
      "            C00829  Naphthalene\n",
      "            C02909  (2-Naphthyl)methanol\n",
      "            C02923  2,3-Dihydroxytoluene\n",
      "            C03012  Naphthalene-1,2-diol\n",
      "            C03203  1-Hydroxy-2-naphthoate\n",
      "            C04314  cis-1,2-Dihydronaphthalene-1,2-diol\n",
      "            C06202  Salicylaldehyde\n",
      "            C06203  trans-o-Hydroxybenzylidenepyruvate\n",
      "            C06204  2-Hydroxychromene-2-carboxylate\n",
      "            C06730  4-Methylcatechol\n",
      "            C14082  1-Methylnaphthalene\n",
      "            C14083  cis-1,2-Dihydroxy-1,2-dihydro-8-methylnaphthalene\n",
      "            C14084  1,2-Dihydroxy-8-methylnaphthalene\n",
      "            C14085  2-Hydroxy-8-methylchromene-2-carboxylate\n",
      "            C14086  2-Hydroxy-3-methylbenzalpyruvate\n",
      "            C14087  3-Methylsalicylaldehyde\n",
      "            C14088  3-Methylsalicylate\n",
      "            C14089  1-Hydroxymethylnaphthalene\n",
      "            C14090  1-Naphthaldehyde\n",
      "            C14091  1-Naphthoic acid\n",
      "            C14092  cis-1,2-Dihydroxy-1,2-dihydro-8-carboxynaphthalene\n",
      "            C14093  1,2-Dihydroxy-8-carboxynaphthalene\n",
      "            C14094  2-Carboxy-2-hydroxy-8-carboxychromene\n",
      "            C14095  2-Hydroxy-3-carboxybenzalpyruvate\n",
      "            C14096  3-Formylsalicylic acid\n",
      "            C14097  2-Hydroxyisophthalic acid\n",
      "            C14098  2-Methylnaphthalene\n",
      "            C14099  2-Naphthaldehyde\n",
      "            C14100  4-Formylsalicylic acid\n",
      "            C14101  2-Naphthoic acid\n",
      "            C14102  cis-1,2-Dihydroxy-1,2-dihydro-7-methylnaphthalene\n",
      "            C14103  4-Methylsalicylate\n",
      "            C14104  cis-1,2-Dihydroxy-1,2-dihydro-7-hydroxymethylnaphthalene\n",
      "            C14105  1,2-Dihydroxy-7-hydroxymethylnaphthalene\n",
      "            C14106  2-Hydroxy-7-hydroxymethylchromene-2-carboxylate\n",
      "            C14107  2-Hydroxy-4-hydroxymethylbenzalpyruvate\n",
      "            C14108  4-Hydroxymethylsalicylaldehyde\n",
      "            C14109  4-Hydroxymethylsalicylate\n",
      "            C14110  4-Hydroxymethylcatechol\n",
      "            C14111  5,6,7,8-Tetrahydro-2-naphthoic acid\n",
      "            C14112  cis-2-Carboxycyclohexyl-acetic acid\n",
      "            C14113  Decahydro-2-naphthoic acid\n",
      "            C14114  Tetralin\n",
      "            C14115  2-Naphtylmethylsuccinic acid\n",
      "            C14116  Naphthyl-2-methyl-succinyl-CoA\n",
      "            C14117  Naphthyl-2-methylene-succinyl-CoA\n",
      "            C14118  Naphthyl-2-hydroxymethyl-succinyl CoA\n",
      "            C14119  Naphthyl-2-oxomethyl-succinyl-CoA\n",
      "            C14120  2-Naphthoyl-CoA\n",
      "            C16193  1,6-Naphthalenedisulfonic acid\n",
      "            C16194  2,6-Naphthalenedisulfonic acid\n",
      "            C16196  1,2-Dihydroxynaphthalene-6-sulfonate\n",
      "            C16198  (Z)-4-(2-Hydroxy-5-sulfonatophenyl)-2-oxo-3-butenoate\n",
      "            C16199  5-Sulfosalicylate\n",
      "            C16201  1-Naphthalenesulfonic acid\n",
      "            C16202  2-Naphthalenesulfonic acid\n",
      "REFERENCE   PMID:16461648\n",
      "  AUTHORS   Kim SJ, Kweon O, Freeman JP, Jones RC, Adjei MD, Jhoo JW, Edmondson RD, Cerniglia CE.\n",
      "  TITLE     Molecular cloning and expression of genes encoding a novel dioxygenase involved in low- and high-molecular-weight polycyclic aromatic hydrocarbon degradation in Mycobacterium vanbaalenii PYR-1.\n",
      "  JOURNAL   Appl Environ Microbiol 72:1045-54 (2006)\n",
      "            DOI:10.1128/AEM.72.2.1045-1054.2006\n",
      "REFERENCE   PMID:8226631\n",
      "  AUTHORS   Denome SA, Stanley DC, Olson ES, Young KD\n",
      "  TITLE     Metabolism of dibenzothiophene and naphthalene in Pseudomonas strains: complete DNA sequence of an upper naphthalene catabolic pathway.\n",
      "  JOURNAL   J Bacteriol 175:6890-901 (1993)\n",
      "            DOI:10.1128/JB.175.21.6890-6901.1993\n",
      "REFERENCE   PMID:1447127\n",
      "  AUTHORS   Eaton RW, Chapman PJ\n",
      "  TITLE     Bacterial metabolism of naphthalene: construction and use of recombinant bacteria to study ring cleavage of 1,2-dihydroxynaphthalene and subsequent reactions.\n",
      "  JOURNAL   J Bacteriol 174:7542-54 (1992)\n",
      "            DOI:10.1128/JB.174.23.7542-7554.1992\n",
      "REFERENCE   PMID:8981987\n",
      "  AUTHORS   Morawski B, Eaton RW, Rossiter JT, Guoping S, Griengl H, Ribbons DW\n",
      "  TITLE     2-Naphthoate catabolic pathway in Burkholderia strain JT 1500.\n",
      "  JOURNAL   J Bacteriol 179:115-21 (1997)\n",
      "            DOI:10.1128/JB.179.1.115-121.1997\n",
      "REFERENCE   PMID:15500985\n",
      "  AUTHORS   Safinowski M, Meckenstock RU\n",
      "  TITLE     Enzymatic reactions in anaerobic 2-methylnaphthalene degradation by the sulphate-reducing enrichment culture N 47.\n",
      "  JOURNAL   FEMS Microbiol Lett 240:99-104 (2004)\n",
      "            DOI:10.1016/j.femsle.2004.09.014\n",
      "REFERENCE   PMID:11097910\n",
      "  AUTHORS   Annweiler E, Materna A, Safinowski M, Kappler A, Richnow HH, Michaelis W, Meckenstock RU\n",
      "  TITLE     Anaerobic degradation of 2-methylnaphthalene by a sulfate-reducing enrichment culture.\n",
      "  JOURNAL   Appl Environ Microbiol 66:5329-33 (2000)\n",
      "            DOI:10.1128/AEM.66.12.5329-5333.2000\n",
      "REFERENCE   PMID:11823228\n",
      "  AUTHORS   Annweiler E, Michaelis W, Meckenstock RU\n",
      "  TITLE     Identical ring cleavage products during anaerobic degradation of naphthalene, 2-methylnaphthalene, and tetralin indicate a new metabolic pathway.\n",
      "  JOURNAL   Appl Environ Microbiol 68:852-8 (2002)\n",
      "            DOI:10.1128/AEM.68.2.852-858.2002\n",
      "REFERENCE   PMID:15372028\n",
      "  AUTHORS   Aitken CM, Jones DM, Larter SR\n",
      "  TITLE     Anaerobic hydrocarbon biodegradation in deep subsurface oil reservoirs.\n",
      "  JOURNAL   Nature 431:291-4 (2004)\n",
      "            DOI:10.1038/nature02922\n",
      "REFERENCE   PMID:19854898\n",
      "  AUTHORS   Selesi D, Jehmlich N, von Bergen M, Schmidt F, Rattei T, Tischler P, Lueders T, Meckenstock RU\n",
      "  TITLE     Combined genomic and proteomic approaches identify gene clusters involved in anaerobic 2-methylnaphthalene degradation in the sulfate-reducing enrichment culture N47.\n",
      "  JOURNAL   J Bacteriol 192:295-306 (2010)\n",
      "            DOI:10.1128/JB.00874-09\n",
      "REL_PATHWAY pae00350  Tyrosine metabolism\n",
      "            pae00362  Benzoate degradation\n",
      "            pae00622  Xylene degradation\n",
      "KO_PATHWAY  ko00626\n",
      "///\n",
      "path:pae00627\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00627                    Pathway\n",
      "NAME        Aminobenzoate degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Xenobiotics biodegradation and metabolism\n",
      "PATHWAY_MAP pae00627  Aminobenzoate degradation\n",
      "MODULE      pae_M00637  Anthranilate degradation, anthranilate => catechol [PATH:pae00627]\n",
      "DBLINKS     GO: 0043421\n",
      "            UMBBD: dcz 2,4-d pcp 2,4,5-t box dca\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0704  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA0704  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA0202  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA0202  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA3366  amiE; acylamide amidohydrolase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA3366  amiE; acylamide amidohydrolase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA4342  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA4342  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA4163  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA4163  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA0954  acylphosphatase [KO:K01512] [EC:3.6.1.7]\n",
      "line in gene PA0954  acylphosphatase [KO:K01512] [EC:3.6.1.7]\n",
      "            PA4901  mdlC; benzoylformate decarboxylase [KO:K01576] [EC:4.1.1.7]\n",
      "line in gene PA4901  mdlC; benzoylformate decarboxylase [KO:K01576] [EC:4.1.1.7]\n",
      "            PA4019  aromatic acid decarboxylase [KO:K03186] [EC:2.5.1.129]\n",
      "line in gene PA4019  aromatic acid decarboxylase [KO:K03186] [EC:2.5.1.129]\n",
      "            PA2512  antA; anthranilate dioxygenase large subunit [KO:K05599] [EC:1.14.12.1]\n",
      "line in gene PA2512  antA; anthranilate dioxygenase large subunit [KO:K05599] [EC:1.14.12.1]\n",
      "            PA2513  antB; anthranilate dioxygenase small subunit [KO:K05600] [EC:1.14.12.1]\n",
      "line in gene PA2513  antB; anthranilate dioxygenase small subunit [KO:K05600] [EC:1.14.12.1]\n",
      "            PA2514  antC; anthranilate dioxygenase reductase [KO:K11311] [EC:1.18.1.-]\n",
      "line in gene PA2514  antC; anthranilate dioxygenase reductase [KO:K11311] [EC:1.18.1.-]\n",
      "            PA4904  vanA; vanillate O-demethylase oxygenase [KO:K03862] [EC:1.14.13.82]\n",
      "line in gene PA4904  vanA; vanillate O-demethylase oxygenase [KO:K03862] [EC:1.14.13.82]\n",
      "            PA4905  vanB; vanillate O-demethylase [KO:K03863]\n",
      "line in gene PA4905  vanB; vanillate O-demethylase [KO:K03863]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "COMPOUND    C00090  Catechol\n",
      "            C00108  Anthranilate\n",
      "            C00146  Phenol\n",
      "            C00156  4-Hydroxybenzoate\n",
      "            C00180  Benzoate\n",
      "            C00196  2,3-Dihydroxybenzoate\n",
      "            C00230  3,4-Dihydroxybenzoate\n",
      "            C00261  Benzaldehyde\n",
      "            C00292  Aniline\n",
      "            C00472  p-Benzoquinone\n",
      "            C00512  Benzoyl-CoA\n",
      "            C00530  Hydroquinone\n",
      "            C00568  4-Aminobenzoate\n",
      "            C00632  3-Hydroxyanthranilate\n",
      "            C00633  4-Hydroxybenzaldehyde\n",
      "            C00755  4-Hydroxy-3-methoxy-benzaldehyde\n",
      "            C00870  4-Nitrophenol\n",
      "            C00877  Crotonoyl-CoA\n",
      "            C01108  1,2,3-Trihydroxybenzene\n",
      "            C01144  (S)-3-Hydroxybutanoyl-CoA\n",
      "            C01424  Gallate\n",
      "            C01502  o-Methoxyphenol\n",
      "            C01983  (R)-Mandelate\n",
      "            C01984  (S)-Mandelate\n",
      "            C01987  2-Aminophenol\n",
      "            C02137  alpha-Oxo-benzeneacetic acid\n",
      "            C02183  Phloroglucinol\n",
      "            C02222  2-Maleylacetate\n",
      "            C02235  4-Nitrocatechol\n",
      "            C02247  Anthraniloyl-CoA\n",
      "            C02372  4-Hydroxyaniline\n",
      "            C02720  Hydroxylaminobenzene\n",
      "            C02734  Phenolic phosphate\n",
      "            C02814  Benzene-1,2,4-triol\n",
      "            C02949  4-Hydroxybenzoyl-CoA\n",
      "            C03198  (S)-4-Hydroxymandelate\n",
      "            C03360  4-Nitrophenyl phosphate\n",
      "            C03590  4-Hydroxyphenylglyoxylate\n",
      "            C03671  2-Pyrone-4,6-dicarboxylate\n",
      "            C03824  2-Aminomuconate semialdehyde\n",
      "            C04434  (1E)-4-Oxobut-1-ene-1,2,4-tricarboxylate\n",
      "            C04484  4-Carboxy-2-hydroxymuconate semialdehyde\n",
      "            C05165  2-Amino-5-oxocyclohex-1-enecarbonyl-CoA\n",
      "            C05616  3-O-Methylgallate\n",
      "            C06206  Benzoyl phosphate\n",
      "            C06317  Vanillyl alcohol\n",
      "            C06335  4-Aminobenzenesulfonate\n",
      "            C06337  Terephthalate\n",
      "            C06604  Parathion\n",
      "            C06605  Aminoparathion\n",
      "            C06606  Paraoxon\n",
      "            C06607  Diethylthiophosphoric acid\n",
      "            C06608  Diethylphosphoric acid\n",
      "            C06672  Vanillate\n",
      "            C06673  4-Carboxy-4'-sulfoazobenzene\n",
      "            C06674  4-Sulfocatechol\n",
      "            C06675  3-Sulfomuconate\n",
      "            C06676  4-Sulfolactone\n",
      "            C06719  Dihydrophloroglucinol\n",
      "            C06813  Nitrobenzene\n",
      "            C06876  Nitrosobenzene\n",
      "            C07103  2-Hydroxy-1,4-benzoquinone\n",
      "            C07301  (R)-Mandelamide\n",
      "            C08063  1,4-Cyclohexanedione\n",
      "            C09814  Benzonitrile\n",
      "            C09815  Benzamide\n",
      "            C10833  Syringic acid\n",
      "            C14418  3-Nitrophenol\n",
      "            C14602  3-Hydroxyaminophenol\n",
      "            C14604  Aminohydroquinone\n",
      "            C16200  Phenylboronic acid\n",
      "            C16234  o-Nitrobenzoate\n",
      "            C16235  o-Hydroxylaminobenzoate\n",
      "            C16267  Cyclopropanecarboxylate\n",
      "            C16268  Cyclopropanecarboxyl-CoA\n",
      "            C16272  3-Hydroxy-5-oxohexanoate\n",
      "            C16273  3-Hydroxy-5-oxohexanoyl-CoA\n",
      "            C16281  Thiobenzamide\n",
      "            C16283  Thiobenzamide S-oxide\n",
      "            C16285  Thiobenzamide S,S-dioxide\n",
      "            C18338  5-Carboxyvanillic acid\n",
      "            C18345  4-Carboxy-2-hydroxy-6-methoxy-6-oxohexa-2,4-dienoate\n",
      "            C18347  5,5'-Dehydrodivanillate\n",
      "            C18348  2,2',3-Trihydroxy-3'-methoxy-5,5'-dicarboxybiphenyl\n",
      "            C18349  4-[2-(5-Carboxy-2-hydroxy-3-methoxyphenyl)-2-oxoethylidene]-2-hydroxy-2-pentenedioate\n",
      "            C18351  4-Aminocatechol\n",
      "REFERENCE   PMID:3078743\n",
      "  AUTHORS   Fewson CA\n",
      "  TITLE     Microbial metabolism of mandelate: a microcosm of diversity.\n",
      "  JOURNAL   FEMS Microbiol Rev 4:85-110 (1988)\n",
      "            DOI:10.1111/j.1574-6968.1988.tb02737.x\n",
      "REFERENCE   PMID:12670968\n",
      "  AUTHORS   McLeish MJ, Kneen MM, Gopalakrishna KN, Koo CW, Babbitt PC, Gerlt JA, Kenyon GL\n",
      "  TITLE     Identification and characterization of a mandelamide hydrolase and an NAD(P)+-dependent benzaldehyde dehydrogenase from Pseudomonas putida ATCC 12633.\n",
      "  JOURNAL   J Bacteriol 185:2451-6 (2003)\n",
      "            DOI:10.1128/JB.185.8.2451-2456.2003\n",
      "REFERENCE   PMID:8782397\n",
      "  AUTHORS   Shimao M, Nakamura T, Okuda A, Harayama S.\n",
      "  TITLE     Characteristics of transposon insertion mutants of mandelic acid-utilizing Pseudomonas putida strain A10L.\n",
      "  JOURNAL   Biosci Biotechnol Biochem 60:1051-5 (1996)\n",
      "            DOI:10.1271/bbb.60.1051\n",
      "REFERENCE   PMID:11344134\n",
      "  AUTHORS   Naidu D, Ragsdale SW.\n",
      "  TITLE     Characterization of a three-component vanillate O-demethylase from Moorella thermoacetica.\n",
      "  JOURNAL   J Bacteriol 183:3276-81 (2001)\n",
      "            DOI:10.1128/JB.183.11.3276-3281.2001\n",
      "REFERENCE   PMID:14711645\n",
      "  AUTHORS   Toraya T, Oka T, Ando M, Yamanishi M, Nishihara H.\n",
      "  TITLE     Novel pathway for utilization of cyclopropanecarboxylate by Rhodococcus rhodochrous.\n",
      "  JOURNAL   Appl Environ Microbiol 70:224-8 (2004)\n",
      "            DOI:10.1128/AEM.70.1.224-228.2004\n",
      "REL_PATHWAY pae00361  Chlorocyclohexane and chlorobenzene degradation\n",
      "            pae00362  Benzoate degradation\n",
      "            pae00380  Tryptophan metabolism\n",
      "KO_PATHWAY  ko00627\n",
      "///\n",
      "path:pae00630\n",
      "ENTRY       pae00630                    Pathway\n",
      "NAME        Glyoxylate and dicarboxylate metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "DBLINKS     GO: 0046487 0043648\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2634  aceA; isocitrate lyase [KO:K01637] [EC:4.1.3.1]\n",
      "line in gene PA2634  aceA; isocitrate lyase [KO:K01637] [EC:4.1.3.1]\n",
      "            PA0887  acsA; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA0887  acsA; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "            PA4733  acsB; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA4733  acsB; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "            PA2555  AMP-binding protein [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA2555  AMP-binding protein [KO:K01895] [EC:6.2.1.1]\n",
      "            PA0482  glcB; malate synthase G [KO:K01638] [EC:2.3.3.9]\n",
      "line in gene PA0482  glcB; malate synthase G [KO:K01638] [EC:2.3.3.9]\n",
      "            PA1580  gltA; citrate synthase [KO:K01647] [EC:2.3.3.1]\n",
      "line in gene PA1580  gltA; citrate synthase [KO:K01647] [EC:2.3.3.1]\n",
      "            PA1562  acnA; aconitate hydratase [KO:K01681] [EC:4.2.1.3]\n",
      "line in gene PA1562  acnA; aconitate hydratase [KO:K01681] [EC:4.2.1.3]\n",
      "            PA1787  acnB; aconitate hydratase B [KO:K01682] [EC:4.2.1.3 4.2.1.99]\n",
      "line in gene PA1787  acnB; aconitate hydratase B [KO:K01682] [EC:4.2.1.3 4.2.1.99]\n",
      "            PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3302  hypothetical protein [KO:K17865] [EC:4.2.1.55]\n",
      "line in gene PA3302  hypothetical protein [KO:K17865] [EC:4.2.1.55]\n",
      "            PA5355  glcD; glycolate oxidase subunit GlcD [KO:K00104] [EC:1.1.3.15]\n",
      "line in gene PA5355  glcD; glycolate oxidase subunit GlcD [KO:K00104] [EC:1.1.3.15]\n",
      "            PA5354  glcE; glycolate oxidase FAD binding subunit [KO:K11472]\n",
      "line in gene PA5354  glcE; glycolate oxidase FAD binding subunit [KO:K11472]\n",
      "            PA5353  glcF; glycolate oxidase iron-sulfur subunit [KO:K11473]\n",
      "line in gene PA5353  glcF; glycolate oxidase iron-sulfur subunit [KO:K11473]\n",
      "            PA2147  katE; catalase HPII [KO:K03781] [EC:1.11.1.6]\n",
      "line in gene PA2147  katE; catalase HPII [KO:K03781] [EC:1.11.1.6]\n",
      "            PA4613  katB; catalase [KO:K03781] [EC:1.11.1.6]\n",
      "line in gene PA4613  katB; catalase [KO:K03781] [EC:1.11.1.6]\n",
      "            PA4236  katA; catalase [KO:K03781] [EC:1.11.1.6]\n",
      "line in gene PA4236  katA; catalase [KO:K03781] [EC:1.11.1.6]\n",
      "            PA3896  2-hydroxyacid dehydrogenase [KO:K00090] [EC:1.1.1.79 1.1.1.81 1.1.1.215]\n",
      "line in gene PA3896  2-hydroxyacid dehydrogenase [KO:K00090] [EC:1.1.1.79 1.1.1.81 1.1.1.215]\n",
      "            PA4626  hprA; glycerate dehydrogenase [KO:K00018] [EC:1.1.1.29]\n",
      "line in gene PA4626  hprA; glycerate dehydrogenase [KO:K00018] [EC:1.1.1.29]\n",
      "            PA0608  phosphoglycolate phosphatase [KO:K01091] [EC:3.1.3.18]\n",
      "line in gene PA0608  phosphoglycolate phosphatase [KO:K01091] [EC:3.1.3.18]\n",
      "            PA2974  hydrolase [KO:K01091] [EC:3.1.3.18]\n",
      "line in gene PA2974  hydrolase [KO:K01091] [EC:3.1.3.18]\n",
      "            PA0065  5'-nucleotidase [KO:K01091] [EC:3.1.3.18]\n",
      "line in gene PA0065  5'-nucleotidase [KO:K01091] [EC:3.1.3.18]\n",
      "            PA0296  spuI; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA0296  spuI; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA0298  spuB; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA0298  spuB; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA2040  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA2040  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5119  glnA; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5119  glnA; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5522  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5522  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA1566  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA1566  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "            PA3356  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA3356  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5508  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5508  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA2444  glyA2; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA2444  glyA2; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA5415  glyA1; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA5415  glyA1; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA4602  glyA3; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA4602  glyA3; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA2445  gcvP2; glycine dehydrogenase [KO:K00281] [EC:1.4.4.2]\n",
      "line in gene PA2445  gcvP2; glycine dehydrogenase [KO:K00281] [EC:1.4.4.2]\n",
      "            PA5213  gcvP1; glycine dehydrogenase [KO:K00281] [EC:1.4.4.2]\n",
      "line in gene PA5213  gcvP1; glycine dehydrogenase [KO:K00281] [EC:1.4.4.2]\n",
      "            PA2442  gcvT2; glycine cleavage system protein T2 [KO:K00605] [EC:2.1.2.10]\n",
      "line in gene PA2442  gcvT2; glycine cleavage system protein T2 [KO:K00605] [EC:2.1.2.10]\n",
      "            PA5215  gcvT1; glycine cleavage system aminomethyltransferase T [KO:K00605] [EC:2.1.2.10]\n",
      "line in gene PA5215  gcvT1; glycine cleavage system aminomethyltransferase T [KO:K00605] [EC:2.1.2.10]\n",
      "            PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA2446  gcvH2; glycine cleavage system protein H [KO:K02437]\n",
      "line in gene PA2446  gcvH2; glycine cleavage system protein H [KO:K02437]\n",
      "            PA5214  gcvH1; glycine cleavage system protein H [KO:K02437]\n",
      "line in gene PA5214  gcvH1; glycine cleavage system protein H [KO:K02437]\n",
      "            PA4796  hypothetical protein [KO:K02437]\n",
      "line in gene PA4796  hypothetical protein [KO:K02437]\n",
      "            PA1502  gcl; glyoxylate carboligase [KO:K01608] [EC:4.1.1.47]\n",
      "line in gene PA1502  gcl; glyoxylate carboligase [KO:K01608] [EC:4.1.1.47]\n",
      "            PA1501  hypothetical protein [KO:K01816] [EC:5.3.1.22]\n",
      "line in gene PA1501  hypothetical protein [KO:K01816] [EC:5.3.1.22]\n",
      "            PA1500  2-hydroxy-3-oxopropionate reductase [KO:K00042] [EC:1.1.1.60]\n",
      "line in gene PA1500  2-hydroxy-3-oxopropionate reductase [KO:K00042] [EC:1.1.1.60]\n",
      "            PA1052  hypothetical protein [KO:K00865] [EC:2.7.1.165]\n",
      "line in gene PA1052  hypothetical protein [KO:K00865] [EC:2.7.1.165]\n",
      "            PA1499  hypothetical protein [KO:K11529] [EC:2.7.1.165]\n",
      "line in gene PA1499  hypothetical protein [KO:K11529] [EC:2.7.1.165]\n",
      "            PA3181  2-dehydro-3-deoxy-phosphogluconate aldolase [KO:K01625] [EC:4.1.2.14 4.1.3.42]\n",
      "line in gene PA3181  2-dehydro-3-deoxy-phosphogluconate aldolase [KO:K01625] [EC:4.1.2.14 4.1.3.42]\n",
      "            PA3131  aldolase [KO:K01625] [EC:4.1.2.14 4.1.3.42]\n",
      "line in gene PA3131  aldolase [KO:K01625] [EC:4.1.2.14 4.1.3.42]\n",
      "            PA4812  fdnG; formate dehydrogenase-O major subunit [KO:K00123] [EC:1.17.1.9]\n",
      "line in gene PA4812  fdnG; formate dehydrogenase-O major subunit [KO:K00123] [EC:1.17.1.9]\n",
      "            PA4811  fdnH; nitrate-inducible formate dehydrogenase subunit beta [KO:K00124]\n",
      "line in gene PA4811  fdnH; nitrate-inducible formate dehydrogenase subunit beta [KO:K00124]\n",
      "            PA4810  fdnI; nitrate-inducible formate dehydrogenase subunit gamma [KO:K00127]\n",
      "line in gene PA4810  fdnI; nitrate-inducible formate dehydrogenase subunit gamma [KO:K00127]\n",
      "            PA2081  kynB; kynurenine formamidase KynB [KO:K07130] [EC:3.5.1.9]\n",
      "line in gene PA2081  kynB; kynurenine formamidase KynB [KO:K07130] [EC:3.5.1.9]\n",
      "            PA5420  purU2; formyltetrahydrofolate deformylase [KO:K01433] [EC:3.5.1.10]\n",
      "line in gene PA5420  purU2; formyltetrahydrofolate deformylase [KO:K01433] [EC:3.5.1.10]\n",
      "            PA4314  purU1; formyltetrahydrofolate deformylase [KO:K01433] [EC:3.5.1.10]\n",
      "line in gene PA4314  purU1; formyltetrahydrofolate deformylase [KO:K01433] [EC:3.5.1.10]\n",
      "            PA5091  hutG; N-formylglutamate amidohydrolase [KO:K01458] [EC:3.5.1.68]\n",
      "line in gene PA5091  hutG; N-formylglutamate amidohydrolase [KO:K01458] [EC:3.5.1.68]\n",
      "COMPOUND    C00007  Oxygen\n",
      "            C00011  CO2\n",
      "            C00014  Ammonia\n",
      "            C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00025  L-Glutamate\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00027  Hydrogen peroxide\n",
      "            C00033  Acetate\n",
      "            C00036  Oxaloacetate\n",
      "            C00037  Glycine\n",
      "            C00042  Succinate\n",
      "            C00048  Glyoxylate\n",
      "            C00058  Formate\n",
      "            C00064  L-Glutamine\n",
      "            C00065  L-Serine\n",
      "            C00091  Succinyl-CoA\n",
      "            C00100  Propanoyl-CoA\n",
      "            C00136  Butanoyl-CoA\n",
      "            C00149  (S)-Malate\n",
      "            C00158  Citrate\n",
      "            C00160  Glycolate\n",
      "            C00168  Hydroxypyruvate\n",
      "            C00197  3-Phospho-D-glycerate\n",
      "            C00209  Oxalate\n",
      "            C00258  D-Glycerate\n",
      "            C00266  Glycolaldehyde\n",
      "            C00311  Isocitrate\n",
      "            C00313  Oxalyl-CoA\n",
      "            C00332  Acetoacetyl-CoA\n",
      "            C00417  cis-Aconitate\n",
      "            C00552  meso-Tartaric acid\n",
      "            C00631  2-Phospho-D-glycerate\n",
      "            C00683  (S)-Methylmalonyl-CoA\n",
      "            C00798  Formyl-CoA\n",
      "            C00877  Crotonoyl-CoA\n",
      "            C00888  Pentanoyl-CoA\n",
      "            C00898  (R,R)-Tartaric acid\n",
      "            C00975  Dihydroxyfumarate\n",
      "            C00988  2-Phosphoglycolate\n",
      "            C01127  4-Hydroxy-2-oxoglutarate\n",
      "            C01146  2-Hydroxy-3-oxopropanoate\n",
      "            C01182  D-Ribulose 1,5-bisphosphate\n",
      "            C01213  (R)-Methylmalonyl-CoA\n",
      "            C01380  Ethylene glycol\n",
      "            C01732  Mesaconate\n",
      "            C01989  3-Ethylmalate\n",
      "            C01990  3-Oxalomalate\n",
      "            C02123  3-Propylmalate\n",
      "            C02405  Formyl phosphate\n",
      "            C03217  2-Hydroxy-3-oxoadipate\n",
      "            C03459  2-Hydroxy-3-oxosuccinate\n",
      "            C03548  trans-2,3-Epoxysuccinate\n",
      "            C03561  (R)-3-Hydroxybutanoyl-CoA\n",
      "            C03618  L-threo-3-Methylaspartate\n",
      "            C04348  L-Malyl-CoA\n",
      "            C06027  L-erythro-3-Methylmalyl-CoA\n",
      "            C06028  2-Methylfumaryl-CoA\n",
      "            C06049  N-Formylderivatives\n",
      "            C18026  (2S)-Ethylmalonyl-CoA\n",
      "            C18324  (2S)-Methylsuccinyl-CoA\n",
      "            C20238  (2R)-Ethylmalonyl-CoA\n",
      "REFERENCE   PMID:10978349\n",
      "  AUTHORS   Njau RK, Herndon CA, Hawes JW.\n",
      "  TITLE     Novel beta-hydroxyacid dehydrogenases in Escherichia coli and Haemophilus influenzae.\n",
      "  JOURNAL   J Biol Chem 275:38780-6 (2000)\n",
      "            DOI:10.1074/jbc.M007432200\n",
      "REFERENCE   PMID:18065535\n",
      "  AUTHORS   Zarzycki J, Schlichting A, Strychalsky N, Muller M, Alber BE, Fuchs G\n",
      "  TITLE     Mesaconyl-coenzyme A hydratase, a new enzyme of two central carbon metabolic pathways in bacteria.\n",
      "  JOURNAL   J Bacteriol 190:1366-74 (2008)\n",
      "            DOI:10.1128/JB.01621-07\n",
      "REFERENCE   PMID:18819910\n",
      "  AUTHORS   Erb TJ, Retey J, Fuchs G, Alber BE\n",
      "  TITLE     Ethylmalonyl-CoA mutase from Rhodobacter sphaeroides defines a new subclade of coenzyme B12-dependent acyl-CoA mutases.\n",
      "  JOURNAL   J Biol Chem 283:32283-93 (2008)\n",
      "            DOI:10.1074/jbc.M805527200\n",
      "REFERENCE   PMID:9596633\n",
      "  AUTHORS   Coschigano KT, Melo-Oliveira R, Lim J, Coruzzi GM\n",
      "  TITLE     Arabidopsis gls mutants and distinct Fd-GOGAT genes. Implications for photorespiration and primary nitrogen assimilation.\n",
      "  JOURNAL   Plant Cell 10:741-52 (1998)\n",
      "            DOI:10.1105/tpc.10.5.741\n",
      "REFERENCE   PMID:16407450\n",
      "  AUTHORS   Masclaux-Daubresse C, Reisdorf-Cren M, Pageau K, Lelandais M, Grandjean O, Kronenberger J, Valadier MH, Feraud M, Jouglet T, Suzuki A\n",
      "  TITLE     Glutamine synthetase-glutamate synthase pathway and glutamate dehydrogenase play distinct roles in the sink-source nitrogen cycle in tobacco.\n",
      "  JOURNAL   Plant Physiol 140:444-56 (2006)\n",
      "            DOI:10.1104/pp.105.071910\n",
      "REFERENCE   PMID:21252347\n",
      "  AUTHORS   Khomyakova M, Bukmez O, Thomas LK, Erb TJ, Berg IA\n",
      "  TITLE     A methylaspartate cycle in haloarchaea.\n",
      "  JOURNAL   Science 331:334-7 (2011)\n",
      "            DOI:10.1126/science.1196544\n",
      "REFERENCE   PMID:11513957\n",
      "  AUTHORS   Serrano JA, Bonete MJ\n",
      "  TITLE     Sequencing, phylogenetic and transcriptional analysis of the glyoxylate bypass operon (ace) in the halophilic archaeon Haloferax volcanii.\n",
      "  JOURNAL   Biochim Biophys Acta 1520:154-62 (2001)\n",
      "            DOI:10.1016/S0167-4781(01)00263-9\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00053  Ascorbate and aldarate metabolism\n",
      "            pae00071  Fatty acid degradation\n",
      "            pae00230  Purine metabolism\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00750  Vitamin B6 metabolism\n",
      "            pae00910  Nitrogen metabolism\n",
      "KO_PATHWAY  ko00630\n",
      "///\n",
      "path:pae00633\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00633                    Pathway\n",
      "NAME        Nitrotoluene degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Xenobiotics biodegradation and metabolism\n",
      "PATHWAY_MAP pae00633  Nitrotoluene degradation\n",
      "DBLINKS     UMBBD: tnt tnt2 dnt\n",
      "            GO: 0046260\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2932  morB; morphinone reductase [KO:K10680] [EC:1.-.-.-]\n",
      "line in gene PA2932  morB; morphinone reductase [KO:K10680] [EC:1.-.-.-]\n",
      "            PA1334  oxidoreductase [KO:K10680] [EC:1.-.-.-]\n",
      "line in gene PA1334  oxidoreductase [KO:K10680] [EC:1.-.-.-]\n",
      "COMPOUND    C01468  4-Cresol\n",
      "            C06002  (S)-Methylmalonate semialdehyde\n",
      "            C11006  2,4-Dinitrotoluene\n",
      "            C14401  2,4-Diaminotoluene\n",
      "            C16391  Trinitrotoluene\n",
      "            C16392  4-Hydroxylamino-2,6-dinitrotoluene\n",
      "            C16393  2-Hydroxylamino-4,6-dinitrotoluene\n",
      "            C16394  4-Amino-2,6-dinitrotoluene\n",
      "            C16395  2-Amino-4,6-dinitrotoluene\n",
      "            C16396  2,4-Diamino-6-nitrotoluene\n",
      "            C16397  2,6-Diamino-4-nitrotoluene\n",
      "            C16398  2-Amino-4-nitrotoluene\n",
      "            C16399  2,4-Diamino-6-hydroxylaminotoluene\n",
      "            C16400  2,4,6-Triaminotoluene\n",
      "            C16401  2,4,6-Trihydroxytoluene\n",
      "            C16402  2,4-Dihydroxylamino-6-nitrotoluene\n",
      "            C16403  4-Amino-2-hydroxylamino-6-nitrotoluene\n",
      "            C16411  4,4',6,6'-Tetranitro-2,2'-azoxytoluene\n",
      "            C16412  2,4',6,6'-Tetranitro-2',4-azoxytoluene\n",
      "            C16413  2,2',6,6'-Tetranitro-4,4'-azoxytoluene\n",
      "            C16414  2-Amino-5-hydroxyl-4-hydroxylamino-6-nitrotoluene\n",
      "            C16420  4-Acetamido-2-amino-6-nitrotoluene\n",
      "            C18315  4-Methyl-5-nitrocatechol\n",
      "            C18316  2-Hydroxy-5-methylquinone\n",
      "            C18317  2,4,5-Trihydroxytoluene\n",
      "            C18318  cis,cis-2,4-Dihydroxy-5-methyl-6-oxo-2,4-hexadienoate\n",
      "            C18320  4-Amino-2-nitrotoluene\n",
      "REFERENCE   \n",
      "  AUTHORS   Esteve-Nunez A, Ramos JL.\n",
      "  TITLE     Metabolism of 2,4,6-Trinitrotoluene by Pseudomonas sp. JLR11.\n",
      "  JOURNAL   Environ Sci Tech 32:3802-3808 (1998)\n",
      "            DOI:10.1021/es9803308\n",
      "REFERENCE   PMID:8534088\n",
      "  AUTHORS   Gilcrease PC, Murphy VG.\n",
      "  TITLE     Bioconversion of 2,4-diamino-6-nitrotoluene to a novel metabolite under anoxic and aerobic conditions.\n",
      "  JOURNAL   Appl Environ Microbiol 61:4209-14 (1995)\n",
      "            DOI:10.1128/AEM.61.12.4209-4214.1995\n",
      "REFERENCE   PMID:11527999\n",
      "  AUTHORS   Esteve-Nunez A, Caballero A, Ramos JL.\n",
      "  TITLE     Biological degradation of 2,4,6-trinitrotoluene.\n",
      "  JOURNAL   Microbiol Mol Biol Rev 65:335-52, table of contents (2001)\n",
      "            DOI:10.1128/MMBR.65.3.335-352.2001\n",
      "REFERENCE   PMID:14751225\n",
      "  AUTHORS   Fleischmann TJ, Walker KC, Spain JC, Hughes JB, Morrie Craig A.\n",
      "  TITLE     Anaerobic transformation of 2,4,6-TNT by bovine ruminal microbes.\n",
      "  JOURNAL   Biochem Biophys Res Commun 314:957-63 (2004)\n",
      "            DOI:10.1016/j.bbrc.2003.12.193\n",
      "REFERENCE   PMID:14753774\n",
      "  AUTHORS   Borch T, Gerlach R.\n",
      "  TITLE     Use of reversed-phase high-performance liquid chromatography-diode array detection for complete separation of 2,4,6-trinitrotoluene metabolites and EPA Method 8330 explosives: influence of temperature and an ion-pair reagent.\n",
      "  JOURNAL   J Chromatogr A 1022:83-94 (2004)\n",
      "            DOI:10.1016/j.chroma.2003.09.067\n",
      "REFERENCE   PMID:15490158\n",
      "  AUTHORS   Yin H, Wood TK, Smets BF.\n",
      "  TITLE     Reductive transformation of TNT by Escherichia coli: pathway description.\n",
      "  JOURNAL   Appl Microbiol Biotechnol 67:397-404 (2005)\n",
      "            DOI:10.1007/s00253-004-1736-x\n",
      "REFERENCE   PMID:16187099\n",
      "  AUTHORS   Kutty R, Bennett GN.\n",
      "  TITLE     Biochemical characterization of trinitrotoluene transforming oxygen-insensitive nitroreductases from Clostridium acetobutylicum ATCC 824.\n",
      "  JOURNAL   Arch Microbiol 184:158-67 (2005)\n",
      "            DOI:10.1007/s00203-005-0036-x\n",
      "REFERENCE   PMID:16535610\n",
      "  AUTHORS   Fiorella PD, Spain JC.\n",
      "  TITLE     Transformation of 2,4,6-Trinitrotoluene by Pseudomonas pseudoalcaligenes JS52.\n",
      "  JOURNAL   Appl Environ Microbiol 63:2007-2015 (1997)\n",
      "            DOI:10.1128/AEM.63.5.2007-2015.1997\n",
      "REFERENCE   PMID:17504490\n",
      "  AUTHORS   Gonzalez-Perez MM, van Dillewijn P, Wittich RM, Ramos JL.\n",
      "  TITLE     Escherichia coli has multiple enzymes that attack TNT and release nitrogen for growth.\n",
      "  JOURNAL   Environ Microbiol 9:1535-40 (2007)\n",
      "            DOI:10.1111/j.1462-2920.2007.01272.x\n",
      "REFERENCE   PMID:16535348\n",
      "  AUTHORS   Noguera DR, Freedman DL\n",
      "  TITLE     Reduction and Acetylation of 2,4-Dinitrotoluene by a Pseudomonas aeruginosa Strain.\n",
      "  JOURNAL   Appl Environ Microbiol 62:2257-63 (1996)\n",
      "            DOI:10.1128/AEM.62.7.2257-2263.1996\n",
      "REFERENCE   PMID:9922262\n",
      "  AUTHORS   Haigler BE, Johnson GR, Suen WC, Spain JC\n",
      "  TITLE     Biochemical and genetic evidence for meta-ring cleavage of 2,4, 5-trihydroxytoluene in Burkholderia sp. strain DNT.\n",
      "  JOURNAL   J Bacteriol 181:965-72 (1999)\n",
      "            DOI:10.1128/JB.181.3.965-972.1999\n",
      "REFERENCE   PMID:12107140\n",
      "  AUTHORS   Johnson GR, Jain RK, Spain JC\n",
      "  TITLE     Origins of the 2,4-dinitrotoluene pathway.\n",
      "  JOURNAL   J Bacteriol 184:4219-32 (2002)\n",
      "            DOI:10.1128/JB.184.15.4219-4232.2002\n",
      "REL_PATHWAY pae00280  Valine, leucine and isoleucine degradation\n",
      "            pae00623  Toluene degradation\n",
      "KO_PATHWAY  ko00633\n",
      "///\n",
      "path:pae00640\n",
      "ENTRY       pae00640                    Pathway\n",
      "NAME        Propanoate metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00640  Propanoate metabolism\n",
      "DBLINKS     GO: 0019541\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0887  acsA; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA0887  acsA; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "            PA4733  acsB; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA4733  acsB; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "            PA2555  AMP-binding protein [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA2555  AMP-binding protein [KO:K01895] [EC:6.2.1.1]\n",
      "            PA3568  propionyl-CoA synthetase [KO:K01908] [EC:6.2.1.17]\n",
      "line in gene PA3568  propionyl-CoA synthetase [KO:K01908] [EC:6.2.1.17]\n",
      "            PA1017  pauA; pimeloyl-CoA synthetase [KO:K24012] [EC:6.2.1.13]\n",
      "line in gene PA1017  pauA; pimeloyl-CoA synthetase [KO:K24012] [EC:6.2.1.13]\n",
      "            PA4978  hypothetical protein [KO:K24012] [EC:6.2.1.13]\n",
      "line in gene PA4978  hypothetical protein [KO:K24012] [EC:6.2.1.13]\n",
      "            PA0836  ackA; acetate kinase [KO:K00925] [EC:2.7.2.1]\n",
      "line in gene PA0836  ackA; acetate kinase [KO:K00925] [EC:2.7.2.1]\n",
      "            PA0835  pta; phosphate acetyltransferase [KO:K13788] [EC:2.3.1.8]\n",
      "line in gene PA0835  pta; phosphate acetyltransferase [KO:K13788] [EC:2.3.1.8]\n",
      "            PA2247  bkdA1; 2-oxoisovalerate dehydrogenase subunit alpha [KO:K00166] [EC:1.2.4.4]\n",
      "line in gene PA2247  bkdA1; 2-oxoisovalerate dehydrogenase subunit alpha [KO:K00166] [EC:1.2.4.4]\n",
      "            PA3417  pyruvate dehydrogenase E1 component subunit alpha [KO:K00166] [EC:1.2.4.4]\n",
      "line in gene PA3417  pyruvate dehydrogenase E1 component subunit alpha [KO:K00166] [EC:1.2.4.4]\n",
      "            PA2248  bkdA2; 2-oxoisovalerate dehydrogenase subunit beta [KO:K00167] [EC:1.2.4.4]\n",
      "line in gene PA2248  bkdA2; 2-oxoisovalerate dehydrogenase subunit beta [KO:K00167] [EC:1.2.4.4]\n",
      "            PA3416  pyruvate dehydrogenase E1 component subunit beta [KO:K00167] [EC:1.2.4.4]\n",
      "line in gene PA3416  pyruvate dehydrogenase E1 component subunit beta [KO:K00167] [EC:1.2.4.4]\n",
      "            PA2249  bkdB; branched-chain alpha-keto acid dehydrogenase complex lipoamide acyltransferase [KO:K09699] [EC:2.3.1.168]\n",
      "line in gene PA2249  bkdB; branched-chain alpha-keto acid dehydrogenase complex lipoamide acyltransferase [KO:K09699] [EC:2.3.1.168]\n",
      "            PA3415  branched-chain alpha-keto acid dehydrogenase subunit E2 [KO:K09699] [EC:2.3.1.168]\n",
      "line in gene PA3415  branched-chain alpha-keto acid dehydrogenase subunit E2 [KO:K09699] [EC:2.3.1.168]\n",
      "            PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA2250  lpdV; branched-chain alpha-keto acid dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA4829  lpd3; dihydrolipoamide dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "line in gene PA1587  lpdG; 2-oxoglutarate dehydrogenase complex dihydrolipoyl dehydrogenase [KO:K00382] [EC:1.8.1.4]\n",
      "            PA2552  acyl-CoA dehydrogenase [KO:K00248] [EC:1.3.8.1]\n",
      "line in gene PA2552  acyl-CoA dehydrogenase [KO:K00248] [EC:1.3.8.1]\n",
      "            PA1833  oxidoreductase [KO:K19745] [EC:1.3.1.-]\n",
      "line in gene PA1833  oxidoreductase [KO:K19745] [EC:1.3.1.-]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "line in gene PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "            PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "line in gene PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "            PA3639  accA; acetyl-CoA carboxylase carboxyltransferase subunit alpha [KO:K01962] [EC:6.4.1.2 2.1.3.15]\n",
      "line in gene PA3639  accA; acetyl-CoA carboxylase carboxyltransferase subunit alpha [KO:K01962] [EC:6.4.1.2 2.1.3.15]\n",
      "            PA4847  accB; acetyl-CoA carboxylase biotin carboxyl carrier protein subunit [KO:K02160]\n",
      "line in gene PA4847  accB; acetyl-CoA carboxylase biotin carboxyl carrier protein subunit [KO:K02160]\n",
      "            PA0493  acetyl-CoA carboxylase biotin carboxyl carrier protein subunit [KO:K02160]\n",
      "line in gene PA0493  acetyl-CoA carboxylase biotin carboxyl carrier protein subunit [KO:K02160]\n",
      "            PA4848  accC; acetyl-CoA carboxylase biotin carboxylase subunit [KO:K01961] [EC:6.4.1.2 6.3.4.14]\n",
      "line in gene PA4848  accC; acetyl-CoA carboxylase biotin carboxylase subunit [KO:K01961] [EC:6.4.1.2 6.3.4.14]\n",
      "            PA0494  acetyl-CoA carboxylase biotin carboxylase subunit [KO:K01961] [EC:6.4.1.2 6.3.4.14]\n",
      "line in gene PA0494  acetyl-CoA carboxylase biotin carboxylase subunit [KO:K01961] [EC:6.4.1.2 6.3.4.14]\n",
      "            PA3112  accD; acetyl-CoA carboxylase carboxyltransferase subunit beta [KO:K01963] [EC:6.4.1.2 2.1.3.15]\n",
      "line in gene PA3112  accD; acetyl-CoA carboxylase carboxyltransferase subunit beta [KO:K01963] [EC:6.4.1.2 2.1.3.15]\n",
      "            PA0132  beta alanine--pyruvate transaminase [KO:K00822] [EC:2.6.1.18]\n",
      "line in gene PA0132  beta alanine--pyruvate transaminase [KO:K00822] [EC:2.6.1.18]\n",
      "            PA5313  omega amino acid--pyruvate transaminase [KO:K00822] [EC:2.6.1.18]\n",
      "line in gene PA5313  omega amino acid--pyruvate transaminase [KO:K00822] [EC:2.6.1.18]\n",
      "            PA1588  sucC; succinyl-CoA ligase subunit beta [KO:K01903] [EC:6.2.1.5]\n",
      "line in gene PA1588  sucC; succinyl-CoA ligase subunit beta [KO:K01903] [EC:6.2.1.5]\n",
      "            PA1589  sucD; succinyl-CoA ligase subunit alpha [KO:K01902] [EC:6.2.1.5]\n",
      "line in gene PA1589  sucD; succinyl-CoA ligase subunit alpha [KO:K01902] [EC:6.2.1.5]\n",
      "            PA3570  mmsA; methylmalonate-semialdehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "line in gene PA3570  mmsA; methylmalonate-semialdehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "            PA0747  aldehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "line in gene PA0747  aldehyde dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "            PA0130  3-oxopropanoate dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "line in gene PA0130  3-oxopropanoate dehydrogenase [KO:K00140] [EC:1.2.1.18 1.2.1.27]\n",
      "            PA0795  prpC; methylcitrate synthase [KO:K01659] [EC:2.3.3.5]\n",
      "line in gene PA0795  prpC; methylcitrate synthase [KO:K01659] [EC:2.3.3.5]\n",
      "            PA0792  prpD; 2-methylcitrate dehydratase [KO:K01720] [EC:4.2.1.79]\n",
      "line in gene PA0792  prpD; 2-methylcitrate dehydratase [KO:K01720] [EC:4.2.1.79]\n",
      "            PA0794  aconitate hydratase [KO:K20455] [EC:4.2.1.117]\n",
      "line in gene PA0794  aconitate hydratase [KO:K20455] [EC:4.2.1.117]\n",
      "            PA0793  hypothetical protein [KO:K09788] [EC:5.3.3.-]\n",
      "line in gene PA0793  hypothetical protein [KO:K09788] [EC:5.3.3.-]\n",
      "            PA1787  acnB; aconitate hydratase B [KO:K01682] [EC:4.2.1.3 4.2.1.99]\n",
      "line in gene PA1787  acnB; aconitate hydratase B [KO:K01682] [EC:4.2.1.3 4.2.1.99]\n",
      "            PA0796  prpB; 2-methylisocitrate lyase [KO:K03417] [EC:4.1.3.30]\n",
      "line in gene PA0796  prpB; 2-methylisocitrate lyase [KO:K03417] [EC:4.1.3.30]\n",
      "COMPOUND    C00024  Acetyl-CoA\n",
      "            C00042  Succinate\n",
      "            C00068  Thiamin diphosphate\n",
      "            C00083  Malonyl-CoA\n",
      "            C00091  Succinyl-CoA\n",
      "            C00099  beta-Alanine\n",
      "            C00100  Propanoyl-CoA\n",
      "            C00109  2-Oxobutanoate\n",
      "            C00111  Glycerone phosphate\n",
      "            C00163  Propanoate\n",
      "            C00186  (S)-Lactate\n",
      "            C00222  3-Oxopropanoate\n",
      "            C00424  (S)-Lactaldehyde\n",
      "            C00479  Propanal\n",
      "            C00546  Methylglyoxal\n",
      "            C00583  Propane-1,2-diol\n",
      "            C00683  (S)-Methylmalonyl-CoA\n",
      "            C00804  Propynoate\n",
      "            C00827  Lactoyl-CoA\n",
      "            C00894  Propenoyl-CoA\n",
      "            C01013  3-Hydroxypropanoate\n",
      "            C01213  (R)-Methylmalonyl-CoA\n",
      "            C02170  Methylmalonate\n",
      "            C02225  2-Methylcitrate\n",
      "            C02335  beta-Alanyl-CoA\n",
      "            C02876  Propanoyl phosphate\n",
      "            C04225  (Z)-But-2-ene-1,2,3-tricarboxylate\n",
      "            C04593  (2S,3R)-3-Hydroxybutane-1,2,3-tricarboxylate\n",
      "            C05235  Hydroxyacetone\n",
      "            C05668  3-Hydroxypropionyl-CoA\n",
      "            C05979  Propane-1-ol\n",
      "            C05983  Propionyladenylate\n",
      "            C05984  2-Hydroxybutanoic acid\n",
      "            C05989  3-Oxopropionyl-CoA\n",
      "            C06002  (S)-Methylmalonate semialdehyde\n",
      "            C15972  Enzyme N6-(lipoyl)lysine\n",
      "            C15973  Enzyme N6-(dihydrolipoyl)lysine\n",
      "            C21017  2-(alpha-Hydroxypropyl)thiamine diphosphate\n",
      "            C21018  Enzyme N6-(S-propyldihydrolipoyl)lysine\n",
      "            C21250  2-Methyl-trans-aconitate\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00280  Valine, leucine and isoleucine degradation\n",
      "            pae00410  beta-Alanine metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00660  C5-Branched dibasic acid metabolism\n",
      "            pae00770  Pantothenate and CoA biosynthesis\n",
      "KO_PATHWAY  ko00640\n",
      "///\n",
      "path:pae00643\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00643                    Pathway\n",
      "NAME        Styrene degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Xenobiotics biodegradation and metabolism\n",
      "PATHWAY_MAP pae00643  Styrene degradation\n",
      "DBLINKS     UMBBD: sty acr pao\n",
      "            GO: 0042207\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4073  aldehyde dehydrogenase [KO:K00146] [EC:1.2.1.39]\n",
      "line in gene PA4073  aldehyde dehydrogenase [KO:K00146] [EC:1.2.1.39]\n",
      "            PA2009  hmgA; homogentisate 1,2-dioxygenase [KO:K00451] [EC:1.13.11.5]\n",
      "line in gene PA2009  hmgA; homogentisate 1,2-dioxygenase [KO:K00451] [EC:1.13.11.5]\n",
      "            PA2473  glutathione S-transferase [KO:K01800] [EC:5.2.1.2]\n",
      "line in gene PA2473  glutathione S-transferase [KO:K01800] [EC:5.2.1.2]\n",
      "            PA2007  maiA; maleylacetoacetate isomerase [KO:K01800] [EC:5.2.1.2]\n",
      "line in gene PA2007  maiA; maleylacetoacetate isomerase [KO:K01800] [EC:5.2.1.2]\n",
      "            PA2008  fahA; fumarylacetoacetase [KO:K01555] [EC:3.7.1.2]\n",
      "line in gene PA2008  fahA; fumarylacetoacetase [KO:K01555] [EC:3.7.1.2]\n",
      "            PA0704  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA0704  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA0202  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA0202  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA3366  amiE; acylamide amidohydrolase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA3366  amiE; acylamide amidohydrolase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA4342  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA4342  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA4163  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "line in gene PA4163  amidase [KO:K01426] [EC:3.5.1.4]\n",
      "            PA0226  CoA transferase subunit A [KO:K01039] [EC:2.8.3.12]\n",
      "line in gene PA0226  CoA transferase subunit A [KO:K01039] [EC:2.8.3.12]\n",
      "            PA0227  CoA transferase subunit B [KO:K01040] [EC:2.8.3.12]\n",
      "line in gene PA0227  CoA transferase subunit B [KO:K01040] [EC:2.8.3.12]\n",
      "COMPOUND    C00122  Fumarate\n",
      "            C00164  Acetoacetate\n",
      "            C00186  (S)-Lactate\n",
      "            C00511  Acrylic acid\n",
      "            C00544  Homogentisate\n",
      "            C00596  2-Hydroxy-2,4-pentadienoate\n",
      "            C00601  Phenylacetaldehyde\n",
      "            C00827  Lactoyl-CoA\n",
      "            C00894  Propenoyl-CoA\n",
      "            C01036  4-Maleylacetoacetate\n",
      "            C01061  4-Fumarylacetoacetate\n",
      "            C01659  Acrylamide\n",
      "            C01998  Acrylonitrile\n",
      "            C02505  2-Phenylacetamide\n",
      "            C05593  3-Hydroxyphenylacetate\n",
      "            C05852  2-Hydroxyphenylacetate\n",
      "            C07083  Styrene\n",
      "            C07084  Styrene-cis-2,3-dihydrodiol\n",
      "            C07085  3-Vinylcatechol\n",
      "            C07086  Phenylacetic acid\n",
      "            C07087  2-Hydroxy-6-oxoocta-2,4,7-trienoate\n",
      "            C16074  Phenylacetonitrile\n",
      "            C16075  (Z)-Phenylacetaldehyde oxime\n",
      "            C20782  (S)-2-Phenyloxirane\n",
      "REFERENCE   PMID:11544206\n",
      "  AUTHORS   Rodriguez-Saiz M, Barredo JL, Moreno MA, Fernandez-Canon JM, Penalva MA, Diez B.\n",
      "  TITLE     Reduced function of a phenylacetate-oxidizing cytochrome p450 caused strong genetic improvement in early phylogeny of penicillin-producing strains.\n",
      "  JOURNAL   J Bacteriol 183:5465-71 (2001)\n",
      "            DOI:10.1128/JB.183.19.5465-5471.2001\n",
      "REFERENCE   PMID:17189487\n",
      "  AUTHORS   Ferrer-Sevillano F, Fernandez-Canon JM.\n",
      "  TITLE     Novel phacB-encoded cytochrome P450 monooxygenase from Aspergillus nidulans with 3-hydroxyphenylacetate 6-hydroxylase and 3,4-dihydroxyphenylacetate 6-hydroxylase activities.\n",
      "  JOURNAL   Eukaryot Cell 6:514-20 (2007)\n",
      "            DOI:10.1128/EC.00226-06\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00360  Phenylalanine metabolism\n",
      "            pae00362  Benzoate degradation\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00640  Propanoate metabolism\n",
      "KO_PATHWAY  ko00643\n",
      "///\n",
      "path:pae00650\n",
      "ENTRY       pae00650                    Pathway\n",
      "NAME        Butanoate metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00650  Butanoate metabolism\n",
      "DBLINKS     GO: 0019605\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3590  3-hydroxyacyl-CoA dehydrogenase [KO:K00074] [EC:1.1.1.157]\n",
      "line in gene PA3590  3-hydroxyacyl-CoA dehydrogenase [KO:K00074] [EC:1.1.1.157]\n",
      "            PA1628  3-hydroxyacyl-CoA dehydrogenase [KO:K00074] [EC:1.1.1.157]\n",
      "line in gene PA1628  3-hydroxyacyl-CoA dehydrogenase [KO:K00074] [EC:1.1.1.157]\n",
      "            PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "line in gene PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "            PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "line in gene PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA2552  acyl-CoA dehydrogenase [KO:K00248] [EC:1.3.8.1]\n",
      "line in gene PA2552  acyl-CoA dehydrogenase [KO:K00248] [EC:1.3.8.1]\n",
      "            PA2950  reductase [KO:K00209] [EC:1.3.1.9 1.3.1.44]\n",
      "line in gene PA2950  reductase [KO:K00209] [EC:1.3.1.9 1.3.1.44]\n",
      "            PA3302  hypothetical protein [KO:K17865] [EC:4.2.1.55]\n",
      "line in gene PA3302  hypothetical protein [KO:K17865] [EC:4.2.1.55]\n",
      "            PA5056  phaC1; poly(3-hydroxyalkanoic acid) synthase [KO:K03821] [EC:2.3.1.304]\n",
      "line in gene PA5056  phaC1; poly(3-hydroxyalkanoic acid) synthase [KO:K03821] [EC:2.3.1.304]\n",
      "            PA5058  phaC2; poly(3-hydroxyalkanoic acid) synthase [KO:K03821] [EC:2.3.1.304]\n",
      "line in gene PA5058  phaC2; poly(3-hydroxyalkanoic acid) synthase [KO:K03821] [EC:2.3.1.304]\n",
      "            PA0226  CoA transferase subunit A [KO:K01039] [EC:2.8.3.12]\n",
      "line in gene PA0226  CoA transferase subunit A [KO:K01039] [EC:2.8.3.12]\n",
      "            PA0227  CoA transferase subunit B [KO:K01040] [EC:2.8.3.12]\n",
      "line in gene PA0227  CoA transferase subunit B [KO:K01040] [EC:2.8.3.12]\n",
      "            PA1583  sdhA; succinate dehydrogenase flavoprotein subunit [KO:K00239] [EC:1.3.5.1 1.3.5.4]\n",
      "line in gene PA1583  sdhA; succinate dehydrogenase flavoprotein subunit [KO:K00239] [EC:1.3.5.1 1.3.5.4]\n",
      "            PA1584  sdhB; succinate dehydrogenase iron-sulfur subunit [KO:K00240] [EC:1.3.5.1 1.3.5.4]\n",
      "line in gene PA1584  sdhB; succinate dehydrogenase iron-sulfur subunit [KO:K00240] [EC:1.3.5.1 1.3.5.4]\n",
      "            PA1581  sdhC; succinate dehydrogenase subunit C [KO:K00241]\n",
      "line in gene PA1581  sdhC; succinate dehydrogenase subunit C [KO:K00241]\n",
      "            PA1582  sdhD; succinate dehydrogenase subunit D [KO:K00242]\n",
      "line in gene PA1582  sdhD; succinate dehydrogenase subunit D [KO:K00242]\n",
      "            PA0266  5-aminovalerate aminotransferase DavT [KO:K14268] [EC:2.6.1.48 2.6.1.19]\n",
      "line in gene PA0266  5-aminovalerate aminotransferase DavT [KO:K14268] [EC:2.6.1.48 2.6.1.19]\n",
      "            PA0265  gabD; glutarate-semialdehyde dehydrogenase DavD [KO:K00135] [EC:1.2.1.16 1.2.1.79 1.2.1.20]\n",
      "line in gene PA0265  gabD; glutarate-semialdehyde dehydrogenase DavD [KO:K00135] [EC:1.2.1.16 1.2.1.79 1.2.1.20]\n",
      "            PA5445  coenzyme A transferase [KO:K18118] [EC:2.8.3.18]\n",
      "line in gene PA5445  coenzyme A transferase [KO:K18118] [EC:2.8.3.18]\n",
      "            PA2011  liuE; 3-hydroxy-3-isohexenylglutaryl-CoA/hydroxy-methylglutaryl-CoA lyase [KO:K01640] [EC:4.1.3.4]\n",
      "line in gene PA2011  liuE; 3-hydroxy-3-isohexenylglutaryl-CoA/hydroxy-methylglutaryl-CoA lyase [KO:K01640] [EC:4.1.3.4]\n",
      "            PA1999  dhcA; dehydrocarnitine CoA transferase subunit A [KO:K01028] [EC:2.8.3.5]\n",
      "line in gene PA1999  dhcA; dehydrocarnitine CoA transferase subunit A [KO:K01028] [EC:2.8.3.5]\n",
      "            PA2000  dhcB; dehydrocarnitine CoA transferase subunit B [KO:K01029] [EC:2.8.3.5]\n",
      "line in gene PA2000  dhcB; dehydrocarnitine CoA transferase subunit B [KO:K01029] [EC:2.8.3.5]\n",
      "            PA1997  acetoacetyl-CoA synthetase [KO:K01907] [EC:6.2.1.16]\n",
      "line in gene PA1997  acetoacetyl-CoA synthetase [KO:K01907] [EC:6.2.1.16]\n",
      "            PA2003  bdhA; 3-hydroxybutyrate dehydrogenase [KO:K00019] [EC:1.1.1.30]\n",
      "line in gene PA2003  bdhA; 3-hydroxybutyrate dehydrogenase [KO:K00019] [EC:1.1.1.30]\n",
      "            PA2035  thiamine pyrophosphate protein [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA2035  thiamine pyrophosphate protein [KO:K01652] [EC:2.2.1.6]\n",
      "            PA4180  acetolactate synthase [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA4180  acetolactate synthase [KO:K01652] [EC:2.2.1.6]\n",
      "            PA4696  ilvI; acetolactate synthase 3 catalytic subunit [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA4696  ilvI; acetolactate synthase 3 catalytic subunit [KO:K01652] [EC:2.2.1.6]\n",
      "            PA1417  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA1417  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "            PA3506  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA3506  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "            PA4695  ilvH; acetolactate synthase small subunit [KO:K01653] [EC:2.2.1.6]\n",
      "line in gene PA4695  ilvH; acetolactate synthase small subunit [KO:K01653] [EC:2.2.1.6]\n",
      "            PA4097  alcohol dehydrogenase [KO:K00004] [EC:1.1.1.4 1.1.1.- 1.1.1.303]\n",
      "line in gene PA4097  alcohol dehydrogenase [KO:K00004] [EC:1.1.1.4 1.1.1.- 1.1.1.303]\n",
      "            PA4153  2,3-butanediol dehydrogenase [KO:K00004] [EC:1.1.1.4 1.1.1.- 1.1.1.303]\n",
      "line in gene PA4153  2,3-butanediol dehydrogenase [KO:K00004] [EC:1.1.1.4 1.1.1.- 1.1.1.303]\n",
      "            PA4148  short-chain dehydrogenase [KO:K03366] [EC:1.1.1.- 1.1.1.76 1.1.1.304]\n",
      "line in gene PA4148  short-chain dehydrogenase [KO:K03366] [EC:1.1.1.- 1.1.1.76 1.1.1.304]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00025  L-Glutamate\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00042  Succinate\n",
      "            C00091  Succinyl-CoA\n",
      "            C00122  Fumarate\n",
      "            C00132  Methanol\n",
      "            C00136  Butanoyl-CoA\n",
      "            C00164  Acetoacetate\n",
      "            C00207  Acetone\n",
      "            C00232  Succinate semialdehyde\n",
      "            C00246  Butanoic acid\n",
      "            C00332  Acetoacetyl-CoA\n",
      "            C00334  4-Aminobutanoate\n",
      "            C00356  (S)-3-Hydroxy-3-methylglutaryl-CoA\n",
      "            C00497  (R)-Malate\n",
      "            C00741  Diacetyl\n",
      "            C00810  (R)-Acetoin\n",
      "            C00877  Crotonoyl-CoA\n",
      "            C00989  4-Hydroxybutanoic acid\n",
      "            C01089  (R)-3-Hydroxybutanoate\n",
      "            C01144  (S)-3-Hydroxybutanoyl-CoA\n",
      "            C01384  Maleic acid\n",
      "            C01412  Butanal\n",
      "            C01769  (S)-Acetoin\n",
      "            C01845  Propan-2-ol\n",
      "            C02331  Vinylacetyl-CoA\n",
      "            C02411  Glutaconyl-1-CoA\n",
      "            C02527  Butanoylphosphate\n",
      "            C02630  2-Hydroxyglutarate\n",
      "            C03044  (R,R)-Butane-2,3-diol\n",
      "            C03046  (S,S)-Butane-2,3-diol\n",
      "            C03058  2-Hydroxyglutaryl-CoA\n",
      "            C03561  (R)-3-Hydroxybutanoyl-CoA\n",
      "            C04546  (R)-3-((R)-3-Hydroxybutanoyloxy)butanoate\n",
      "            C06010  (S)-2-Acetolactate\n",
      "            C06142  1-Butanol\n",
      "            C06143  Poly-beta-hydroxybutyrate\n",
      "            C06144  3-Butynoate\n",
      "            C06145  3-Butyn-1-al\n",
      "            C06146  3-Butyn-1-ol\n",
      "            C11062  4-Hydroxybutyryl-CoA\n",
      "            C17530  Methyl acetate\n",
      "            C20657  meso-2,3-Butanediol\n",
      "            C20783  Propane\n",
      "            C21390  Butane\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00330  Arginine and proline metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00750  Vitamin B6 metabolism\n",
      "KO_PATHWAY  ko00650\n",
      "///\n",
      "path:pae00660\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00660                    Pathway\n",
      "NAME        C5-Branched dibasic acid metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Carbohydrate metabolism\n",
      "PATHWAY_MAP pae00660  C5-Branched dibasic acid metabolism\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2035  thiamine pyrophosphate protein [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA2035  thiamine pyrophosphate protein [KO:K01652] [EC:2.2.1.6]\n",
      "            PA4180  acetolactate synthase [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA4180  acetolactate synthase [KO:K01652] [EC:2.2.1.6]\n",
      "            PA4696  ilvI; acetolactate synthase 3 catalytic subunit [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA4696  ilvI; acetolactate synthase 3 catalytic subunit [KO:K01652] [EC:2.2.1.6]\n",
      "            PA1417  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA1417  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "            PA3506  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA3506  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "            PA4695  ilvH; acetolactate synthase small subunit [KO:K01653] [EC:2.2.1.6]\n",
      "line in gene PA4695  ilvH; acetolactate synthase small subunit [KO:K01653] [EC:2.2.1.6]\n",
      "            PA0882  hypothetical protein [KO:K18289] [EC:2.8.3.- 2.8.3.22]\n",
      "line in gene PA0882  hypothetical protein [KO:K18289] [EC:2.8.3.- 2.8.3.22]\n",
      "            PA1588  sucC; succinyl-CoA ligase subunit beta [KO:K01903] [EC:6.2.1.5]\n",
      "line in gene PA1588  sucC; succinyl-CoA ligase subunit beta [KO:K01903] [EC:6.2.1.5]\n",
      "            PA1589  sucD; succinyl-CoA ligase subunit alpha [KO:K01902] [EC:6.2.1.5]\n",
      "line in gene PA1589  sucD; succinyl-CoA ligase subunit alpha [KO:K01902] [EC:6.2.1.5]\n",
      "            PA0878  hypothetical protein [KO:K18291] [EC:4.2.1.56 4.2.1.-]\n",
      "line in gene PA0878  hypothetical protein [KO:K18291] [EC:4.2.1.56 4.2.1.-]\n",
      "            PA0883  acyl-CoA lyase subunit beta [KO:K18292] [EC:4.1.3.25]\n",
      "line in gene PA0883  acyl-CoA lyase subunit beta [KO:K18292] [EC:4.1.3.25]\n",
      "            PA3121  leuC; 3-isopropylmalate dehydratase large subunit [KO:K01703] [EC:4.2.1.33 4.2.1.35]\n",
      "line in gene PA3121  leuC; 3-isopropylmalate dehydratase large subunit [KO:K01703] [EC:4.2.1.33 4.2.1.35]\n",
      "            PA3120  leuD; isopropylmalate isomerase small subunit [KO:K01704] [EC:4.2.1.33 4.2.1.35]\n",
      "line in gene PA3120  leuD; isopropylmalate isomerase small subunit [KO:K01704] [EC:4.2.1.33 4.2.1.35]\n",
      "            PA3118  leuB; 3-isopropylmalate dehydrogenase [KO:K00052] [EC:1.1.1.85]\n",
      "line in gene PA3118  leuB; 3-isopropylmalate dehydrogenase [KO:K00052] [EC:1.1.1.85]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00025  L-Glutamate\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00033  Acetate\n",
      "            C00048  Glyoxylate\n",
      "            C00100  Propanoyl-CoA\n",
      "            C00109  2-Oxobutanoate\n",
      "            C00417  cis-Aconitate\n",
      "            C00490  Itaconate\n",
      "            C00531  Itaconyl-CoA\n",
      "            C00651  4-Methylene-L-glutamate\n",
      "            C00810  (R)-Acetoin\n",
      "            C01011  (3S)-Citramalyl-CoA\n",
      "            C01109  4-Methylene-L-glutamine\n",
      "            C01732  Mesaconate\n",
      "            C02226  2-Methylmaleate\n",
      "            C02341  trans-Aconitate\n",
      "            C02612  (R)-2-Methylmalate\n",
      "            C02614  (S)-2-Methylmalate\n",
      "            C02630  2-Hydroxyglutarate\n",
      "            C02876  Propanoyl phosphate\n",
      "            C03618  L-threo-3-Methylaspartate\n",
      "            C06010  (S)-2-Acetolactate\n",
      "            C06027  L-erythro-3-Methylmalyl-CoA\n",
      "            C06028  2-Methylfumaryl-CoA\n",
      "            C06029  L-threo-3-Methylmalate\n",
      "            C06030  Methyloxaloacetate\n",
      "            C06031  D-threo-3-Methylmalate\n",
      "            C06032  D-erythro-3-Methylmalate\n",
      "            C06033  Parapyruvate\n",
      "            C06034  4-Hydroxy-4-methylglutamate\n",
      "            C06035  4-Methylene-2-oxoglutarate\n",
      "            C06234  4-Methyl-L-glutamate\n",
      "            C21632  2-Hydroxyparaconate\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00290  Valine, leucine and isoleucine biosynthesis\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00640  Propanoate metabolism\n",
      "            pae00760  Nicotinate and nicotinamide metabolism\n",
      "KO_PATHWAY  ko00660\n",
      "///\n",
      "path:pae00670\n",
      "ENTRY       pae00670                    Pathway\n",
      "NAME        One carbon pool by folate - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of cofactors and vitamins\n",
      "PATHWAY_MAP pae00670  One carbon pool by folate\n",
      "MODULE      pae_M00126  Tetrahydrofolate biosynthesis, GTP => THF [PATH:pae00670]\n",
      "DBLINKS     GO: 0035999\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0350  folA; dihydrofolate reductase [KO:K00287] [EC:1.5.1.3]\n",
      "line in gene PA0350  folA; dihydrofolate reductase [KO:K00287] [EC:1.5.1.3]\n",
      "            PA3437  folM; short-chain dehydrogenase [KO:K13938] [EC:1.5.1.50 1.5.1.3]\n",
      "line in gene PA3437  folM; short-chain dehydrogenase [KO:K13938] [EC:1.5.1.50 1.5.1.3]\n",
      "            PA1796  folD; bifunctional 5,10-methylene-tetrahydrofolate dehydrogenase/5,10-methylene-tetrahydrofolate cyclohydrolase [KO:K01491] [EC:1.5.1.5 3.5.4.9]\n",
      "line in gene PA1796  folD; bifunctional 5,10-methylene-tetrahydrofolate dehydrogenase/5,10-methylene-tetrahydrofolate cyclohydrolase [KO:K01491] [EC:1.5.1.5 3.5.4.9]\n",
      "            PA2444  glyA2; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA2444  glyA2; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA5415  glyA1; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA5415  glyA1; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA4602  glyA3; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA4602  glyA3; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA0944  purN; phosphoribosylglycinamide formyltransferase [KO:K11175] [EC:2.1.2.2]\n",
      "line in gene PA0944  purN; phosphoribosylglycinamide formyltransferase [KO:K11175] [EC:2.1.2.2]\n",
      "            PA4854  purH; bifunctional phosphoribosylaminoimidazolecarboxamide formyltransferase/IMP cyclohydrolase [KO:K00602] [EC:2.1.2.3 3.5.4.10]\n",
      "line in gene PA4854  purH; bifunctional phosphoribosylaminoimidazolecarboxamide formyltransferase/IMP cyclohydrolase [KO:K00602] [EC:2.1.2.3 3.5.4.10]\n",
      "            PA0018  fmt; methionyl-tRNA formyltransferase [KO:K00604] [EC:2.1.2.9]\n",
      "line in gene PA0018  fmt; methionyl-tRNA formyltransferase [KO:K00604] [EC:2.1.2.9]\n",
      "            PA2442  gcvT2; glycine cleavage system protein T2 [KO:K00605] [EC:2.1.2.10]\n",
      "line in gene PA2442  gcvT2; glycine cleavage system protein T2 [KO:K00605] [EC:2.1.2.10]\n",
      "            PA5215  gcvT1; glycine cleavage system aminomethyltransferase T [KO:K00605] [EC:2.1.2.10]\n",
      "line in gene PA5215  gcvT1; glycine cleavage system aminomethyltransferase T [KO:K00605] [EC:2.1.2.10]\n",
      "            PA1843  metH; B12-dependent methionine synthase [KO:K00548] [EC:2.1.1.13]\n",
      "line in gene PA1843  metH; B12-dependent methionine synthase [KO:K00548] [EC:2.1.1.13]\n",
      "            PA0342  thyA; thymidylate synthase [KO:K00560] [EC:2.1.1.45]\n",
      "line in gene PA0342  thyA; thymidylate synthase [KO:K00560] [EC:2.1.1.45]\n",
      "            PA5420  purU2; formyltetrahydrofolate deformylase [KO:K01433] [EC:3.5.1.10]\n",
      "line in gene PA5420  purU2; formyltetrahydrofolate deformylase [KO:K01433] [EC:3.5.1.10]\n",
      "            PA4314  purU1; formyltetrahydrofolate deformylase [KO:K01433] [EC:3.5.1.10]\n",
      "line in gene PA4314  purU1; formyltetrahydrofolate deformylase [KO:K01433] [EC:3.5.1.10]\n",
      "            PA0430  metF; 5,10-methylenetetrahydrofolate reductase [KO:K00297] [EC:1.5.1.54]\n",
      "line in gene PA0430  metF; 5,10-methylenetetrahydrofolate reductase [KO:K00297] [EC:1.5.1.54]\n",
      "            PA5228  5-formyltetrahydrofolate cyclo-ligase [KO:K01934] [EC:6.3.3.2]\n",
      "line in gene PA5228  5-formyltetrahydrofolate cyclo-ligase [KO:K01934] [EC:6.3.3.2]\n",
      "COMPOUND    C00101  Tetrahydrofolate\n",
      "            C00143  5,10-Methylenetetrahydrofolate\n",
      "            C00234  10-Formyltetrahydrofolate\n",
      "            C00415  Dihydrofolate\n",
      "            C00440  5-Methyltetrahydrofolate\n",
      "            C00445  5,10-Methenyltetrahydrofolate\n",
      "            C00504  Folate\n",
      "            C00664  5-Formiminotetrahydrofolate\n",
      "            C03479  Folinic acid\n",
      "REL_PATHWAY pae00790  Folate biosynthesis\n",
      "KO_PATHWAY  ko00670\n",
      "///\n",
      "path:pae00680\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00680                    Pathway\n",
      "NAME        Methane metabolism - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Methane is metabolized principally by methanotrophs and methanogens in the global carbon cycle. Methanotrophs consume methane as the only source of carbon, while methanogens produce methane as a metabolic byproduct. Methylotrophs, which are microorganisms that can obtain energy for growth by oxidizing one-carbon compounds, such as methanol and methane, are situated between methanotrophs and methanogens. Methanogens can obtain energy for growth by converting a limited number of substrates to methane under anaerobic conditions. Three types of methanogenic pathways are known: CO2 to methane [MD:M00567], methanol to methane [MD:M00356], and acetate to methane [MD:M00357]. Methanogens use 2-mercaptoethanesulfonate (CoM; coenzyme M) as the terminal methyl carrier in methanogenesis and have four enzymes for CoM biosynthesis [MD:M00358]. Coenzyme B-Coenzyme M heterodisulfide reductase (Hdr), requiring for the final reaction steps of methanogenic pathway, is divided into two types: cytoplasmic HdrABC in most methanogens and membrane-bound HdrED in Methanosarcina species. In methanotrophs and methyltrophs methane is oxidized to form formaldehyde, which is at the diverging point for further oxidation to CO2 for energy source and assimilation for biosynthesis. There are three pathways that convert formaldehyde to C2 or C3 compounds: serine pathway [MD:M00346], ribulose monophosphate pathway [MD:M00345], and xylulose monophosphate pathway [MD:M00344]. The first two pathways are found in prokaryotes and the third is found in yeast. As a special case of methylotrophs, various amines can be used as carbon sources in trimethylamine metabolism [MD:M00563].\n",
      "CLASS       Metabolism; Energy metabolism\n",
      "PATHWAY_MAP pae00680  Methane metabolism\n",
      "MODULE      pae_M00020  Serine biosynthesis, glycerate-3P => serine [PATH:pae00680]\n",
      "DBLINKS     GO: 0015947\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA5421  fdhA; glutathione-independent formaldehyde dehydrogenase [KO:K00148] [EC:1.2.1.46]\n",
      "line in gene PA5421  fdhA; glutathione-independent formaldehyde dehydrogenase [KO:K00148] [EC:1.2.1.46]\n",
      "            PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "line in gene PA3629  adhC; alcohol dehydrogenase [KO:K00121] [EC:1.1.1.284 1.1.1.1]\n",
      "            PA3628  esterase [KO:K01070] [EC:3.1.2.12]\n",
      "line in gene PA3628  esterase [KO:K01070] [EC:3.1.2.12]\n",
      "            PA4812  fdnG; formate dehydrogenase-O major subunit [KO:K00123] [EC:1.17.1.9]\n",
      "line in gene PA4812  fdnG; formate dehydrogenase-O major subunit [KO:K00123] [EC:1.17.1.9]\n",
      "            PA4811  fdnH; nitrate-inducible formate dehydrogenase subunit beta [KO:K00124]\n",
      "line in gene PA4811  fdnH; nitrate-inducible formate dehydrogenase subunit beta [KO:K00124]\n",
      "            PA4810  fdnI; nitrate-inducible formate dehydrogenase subunit gamma [KO:K00127]\n",
      "line in gene PA4810  fdnI; nitrate-inducible formate dehydrogenase subunit gamma [KO:K00127]\n",
      "            PA2444  glyA2; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA2444  glyA2; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA5415  glyA1; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA5415  glyA1; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA4602  glyA3; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "line in gene PA4602  glyA3; serine hydroxymethyltransferase [KO:K00600] [EC:2.1.2.1]\n",
      "            PA4626  hprA; glycerate dehydrogenase [KO:K00018] [EC:1.1.1.29]\n",
      "line in gene PA4626  hprA; glycerate dehydrogenase [KO:K00018] [EC:1.1.1.29]\n",
      "            PA1499  hypothetical protein [KO:K11529] [EC:2.7.1.165]\n",
      "line in gene PA1499  hypothetical protein [KO:K11529] [EC:2.7.1.165]\n",
      "            PA3635  eno; enolase [KO:K01689] [EC:4.2.1.11]\n",
      "line in gene PA3635  eno; enolase [KO:K01689] [EC:4.2.1.11]\n",
      "            PA3687  ppc; phosphoenolpyruvate carboxylase [KO:K01595] [EC:4.1.1.31]\n",
      "line in gene PA3687  ppc; phosphoenolpyruvate carboxylase [KO:K01595] [EC:4.1.1.31]\n",
      "            PA0555  fda; fructose-1,6-bisphosphate aldolase [KO:K01624] [EC:4.1.2.13]\n",
      "line in gene PA0555  fda; fructose-1,6-bisphosphate aldolase [KO:K01624] [EC:4.1.2.13]\n",
      "            PA5110  fbp; fructose-1,6-bisphosphatase [KO:K03841] [EC:3.1.3.11]\n",
      "line in gene PA5110  fbp; fructose-1,6-bisphosphatase [KO:K03841] [EC:3.1.3.11]\n",
      "            PA0836  ackA; acetate kinase [KO:K00925] [EC:2.7.2.1]\n",
      "line in gene PA0836  ackA; acetate kinase [KO:K00925] [EC:2.7.2.1]\n",
      "            PA0835  pta; phosphate acetyltransferase [KO:K13788] [EC:2.3.1.8]\n",
      "line in gene PA0835  pta; phosphate acetyltransferase [KO:K13788] [EC:2.3.1.8]\n",
      "            PA0887  acsA; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA0887  acsA; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "            PA4733  acsB; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA4733  acsB; acetyl-CoA synthetase [KO:K01895] [EC:6.2.1.1]\n",
      "            PA2555  AMP-binding protein [KO:K01895] [EC:6.2.1.1]\n",
      "line in gene PA2555  AMP-binding protein [KO:K01895] [EC:6.2.1.1]\n",
      "            PA1770  ppsA; phosphoenolpyruvate synthase [KO:K01007] [EC:2.7.9.2]\n",
      "line in gene PA1770  ppsA; phosphoenolpyruvate synthase [KO:K01007] [EC:2.7.9.2]\n",
      "            PA5131  pgm; 2,3-bisphosphoglycerate-independent phosphoglycerate mutase [KO:K15633] [EC:5.4.2.12]\n",
      "line in gene PA5131  pgm; 2,3-bisphosphoglycerate-independent phosphoglycerate mutase [KO:K15633] [EC:5.4.2.12]\n",
      "            PA0316  serA; D-3-phosphoglycerate dehydrogenase [KO:K00058] [EC:1.1.1.95 1.1.1.399]\n",
      "line in gene PA0316  serA; D-3-phosphoglycerate dehydrogenase [KO:K00058] [EC:1.1.1.95 1.1.1.399]\n",
      "            PA3167  serC; 3-phosphoserine/phosphohydroxythreonine aminotransferase [KO:K00831] [EC:2.6.1.52]\n",
      "line in gene PA3167  serC; 3-phosphoserine/phosphohydroxythreonine aminotransferase [KO:K00831] [EC:2.6.1.52]\n",
      "            PA4960  phosphoserine phosphatase [KO:K01079] [EC:3.1.3.3]\n",
      "line in gene PA4960  phosphoserine phosphatase [KO:K01079] [EC:3.1.3.3]\n",
      "            PA1757  thrH; phosphoserine phosphatase [KO:K02203] [EC:3.1.3.3 2.7.1.39]\n",
      "line in gene PA1757  thrH; phosphoserine phosphatase [KO:K02203] [EC:3.1.3.3 2.7.1.39]\n",
      "COMPOUND    C00011  CO2\n",
      "            C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00033  Acetate\n",
      "            C00036  Oxaloacetate\n",
      "            C00037  Glycine\n",
      "            C00048  Glyoxylate\n",
      "            C00058  Formate\n",
      "            C00065  L-Serine\n",
      "            C00067  Formaldehyde\n",
      "            C00074  Phosphoenolpyruvate\n",
      "            C00082  L-Tyrosine\n",
      "            C00085  D-Fructose 6-phosphate\n",
      "            C00101  Tetrahydrofolate\n",
      "            C00111  Glycerone phosphate\n",
      "            C00118  D-Glyceraldehyde 3-phosphate\n",
      "            C00132  Methanol\n",
      "            C00143  5,10-Methylenetetrahydrofolate\n",
      "            C00149  (S)-Malate\n",
      "            C00168  Hydroxypyruvate\n",
      "            C00184  Glycerone\n",
      "            C00197  3-Phospho-D-glycerate\n",
      "            C00199  D-Ribulose 5-phosphate\n",
      "            C00218  Methylamine\n",
      "            C00227  Acetyl phosphate\n",
      "            C00231  D-Xylulose 5-phosphate\n",
      "            C00237  CO\n",
      "            C00258  D-Glycerate\n",
      "            C00322  2-Oxoadipate\n",
      "            C00354  D-Fructose 1,6-bisphosphate\n",
      "            C00483  Tyramine\n",
      "            C00543  Dimethylamine\n",
      "            C00565  Trimethylamine\n",
      "            C00593  Sulfoacetaldehyde\n",
      "            C00631  2-Phospho-D-glycerate\n",
      "            C00862  Methanofuran\n",
      "            C00876  Coenzyme F420\n",
      "            C01001  Formylmethanofuran\n",
      "            C01005  O-Phospho-L-serine\n",
      "            C01031  S-Formylglutathione\n",
      "            C01046  N-Methyl-L-glutamate\n",
      "            C01080  Reduced coenzyme F420\n",
      "            C01104  Trimethylamine N-oxide\n",
      "            C01217  5,6,7,8-Tetrahydromethanopterin\n",
      "            C01274  5-Formyl-5,6,7,8-tetrahydromethanopterin\n",
      "            C01438  Methane\n",
      "            C03232  3-Phosphonooxypyruvate\n",
      "            C03576  Coenzyme M\n",
      "            C03920  2-(Methylthio)ethanesulfonate\n",
      "            C04330  5,10-Methenyltetrahydromethanopterin\n",
      "            C04348  L-Malyl-CoA\n",
      "            C04377  5,10-Methylenetetrahydromethanopterin\n",
      "            C04488  5-Methyl-5,6,7,8-tetrahydromethanopterin\n",
      "            C04628  Coenzyme B\n",
      "            C04732  5-Amino-6-(1-D-ribitylamino)uracil\n",
      "            C04832  Coenzyme M 7-mercaptoheptanoylthreonine-phosphate heterodisulfide\n",
      "            C05528  3-Sulfopyruvate\n",
      "            C06019  D-arabino-Hex-3-ulose 6-phosphate\n",
      "            C11536  (2R)-O-Phospho-3-sulfolactate\n",
      "            C11537  (2R)-3-Sulfolactate\n",
      "            C14180  S-(Hydroxymethyl)glutathione\n",
      "            C16583  (R)-(Homo)2-citrate\n",
      "            C16588  2-Oxopimelate\n",
      "            C16589  2-Oxosuberate\n",
      "            C16590  7-Oxoheptanoic acid\n",
      "            C16593  7-Mercaptoheptanoic acid\n",
      "            C16594  7-Mercaptoheptanoylthreonine\n",
      "            C16597  (-)-threo-Iso(homo)2-citrate\n",
      "            C16598  (R)-(Homo)3-citrate\n",
      "            C16600  (-)-threo-Iso(homo)3-citrate\n",
      "            C18799  5-Methyl-H4SPT\n",
      "            C18802  Tetrahydrosarcinapterin\n",
      "            C19151  Coenzyme F420-3\n",
      "            C19152  Coenzyme F420-1\n",
      "            C19153  Coenzyme F420-0\n",
      "            C19154  7,8-Didemethyl-8-hydroxy-5-deazariboflavin\n",
      "            C19155  (2S)-Lactyl-2-diphospho-5'-guanosine\n",
      "            C19156  (2S)-2-Phospholactate\n",
      "            C20581  cis-(Homo)2-aconitate\n",
      "            C20582  cis-(Homo)3-aconitate\n",
      "            C20926  gamma-Glutamyltyramine\n",
      "            C20954  (5-Formylfuran-3-yl)methyl phosphate\n",
      "            C21068  [5-(Aminomethyl)furan-3-yl]methyl phosphate\n",
      "            C21069  [5-(Aminomethyl)furan-3-yl]methyl diphosphate\n",
      "            C21070  (4-{4-[2-(gamma-L-Glutamylamino)ethyl]phenoxymethyl}furan-2-yl)methanamine\n",
      "            C21971  5-Amino-5-(4-hydroxybenzyl)-6-(D-ribitylimino)-5,6-dihydrouracil\n",
      "            C22277  Dehydro coenzyme F420-0\n",
      "            C22297  Enolpyruvoyl-2-diphospho-5'-guanosine\n",
      "REFERENCE   PMID:11830598\n",
      "  AUTHORS   Graham DE, Xu H, White RH\n",
      "  TITLE     Identification of coenzyme M biosynthetic phosphosulfolactate synthase: a new family of sulfonate-biosynthesizing enzymes.\n",
      "  JOURNAL   J Biol Chem 277:13421-9 (2002)\n",
      "            DOI:10.1074/jbc.M201011200\n",
      "REFERENCE   PMID:15168610\n",
      "  AUTHORS   Deppenmeier U\n",
      "  TITLE     The membrane-bound electron transport system of Methanosarcina species.\n",
      "  JOURNAL   J Bioenerg Biomembr 36:55-64 (2004)\n",
      "            DOI:10.1023/B:JOBB.0000019598.64642.97\n",
      "REFERENCE   PMID:15353801\n",
      "  AUTHORS   Hallam SJ, Putnam N, Preston CM, Detter JC, Rokhsar D, Richardson PM, DeLong EF\n",
      "  TITLE     Reverse methanogenesis: testing the hypothesis with environmental genomics.\n",
      "  JOURNAL   Science 305:1457-62 (2004)\n",
      "            DOI:10.1126/science.1100025\n",
      "REFERENCE   PMID:16024727\n",
      "  AUTHORS   Welander PV, Metcalf WW\n",
      "  TITLE     Loss of the mtr operon in Methanosarcina blocks growth on methanol, but not methanogenesis, and reveals an unknown methanogenic pathway.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 102:10664-9 (2005)\n",
      "            DOI:10.1073/pnas.0502623102\n",
      "REFERENCE   PMID:16278835\n",
      "  AUTHORS   Yurimoto H, Kato N, Sakai Y\n",
      "  TITLE     Assimilation, dissimilation, and detoxification of formaldehyde, a central metabolic intermediate of methylotrophic metabolism.\n",
      "  JOURNAL   Chem Rec 5:367-75 (2005)\n",
      "            DOI:10.1002/tcr.20056\n",
      "REFERENCE   PMID:16385054\n",
      "  AUTHORS   Fricke WF, Seedorf H, Henne A, Kruer M, Liesegang H, Hedderich R, Gottschalk G, Thauer RK.\n",
      "  TITLE     The genome sequence of Methanosphaera stadtmanae reveals why this human intestinal archaeon is restricted to methanol and H2 for methane formation and ATP synthesis.\n",
      "  JOURNAL   J Bacteriol 188:642-58 (2006)\n",
      "            DOI:10.1128/JB.188.2.642-658.2006\n",
      "REFERENCE   PMID:16428816\n",
      "  AUTHORS   Kato N, Yurimoto H, Thauer RK\n",
      "  TITLE     The physiological role of the ribulose monophosphate pathway in bacteria and archaea.\n",
      "  JOURNAL   Biosci Biotechnol Biochem 70:10-21 (2006)\n",
      "            DOI:10.1271/bbb.70.10\n",
      "REFERENCE   PMID:18587410\n",
      "  AUTHORS   Thauer RK, Kaster AK, Seedorf H, Buckel W, Hedderich R\n",
      "  TITLE     Methanogenic archaea: ecologically relevant differences in energy conservation.\n",
      "  JOURNAL   Nat Rev Microbiol 6:579-91 (2008)\n",
      "            DOI:10.1038/nrmicro1931\n",
      "REFERENCE   PMID:20437165\n",
      "  AUTHORS   Liffourrena AS, Salvano MA, Lucchesi GI\n",
      "  TITLE     Pseudomonas putida A ATCC 12633 oxidizes trimethylamine aerobically via two different pathways.\n",
      "  JOURNAL   Arch Microbiol 192:471-6 (2010)\n",
      "            DOI:10.1007/s00203-010-0577-5\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00030  Pentose phosphate pathway\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00300  Lysine biosynthesis\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00740  Riboflavin metabolism\n",
      "            pae00790  Folate biosynthesis\n",
      "            pae00910  Nitrogen metabolism\n",
      "            pae00920  Sulfur metabolism\n",
      "KO_PATHWAY  ko00680\n",
      "///\n",
      "path:pae00730\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00730                    Pathway\n",
      "NAME        Thiamine metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of cofactors and vitamins\n",
      "PATHWAY_MAP pae00730  Thiamine metabolism\n",
      "DBLINKS     GO: 0006772\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4973  thiC; thiamine biosynthesis protein ThiC [KO:K03147] [EC:4.1.99.17]\n",
      "line in gene PA4973  thiC; thiamine biosynthesis protein ThiC [KO:K03147] [EC:4.1.99.17]\n",
      "            PA3975  thiD; phosphomethylpyrimidine kinase [KO:K00941] [EC:2.7.1.49 2.7.4.7]\n",
      "line in gene PA3975  thiD; phosphomethylpyrimidine kinase [KO:K00941] [EC:2.7.1.49 2.7.4.7]\n",
      "            PA3814  iscS; cysteine desulfurase [KO:K04487] [EC:2.8.1.7]\n",
      "line in gene PA3814  iscS; cysteine desulfurase [KO:K04487] [EC:2.8.1.7]\n",
      "            PA2062  pyridoxal-phosphate dependent protein [KO:K04487] [EC:2.8.1.7]\n",
      "line in gene PA2062  pyridoxal-phosphate dependent protein [KO:K04487] [EC:2.8.1.7]\n",
      "            PA5118  thiI; thiamine biosynthesis protein ThiI [KO:K03151] [EC:2.8.1.4]\n",
      "line in gene PA5118  thiI; thiamine biosynthesis protein ThiI [KO:K03151] [EC:2.8.1.4]\n",
      "            PA4548  D-amino acid oxidase [KO:K03153] [EC:1.4.3.19]\n",
      "line in gene PA4548  D-amino acid oxidase [KO:K03153] [EC:1.4.3.19]\n",
      "            PA4044  dxs; 1-deoxy-D-xylulose-5-phosphate synthase [KO:K01662] [EC:2.2.1.7]\n",
      "line in gene PA4044  dxs; 1-deoxy-D-xylulose-5-phosphate synthase [KO:K01662] [EC:2.2.1.7]\n",
      "            PA0381  thiG; thiazole synthase [KO:K03149] [EC:2.8.1.10]\n",
      "line in gene PA0381  thiG; thiazole synthase [KO:K03149] [EC:2.8.1.10]\n",
      "            PA3976  thiE; thiamine-phosphate pyrophosphorylase [KO:K00788] [EC:2.5.1.3]\n",
      "line in gene PA3976  thiE; thiamine-phosphate pyrophosphorylase [KO:K00788] [EC:2.5.1.3]\n",
      "            PA4051  thiL; thiamine monophosphate kinase [KO:K00946] [EC:2.7.4.16]\n",
      "line in gene PA4051  thiL; thiamine monophosphate kinase [KO:K00946] [EC:2.7.4.16]\n",
      "            PA4952  ribosome-associated GTPase [KO:K06949] [EC:3.6.1.- 3.1.3.100]\n",
      "line in gene PA4952  ribosome-associated GTPase [KO:K06949] [EC:3.6.1.- 3.1.3.100]\n",
      "            PA3296  phoA; alkaline phosphatase [KO:K01077] [EC:3.1.3.1]\n",
      "line in gene PA3296  phoA; alkaline phosphatase [KO:K01077] [EC:3.1.3.1]\n",
      "            PA3686  adk; adenylate kinase [KO:K00939] [EC:2.7.4.3]\n",
      "line in gene PA3686  adk; adenylate kinase [KO:K00939] [EC:2.7.4.3]\n",
      "COMPOUND    C00003  NAD+\n",
      "            C00018  Pyridoxal phosphate\n",
      "            C00022  Pyruvate\n",
      "            C00037  Glycine\n",
      "            C00068  Thiamin diphosphate\n",
      "            C00082  L-Tyrosine\n",
      "            C00097  L-Cysteine\n",
      "            C00118  D-Glyceraldehyde 3-phosphate\n",
      "            C00378  Thiamine\n",
      "            C01081  Thiamin monophosphate\n",
      "            C01279  4-Amino-5-hydroxymethyl-2-methylpyrimidine\n",
      "            C02892  Thiamine acetic acid\n",
      "            C03028  Thiamin triphosphate\n",
      "            C03373  Aminoimidazole ribotide\n",
      "            C04294  5-(2-Hydroxyethyl)-4-methylthiazole\n",
      "            C04327  4-Methyl-5-(2-phosphooxyethyl)thiazole\n",
      "            C04556  4-Amino-2-methyl-5-(phosphooxymethyl)pyrimidine\n",
      "            C04752  4-Amino-5-hydroxymethyl-2-methylpyrimidine diphosphate\n",
      "            C05856  Thiamine aldehyde\n",
      "            C11437  1-Deoxy-D-xylulose 5-phosphate\n",
      "            C15809  Iminoglycine\n",
      "            C15810  Sulfur-carrier protein\n",
      "            C15811  [Enzyme]-cysteine\n",
      "            C15812  [Enzyme]-S-sulfanylcysteine\n",
      "            C15813  Adenylyl-[sulfur-carrier protein]\n",
      "            C15814  Thiocarboxy-[sulfur-carrier protein]\n",
      "            C19872  N-Formyl-4-amino-5-aminomethyl-2-methylpyrimidine\n",
      "            C20246  2-[(2R,5Z)-2-Carboxy-4-methylthiazol-5(2H)-ylidene]ethyl phosphate\n",
      "            C20247  2-(2-Carboxy-4-methylthiazol-5-yl)ethyl phosphate\n",
      "            C20267  4-Amino-5-aminomethyl-2-methylpyrimidine\n",
      "            C20784  ADP-5-ethyl-4-methylthiazole-2-carboxylate\n",
      "REFERENCE   PMID:15614489\n",
      "  AUTHORS   Kawasaki Y, Onozuka M, Mizote T, Nosaka K.\n",
      "  TITLE     Biosynthesis of hydroxymethylpyrimidine pyrophosphate in Saccharomyces cerevisiae.\n",
      "  JOURNAL   Curr Genet 47:156-62 (2005)\n",
      "            DOI:10.1007/s00294-004-0557-x\n",
      "REFERENCE   PMID:14567704\n",
      "  AUTHORS   Park JH, Dorrestein PC, Zhai H, Kinsland C, McLafferty FW, Begley TP.\n",
      "  TITLE     Biosynthesis of the thiazole moiety of thiamin pyrophosphate (vitamin B1).\n",
      "  JOURNAL   Biochemistry 42:12430-8 (2003)\n",
      "            DOI:10.1021/bi034902z\n",
      "REFERENCE   PMID:22616866\n",
      "  AUTHORS   Begley TP, Ealick SE, McLafferty FW\n",
      "  TITLE     Thiamin biosynthesis: still yielding fascinating biological chemistry.\n",
      "  JOURNAL   Biochem Soc Trans 40:555-60 (2012)\n",
      "            DOI:10.1042/BST20120084\n",
      "REFERENCE   PMID:12777485\n",
      "  AUTHORS   Wightman R, Meacock PA\n",
      "  TITLE     The THI5 gene family of Saccharomyces cerevisiae: distribution of homologues among the hemiascomycetes and functional redundancy in the aerobic biosynthesis of thiamin from pyridoxine.\n",
      "  JOURNAL   Microbiology 149:1447-60 (2003)\n",
      "            DOI:10.1099/mic.0.26194-0\n",
      "REFERENCE   PMID:22031445\n",
      "  AUTHORS   Chatterjee A, Abeydeera ND, Bale S, Pai PJ, Dorrestein PC, Russell DH, Ealick SE, Begley TP\n",
      "  TITLE     Saccharomyces cerevisiae THI4p is a suicide thiamine thiazole synthase.\n",
      "  JOURNAL   Nature 478:542-6 (2011)\n",
      "            DOI:10.1038/nature10503\n",
      "REFERENCE   PMID:19348578\n",
      "  AUTHORS   Jurgenson CT, Begley TP, Ealick SE\n",
      "  TITLE     The structural and biochemical foundations of thiamin biosynthesis.\n",
      "  JOURNAL   Annu Rev Biochem 78:569-603 (2009)\n",
      "            DOI:10.1146/annurev.biochem.78.072407.102340\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00230  Purine metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "            pae00750  Vitamin B6 metabolism\n",
      "KO_PATHWAY  ko00730\n",
      "///\n",
      "path:pae00740\n",
      "ENTRY       pae00740                    Pathway\n",
      "NAME        Riboflavin metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of cofactors and vitamins\n",
      "PATHWAY_MAP pae00740  Riboflavin metabolism\n",
      "MODULE      pae_M00125  Riboflavin biosynthesis, plants and bacteria, GTP => riboflavin/FMN/FAD [PATH:pae00740]\n",
      "DBLINKS     GO: 0006771\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4047  ribA; GTP cyclohydrolase II [KO:K01497] [EC:3.5.4.25]\n",
      "line in gene PA4047  ribA; GTP cyclohydrolase II [KO:K01497] [EC:3.5.4.25]\n",
      "            PA4054  ribB; 3,4-dihydroxy-2-butanone-4-phosphate synthase [KO:K14652] [EC:4.1.99.12 3.5.4.25]\n",
      "line in gene PA4054  ribB; 3,4-dihydroxy-2-butanone-4-phosphate synthase [KO:K14652] [EC:4.1.99.12 3.5.4.25]\n",
      "            PA4056  ribD; riboflavin-specific deaminase/reductase [KO:K11752] [EC:3.5.4.26 1.1.1.193]\n",
      "line in gene PA4056  ribD; riboflavin-specific deaminase/reductase [KO:K11752] [EC:3.5.4.26 1.1.1.193]\n",
      "            PA5281  hydrolase [KO:K20862] [EC:3.1.3.102 3.1.3.104]\n",
      "line in gene PA5281  hydrolase [KO:K20862] [EC:3.1.3.102 3.1.3.104]\n",
      "            PA4053  ribE; 6,7-dimethyl-8-ribityllumazine synthase [KO:K00794] [EC:2.5.1.78]\n",
      "line in gene PA4053  ribE; 6,7-dimethyl-8-ribityllumazine synthase [KO:K00794] [EC:2.5.1.78]\n",
      "            PA4055  ribC; riboflavin synthase subunit alpha [KO:K00793] [EC:2.5.1.9]\n",
      "line in gene PA4055  ribC; riboflavin synthase subunit alpha [KO:K00793] [EC:2.5.1.9]\n",
      "            PA4561  ribF; bifunctional riboflavin kinase/FMN adenylyltransferase [KO:K11753] [EC:2.7.1.26 2.7.7.2]\n",
      "line in gene PA4561  ribF; bifunctional riboflavin kinase/FMN adenylyltransferase [KO:K11753] [EC:2.7.1.26 2.7.7.2]\n",
      "            PA0190  acid phosphatase [KO:K09474] [EC:3.1.3.2]\n",
      "line in gene PA0190  acid phosphatase [KO:K09474] [EC:3.1.3.2]\n",
      "            PA4092  hpaC; 4-hydroxyphenylacetate 3-monooxygenase small subunit [KO:K00484] [EC:1.5.1.36]\n",
      "line in gene PA4092  hpaC; 4-hydroxyphenylacetate 3-monooxygenase small subunit [KO:K00484] [EC:1.5.1.36]\n",
      "            PA2357  msuE; FMN reductase MsuE [KO:K00299] [EC:1.5.1.38]\n",
      "line in gene PA2357  msuE; FMN reductase MsuE [KO:K00299] [EC:1.5.1.38]\n",
      "            PA3446  NAD(P)H-dependent FMN reductase [KO:K00299] [EC:1.5.1.38]\n",
      "line in gene PA3446  NAD(P)H-dependent FMN reductase [KO:K00299] [EC:1.5.1.38]\n",
      "            PA1274  5,6-dimethylbenzimidazole synthase [KO:K04719] [EC:1.13.11.79]\n",
      "line in gene PA1274  5,6-dimethylbenzimidazole synthase [KO:K04719] [EC:1.13.11.79]\n",
      "            PA4971  aspP; adenosine diphosphate sugar pyrophosphatase [KO:K01515] [EC:3.6.1.13 3.6.1.-]\n",
      "line in gene PA4971  aspP; adenosine diphosphate sugar pyrophosphatase [KO:K01515] [EC:3.6.1.13 3.6.1.-]\n",
      "            PA4019  aromatic acid decarboxylase [KO:K03186] [EC:2.5.1.129]\n",
      "line in gene PA4019  aromatic acid decarboxylase [KO:K03186] [EC:2.5.1.129]\n",
      "COMPOUND    C00016  FAD\n",
      "            C00044  GTP\n",
      "            C00061  FMN\n",
      "            C00199  D-Ribulose 5-phosphate\n",
      "            C00235  Dimethylallyl diphosphate\n",
      "            C00255  Riboflavin\n",
      "            C00474  Ribitol\n",
      "            C01007  Reduced riboflavin\n",
      "            C01268  5-Amino-6-(5'-phosphoribosylamino)uracil\n",
      "            C01304  2,5-Diamino-6-(5-phospho-D-ribosylamino)pyrimidin-4(3H)-one\n",
      "            C01352  FADH2\n",
      "            C01390  Prenol\n",
      "            C01727  Lumichrome\n",
      "            C01847  Reduced FMN\n",
      "            C03114  Dimethylbenzimidazole\n",
      "            C04332  6,7-Dimethyl-8-(D-ribityl)lumazine\n",
      "            C04454  5-Amino-6-(5'-phospho-D-ribitylamino)uracil\n",
      "            C04732  5-Amino-6-(1-D-ribitylamino)uracil\n",
      "            C05995  7-Hydroxy-6-methyl-8-ribityllumazine\n",
      "            C15556  L-3,4-Dihydroxybutan-2-one 4-phosphate\n",
      "            C15563  2-Amino-5-formylamino-6-(5-phospho-D-ribosylamino)pyrimidin-4(3H)-one\n",
      "            C18910  2,5-Diamino-6-(5-phospho-D-ribitylamino)pyrimidin-4(3H)-one\n",
      "            C21214  Dimethylallyl phosphate\n",
      "            C21215  Prenylated FMNH2\n",
      "REFERENCE   PMID:12136096\n",
      "  AUTHORS   Vitreschak AG, Rodionov DA, Mironov AA, Gelfand MS.\n",
      "  TITLE     Regulation of riboflavin biosynthesis and transport genes in bacteria by transcriptional and translational attenuation.\n",
      "  JOURNAL   Nucleic Acids Res 30:3141-51 (2002)\n",
      "            DOI:10.1093/nar/30.14.3141\n",
      "REFERENCE   PMID:11856310\n",
      "  AUTHORS   Fischer M, Haase I, Feicht R, Richter G, Gerhardt S, Changeux JP, Huber R, Bacher A.\n",
      "  TITLE     Biosynthesis of riboflavin: 6,7-dimethyl-8-ribityllumazine synthase of Schizosaccharomyces pombe.\n",
      "  JOURNAL   Eur J Biochem 269:519-26 (2002)\n",
      "            DOI:10.1046/j.0014-2956.2001.02674.x\n",
      "REFERENCE   PMID:12867427\n",
      "  AUTHORS   Haase I, Fischer M, Bacher A, Schramek N.\n",
      "  TITLE     Temperature-dependent presteady state kinetics of lumazine synthase from the hyperthermophilic eubacterium Aquifex aeolicus.\n",
      "  JOURNAL   J Biol Chem 278:37909-15 (2003)\n",
      "            DOI:10.1074/jbc.M303090200\n",
      "REFERENCE   PMID:16042598\n",
      "  AUTHORS   Fischer M, Romisch W, Illarionov B, Eisenreich W, Bacher A.\n",
      "  TITLE     Structures and reaction mechanisms of riboflavin synthases of eubacterial and archaeal origin.\n",
      "  JOURNAL   Biochem Soc Trans 33:780-4 (2005)\n",
      "            DOI:10.1042/BST0330780\n",
      "REL_PATHWAY pae00030  Pentose phosphate pathway\n",
      "            pae00040  Pentose and glucuronate interconversions\n",
      "            pae00230  Purine metabolism\n",
      "            pae00860  Porphyrin metabolism\n",
      "            pae00900  Terpenoid backbone biosynthesis\n",
      "KO_PATHWAY  ko00740\n",
      "///\n",
      "path:pae00750\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00750                    Pathway\n",
      "NAME        Vitamin B6 metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of cofactors and vitamins\n",
      "PATHWAY_MAP pae00750  Vitamin B6 metabolism\n",
      "MODULE      pae_M00124  Pyridoxal-P biosynthesis, erythrose-4P => pyridoxal-P [PATH:pae00750]\n",
      "DBLINKS     GO: 0042816\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1049  pdxH; pyridoxine/pyridoxamine 5'-phosphate oxidase [KO:K00275] [EC:1.4.3.5]\n",
      "line in gene PA1049  pdxH; pyridoxine/pyridoxamine 5'-phosphate oxidase [KO:K00275] [EC:1.4.3.5]\n",
      "            PA5516  pdxY; pyridoxamine kinase [KO:K00868] [EC:2.7.1.35]\n",
      "line in gene PA5516  pdxY; pyridoxamine kinase [KO:K00868] [EC:2.7.1.35]\n",
      "            PA0551  epd; D-erythrose 4-phosphate dehydrogenase [KO:K03472] [EC:1.2.1.72]\n",
      "line in gene PA0551  epd; D-erythrose 4-phosphate dehydrogenase [KO:K03472] [EC:1.2.1.72]\n",
      "            PA1375  pdxB; erythronate-4-phosphate dehydrogenase [KO:K03473] [EC:1.1.1.290]\n",
      "line in gene PA1375  pdxB; erythronate-4-phosphate dehydrogenase [KO:K03473] [EC:1.1.1.290]\n",
      "            PA3167  serC; 3-phosphoserine/phosphohydroxythreonine aminotransferase [KO:K00831] [EC:2.6.1.52]\n",
      "line in gene PA3167  serC; 3-phosphoserine/phosphohydroxythreonine aminotransferase [KO:K00831] [EC:2.6.1.52]\n",
      "            PA0593  pdxA; 4-hydroxythreonine-4-phosphate dehydrogenase [KO:K00097] [EC:1.1.1.262]\n",
      "line in gene PA0593  pdxA; 4-hydroxythreonine-4-phosphate dehydrogenase [KO:K00097] [EC:1.1.1.262]\n",
      "            PA2212  4-hydroxythreonine-4-phosphate dehydrogenase [KO:K00097] [EC:1.1.1.262]\n",
      "line in gene PA2212  4-hydroxythreonine-4-phosphate dehydrogenase [KO:K00097] [EC:1.1.1.262]\n",
      "            PA0773  pdxJ; pyridoxine 5'-phosphate synthase [KO:K03474] [EC:2.6.99.2]\n",
      "line in gene PA0773  pdxJ; pyridoxine 5'-phosphate synthase [KO:K03474] [EC:2.6.99.2]\n",
      "            PA3735  thrC; threonine synthase [KO:K01733] [EC:4.2.3.1]\n",
      "line in gene PA3735  thrC; threonine synthase [KO:K01733] [EC:4.2.3.1]\n",
      "COMPOUND    C00018  Pyridoxal phosphate\n",
      "            C00064  L-Glutamine\n",
      "            C00117  D-Ribose 5-phosphate\n",
      "            C00118  D-Glyceraldehyde 3-phosphate\n",
      "            C00232  Succinate semialdehyde\n",
      "            C00250  Pyridoxal\n",
      "            C00266  Glycolaldehyde\n",
      "            C00279  D-Erythrose 4-phosphate\n",
      "            C00314  Pyridoxine\n",
      "            C00534  Pyridoxamine\n",
      "            C00627  Pyridoxine phosphate\n",
      "            C00647  Pyridoxamine phosphate\n",
      "            C00847  4-Pyridoxate\n",
      "            C00971  4-Pyridoxolactone\n",
      "            C01215  2-(Acetamidomethylene)succinate\n",
      "            C01270  3-Hydroxy-2-methylpyridine-5-carboxylate\n",
      "            C03393  4-Phospho-D-erythronate\n",
      "            C04106  2-(Hydroxymethyl)-4-oxobutanoate\n",
      "            C04604  3-Hydroxy-2-methylpyridine-4,5-dicarboxylate\n",
      "            C04690  2-(Hydroxymethyl)-3-(acetamidomethylene)succinate\n",
      "            C04773  3-Hydroxy-4-hydroxymethyl-2-methylpyridine-5-carboxylate\n",
      "            C06050  2-Methyl-3-hydroxy-5-formylpyridine-4-carboxylate\n",
      "            C06051  Isopyridoxal\n",
      "            C06052  5-Pyridoxolactone\n",
      "            C06054  2-Oxo-3-hydroxy-4-phosphobutanoate\n",
      "            C06055  O-Phospho-4-hydroxy-L-threonine\n",
      "            C06056  4-Hydroxy-L-threonine\n",
      "            C07335  2-Amino-3-oxo-4-phosphonooxybutyrate\n",
      "            C11638  3-Amino-2-oxopropyl phosphate\n",
      "REFERENCE   PMID:10753941\n",
      "  AUTHORS   Tazoe M, Ichikawa K, Hoshino T.\n",
      "  TITLE     Biosynthesis of vitamin B(6) in rhizobium.\n",
      "  JOURNAL   J Biol Chem 275:11300-5 (2000)\n",
      "            DOI:10.1074/jbc.275.15.11300\n",
      "REFERENCE   PMID:9696782\n",
      "  AUTHORS   Yang Y, Zhao G, Man TK, Winkler ME.\n",
      "  TITLE     Involvement of the gapA- and epd (gapB)-encoded dehydrogenases in pyridoxal 5'-phosphate coenzyme biosynthesis in Escherichia coli K-12.\n",
      "  JOURNAL   J Bacteriol 180:4294-9 (1998)\n",
      "            DOI:10.1128/JB.180.16.4294-4299.1998\n",
      "REFERENCE   PMID:12269807\n",
      "  AUTHORS   Yeh JI, Du S, Pohl E, Cane DE.\n",
      "  TITLE     Multistate binding in pyridoxine 5'-phosphate synthase: 1.96 A crystal structure in complex with 1-deoxy-D-xylulose phosphate.\n",
      "  JOURNAL   Biochemistry 41:11649-57 (2002)\n",
      "            DOI:10.1021/bi026292t\n",
      "REFERENCE   PMID:10225425\n",
      "  AUTHORS   Laber B, Maurer W, Scharf S, Stepusin K, Schmidt FS.\n",
      "  TITLE     Vitamin B6 biosynthesis: formation of pyridoxine 5'-phosphate from 4-(phosphohydroxy)-L-threonine and 1-deoxy-D-xylulose-5-phosphate by PdxA and PdxJ protein.\n",
      "  JOURNAL   FEBS Lett 449:45-8 (1999)\n",
      "            DOI:10.1016/S0014-5793(99)00393-2\n",
      "REFERENCE   PMID:16157873\n",
      "  AUTHORS   Tambasco-Studart M, Titiz O, Raschle T, Forster G, Amrhein N, Fitzpatrick TB.\n",
      "  TITLE     Vitamin B6 biosynthesis in higher plants.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 102:13687-92 (2005)\n",
      "            DOI:10.1073/pnas.0506228102\n",
      "REFERENCE   PMID:22276911\n",
      "  AUTHORS   Nagase T, Mugo AN, Chu HN, Yoshikane Y, Ohnishi K, Yagi T\n",
      "  TITLE     The mll6786 gene encodes a repressor protein controlling the degradation pathway for vitamin B6 in Mesorhizobium loti.\n",
      "  JOURNAL   FEMS Microbiol Lett 329:116-22 (2012)\n",
      "            DOI:10.1111/j.1574-6968.2012.02510.x\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00030  Pentose phosphate pathway\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00650  Butanoate metabolism\n",
      "            pae00730  Thiamine metabolism\n",
      "KO_PATHWAY  ko00750\n",
      "///\n",
      "path:pae00760\n",
      "ENTRY       pae00760                    Pathway\n",
      "NAME        Nicotinate and nicotinamide metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of cofactors and vitamins\n",
      "PATHWAY_MAP pae00760  Nicotinate and nicotinamide metabolism\n",
      "MODULE      pae_M00115  NAD biosynthesis, aspartate => quinolinate => NAD [PATH:pae00760]\n",
      "DBLINKS     UMBBD: nic\n",
      "            GO: 0046497 0006769\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0761  nadB; L-aspartate oxidase [KO:K00278] [EC:1.4.3.16]\n",
      "line in gene PA0761  nadB; L-aspartate oxidase [KO:K00278] [EC:1.4.3.16]\n",
      "            PA3505  L-aspartate dehydrogenase [KO:K06989] [EC:1.4.1.21]\n",
      "line in gene PA3505  L-aspartate dehydrogenase [KO:K06989] [EC:1.4.1.21]\n",
      "            PA1004  nadA; quinolinate synthetase [KO:K03517] [EC:2.5.1.72]\n",
      "line in gene PA1004  nadA; quinolinate synthetase [KO:K03517] [EC:2.5.1.72]\n",
      "            PA4524  nadC; nicotinate-nucleotide pyrophosphorylase [KO:K00767] [EC:2.4.2.19]\n",
      "line in gene PA4524  nadC; nicotinate-nucleotide pyrophosphorylase [KO:K00767] [EC:2.4.2.19]\n",
      "            PA4919  pncB1; nicotinate phosphoribosyltransferase [KO:K00763] [EC:6.3.4.21]\n",
      "line in gene PA4919  pncB1; nicotinate phosphoribosyltransferase [KO:K00763] [EC:6.3.4.21]\n",
      "            PA4376  pncB2; nicotinate phosphoribosyltransferase [KO:K00763] [EC:6.3.4.21]\n",
      "line in gene PA4376  pncB2; nicotinate phosphoribosyltransferase [KO:K00763] [EC:6.3.4.21]\n",
      "            PA3625  surE; 5'-nucleotidase SurE [KO:K03787] [EC:3.1.3.5 3.1.3.6]\n",
      "line in gene PA3625  surE; 5'-nucleotidase SurE [KO:K03787] [EC:3.1.3.5 3.1.3.6]\n",
      "            PA4006  nadD; nicotinic acid mononucleotide adenylyltransferase [KO:K00969] [EC:2.7.7.18]\n",
      "line in gene PA4006  nadD; nicotinic acid mononucleotide adenylyltransferase [KO:K00969] [EC:2.7.7.18]\n",
      "            PA0935  nucleoside triphosphate pyrophosphohydrolase [KO:K04765] [EC:3.6.1.9]\n",
      "line in gene PA0935  nucleoside triphosphate pyrophosphohydrolase [KO:K04765] [EC:3.6.1.9]\n",
      "            PA1823  NADH pyrophosphatase [KO:K03426] [EC:3.6.1.22]\n",
      "line in gene PA1823  NADH pyrophosphatase [KO:K03426] [EC:3.6.1.22]\n",
      "            PA4920  nadE; NAD synthetase [KO:K01916] [EC:6.3.1.5]\n",
      "line in gene PA4920  nadE; NAD synthetase [KO:K01916] [EC:6.3.1.5]\n",
      "            PA3618  hypothetical protein [KO:K03743] [EC:3.5.1.42]\n",
      "line in gene PA3618  hypothetical protein [KO:K03743] [EC:3.5.1.42]\n",
      "            PA2141  hypothetical protein [KO:K03743] [EC:3.5.1.42]\n",
      "line in gene PA2141  hypothetical protein [KO:K03743] [EC:3.5.1.42]\n",
      "            PA4918  hypothetical protein [KO:K08281] [EC:3.5.1.19 3.5.1.-]\n",
      "line in gene PA4918  hypothetical protein [KO:K08281] [EC:3.5.1.19 3.5.1.-]\n",
      "            PA1125  NAD-dependent protein deacylase [KO:K12410] [EC:2.3.1.286]\n",
      "line in gene PA1125  NAD-dependent protein deacylase [KO:K12410] [EC:2.3.1.286]\n",
      "            PA1197  NAD-dependent protein deacylase [KO:K12410] [EC:2.3.1.286]\n",
      "line in gene PA1197  NAD-dependent protein deacylase [KO:K12410] [EC:2.3.1.286]\n",
      "            PA3088  NAD kinase [KO:K00858] [EC:2.7.1.23]\n",
      "line in gene PA3088  NAD kinase [KO:K00858] [EC:2.7.1.23]\n",
      "            PA2991  sth; soluble pyridine nucleotide transhydrogenase [KO:K00322] [EC:1.6.1.1]\n",
      "line in gene PA2991  sth; soluble pyridine nucleotide transhydrogenase [KO:K00322] [EC:1.6.1.1]\n",
      "            PA0195  pntAA; NAD(P) transhydrogenase subunit alpha [KO:K00324] [EC:1.6.1.2 7.1.1.1]\n",
      "line in gene PA0195  pntAA; NAD(P) transhydrogenase subunit alpha [KO:K00324] [EC:1.6.1.2 7.1.1.1]\n",
      "            PA0195.1  pntAB; NAD(P) transhydrogenase subunit alpha [KO:K00324] [EC:1.6.1.2 7.1.1.1]\n",
      "line in gene PA0195.1  pntAB; NAD(P) transhydrogenase subunit alpha [KO:K00324] [EC:1.6.1.2 7.1.1.1]\n",
      "            PA0196  pntB; pyridine nucleotide transhydrogenase subunit beta [KO:K00325] [EC:1.6.1.2 7.1.1.1]\n",
      "line in gene PA0196  pntB; pyridine nucleotide transhydrogenase subunit beta [KO:K00325] [EC:1.6.1.2 7.1.1.1]\n",
      "            PA1602  oxidoreductase [KO:K18029] [EC:1.17.2.1]\n",
      "line in gene PA1602  oxidoreductase [KO:K18029] [EC:1.17.2.1]\n",
      "            PA1601  aldehyde dehydrogenase [KO:K18030] [EC:1.17.2.1]\n",
      "line in gene PA1601  aldehyde dehydrogenase [KO:K18030] [EC:1.17.2.1]\n",
      "            PA0265  gabD; glutarate-semialdehyde dehydrogenase DavD [KO:K00135] [EC:1.2.1.16 1.2.1.79 1.2.1.20]\n",
      "line in gene PA0265  gabD; glutarate-semialdehyde dehydrogenase DavD [KO:K00135] [EC:1.2.1.16 1.2.1.79 1.2.1.20]\n",
      "COMPOUND    C00003  NAD+\n",
      "            C00006  NADP+\n",
      "            C00022  Pyruvate\n",
      "            C00042  Succinate\n",
      "            C00049  L-Aspartate\n",
      "            C00111  Glycerone phosphate\n",
      "            C00122  Fumarate\n",
      "            C00153  Nicotinamide\n",
      "            C00163  Propanoate\n",
      "            C00232  Succinate semialdehyde\n",
      "            C00253  Nicotinate\n",
      "            C00334  4-Aminobutanoate\n",
      "            C00455  Nicotinamide D-ribonucleotide\n",
      "            C00857  Deamino-NAD+\n",
      "            C00922  2,3-Dimethylmaleate\n",
      "            C01004  N-Methylnicotinate\n",
      "            C01020  6-Hydroxynicotinate\n",
      "            C01056  (S)-6-Hydroxynicotine\n",
      "            C01059  2,5-Dihydroxypyridine\n",
      "            C01185  Nicotinate D-ribonucleotide\n",
      "            C01297  6-Hydroxypseudooxynicotine\n",
      "            C01384  Maleic acid\n",
      "            C01596  Maleamate\n",
      "            C02295  Methylitaconate\n",
      "            C02918  1-Methylnicotinamide\n",
      "            C02930  2-Methyleneglutarate\n",
      "            C03043  (R)-6-Hydroxynicotine\n",
      "            C03056  2,6-Dihydroxypyridine\n",
      "            C03150  Nicotinamide-beta-riboside\n",
      "            C03458  2,3,6-Trihydroxypyridine\n",
      "            C03652  (2R,3S)-2,3-Dimethylmalate\n",
      "            C03722  Quinolinate\n",
      "            C04226  6-Oxo-1,4,5,6-tetrahydronicotinate\n",
      "            C05380  Nicotinurate\n",
      "            C05840  Iminoaspartate\n",
      "            C05841  Nicotinate D-ribonucleoside\n",
      "            C05842  N1-Methyl-2-pyridone-5-carboxamide\n",
      "            C05843  N1-Methyl-4-pyridone-5-carboxamide\n",
      "            C06178  1-Methylpyrrolinium\n",
      "            C15523  2,6-Dihydroxynicotinate\n",
      "            C15986  2,6-Dihydroxypseudooxynicotine\n",
      "            C15987  4-Methylaminobutyrate\n",
      "            C16150  (R,S)-Nicotine\n",
      "            C16151  2,6-Dihydroxy-N-methylmyosmine\n",
      "            C16152  Blue pigment\n",
      "            C16159  2-Formylglutarate\n",
      "            C16390  (S)-2-(Hydroxymethyl)glutarate\n",
      "            C18232  N-Formylmaleamic acid\n",
      "            C19567  4-Oxo-1-(3-pyridyl)-1-butanone\n",
      "            C19569  3-Succinoylpyridine\n",
      "            C19631  6-Hydroxy-3-succinoylpyridine\n",
      "            C20361  Pseudooxynicotine\n",
      "            C20381  3,3'-Bipyridine-2,2',5,5',6,6'-hexol\n",
      "            C21075  5-(N-Methyl-4,5-dihydro-1H-pyrrol-2-yl)pyridin-2-ol\n",
      "            C21154  N-Methylmyosmine\n",
      "REFERENCE   PMID:13475371\n",
      "  AUTHORS   BEHRMAN EJ, STANIER RY.\n",
      "  TITLE     The bacterial oxidation of nicotinic acid.\n",
      "  JOURNAL   J Biol Chem 228:923-45 (1957)\n",
      "REFERENCE   PMID:16894175\n",
      "  AUTHORS   Alhapel A, Darley DJ, Wagener N, Eckel E, Elsner N, Pierik AJ.\n",
      "  TITLE     Molecular and functional analysis of nicotinate catabolism in Eubacterium barkeri.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 103:12341-6 (2006)\n",
      "            DOI:10.1073/pnas.0601635103\n",
      "REFERENCE   PMID:11514508\n",
      "  AUTHORS   Baitsch D, Sandu C, Brandsch R, Igloi GL.\n",
      "  TITLE     Gene cluster on pAO1 of Arthrobacter nicotinovorans involved in degradation of the plant alkaloid nicotine: cloning, purification, and characterization of 2,6-dihydroxypyridine 3-hydroxylase.\n",
      "  JOURNAL   J Bacteriol 183:5262-7 (2001)\n",
      "            DOI:10.1128/JB.183.18.5262-5267.2001\n",
      "REFERENCE   PMID:16321959\n",
      "  AUTHORS   Sachelaru P, Schiltz E, Igloi GL, Brandsch R.\n",
      "  TITLE     An alpha/beta-fold C--C bond hydrolase is involved in a central step of nicotine catabolism by Arthrobacter nicotinovorans.\n",
      "  JOURNAL   J Bacteriol 187:8516-9 (2005)\n",
      "            DOI:10.1128/JB.187.24.8516-8519.2005\n",
      "REFERENCE   PMID:5835946\n",
      "  AUTHORS   Gherna RL, Richardson SH, Rittenberg SC\n",
      "  TITLE     The bacterial oxidation of nicotine. VI. The metabolism of 2,6-dihydroxypseudooxynicotine.\n",
      "  JOURNAL   J Biol Chem 240:3669-74 (1965)\n",
      "REFERENCE   PMID:21953451\n",
      "  AUTHORS   Galeazzi L, Bocci P, Amici A, Brunetti L, Ruggieri S, Romine M, Reed S, Osterman AL, Rodionov DA, Sorci L, Raffaelli N\n",
      "  TITLE     Identification of nicotinamide mononucleotide deamidase of the bacterial pyridine nucleotide cycle reveals a novel broadly conserved amidohydrolase family.\n",
      "  JOURNAL   J Biol Chem 286:40365-75 (2011)\n",
      "            DOI:10.1074/jbc.M111.275818\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00380  Tryptophan metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00660  C5-Branched dibasic acid metabolism\n",
      "KO_PATHWAY  ko00760\n",
      "///\n",
      "path:pae00770\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00770                    Pathway\n",
      "NAME        Pantothenate and CoA biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of cofactors and vitamins\n",
      "PATHWAY_MAP pae00770  Pantothenate and CoA biosynthesis\n",
      "MODULE      pae_M00019  Valine/isoleucine biosynthesis, pyruvate => valine / 2-oxobutanoate => isoleucine [PATH:pae00770]\n",
      "            pae_M00119  Pantothenate biosynthesis, valine/L-aspartate => pantothenate [PATH:pae00770]\n",
      "            pae_M00120  Coenzyme A biosynthesis, pantothenate => CoA [PATH:pae00770]\n",
      "DBLINKS     GO: 0015940 0015937\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2035  thiamine pyrophosphate protein [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA2035  thiamine pyrophosphate protein [KO:K01652] [EC:2.2.1.6]\n",
      "            PA4180  acetolactate synthase [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA4180  acetolactate synthase [KO:K01652] [EC:2.2.1.6]\n",
      "            PA4696  ilvI; acetolactate synthase 3 catalytic subunit [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA4696  ilvI; acetolactate synthase 3 catalytic subunit [KO:K01652] [EC:2.2.1.6]\n",
      "            PA1417  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA1417  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "            PA3506  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "line in gene PA3506  hypothetical protein [KO:K01652] [EC:2.2.1.6]\n",
      "            PA4695  ilvH; acetolactate synthase small subunit [KO:K01653] [EC:2.2.1.6]\n",
      "line in gene PA4695  ilvH; acetolactate synthase small subunit [KO:K01653] [EC:2.2.1.6]\n",
      "            PA4694  ilvC; ketol-acid reductoisomerase [KO:K00053] [EC:1.1.1.86]\n",
      "line in gene PA4694  ilvC; ketol-acid reductoisomerase [KO:K00053] [EC:1.1.1.86]\n",
      "            PA0353  ilvD; dihydroxy-acid dehydratase [KO:K01687] [EC:4.2.1.9]\n",
      "line in gene PA0353  ilvD; dihydroxy-acid dehydratase [KO:K01687] [EC:4.2.1.9]\n",
      "            PA4729  panB; 3-methyl-2-oxobutanoate hydroxymethyltransferase [KO:K00606] [EC:2.1.2.11]\n",
      "line in gene PA4729  panB; 3-methyl-2-oxobutanoate hydroxymethyltransferase [KO:K00606] [EC:2.1.2.11]\n",
      "            PA1598  3-methyl-2-oxobutanoate hydroxymethyltransferase [KO:K00606] [EC:2.1.2.11]\n",
      "line in gene PA1598  3-methyl-2-oxobutanoate hydroxymethyltransferase [KO:K00606] [EC:2.1.2.11]\n",
      "            PA4397  panE; 2-dehydropantoate 2-reductase [KO:K00077] [EC:1.1.1.169]\n",
      "line in gene PA4397  panE; 2-dehydropantoate 2-reductase [KO:K00077] [EC:1.1.1.169]\n",
      "            PA1752  2-dehydropantoate 2-reductase [KO:K00077] [EC:1.1.1.169]\n",
      "line in gene PA1752  2-dehydropantoate 2-reductase [KO:K00077] [EC:1.1.1.169]\n",
      "            PA4730  panC; pantoate--beta-alanine ligase [KO:K01918] [EC:6.3.2.1]\n",
      "line in gene PA4730  panC; pantoate--beta-alanine ligase [KO:K01918] [EC:6.3.2.1]\n",
      "            PA4279  pantothenate kinase [KO:K03525] [EC:2.7.1.33]\n",
      "line in gene PA4279  pantothenate kinase [KO:K03525] [EC:2.7.1.33]\n",
      "            PA5320  coaC; bifunctional phosphopantothenoylcysteine decarboxylase/phosphopantothenate synthase [KO:K13038] [EC:4.1.1.36 6.3.2.5]\n",
      "line in gene PA5320  coaC; bifunctional phosphopantothenoylcysteine decarboxylase/phosphopantothenate synthase [KO:K13038] [EC:4.1.1.36 6.3.2.5]\n",
      "            PA0935  nucleoside triphosphate pyrophosphohydrolase [KO:K04765] [EC:3.6.1.9]\n",
      "line in gene PA0935  nucleoside triphosphate pyrophosphohydrolase [KO:K04765] [EC:3.6.1.9]\n",
      "            PA0363  coaD; phosphopantetheine adenylyltransferase [KO:K00954] [EC:2.7.7.3]\n",
      "line in gene PA0363  coaD; phosphopantetheine adenylyltransferase [KO:K00954] [EC:2.7.7.3]\n",
      "            PA4529  coaE; dephospho-CoA kinase [KO:K00859] [EC:2.7.1.24]\n",
      "line in gene PA4529  coaE; dephospho-CoA kinase [KO:K00859] [EC:2.7.1.24]\n",
      "            PA4353  hypothetical protein [KO:K08682] [EC:3.1.4.14]\n",
      "line in gene PA4353  hypothetical protein [KO:K08682] [EC:3.1.4.14]\n",
      "            PA5013  ilvE; branched-chain amino acid aminotransferase [KO:K00826] [EC:2.6.1.42]\n",
      "line in gene PA5013  ilvE; branched-chain amino acid aminotransferase [KO:K00826] [EC:2.6.1.42]\n",
      "            PA0440  oxidoreductase [KO:K17722] [EC:1.3.1.1]\n",
      "line in gene PA0440  oxidoreductase [KO:K17722] [EC:1.3.1.1]\n",
      "            PA0439  dihydropyrimidine dehydrogenase [KO:K17723] [EC:1.3.1.1]\n",
      "line in gene PA0439  dihydropyrimidine dehydrogenase [KO:K17723] [EC:1.3.1.1]\n",
      "            PA0441  dht; D-hydantoinase/dihydropyrimidinase [KO:K01464] [EC:3.5.2.2]\n",
      "line in gene PA0441  dht; D-hydantoinase/dihydropyrimidinase [KO:K01464] [EC:3.5.2.2]\n",
      "            PA4731  panD; aspartate alpha-decarboxylase [KO:K01579] [EC:4.1.1.11]\n",
      "line in gene PA4731  panD; aspartate alpha-decarboxylase [KO:K01579] [EC:4.1.1.11]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "COMPOUND    C00010  CoA\n",
      "            C00022  Pyruvate\n",
      "            C00049  L-Aspartate\n",
      "            C00054  Adenosine 3',5'-bisphosphate\n",
      "            C00097  L-Cysteine\n",
      "            C00099  beta-Alanine\n",
      "            C00106  Uracil\n",
      "            C00141  3-Methyl-2-oxobutanoic acid\n",
      "            C00183  L-Valine\n",
      "            C00229  Acyl-carrier protein\n",
      "            C00429  5,6-Dihydrouracil\n",
      "            C00522  (R)-Pantoate\n",
      "            C00750  Spermine\n",
      "            C00831  Pantetheine\n",
      "            C00864  Pantothenate\n",
      "            C00882  Dephospho-CoA\n",
      "            C00966  2-Dehydropantoate\n",
      "            C01053  (R)-4-Dehydropantoate\n",
      "            C01088  (R)-3,3-Dimethylmalate\n",
      "            C01134  Pantetheine 4'-phosphate\n",
      "            C02642  3-Ureidopropionate\n",
      "            C03492  D-4'-Phosphopantothenate\n",
      "            C03688  Apo-[acyl-carrier-protein]\n",
      "            C04079  N-((R)-Pantothenoyl)-L-cysteine\n",
      "            C04272  (R)-2,3-Dihydroxy-3-methylbutanoate\n",
      "            C04352  (R)-4'-Phosphopantothenoyl-L-cysteine\n",
      "            C05665  3-Aminopropanal\n",
      "            C05944  Pantothenol\n",
      "            C06010  (S)-2-Acetolactate\n",
      "            C18911  (R)-4-Phosphopantoate\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00330  Arginine and proline metabolism\n",
      "            pae00410  beta-Alanine metabolism\n",
      "            pae00640  Propanoate metabolism\n",
      "KO_PATHWAY  ko00770\n",
      "///\n",
      "path:pae00780\n",
      "ENTRY       pae00780                    Pathway\n",
      "NAME        Biotin metabolism - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Biotin (vitamin H or vitamin B7) is the essential cofactor of biotin-dependent carboxylases, such as pyruvate carboxylase and acetyl-CoA carboxylase. Mammals cannot synthesize biotin, while in bacteria, fungi, and plants it is synthesized from pimelate thioester through different pathways. In E. coli and many organisms, pimelate thioester is derived from malonyl-ACP. The pathway starts with the methylation to malonyl-ACP methyl ester, followed by the fatty acid chain elongation cycle to form pimeloyl-ACP methyl ester, which is then demethylated to form pimeloyl-ACP [MD:M00572]. Pimeloyl-ACP is converted to biotin through the final four steps in the biotin bicyclic ring assembly, which are conserved among biotin-producing organisms [MD:M00123]. In B. subtilis, biotin is derived from pimeloyl-ACP formed by oxidative cleavage of long-chain acyl-ACPs [MD:M00573]. Some bacteria synthesize biotin from pimeloyl-CoA derived from pimelate [MD:M00577]. Biotin is covalently attached to biotin-dependent carboxylase by biotin protein ligase, also known as holocarboxylase synthase in mammals, to form an active holocarboxylase. After degradation of the biotinylated carboxylase into biocytin, it is further degraded by biotinidase to release free biotin, which is recycled in holocarboxylase synthesis. Biotin is catabolized by beta-oxidation of the valeric acid side chain or oxidation of sulfur in the heterocyclic ring.\n",
      "CLASS       Metabolism; Metabolism of cofactors and vitamins\n",
      "PATHWAY_MAP pae00780  Biotin metabolism\n",
      "MODULE      pae_M00123  Biotin biosynthesis, pimeloyl-ACP/CoA => biotin [PATH:pae00780]\n",
      "            pae_M00572  Pimeloyl-ACP biosynthesis, BioC-BioH pathway, malonyl-ACP => pimeloyl-ACP [PATH:pae00780]\n",
      "DBLINKS     GO: 0006768\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0503  biotin synthesis protein BioC [KO:K02169] [EC:2.1.1.197]\n",
      "line in gene PA0503  biotin synthesis protein BioC [KO:K02169] [EC:2.1.1.197]\n",
      "            PA1609  fabB; 3-oxoacyl-ACP synthase [KO:K00647] [EC:2.3.1.41]\n",
      "line in gene PA1609  fabB; 3-oxoacyl-ACP synthase [KO:K00647] [EC:2.3.1.41]\n",
      "            PA0098  3-oxoacyl-ACP synthase [KO:K00647] [EC:2.3.1.41]\n",
      "line in gene PA0098  3-oxoacyl-ACP synthase [KO:K00647] [EC:2.3.1.41]\n",
      "            PA2965  fabF1; 3-oxoacyl-ACP synthase [KO:K09458] [EC:2.3.1.179]\n",
      "line in gene PA2965  fabF1; 3-oxoacyl-ACP synthase [KO:K09458] [EC:2.3.1.179]\n",
      "            PA1373  fabF2; 3-oxoacyl-ACP synthase [KO:K09458] [EC:2.3.1.179]\n",
      "line in gene PA1373  fabF2; 3-oxoacyl-ACP synthase [KO:K09458] [EC:2.3.1.179]\n",
      "            PA5524  short-chain dehydrogenase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA5524  short-chain dehydrogenase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA1470  short-chain dehydrogenase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA1470  short-chain dehydrogenase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA4389  speA; 3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA4389  speA; 3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA2967  fabG; 3-oxoacyl-[acyl-carrier-protein] reductase FabG [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA2967  fabG; 3-oxoacyl-[acyl-carrier-protein] reductase FabG [KO:K00059] [EC:1.1.1.100]\n",
      "            PA4786  3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA4786  3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA4089  3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA4089  3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA0182  3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA0182  3-ketoacyl-ACP reductase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA1827  short-chain dehydrogenase [KO:K00059] [EC:1.1.1.100]\n",
      "line in gene PA1827  short-chain dehydrogenase [KO:K00059] [EC:1.1.1.100]\n",
      "            PA3645  fabZ; 3-hydroxyacyl-[acyl-carrier-protein] dehydratase FabZ [KO:K02372] [EC:4.2.1.59]\n",
      "line in gene PA3645  fabZ; 3-hydroxyacyl-[acyl-carrier-protein] dehydratase FabZ [KO:K02372] [EC:4.2.1.59]\n",
      "            PA1806  fabI; NADH-dependent enoyl-ACP reductase [KO:K00208] [EC:1.3.1.9 1.3.1.10]\n",
      "line in gene PA1806  fabI; NADH-dependent enoyl-ACP reductase [KO:K00208] [EC:1.3.1.9 1.3.1.10]\n",
      "            PA0502  biotin biosynthesis protein BioH [KO:K02170] [EC:3.1.1.85]\n",
      "line in gene PA0502  biotin biosynthesis protein BioH [KO:K02170] [EC:3.1.1.85]\n",
      "            PA0501  bioF; 8-amino-7-oxononanoate synthase [KO:K00652] [EC:2.3.1.47]\n",
      "line in gene PA0501  bioF; 8-amino-7-oxononanoate synthase [KO:K00652] [EC:2.3.1.47]\n",
      "            PA0420  bioA; adenosylmethionine--8-amino-7-oxononanoate aminotransferase BioA [KO:K00833] [EC:2.6.1.62]\n",
      "line in gene PA0420  bioA; adenosylmethionine--8-amino-7-oxononanoate aminotransferase BioA [KO:K00833] [EC:2.6.1.62]\n",
      "            PA0504  bioD; ATP-dependent dethiobiotin synthetase BioD [KO:K01935] [EC:6.3.3.3]\n",
      "line in gene PA0504  bioD; ATP-dependent dethiobiotin synthetase BioD [KO:K01935] [EC:6.3.3.3]\n",
      "            PA0500  bioB; biotin synthase [KO:K01012] [EC:2.8.1.6]\n",
      "line in gene PA0500  bioB; biotin synthase [KO:K01012] [EC:2.8.1.6]\n",
      "            PA4280  birA; biotin--protein ligase [KO:K03524] [EC:6.3.4.15]\n",
      "line in gene PA4280  birA; biotin--protein ligase [KO:K03524] [EC:6.3.4.15]\n",
      "COMPOUND    C00047  L-Lysine\n",
      "            C00086  Urea\n",
      "            C00120  Biotin\n",
      "            C01037  7,8-Diaminononanoate\n",
      "            C01063  6-Carboxyhexanoyl-CoA\n",
      "            C01092  8-Amino-7-oxononanoate\n",
      "            C01209  Malonyl-[acyl-carrier protein]\n",
      "            C01894  Biotinyl-CoA\n",
      "            C01909  Dethiobiotin\n",
      "            C02656  Pimelate\n",
      "            C05552  Biocytin\n",
      "            C05921  Biotinyl-5'-AMP\n",
      "            C06250  Holo-[carboxylase]\n",
      "            C19673  Malonyl-[acp] methyl ester\n",
      "            C19845  Pimeloyl-[acyl-carrier protein]\n",
      "            C19846  Pimeloyl-[acyl-carrier protein] methyl ester\n",
      "            C20372  3-Ketoglutaryl-[acp] methyl ester\n",
      "            C20373  3-Hydroxyglutaryl-[acp] methyl ester\n",
      "            C20374  Enoylglutaryl-[acp] methyl ester\n",
      "            C20375  Glutaryl-[acp] methyl ester\n",
      "            C20376  3-Ketopimeloyl-[acp] methyl ester\n",
      "            C20377  3-Hydroxypimeloyl-[acp] methyl ester\n",
      "            C20378  Enoylpimeloyl-[acp] methyl ester\n",
      "            C20384  Bisnorbiotin\n",
      "            C20385  Tetranorbiotin\n",
      "            C20386  Biotin sulfoxide\n",
      "            C20387  Biotin sulfone\n",
      "            C20683  Long-chain acyl-[acyl-carrier protein]\n",
      "            C22458  8-Amino-7-(carboxyamino)nonanoate\n",
      "REFERENCE   PMID:21437340\n",
      "  AUTHORS   Lin S, Cronan JE\n",
      "  TITLE     Closing in on complete pathways of biotin biosynthesis.\n",
      "  JOURNAL   Mol Biosyst 7:1811-21 (2011)\n",
      "            DOI:10.1039/c1mb05022b\n",
      "REFERENCE   PMID:22965231\n",
      "  AUTHORS   Lin S, Cronan JE\n",
      "  TITLE     The BioC O-methyltransferase catalyzes methyl esterification of malonyl-acyl carrier protein, an essential step in biotin synthesis.\n",
      "  JOURNAL   J Biol Chem 287:37010-20 (2012)\n",
      "            DOI:10.1074/jbc.M112.410290\n",
      "REFERENCE   PMID:20693992\n",
      "  AUTHORS   Lin S, Hanson RE, Cronan JE\n",
      "  TITLE     Biotin synthesis begins by hijacking the fatty acid synthetic pathway.\n",
      "  JOURNAL   Nat Chem Biol 6:682-8 (2010)\n",
      "            DOI:10.1038/nchembio.420\n",
      "REFERENCE   PMID:21435937\n",
      "  AUTHORS   Cronan JE, Lin S\n",
      "  TITLE     Synthesis of the alpha,omega-dicarboxylic acid precursor of biotin by the canonical fatty acid biosynthetic pathway.\n",
      "  JOURNAL   Curr Opin Chem Biol 15:407-13 (2011)\n",
      "            DOI:10.1016/j.cbpa.2011.03.001\n",
      "REFERENCE   PMID:11368323\n",
      "  AUTHORS   Stok JE, De Voss J\n",
      "  TITLE     Expression, purification, and characterization of BioI: a carbon-carbon bond cleaving cytochrome P450 involved in biotin biosynthesis in Bacillus subtilis.\n",
      "  JOURNAL   Arch Biochem Biophys 384:351-60 (2000)\n",
      "            DOI:10.1006/abbi.2000.2067\n",
      "REFERENCE   PMID:18838690\n",
      "  AUTHORS   Cryle MJ, Schlichting I\n",
      "  TITLE     Structural insights from a P450 Carrier Protein complex reveal how specificity is achieved in the P450(BioI) ACP complex.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 105:15696-701 (2008)\n",
      "            DOI:10.1073/pnas.0805983105\n",
      "REFERENCE   PMID:8763940\n",
      "  AUTHORS   Bower S, Perkins JB, Yocum RR, Howitt CL, Rahaim P, Pero J\n",
      "  TITLE     Cloning, sequencing, and characterization of the Bacillus subtilis biotin biosynthetic operon.\n",
      "  JOURNAL   J Bacteriol 178:4122-30 (1996)\n",
      "            DOI:10.1128/JB.178.14.4122-4130.1996\n",
      "REFERENCE   PMID:1445232\n",
      "  AUTHORS   Ploux O, Soularue P, Marquet A, Gloeckler R, Lemoine Y\n",
      "  TITLE     Investigation of the first step of biotin biosynthesis in Bacillus sphaericus. Purification and characterization of the pimeloyl-CoA synthase, and uptake of pimelate.\n",
      "  JOURNAL   Biochem J 287 ( Pt 3):685-90 (1992)\n",
      "            DOI:10.1042/bj2870685\n",
      "REFERENCE   PMID:12368242\n",
      "  AUTHORS   Rodionov DA, Mironov AA, Gelfand MS\n",
      "  TITLE     Conservation of the biotin regulon and the BirA regulatory signal in Eubacteria and Archaea.\n",
      "  JOURNAL   Genome Res 12:1507-16 (2002)\n",
      "            DOI:10.1101/gr.314502\n",
      "REFERENCE   PMID:11320134\n",
      "  AUTHORS   Sullivan JT, Brown SD, Yocum RR, Ronson CW\n",
      "  TITLE     The bio operon on the acquired symbiosis island of Mesorhizobium sp. strain R7A includes a novel gene involved in pimeloyl-CoA synthesis.\n",
      "  JOURNAL   Microbiology 147:1315-22 (2001)\n",
      "            DOI:10.1099/00221287-147-5-1315\n",
      "REFERENCE   PMID:12039741\n",
      "  AUTHORS   Entcheva P, Phillips DA, Streit WR\n",
      "  TITLE     Functional analysis of Sinorhizobium meliloti genes involved in biotin synthesis and transport.\n",
      "  JOURNAL   Appl Environ Microbiol 68:2843-8 (2002)\n",
      "            DOI:10.1128/AEM.68.6.2843-2848.2002\n",
      "REFERENCE   PMID:19319844\n",
      "  AUTHORS   Zempleni J, Wijeratne SS, Hassan YI\n",
      "  TITLE     Biotin.\n",
      "  JOURNAL   Biofactors 35:36-46 (2009)\n",
      "            DOI:10.1002/biof.8\n",
      "REFERENCE   PMID:9022537\n",
      "  AUTHORS   Zempleni J, McCormick DB, Mock DM\n",
      "  TITLE     Identification of biotin sulfone, bisnorbiotin methyl ketone, and tetranorbiotin-l-sulfoxide in human urine.\n",
      "  JOURNAL   Am J Clin Nutr 65:508-11 (1997)\n",
      "            DOI:10.1093/ajcn/65.2.508\n",
      "REFERENCE   PMID:5775781\n",
      "  AUTHORS   Iwahara S, McCormick DB, Wright LD, Li HC\n",
      "  TITLE     Bacterial degradation of biotin. 3. Metabolism of 14C-carbonyl-labeled biotin.\n",
      "  JOURNAL   J Biol Chem 244:1393-8 (1969)\n",
      "REFERENCE   PMID:2180922\n",
      "  AUTHORS   Pierson DE, Campbell A\n",
      "  TITLE     Cloning and nucleotide sequence of bisC, the structural gene for biotin sulfoxide reductase in Escherichia coli.\n",
      "  JOURNAL   J Bacteriol 172:2194-8 (1990)\n",
      "            DOI:10.1128/JB.172.4.2194-2198.1990\n",
      "REFERENCE   PMID:15601707\n",
      "  AUTHORS   Ezraty B, Bos J, Barras F, Aussel L\n",
      "  TITLE     Methionine sulfoxide reduction and assimilation in Escherichia coli: new role for the biotin sulfoxide reductase BisC.\n",
      "  JOURNAL   J Bacteriol 187:231-7 (2005)\n",
      "            DOI:10.1128/JB.187.1.231-237.2005\n",
      "REFERENCE   PMID:374979\n",
      "  AUTHORS   del Campillo-Campbell A, Dykhuizen D, Cleary PP\n",
      "  TITLE     Enzymic reduction of d-biotin d-sulfoxide to d-biotin.\n",
      "  JOURNAL   Methods Enzymol 62:379-85 (1979)\n",
      "            DOI:10.1016/0076-6879(79)62244-9\n",
      "REFERENCE   PMID:9013576\n",
      "  AUTHORS   Pollock VV, Barber MJ\n",
      "  TITLE     Biotin sulfoxide reductase. Heterologous expression and characterization of a functional molybdopterin guanine dinucleotide-containing enzyme.\n",
      "  JOURNAL   J Biol Chem 272:3355-62 (1997)\n",
      "            DOI:10.1074/jbc.272.6.3355\n",
      "REFERENCE   PMID:15366932\n",
      "  AUTHORS   Nelson KJ, Rajagopalan KV\n",
      "  TITLE     Studies on the interaction of NADPH with Rhodobacter sphaeroides biotin sulfoxide reductase.\n",
      "  JOURNAL   Biochemistry 43:11226-37 (2004)\n",
      "            DOI:10.1021/bi0490845\n",
      "REFERENCE   PMID:16299174\n",
      "  AUTHORS   Pinon V, Ravanel S, Douce R, Alban C\n",
      "  TITLE     Biotin synthesis in plants. The first committed step of the pathway is catalyzed by a cytosolic 7-keto-8-aminopelargonic acid synthase.\n",
      "  JOURNAL   Plant Physiol 139:1666-76 (2005)\n",
      "            DOI:10.1104/pp.105.070144\n",
      "REL_PATHWAY pae00061  Fatty acid biosynthesis\n",
      "            pae00310  Lysine degradation\n",
      "KO_PATHWAY  ko00780\n",
      "///\n",
      "path:pae00785\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00785                    Pathway\n",
      "NAME        Lipoic acid metabolism - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of cofactors and vitamins\n",
      "PATHWAY_MAP pae00785  Lipoic acid metabolism\n",
      "MODULE      pae_M00881  Lipoic acid biosynthesis, plants and bacteria, octanoyl-ACP => dihydrolipoyl-E2/H [PATH:pae00785]\n",
      "DBLINKS     GO: 0009106\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3997  lipB; lipoate-protein ligase B [KO:K03801] [EC:2.3.1.181]\n",
      "line in gene PA3997  lipB; lipoate-protein ligase B [KO:K03801] [EC:2.3.1.181]\n",
      "            PA3996  lis; lipoyl synthase [KO:K03644] [EC:2.8.1.8]\n",
      "line in gene PA3996  lis; lipoyl synthase [KO:K03644] [EC:2.8.1.8]\n",
      "COMPOUND    C01944  Octanoyl-CoA\n",
      "            C02051  Lipoylprotein\n",
      "            C02147  Dihydrolipoate\n",
      "            C02972  Dihydrolipoylprotein\n",
      "            C05752  Octanoyl-[acp]\n",
      "            C06423  Octanoic acid\n",
      "            C15972  Enzyme N6-(lipoyl)lysine\n",
      "            C15973  Enzyme N6-(dihydrolipoyl)lysine\n",
      "            C16238  Lipoyl-AMP\n",
      "            C16241  (R)-Lipoate\n",
      "            C22159  [Glycine cleavage system H]-N6-octanoyl-L-lysine\n",
      "            C22160  [Lipoyl-carrier protein E2]-N6-octanoyl-L-lysine\n",
      "            C22161  Dihydrolipoyl-AMP\n",
      "REFERENCE   PMID:16246025\n",
      "  AUTHORS   Gunther S, McMillan PJ, Wallace LJ, Muller S.\n",
      "  TITLE     Plasmodium falciparum possesses organelle-specific alpha-keto acid dehydrogenase complexes and lipoylation pathways.\n",
      "  JOURNAL   Biochem Soc Trans 33:977-80 (2005)\n",
      "            DOI:10.1042/BST20050977\n",
      "REFERENCE   PMID:11382754\n",
      "  AUTHORS   Fujiwara K, Takeuchi S, Okamura-Ikeda K, Motokawa Y.\n",
      "  TITLE     Purification, characterization, and cDNA cloning of lipoate-activating enzyme from bovine liver.\n",
      "  JOURNAL   J Biol Chem 276:28819-23 (2001)\n",
      "            DOI:10.1074/jbc.M101748200\n",
      "REFERENCE   PMID:9808738\n",
      "  AUTHORS   Yasuno R, Wada H.\n",
      "  TITLE     Biosynthesis of lipoic acid in Arabidopsis: cloning and characterization of the cDNA for lipoic acid synthase.\n",
      "  JOURNAL   Plant Physiol 118:935-43 (1998)\n",
      "            DOI:10.1104/pp.118.3.935\n",
      "REFERENCE   PMID:10671542\n",
      "  AUTHORS   Gueguen V, Macherel D, Jaquinod M, Douce R, Bourguignon J.\n",
      "  TITLE     Fatty acid and lipoic acid biosynthesis in higher plant mitochondria.\n",
      "  JOURNAL   J Biol Chem 275:5016-25 (2000)\n",
      "            DOI:10.1074/jbc.275.7.5016\n",
      "REFERENCE   PMID:11427685\n",
      "  AUTHORS   Wada M, Yasuno R, Jordan SW, Cronan JE Jr, Wada H.\n",
      "  TITLE     Lipoic acid metabolism in Arabidopsis thaliana: cloning and characterization of a cDNA encoding lipoyltransferase.\n",
      "  JOURNAL   Plant Cell Physiol 42:650-6 (2001)\n",
      "            DOI:10.1093/pcp/pce081\n",
      "REFERENCE   PMID:8206909\n",
      "  AUTHORS   Morris TW, Reed KE, Cronan JE Jr.\n",
      "  TITLE     Identification of the gene encoding lipoate-protein ligase A of Escherichia coli. Molecular cloning and characterization of the lplA gene and gene product.\n",
      "  JOURNAL   J Biol Chem 269:16091-100 (1994)\n",
      "REFERENCE   PMID:14700636\n",
      "  AUTHORS   Zhao X, Miller JR, Jiang Y, Marletta MA, Cronan JE.\n",
      "  TITLE     Assembly of the covalent linkage between lipoic acid and its cognate enzymes.\n",
      "  JOURNAL   Chem Biol 10:1293-302 (2003)\n",
      "            DOI:10.1016/j.chembiol.2003.11.016\n",
      "REFERENCE   PMID:15642479\n",
      "  AUTHORS   Nesbitt NM, Baleanu-Gogonea C, Cicchillo RM, Goodson K, Iwig DF, Broadwater JA, Haas JA, Fox BG, Booker SJ.\n",
      "  TITLE     Expression, purification, and physical characterization of Escherichia coli lipoyl(octanoyl)transferase.\n",
      "  JOURNAL   Protein Expr Purif 39:269-82 (2005)\n",
      "            DOI:10.1016/j.pep.2004.10.021\n",
      "REFERENCE   PMID:29191830\n",
      "  AUTHORS   Solmonson A, DeBerardinis RJ\n",
      "  TITLE     Lipoic acid metabolism and mitochondrial redox regulation.\n",
      "  JOURNAL   J Biol Chem 293:7522-7530 (2018)\n",
      "            DOI:10.1074/jbc.TM117.000259\n",
      "REFERENCE   PMID:27074917\n",
      "  AUTHORS   Cronan JE\n",
      "  TITLE     Assembly of Lipoic Acid on Its Cognate Enzymes: an Extraordinary and Essential Biosynthetic Pathway.\n",
      "  JOURNAL   Microbiol Mol Biol Rev 80:429-50 (2016)\n",
      "            DOI:10.1128/MMBR.00073-15\n",
      "REFERENCE   PMID:24777537\n",
      "  AUTHORS   Mayr JA, Feichtinger RG, Tort F, Ribes A, Sperl W\n",
      "  TITLE     Lipoic acid biosynthesis defects.\n",
      "  JOURNAL   J Inherit Metab Dis 37:553-63 (2014)\n",
      "            DOI:10.1007/s10545-014-9705-8\n",
      "REFERENCE   PMID:29987032\n",
      "  AUTHORS   Cao X, Zhu L, Song X, Hu Z, Cronan JE\n",
      "  TITLE     Protein moonlighting elucidates the essential human pathway catalyzing lipoic acid assembly on its cognate enzymes.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 115:E7063-E7072 (2018)\n",
      "            DOI:10.1073/pnas.1805862115\n",
      "REFERENCE   PMID:24341803\n",
      "  AUTHORS   Soreze Y, Boutron A, Habarou F, Barnerias C, Nonnenmacher L, Delpech H, Mamoune A, Chretien D, Hubert L, Bole-Feysot C, Nitschke P, Correia I, Sardet C, Boddaert N, Hamel Y, Delahodde A, Ottolenghi C, de Lonlay P\n",
      "  TITLE     Mutations in human lipoyltransferase gene LIPT1 cause a Leigh disease with secondary deficiency for pyruvate and alpha-ketoglutarate dehydrogenase.\n",
      "  JOURNAL   Orphanet J Rare Dis 8:192 (2013)\n",
      "            DOI:10.1186/1750-1172-8-192\n",
      "REFERENCE   PMID:28757203\n",
      "  AUTHORS   Habarou F, Hamel Y, Haack TB, Feichtinger RG, Lebigot E, Marquardt I, Busiah K, Laroche C, Madrange M, Grisel C, Pontoizeau C, Eisermann M, Boutron A, Chretien D, Chadefaux-Vekemans B, Barouki R, Bole-Feysot C, Nitschke P, Goudin N, Boddaert N, Nemazanyy I, Delahodde A, Kolker S, Rodenburg RJ, Korenke GC, Meitinger T, Strom TM, Prokisch H, Rotig A, Ottolenghi C, Mayr JA, de Lonlay P\n",
      "  TITLE     Biallelic Mutations in LIPT2 Cause a Mitochondrial Lipoylation Defect Associated with Severe Neonatal Encephalopathy.\n",
      "  JOURNAL   Am J Hum Genet 101:283-290 (2017)\n",
      "            DOI:10.1016/j.ajhg.2017.07.001\n",
      "REFERENCE   PMID:20508247\n",
      "  AUTHORS   Spalding MD, Prigge ST\n",
      "  TITLE     Lipoic acid metabolism in microbial pathogens.\n",
      "  JOURNAL   Microbiol Mol Biol Rev 74:200-28 (2010)\n",
      "            DOI:10.1128/MMBR.00008-10\n",
      "REFERENCE   PMID:28543853\n",
      "  AUTHORS   Guerra AJ, Afanador GA, Prigge ST\n",
      "  TITLE     Crystal structure of lipoate-bound lipoate ligase 1, LipL1, from Plasmodium falciparum.\n",
      "  JOURNAL   Proteins 85:1777-1783 (2017)\n",
      "            DOI:10.1002/prot.25324\n",
      "REFERENCE   PMID:25116855\n",
      "  AUTHORS   Afanador GA, Matthews KA, Bartee D, Gisselberg JE, Walters MS, Freel Meyers CL, Prigge ST\n",
      "  TITLE     Redox-dependent lipoylation of mitochondrial proteins in Plasmodium falciparum.\n",
      "  JOURNAL   Mol Microbiol 94:156-71 (2014)\n",
      "            DOI:10.1111/mmi.12753\n",
      "REFERENCE   PMID:28836704\n",
      "  AUTHORS   Afanador GA, Guerra AJ, Swift RP, Rodriguez RE, Bartee D, Matthews KA, Schon A, Freire E, Freel Meyers CL, Prigge ST\n",
      "  TITLE     A novel lipoate attachment enzyme is shared by Plasmodium and Chlamydia species.\n",
      "  JOURNAL   Mol Microbiol 106:439-451 (2017)\n",
      "            DOI:10.1111/mmi.13776\n",
      "REFERENCE   PMID:22607141\n",
      "  AUTHORS   Storm J, Muller S\n",
      "  TITLE     Lipoic acid metabolism of Plasmodium--a suitable drug target.\n",
      "  JOURNAL   Curr Pharm Des 18:3480-9 (2012)\n",
      "            DOI:10.2174/138161212801327266\n",
      "REFERENCE   PMID:18069893\n",
      "  AUTHORS   Gunther S, Wallace L, Patzewitz EM, McMillan PJ, Storm J, Wrenger C, Bissett R, Smith TK, Muller S\n",
      "  TITLE     Apicoplast lipoic acid protein ligase B is not essential for Plasmodium falciparum.\n",
      "  JOURNAL   PLoS Pathog 3:e189 (2007)\n",
      "            DOI:10.1371/journal.ppat.0030189\n",
      "REFERENCE   PMID:15225307\n",
      "  AUTHORS   Wrenger C, Muller S\n",
      "  TITLE     The human malaria parasite Plasmodium falciparum has distinct organelle-specific lipoylation pathways.\n",
      "  JOURNAL   Mol Microbiol 53:103-13 (2004)\n",
      "            DOI:10.1111/j.1365-2958.2004.04112.x\n",
      "REFERENCE   PMID:21338420\n",
      "  AUTHORS   Martin N, Christensen QH, Mansilla MC, Cronan JE, de Mendoza D\n",
      "  TITLE     A novel two-gene requirement for the octanoyltransfer reaction of Bacillus subtilis lipoic acid biosynthesis.\n",
      "  JOURNAL   Mol Microbiol 80:335-49 (2011)\n",
      "            DOI:10.1111/j.1365-2958.2011.07597.x\n",
      "REFERENCE   PMID:21338421\n",
      "  AUTHORS   Christensen QH, Martin N, Mansilla MC, de Mendoza D, Cronan JE\n",
      "  TITLE     A novel amidotransferase required for lipoic acid cofactor assembly in Bacillus subtilis.\n",
      "  JOURNAL   Mol Microbiol 80:350-63 (2011)\n",
      "            DOI:10.1111/j.1365-2958.2011.07598.x\n",
      "REL_PATHWAY pae00061  Fatty acid biosynthesis\n",
      "KO_PATHWAY  ko00785\n",
      "///\n",
      "path:pae00790\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00790                    Pathway\n",
      "NAME        Folate biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of cofactors and vitamins\n",
      "PATHWAY_MAP pae00790  Folate biosynthesis\n",
      "MODULE      pae_M00126  Tetrahydrofolate biosynthesis, GTP => THF [PATH:pae00790]\n",
      "            pae_M00880  Molybdenum cofactor biosynthesis, GTP => molybdenum cofactor [PATH:pae00790]\n",
      "DBLINKS     GO: 0046656\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3438  folE1; GTP cyclohydrolase I [KO:K01495] [EC:3.5.4.16]\n",
      "line in gene PA3438  folE1; GTP cyclohydrolase I [KO:K01495] [EC:3.5.4.16]\n",
      "            PA1674  folE2; GTP cyclohydrolase I [KO:K01495] [EC:3.5.4.16]\n",
      "line in gene PA1674  folE2; GTP cyclohydrolase I [KO:K01495] [EC:3.5.4.16]\n",
      "            PA5539  GTP cyclohydrolase [KO:K09007] [EC:3.5.4.16]\n",
      "line in gene PA5539  GTP cyclohydrolase [KO:K09007] [EC:3.5.4.16]\n",
      "            PA3296  phoA; alkaline phosphatase [KO:K01077] [EC:3.1.3.1]\n",
      "line in gene PA3296  phoA; alkaline phosphatase [KO:K01077] [EC:3.1.3.1]\n",
      "            PA3910  eddA; extracelullar DNA degradation protein EddA [KO:K01113] [EC:3.1.3.1]\n",
      "line in gene PA3910  eddA; extracelullar DNA degradation protein EddA [KO:K01113] [EC:3.1.3.1]\n",
      "            PA0582  folB; dihydroneopterin aldolase [KO:K01633] [EC:4.1.2.25 5.1.99.8 1.13.11.81]\n",
      "line in gene PA0582  folB; dihydroneopterin aldolase [KO:K01633] [EC:4.1.2.25 5.1.99.8 1.13.11.81]\n",
      "            PA4728  folK; 2-amino-4-hydroxy-6-hydroxymethyldihydropteridine pyrophosphokinase [KO:K00950] [EC:2.7.6.3]\n",
      "line in gene PA4728  folK; 2-amino-4-hydroxy-6-hydroxymethyldihydropteridine pyrophosphokinase [KO:K00950] [EC:2.7.6.3]\n",
      "            PA0583  hypothetical protein [KO:K00950] [EC:2.7.6.3]\n",
      "line in gene PA0583  hypothetical protein [KO:K00950] [EC:2.7.6.3]\n",
      "            PA4750  folP; dihydropteroate synthase [KO:K00796] [EC:2.5.1.15]\n",
      "line in gene PA4750  folP; dihydropteroate synthase [KO:K00796] [EC:2.5.1.15]\n",
      "            PA0350  folA; dihydrofolate reductase [KO:K00287] [EC:1.5.1.3]\n",
      "line in gene PA0350  folA; dihydrofolate reductase [KO:K00287] [EC:1.5.1.3]\n",
      "            PA3111  folC; folylpolyglutamate synthetase [KO:K11754] [EC:6.3.2.12 6.3.2.17]\n",
      "line in gene PA3111  folC; folylpolyglutamate synthetase [KO:K11754] [EC:6.3.2.12 6.3.2.17]\n",
      "            PA4047  ribA; GTP cyclohydrolase II [KO:K01497] [EC:3.5.4.25]\n",
      "line in gene PA4047  ribA; GTP cyclohydrolase II [KO:K01497] [EC:3.5.4.25]\n",
      "            PA4054  ribB; 3,4-dihydroxy-2-butanone-4-phosphate synthase [KO:K14652] [EC:4.1.99.12 3.5.4.25]\n",
      "line in gene PA4054  ribB; 3,4-dihydroxy-2-butanone-4-phosphate synthase [KO:K14652] [EC:4.1.99.12 3.5.4.25]\n",
      "            PA1758  pabB; para-aminobenzoate synthase component I [KO:K01665] [EC:2.6.1.85]\n",
      "line in gene PA1758  pabB; para-aminobenzoate synthase component I [KO:K01665] [EC:2.6.1.85]\n",
      "            PA2964  pabC; 4-amino-4-deoxychorismate lyase [KO:K02619] [EC:4.1.3.38]\n",
      "line in gene PA2964  pabC; 4-amino-4-deoxychorismate lyase [KO:K02619] [EC:4.1.3.38]\n",
      "            PA3439  folX; D-erythro-7,8-dihydroneopterin triphosphate 2'-epimerase [KO:K07589] [EC:5.1.99.7]\n",
      "line in gene PA3439  folX; D-erythro-7,8-dihydroneopterin triphosphate 2'-epimerase [KO:K07589] [EC:5.1.99.7]\n",
      "            PA3437  folM; short-chain dehydrogenase [KO:K13938] [EC:1.5.1.50 1.5.1.3]\n",
      "line in gene PA3437  folM; short-chain dehydrogenase [KO:K13938] [EC:1.5.1.50 1.5.1.3]\n",
      "            PA2666  6-carboxytetrahydropterin synthase QueD [KO:K01737] [EC:4.2.3.12 4.1.2.50]\n",
      "line in gene PA2666  6-carboxytetrahydropterin synthase QueD [KO:K01737] [EC:4.2.3.12 4.1.2.50]\n",
      "            PA0871  phhB; pterin-4-alpha-carbinolamine dehydratase [KO:K01724] [EC:4.2.1.96]\n",
      "line in gene PA0871  phhB; pterin-4-alpha-carbinolamine dehydratase [KO:K01724] [EC:4.2.1.96]\n",
      "            PA0872  phhA; phenylalanine 4-monooxygenase [KO:K00500] [EC:1.14.16.1]\n",
      "line in gene PA0872  phhA; phenylalanine 4-monooxygenase [KO:K00500] [EC:1.14.16.1]\n",
      "            PA0975  radical activating enzyme [KO:K10026] [EC:4.3.99.3]\n",
      "line in gene PA0975  radical activating enzyme [KO:K10026] [EC:4.3.99.3]\n",
      "            PA0976  7-cyano-7-deazaguanine synthase [KO:K06920] [EC:6.3.4.20]\n",
      "line in gene PA0976  7-cyano-7-deazaguanine synthase [KO:K06920] [EC:6.3.4.20]\n",
      "            PA2806  7-cyano-7-deazaguanine reductase [KO:K06879] [EC:1.7.1.13]\n",
      "line in gene PA2806  7-cyano-7-deazaguanine reductase [KO:K06879] [EC:1.7.1.13]\n",
      "            PA1505  moaA2; cyclic pyranopterin monophosphate synthase [KO:K03639] [EC:4.1.99.22]\n",
      "line in gene PA1505  moaA2; cyclic pyranopterin monophosphate synthase [KO:K03639] [EC:4.1.99.22]\n",
      "            PA3870  moaA1; molybdenum cofactor biosynthesis protein A [KO:K03639] [EC:4.1.99.22]\n",
      "line in gene PA3870  moaA1; molybdenum cofactor biosynthesis protein A [KO:K03639] [EC:4.1.99.22]\n",
      "            PA3918  moaC; molybdenum cofactor biosynthesis protein MoaC [KO:K03637] [EC:4.6.1.17]\n",
      "line in gene PA3918  moaC; molybdenum cofactor biosynthesis protein MoaC [KO:K03637] [EC:4.6.1.17]\n",
      "            PA3916  moaE; molybdopterin converting factor large subunit [KO:K03635] [EC:2.8.1.12]\n",
      "line in gene PA3916  moaE; molybdopterin converting factor large subunit [KO:K03635] [EC:2.8.1.12]\n",
      "            PA3029  moaB2; molybdopterin biosynthesis protein B [KO:K03638] [EC:2.7.7.75]\n",
      "line in gene PA3029  moaB2; molybdopterin biosynthesis protein B [KO:K03638] [EC:2.7.7.75]\n",
      "            PA3915  moaB1; molybdopterin biosynthesis protein B [KO:K03638] [EC:2.7.7.75]\n",
      "line in gene PA3915  moaB1; molybdopterin biosynthesis protein B [KO:K03638] [EC:2.7.7.75]\n",
      "            PA3914  moeA1; molybdenum cofactor biosynthesis protein A [KO:K03750] [EC:2.10.1.1]\n",
      "line in gene PA3914  moeA1; molybdenum cofactor biosynthesis protein A [KO:K03750] [EC:2.10.1.1]\n",
      "            PA3028  moeA2; molybdenum cofactor biosynthesis protein A [KO:K03750] [EC:2.10.1.1]\n",
      "line in gene PA3028  moeA2; molybdenum cofactor biosynthesis protein A [KO:K03750] [EC:2.10.1.1]\n",
      "            PA3030  mobA; molybdenum cofactor guanylyltransferase [KO:K03752] [EC:2.7.7.77]\n",
      "line in gene PA3030  mobA; molybdenum cofactor guanylyltransferase [KO:K03752] [EC:2.7.7.77]\n",
      "COMPOUND    C00044  GTP\n",
      "            C00101  Tetrahydrofolate\n",
      "            C00119  5-Phospho-alpha-D-ribose 1-diphosphate\n",
      "            C00156  4-Hydroxybenzoate\n",
      "            C00251  Chorismate\n",
      "            C00266  Glycolaldehyde\n",
      "            C00268  Dihydrobiopterin\n",
      "            C00272  Tetrahydrobiopterin\n",
      "            C00415  Dihydrofolate\n",
      "            C00504  Folate\n",
      "            C00568  4-Aminobenzoate\n",
      "            C00835  Sepiapterin\n",
      "            C00921  Dihydropteroate\n",
      "            C01217  5,6,7,8-Tetrahydromethanopterin\n",
      "            C01300  6-(Hydroxymethyl)-7,8-dihydropterin\n",
      "            C01304  2,5-Diamino-6-(5-phospho-D-ribosylamino)pyrimidin-4(3H)-one\n",
      "            C02587  Pyrimidodiazepine\n",
      "            C02953  7,8-Dihydrobiopterin\n",
      "            C03541  THF-polyglutamate\n",
      "            C03684  6-Pyruvoyltetrahydropterin\n",
      "            C04244  6-Lactoyl-5,6,7,8-tetrahydropterin\n",
      "            C04807  6-Hydroxymethyl-7,8-dihydropterin diphosphate\n",
      "            C04874  7,8-Dihydroneopterin\n",
      "            C04895  7,8-Dihydroneopterin 3'-triphosphate\n",
      "            C05922  Formamidopyrimidine nucleoside triphosphate\n",
      "            C05923  2,5-Diaminopyrimidine nucleoside triphosphate\n",
      "            C05924  Molybdopterin\n",
      "            C05925  Dihydroneopterin phosphate\n",
      "            C05926  Neopterin\n",
      "            C05927  7,8-Dihydromethanopterin\n",
      "            C06148  2,5-Diamino-6-(5'-triphosphoryl-3',4'-trihydroxy-2'-oxopentyl)-amino-4-oxopyrimidine\n",
      "            C06313  Biopterin\n",
      "            C09332  THF-L-glutamate\n",
      "            C11355  4-Amino-4-deoxychorismate\n",
      "            C15522  4a-Hydroxytetrahydrobiopterin\n",
      "            C15996  7-Cyano-7-carbaguanine\n",
      "            C16675  7-Aminomethyl-7-carbaguanine\n",
      "            C18237  Molybdoenzyme molybdenum cofactor\n",
      "            C18239  Precursor Z\n",
      "            C18802  Tetrahydrosarcinapterin\n",
      "            C19848  Adenylated molybdopterin\n",
      "            C19871  Guanylyl molybdenum cofactor\n",
      "            C20239  6-Carboxy-5,6,7,8-tetrahydropterin\n",
      "            C20248  7-Carboxy-7-carbaguanine\n",
      "            C20264  L-threo-Tetrahydrobiopterin\n",
      "            C20559  4-(beta-D-Ribofuranosyl)aniline 5'-phosphate\n",
      "            C20562  N-[(7,8-Dihydropterin-6-yl)methyl]-4-(beta-D-ribofuranosyl)aniline 5'-phosphate\n",
      "            C20566  7,8-Dihydroneopterin 2'-phosphate\n",
      "            C20567  7,8-Dihydroneopterin 2',3'-cyclic phosphate\n",
      "            C21007  5,6,7,8-Tetrahydromonapterin\n",
      "            C21008  7,8-Dihydromonapterin\n",
      "            C21065  7,8-Dihydroxanthopterin\n",
      "            C21094  7,8-Dihydromonapterin 3'-triphosphate\n",
      "            C21107  4-(beta-D-Ribofuranosyl)phenol 5'-phosphate\n",
      "            C21310  (8S)-3',8-Cyclo-7,8-dihydroguanosine 5'-triphosphate\n",
      "            C21485  Cytidylyl molybdenum cofactor\n",
      "            C21486  Thio-molybdenum cofactor\n",
      "            C21640  6-(1'-Hydroxy-2'-oxopropyl)-tetrahydropterin\n",
      "REFERENCE   \n",
      "  AUTHORS   Nishizuka Y, Seyama Y, Ikai A, Ishimura Y, Kawaguchi A (eds).\n",
      "  TITLE     [Cellular Functions and Metabolic Maps] (In Japanese)\n",
      "  JOURNAL   Tokyo Kagaku Dojin (1997)\n",
      "REFERENCE   PMID:19675644\n",
      "  AUTHORS   Schwarz G, Mendel RR, Ribbe MW\n",
      "  TITLE     Molybdenum cofactors, enzymes and pathways.\n",
      "  JOURNAL   Nature 460:839-47 (2009)\n",
      "            DOI:10.1038/nature08302\n",
      "REFERENCE   PMID:16632608\n",
      "  AUTHORS   Hanzelmann P, Schindelin H\n",
      "  TITLE     Binding of 5'-GTP to the C-terminal FeS cluster of the radical S-adenosylmethionine enzyme MoaA provides insights into its mechanism.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 103:6829-34 (2006)\n",
      "            DOI:10.1073/pnas.0510711103\n",
      "REFERENCE   PMID:18092812\n",
      "  AUTHORS   Daniels JN, Wuebbens MM, Rajagopalan KV, Schindelin H\n",
      "  TITLE     Crystal structure of a molybdopterin synthase-precursor Z complex: insight into its sulfur transfer mechanism and its role in molybdenum cofactor deficiency.\n",
      "  JOURNAL   Biochemistry 47:615-26 (2008)\n",
      "            DOI:10.1021/bi701734g\n",
      "REFERENCE   PMID:12571226\n",
      "  AUTHORS   Wuebbens MM, Rajagopalan KV\n",
      "  TITLE     Mechanistic and mutational studies of Escherichia coli molybdopterin synthase clarify the final step of molybdopterin biosynthesis.\n",
      "  JOURNAL   J Biol Chem 278:14523-32 (2003)\n",
      "            DOI:10.1074/jbc.M300453200\n",
      "REFERENCE   PMID:21867484\n",
      "  AUTHORS   Werner ER, Blau N, Thony B\n",
      "  TITLE     Tetrahydrobiopterin: biochemistry and pathophysiology.\n",
      "  JOURNAL   Biochem J 438:397-414 (2011)\n",
      "            DOI:10.1042/BJ20110293\n",
      "REFERENCE   PMID:12893295\n",
      "  AUTHORS   Iino T, Tabata M, Takikawa S, Sawada H, Shintaku H, Ishikura S, Hara A\n",
      "  TITLE     Tetrahydrobiopterin is synthesized from 6-pyruvoyl-tetrahydropterin by the human aldo-keto reductase AKR1 family members.\n",
      "  JOURNAL   Arch Biochem Biophys 416:180-7 (2003)\n",
      "            DOI:10.1016/S0003-9861(03)00295-9\n",
      "REFERENCE   PMID:1902669\n",
      "  AUTHORS   Park YS, Heizmann CW, Wermuth B, Levine RA, Steinerstauch P, Guzman J, Blau N\n",
      "  TITLE     Human carbonyl and aldose reductases: new catalytic functions in tetrahydrobiopterin biosynthesis.\n",
      "  JOURNAL   Biochem Biophys Res Commun 175:738-44 (1991)\n",
      "            DOI:10.1016/0006-291X(91)91628-P\n",
      "REL_PATHWAY pae00230  Purine metabolism\n",
      "            pae00362  Benzoate degradation\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "            pae00670  One carbon pool by folate\n",
      "            pae00680  Methane metabolism\n",
      "KO_PATHWAY  ko00790\n",
      "///\n",
      "path:pae00791\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00791                    Pathway\n",
      "NAME        Atrazine degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Xenobiotics biodegradation and metabolism\n",
      "PATHWAY_MAP pae00791  Atrazine degradation\n",
      "DBLINKS     UMBBD: atr cya cnm cpm\n",
      "            GO: 0019381\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0495  hypothetical protein [KO:K01941] [EC:6.3.4.6]\n",
      "line in gene PA0495  hypothetical protein [KO:K01941] [EC:6.3.4.6]\n",
      "            PA4868  ureC; urease subunit alpha [KO:K01428] [EC:3.5.1.5]\n",
      "line in gene PA4868  ureC; urease subunit alpha [KO:K01428] [EC:3.5.1.5]\n",
      "            PA4867  ureB; urease subunit beta [KO:K01429] [EC:3.5.1.5]\n",
      "line in gene PA4867  ureB; urease subunit beta [KO:K01429] [EC:3.5.1.5]\n",
      "            PA4865  ureA; urease subunit gamma [KO:K01430] [EC:3.5.1.5]\n",
      "line in gene PA4865  ureA; urease subunit gamma [KO:K01430] [EC:3.5.1.5]\n",
      "COMPOUND    C00011  CO2\n",
      "            C00086  Urea\n",
      "            C01010  Urea-1-carboxylate\n",
      "            C01566  Cyanamide\n",
      "            C06551  Atrazine\n",
      "            C06552  Hydroxyatrazine\n",
      "            C06553  N-Isopropylammelide\n",
      "            C06554  Cyanuric acid\n",
      "            C06555  Biuret\n",
      "            C06556  Deisopropylatrazine\n",
      "            C06557  Deisopropylhydroxyatrazine\n",
      "            C06558  N-Ethylammelide\n",
      "            C06559  Deethylatrazine\n",
      "            C06560  Deisopropyldeethylatrazine\n",
      "            C08733  Ammeline\n",
      "            C08734  Ammelide\n",
      "            C08735  2-Chloro-4-hydroxy-6-amino-1,3,5-triazine\n",
      "            C08737  Melamine\n",
      "            C14147  Cyromazine\n",
      "            C14148  N-Cyclopropylammeline\n",
      "            C14149  N-Cyclopropylammelide\n",
      "            C22084  1-Carboxybiuret\n",
      "            C22085  Urea-1,3-dicarboxylate\n",
      "REFERENCE   PMID:16597948\n",
      "  AUTHORS   Shapir N, Cheng G, Sadowsky MJ, Wackett LP\n",
      "  TITLE     Purification and characterization of TrzF: biuret hydrolysis by allophanate hydrolase supports growth.\n",
      "  JOURNAL   Appl Environ Microbiol 72:2491-5 (2006)\n",
      "            DOI:10.1128/AEM.72.4.2491-2495.2006\n",
      "REFERENCE   PMID:16085834\n",
      "  AUTHORS   Cheng G, Shapir N, Sadowsky MJ, Wackett LP.\n",
      "  TITLE     Allophanate hydrolase, not urease, functions in bacterial cyanuric acid metabolism.\n",
      "  JOURNAL   Appl Environ Microbiol 71:4437-45 (2005)\n",
      "            DOI:10.1128/AEM.71.8.4437-4445.2005\n",
      "REFERENCE   PMID:25847245\n",
      "  AUTHORS   Li J, Biss M, Fu Y, Xu X, Moore SA, Xiao W\n",
      "  TITLE     Two duplicated genes DDI2 and DDI3 in budding yeast encode a cyanamide hydratase and are induced by cyanamide.\n",
      "  JOURNAL   J Biol Chem 290:12664-75 (2015)\n",
      "            DOI:10.1074/jbc.M115.645408\n",
      "KO_PATHWAY  ko00791\n",
      "///\n",
      "path:pae00860\n",
      "ENTRY       pae00860                    Pathway\n",
      "NAME        Porphyrin metabolism - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Porphyrins are a group of chemical compounds with the backbone ring structure consisting of four linked pyrrole units. Metal complexes of porphyrins, as well as those of corrinoids with the one-carbon shorter ring structure, play important biological roles. They include iron-containing heme, magnesium-containing chlorophyll, nickel-containing coenzyme F430 and cobalt-containing cobamide (such as vitamin B12 coenzyme) as shown in this map.\n",
      "CLASS       Metabolism; Metabolism of cofactors and vitamins\n",
      "PATHWAY_MAP pae00860  Porphyrin metabolism\n",
      "MODULE      pae_M00121  Heme biosynthesis, plants and bacteria, glutamate => heme [PATH:pae00860]\n",
      "            pae_M00122  Cobalamin biosynthesis, cobyrinate a,c-diamide => cobalamin [PATH:pae00860]\n",
      "            pae_M00846  Siroheme biosynthesis, glutamyl-tRNA => siroheme [PATH:pae00860]\n",
      "DBLINKS     GO: 0006778\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3134  gltX; glutamate--tRNA ligase [KO:K01885] [EC:6.1.1.17]\n",
      "line in gene PA3134  gltX; glutamate--tRNA ligase [KO:K01885] [EC:6.1.1.17]\n",
      "            PA4666  hemA; glutamyl-tRNA reductase [KO:K02492] [EC:1.2.1.70]\n",
      "line in gene PA4666  hemA; glutamyl-tRNA reductase [KO:K02492] [EC:1.2.1.70]\n",
      "            PA3977  hemL; glutamate-1-semialdehyde aminotransferase [KO:K01845] [EC:5.4.3.8]\n",
      "line in gene PA3977  hemL; glutamate-1-semialdehyde aminotransferase [KO:K01845] [EC:5.4.3.8]\n",
      "            PA4088  aminotransferase [KO:K01845] [EC:5.4.3.8]\n",
      "line in gene PA4088  aminotransferase [KO:K01845] [EC:5.4.3.8]\n",
      "            PA5523  aminotransferase [KO:K01845] [EC:5.4.3.8]\n",
      "line in gene PA5523  aminotransferase [KO:K01845] [EC:5.4.3.8]\n",
      "            PA5243  hemB; delta-aminolevulinic acid dehydratase [KO:K01698] [EC:4.2.1.24]\n",
      "line in gene PA5243  hemB; delta-aminolevulinic acid dehydratase [KO:K01698] [EC:4.2.1.24]\n",
      "            PA5260  hemC; porphobilinogen deaminase [KO:K01749] [EC:2.5.1.61]\n",
      "line in gene PA5260  hemC; porphobilinogen deaminase [KO:K01749] [EC:2.5.1.61]\n",
      "            PA5259  hemD; uroporphyrinogen-III synthase [KO:K01719] [EC:4.2.1.75]\n",
      "line in gene PA5259  hemD; uroporphyrinogen-III synthase [KO:K01719] [EC:4.2.1.75]\n",
      "            PA5034  hemE; uroporphyrinogen decarboxylase [KO:K01599] [EC:4.1.1.37]\n",
      "line in gene PA5034  hemE; uroporphyrinogen decarboxylase [KO:K01599] [EC:4.1.1.37]\n",
      "            PA0024  hemF; coproporphyrinogen III oxidase [KO:K00228] [EC:1.3.3.3]\n",
      "line in gene PA0024  hemF; coproporphyrinogen III oxidase [KO:K00228] [EC:1.3.3.3]\n",
      "            PA1546  hemN; oxygen-independent coproporphyrinogen-III oxidase [KO:K02495] [EC:1.3.98.3]\n",
      "line in gene PA1546  hemN; oxygen-independent coproporphyrinogen-III oxidase [KO:K02495] [EC:1.3.98.3]\n",
      "            PA0661  hypothetical protein [KO:K08973] [EC:1.3.99.-]\n",
      "line in gene PA0661  hypothetical protein [KO:K08973] [EC:1.3.99.-]\n",
      "            PA4655  hemH; ferrochelatase [KO:K01772] [EC:4.98.1.1 4.99.1.9]\n",
      "line in gene PA4655  hemH; ferrochelatase [KO:K01772] [EC:4.98.1.1 4.99.1.9]\n",
      "            PA5258  hypothetical protein [KO:K02496] [EC:2.1.1.107]\n",
      "line in gene PA5258  hypothetical protein [KO:K02496] [EC:2.1.1.107]\n",
      "            PA2611  cysG; siroheme synthase [KO:K02302] [EC:2.1.1.107 1.3.1.76 4.99.1.4]\n",
      "line in gene PA2611  cysG; siroheme synthase [KO:K02302] [EC:2.1.1.107 1.3.1.76 4.99.1.4]\n",
      "            PA1778  cobA; uroporphyrin-III C-methyltransferase [KO:K02303] [EC:2.1.1.107]\n",
      "line in gene PA1778  cobA; uroporphyrin-III C-methyltransferase [KO:K02303] [EC:2.1.1.107]\n",
      "            PA0510  uroporphyrin-III C-methyltransferase [KO:K02303] [EC:2.1.1.107]\n",
      "line in gene PA0510  uroporphyrin-III C-methyltransferase [KO:K02303] [EC:2.1.1.107]\n",
      "            PA0515  heme d1 biosynthesis protein NirD [KO:K22225] [EC:4.1.1.111]\n",
      "line in gene PA0515  heme d1 biosynthesis protein NirD [KO:K22225] [EC:4.1.1.111]\n",
      "            PA0513  heme d1 biosynthesis protein NirG [KO:K22225] [EC:4.1.1.111]\n",
      "line in gene PA0513  heme d1 biosynthesis protein NirG [KO:K22225] [EC:4.1.1.111]\n",
      "            PA0514  nirL; heme d1 biosynthesis protein NirL [KO:K22225] [EC:4.1.1.111]\n",
      "line in gene PA0514  nirL; heme d1 biosynthesis protein NirL [KO:K22225] [EC:4.1.1.111]\n",
      "            PA0512  heme d1 biosynthesis protein NirH [KO:K22225] [EC:4.1.1.111]\n",
      "line in gene PA0512  heme d1 biosynthesis protein NirH [KO:K22225] [EC:4.1.1.111]\n",
      "            PA0509  nirN; cytochrome C [KO:K24867]\n",
      "line in gene PA0509  nirN; cytochrome C [KO:K24867]\n",
      "            PA0113  protoheme IX farnesyltransferase [KO:K02257] [EC:2.5.1.141]\n",
      "line in gene PA0113  protoheme IX farnesyltransferase [KO:K02257] [EC:2.5.1.141]\n",
      "            PA1321  cyoE; protoheme IX farnesyltransferase [KO:K02257] [EC:2.5.1.141]\n",
      "line in gene PA1321  cyoE; protoheme IX farnesyltransferase [KO:K02257] [EC:2.5.1.141]\n",
      "            PA0112  hypothetical protein [KO:K02259] [EC:1.17.99.9]\n",
      "line in gene PA0112  hypothetical protein [KO:K02259] [EC:1.17.99.9]\n",
      "            PA2904  cobI; precorrin-2 C(20)-methyltransferase [KO:K03394] [EC:2.1.1.130 2.1.1.151]\n",
      "line in gene PA2904  cobI; precorrin-2 C(20)-methyltransferase [KO:K03394] [EC:2.1.1.130 2.1.1.151]\n",
      "            PA2906  oxidoreductase [KO:K02229] [EC:1.14.13.83]\n",
      "line in gene PA2906  oxidoreductase [KO:K02229] [EC:1.14.13.83]\n",
      "            PA2903  cobJ; precorrin-3 methylase CobJ [KO:K13541] [EC:3.7.1.12 2.1.1.272 2.1.1.131]\n",
      "line in gene PA2903  cobJ; precorrin-3 methylase CobJ [KO:K13541] [EC:3.7.1.12 2.1.1.272 2.1.1.131]\n",
      "            PA2948  cobM; precorrin-4 C(11)-methyltransferase [KO:K05936] [EC:2.1.1.133 2.1.1.271]\n",
      "line in gene PA2948  cobM; precorrin-4 C(11)-methyltransferase [KO:K05936] [EC:2.1.1.133 2.1.1.271]\n",
      "            PA2947  hypothetical protein [KO:K02189] [EC:3.7.1.12]\n",
      "line in gene PA2947  hypothetical protein [KO:K02189] [EC:3.7.1.12]\n",
      "            PA2909  cobalt-precorrin-6x reductase [KO:K05895] [EC:1.3.1.54 1.3.1.106]\n",
      "line in gene PA2909  cobalt-precorrin-6x reductase [KO:K05895] [EC:1.3.1.54 1.3.1.106]\n",
      "            PA2908  cbiD; cobalt-precorrin-5B C(1)-methyltransferase [KO:K02188] [EC:2.1.1.195]\n",
      "line in gene PA2908  cbiD; cobalt-precorrin-5B C(1)-methyltransferase [KO:K02188] [EC:2.1.1.195]\n",
      "            PA2907  cobL; precorrin-6y-dependent methyltransferase CobL [KO:K00595] [EC:2.1.1.132 2.1.1.289 2.1.1.196]\n",
      "line in gene PA2907  cobL; precorrin-6y-dependent methyltransferase CobL [KO:K00595] [EC:2.1.1.132 2.1.1.289 2.1.1.196]\n",
      "            PA2905  cobH; precorrin-8X methylmutase [KO:K06042] [EC:5.4.99.61 5.4.99.60]\n",
      "line in gene PA2905  cobH; precorrin-8X methylmutase [KO:K06042] [EC:5.4.99.61 5.4.99.60]\n",
      "            PA1273  cobB; hydrogenobyrinate a,c-diamide synthase [KO:K02224] [EC:6.3.5.9 6.3.5.11]\n",
      "line in gene PA1273  cobB; hydrogenobyrinate a,c-diamide synthase [KO:K02224] [EC:6.3.5.9 6.3.5.11]\n",
      "            PA1923  cobaltochelatase subunit CobN [KO:K02230] [EC:6.6.1.2]\n",
      "line in gene PA1923  cobaltochelatase subunit CobN [KO:K02230] [EC:6.6.1.2]\n",
      "            PA2944  cobN; cobaltochelatase subunit CobN [KO:K02230] [EC:6.6.1.2]\n",
      "line in gene PA2944  cobN; cobaltochelatase subunit CobN [KO:K02230] [EC:6.6.1.2]\n",
      "            PA4399  hypothetical protein [KO:K00798] [EC:2.5.1.17]\n",
      "line in gene PA4399  hypothetical protein [KO:K00798] [EC:2.5.1.17]\n",
      "            PA1272  cobO; cob(I)yrinic acid a,c-diamide adenosyltransferase [KO:K19221] [EC:2.5.1.17]\n",
      "line in gene PA1272  cobO; cob(I)yrinic acid a,c-diamide adenosyltransferase [KO:K19221] [EC:2.5.1.17]\n",
      "            PA1277  cobQ; cobyric acid synthase [KO:K02232] [EC:6.3.5.10]\n",
      "line in gene PA1277  cobQ; cobyric acid synthase [KO:K02232] [EC:6.3.5.10]\n",
      "            PA1275  cobD; cobalamin biosynthesis protein CobD [KO:K02227] [EC:6.3.1.10]\n",
      "line in gene PA1275  cobD; cobalamin biosynthesis protein CobD [KO:K02227] [EC:6.3.1.10]\n",
      "            PA1276  cobC; threonine-phosphate decarboxylase [KO:K02225]\n",
      "line in gene PA1276  cobC; threonine-phosphate decarboxylase [KO:K02225]\n",
      "            PA1278  cobP; bifunctional adenosylcobinamide kinase/adenosylcobinamide-phosphate guanylyltransferase [KO:K02231] [EC:2.7.1.156 2.7.7.62]\n",
      "line in gene PA1278  cobP; bifunctional adenosylcobinamide kinase/adenosylcobinamide-phosphate guanylyltransferase [KO:K02231] [EC:2.7.1.156 2.7.7.62]\n",
      "            PA1281  cobV; adenosylcobinamide-GDP ribazoletransferase [KO:K02233] [EC:2.7.8.26]\n",
      "line in gene PA1281  cobV; adenosylcobinamide-GDP ribazoletransferase [KO:K02233] [EC:2.7.8.26]\n",
      "            PA1279  cobU; nicotinate-nucleotide--dimethylbenzimidazole phosphoribosyltransferase [KO:K00768] [EC:2.4.2.21]\n",
      "line in gene PA1279  cobU; nicotinate-nucleotide--dimethylbenzimidazole phosphoribosyltransferase [KO:K00768] [EC:2.4.2.21]\n",
      "            PA1280  hypothetical protein [KO:K02226] [EC:3.1.3.73]\n",
      "line in gene PA1280  hypothetical protein [KO:K02226] [EC:3.1.3.73]\n",
      "            PA2942  magnesium chelatase [KO:K03405] [EC:6.6.1.1]\n",
      "line in gene PA2942  magnesium chelatase [KO:K03405] [EC:6.6.1.1]\n",
      "            PA0672  hemO; heme oxygenase [KO:K07215] [EC:1.14.99.58]\n",
      "line in gene PA0672  hemO; heme oxygenase [KO:K07215] [EC:1.14.99.58]\n",
      "            PA4388  hypothetical protein [KO:K07226] [EC:1.14.99.58]\n",
      "line in gene PA4388  hypothetical protein [KO:K07226] [EC:1.14.99.58]\n",
      "            PA4880  bacterioferritin [KO:K03594] [EC:1.16.3.1]\n",
      "line in gene PA4880  bacterioferritin [KO:K03594] [EC:1.16.3.1]\n",
      "            PA3531  bfrB; bacterioferritin [KO:K03594] [EC:1.16.3.1]\n",
      "line in gene PA3531  bfrB; bacterioferritin [KO:K03594] [EC:1.16.3.1]\n",
      "            PA4235  bfrA; bacterioferritin [KO:K03594] [EC:1.16.3.1]\n",
      "line in gene PA4235  bfrA; bacterioferritin [KO:K03594] [EC:1.16.3.1]\n",
      "COMPOUND    C00025  L-Glutamate\n",
      "            C00032  Heme\n",
      "            C00037  Glycine\n",
      "            C00188  L-Threonine\n",
      "            C00194  Cobamide coenzyme\n",
      "            C00430  5-Aminolevulinate\n",
      "            C00486  Bilirubin\n",
      "            C00500  Biliverdin\n",
      "            C00524  Cytochrome c\n",
      "            C00541  Cob(II)alamin\n",
      "            C00748  Siroheme\n",
      "            C00853  Cob(I)alamin\n",
      "            C00931  Porphobilinogen\n",
      "            C00992  Aquacob(III)alamin\n",
      "            C01024  Hydroxymethylbilane\n",
      "            C01051  Uroporphyrinogen III\n",
      "            C01079  Protoporphyrinogen IX\n",
      "            C01708  Hemoglobin\n",
      "            C02139  Chlorophyllide\n",
      "            C02191  Protoporphyrin\n",
      "            C02463  Precorrin 2\n",
      "            C02469  Uroporphyrin III\n",
      "            C02800  4,5-Dioxopentanoate\n",
      "            C02823  Cyanocobalamin\n",
      "            C02880  Protochlorophyllide\n",
      "            C02987  L-Glutamyl-tRNA(Glu)\n",
      "            C03029  Transferrin[Fe(II)]2\n",
      "            C03114  Dimethylbenzimidazole\n",
      "            C03179  Transferrin[Fe(III)]2\n",
      "            C03194  (R)-1-Aminopropan-2-ol\n",
      "            C03263  Coproporphyrinogen III\n",
      "            C03373  Aminoimidazole ribotide\n",
      "            C03516  Magnesium protoporphyrin\n",
      "            C03741  (S)-4-Amino-5-oxopentanoate\n",
      "            C04122  D-1-Aminopropan-2-ol O-phosphate\n",
      "            C04536  Magnesium protoporphyrin monomethyl ester\n",
      "            C04778  N1-(5-Phospho-alpha-D-ribosyl)-5,6-dimethylbenzimidazole\n",
      "            C05306  Chlorophyll a\n",
      "            C05307  Chlorophyll b\n",
      "            C05766  Uroporphyrinogen I\n",
      "            C05767  Uroporphyrin I\n",
      "            C05768  Coproporphyrinogen I\n",
      "            C05769  Coproporphyrin I\n",
      "            C05770  Coproporphyrin III\n",
      "            C05772  Precorrin 3A\n",
      "            C05773  Cobyrinate\n",
      "            C05774  Cobinamide\n",
      "            C05775  alpha-Ribazole\n",
      "            C05777  Coenzyme F430\n",
      "            C05778  Sirohydrochlorin\n",
      "            C05779  Apoferritin\n",
      "            C05780  Apotransferrin\n",
      "            C05781  Oxyhemoglobin\n",
      "            C05782  Myoglobin\n",
      "            C05783  Cytochrome a\n",
      "            C05784  Catalase\n",
      "            C05785  Peroxidase\n",
      "            C05786  (3Z)-Phycocyanobilin\n",
      "            C05787  Bilirubin beta-diglucuronide\n",
      "            C05789  L-Urobilinogen\n",
      "            C05790  I-Urobilinogen\n",
      "            C05791  D-Urobilinogen\n",
      "            C05793  L-Urobilin\n",
      "            C05794  Urobilin\n",
      "            C05795  D-Urobilin\n",
      "            C05797  Pheophytin a\n",
      "            C05798  Bacteriopheophytin\n",
      "            C05912  (3Z)-Phycoerythrobilin\n",
      "            C05913  (3Z)-Phytochromobilin\n",
      "            C06319  Precorrin 6Y\n",
      "            C06320  Precorrin 6X\n",
      "            C06399  Hydrogenobyrinate\n",
      "            C06406  Precorrin 3B\n",
      "            C06407  Precorrin 4\n",
      "            C06408  Precorrin 8X\n",
      "            C06416  Precorrin 5\n",
      "            C06503  Hydrogenobyrinate a,c diamide\n",
      "            C06504  Cob(II)yrinate a,c diamide\n",
      "            C06505  Cob(I)yrinate a,c diamide\n",
      "            C06506  Adenosyl cobyrinate a,c diamide\n",
      "            C06507  Adenosyl cobyrinate hexaamide\n",
      "            C06508  Adenosyl cobinamide\n",
      "            C06509  Adenosyl cobinamide phosphate\n",
      "            C06510  Adenosine-GDP-cobinamide\n",
      "            C11242  Bacteriochlorophyll a\n",
      "            C11243  Bacteriochlorophyll b\n",
      "            C11538  Cobalt-sirohydrochlorin\n",
      "            C11540  Cobalt-precorrin 4\n",
      "            C11542  Cobalt-precorrin 6\n",
      "            C11543  Cobalt-dihydro-precorrin 6\n",
      "            C11545  Cobalt-precorrin 8\n",
      "            C11630  15,16-Dihydrobiliverdin\n",
      "            C11829  13(1)-Hydroxy-magnesium-protoporphyrin IX 13-monomethyl ester\n",
      "            C11830  13(1)-Oxo-magnesium-protoporphyrin IX 13-monomethyl ester\n",
      "            C11831  Divinylprotochlorophyllide\n",
      "            C11832  Divinyl chlorophyllide a\n",
      "            C11850  Divinylchlorophyll a\n",
      "            C11851  Zn-Bacteriochlorophyll a\n",
      "            C12147  L-Threonine O-3-phosphate\n",
      "            C14818  Fe2+\n",
      "            C14819  Fe3+\n",
      "            C15670  Heme A\n",
      "            C15672  Heme O\n",
      "            C16242  Cobalt-precorrin 5A\n",
      "            C16243  Cobalt-precorrin 5B\n",
      "            C16244  Cobalt-precorrin 7\n",
      "            C16540  7(1)-Hydroxychlorophyllide a\n",
      "            C16541  Chlorophyllide b\n",
      "            C17401  Cobalt-factor III\n",
      "            C18021  Pheophorbide a\n",
      "            C18022  Red chlorophyll catabolite\n",
      "            C18064  Pyropheophorbide a\n",
      "            C18098  Primary fluorescent chlorophyll catabolite\n",
      "            C18151  7(1)-Hydroxychlorophyll a\n",
      "            C18152  3-Vinylbacteriochlorophyllide a\n",
      "            C18153  3-Hydroxyethylbacteriochlorophyllide a\n",
      "            C18154  3-Hydroxyethylchlorophyllide a\n",
      "            C18155  Bacteriochlorophyllide a\n",
      "            C18156  Bacteriochlorophyllide b\n",
      "            C18157  8-Ethyl-12-methyl-3-vinylbacteriochlorophyllide d\n",
      "            C18160  Bacteriochlorophyllide d\n",
      "            C18161  Bacteriochlorophyllide c\n",
      "            C18162  Bacteriochlorophyll d\n",
      "            C18163  Bacteriochlorophyll c\n",
      "            C19608  Ferritin\n",
      "            C20666  5-Oxo-delta-bilirubin\n",
      "            C21217  Geranylgeranyl bacteriochlorophyllide a\n",
      "            C21284  Fe-coproporphyrin III\n",
      "            C21427  8,12-Diethyl-3-vinylbacteriochlorophyllide d\n",
      "            C21428  12-Ethyl-8-propyl-3-vinylbacteriochlorophyllide d\n",
      "            C21429  12-Ethyl-8-isobutyl-3-vinylbacteriochlorophyllide d\n",
      "            C21431  8-Ethyl-12-methylbacteriochlorophyllide d\n",
      "            C21432  8,12-Diethylbacteriochlorophyllide d\n",
      "            C21433  12-Ethyl-8-propylbacteriochlorophyllide d\n",
      "            C21434  12-Ethyl-8-isobutylbacteriochlorophyllide d\n",
      "            C21435  Geranylgeranyl bacteriochlorophyllide b\n",
      "            C21510  Nickel-sirohydrochlorin\n",
      "            C21511  Nickel-sirohydrochlorin a,c-diamide\n",
      "            C21512  15,17(3)-Seco-F430-17(3)-acid\n",
      "            C21582  Bacteriochlorophyllide e\n",
      "            C21723  Biliverdin-IX-delta\n",
      "            C21724  Biliverdin-IX-beta\n",
      "            C21764  5-Hydroxybenzimidazole\n",
      "            C21835  12,18-Didecarboxysiroheme\n",
      "            C22338  Dihydro-heme d1\n",
      "            C22339  Heme d1\n",
      "            C22450  5-Methoxybenzimidazole\n",
      "            C22451  5-Methoxy-6-methylbenzimidazole\n",
      "REFERENCE   PMID:9461500\n",
      "  AUTHORS   Woodcock SC, Raux E, Levillayer F, Thermes C, Rambach A, Warren MJ.\n",
      "  TITLE     Effect of mutations in the transmethylase and dehydrogenase/chelatase domains of sirohaem synthase (CysG) on sirohaem and cobalamin biosynthesis.\n",
      "  JOURNAL   Biochem J 330:121-9 (1998)\n",
      "            DOI:10.1042/bj3300121\n",
      "REFERENCE   PMID:10217486\n",
      "  AUTHORS   Johansson P, Hederstedt L\n",
      "  TITLE     Organization of genes for tetrapyrrole biosynthesis in gram--positive bacteria.\n",
      "  JOURNAL   Microbiology 145 ( Pt 3):529-38 (1999)\n",
      "            DOI:10.1099/13500872-145-3-529\n",
      "REFERENCE   PMID:10518530\n",
      "  AUTHORS   Maggio-Hall LA, Escalante-Semerena JC.\n",
      "  TITLE     In vitro synthesis of the nucleotide loop of cobalamin by Salmonella typhimurium enzymes.\n",
      "  JOURNAL   Proc Natl Acad Sci USA 96:11798-803 (1999)\n",
      "            DOI:10.1073/pnas.96.21.11798\n",
      "REFERENCE   PMID:11007789\n",
      "  AUTHORS   Roper JM, Raux E, Brindley AA, Schubert HL, Gharbia SE, Shah HN, Warren MJ.\n",
      "  TITLE     The enigma of cobalamin (Vitamin B12) biosynthesis in Porphyromonas gingivalis. Identification and characterization of a functional corrin pathway.\n",
      "  JOURNAL   J Biol Chem 275:40316-23 (2000)\n",
      "            DOI:10.1074/jbc.M007146200\n",
      "REFERENCE   PMID:12196148\n",
      "  AUTHORS   Scott AI, Roessner CA.\n",
      "  TITLE     Biosynthesis of cobalamin (vitamin B(12)).\n",
      "  JOURNAL   Biochem Soc Trans 30:613-20 (2002)\n",
      "            DOI:10.1042/bst0300613\n",
      "REFERENCE   PMID:12195810\n",
      "  AUTHORS   Warren MJ, Raux E, Schubert HL, Escalante-Semerena JC.\n",
      "  TITLE     The biosynthesis of adenosylcobalamin (vitamin B12).\n",
      "  JOURNAL   Nat Prod Rep 19:390-412 (2002)\n",
      "            DOI:10.1039/b108967f\n",
      "REFERENCE   PMID:12055304\n",
      "  AUTHORS   Roessner CA, Huang KX, Warren MJ, Raux E, Scott AI.\n",
      "  TITLE     Isolation and characterization of 14 additional genes specifying the anaerobic biosynthesis of cobalamin (vitamin B12) in Propionibacterium freudenreichii (P. shermanii).\n",
      "  JOURNAL   Microbiology 148:1845-53 (2002)\n",
      "            DOI:10.1099/00221287-148-6-1845\n",
      "REFERENCE   PMID:11283349\n",
      "  AUTHORS   Frankenberg N, Mukougawa K, Kohchi T, Lagarias JC.\n",
      "  TITLE     Functional genomic analysis of the HY2 family of ferredoxin-dependent bilin reductases from oxygenic photosynthetic organisms.\n",
      "  JOURNAL   Plant Cell 13:965-78 (2001)\n",
      "            DOI:10.1105/tpc.13.4.965\n",
      "REFERENCE   PMID:17506685\n",
      "  AUTHORS   Chew AG, Bryant DA\n",
      "  TITLE     Chlorophyll biosynthesis in bacteria: the origins of structural and functional diversity.\n",
      "  JOURNAL   Annu Rev Microbiol 61:113-29 (2007)\n",
      "            DOI:10.1146/annurev.micro.61.080706.093242\n",
      "REFERENCE   PMID:16245042\n",
      "  AUTHORS   Frigaard NU, Chew AG, Li H, Maresca JA, Bryant DA\n",
      "  TITLE     Chlorobium tepidum: insights into the structure, physiology, and metabolism of a green sulfur bacterium derived from the complete genome sequence.\n",
      "  JOURNAL   Photosynth Res 78:93-117 (2003)\n",
      "            DOI:10.1023/B:PRES.0000004310.96189.b4\n",
      "REFERENCE   PMID:23042036\n",
      "  AUTHORS   Deery E, Schroeder S, Lawrence AD, Taylor SL, Seyedarabi A, Waterman J, Wilson KS, Brown D, Geeves MA, Howard MJ, Pickersgill RW, Warren MJ\n",
      "  TITLE     An enzyme-trap approach allows isolation of intermediates in cobalamin biosynthesis.\n",
      "  JOURNAL   Nat Chem Biol 8:933-40 (2012)\n",
      "            DOI:10.1038/nchembio.1086\n",
      "REFERENCE   PMID:23922391\n",
      "  AUTHORS   Moore SJ, Lawrence AD, Biedendieck R, Deery E, Frank S, Howard MJ, Rigby SE, Warren MJ\n",
      "  TITLE     Elucidation of the anaerobic pathway for the corrin component of cobalamin (vitamin B12).\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 110:14906-11 (2013)\n",
      "            DOI:10.1073/pnas.1308098110\n",
      "REL_PATHWAY pae00230  Purine metabolism\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00740  Riboflavin metabolism\n",
      "KO_PATHWAY  ko00860\n",
      "///\n",
      "path:pae00900\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00900                    Pathway\n",
      "NAME        Terpenoid backbone biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Terpenoids, also known as isoprenoids, are a large class of natural products consisting of isoprene (C5) units. There are two biosynthetic pathways, the mevalonate pathway [MD:M00095] and the non-mevalonate pathway or the MEP/DOXP pathway [MD:M00096], for the terpenoid building blocks: isopentenyl diphosphate (IPP) and dimethylallyl diphosphate (DMAPP). The action of prenyltransferases then generates higher-order building blocks: geranyl diphosphate (GPP), farsenyl diphosphate (FPP), and geranylgeranyl diphosphate (GGPP), which are the precursors of monoterpenoids (C10), sesquiterpenoids (C15), and diterpenoids (C20), respectively. Condensation of these building blocks gives rise to the precursors of sterols (C30) and carotenoids (C40). The MEP/DOXP pathway is absent in higher animals and fungi, but in green plants the MEP/DOXP and mevalonate pathways co-exist in separate cellular compartments. The MEP/DOXP pathway, operating in the plastids, is responsible for the formation of essential oil monoterpenes and linalyl acetate, some sesquiterpenes, diterpenes, and carotenoids and phytol. The mevalonate pathway, operating in the cytosol, gives rise to triterpenes, sterols, and most sesquiterpenes.\n",
      "CLASS       Metabolism; Metabolism of terpenoids and polyketides\n",
      "PATHWAY_MAP pae00900  Terpenoid backbone biosynthesis\n",
      "DBLINKS     GO: 0009240 0016091\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4044  dxs; 1-deoxy-D-xylulose-5-phosphate synthase [KO:K01662] [EC:2.2.1.7]\n",
      "line in gene PA4044  dxs; 1-deoxy-D-xylulose-5-phosphate synthase [KO:K01662] [EC:2.2.1.7]\n",
      "            PA3650  dxr; 1-deoxy-D-xylulose 5-phosphate reductoisomerase [KO:K00099] [EC:1.1.1.267]\n",
      "line in gene PA3650  dxr; 1-deoxy-D-xylulose 5-phosphate reductoisomerase [KO:K00099] [EC:1.1.1.267]\n",
      "            PA3633  ygbP; 2-C-methyl-D-erythritol 4-phosphate cytidylyltransferase [KO:K00991] [EC:2.7.7.60]\n",
      "line in gene PA3633  ygbP; 2-C-methyl-D-erythritol 4-phosphate cytidylyltransferase [KO:K00991] [EC:2.7.7.60]\n",
      "            PA4669  ipk; 4-diphosphocytidyl-2C-methyl-D-erythritol kinase [KO:K00919] [EC:2.7.1.148]\n",
      "line in gene PA4669  ipk; 4-diphosphocytidyl-2C-methyl-D-erythritol kinase [KO:K00919] [EC:2.7.1.148]\n",
      "            PA3627  ygbB; 2-C-methyl-D-erythritol 2,4-cyclodiphosphate synthase [KO:K01770] [EC:4.6.1.12]\n",
      "line in gene PA3627  ygbB; 2-C-methyl-D-erythritol 2,4-cyclodiphosphate synthase [KO:K01770] [EC:4.6.1.12]\n",
      "            PA3803  gcpE; 4-hydroxy-3-methylbut-2-en-1-yl diphosphate synthase (flavodoxin) [KO:K03526] [EC:1.17.7.1 1.17.7.3]\n",
      "line in gene PA3803  gcpE; 4-hydroxy-3-methylbut-2-en-1-yl diphosphate synthase (flavodoxin) [KO:K03526] [EC:1.17.7.1 1.17.7.3]\n",
      "            PA4557  lytB; 4-hydroxy-3-methylbut-2-enyl diphosphate reductase [KO:K03527] [EC:1.17.7.4]\n",
      "line in gene PA4557  lytB; 4-hydroxy-3-methylbut-2-enyl diphosphate reductase [KO:K03527] [EC:1.17.7.4]\n",
      "            PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA4019  aromatic acid decarboxylase [KO:K03186] [EC:2.5.1.129]\n",
      "line in gene PA4019  aromatic acid decarboxylase [KO:K03186] [EC:2.5.1.129]\n",
      "            PA4043  ispA; geranyltranstransferase [KO:K13789] [EC:2.5.1.1 2.5.1.10 2.5.1.29]\n",
      "line in gene PA4043  ispA; geranyltranstransferase [KO:K13789] [EC:2.5.1.1 2.5.1.10 2.5.1.29]\n",
      "            PA4569  ispB; octaprenyl-diphosphate synthase [KO:K02523] [EC:2.5.1.90]\n",
      "line in gene PA4569  ispB; octaprenyl-diphosphate synthase [KO:K02523] [EC:2.5.1.90]\n",
      "            PA3652  uppS; ditrans,polycis-undecaprenyl-diphosphate synthase [KO:K00806] [EC:2.5.1.31]\n",
      "line in gene PA3652  uppS; ditrans,polycis-undecaprenyl-diphosphate synthase [KO:K00806] [EC:2.5.1.31]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00118  D-Glyceraldehyde 3-phosphate\n",
      "            C00129  Isopentenyl diphosphate\n",
      "            C00235  Dimethylallyl diphosphate\n",
      "            C00332  Acetoacetyl-CoA\n",
      "            C00341  Geranyl diphosphate\n",
      "            C00353  Geranylgeranyl diphosphate\n",
      "            C00356  (S)-3-Hydroxy-3-methylglutaryl-CoA\n",
      "            C00418  (R)-Mevalonate\n",
      "            C00448  trans,trans-Farnesyl diphosphate\n",
      "            C01107  (R)-5-Phosphomevalonate\n",
      "            C01126  (2E,6E)-Farnesol\n",
      "            C01143  (R)-5-Diphosphomevalonate\n",
      "            C01230  all-trans-Hexaprenyl diphosphate\n",
      "            C02321  cis-1,4-Polyisoprene\n",
      "            C03461  2-trans,6-trans-Farnesal\n",
      "            C04145  all-trans-Nonaprenyl diphosphate\n",
      "            C04146  all-trans-Octaprenyl diphosphate\n",
      "            C04216  all-trans-Heptaprenyl diphosphate\n",
      "            C04506  Protein C-terminal S-farnesyl-L-cysteine\n",
      "            C04574  di-trans,poly-cis-Undecaprenyl diphosphate\n",
      "            C04748  Protein C-terminal S-farnesyl-L-cysteine methyl ester\n",
      "            C05427  Phytyl diphosphate\n",
      "            C05859  Dehydrodolichol diphosphate\n",
      "            C06081  Polyprenol\n",
      "            C11434  2-C-Methyl-D-erythritol 4-phosphate\n",
      "            C11435  4-(Cytidine 5'-diphospho)-2-C-methyl-D-erythritol\n",
      "            C11436  2-Phospho-4-(cytidine 5'-diphospho)-2-C-methyl-D-erythritol\n",
      "            C11437  1-Deoxy-D-xylulose 5-phosphate\n",
      "            C11453  2-C-Methyl-D-erythritol 2,4-cyclodiphosphate\n",
      "            C11811  1-Hydroxy-2-methyl-2-butenyl 4-diphosphate\n",
      "            C16521  Isoprene\n",
      "            C16826  2-cis,6-trans-Farnesyl diphosphate\n",
      "            C17432  all-trans-Decaprenyl diphosphate\n",
      "            C18321  trans,octacis-Decaprenyl diphosphate\n",
      "            C19691  Farnesylcysteine\n",
      "            C19760  (2Z,6Z)-Farnesyl diphosphate\n",
      "            C19852  tritrans,heptacis-Undecaprenyl diphosphate\n",
      "            C19853  trans,polycis-Polyprenyl diphosphate\n",
      "            C20120  S-Farnesyl protein\n",
      "            C20121  trans,trans-Farnesyl phosphate\n",
      "            C20345  Isopentenyl phosphate\n",
      "            C20847  (R)-Mevalonate 3-phosphate\n",
      "            C20848  (R)-Mevalonate 3,5-bisphosphate\n",
      "            C22453  trans-Anhydromevalonate phosphate\n",
      "REFERENCE   PMID:12777052\n",
      "  AUTHORS   Lange BM, Ghassemian M.\n",
      "  TITLE     Genome organization in Arabidopsis thaliana: a survey for genes involved in isoprenoid and chlorophyll metabolism.\n",
      "  JOURNAL   Plant Mol Biol 51:925-48 (2003)\n",
      "            DOI:10.1023/A:1023005504702\n",
      "REFERENCE   PMID:16262699\n",
      "  AUTHORS   Saiki R, Nagata A, Kainou T, Matsuda H, Kawamukai M\n",
      "  TITLE     Characterization of solanesyl and decaprenyl diphosphate synthases in mice and humans.\n",
      "  JOURNAL   FEBS J 272:5606-22 (2005)\n",
      "            DOI:10.1111/j.1742-4658.2005.04956.x\n",
      "REFERENCE   PMID:9858571\n",
      "  AUTHORS   Sato M, Sato K, Nishikawa S, Hirata A, Kato J, Nakano A\n",
      "  TITLE     The yeast RER2 gene, identified by endoplasmic reticulum protein localization mutations, encodes cis-prenyltransferase, a key enzyme in dolichol synthesis.\n",
      "  JOURNAL   Mol Cell Biol 19:471-83 (1999)\n",
      "            DOI:10.1128/MCB.19.1.471\n",
      "REFERENCE   PMID:24375100\n",
      "  AUTHORS   Vannice JC, Skaff DA, Keightley A, Addo JK, Wyckoff GJ, Miziorko HM\n",
      "  TITLE     Identification in Haloferax volcanii of phosphomevalonate decarboxylase and isopentenyl phosphate kinase as catalysts of the terminal enzyme reactions in an  archaeal alternate mevalonate pathway.\n",
      "  JOURNAL   J Bacteriol 196:1055-63 (2014)\n",
      "            DOI:10.1128/JB.01230-13\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00130  Ubiquinone and other terpenoid-quinone biosynthesis\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "KO_PATHWAY  ko00900\n",
      "///\n",
      "path:pae00903\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00903                    Pathway\n",
      "NAME        Limonene and pinene degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of terpenoids and polyketides\n",
      "PATHWAY_MAP pae00903  Limonene and pinene degradation\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2098  esterase [KO:K14731] [EC:3.1.1.83]\n",
      "line in gene PA2098  esterase [KO:K14731] [EC:3.1.1.83]\n",
      "            PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA1027  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "line in gene PA4899  aldehyde dehydrogenase [KO:K00128] [EC:1.2.1.3]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "line in gene PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "            PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "line in gene PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "COMPOUND    C00521  (S)-Limonene\n",
      "            C00964  (-)-trans-Carveol\n",
      "            C01767  (-)-Carvone\n",
      "            C02452  (-)-Perillyl alcohol\n",
      "            C02576  Perillyl aldehyde\n",
      "            C02759  alpha-Pinene-oxide\n",
      "            C04435  (Z)-2-Methyl-5-isopropylhexa-2,5-dienal\n",
      "            C06099  (R)-Limonene\n",
      "            C07271  Limonene-1,2-epoxide\n",
      "            C07276  Limonene-1,2-diol\n",
      "            C09880  alpha-Pinene\n",
      "            C09884  Pinocarvone\n",
      "            C11383  (+)-(S)-Carvone\n",
      "            C11395  (4R,6R)-cis-Carveol\n",
      "            C11396  (1R,2R,4R)-Dihydrocarveol\n",
      "            C11397  (1R,2S,4R)-Neo-dihydrocarveol\n",
      "            C11398  (1R,4R)-Dihydrocarvone\n",
      "            C11399  (1S,2S,4R)-Iso-dihydrocarveol\n",
      "            C11400  (1S,2R,4R)-Neoiso-dihydrocarveol\n",
      "            C11401  (1S,4R)-Iso-dihydrocarvone\n",
      "            C11402  (4R,7R)-4-Isopropenyl-7-methyl-2-oxo-oxepanone\n",
      "            C11403  (3S,6R)-6-Isopropenyl-3-methyl-2-oxo-oxepanone\n",
      "            C11404  (3R)-6-Hydroxy-3-isopropenyl-heptanoate\n",
      "            C11405  (3R)-3-Isopropenyl-6-oxoheptanoate\n",
      "            C11406  (5R)-6-Hydroxy-5-isopropenyl-2-methylhexanoate\n",
      "            C11407  (3R)-3-Isopropenyl-6-oxoheptanoyl-CoA\n",
      "            C11408  (4S,6S)-cis-Carveol\n",
      "            C11409  (+)-trans-Carveol\n",
      "            C11410  (1R,2S,4S)-Neoiso-dihydrocarveol\n",
      "            C11411  (1R,2R,4S)-Iso-dihydrocarveol\n",
      "            C11412  (1R,4S)-Iso-dihydrocarvone\n",
      "            C11413  (1S,2S,4S)-Dihydrocarveol\n",
      "            C11414  (4S,7R)-4-Isopropenyl-7-methyl-2-oxo-oxepanone\n",
      "            C11415  (1S,4S)-Dihydrocarvone\n",
      "            C11416  (1S,2R,4S)-Neo-dihydrocarveol\n",
      "            C11417  (3S)-6-Hydroxy-3-isopropenyl-heptanoate\n",
      "            C11418  (3S,6S)-6-Isopropenyl-3-methyl-2-oxo-oxepanone\n",
      "            C11419  (3S)-3-Isopropenyl-6-oxoheptanoate\n",
      "            C11420  (5S)-6-Hydroxy-5-isopropenyl-2-methylhexanoate\n",
      "            C11421  (3S)-3-Isopropenyl-6-oxoheptanoyl-CoA\n",
      "            C11924  Perillic acid\n",
      "            C11929  Perillyl-CoA\n",
      "            C11934  2-Hydroxy-4-isopropenylcyclohexane-1-carboxyl-CoA\n",
      "            C11935  4-Isopropenyl-2-oxy-cyclohexanecarboxyl-CoA\n",
      "            C11936  3-Isopropenylpimelyl-CoA\n",
      "            C11937  (1S,4R)-1-Hydroxy-2-oxolimonene\n",
      "            C11938  Myrtenol\n",
      "            C11939  Myrtenal\n",
      "            C11940  Myrtenic acid\n",
      "            C11941  Pinocarveol\n",
      "            C11942  trans-2-Methyl-5-isopropylhexa-2,5-dienal\n",
      "            C11943  trans-2-Methyl-5-isopropylhexa-2,5-dienoic acid\n",
      "            C11944  cis-2-Methyl-5-isopropylhexa-2,5-dienoic acid\n",
      "            C11945  trans-2-Methyl-5-isopropylhexa-2,5-dienoyl-CoA\n",
      "            C11946  cis-2-Methyl-5-isopropylhexa-2,5-dienoyl-CoA\n",
      "            C11947  3-Hydroxy-2,6-dimethyl-5-methylene-heptanoyl-CoA\n",
      "            C11948  2,6-Dimethyl-5-methylene-3-oxo-heptanoyl-CoA\n",
      "            C11949  3-Isopropylbut-3-enoyl-CoA\n",
      "            C11950  3-Isopropyl-3-butenoic acid\n",
      "            C19079  (4R)-7-Hydroxy-4-isopropenyl-7-methyl-2-oxo-oxepanone\n",
      "            C19081  (4S)-Limonene-1,2-epoxide\n",
      "            C19082  (1R,2R,4S)-Limonene-1,2-diol\n",
      "            C19083  (1R,4S)-1-Hydroxy-2-oxolimonene\n",
      "            C19084  (4S)-7-Hydroxy-4-isopropenyl-7-methyl-2-oxo-oxepanone\n",
      "REFERENCE   PMID:10224006\n",
      "  AUTHORS   van der Werf MJ, Swarts HJ, de Bont JA\n",
      "  TITLE     Rhodococcus erythropolis DCL14 contains a novel degradation pathway for limonene.\n",
      "  JOURNAL   Appl Environ Microbiol 65:2092-102 (1999)\n",
      "            DOI:10.1128/AEM.65.5.2092-2102.1999\n",
      "REFERENCE   PMID:10832640\n",
      "  AUTHORS   van der Werf MJ, Boot AM\n",
      "  TITLE     Metabolism of carveol and dihydrocarveol in Rhodococcus erythropolis DCL14.\n",
      "  JOURNAL   Microbiology 146 ( Pt 5):1129-41 (2000)\n",
      "            DOI:10.1099/00221287-146-5-1129\n",
      "KO_PATHWAY  ko00903\n",
      "///\n",
      "path:pae00910\n",
      "ENTRY       pae00910                    Pathway\n",
      "NAME        Nitrogen metabolism - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION The biological process of the nitrogen cycle is a complex interplay among many microorganisms catalyzing different reactions, where nitrogen is found in various oxidation states ranging from +5 in nitrate to -3 in ammonia. The core nitrogen cycle involves four reduction pathways and two oxidation pathways. Nitrogen fixation [MD:M00175] is the process of reducing atmospheric molecular nitrogen to ammonia, a biologically useful reduced form incorporated into amino acids and other vital compounds. The ability of fixing atmospheric nitrogen by the nitrogenase enzyme complex is present in restricted prokaryotes (diazotrophs). The other reduction pathways are assimilatory nitrate reduction [MD:M00531] and dissimilatory nitrate reduction [MD:M00530] both for conversion to ammonia, and denitrification [MD:M00529]. Denitrification is a respiration in which nitrate or nitrite is reduced as a terminal electron acceptor under low oxygen or anoxic conditions, producing gaseous nitrogen compounds (N2, NO and N2O) to the atmosphere. The two oxidation pathways are nitrification [MD:M00528] and anaerobic ammonium oxidation (anammox). Nitrification is the oxidation of ammonia (NH3) with oxygen into nitrite followed by the oxidation of nitrite into nitrate. The first step is performed by ammonia-oxidizing microorganisms (e.g., Nitrosomonas and Nitrosococcus) and the second step by nitrite-oxidizing microorganisms (e.g., Nitrobacter). Anammox is a biochemical process of oxidizing ammonium (NH4+) into dinitrogen gas (N2) using nitrite as an electron acceptor. It occurs in the anammoxosome, a membrane bound compartment inside the cytoplasm, of anammox bacteria (e.g., Planctomycetes).\n",
      "CLASS       Metabolism; Energy metabolism\n",
      "PATHWAY_MAP pae00910  Nitrogen metabolism\n",
      "MODULE      pae_M00529  Denitrification, nitrate => nitrogen [PATH:pae00910]\n",
      "            pae_M00530  Dissimilatory nitrate reduction, nitrate => ammonia [PATH:pae00910]\n",
      "DBLINKS     GO: 0006807\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1783  nasA; nitrate transporter [KO:K02575]\n",
      "line in gene PA1783  nasA; nitrate transporter [KO:K02575]\n",
      "            PA3876  narK2; nitrite extrusion protein 2 [KO:K02575]\n",
      "line in gene PA3876  narK2; nitrite extrusion protein 2 [KO:K02575]\n",
      "            PA3877  narK1; nitrite extrusion protein 1 [KO:K02575]\n",
      "line in gene PA3877  narK1; nitrite extrusion protein 1 [KO:K02575]\n",
      "            PA3875  narG; respiratory nitrate reductase subunit alpha [KO:K00370] [EC:1.7.5.1 1.7.99.-]\n",
      "line in gene PA3875  narG; respiratory nitrate reductase subunit alpha [KO:K00370] [EC:1.7.5.1 1.7.99.-]\n",
      "            PA3874  narH; respiratory nitrate reductase subunit beta [KO:K00371] [EC:1.7.5.1 1.7.99.-]\n",
      "line in gene PA3874  narH; respiratory nitrate reductase subunit beta [KO:K00371] [EC:1.7.5.1 1.7.99.-]\n",
      "            PA3872  narI; respiratory nitrate reductase subunit gamma [KO:K00374] [EC:1.7.5.1 1.7.99.-]\n",
      "line in gene PA3872  narI; respiratory nitrate reductase subunit gamma [KO:K00374] [EC:1.7.5.1 1.7.99.-]\n",
      "            PA1174  napA; nitrate reductase catalytic subunit [KO:K02567] [EC:1.9.6.1]\n",
      "line in gene PA1174  napA; nitrate reductase catalytic subunit [KO:K02567] [EC:1.9.6.1]\n",
      "            PA1173  napB; cytochrome C protein NapB [KO:K02568]\n",
      "line in gene PA1173  napB; cytochrome C protein NapB [KO:K02568]\n",
      "            PA1779  assimilatory nitrate reductase [KO:K00372] [EC:1.7.99.-]\n",
      "line in gene PA1779  assimilatory nitrate reductase [KO:K00372] [EC:1.7.99.-]\n",
      "            PA1781  nirB; assimilatory nitrite reductase large subunit [KO:K00362] [EC:1.7.1.15]\n",
      "line in gene PA1781  nirB; assimilatory nitrite reductase large subunit [KO:K00362] [EC:1.7.1.15]\n",
      "            PA1780  nirD; assimilatory nitrite reductase small subunit [KO:K00363] [EC:1.7.1.15]\n",
      "line in gene PA1780  nirD; assimilatory nitrite reductase small subunit [KO:K00363] [EC:1.7.1.15]\n",
      "            PA0519  nirS; nitrite reductase [KO:K15864] [EC:1.7.2.1 1.7.99.1]\n",
      "line in gene PA0519  nirS; nitrite reductase [KO:K15864] [EC:1.7.2.1 1.7.99.1]\n",
      "            PA0524  norB; nitric oxide reductase subunit B [KO:K04561] [EC:1.7.2.5]\n",
      "line in gene PA0524  norB; nitric oxide reductase subunit B [KO:K04561] [EC:1.7.2.5]\n",
      "            PA0523  norC; nitric oxide reductase subunit C [KO:K02305]\n",
      "line in gene PA0523  norC; nitric oxide reductase subunit C [KO:K02305]\n",
      "            PA3392  nosZ; nitrous-oxide reductase [KO:K00376] [EC:1.7.2.4]\n",
      "line in gene PA3392  nosZ; nitrous-oxide reductase [KO:K00376] [EC:1.7.2.4]\n",
      "            PA4202  hypothetical protein [KO:K00459] [EC:1.13.12.16]\n",
      "line in gene PA4202  hypothetical protein [KO:K00459] [EC:1.13.12.16]\n",
      "            PA0660  hypothetical protein [KO:K00459] [EC:1.13.12.16]\n",
      "line in gene PA0660  hypothetical protein [KO:K00459] [EC:1.13.12.16]\n",
      "            PA3068  gdhB; NAD-specific glutamate dehydrogenase [KO:K15371] [EC:1.4.1.2]\n",
      "line in gene PA3068  gdhB; NAD-specific glutamate dehydrogenase [KO:K15371] [EC:1.4.1.2]\n",
      "            PA4588  gdhA; glutamate dehydrogenase [KO:K00262] [EC:1.4.1.4]\n",
      "line in gene PA4588  gdhA; glutamate dehydrogenase [KO:K00262] [EC:1.4.1.4]\n",
      "            PA0296  spuI; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA0296  spuI; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA0298  spuB; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA0298  spuB; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA2040  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA2040  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5119  glnA; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5119  glnA; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5522  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5522  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA1566  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA1566  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "            PA3356  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA3356  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5508  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5508  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5036  gltB; glutamate synthase subunit alpha [KO:K00265] [EC:1.4.1.13]\n",
      "line in gene PA5036  gltB; glutamate synthase subunit alpha [KO:K00265] [EC:1.4.1.13]\n",
      "            PA5035  gltD; glutamate synthase subunit beta [KO:K00266] [EC:1.4.1.13]\n",
      "line in gene PA5035  gltD; glutamate synthase subunit beta [KO:K00266] [EC:1.4.1.13]\n",
      "            PA2052  cynS; cyanate hydratase [KO:K01725] [EC:4.2.1.104]\n",
      "line in gene PA2052  cynS; cyanate hydratase [KO:K01725] [EC:4.2.1.104]\n",
      "            PA5173  arcC; carbamate kinase [KO:K00926] [EC:2.7.2.2]\n",
      "line in gene PA5173  arcC; carbamate kinase [KO:K00926] [EC:2.7.2.2]\n",
      "            PA2053  cynT; carbonic anhydrase [KO:K01673] [EC:4.2.1.1]\n",
      "line in gene PA2053  cynT; carbonic anhydrase [KO:K01673] [EC:4.2.1.1]\n",
      "            PA0102  carbonic anhydrase [KO:K01673] [EC:4.2.1.1]\n",
      "line in gene PA0102  carbonic anhydrase [KO:K01673] [EC:4.2.1.1]\n",
      "            PA4676  carbonic anhydrase [KO:K01673] [EC:4.2.1.1]\n",
      "line in gene PA4676  carbonic anhydrase [KO:K01673] [EC:4.2.1.1]\n",
      "COMPOUND    C00011  CO2\n",
      "            C00014  Ammonia\n",
      "            C00025  L-Glutamate\n",
      "            C00058  Formate\n",
      "            C00064  L-Glutamine\n",
      "            C00088  Nitrite\n",
      "            C00169  Carbamoyl phosphate\n",
      "            C00192  Hydroxylamine\n",
      "            C00244  Nitrate\n",
      "            C00288  HCO3-\n",
      "            C00488  Formamide\n",
      "            C00533  Nitric oxide\n",
      "            C00697  Nitrogen\n",
      "            C00726  Nitrile\n",
      "            C00887  Nitrous oxide\n",
      "            C01417  Cyanate\n",
      "            C01563  Carbamate\n",
      "            C05361  Hydrazine\n",
      "            C06058  Nitroalkane\n",
      "REFERENCE   PMID:14702404\n",
      "  AUTHORS   Scott JD, Ludwig RA\n",
      "  TITLE     Azorhizobium caulinodans electron-transferring flavoprotein N electrochemically couples pyruvate dehydrogenase complex activity to N2 fixation.\n",
      "  JOURNAL   Microbiology 150:117-26 (2004)\n",
      "            DOI:10.1099/mic.0.26603-0\n",
      "REFERENCE   PMID:17408485\n",
      "  AUTHORS   Kneip C, Lockhart P, Voss C, Maier UG\n",
      "  TITLE     Nitrogen fixation in eukaryotes--new models for symbiosis.\n",
      "  JOURNAL   BMC Evol Biol 7:55 (2007)\n",
      "            DOI:10.1186/1471-2148-7-55\n",
      "REFERENCE   PMID:11004450\n",
      "  AUTHORS   Whittaker M, Bergmann D, Arciero D, Hooper AB\n",
      "  TITLE     Electron transfer during the oxidation of ammonia by the chemolithotrophic bacterium Nitrosomonas europaea.\n",
      "  JOURNAL   Biochim Biophys Acta 1459:346-55 (2000)\n",
      "            DOI:10.1016/S0005-2728(00)00171-7\n",
      "REFERENCE   PMID:15528644\n",
      "  AUTHORS   Cabello P, Roldan MD, Moreno-Vivian C\n",
      "  TITLE     Nitrate reduction and the nitrogen cycle in archaea.\n",
      "  JOURNAL   Microbiology 150:3527-46 (2004)\n",
      "            DOI:10.1099/mic.0.27303-0\n",
      "REFERENCE   PMID:11921398\n",
      "  AUTHORS   Stolz JF, Basu P\n",
      "  TITLE     Evolution of nitrate reductase: molecular and structural variations on a common function.\n",
      "  JOURNAL   Chembiochem 3:198-206 (2002)\n",
      "            DOI:10.1002/1439-7633(20020301)3:2/3<198::AID-CBIC198>3.0.CO;2-C\n",
      "REFERENCE   PMID:18021072\n",
      "  AUTHORS   Morozkina EV, Zvyagilskaya RA\n",
      "  TITLE     Nitrate reductases: structure, functions, and effect of stress factors.\n",
      "  JOURNAL   Biochemistry (Mosc) 72:1151-60 (2007)\n",
      "            DOI:10.1134/S0006297907100124\n",
      "REFERENCE   PMID:19247843\n",
      "  AUTHORS   Jetten MS, Niftrik L, Strous M, Kartal B, Keltjens JT, Op den Camp HJ\n",
      "  TITLE     Biochemistry and molecular biology of anammox bacteria.\n",
      "  JOURNAL   Crit Rev Biochem Mol Biol 44:65-84 (2009)\n",
      "            DOI:10.1080/10409230902722783\n",
      "REFERENCE   PMID:22221911\n",
      "  AUTHORS   Luesken FA, Wu ML, Op den Camp HJ, Keltjens JT, Stunnenberg H, Francoijs KJ, Strous M, Jetten MS\n",
      "  TITLE     Effect of oxygen on the anaerobic methanotroph 'Candidatus Methylomirabilis oxyfera': kinetic and transcriptional analysis.\n",
      "  JOURNAL   Environ Microbiol 14:1024-34 (2012)\n",
      "            DOI:10.1111/j.1462-2920.2011.02682.x\n",
      "REFERENCE   PMID:26610024\n",
      "  AUTHORS   Daims H, Lebedeva EV, Pjevac P, Han P, Herbold C, Albertsen M, Jehmlich N, Palatinszky M, Vierheilig J, Bulaev A, Kirkegaard RH, von Bergen M, Rattei T, Bendinger B, Nielsen PH, Wagner M\n",
      "  TITLE     Complete nitrification by Nitrospira bacteria.\n",
      "  JOURNAL   Nature 528:504-9 (2015)\n",
      "            DOI:10.1038/nature16461\n",
      "REFERENCE   PMID:26610025\n",
      "  AUTHORS   van Kessel MA, Speth DR, Albertsen M, Nielsen PH, Op den Camp HJ, Kartal B, Jetten MS, Lucker S\n",
      "  TITLE     Complete nitrification by a single microorganism.\n",
      "  JOURNAL   Nature 528:555-9 (2015)\n",
      "            DOI:10.1038/nature16459\n",
      "REL_PATHWAY pae00220  Arginine biosynthesis\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00460  Cyanoamino acid metabolism\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00680  Methane metabolism\n",
      "KO_PATHWAY  ko00910\n",
      "///\n",
      "path:pae00920\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00920                    Pathway\n",
      "NAME        Sulfur metabolism - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Sulfur is an essential element for life and the metabolism of organic sulfur compounds plays an important role in the global sulfur cycle. Sulfur occurs in various oxidation states ranging from +6 in sulfate to -2 in sulfide (H2S). Sulfate reduction can occur in both an energy consuming assimilatory pathway and an energy producing dissimilatory pathway. The assimilatory pathway, which is found in a wide range of organisms, produces reduced sulfur compounds for the biosynthesis of S-containing amino acids and does not lead to direct excretion of sulfide. In the dissimilatory pathway, which is restricted to obligatory anaerobic bacterial and archaeal lineages, sulfate (or sulfur) is the terminal electron acceptor of the respiratory chain producing large quantities of inorganic sulfide. Both pathways start from the activation of sulfate by reaction with ATP to form adenylyl sulfate (APS). In the assimilatory pathway [MD:M00176] APS is converted to 3'-phosphoadenylyl sulfate (PAPS) and then reduced to sulfite, and sulfite is further reduced to sulfide by the assimilatory sulfite reductase. In the dissimilatory pathway [MD:M00596] APS is directly reduced to sulfite, and sulfite is further reduced to sulfide by the dissimilatory sulfite reductase. The capacity for oxidation of sulfur is quite widespread among bacteria and archaea, comprising phototrophs and chemolithoautotrophs. The SOX (sulfur-oxidation) system [MD:M00595] is a well-known sulfur oxidation pathway and is found in both photosynthetic and non-photosynthetic sulfur-oxidizing bacteria. Green sulfur bacteria and purple sulfur bacteria carry out anoxygenic photosynthesis with reduced sulfur compounds such as sulfide and elemental sulfur, as well as thiosulfate (in some species with the SOX system), as the electron donor for photoautotrophic growth. In some chemolithoautotrophic sulfur oxidizers (such as Thiobacillus denitrificans), it has been suggested that dissimilatory sulfur reduction enzymes operate in the reverse direction, forming a sulfur oxidation pathway from sulfite to APS and then to sulfate.\n",
      "CLASS       Metabolism; Energy metabolism\n",
      "PATHWAY_MAP pae00920  Sulfur metabolism\n",
      "MODULE      pae_M00021  Cysteine biosynthesis, serine => cysteine [PATH:pae00920]\n",
      "            pae_M00176  Assimilatory sulfate reduction, sulfate => H2S [PATH:pae00920]\n",
      "DBLINKS     GO: 0006790\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1493  cysP; sulfate ABC transporter substrate-binding protein [KO:K02048]\n",
      "line in gene PA1493  cysP; sulfate ABC transporter substrate-binding protein [KO:K02048]\n",
      "            PA0283  sbp; sulfate-binding protein [KO:K23163]\n",
      "line in gene PA0283  sbp; sulfate-binding protein [KO:K23163]\n",
      "            PA0282  cysT; sulfate transporter CysT [KO:K02046]\n",
      "line in gene PA0282  cysT; sulfate transporter CysT [KO:K02046]\n",
      "            PA0281  cysW; sulfate transporter CysW [KO:K02047]\n",
      "line in gene PA0281  cysW; sulfate transporter CysW [KO:K02047]\n",
      "            PA0280  cysA; sulfate.thiosulfate ABC transporter ATP-binding protein CysA [KO:K02045] [EC:7.3.2.3]\n",
      "line in gene PA0280  cysA; sulfate.thiosulfate ABC transporter ATP-binding protein CysA [KO:K02045] [EC:7.3.2.3]\n",
      "            PA3938  taurine-binding protein [KO:K15551]\n",
      "line in gene PA3938  taurine-binding protein [KO:K15551]\n",
      "            PA3936  taurine ABC transporter permease [KO:K15552]\n",
      "line in gene PA3936  taurine ABC transporter permease [KO:K15552]\n",
      "            PA3937  taurine ABC transporter ATP-binding protein [KO:K10831] [EC:7.6.2.7]\n",
      "line in gene PA3937  taurine ABC transporter ATP-binding protein [KO:K10831] [EC:7.6.2.7]\n",
      "            PA2310  hypothetical protein [KO:K03119] [EC:1.14.11.17]\n",
      "line in gene PA2310  hypothetical protein [KO:K03119] [EC:1.14.11.17]\n",
      "            PA3935  tauD; taurine dioxygenase [KO:K03119] [EC:1.14.11.17]\n",
      "line in gene PA3935  tauD; taurine dioxygenase [KO:K03119] [EC:1.14.11.17]\n",
      "            PA3449  hypothetical protein [KO:K15553]\n",
      "line in gene PA3449  hypothetical protein [KO:K15553]\n",
      "            PA2595  hypothetical protein [KO:K15553]\n",
      "line in gene PA2595  hypothetical protein [KO:K15553]\n",
      "            PA2594  hypothetical protein [KO:K15553]\n",
      "line in gene PA2594  hypothetical protein [KO:K15553]\n",
      "            PA3445  hypothetical protein [KO:K15553]\n",
      "line in gene PA3445  hypothetical protein [KO:K15553]\n",
      "            PA0186  ABC transporter [KO:K15553]\n",
      "line in gene PA0186  ABC transporter [KO:K15553]\n",
      "            PA2596  hypothetical protein [KO:K15553]\n",
      "line in gene PA2596  hypothetical protein [KO:K15553]\n",
      "            PA2599  hypothetical protein [KO:K15553]\n",
      "line in gene PA2599  hypothetical protein [KO:K15553]\n",
      "            PA2309  hypothetical protein [KO:K15553]\n",
      "line in gene PA2309  hypothetical protein [KO:K15553]\n",
      "            PA2296  hypothetical protein [KO:K15553]\n",
      "line in gene PA2296  hypothetical protein [KO:K15553]\n",
      "            PA3443  ABC transporter permease [KO:K15554]\n",
      "line in gene PA3443  ABC transporter permease [KO:K15554]\n",
      "            PA3448  ABC transporter permease [KO:K15554]\n",
      "line in gene PA3448  ABC transporter permease [KO:K15554]\n",
      "            PA0185  ABC transporter permease [KO:K15554]\n",
      "line in gene PA0185  ABC transporter permease [KO:K15554]\n",
      "            PA2307  ABC transporter permease [KO:K15554]\n",
      "line in gene PA2307  ABC transporter permease [KO:K15554]\n",
      "            PA2295  ABC transporter permease [KO:K15554]\n",
      "line in gene PA2295  ABC transporter permease [KO:K15554]\n",
      "            PA3442  aliphatic sulfonates ABC transporter ATP-binding subunit [KO:K15555] [EC:7.6.2.14]\n",
      "line in gene PA3442  aliphatic sulfonates ABC transporter ATP-binding subunit [KO:K15555] [EC:7.6.2.14]\n",
      "            PA0184  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "line in gene PA0184  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "            PA3447  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "line in gene PA3447  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "            PA2308  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "line in gene PA2308  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "            PA2294  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "line in gene PA2294  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "            PA2356  msuD; methanesulfonate monooxygenase [KO:K04091] [EC:1.14.14.5 1.14.14.34]\n",
      "line in gene PA2356  msuD; methanesulfonate monooxygenase [KO:K04091] [EC:1.14.14.5 1.14.14.34]\n",
      "            PA3444  alkanesulfonate monooxygenase [KO:K04091] [EC:1.14.14.5 1.14.14.34]\n",
      "line in gene PA3444  alkanesulfonate monooxygenase [KO:K04091] [EC:1.14.14.5 1.14.14.34]\n",
      "            PA2600  hypothetical protein [KO:K04091] [EC:1.14.14.5 1.14.14.34]\n",
      "line in gene PA2600  hypothetical protein [KO:K04091] [EC:1.14.14.5 1.14.14.34]\n",
      "            PA2598  hypothetical protein [KO:K04091] [EC:1.14.14.5 1.14.14.34]\n",
      "line in gene PA2598  hypothetical protein [KO:K04091] [EC:1.14.14.5 1.14.14.34]\n",
      "            PA2357  msuE; FMN reductase MsuE [KO:K00299] [EC:1.5.1.38]\n",
      "line in gene PA2357  msuE; FMN reductase MsuE [KO:K00299] [EC:1.5.1.38]\n",
      "            PA3446  NAD(P)H-dependent FMN reductase [KO:K00299] [EC:1.5.1.38]\n",
      "line in gene PA3446  NAD(P)H-dependent FMN reductase [KO:K00299] [EC:1.5.1.38]\n",
      "            PA4442  cysN; bifunctional sulfate adenylyltransferase subunit 1/adenylylsulfate kinase [KO:K00955] [EC:2.7.7.4 2.7.1.25]\n",
      "line in gene PA4442  cysN; bifunctional sulfate adenylyltransferase subunit 1/adenylylsulfate kinase [KO:K00955] [EC:2.7.7.4 2.7.1.25]\n",
      "            PA4443  cysD; sulfate adenylyltransferase subunit 2 [KO:K00957] [EC:2.7.7.4]\n",
      "line in gene PA4443  cysD; sulfate adenylyltransferase subunit 2 [KO:K00957] [EC:2.7.7.4]\n",
      "            PA1393  cysC; adenylyl-sulfate kinase [KO:K00860] [EC:2.7.1.25]\n",
      "line in gene PA1393  cysC; adenylyl-sulfate kinase [KO:K00860] [EC:2.7.1.25]\n",
      "            PA5175  cysQ; 3'(2'),5'-bisphosphate nucleotidase CysQ [KO:K01082] [EC:3.1.3.7]\n",
      "line in gene PA5175  cysQ; 3'(2'),5'-bisphosphate nucleotidase CysQ [KO:K01082] [EC:3.1.3.7]\n",
      "            PA1756  cysH; phosphoadenosine phosphosulfate reductase [KO:K00390] [EC:1.8.4.8 1.8.4.10]\n",
      "line in gene PA1756  cysH; phosphoadenosine phosphosulfate reductase [KO:K00390] [EC:1.8.4.8 1.8.4.10]\n",
      "            PA4513  oxidoreductase [KO:K00380] [EC:1.8.1.2]\n",
      "line in gene PA4513  oxidoreductase [KO:K00380] [EC:1.8.1.2]\n",
      "            PA1838  cysI; sulfite reductase [KO:K00381] [EC:1.8.1.2]\n",
      "line in gene PA1838  cysI; sulfite reductase [KO:K00381] [EC:1.8.1.2]\n",
      "            PA4130  sulfite/nitrite reductase [KO:K00381] [EC:1.8.1.2]\n",
      "line in gene PA4130  sulfite/nitrite reductase [KO:K00381] [EC:1.8.1.2]\n",
      "            PA2345  hypothetical protein [KO:K17218] [EC:1.8.5.4]\n",
      "line in gene PA2345  hypothetical protein [KO:K17218] [EC:1.8.5.4]\n",
      "            PA2566  hypothetical protein [KO:K17218] [EC:1.8.5.4]\n",
      "line in gene PA2566  hypothetical protein [KO:K17218] [EC:1.8.5.4]\n",
      "            PA1292  3-mercaptopyruvate sulfurtransferase [KO:K01011] [EC:2.8.1.1 2.8.1.2]\n",
      "line in gene PA1292  3-mercaptopyruvate sulfurtransferase [KO:K01011] [EC:2.8.1.1 2.8.1.2]\n",
      "            PA4956  rhdA; thiosulfate:cyanide sulfurtransferase [KO:K01011] [EC:2.8.1.1 2.8.1.2]\n",
      "line in gene PA4956  rhdA; thiosulfate:cyanide sulfurtransferase [KO:K01011] [EC:2.8.1.1 2.8.1.2]\n",
      "            PA0589  thiosulfate sulfurtransferase [KO:K02439] [EC:2.8.1.1]\n",
      "line in gene PA0589  thiosulfate sulfurtransferase [KO:K02439] [EC:2.8.1.1]\n",
      "            PA3816  cysE; O-acetylserine synthase [KO:K00640] [EC:2.3.1.30]\n",
      "line in gene PA3816  cysE; O-acetylserine synthase [KO:K00640] [EC:2.3.1.30]\n",
      "            PA2105  acetyltransferase [KO:K00640] [EC:2.3.1.30]\n",
      "line in gene PA2105  acetyltransferase [KO:K00640] [EC:2.3.1.30]\n",
      "            PA1061  hypothetical protein [KO:K01738] [EC:2.5.1.47]\n",
      "line in gene PA1061  hypothetical protein [KO:K01738] [EC:2.5.1.47]\n",
      "            PA2104  cysteine synthase [KO:K01738] [EC:2.5.1.47]\n",
      "line in gene PA2104  cysteine synthase [KO:K01738] [EC:2.5.1.47]\n",
      "            PA2709  cysK; cysteine synthase A [KO:K01738] [EC:2.5.1.47]\n",
      "line in gene PA2709  cysK; cysteine synthase A [KO:K01738] [EC:2.5.1.47]\n",
      "            PA0390  metX; homoserine O-acetyltransferase [KO:K00641] [EC:2.3.1.31 2.3.1.46]\n",
      "line in gene PA0390  metX; homoserine O-acetyltransferase [KO:K00641] [EC:2.3.1.31 2.3.1.46]\n",
      "            PA3107  metZ; O-succinylhomoserine sulfhydrylase [KO:K10764] [EC:2.5.1.-]\n",
      "line in gene PA3107  metZ; O-succinylhomoserine sulfhydrylase [KO:K10764] [EC:2.5.1.-]\n",
      "            PA3954  hypothetical protein [KO:K17228] [EC:1.14.14.35]\n",
      "line in gene PA3954  hypothetical protein [KO:K17228] [EC:1.14.14.35]\n",
      "COMPOUND    C00033  Acetate\n",
      "            C00042  Succinate\n",
      "            C00053  3'-Phosphoadenylyl sulfate\n",
      "            C00054  Adenosine 3',5'-bisphosphate\n",
      "            C00059  Sulfate\n",
      "            C00065  L-Serine\n",
      "            C00084  Acetaldehyde\n",
      "            C00087  Sulfur\n",
      "            C00094  Sulfite\n",
      "            C00097  L-Cysteine\n",
      "            C00155  L-Homocysteine\n",
      "            C00224  Adenylyl sulfate\n",
      "            C00245  Taurine\n",
      "            C00263  L-Homoserine\n",
      "            C00283  Hydrogen sulfide\n",
      "            C00320  Thiosulfate\n",
      "            C00409  Methanethiol\n",
      "            C00580  Dimethyl sulfide\n",
      "            C00979  O-Acetyl-L-serine\n",
      "            C01118  O-Succinyl-L-homoserine\n",
      "            C01861  Trithionate\n",
      "            C02084  Tetrathionate\n",
      "            C03920  2-(Methylthio)ethanesulfonate\n",
      "            C04022  S,S-Dimethyl-beta-propiothetin\n",
      "            C08276  3-(Methylthio)propanoate\n",
      "            C11142  Dimethyl sulfone\n",
      "            C11143  Dimethyl sulfoxide\n",
      "            C11145  Methanesulfonic acid\n",
      "            C15521  Alkanesulfonate\n",
      "            C17267  S-Sulfanylglutathione\n",
      "            C19692  Polysulfide\n",
      "            C20870  3-(Methylthio)propanoyl-CoA\n",
      "            C20955  3-(Methylthio)acryloyl-CoA\n",
      "REFERENCE   PMID:22982583\n",
      "  AUTHORS   Grein F, Ramos AR, Venceslau SS, Pereira IA\n",
      "  TITLE     Unifying concepts in anaerobic respiration: Insights from dissimilatory sulfur metabolism.\n",
      "  JOURNAL   Biochim Biophys Acta 1827:145-60 (2013)\n",
      "            DOI:10.1016/j.bbabio.2012.09.001\n",
      "REFERENCE   PMID:22633058\n",
      "  AUTHORS   Fauque GD, Barton LL\n",
      "  TITLE     Hemoproteins in dissimilatory sulfate- and sulfur-reducing prokaryotes.\n",
      "  JOURNAL   Adv Microb Physiol 60:1-90 (2012)\n",
      "            DOI:10.1016/B978-0-12-398264-3.00001-2\n",
      "REFERENCE   PMID:20143161\n",
      "  AUTHORS   Sakurai H, Ogawa T, Shiga M, Inoue K\n",
      "  TITLE     Inorganic sulfur oxidizing system in green sulfur bacteria.\n",
      "  JOURNAL   Photosynth Res 104:163-76 (2010)\n",
      "            DOI:10.1007/s11120-010-9531-2\n",
      "REFERENCE   PMID:22092713\n",
      "  AUTHORS   Falkenby LG, Szymanska M, Holkenbrink C, Habicht KS, Andersen JS, Miller M, Frigaard NU\n",
      "  TITLE     Quantitative proteomics of Chlorobaculum tepidum: insights into the sulfur metabolism of a phototrophic green sulfur bacterium.\n",
      "  JOURNAL   FEMS Microbiol Lett 323:142-50 (2011)\n",
      "            DOI:10.1111/j.1574-6968.2011.02370.x\n",
      "REFERENCE   PMID:21833341\n",
      "  AUTHORS   Gregersen LH, Bryant DA, Frigaard NU\n",
      "  TITLE     Mechanisms and evolution of oxidative sulfur metabolism in green sulfur bacteria.\n",
      "  JOURNAL   Front Microbiol 2:116 (2011)\n",
      "            DOI:10.3389/fmicb.2011.00116\n",
      "REFERENCE   PMID:16452431\n",
      "  AUTHORS   Beller HR, Chain PS, Letain TE, Chakicherla A, Larimer FW, Richardson PM, Coleman MA, Wood AP, Kelly DP.\n",
      "  TITLE     The genome sequence of the obligately chemolithoautotrophic, facultatively anaerobic bacterium Thiobacillus denitrificans.\n",
      "  JOURNAL   J Bacteriol 188:1473-88 (2006)\n",
      "            DOI:10.1128/JB.188.4.1473-1488.2006\n",
      "REFERENCE   PMID:9695921\n",
      "  AUTHORS   Pott AS, Dahl C\n",
      "  TITLE     Sirohaem sulfite reductase and other proteins encoded by genes at the dsr locus of Chromatium vinosum are involved in the oxidation of intracellular sulfur.\n",
      "  JOURNAL   Microbiology 144 ( Pt 7):1881-94 (1998)\n",
      "            DOI:10.1099/00221287-144-7-1881\n",
      "REFERENCE   PMID:18929068\n",
      "  AUTHORS   Frigaard NU, Dahl C\n",
      "  TITLE     Sulfur metabolism in phototrophic sulfur bacteria.\n",
      "  JOURNAL   Adv Microb Physiol 54:103-200 (2009)\n",
      "            DOI:10.1016/S0065-2911(08)00002-7\n",
      "REL_PATHWAY pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00680  Methane metabolism\n",
      "KO_PATHWAY  ko00920\n",
      "///\n",
      "path:pae00930\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00930                    Pathway\n",
      "NAME        Caprolactam degradation - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Xenobiotics biodegradation and metabolism\n",
      "PATHWAY_MAP pae00930  Caprolactam degradation\n",
      "DBLINKS     GO: 0019384\n",
      "            UMBBD: cap chx\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1525  alkB2; alkane-1 monooxygenase [KO:K00496] [EC:1.14.15.3]\n",
      "line in gene PA1525  alkB2; alkane-1 monooxygenase [KO:K00496] [EC:1.14.15.3]\n",
      "            PA2574  alkB1; alkane-1 monooxygenase [KO:K00496] [EC:1.14.15.3]\n",
      "line in gene PA2574  alkB1; alkane-1 monooxygenase [KO:K00496] [EC:1.14.15.3]\n",
      "            PA2098  esterase [KO:K14731] [EC:3.1.1.83]\n",
      "line in gene PA2098  esterase [KO:K14731] [EC:3.1.1.83]\n",
      "            PA1631  acyl-CoA dehydrogenase [KO:K06446] [EC:1.3.99.-]\n",
      "line in gene PA1631  acyl-CoA dehydrogenase [KO:K06446] [EC:1.3.99.-]\n",
      "            PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1821  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1748  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA3426  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "line in gene PA1629  enoyl-CoA hydratase [KO:K01692] [EC:4.2.1.17]\n",
      "            PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "line in gene PA3014  faoA; fatty acid oxidation complex subunit alpha [KO:K01825] [EC:1.1.1.35 4.2.1.17 5.1.2.3 5.3.3.8]\n",
      "            PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "line in gene PA1737  3-hydroxyacyl-CoA dehydrogenase [KO:K01782] [EC:1.1.1.35 4.2.1.17 5.1.2.3]\n",
      "COMPOUND    C00414  Cyclohexanone\n",
      "            C00571  Cyclohexylamine\n",
      "            C00854  Cyclohexanol\n",
      "            C01147  2-Hydroxycyclohexan-1-one\n",
      "            C01255  N-(6-Aminohexanoyl)-6-aminohexanoate\n",
      "            C01880  epsilon-Caprolactone\n",
      "            C02232  3-Oxoadipyl-CoA\n",
      "            C02378  6-Aminohexanoate\n",
      "            C02824  Cyclohexylsulfamate\n",
      "            C03241  6-Hydroxyhexan-6-olide\n",
      "            C03739  trans-1,2-Cyclohexanediol\n",
      "            C04277  1,8-Diazacyclotetradecane-2,9-dione\n",
      "            C06102  Adipate semialdehyde\n",
      "            C06103  6-Hydroxyhexanoic acid\n",
      "            C06104  Adipate\n",
      "            C06105  Cyclohexane-1,2-dione\n",
      "            C06593  epsilon-Caprolactam\n",
      "            C11249  Cyclohexane\n",
      "            C11519  N-Cyclohexylformamide\n",
      "            C11520  Cyclohexyl isocyanide\n",
      "            C12313  cis-1,2-Cyclohexanediol\n",
      "            C14143  Adipyl-CoA\n",
      "            C14144  5-Carboxy-2-pentenoyl-CoA\n",
      "            C14145  (3S)-3-Hydroxyadipyl-CoA\n",
      "            C20988  [N-(6-Aminohexanoyl)]n\n",
      "            C20990  6-Aminohexanoate cyclic oligomer\n",
      "REFERENCE   PMID:11571189\n",
      "  AUTHORS   Parke D, Garcia MA, Ornston LN\n",
      "  TITLE     Cloning and genetic characterization of dca genes required for beta-oxidation of straight-chain dicarboxylic acids in Acinetobacter sp. strain ADP1.\n",
      "  JOURNAL   Appl Environ Microbiol 67:4817-27 (2001)\n",
      "            DOI:10.1128/AEM.67.10.4817-4827.2001\n",
      "REFERENCE   PMID:12535077\n",
      "  AUTHORS   Campbell JW, Morgan-Kiss RM, Cronan JE Jr\n",
      "  TITLE     A new Escherichia coli metabolic competency: growth on fatty acids by a novel anaerobic beta-oxidation pathway.\n",
      "  JOURNAL   Mol Microbiol 47:793-805 (2003)\n",
      "            DOI:10.1046/j.1365-2958.2003.03341.x\n",
      "REFERENCE   PMID:15658985\n",
      "  AUTHORS   Brzostowicz PC, Walters DM, Jackson RE, Halsey KH, Ni H, Rouviere PE\n",
      "  TITLE     Proposed involvement of a soluble methane monooxygenase homologue in the cyclohexane-dependent growth of a new Brachymonas species.\n",
      "  JOURNAL   Environ Microbiol 7:179-90 (2005)\n",
      "            DOI:10.1111/j.1462-2920.2004.00681.x\n",
      "REL_PATHWAY pae00362  Benzoate degradation\n",
      "KO_PATHWAY  ko00930\n",
      "///\n",
      "path:pae00970\n",
      "ENTRY       pae00970                    Pathway\n",
      "NAME        Aminoacyl-tRNA biosynthesis - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Genetic Information Processing; Translation\n",
      "PATHWAY_MAP pae00970  Aminoacyl-tRNA biosynthesis\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3133.2  tRNA-Ala [KO:K14218]\n",
      "line in gene PA3133.2  tRNA-Ala [KO:K14218]\n",
      "            PA3133.4  tRNA-Ala [KO:K14218]\n",
      "line in gene PA3133.4  tRNA-Ala [KO:K14218]\n",
      "            PA0668.3  tRNA-Ala [KO:K14218]\n",
      "line in gene PA0668.3  tRNA-Ala [KO:K14218]\n",
      "            PA4280.3  tRNA-Ala [KO:K14218]\n",
      "line in gene PA4280.3  tRNA-Ala [KO:K14218]\n",
      "            PA5369.3  tRNA-Ala [KO:K14218]\n",
      "line in gene PA5369.3  tRNA-Ala [KO:K14218]\n",
      "            PA4690.3  tRNA-Ala [KO:K14218]\n",
      "line in gene PA4690.3  tRNA-Ala [KO:K14218]\n",
      "            PA0263.1  tRNA-Arg [KO:K14219]\n",
      "line in gene PA0263.1  tRNA-Arg [KO:K14219]\n",
      "            PA3368.1  tRNA-Arg [KO:K14219]\n",
      "line in gene PA3368.1  tRNA-Arg [KO:K14219]\n",
      "            PA1796.1  tRNA-Arg [KO:K14219]\n",
      "line in gene PA1796.1  tRNA-Arg [KO:K14219]\n",
      "            PA0905.2  tRNA-Arg [KO:K14219]\n",
      "line in gene PA0905.2  tRNA-Arg [KO:K14219]\n",
      "            PA0905.3  tRNA-Arg [KO:K14219]\n",
      "line in gene PA0905.3  tRNA-Arg [KO:K14219]\n",
      "            PA4581.1  tRNA-Arg [KO:K14219]\n",
      "line in gene PA4581.1  tRNA-Arg [KO:K14219]\n",
      "            PA3139.1  tRNA-Asn [KO:K14220]\n",
      "line in gene PA3139.1  tRNA-Asn [KO:K14220]\n",
      "            PA4541.3  tRNA-Asn [KO:K14220]\n",
      "line in gene PA4541.3  tRNA-Asn [KO:K14220]\n",
      "            PA1804.1  tRNA-Asp [KO:K14221]\n",
      "line in gene PA1804.1  tRNA-Asp [KO:K14221]\n",
      "            PA3094.1  tRNA-Asp [KO:K14221]\n",
      "line in gene PA3094.1  tRNA-Asp [KO:K14221]\n",
      "            PA3262.1  tRNA-Asp [KO:K14221]\n",
      "line in gene PA3262.1  tRNA-Asp [KO:K14221]\n",
      "            PA3094.2  tRNA-Asp [KO:K14221]\n",
      "line in gene PA3094.2  tRNA-Asp [KO:K14221]\n",
      "            PA2581.1  tRNA-Cys [KO:K14222]\n",
      "line in gene PA2581.1  tRNA-Cys [KO:K14222]\n",
      "            PA4669.1  tRNA-Gln [KO:K14223]\n",
      "line in gene PA4669.1  tRNA-Gln [KO:K14223]\n",
      "            PA2819.3  tRNA-Glu [KO:K14224]\n",
      "line in gene PA2819.3  tRNA-Glu [KO:K14224]\n",
      "            PA3133.3  tRNA-Glu [KO:K14224]\n",
      "line in gene PA3133.3  tRNA-Glu [KO:K14224]\n",
      "            PA3133.1  tRNA-Glu [KO:K14224]\n",
      "line in gene PA3133.1  tRNA-Glu [KO:K14224]\n",
      "            PA0729.1  tRNA-Gly [KO:K14225]\n",
      "line in gene PA0729.1  tRNA-Gly [KO:K14225]\n",
      "            PA4277.2  tRNA-Gly [KO:K14225]\n",
      "line in gene PA4277.2  tRNA-Gly [KO:K14225]\n",
      "            PA2583.1  tRNA-Gly [KO:K14225]\n",
      "line in gene PA2583.1  tRNA-Gly [KO:K14225]\n",
      "            PA2819.2  tRNA-Gly [KO:K14225]\n",
      "line in gene PA2819.2  tRNA-Gly [KO:K14225]\n",
      "            PA2819.1  tRNA-Gly [KO:K14225]\n",
      "line in gene PA2819.1  tRNA-Gly [KO:K14225]\n",
      "            PA1796.2  tRNA-His [KO:K14226]\n",
      "line in gene PA1796.2  tRNA-His [KO:K14226]\n",
      "            PA1796.4  tRNA-His [KO:K14226]\n",
      "line in gene PA1796.4  tRNA-His [KO:K14226]\n",
      "            PA0668.2  tRNA-Ile [KO:K14227]\n",
      "line in gene PA0668.2  tRNA-Ile [KO:K14227]\n",
      "            PA4690.4  tRNA-Ile [KO:K14227]\n",
      "line in gene PA4690.4  tRNA-Ile [KO:K14227]\n",
      "            PA4280.4  tRNA-Ile [KO:K14227]\n",
      "line in gene PA4280.4  tRNA-Ile [KO:K14227]\n",
      "            PA5369.4  tRNA-Ile [KO:K14227]\n",
      "line in gene PA5369.4  tRNA-Ile [KO:K14227]\n",
      "            PA4746.2  tRNA-Leu [KO:K14228]\n",
      "line in gene PA4746.2  tRNA-Leu [KO:K14228]\n",
      "            PA1796.3  tRNA-Leu [KO:K14228]\n",
      "line in gene PA1796.3  tRNA-Leu [KO:K14228]\n",
      "            PA2570.1  tRNA-Leu [KO:K14228]\n",
      "line in gene PA2570.1  tRNA-Leu [KO:K14228]\n",
      "            PA4937.1  tRNA-Leu [KO:K14228]\n",
      "line in gene PA4937.1  tRNA-Leu [KO:K14228]\n",
      "            PA4937.2  tRNA-Leu [KO:K14228]\n",
      "line in gene PA4937.2  tRNA-Leu [KO:K14228]\n",
      "            PA3824.1  tRNA-Leu [KO:K14228]\n",
      "line in gene PA3824.1  tRNA-Leu [KO:K14228]\n",
      "            PA0976.1  tRNA-Lys [KO:K14229]\n",
      "line in gene PA0976.1  tRNA-Lys [KO:K14229]\n",
      "            PA4541.1  tRNA-Lys [KO:K14229]\n",
      "line in gene PA4541.1  tRNA-Lys [KO:K14229]\n",
      "            PA0574.1  tRNA-Met [KO:K14230]\n",
      "line in gene PA0574.1  tRNA-Met [KO:K14230]\n",
      "            PA4673.1  tRNA-Met [KO:K14230]\n",
      "line in gene PA4673.1  tRNA-Met [KO:K14230]\n",
      "            PA0922.1  tRNA-Met [KO:K14230]\n",
      "line in gene PA0922.1  tRNA-Met [KO:K14230]\n",
      "            PA4746.1  tRNA-Met [KO:K14230]\n",
      "line in gene PA4746.1  tRNA-Met [KO:K14230]\n",
      "            PA5149.1  tRNA-Phe [KO:K14231]\n",
      "line in gene PA5149.1  tRNA-Phe [KO:K14231]\n",
      "            PA3031.1  tRNA-Pro [KO:K14232]\n",
      "line in gene PA3031.1  tRNA-Pro [KO:K14232]\n",
      "            PA2736.1  tRNA-Pro [KO:K14232]\n",
      "line in gene PA2736.1  tRNA-Pro [KO:K14232]\n",
      "            PA4541.2  tRNA-Pro [KO:K14232]\n",
      "line in gene PA4541.2  tRNA-Pro [KO:K14232]\n",
      "            PA2603.1  tRNA-Ser [KO:K14233]\n",
      "line in gene PA2603.1  tRNA-Ser [KO:K14233]\n",
      "            PA0905.1  tRNA-Ser [KO:K14233]\n",
      "line in gene PA0905.1  tRNA-Ser [KO:K14233]\n",
      "            PA1013.1  tRNA-Ser [KO:K14233]\n",
      "line in gene PA1013.1  tRNA-Ser [KO:K14233]\n",
      "            PA2852.1  tRNA-Ser [KO:K14233]\n",
      "line in gene PA2852.1  tRNA-Ser [KO:K14233]\n",
      "            PA5160.1  tRNA-Thr [KO:K14234]\n",
      "line in gene PA5160.1  tRNA-Thr [KO:K14234]\n",
      "            PA4277.1  tRNA-Thr [KO:K14234]\n",
      "line in gene PA4277.1  tRNA-Thr [KO:K14234]\n",
      "            PA4524.1  tRNA-Thr [KO:K14234]\n",
      "line in gene PA4524.1  tRNA-Thr [KO:K14234]\n",
      "            PA4276.1  tRNA-Trp [KO:K14235]\n",
      "line in gene PA4276.1  tRNA-Trp [KO:K14235]\n",
      "            PA4277.3  tRNA-Tyr [KO:K14236]\n",
      "line in gene PA4277.3  tRNA-Tyr [KO:K14236]\n",
      "            PA3094.3  tRNA-Val [KO:K14237]\n",
      "line in gene PA3094.3  tRNA-Val [KO:K14237]\n",
      "            PA3262.2  tRNA-Val [KO:K14237]\n",
      "line in gene PA3262.2  tRNA-Val [KO:K14237]\n",
      "            PA2775.1  tRNA-Val [KO:K14237]\n",
      "line in gene PA2775.1  tRNA-Val [KO:K14237]\n",
      "            PA4802.1  tRNA-Sec [KO:K14238]\n",
      "line in gene PA4802.1  tRNA-Sec [KO:K14238]\n",
      "            PA3134  gltX; glutamate--tRNA ligase [KO:K01885] [EC:6.1.1.17]\n",
      "line in gene PA3134  gltX; glutamate--tRNA ligase [KO:K01885] [EC:6.1.1.17]\n",
      "            PA4483  gatA; aspartyl/glutamyl-tRNA amidotransferase subunit A [KO:K02433] [EC:6.3.5.6 6.3.5.7]\n",
      "line in gene PA4483  gatA; aspartyl/glutamyl-tRNA amidotransferase subunit A [KO:K02433] [EC:6.3.5.6 6.3.5.7]\n",
      "            PA4484  gatB; aspartyl/glutamyl-tRNA amidotransferase subunit B [KO:K02434] [EC:6.3.5.6 6.3.5.7]\n",
      "line in gene PA4484  gatB; aspartyl/glutamyl-tRNA amidotransferase subunit B [KO:K02434] [EC:6.3.5.6 6.3.5.7]\n",
      "            PA4482  gatC; aspartyl/glutamyl-tRNA amidotransferase subunit C [KO:K02435] [EC:6.3.5.6 6.3.5.7]\n",
      "line in gene PA4482  gatC; aspartyl/glutamyl-tRNA amidotransferase subunit C [KO:K02435] [EC:6.3.5.6 6.3.5.7]\n",
      "            PA1794  glnS; glutamine--tRNA ligase [KO:K01886] [EC:6.1.1.18]\n",
      "line in gene PA1794  glnS; glutamine--tRNA ligase [KO:K01886] [EC:6.1.1.18]\n",
      "            PA0903  alaS; alanine--tRNA ligase [KO:K01872] [EC:6.1.1.7]\n",
      "line in gene PA0903  alaS; alanine--tRNA ligase [KO:K01872] [EC:6.1.1.7]\n",
      "            PA0963  aspS; aspartate--tRNA ligase [KO:K01876] [EC:6.1.1.12]\n",
      "line in gene PA0963  aspS; aspartate--tRNA ligase [KO:K01876] [EC:6.1.1.12]\n",
      "            PA0009  glyQ; glycine--tRNA ligase subunit alpha [KO:K01878] [EC:6.1.1.14]\n",
      "line in gene PA0009  glyQ; glycine--tRNA ligase subunit alpha [KO:K01878] [EC:6.1.1.14]\n",
      "            PA0008  glyS; glycine--tRNA ligase subunit beta [KO:K01879] [EC:6.1.1.14]\n",
      "line in gene PA0008  glyS; glycine--tRNA ligase subunit beta [KO:K01879] [EC:6.1.1.14]\n",
      "            PA2744  thrS; threonine--tRNA ligase [KO:K01868] [EC:6.1.1.3]\n",
      "line in gene PA2744  thrS; threonine--tRNA ligase [KO:K01868] [EC:6.1.1.3]\n",
      "            PA2612  serS; serine--tRNA ligase [KO:K01875] [EC:6.1.1.11]\n",
      "line in gene PA2612  serS; serine--tRNA ligase [KO:K01875] [EC:6.1.1.11]\n",
      "            PA4808  selA; selenocysteine synthase [KO:K01042] [EC:2.9.1.1]\n",
      "line in gene PA4808  selA; selenocysteine synthase [KO:K01042] [EC:2.9.1.1]\n",
      "            PA1795  cysS; cysteine--tRNA ligase [KO:K01883] [EC:6.1.1.16]\n",
      "line in gene PA1795  cysS; cysteine--tRNA ligase [KO:K01883] [EC:6.1.1.16]\n",
      "            PA3482  metG; methionine--tRNA ligase [KO:K01874] [EC:6.1.1.10]\n",
      "line in gene PA3482  metG; methionine--tRNA ligase [KO:K01874] [EC:6.1.1.10]\n",
      "            PA0018  fmt; methionyl-tRNA formyltransferase [KO:K00604] [EC:2.1.2.9]\n",
      "line in gene PA0018  fmt; methionyl-tRNA formyltransferase [KO:K00604] [EC:2.1.2.9]\n",
      "            PA3834  valS; valine--tRNA ligase [KO:K01873] [EC:6.1.1.9]\n",
      "line in gene PA3834  valS; valine--tRNA ligase [KO:K01873] [EC:6.1.1.9]\n",
      "            PA3987  leuS; leucine--tRNA ligase [KO:K01869] [EC:6.1.1.4]\n",
      "line in gene PA3987  leuS; leucine--tRNA ligase [KO:K01869] [EC:6.1.1.4]\n",
      "            PA4560  ileS; isoleucine--tRNA ligase [KO:K01870] [EC:6.1.1.5]\n",
      "line in gene PA4560  ileS; isoleucine--tRNA ligase [KO:K01870] [EC:6.1.1.5]\n",
      "            PA3700  lysS; lysine--tRNA ligase [KO:K04567] [EC:6.1.1.6]\n",
      "line in gene PA3700  lysS; lysine--tRNA ligase [KO:K04567] [EC:6.1.1.6]\n",
      "            PA5051  argS; arginine--tRNA ligase [KO:K01887] [EC:6.1.1.19]\n",
      "line in gene PA5051  argS; arginine--tRNA ligase [KO:K01887] [EC:6.1.1.19]\n",
      "            PA0956  proS; proline--tRNA ligase [KO:K01881] [EC:6.1.1.15]\n",
      "line in gene PA0956  proS; proline--tRNA ligase [KO:K01881] [EC:6.1.1.15]\n",
      "            PA3802  hisS; histidine--tRNA ligase [KO:K01892] [EC:6.1.1.21]\n",
      "line in gene PA3802  hisS; histidine--tRNA ligase [KO:K01892] [EC:6.1.1.21]\n",
      "            PA2740  pheS; phenylalanine--tRNA ligase subunit alpha [KO:K01889] [EC:6.1.1.20]\n",
      "line in gene PA2740  pheS; phenylalanine--tRNA ligase subunit alpha [KO:K01889] [EC:6.1.1.20]\n",
      "            PA2739  pheT; phenylalanine--tRNA ligase subunit beta [KO:K01890] [EC:6.1.1.20]\n",
      "line in gene PA2739  pheT; phenylalanine--tRNA ligase subunit beta [KO:K01890] [EC:6.1.1.20]\n",
      "            PA0668  tyrZ; tyrosine--tRNA ligase [KO:K01866] [EC:6.1.1.1]\n",
      "line in gene PA0668  tyrZ; tyrosine--tRNA ligase [KO:K01866] [EC:6.1.1.1]\n",
      "            PA4138  tyrS; tyrosine--tRNA ligase [KO:K01866] [EC:6.1.1.1]\n",
      "line in gene PA4138  tyrS; tyrosine--tRNA ligase [KO:K01866] [EC:6.1.1.1]\n",
      "            PA4439  trpS; tryptophan--tRNA ligase [KO:K01867] [EC:6.1.1.2]\n",
      "line in gene PA4439  trpS; tryptophan--tRNA ligase [KO:K01867] [EC:6.1.1.2]\n",
      "COMPOUND    C00025  L-Glutamate\n",
      "            C00037  Glycine\n",
      "            C00041  L-Alanine\n",
      "            C00047  L-Lysine\n",
      "            C00049  L-Aspartate\n",
      "            C00062  L-Arginine\n",
      "            C00064  L-Glutamine\n",
      "            C00065  L-Serine\n",
      "            C00073  L-Methionine\n",
      "            C00078  L-Tryptophan\n",
      "            C00079  L-Phenylalanine\n",
      "            C00082  L-Tyrosine\n",
      "            C00097  L-Cysteine\n",
      "            C00101  Tetrahydrofolate\n",
      "            C00123  L-Leucine\n",
      "            C00135  L-Histidine\n",
      "            C00148  L-Proline\n",
      "            C00152  L-Asparagine\n",
      "            C00183  L-Valine\n",
      "            C00188  L-Threonine\n",
      "            C00234  10-Formyltetrahydrofolate\n",
      "            C00407  L-Isoleucine\n",
      "            C00886  L-Alanyl-tRNA\n",
      "            C01005  O-Phospho-L-serine\n",
      "            C01931  L-Lysyl-tRNA\n",
      "            C02047  L-Leucyl-tRNA\n",
      "            C02163  L-Arginyl-tRNA(Arg)\n",
      "            C02282  Glutaminyl-tRNA\n",
      "            C02412  Glycyl-tRNA(Gly)\n",
      "            C02430  L-Methionyl-tRNA\n",
      "            C02553  L-Seryl-tRNA(Ser)\n",
      "            C02554  L-Valyl-tRNA(Val)\n",
      "            C02702  L-Prolyl-tRNA(Pro)\n",
      "            C02839  L-Tyrosyl-tRNA(Tyr)\n",
      "            C02984  L-Aspartyl-tRNA(Asp)\n",
      "            C02987  L-Glutamyl-tRNA(Glu)\n",
      "            C02988  L-Histidyl-tRNA(His)\n",
      "            C02992  L-Threonyl-tRNA(Thr)\n",
      "            C03125  L-Cysteinyl-tRNA(Cys)\n",
      "            C03127  L-Isoleucyl-tRNA(Ile)\n",
      "            C03294  N-Formylmethionyl-tRNA\n",
      "            C03402  L-Asparaginyl-tRNA(Asn)\n",
      "            C03511  L-Phenylalanyl-tRNA(Phe)\n",
      "            C03512  L-Tryptophanyl-tRNA(Trp)\n",
      "            C06112  L-Glutamyl-tRNA(Gln)\n",
      "            C06113  L-Aspartyl-tRNA(Asn)\n",
      "            C06481  L-Seryl-tRNA(Sec)\n",
      "            C06482  L-Selenocysteinyl-tRNA(Sec)\n",
      "            C16138  L-Pyrrolysine\n",
      "            C16638  O-Phosphoseryl-tRNA(Sec)\n",
      "            C17022  O-Phosphoseryl-tRNA(Cys)\n",
      "            C17027  L-Pyrrolysyl-tRNA(Pyl)\n",
      "REFERENCE   PMID:18252769\n",
      "  AUTHORS   Sheppard K, Yuan J, Hohn MJ, Jester B, Devine KM, Soll D\n",
      "  TITLE     From one amino acid to another: tRNA-dependent amino acid biosynthesis.\n",
      "  JOURNAL   Nucleic Acids Res 36:1813-25 (2008)\n",
      "            DOI:10.1093/nar/gkn015\n",
      "REFERENCE   PMID:15314242\n",
      "  AUTHORS   Polycarpo C, Ambrogelly A, Berube A, Winbush SM, McCloskey JA, Crain PF, Wood JL, Soll D\n",
      "  TITLE     An aminoacyl-tRNA synthetase that specifically activates pyrrolysine.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 101:12450-4 (2004)\n",
      "            DOI:10.1073/pnas.0405362101\n",
      "REL_PATHWAY pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00290  Valine, leucine and isoleucine biosynthesis\n",
      "            pae00300  Lysine biosynthesis\n",
      "            pae00330  Arginine and proline metabolism\n",
      "            pae00340  Histidine metabolism\n",
      "            pae00380  Tryptophan metabolism\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "KO_PATHWAY  ko00970\n",
      "///\n",
      "path:pae00997\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae00997                    Pathway\n",
      "NAME        Biosynthesis of various other secondary metabolites - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Biosynthesis of other secondary metabolites\n",
      "PATHWAY_MAP pae00997  Biosynthesis of various other secondary metabolites\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2254  pvcA; paerucumarin biosynthesis protein PvcA [KO:K22952] [EC:4.1.99.24]\n",
      "line in gene PA2254  pvcA; paerucumarin biosynthesis protein PvcA [KO:K22952] [EC:4.1.99.24]\n",
      "            PA2255  pvcB; paerucumarin biosynthesis protein PvcB [KO:K22874] [EC:1.14.20.9 1.14.20.10]\n",
      "line in gene PA2255  pvcB; paerucumarin biosynthesis protein PvcB [KO:K22874] [EC:1.14.20.9 1.14.20.10]\n",
      "            PA2256  pvcC; paerucumarin biosynthesis protein PvcC [KO:K25524] [EC:1.14.-.-]\n",
      "line in gene PA2256  pvcC; paerucumarin biosynthesis protein PvcC [KO:K25524] [EC:1.14.-.-]\n",
      "            PA2257  pvcD; paerucumarin biosynthesis protein PvcD [KO:K25525]\n",
      "line in gene PA2257  pvcD; paerucumarin biosynthesis protein PvcD [KO:K25525]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00024  Acetyl-CoA\n",
      "            C00025  L-Glutamate\n",
      "            C00026  2-Oxoglutarate\n",
      "            C00036  Oxaloacetate\n",
      "            C00041  L-Alanine\n",
      "            C00047  L-Lysine\n",
      "            C00049  L-Aspartate\n",
      "            C00065  L-Serine\n",
      "            C00077  L-Ornithine\n",
      "            C00078  L-Tryptophan\n",
      "            C00079  L-Phenylalanine\n",
      "            C00082  L-Tyrosine\n",
      "            C00083  Malonyl-CoA\n",
      "            C00108  Anthranilate\n",
      "            C00129  Isopentenyl diphosphate\n",
      "            C00197  3-Phospho-D-glycerate\n",
      "            C00251  Chorismate\n",
      "            C00279  D-Erythrose 4-phosphate\n",
      "            C00353  Geranylgeranyl diphosphate\n",
      "            C00441  L-Aspartate 4-semialdehyde\n",
      "            C00448  trans,trans-Farnesyl diphosphate\n",
      "            C00515  D-Ornithine\n",
      "            C01005  O-Phospho-L-serine\n",
      "            C01028  N6-Hydroxy-L-lysine\n",
      "            C03401  L-2,3-Diaminopropanoate\n",
      "            C03955  N6-Acetyl-N6-hydroxy-L-lysine\n",
      "            C04075  2-Amino-4-chloro-4-pentenoic acid\n",
      "            C05345  beta-D-Fructose 6-phosphate\n",
      "            C05554  Aerobactin\n",
      "            C12115  3-Amino-4-hydroxybenzoate\n",
      "            C20333  N2-Citryl-N6-acetyl-N6-hydroxy-L-lysine\n",
      "            C20644  2-Amino-4,5-dihydroxy-6-oxo-7-(phosphooxy)heptanoate\n",
      "            C20799  Grixazone B\n",
      "            C20834  Cyclooctat-9-en-7-ol\n",
      "            C20851  Dihydromonacolin L acid\n",
      "            C20852  3alpha-Hydroxy-3,5-dihydromonacolin L acid\n",
      "            C20853  Monacolin L acid\n",
      "            C20854  Monacolin J acid\n",
      "            C21108  Dihydromonacolin L-[acyl-carrier protein]\n",
      "            C21129  (S)-2-Methylbutanoyl-[acp]\n",
      "            C21130  Lovastatin acid\n",
      "            C21559  N-[(2S)-2-Amino-2-carboxyethyl]-L-glutamate\n",
      "            C21889  (2S)-3-(4-Hydroxyphenyl)-2-isocyanopropanoate\n",
      "            C21891  (2E)-3-(4-Hydroxyphenyl)-2-isocyanoprop-2-enoate\n",
      "            C21978  Cyclooctat-9-ene-5,7-diol\n",
      "            C21979  Cyclooctatin\n",
      "            C22072  2-[(L-Alanin-3-ylcarbamoyl)methyl]-2-hydroxybutanedioate\n",
      "            C22073  2-[(2-Aminoethylcarbamoyl)methyl]-2-hydroxybutanedioate\n",
      "            C22074  2-[(L-Alanin-3-ylcarbamoyl)methyl]-3-(2-aminoethylcarbamoyl)-2-hydroxypropanoate\n",
      "            C22075  Staphyloferrin B\n",
      "            C22101  N5-Citryl-D-ornithine\n",
      "            C22102  Staphyloferrin A\n",
      "            C22137  4-Chloro-L-lysine\n",
      "            C22138  L-Propargylglycine\n",
      "            C22139  gamma-L-Glutamyl-L-propargylglycine\n",
      "            C22140  L-gamma-Glutamyl-(3R)-L-beta-ethynylserine\n",
      "            C22141  L-beta-Ethynylserine\n",
      "            C22142  Cyclo(L-Trp-L-Phe)\n",
      "            C22143  N-Methyl-cyclo(L-Trp-L-Phe)\n",
      "            C22144  (-)-Ditryptophenaline\n",
      "            C22145  Fumiquinazoline F\n",
      "            C22146  Fumiquinazoline F-indoline-2',3'-diol\n",
      "            C22147  Fumiquinazoline A\n",
      "            C22148  Fumiquinazoline C\n",
      "            C22149  Fumiquinazoline D\n",
      "            C22449  Paerucumarin\n",
      "REFERENCE   PMID:24677498\n",
      "  AUTHORS   Saruwatari T, Yagishita F, Mino T, Noguchi H, Hotta K, Watanabe K\n",
      "  TITLE     Cytochrome P450 as dimerization catalyst in diketopiperazine alkaloid biosynthesis.\n",
      "  JOURNAL   Chembiochem 15:656-9 (2014)\n",
      "            DOI:10.1002/cbic.201300751\n",
      "REFERENCE   PMID:24612080\n",
      "  AUTHORS   Lim FY, Ames B, Walsh CT, Keller NP\n",
      "  TITLE     Co-ordination between BrlA regulation and secretion of the oxidoreductase FmqD directs selective accumulation of fumiquinazoline C to conidial tissues in Aspergillus fumigatus.\n",
      "  JOURNAL   Cell Microbiol 16:1267-83 (2014)\n",
      "            DOI:10.1111/cmi.12284\n",
      "REFERENCE   PMID:18689486\n",
      "  AUTHORS   Clarke-Pearson MF, Brady SF\n",
      "  TITLE     Paerucumarin, a new metabolite produced by the pvc gene cluster from Pseudomonas  aeruginosa.\n",
      "  JOURNAL   J Bacteriol 190:6927-30 (2008)\n",
      "            DOI:10.1128/JB.00801-08\n",
      "REFERENCE   PMID:19775248\n",
      "  AUTHORS   Cheung J, Beasley FC, Liu S, Lajoie GA, Heinrichs DE\n",
      "  TITLE     Molecular characterization of staphyloferrin B biosynthesis in Staphylococcus aureus.\n",
      "  JOURNAL   Mol Microbiol 74:594-608 (2009)\n",
      "            DOI:10.1111/j.1365-2958.2009.06880.x\n",
      "REFERENCE   PMID:19635410\n",
      "  AUTHORS   Kim SY, Zhao P, Igarashi M, Sawa R, Tomita T, Nishiyama M, Kuzuyama T\n",
      "  TITLE     Cloning and heterologous expression of the cyclooctatin biosynthetic gene cluster afford a diterpene cyclase and two p450 hydroxylases.\n",
      "  JOURNAL   Chem Biol 16:736-43 (2009)\n",
      "            DOI:10.1016/j.chembiol.2009.06.007\n",
      "REFERENCE   PMID:23653178\n",
      "  AUTHORS   Xu W, Chooi YH, Choi JW, Li S, Vederas JC, Da Silva NA, Tang Y\n",
      "  TITLE     LovG: the thioesterase required for dihydromonacolin L release and lovastatin nonaketide synthase turnover in lovastatin biosynthesis.\n",
      "  JOURNAL   Angew Chem Int Ed Engl 52:6472-5 (2013)\n",
      "            DOI:10.1002/anie.201302406\n",
      "REFERENCE   PMID:17003031\n",
      "  AUTHORS   Suzuki H, Ohnishi Y, Furusho Y, Sakuda S, Horinouchi S\n",
      "  TITLE     Novel benzene ring biosynthesis from C(3) and C(4) primary metabolites by two enzymes.\n",
      "  JOURNAL   J Biol Chem 281:36944-51 (2006)\n",
      "            DOI:10.1074/jbc.M608103200\n",
      "REFERENCE   PMID:19138128\n",
      "  AUTHORS   Cotton JL, Tao J, Balibar CJ\n",
      "  TITLE     Identification and characterization of the Staphylococcus aureus gene cluster coding for staphyloferrin A.\n",
      "  JOURNAL   Biochemistry 48:1025-35 (2009)\n",
      "            DOI:10.1021/bi801844c\n",
      "REFERENCE   PMID:30867596\n",
      "  AUTHORS   Marchand JA, Neugebauer ME, Ing MC, Lin CI, Pelton JG, Chang MCY\n",
      "  TITLE     Discovery of a pathway for terminal-alkyne amino acid biosynthesis.\n",
      "  JOURNAL   Nature 567:420-424 (2019)\n",
      "            DOI:10.1038/s41586-019-1020-y\n",
      "REFERENCE   PMID:1393837\n",
      "  AUTHORS   Neilands JB\n",
      "  TITLE     Mechanism and regulation of synthesis of aerobactin in Escherichia coli K12 (pColV-K30).\n",
      "  JOURNAL   Can J Microbiol 38:728-33 (1992)\n",
      "            DOI:10.1139/m92-119\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "            pae00900  Terpenoid backbone biosynthesis\n",
      "KO_PATHWAY  ko00997\n",
      "///\n",
      "path:pae00999\n",
      "ENTRY       pae00999                    Pathway\n",
      "NAME        Biosynthesis of various plant secondary metabolites - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Biosynthesis of other secondary metabolites\n",
      "PATHWAY_MAP pae00999  Biosynthesis of various plant secondary metabolites\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0546  metK; S-adenosylmethionine synthetase [KO:K00789] [EC:2.5.1.6]\n",
      "line in gene PA0546  metK; S-adenosylmethionine synthetase [KO:K00789] [EC:2.5.1.6]\n",
      "            PA0025  aroE; shikimate dehydrogenase [KO:K00014] [EC:1.1.1.25]\n",
      "line in gene PA0025  aroE; shikimate dehydrogenase [KO:K00014] [EC:1.1.1.25]\n",
      "            PA0244  shikimate 5-dehydrogenase [KO:K00014] [EC:1.1.1.25]\n",
      "line in gene PA0244  shikimate 5-dehydrogenase [KO:K00014] [EC:1.1.1.25]\n",
      "            PA1726  bglX; beta-glucosidase [KO:K05349] [EC:3.2.1.21]\n",
      "line in gene PA1726  bglX; beta-glucosidase [KO:K05349] [EC:3.2.1.21]\n",
      "COMPOUND    C00019  S-Adenosyl-L-methionine\n",
      "            C00024  Acetyl-CoA\n",
      "            C00036  Oxaloacetate\n",
      "            C00049  L-Aspartate\n",
      "            C00073  L-Methionine\n",
      "            C00074  Phosphoenolpyruvate\n",
      "            C00079  L-Phenylalanine\n",
      "            C00082  L-Tyrosine\n",
      "            C00083  Malonyl-CoA\n",
      "            C00223  p-Coumaroyl-CoA\n",
      "            C00254  Prephenate\n",
      "            C00323  Caffeoyl-CoA\n",
      "            C00353  Geranylgeranyl diphosphate\n",
      "            C00406  Feruloyl-CoA\n",
      "            C00423  trans-Cinnamate\n",
      "            C00448  trans,trans-Farnesyl diphosphate\n",
      "            C00493  Shikimate\n",
      "            C00590  Coniferyl alcohol\n",
      "            C00758  Bergaptol\n",
      "            C00841  Xanthotoxol\n",
      "            C01158  1-O-Galloyl-beta-D-glucose\n",
      "            C01198  3-(2-Hydroxyphenyl)propanoate\n",
      "            C01424  Gallate\n",
      "            C01527  Scopolin\n",
      "            C01557  Bergapten\n",
      "            C01752  Scopoletin\n",
      "            C01772  trans-2-Hydroxycinnamate\n",
      "            C01864  Xanthotoxin\n",
      "            C02162  Isopimpinellin\n",
      "            C02274  Dihydrocoumarin\n",
      "            C02637  3-Dehydroshikimate\n",
      "            C02951  5-Hydroxyxanthotoxin\n",
      "            C04576  Pentagalloylglucose\n",
      "            C05158  trans-beta-D-Glucosyl-2-hydroxycinnamate\n",
      "            C05324  Nicotianamine\n",
      "            C05366  (+)-Pinoresinol\n",
      "            C05838  cis-2-Hydroxycinnamate\n",
      "            C05839  cis-beta-D-Glucosyl-2-hydroxycinnamate\n",
      "            C05851  Coumarin\n",
      "            C06098  Zeaxanthin\n",
      "            C06972  Dronabinol\n",
      "            C07578  Cannabidiol\n",
      "            C08588  Crocetin\n",
      "            C08589  Crocin\n",
      "            C09060  Angelicin\n",
      "            C09263  Esculetin\n",
      "            C09265  Fraxetin\n",
      "            C09276  Marmesin\n",
      "            C09305  Psoralen\n",
      "            C09311  Scoparone\n",
      "            C09315  Umbelliferone\n",
      "            C10556  Deoxypodophyllotoxin\n",
      "            C10557  Dihydroanhydropodorhizol\n",
      "            C10646  Lariciresinol\n",
      "            C10682  Matairesinol\n",
      "            C10784  Cannabidiolic acid\n",
      "            C10874  Podophyllotoxin\n",
      "            C15485  2'-Deoxymugineic acid\n",
      "            C15486  3''-Deamino-3''-oxonicotianamine\n",
      "            C15500  Mugineic acid\n",
      "            C15501  3-Epihydroxymugineic acid\n",
      "            C15502  3-Epihydroxy-2'-deoxymugineic acid\n",
      "            C17055  Picrocrocin\n",
      "            C17062  Safranal\n",
      "            C17513  trans-Crocetin (beta-D-glucosyl) (beta-D-gentibiosyl) ester\n",
      "            C18079  Isoscopoletin\n",
      "            C18080  Osthenol\n",
      "            C18083  Demethylsuberosin\n",
      "            C18167  Secoisolariciresinol\n",
      "            C19707  (+)-Columbianetin\n",
      "            C19730  Crocetin dialdehyde\n",
      "            C19731  (3S)-3-Hydroxycyclocitral\n",
      "            C19867  beta-D-Glucosyl crocetin\n",
      "            C19868  Bis(beta-D-glucosyl) crocetin\n",
      "            C19869  beta-D-Gentiobiosyl crocetin\n",
      "            C20406  Cannabigerolate\n",
      "            C20413  3,5,7-Trioxododecanoyl-CoA\n",
      "            C20417  2,4-Dihydroxy-6-pentylbenzoate\n",
      "            C20454  (-)-Lariciresinol\n",
      "            C20455  (-)-Pinoresinol\n",
      "            C20456  (+)-Secoisolariciresinol\n",
      "            C20894  beta-Citraurin\n",
      "            C21183  Bursehernin\n",
      "            C21184  (-)-5'-Demethylyatein\n",
      "            C21191  Pluviatolide\n",
      "            C21459  6-Hydroxyferuloyl-CoA\n",
      "            C21903  Delta9-Tetrahydrocannabinolate\n",
      "            C22007  Sideretin\n",
      "            C22070  2,4-Dihydroxycinnamoyl-CoA\n",
      "            C22152  Bergamottin\n",
      "            C22153  Epoxybergamottin\n",
      "            C22156  6',7'-Dihydroxybergamottin\n",
      "            C22438  3,5-Dehydroshikimate\n",
      "            C22444  5,8-Dihydroxypsoralen\n",
      "            C22445  8-Hydroxybergapten\n",
      "            C22452  6,7,8-Trihydroxycoumarin\n",
      "REFERENCE   PMID:22569263\n",
      "  AUTHORS   Nagatoshi M, Terasaka K, Owaki M, Sota M, Inukai T, Nagatsu A, Mizukami H\n",
      "  TITLE     UGT75L6 and UGT94E5 mediate sequential glucosylation of crocetin to crocin in Gardenia jasminoides.\n",
      "  JOURNAL   FEBS Lett 586:1055-61 (2012)\n",
      "            DOI:10.1016/j.febslet.2012.03.003\n",
      "REFERENCE   PMID:17544411\n",
      "  AUTHORS   Taura F, Sirikantaramas S, Shoyama Y, Yoshikai K, Shoyama Y, Morimoto S\n",
      "  TITLE     Cannabidiolic-acid synthase, the chemotype-determining enzyme in the fiber-type Cannabis sativa.\n",
      "  JOURNAL   FEBS Lett 581:2929-34 (2007)\n",
      "            DOI:10.1016/j.febslet.2007.05.043\n",
      "REFERENCE   PMID:9952442\n",
      "  AUTHORS   Higuchi K, Suzuki K, Nakanishi H, Yamaguchi H, Nishizawa NK, Mori S\n",
      "  TITLE     Cloning of nicotianamine synthase genes, novel genes involved in the biosynthesis of phytosiderophores.\n",
      "  JOURNAL   Plant Physiol 119:471-80 (1999)\n",
      "            DOI:10.1104/pp.119.2.471\n",
      "REFERENCE   PMID:7622460\n",
      "  AUTHORS   Ma JF, Shinada T, Matsuda C, Nomoto K\n",
      "  TITLE     Biosynthesis of phytosiderophores, mugineic acids, associated with methionine cycling.\n",
      "  JOURNAL   J Biol Chem 270:16549-54 (1995)\n",
      "            DOI:10.1074/jbc.270.28.16549\n",
      "REFERENCE   PMID:33346890\n",
      "  AUTHORS   Tahara K, Nishiguchi M, Funke E, Miyazawa SI, Miyama T, Milkowski C\n",
      "  TITLE     Dehydroquinate dehydratase/shikimate dehydrogenases involved in gallate biosynthesis of the aluminum-tolerant tree species Eucalyptus camaldulensis.\n",
      "  JOURNAL   Planta 253:3 (2020)\n",
      "            DOI:10.1007/s00425-020-03516-w\n",
      "REFERENCE   PMID:26558757\n",
      "  AUTHORS   Dugrand-Judek A, Olry A, Hehn A, Costantino G, Ollitrault P, Froelicher Y, Bourgaud F.\n",
      "  TITLE     The Distribution of Coumarins and Furanocoumarins in Citrus Species Closely Matches Citrus Phylogeny and Reflects the Organization of Biosynthetic Pathways.\n",
      "  JOURNAL   PLoS One 10:e0142757 (2015)\n",
      "            DOI:10.1371/journal.pone.0142757\n",
      "REFERENCE   PMID:15009205\n",
      "  AUTHORS   Hehmann M, Lukacin R, Ekiert H, Matern U\n",
      "  TITLE     Furanocoumarin biosynthesis in Ammi majus L. Cloning of bergaptol O-methyltransferase.\n",
      "  JOURNAL   Eur J Biochem 271:932-40 (2004)\n",
      "            DOI:10.1111/j.1432-1033.2004.03995.x\n",
      "REFERENCE   PMID:26359402\n",
      "  AUTHORS   Lau W, Sattely ES\n",
      "  TITLE     Six enzymes from mayapple that complete the biosynthetic pathway to the etoposide aglycone.\n",
      "  JOURNAL   Science 349:1224-8 (2015)\n",
      "            DOI:10.1126/science.aac7202\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "            pae00900  Terpenoid backbone biosynthesis\n",
      "KO_PATHWAY  ko00999\n",
      "///\n",
      "path:pae01040\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae01040                    Pathway\n",
      "NAME        Biosynthesis of unsaturated fatty acids - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Lipid metabolism\n",
      "PATHWAY_MAP pae01040  Biosynthesis of unsaturated fatty acids\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0286  desA; delta-9 fatty acid desaturase DesA [KO:K00507] [EC:1.14.19.1]\n",
      "line in gene PA0286  desA; delta-9 fatty acid desaturase DesA [KO:K00507] [EC:1.14.19.1]\n",
      "            PA4888  desB; acyl-CoA desaturase [KO:K22769] [EC:1.14.19.-]\n",
      "line in gene PA4888  desB; acyl-CoA desaturase [KO:K22769] [EC:1.14.19.-]\n",
      "            PA4889  oxidoreductase [KO:K22770]\n",
      "line in gene PA4889  oxidoreductase [KO:K22770]\n",
      "            PA2856  tesA; esterase TesA [KO:K10804] [EC:3.1.2.- 3.1.2.2 3.1.1.2 3.1.1.5]\n",
      "line in gene PA2856  tesA; esterase TesA [KO:K10804] [EC:3.1.2.- 3.1.2.2 3.1.1.2 3.1.1.5]\n",
      "            PA3942  tesB; acyl-CoA thioesterase II [KO:K10805] [EC:3.1.2.-]\n",
      "line in gene PA3942  tesB; acyl-CoA thioesterase II [KO:K10805] [EC:3.1.2.-]\n",
      "            PA5371  acyl-CoA thioesterase [KO:K10806] [EC:3.1.2.-]\n",
      "line in gene PA5371  acyl-CoA thioesterase [KO:K10806] [EC:3.1.2.-]\n",
      "COMPOUND    C00154  Palmitoyl-CoA\n",
      "            C00219  Arachidonate\n",
      "            C00249  Hexadecanoic acid\n",
      "            C00412  Stearoyl-CoA\n",
      "            C00510  Oleoyl-CoA\n",
      "            C00712  (9Z)-Octadecenoic acid\n",
      "            C01203  Oleoyl-[acyl-carrier protein]\n",
      "            C01530  Octadecanoic acid\n",
      "            C01595  Linoleate\n",
      "            C02041  Eicosanoyl-CoA\n",
      "            C02050  Linoleoyl-CoA\n",
      "            C02249  Arachidonyl-CoA\n",
      "            C03035  gamma-Linolenoyl-CoA\n",
      "            C03242  Dihomo-gamma-linolenate\n",
      "            C03595  8,11,14-Eicosatrienoyl-CoA\n",
      "            C04088  Octadecanoyl-[acyl-carrier protein]\n",
      "            C05764  Hexadecanoyl-[acp]\n",
      "            C06425  Icosanoic acid\n",
      "            C06426  (6Z,9Z,12Z)-Octadecatrienoic acid\n",
      "            C06427  (9Z,12Z,15Z)-Octadecatrienoic acid\n",
      "            C06428  (5Z,8Z,11Z,14Z,17Z)-Icosapentaenoic acid\n",
      "            C06429  (4Z,7Z,10Z,13Z,16Z,19Z)-Docosahexaenoic acid\n",
      "            C08281  Docosanoic acid\n",
      "            C08316  Erucic acid\n",
      "            C08320  Tetracosanoic acid\n",
      "            C08323  (15Z)-Tetracosenoic acid\n",
      "            C16160  Linoleoyl-[acyl-carrier protein]\n",
      "            C16161  alpha-Linolenoyl-ACP\n",
      "            C16162  (9Z,12Z,15Z)-Octadecatrienoyl-CoA\n",
      "            C16163  (6Z,9Z,12Z,15Z)-Octadecatetraenoyl-CoA\n",
      "            C16164  (8Z,11Z,14Z,17Z)-Icosatetraenoyl-CoA\n",
      "            C16165  (5Z,8Z,11Z,14Z,17Z)-Icosapentaenoyl-CoA\n",
      "            C16166  (7Z,10Z,13Z,16Z,19Z)-Docosapentaenoyl-CoA\n",
      "            C16167  (9Z,12Z,15Z,18Z,21Z)-Tetracosapentaenoyl-CoA\n",
      "            C16168  (6Z,9Z,12Z,15Z,18Z,21Z)-Tetracosahexaenoyl-CoA\n",
      "            C16169  (4Z,7Z,10Z,13Z,16Z,19Z)-Docosahexaenoyl-CoA\n",
      "            C16170  (7Z,10Z,13Z,16Z)-Docosatetraenoyl-CoA\n",
      "            C16171  (9Z,12Z,15Z,18Z)-Tetracosatetraenoyl-CoA\n",
      "            C16172  (6Z,9Z,12Z,15Z,18Z)-Tetracosapentaenoyl-CoA\n",
      "            C16173  (4Z,7Z,10Z,13Z,16Z)-Docosapentaenoyl-CoA\n",
      "            C16179  (11Z,14Z,17Z)-Icosatrienoyl-CoA\n",
      "            C16180  (11Z,14Z)-Icosadienoyl-CoA\n",
      "            C16513  (7Z,10Z,13Z,16Z,19Z)-Docosapentaenoic acid\n",
      "            C16522  Icosatrienoic acid\n",
      "            C16525  Icosadienoic acid\n",
      "            C16526  Icosenoic acid\n",
      "            C16527  Adrenic acid\n",
      "            C16528  Docosanoyl-CoA\n",
      "            C16529  Tetracosanoyl-CoA\n",
      "            C16530  Icosenoyl-CoA\n",
      "            C16531  Docosenoyl-CoA\n",
      "            C16532  Tetracosenoyl-CoA\n",
      "            C16533  (13Z,16Z)-Docosadienoic acid\n",
      "            C16645  (13Z,16Z)-Docosadienoyl-CoA\n",
      "            C17279  Ximenoyl-CoA\n",
      "            C21072  Palmitoleoyl-CoA\n",
      "            C21932  Cerotoyl-CoA\n",
      "            C21934  Montanoyl-CoA\n",
      "            C21935  (6Z,9Z)-Octadecadienoyl-CoA\n",
      "            C21937  (8Z,11Z)-Icosadienoyl-CoA\n",
      "            C21939  (5Z,8Z,11Z)-Icosatrienoyl-CoA\n",
      "            C21941  (7Z,10Z,13Z)-Docosatrienoyl-CoA\n",
      "            C21943  Sapienoyl-CoA\n",
      "            C21945  cis-Vaccenoyl-CoA\n",
      "            C21947  Paullinoyl-CoA\n",
      "            C21949  (11Z,14Z,17Z,20Z,23Z)-Hexacosapentaenoyl-CoA\n",
      "            C21951  (13Z,16Z,19Z,22Z,25Z)-Octacosapentaenoyl-CoA\n",
      "            C21953  (8Z,11Z,14Z,17Z,20Z,23Z)-Hexacosahexaenoyl-CoA\n",
      "            C21959  (10Z,13Z,16Z,19Z,22Z,25Z)-Octacosahexaenoyl-CoA\n",
      "            C21961  (11Z,14Z,17Z,20Z)-Hexacosatetraenoyl-CoA\n",
      "            C21963  (13Z,16Z,19Z,22Z)-Octacosatetraenoyl-CoA\n",
      "            C21965  (8Z,11Z,14Z,17Z,20Z)-Hexacosapentaenoyl-CoA\n",
      "            C21967  (10Z,13Z,16Z,19Z,22Z)-Octacosapentaenoyl-CoA\n",
      "            C21981  (6Z,9Z)-Hexadecadienoyl-CoA\n",
      "REFERENCE   PMID:16564093\n",
      "  AUTHORS   Jakobsson A, Westerberg R, Jacobsson A.\n",
      "  TITLE     Fatty acid elongases in mammals: their regulation and roles in metabolism.\n",
      "  JOURNAL   Prog Lipid Res 45:237-49 (2006)\n",
      "            DOI:10.1016/j.plipres.2006.01.004\n",
      "REFERENCE   PMID:12482854\n",
      "  AUTHORS   Moon YA, Horton JD.\n",
      "  TITLE     Identification of two mammalian reductases involved in the two-carbon fatty acyl elongation cascade.\n",
      "  JOURNAL   J Biol Chem 278:7335-43 (2003)\n",
      "            DOI:10.1074/jbc.M211684200\n",
      "REFERENCE   PMID:12087109\n",
      "  AUTHORS   Han G, Gable K, Kohlwein SD, Beaudoin F, Napier JA, Dunn TM.\n",
      "  TITLE     The Saccharomyces cerevisiae YBR159w gene encodes the 3-ketoreductase of the microsomal fatty acid elongase.\n",
      "  JOURNAL   J Biol Chem 277:35440-9 (2002)\n",
      "            DOI:10.1074/jbc.M205620200\n",
      "REFERENCE   PMID:11500517\n",
      "  AUTHORS   Su HM, Moser AB, Moser HW, Watkins PA.\n",
      "  TITLE     Peroxisomal straight-chain Acyl-CoA oxidase and D-bifunctional protein are essential for the retroconversion step in docosahexaenoic acid synthesis.\n",
      "  JOURNAL   J Biol Chem 276:38115-20 (2001)\n",
      "            DOI:10.1074/jbc.M106326200\n",
      "REFERENCE   PMID:16766224\n",
      "  AUTHORS   Huyghe S, Mannaerts GP, Baes M, Van Veldhoven PP.\n",
      "  TITLE     Peroxisomal multifunctional protein-2: the enzyme, the patients and the knockout mouse model.\n",
      "  JOURNAL   Biochim Biophys Acta 1761:973-94 (2006)\n",
      "            DOI:10.1016/j.bbalip.2006.04.006\n",
      "REFERENCE   PMID:17719544\n",
      "  AUTHORS   Denic V, Weissman JS\n",
      "  TITLE     A molecular caliper mechanism for determining very long-chain fatty acid length.\n",
      "  JOURNAL   Cell 130:663-77 (2007)\n",
      "            DOI:10.1016/j.cell.2007.06.031\n",
      "REFERENCE   PMID:24184513\n",
      "  AUTHORS   Ichi I, Kono N, Arita Y, Haga S, Arisawa K, Yamano M, Nagase M, Fujiwara Y, Arai H\n",
      "  TITLE     Identification of genes and pathways involved in the synthesis of Mead acid (20:3n-9), an indicator of essential fatty acid deficiency.\n",
      "  JOURNAL   Biochim Biophys Acta 1841:204-13 (2014)\n",
      "            DOI:10.1016/j.bbalip.2013.10.013\n",
      "REFERENCE   PMID:20018209\n",
      "  AUTHORS   Guillou H, Zadravec D, Martin PG, Jacobsson A\n",
      "  TITLE     The key roles of elongases and desaturases in mammalian fatty acid metabolism: Insights from transgenic mice.\n",
      "  JOURNAL   Prog Lipid Res 49:186-99 (2010)\n",
      "            DOI:10.1016/j.plipres.2009.12.002\n",
      "REFERENCE   PMID:22158834\n",
      "  AUTHORS   Yu M, Benham A, Logan S, Brush RS, Mandal MN, Anderson RE, Agbaga MP\n",
      "  TITLE     ELOVL4 protein preferentially elongates 20:5n3 to very long chain PUFAs over 20:4n6 and 22:6n3.\n",
      "  JOURNAL   J Lipid Res 53:494-504 (2012)\n",
      "            DOI:10.1194/jlr.M021386\n",
      "REFERENCE   PMID:28784662\n",
      "  AUTHORS   Sawai M, Uchida Y, Ohno Y, Miyamoto M, Nishioka C, Itohara S, Sassa T, Kihara A\n",
      "  TITLE     The 3-hydroxyacyl-CoA dehydratases HACD1 and HACD2 exhibit functional redundancy and are active in a wide range of fatty acid elongation pathways.\n",
      "  JOURNAL   J Biol Chem 292:15538-15551 (2017)\n",
      "            DOI:10.1074/jbc.M117.803171\n",
      "REFERENCE   PMID:20937905\n",
      "  AUTHORS   Ohno Y, Suto S, Yamanaka M, Mizutani Y, Mitsutake S, Igarashi Y, Sassa T, Kihara A\n",
      "  TITLE     ELOVL1 production of C24 acyl-CoAs is linked to C24 sphingolipid synthesis.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 107:18439-44 (2010)\n",
      "            DOI:10.1073/pnas.1005572107\n",
      "REFERENCE   PMID:21959040\n",
      "  AUTHORS   Naganuma T, Sato Y, Sassa T, Ohno Y, Kihara A\n",
      "  TITLE     Biochemical characterization of the very long-chain fatty acid elongase ELOVL7.\n",
      "  JOURNAL   FEBS Lett 585:3337-41 (2011)\n",
      "            DOI:10.1016/j.febslet.2011.09.024\n",
      "REFERENCE   PMID:19826053\n",
      "  AUTHORS   Tamura K, Makino A, Hullin-Matsuda F, Kobayashi T, Furihata M, Chung S, Ashida S, Miki T, Fujioka T, Shuin T, Nakamura Y, Nakagawa H\n",
      "  TITLE     Novel lipogenic enzyme ELOVL7 is involved in prostate cancer growth through saturated long-chain fatty acid metabolism.\n",
      "  JOURNAL   Cancer Res 69:8133-40 (2009)\n",
      "            DOI:10.1158/0008-5472.CAN-09-0775\n",
      "REFERENCE   PMID:22562166\n",
      "  AUTHORS   Uemura H\n",
      "  TITLE     Synthesis and production of unsaturated and polyunsaturated fatty acids in yeast: current state and perspectives.\n",
      "  JOURNAL   Appl Microbiol Biotechnol 95:1-12 (2012)\n",
      "            DOI:10.1007/s00253-012-4105-1\n",
      "REFERENCE   PMID:11734571\n",
      "  AUTHORS   Ferdinandusse S, Denis S, Mooijer PA, Zhang Z, Reddy JK, Spector AA, Wanders RJ\n",
      "  TITLE     Identification of the peroxisomal beta-oxidation enzymes involved in the biosynthesis of docosahexaenoic acid.\n",
      "  JOURNAL   J Lipid Res 42:1987-95 (2001)\n",
      "KO_PATHWAY  ko01040\n",
      "///\n",
      "path:pae01053\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae01053                    Pathway\n",
      "NAME        Biosynthesis of siderophore group nonribosomal peptides - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Metabolism; Metabolism of terpenoids and polyketides\n",
      "PATHWAY_MAP pae01053  Biosynthesis of siderophore group nonribosomal peptides\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4231  pchA; salicylate biosynthesis isochorismate synthase [KO:K01851] [EC:5.4.4.2]\n",
      "line in gene PA4231  pchA; salicylate biosynthesis isochorismate synthase [KO:K01851] [EC:5.4.4.2]\n",
      "            PA1165  pcpS; 4'-phosphopantetheinyl transferase [KO:K02362] [EC:6.3.2.14 2.7.8.-]\n",
      "line in gene PA1165  pcpS; 4'-phosphopantetheinyl transferase [KO:K02362] [EC:6.3.2.14 2.7.8.-]\n",
      "            PA4230  pchB; isochorismate-pyruvate lyase [KO:K04782] [EC:4.2.99.21]\n",
      "line in gene PA4230  pchB; isochorismate-pyruvate lyase [KO:K04782] [EC:4.2.99.21]\n",
      "            PA4228  pchD; 2,3-dihydroxybenzoate-AMP ligase [KO:K12238] [EC:6.2.1.61]\n",
      "line in gene PA4228  pchD; 2,3-dihydroxybenzoate-AMP ligase [KO:K12238] [EC:6.2.1.61]\n",
      "            PA4226  pchE; dihydroaeruginoic acid synthetase [KO:K12239] [EC:6.2.1.69]\n",
      "line in gene PA4226  pchE; dihydroaeruginoic acid synthetase [KO:K12239] [EC:6.2.1.69]\n",
      "            PA4225  pchF; pyochelin synthetase [KO:K12240] [EC:6.2.1.69]\n",
      "line in gene PA4225  pchF; pyochelin synthetase [KO:K12240] [EC:6.2.1.69]\n",
      "            PA4224  pchG; pyochelin biosynthetic protein PchG [KO:K12241]\n",
      "line in gene PA4224  pchG; pyochelin biosynthetic protein PchG [KO:K12241]\n",
      "COMPOUND    C00158  Citrate\n",
      "            C00196  2,3-Dihydroxybenzoate\n",
      "            C00230  3,4-Dihydroxybenzoate\n",
      "            C00251  Chorismate\n",
      "            C00315  Spermidine\n",
      "            C00805  Salicylate\n",
      "            C00885  Isochorismate\n",
      "            C02637  3-Dehydroshikimate\n",
      "            C04171  (2S,3S)-2,3-Dihydro-2,3-dihydroxybenzoate\n",
      "            C05821  Enterochelin\n",
      "            C06769  Vibriobactin\n",
      "            C12037  Pyochelin\n",
      "            C12038  Yersiniabactin\n",
      "            C12216  Mycobactin S\n",
      "            C12219  Corynebactin\n",
      "            C12220  Myxochelin A\n",
      "            C12221  Myxochelin B\n",
      "            C20883  3,4-Dihydroxybenzoyl-[aryl-carrier protein]\n",
      "            C20884  (3,4-Dihydroxybenzoyl)adenylate\n",
      "            C20885  N-Citryl-spermidine\n",
      "            C20952  N1-(3,4-Dihydroxybenzoyl)-N8-citryl spermidine\n",
      "            C22004  N8,N'8-Citryl-bis(spermidine)\n",
      "            C22069  N1-(3,4-Dihydroxybenzoyl)-N8,N'8-citryl-bis(spermidine)\n",
      "            C22273  Petrobactin\n",
      "REFERENCE   PMID:12040125\n",
      "  AUTHORS   Crosa JH, Walsh CT.\n",
      "  TITLE     Genetics and assembly line enzymology of siderophore biosynthesis in bacteria.\n",
      "  JOURNAL   Microbiol Mol Biol Rev 66:223-49 (2002)\n",
      "            DOI:10.1128/MMBR.66.2.223-249.2002\n",
      "REFERENCE   PMID:9818149\n",
      "  AUTHORS   Gehring AM, DeMoll E, Fetherston JD, Mori I, Mayhew GF, Blattner FR, Walsh CT, Perry RD.\n",
      "  TITLE     Iron acquisition in plague: modular logic in enzymatic biogenesis of yersiniabactin by Yersinia pestis.\n",
      "  JOURNAL   Chem Biol 5:573-86 (1998)\n",
      "            DOI:10.1016/S1074-5521(98)90115-6\n",
      "REFERENCE   PMID:11029592\n",
      "  AUTHORS   Silakowski B, Kunze B, Nordsiek G, Blocker H, Hofle G, Muller R.\n",
      "  TITLE     The myxochelin iron transport regulon of the myxobacterium Stigmatella aurantiaca Sg a15.\n",
      "  JOURNAL   Eur J Biochem 267:6476-85 (2000)\n",
      "            DOI:10.1046/j.1432-1327.2000.01740.x\n",
      "REFERENCE   PMID:11112537\n",
      "  AUTHORS   Keating TA, Marshall CG, Walsh CT.\n",
      "  TITLE     Vibriobactin biosynthesis in Vibrio cholerae: VibH is an amide synthase homologous to nonribosomal peptide synthetase condensation domains.\n",
      "  JOURNAL   Biochemistry 39:15513-21 (2000)\n",
      "            DOI:10.1021/bi001651a\n",
      "REFERENCE   PMID:10692387\n",
      "  AUTHORS   Hantash FM, Earhart CF.\n",
      "  TITLE     Membrane association of the Escherichia coli enterobactin synthase proteins EntB/G, EntE, and EntF.\n",
      "  JOURNAL   J Bacteriol 182:1768-73 (2000)\n",
      "            DOI:10.1128/JB.182.6.1768-1773.2000\n",
      "REL_PATHWAY pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "KO_PATHWAY  ko01053\n",
      "///\n",
      "path:pae01100\n",
      "ENTRY       pae01100                    Pathway\n",
      "NAME        Metabolic pathways - Pseudomonas aeruginosa PAO1\n",
      "PATHWAY_MAP pae01100  Metabolic pathways\n",
      "MODULE      pae_M00002  Glycolysis, core module involving three-carbon compounds [PATH:pae01100]\n",
      "            pae_M00003  Gluconeogenesis, oxaloacetate => fructose-6P [PATH:pae01100]\n",
      "            pae_M00005  PRPP biosynthesis, ribose 5P => PRPP [PATH:pae01100]\n",
      "            pae_M00007  Pentose phosphate pathway, non-oxidative phase, fructose 6P => ribose 5P [PATH:pae01100]\n",
      "            pae_M00008  Entner-Doudoroff pathway, glucose-6P => glyceraldehyde-3P + pyruvate [PATH:pae01100]\n",
      "            pae_M00009  Citrate cycle (TCA cycle, Krebs cycle) [PATH:pae01100]\n",
      "            pae_M00010  Citrate cycle, first carbon oxidation, oxaloacetate => 2-oxoglutarate [PATH:pae01100]\n",
      "            pae_M00011  Citrate cycle, second carbon oxidation, 2-oxoglutarate => oxaloacetate [PATH:pae01100]\n",
      "            pae_M00015  Proline biosynthesis, glutamate => proline [PATH:pae01100]\n",
      "            pae_M00016  Lysine biosynthesis, succinyl-DAP pathway, aspartate => lysine [PATH:pae01100]\n",
      "            pae_M00018  Threonine biosynthesis, aspartate => homoserine => threonine [PATH:pae01100]\n",
      "            pae_M00019  Valine/isoleucine biosynthesis, pyruvate => valine / 2-oxobutanoate => isoleucine [PATH:pae01100]\n",
      "            pae_M00020  Serine biosynthesis, glycerate-3P => serine [PATH:pae01100]\n",
      "            pae_M00021  Cysteine biosynthesis, serine => cysteine [PATH:pae01100]\n",
      "            pae_M00023  Tryptophan biosynthesis, chorismate => tryptophan [PATH:pae01100]\n",
      "            pae_M00024  Phenylalanine biosynthesis, chorismate => phenylpyruvate => phenylalanine [PATH:pae01100]\n",
      "            pae_M00028  Ornithine biosynthesis, glutamate => ornithine [PATH:pae01100]\n",
      "            pae_M00034  Methionine salvage pathway [PATH:pae01100]\n",
      "            pae_M00036  Leucine degradation, leucine => acetoacetate + acetyl-CoA [PATH:pae01100]\n",
      "            pae_M00044  Tyrosine degradation, tyrosine => homogentisate [PATH:pae01100]\n",
      "            pae_M00045  Histidine degradation, histidine => N-formiminoglutamate => glutamate [PATH:pae01100]\n",
      "            pae_M00046  Pyrimidine degradation, uracil => beta-alanine, thymine => 3-aminoisobutanoate [PATH:pae01100]\n",
      "            pae_M00048  Inosine monophosphate biosynthesis, PRPP + glutamine => IMP [PATH:pae01100]\n",
      "            pae_M00049  Adenine ribonucleotide biosynthesis, IMP => ADP,ATP [PATH:pae01100]\n",
      "            pae_M00050  Guanine ribonucleotide biosynthesis, IMP => GDP,GTP [PATH:pae01100]\n",
      "            pae_M00052  Pyrimidine ribonucleotide biosynthesis, UMP => UDP/UTP,CDP/CTP [PATH:pae01100]\n",
      "            pae_M00053  Pyrimidine deoxyribonuleotide biosynthesis, CDP => dCTP [PATH:pae01100]\n",
      "            pae_M00063  CMP-KDO biosynthesis [PATH:pae01100]\n",
      "            pae_M00064  ADP-L-glycero-D-manno-heptose biosynthesis [PATH:pae01100]\n",
      "            pae_M00082  Fatty acid biosynthesis, initiation [PATH:pae01100]\n",
      "            pae_M00083  Fatty acid biosynthesis, elongation [PATH:pae01100]\n",
      "            pae_M00086  beta-Oxidation, acyl-CoA synthesis [PATH:pae01100]\n",
      "            pae_M00087  beta-Oxidation [PATH:pae01100]\n",
      "            pae_M00093  Phosphatidylethanolamine (PE) biosynthesis, PA => PS => PE [PATH:pae01100]\n",
      "            pae_M00115  NAD biosynthesis, aspartate => quinolinate => NAD [PATH:pae01100]\n",
      "            pae_M00118  Glutathione biosynthesis, glutamate => glutathione [PATH:pae01100]\n",
      "            pae_M00119  Pantothenate biosynthesis, valine/L-aspartate => pantothenate [PATH:pae01100]\n",
      "            pae_M00120  Coenzyme A biosynthesis, pantothenate => CoA [PATH:pae01100]\n",
      "            pae_M00121  Heme biosynthesis, plants and bacteria, glutamate => heme [PATH:pae01100]\n",
      "            pae_M00122  Cobalamin biosynthesis, cobyrinate a,c-diamide => cobalamin [PATH:pae01100]\n",
      "            pae_M00123  Biotin biosynthesis, pimeloyl-ACP/CoA => biotin [PATH:pae01100]\n",
      "            pae_M00124  Pyridoxal-P biosynthesis, erythrose-4P => pyridoxal-P [PATH:pae01100]\n",
      "            pae_M00125  Riboflavin biosynthesis, plants and bacteria, GTP => riboflavin/FMN/FAD [PATH:pae01100]\n",
      "            pae_M00126  Tetrahydrofolate biosynthesis, GTP => THF [PATH:pae01100]\n",
      "            pae_M00135  GABA biosynthesis, eukaryotes, putrescine => GABA [PATH:pae01100]\n",
      "            pae_M00176  Assimilatory sulfate reduction, sulfate => H2S [PATH:pae01100]\n",
      "            pae_M00307  Pyruvate oxidation, pyruvate => acetyl-CoA [PATH:pae01100]\n",
      "            pae_M00338  Cysteine biosynthesis, homocysteine + serine => cysteine [PATH:pae01100]\n",
      "            pae_M00432  Leucine biosynthesis, 2-oxoisovalerate => 2-oxoisocaproate [PATH:pae01100]\n",
      "            pae_M00529  Denitrification, nitrate => nitrogen [PATH:pae01100]\n",
      "            pae_M00530  Dissimilatory nitrate reduction, nitrate => ammonia [PATH:pae01100]\n",
      "            pae_M00533  Homoprotocatechuate degradation, homoprotocatechuate => 2-oxohept-3-enedioate [PATH:pae01100]\n",
      "            pae_M00549  Nucleotide sugar biosynthesis, glucose => UDP-glucose [PATH:pae01100]\n",
      "            pae_M00551  Benzoate degradation, benzoate => catechol / methylbenzoate => methylcatechol [PATH:pae01100]\n",
      "            pae_M00555  Betaine biosynthesis, choline => betaine [PATH:pae01100]\n",
      "            pae_M00568  Catechol ortho-cleavage, catechol => 3-oxoadipate [PATH:pae01100]\n",
      "            pae_M00570  Isoleucine biosynthesis, threonine => 2-oxobutanoate => isoleucine [PATH:pae01100]\n",
      "            pae_M00572  Pimeloyl-ACP biosynthesis, BioC-BioH pathway, malonyl-ACP => pimeloyl-ACP [PATH:pae01100]\n",
      "            pae_M00579  Phosphate acetyltransferase-acetate kinase pathway, acetyl-CoA => acetate [PATH:pae01100]\n",
      "            pae_M00637  Anthranilate degradation, anthranilate => catechol [PATH:pae01100]\n",
      "            pae_M00761  Undecaprenylphosphate alpha-L-Ara4N biosynthesis, UDP-GlcA => undecaprenyl phosphate alpha-L-Ara4N [PATH:pae01100]\n",
      "            pae_M00793  dTDP-L-rhamnose biosynthesis [PATH:pae01100]\n",
      "            pae_M00835  Pyocyanine biosynthesis, chorismate => pyocyanine [PATH:pae01100]\n",
      "            pae_M00844  Arginine biosynthesis, ornithine => arginine [PATH:pae01100]\n",
      "            pae_M00846  Siroheme biosynthesis, glutamyl-tRNA => siroheme [PATH:pae01100]\n",
      "            pae_M00879  Arginine succinyltransferase pathway, arginine => glutamate [PATH:pae01100]\n",
      "            pae_M00880  Molybdenum cofactor biosynthesis, GTP => molybdenum cofactor [PATH:pae01100]\n",
      "            pae_M00881  Lipoic acid biosynthesis, plants and bacteria, octanoyl-ACP => dihydrolipoyl-E2/H [PATH:pae01100]\n",
      "            pae_M00909  UDP-N-acetyl-D-glucosamine biosynthesis, prokaryotes, glucose => UDP-GlcNAc [PATH:pae01100]\n",
      "            pae_M00938  Pyrimidine deoxyribonuleotide biosynthesis, UDP => dTTP [PATH:pae01100]\n",
      "            pae_M00947  D-Arginine racemization, D-arginine => L-arginine [PATH:pae01100]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00030  Pentose phosphate pathway\n",
      "            pae00040  Pentose and glucuronate interconversions\n",
      "            pae00051  Fructose and mannose metabolism\n",
      "            pae00052  Galactose metabolism\n",
      "            pae00053  Ascorbate and aldarate metabolism\n",
      "            pae00061  Fatty acid biosynthesis\n",
      "            pae00071  Fatty acid degradation\n",
      "            pae00130  Ubiquinone and other terpenoid-quinone biosynthesis\n",
      "            pae00190  Oxidative phosphorylation\n",
      "            pae00220  Arginine biosynthesis\n",
      "            pae00230  Purine metabolism\n",
      "            pae00240  Pyrimidine metabolism\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00261  Monobactam biosynthesis\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00280  Valine, leucine and isoleucine degradation\n",
      "            pae00290  Valine, leucine and isoleucine biosynthesis\n",
      "            pae00300  Lysine biosynthesis\n",
      "            pae00310  Lysine degradation\n",
      "            pae00330  Arginine and proline metabolism\n",
      "            pae00332  Carbapenem biosynthesis\n",
      "            pae00340  Histidine metabolism\n",
      "            pae00350  Tyrosine metabolism\n",
      "            pae00360  Phenylalanine metabolism\n",
      "            pae00361  Chlorocyclohexane and chlorobenzene degradation\n",
      "            pae00362  Benzoate degradation\n",
      "            pae00364  Fluorobenzoate degradation\n",
      "            pae00380  Tryptophan metabolism\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "            pae00401  Novobiocin biosynthesis\n",
      "            pae00405  Phenazine biosynthesis\n",
      "            pae00410  beta-Alanine metabolism\n",
      "            pae00430  Taurine and hypotaurine metabolism\n",
      "            pae00440  Phosphonate and phosphinate metabolism\n",
      "            pae00450  Selenocompound metabolism\n",
      "            pae00460  Cyanoamino acid metabolism\n",
      "            pae00470  D-Amino acid metabolism\n",
      "            pae00480  Glutathione metabolism\n",
      "            pae00500  Starch and sucrose metabolism\n",
      "            pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "            pae00521  Streptomycin biosynthesis\n",
      "            pae00523  Polyketide sugar unit biosynthesis\n",
      "            pae00525  Acarbose and validamycin biosynthesis\n",
      "            pae00540  Lipopolysaccharide biosynthesis\n",
      "            pae00541  O-Antigen nucleotide sugar biosynthesis\n",
      "            pae00550  Peptidoglycan biosynthesis\n",
      "            pae00561  Glycerolipid metabolism\n",
      "            pae00562  Inositol phosphate metabolism\n",
      "            pae00564  Glycerophospholipid metabolism\n",
      "            pae00565  Ether lipid metabolism\n",
      "            pae00590  Arachidonic acid metabolism\n",
      "            pae00592  alpha-Linolenic acid metabolism\n",
      "            pae00600  Sphingolipid metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00622  Xylene degradation\n",
      "            pae00623  Toluene degradation\n",
      "            pae00625  Chloroalkane and chloroalkene degradation\n",
      "            pae00626  Naphthalene degradation\n",
      "            pae00627  Aminobenzoate degradation\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00633  Nitrotoluene degradation\n",
      "            pae00640  Propanoate metabolism\n",
      "            pae00643  Styrene degradation\n",
      "            pae00650  Butanoate metabolism\n",
      "            pae00660  C5-Branched dibasic acid metabolism\n",
      "            pae00670  One carbon pool by folate\n",
      "            pae00680  Methane metabolism\n",
      "            pae00730  Thiamine metabolism\n",
      "            pae00740  Riboflavin metabolism\n",
      "            pae00750  Vitamin B6 metabolism\n",
      "            pae00760  Nicotinate and nicotinamide metabolism\n",
      "            pae00770  Pantothenate and CoA biosynthesis\n",
      "            pae00780  Biotin metabolism\n",
      "            pae00785  Lipoic acid metabolism\n",
      "            pae00790  Folate biosynthesis\n",
      "            pae00791  Atrazine degradation\n",
      "            pae00860  Porphyrin metabolism\n",
      "            pae00900  Terpenoid backbone biosynthesis\n",
      "            pae00903  Limonene and pinene degradation\n",
      "            pae00910  Nitrogen metabolism\n",
      "            pae00920  Sulfur metabolism\n",
      "            pae00930  Caprolactam degradation\n",
      "            pae00997  Biosynthesis of various other secondary metabolites\n",
      "            pae00999  Biosynthesis of various plant secondary metabolites\n",
      "            pae01040  Biosynthesis of unsaturated fatty acids\n",
      "            pae01053  Biosynthesis of siderophore group nonribosomal peptides\n",
      "KO_PATHWAY  ko01100\n",
      "///\n",
      "path:pae01110\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae01110                    Pathway\n",
      "NAME        Biosynthesis of secondary metabolites - Pseudomonas aeruginosa PAO1\n",
      "PATHWAY_MAP pae01110  Biosynthesis of secondary metabolites\n",
      "MODULE      pae_M00018  Threonine biosynthesis, aspartate => homoserine => threonine [PATH:pae01110]\n",
      "            pae_M00019  Valine/isoleucine biosynthesis, pyruvate => valine / 2-oxobutanoate => isoleucine [PATH:pae01110]\n",
      "            pae_M00021  Cysteine biosynthesis, serine => cysteine [PATH:pae01110]\n",
      "            pae_M00023  Tryptophan biosynthesis, chorismate => tryptophan [PATH:pae01110]\n",
      "            pae_M00024  Phenylalanine biosynthesis, chorismate => phenylpyruvate => phenylalanine [PATH:pae01110]\n",
      "            pae_M00119  Pantothenate biosynthesis, valine/L-aspartate => pantothenate [PATH:pae01110]\n",
      "            pae_M00121  Heme biosynthesis, plants and bacteria, glutamate => heme [PATH:pae01110]\n",
      "            pae_M00125  Riboflavin biosynthesis, plants and bacteria, GTP => riboflavin/FMN/FAD [PATH:pae01110]\n",
      "            pae_M00432  Leucine biosynthesis, 2-oxoisovalerate => 2-oxoisocaproate [PATH:pae01110]\n",
      "            pae_M00793  dTDP-L-rhamnose biosynthesis [PATH:pae01110]\n",
      "            pae_M00835  Pyocyanine biosynthesis, chorismate => pyocyanine [PATH:pae01110]\n",
      "            pae_M00846  Siroheme biosynthesis, glutamyl-tRNA => siroheme [PATH:pae01110]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00030  Pentose phosphate pathway\n",
      "            pae00053  Ascorbate and aldarate metabolism\n",
      "            pae00130  Ubiquinone and other terpenoid-quinone biosynthesis\n",
      "            pae00220  Arginine biosynthesis\n",
      "            pae00230  Purine metabolism\n",
      "            pae00250  Alanine, aspartate and glutamate metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00261  Monobactam biosynthesis\n",
      "            pae00270  Cysteine and methionine metabolism\n",
      "            pae00280  Valine, leucine and isoleucine degradation\n",
      "            pae00281  Geraniol degradation\n",
      "            pae00290  Valine, leucine and isoleucine biosynthesis\n",
      "            pae00300  Lysine biosynthesis\n",
      "            pae00332  Carbapenem biosynthesis\n",
      "            pae00340  Histidine metabolism\n",
      "            pae00380  Tryptophan metabolism\n",
      "            pae00400  Phenylalanine, tyrosine and tryptophan biosynthesis\n",
      "            pae00401  Novobiocin biosynthesis\n",
      "            pae00405  Phenazine biosynthesis\n",
      "            pae00440  Phosphonate and phosphinate metabolism\n",
      "            pae00460  Cyanoamino acid metabolism\n",
      "            pae00500  Starch and sucrose metabolism\n",
      "            pae00521  Streptomycin biosynthesis\n",
      "            pae00523  Polyketide sugar unit biosynthesis\n",
      "            pae00525  Acarbose and validamycin biosynthesis\n",
      "            pae00564  Glycerophospholipid metabolism\n",
      "            pae00592  alpha-Linolenic acid metabolism\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00650  Butanoate metabolism\n",
      "            pae00740  Riboflavin metabolism\n",
      "            pae00770  Pantothenate and CoA biosynthesis\n",
      "            pae00860  Porphyrin metabolism\n",
      "            pae00900  Terpenoid backbone biosynthesis\n",
      "            pae00997  Biosynthesis of various other secondary metabolites\n",
      "            pae00999  Biosynthesis of various plant secondary metabolites\n",
      "            pae01053  Biosynthesis of siderophore group nonribosomal peptides\n",
      "KO_PATHWAY  ko01110\n",
      "///\n",
      "path:pae01120\n",
      "ENTRY       pae01120                    Pathway\n",
      "NAME        Microbial metabolism in diverse environments - Pseudomonas aeruginosa PAO1\n",
      "PATHWAY_MAP pae01120  Microbial metabolism in diverse environments\n",
      "MODULE      pae_M00007  Pentose phosphate pathway, non-oxidative phase, fructose 6P => ribose 5P [PATH:pae01120]\n",
      "            pae_M00008  Entner-Doudoroff pathway, glucose-6P => glyceraldehyde-3P + pyruvate [PATH:pae01120]\n",
      "            pae_M00176  Assimilatory sulfate reduction, sulfate => H2S [PATH:pae01120]\n",
      "            pae_M00529  Denitrification, nitrate => nitrogen [PATH:pae01120]\n",
      "            pae_M00530  Dissimilatory nitrate reduction, nitrate => ammonia [PATH:pae01120]\n",
      "            pae_M00533  Homoprotocatechuate degradation, homoprotocatechuate => 2-oxohept-3-enedioate [PATH:pae01120]\n",
      "            pae_M00551  Benzoate degradation, benzoate => catechol / methylbenzoate => methylcatechol [PATH:pae01120]\n",
      "            pae_M00568  Catechol ortho-cleavage, catechol => 3-oxoadipate [PATH:pae01120]\n",
      "            pae_M00579  Phosphate acetyltransferase-acetate kinase pathway, acetyl-CoA => acetate [PATH:pae01120]\n",
      "            pae_M00637  Anthranilate degradation, anthranilate => catechol [PATH:pae01120]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "REL_PATHWAY pae00010  Glycolysis / Gluconeogenesis\n",
      "            pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00030  Pentose phosphate pathway\n",
      "            pae00051  Fructose and mannose metabolism\n",
      "            pae00053  Ascorbate and aldarate metabolism\n",
      "            pae00230  Purine metabolism\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00300  Lysine biosynthesis\n",
      "            pae00310  Lysine degradation\n",
      "            pae00350  Tyrosine metabolism\n",
      "            pae00360  Phenylalanine metabolism\n",
      "            pae00361  Chlorocyclohexane and chlorobenzene degradation\n",
      "            pae00362  Benzoate degradation\n",
      "            pae00364  Fluorobenzoate degradation\n",
      "            pae00440  Phosphonate and phosphinate metabolism\n",
      "            pae00562  Inositol phosphate metabolism\n",
      "            pae00620  Pyruvate metabolism\n",
      "            pae00622  Xylene degradation\n",
      "            pae00623  Toluene degradation\n",
      "            pae00625  Chloroalkane and chloroalkene degradation\n",
      "            pae00626  Naphthalene degradation\n",
      "            pae00627  Aminobenzoate degradation\n",
      "            pae00630  Glyoxylate and dicarboxylate metabolism\n",
      "            pae00633  Nitrotoluene degradation\n",
      "            pae00643  Styrene degradation\n",
      "            pae00680  Methane metabolism\n",
      "            pae00750  Vitamin B6 metabolism\n",
      "            pae00760  Nicotinate and nicotinamide metabolism\n",
      "            pae00791  Atrazine degradation\n",
      "            pae00910  Nitrogen metabolism\n",
      "            pae00920  Sulfur metabolism\n",
      "            pae00930  Caprolactam degradation\n",
      "KO_PATHWAY  ko01120\n",
      "///\n",
      "path:pae01200\n",
      "ENTRY       pae01200                    Pathway\n",
      "NAME        Carbon metabolism - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Carbon metabolism is the most basic aspect of life. This map presents an overall view of central carbon metabolism, where the number of carbons is shown for each compound denoted by a circle, excluding a cofactor (CoA, CoM, THF, or THMPT) that is replaced by an asterisk. The map contains carbon utilization pathways of glycolysis (map00010), pentose phosphate pathway (map00030), and citrate cycle (map00020), and six known carbon fixation pathways (map00710 and map00720) as well as some pathways of methane metabolism (map00680). The six carbon fixation pathways are: (1) reductive pentose phosphate cycle (Calvin cycle) in plants and cyanobacteria that perform oxygenic photosynthesis, (2) reductive citrate cycle in photosynthetic green sulfur bacteria and some chemolithoautotrophs, (3) 3-hydroxypropionate bi-cycle in photosynthetic green nonsulfur bacteria, two variants of 4-hydroxybutyrate pathways in Crenarchaeota called (4) hydroxypropionate-hydroxybutyrate cycle and (5) dicarboxylate-hydroxybutyrate cycle, and (6) reductive acetyl-CoA pathway in methanogenic bacteria.\n",
      "PATHWAY_MAP pae01200  Carbon metabolism\n",
      "MODULE      pae_M00002  Glycolysis, core module involving three-carbon compounds [PATH:pae01200]\n",
      "            pae_M00005  PRPP biosynthesis, ribose 5P => PRPP [PATH:pae01200]\n",
      "            pae_M00007  Pentose phosphate pathway, non-oxidative phase, fructose 6P => ribose 5P [PATH:pae01200]\n",
      "            pae_M00008  Entner-Doudoroff pathway, glucose-6P => glyceraldehyde-3P + pyruvate [PATH:pae01200]\n",
      "            pae_M00009  Citrate cycle (TCA cycle, Krebs cycle) [PATH:pae01200]\n",
      "            pae_M00010  Citrate cycle, first carbon oxidation, oxaloacetate => 2-oxoglutarate [PATH:pae01200]\n",
      "            pae_M00011  Citrate cycle, second carbon oxidation, 2-oxoglutarate => oxaloacetate [PATH:pae01200]\n",
      "            pae_M00020  Serine biosynthesis, glycerate-3P => serine [PATH:pae01200]\n",
      "            pae_M00021  Cysteine biosynthesis, serine => cysteine [PATH:pae01200]\n",
      "            pae_M00307  Pyruvate oxidation, pyruvate => acetyl-CoA [PATH:pae01200]\n",
      "            pae_M00579  Phosphate acetyltransferase-acetate kinase pathway, acetyl-CoA => acetate [PATH:pae01200]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "KO_PATHWAY  ko01200\n",
      "///\n",
      "path:pae01210\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae01210                    Pathway\n",
      "NAME        2-Oxocarboxylic acid metabolism - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION 2-Oxocarboxylic acids, also called 2-oxo acids and alpha-keto acids, are the most elementary set of metabolites that includes pyruvate (2-oxopropanoate), 2-oxobutanoate, oxaloacetate (2-oxosuccinate) and 2-oxoglutarate. This diagram illustrates the architecture of chain extension and modification reaction modules for 2-oxocarboxylic acids. The chain extension module RM001 is a tricarboxylic pathway where acetyl-CoA derived carbon is used to extend the chain length by one. The chain modification modules RM002 (including RM032) and RM033, together with a reductive amination step (RC00006 or RC00036), generate basic and branched-chain amino acids, respectively. The modification module RM030 is used in the biosynthesis of glucosinolates, a class of plant secondary metabolites, for conversion to oxime followed by addition of thio-glucose moiety. Furthermore, the chain extension from 2-oxoadipate to 2-oxosuberate is followed by coenzyme B biosynthesis in methonogenic archaea.\n",
      "PATHWAY_MAP pae01210  2-Oxocarboxylic acid metabolism\n",
      "MODULE      pae_M00010  Citrate cycle, first carbon oxidation, oxaloacetate => 2-oxoglutarate [PATH:pae01210]\n",
      "            pae_M00019  Valine/isoleucine biosynthesis, pyruvate => valine / 2-oxobutanoate => isoleucine [PATH:pae01210]\n",
      "            pae_M00028  Ornithine biosynthesis, glutamate => ornithine [PATH:pae01210]\n",
      "            pae_M00432  Leucine biosynthesis, 2-oxoisovalerate => 2-oxoisocaproate [PATH:pae01210]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "            pae00220  Arginine biosynthesis\n",
      "            pae00260  Glycine, serine and threonine metabolism\n",
      "            pae00290  Valine, leucine and isoleucine biosynthesis\n",
      "            pae00300  Lysine biosynthesis\n",
      "            pae00680  Methane metabolism\n",
      "KO_PATHWAY  ko01210\n",
      "///\n",
      "path:pae01212\n",
      "ENTRY       pae01212                    Pathway\n",
      "NAME        Fatty acid metabolism - Pseudomonas aeruginosa PAO1\n",
      "PATHWAY_MAP pae01212  Fatty acid metabolism\n",
      "MODULE      pae_M00082  Fatty acid biosynthesis, initiation [PATH:pae01212]\n",
      "            pae_M00083  Fatty acid biosynthesis, elongation [PATH:pae01212]\n",
      "            pae_M00086  beta-Oxidation, acyl-CoA synthesis [PATH:pae01212]\n",
      "            pae_M00087  beta-Oxidation [PATH:pae01212]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "REL_PATHWAY pae00020  Citrate cycle (TCA cycle)\n",
      "KO_PATHWAY  ko01212\n",
      "///\n",
      "path:pae01220\n",
      "ENTRY       pae01220                    Pathway\n",
      "NAME        Degradation of aromatic compounds - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Microorganisms are known to be capable of degrading diverse chemical substances including man-made chemicals in the environment that are mostly aromatic compounds. This diagram illustrates combination patterns of reaction modules for biodegradation of aromatic compounds, consisting of three main types of ring dihydroxylation modules, followed by meta- or ortho-cleavage modules, together with an optional preprocessing module for converting methyl group to carboxyl group on the aromatic ring.\n",
      "PATHWAY_MAP pae01220  Degradation of aromatic compounds\n",
      "MODULE      pae_M00533  Homoprotocatechuate degradation, homoprotocatechuate => 2-oxohept-3-enedioate [PATH:pae01220]\n",
      "            pae_M00551  Benzoate degradation, benzoate => catechol / methylbenzoate => methylcatechol [PATH:pae01220]\n",
      "            pae_M00568  Catechol ortho-cleavage, catechol => 3-oxoadipate [PATH:pae01220]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "KO_PATHWAY  ko01220\n",
      "///\n",
      "path:pae01230\n",
      "ENTRY       pae01230                    Pathway\n",
      "NAME        Biosynthesis of amino acids - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION This map presents a modular architecture of the biosynthesis pathways of twenty amino acids, which may be viewed as consisting of the core part and its extensions. The core part is the KEGG module for conversion of three-carbon compounds from glyceraldehyde-3P to pyruvate [MD:M00002], together with the pathways around serine and glycine. This KEGG module is the most conserved one in the KEGG MODULE database and is found in almost all the completely sequenced genomes. The extensions are the pathways containing the reaction modules RM001, RM033, RM032, and RM002 for biosynthesis of branched-chain amino acids (left) and basic amino acids (bottom), and the pathways for biosynthesis of histidine and aromatic amino acids (top right). It is interesting to note that the so-called essential amino acids that cannot be synthesized in human and other organisms generally appear in these extensions. Furthermore, the bottom extension of basic amino acids appears to be most divergent containing multiple pathways for lysine biosynthesis and multiple gene sets for arginine biosynthesis.\n",
      "PATHWAY_MAP pae01230  Biosynthesis of amino acids\n",
      "MODULE      pae_M00002  Glycolysis, core module involving three-carbon compounds [PATH:pae01230]\n",
      "            pae_M00005  PRPP biosynthesis, ribose 5P => PRPP [PATH:pae01230]\n",
      "            pae_M00007  Pentose phosphate pathway, non-oxidative phase, fructose 6P => ribose 5P [PATH:pae01230]\n",
      "            pae_M00010  Citrate cycle, first carbon oxidation, oxaloacetate => 2-oxoglutarate [PATH:pae01230]\n",
      "            pae_M00015  Proline biosynthesis, glutamate => proline [PATH:pae01230]\n",
      "            pae_M00016  Lysine biosynthesis, succinyl-DAP pathway, aspartate => lysine [PATH:pae01230]\n",
      "            pae_M00018  Threonine biosynthesis, aspartate => homoserine => threonine [PATH:pae01230]\n",
      "            pae_M00019  Valine/isoleucine biosynthesis, pyruvate => valine / 2-oxobutanoate => isoleucine [PATH:pae01230]\n",
      "            pae_M00020  Serine biosynthesis, glycerate-3P => serine [PATH:pae01230]\n",
      "            pae_M00021  Cysteine biosynthesis, serine => cysteine [PATH:pae01230]\n",
      "            pae_M00023  Tryptophan biosynthesis, chorismate => tryptophan [PATH:pae01230]\n",
      "            pae_M00024  Phenylalanine biosynthesis, chorismate => phenylpyruvate => phenylalanine [PATH:pae01230]\n",
      "            pae_M00028  Ornithine biosynthesis, glutamate => ornithine [PATH:pae01230]\n",
      "            pae_M00338  Cysteine biosynthesis, homocysteine + serine => cysteine [PATH:pae01230]\n",
      "            pae_M00432  Leucine biosynthesis, 2-oxoisovalerate => 2-oxoisocaproate [PATH:pae01230]\n",
      "            pae_M00570  Isoleucine biosynthesis, threonine => 2-oxobutanoate => isoleucine [PATH:pae01230]\n",
      "            pae_M00844  Arginine biosynthesis, ornithine => arginine [PATH:pae01230]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "KO_PATHWAY  ko01230\n",
      "///\n",
      "path:pae01240\n",
      "ENTRY       pae01240                    Pathway\n",
      "NAME        Biosynthesis of cofactors - Pseudomonas aeruginosa PAO1\n",
      "PATHWAY_MAP pae01240  Biosynthesis of cofactors\n",
      "MODULE      pae_M00049  Adenine ribonucleotide biosynthesis, IMP => ADP,ATP [PATH:pae01240]\n",
      "            pae_M00052  Pyrimidine ribonucleotide biosynthesis, UMP => UDP/UTP,CDP/CTP [PATH:pae01240]\n",
      "            pae_M00115  NAD biosynthesis, aspartate => quinolinate => NAD [PATH:pae01240]\n",
      "            pae_M00118  Glutathione biosynthesis, glutamate => glutathione [PATH:pae01240]\n",
      "            pae_M00119  Pantothenate biosynthesis, valine/L-aspartate => pantothenate [PATH:pae01240]\n",
      "            pae_M00120  Coenzyme A biosynthesis, pantothenate => CoA [PATH:pae01240]\n",
      "            pae_M00121  Heme biosynthesis, plants and bacteria, glutamate => heme [PATH:pae01240]\n",
      "            pae_M00122  Cobalamin biosynthesis, cobyrinate a,c-diamide => cobalamin [PATH:pae01240]\n",
      "            pae_M00123  Biotin biosynthesis, pimeloyl-ACP/CoA => biotin [PATH:pae01240]\n",
      "            pae_M00124  Pyridoxal-P biosynthesis, erythrose-4P => pyridoxal-P [PATH:pae01240]\n",
      "            pae_M00125  Riboflavin biosynthesis, plants and bacteria, GTP => riboflavin/FMN/FAD [PATH:pae01240]\n",
      "            pae_M00126  Tetrahydrofolate biosynthesis, GTP => THF [PATH:pae01240]\n",
      "            pae_M00572  Pimeloyl-ACP biosynthesis, BioC-BioH pathway, malonyl-ACP => pimeloyl-ACP [PATH:pae01240]\n",
      "            pae_M00846  Siroheme biosynthesis, glutamyl-tRNA => siroheme [PATH:pae01240]\n",
      "            pae_M00880  Molybdenum cofactor biosynthesis, GTP => molybdenum cofactor [PATH:pae01240]\n",
      "            pae_M00881  Lipoic acid biosynthesis, plants and bacteria, octanoyl-ACP => dihydrolipoyl-E2/H [PATH:pae01240]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "KO_PATHWAY  ko01240\n",
      "///\n",
      "path:pae01250\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae01250                    Pathway\n",
      "NAME        Biosynthesis of nucleotide sugars - Pseudomonas aeruginosa PAO1\n",
      "PATHWAY_MAP pae01250  Biosynthesis of nucleotide sugars\n",
      "MODULE      pae_M00063  CMP-KDO biosynthesis [PATH:pae01250]\n",
      "            pae_M00064  ADP-L-glycero-D-manno-heptose biosynthesis [PATH:pae01250]\n",
      "            pae_M00549  Nucleotide sugar biosynthesis, glucose => UDP-glucose [PATH:pae01250]\n",
      "            pae_M00793  dTDP-L-rhamnose biosynthesis [PATH:pae01250]\n",
      "            pae_M00909  UDP-N-acetyl-D-glucosamine biosynthesis, prokaryotes, glucose => UDP-GlcNAc [PATH:pae01250]\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "KO_PATHWAY  ko01250\n",
      "///\n",
      "path:pae01501\n",
      "ENTRY       pae01501                    Pathway\n",
      "NAME        beta-Lactam resistance - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION The beta-lactam antibiotics are the most widely used group of antibiotics, which exert their effect by interfering with the structural crosslinking of peptidoglycans in bacterial cell walls. Over the past decades, a rapid increase in the number of beta-lactam-resistant clinical isolates is observed. Bacterial resistance to beta-lactam antibiotics can be achieved by any of the following strategies: producing inactivating enzymes called beta-lactamases, altering the beta-lactam targets of penicillin-binding proteins (PBPs), reducing transport of beta-lactams into the periplasmic space via changes in porins, and using the mechanisms for exclusion of beta-lactams by way of efflux pumps. Resistance may result from intrinsic properties of organisms, through gene mutations, and through plasmid- and transposon-specified genes.\n",
      "CLASS       Human Diseases; Drug resistance: antimicrobial\n",
      "PATHWAY_MAP pae01501  beta-Lactam resistance\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "REFERENCE   PMID:23034325\n",
      "  AUTHORS   Fernandez L, Hancock RE\n",
      "  TITLE     Adaptive and mutational resistance: role of porins and efflux pumps in drug resistance.\n",
      "  JOURNAL   Clin Microbiol Rev 25:661-81 (2012)\n",
      "            DOI:10.1128/CMR.00043-12\n",
      "REFERENCE   PMID:21149619\n",
      "  AUTHORS   Muller C, Plesiat P, Jeannot K\n",
      "  TITLE     A two-component regulatory system interconnects resistance to polymyxins, aminoglycosides, fluoroquinolones, and beta-lactams in Pseudomonas aeruginosa.\n",
      "  JOURNAL   Antimicrob Agents Chemother 55:1211-21 (2011)\n",
      "            DOI:10.1128/AAC.01252-10\n",
      "REFERENCE   PMID:20351243\n",
      "  AUTHORS   Tanabe M, Nimigean CM, Iverson TM\n",
      "  TITLE     Structural basis for solute transport, nucleotide regulation, and immunological recognition of Neisseria meningitidis PorB.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 107:6811-6 (2010)\n",
      "            DOI:10.1073/pnas.0912115107\n",
      "REFERENCE   PMID:9797206\n",
      "  AUTHORS   Gill MJ, Simjee S, Al-Hattawi K, Robertson BD, Easmon CS, Ison CA\n",
      "  TITLE     Gonococcal resistance to beta-lactams and tetracycline involves mutation in loop 3 of the porin encoded at the penB locus.\n",
      "  JOURNAL   Antimicrob Agents Chemother 42:2799-803 (1998)\n",
      "            DOI:10.1128/AAC.42.11.2799\n",
      "REFERENCE   PMID:23163477\n",
      "  AUTHORS   Johnson JW, Fisher JF, Mobashery S\n",
      "  TITLE     Bacterial cell-wall recycling.\n",
      "  JOURNAL   Ann N Y Acad Sci 1277:54-75 (2013)\n",
      "            DOI:10.1111/j.1749-6632.2012.06813.x\n",
      "REFERENCE   PMID:22438804\n",
      "  AUTHORS   Amoroso A, Boudet J, Berzigotti S, Duval V, Teller N, Mengin-Lecreulx D, Luxen A, Simorre JP, Joris B\n",
      "  TITLE     A peptidoglycan fragment triggers beta-lactam resistance in Bacillus licheniformis.\n",
      "  JOURNAL   PLoS Pathog 8:e1002571 (2012)\n",
      "            DOI:10.1371/journal.ppat.1002571\n",
      "REFERENCE   PMID:19154333\n",
      "  AUTHORS   Sala C, Haouz A, Saul FA, Miras I, Rosenkrands I, Alzari PM, Cole ST\n",
      "  TITLE     Genome-wide regulon and crystal structure of BlaI (Rv1846c) from Mycobacterium tuberculosis.\n",
      "  JOURNAL   Mol Microbiol 71:1102-16 (2009)\n",
      "            DOI:10.1111/j.1365-2958.2008.06583.x\n",
      "REFERENCE   PMID:18248419\n",
      "  AUTHORS   Zapun A, Contreras-Martel C, Vernet T\n",
      "  TITLE     Penicillin-binding proteins and beta-lactam resistance.\n",
      "  JOURNAL   FEMS Microbiol Rev 32:361-85 (2008)\n",
      "            DOI:10.1111/j.1574-6976.2007.00095.x\n",
      "REFERENCE   PMID:19258280\n",
      "  AUTHORS   Vettoretti L, Plesiat P, Muller C, El Garch F, Phan G, Attree I, Ducruix A, Llanes C\n",
      "  TITLE     Efflux unbalance in Pseudomonas aeruginosa isolates from cystic fibrosis patients.\n",
      "  JOURNAL   Antimicrob Agents Chemother 53:1987-97 (2009)\n",
      "            DOI:10.1128/AAC.01024-08\n",
      "REFERENCE   PMID:15387820\n",
      "  AUTHORS   Cao L, Srikumar R, Poole K\n",
      "  TITLE     MexAB-OprM hyperexpression in NalC-type multidrug-resistant Pseudomonas aeruginosa: identification and characterization of the nalC gene encoding a repressor of PA3720-PA3719.\n",
      "  JOURNAL   Mol Microbiol 53:1423-36 (2004)\n",
      "            DOI:10.1111/j.1365-2958.2004.04210.x\n",
      "REFERENCE   PMID:11895293\n",
      "  AUTHORS   Poole K, Srikumar R\n",
      "  TITLE     Multidrug efflux in Pseudomonas aeruginosa: components, mechanisms and clinical significance.\n",
      "  JOURNAL   Curr Top Med Chem 1:59-71 (2001)\n",
      "            DOI:10.2174/1568026013395605\n",
      "REFERENCE   PMID:15855496\n",
      "  AUTHORS   Sobel ML, Hocquet D, Cao L, Plesiat P, Poole K\n",
      "  TITLE     Mutations in PA3574 (nalD) lead to increased MexAB-OprM expression and multidrug resistance in laboratory and clinical isolates of Pseudomonas aeruginosa.\n",
      "  JOURNAL   Antimicrob Agents Chemother 49:1782-6 (2005)\n",
      "            DOI:10.1128/AAC.49.5.1782-1786.2005\n",
      "REFERENCE   PMID:15105114\n",
      "  AUTHORS   Kaczmarek FS, Gootz TD, Dib-Hajj F, Shang W, Hallowell S, Cronan M\n",
      "  TITLE     Genetic and molecular characterization of beta-lactamase-negative ampicillin-resistant Haemophilus influenzae with unusually high resistance to ampicillin.\n",
      "  JOURNAL   Antimicrob Agents Chemother 48:1630-9 (2004)\n",
      "            DOI:10.1128/AAC.48.5.1630-1639.2004\n",
      "REFERENCE   PMID:16048914\n",
      "  AUTHORS   Dean CR, Narayan S, Daigle DM, Dzink-Fox JL, Puyang X, Bracken KR, Dean KE, Weidmann B, Yuan Z, Jain R, Ryder NS\n",
      "  TITLE     Role of the AcrAB-TolC efflux pump in determining susceptibility of Haemophilus influenzae to the novel peptide deformylase inhibitor LBM415.\n",
      "  JOURNAL   Antimicrob Agents Chemother 49:3129-35 (2005)\n",
      "            DOI:10.1128/AAC.49.8.3129-3135.2005\n",
      "REFERENCE   PMID:22371895\n",
      "  AUTHORS   Rosenfeld N, Bouchier C, Courvalin P, Perichon B\n",
      "  TITLE     Expression of the resistance-nodulation-cell division pump AdeIJK in Acinetobacter baumannii is regulated by AdeN, a TetR-type regulator.\n",
      "  JOURNAL   Antimicrob Agents Chemother 56:2504-10 (2012)\n",
      "            DOI:10.1128/AAC.06422-11\n",
      "REFERENCE   PMID:21173183\n",
      "  AUTHORS   Coyne S, Courvalin P, Perichon B\n",
      "  TITLE     Efflux-mediated antibiotic resistance in Acinetobacter spp.\n",
      "  JOURNAL   Antimicrob Agents Chemother 55:947-53 (2011)\n",
      "            DOI:10.1128/AAC.01388-10\n",
      "REFERENCE   PMID:19026770\n",
      "  AUTHORS   Nikaido H, Takatsuka Y\n",
      "  TITLE     Mechanisms of RND multidrug efflux pumps.\n",
      "  JOURNAL   Biochim Biophys Acta 1794:769-81 (2009)\n",
      "            DOI:10.1016/j.bbapap.2008.10.004\n",
      "REFERENCE   PMID:17277059\n",
      "  AUTHORS   Truong-Bolduc QC, Hooper DC\n",
      "  TITLE     The transcriptional regulators NorG and MgrA modulate resistance to both quinolones and beta-lactams in Staphylococcus aureus.\n",
      "  JOURNAL   J Bacteriol 189:2996-3005 (2007)\n",
      "            DOI:10.1128/JB.01819-06\n",
      "REFERENCE   PMID:12222968\n",
      "  AUTHORS   Sauvage E, Kerff F, Fonze E, Herman R, Schoot B, Marquette JP, Taburet Y, Prevost D, Dumas J, Leonard G, Stefanic P, Coyette J, Charlier P\n",
      "  TITLE     The 2.4-A crystal structure of the penicillin-resistant penicillin-binding protein PBP5fm from Enterococcus faecium in complex with benzylpenicillin.\n",
      "  JOURNAL   Cell Mol Life Sci 59:1223-32 (2002)\n",
      "            DOI:10.1007/s00018-002-8500-0\n",
      "REFERENCE   PMID:16143832\n",
      "  AUTHORS   Fuda CC, Fisher JF, Mobashery S\n",
      "  TITLE     Beta-lactam resistance in Staphylococcus aureus: the adaptive resistance of a plastic genome.\n",
      "  JOURNAL   Cell Mol Life Sci 62:2617-33 (2005)\n",
      "            DOI:10.1007/s00018-005-5148-6\n",
      "REL_PATHWAY pae00550  Peptidoglycan biosynthesis\n",
      "KO_PATHWAY  ko01501\n",
      "///\n",
      "path:pae01502\n",
      "ENTRY       pae01502                    Pathway\n",
      "NAME        Vancomycin resistance - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Vancomycin (VCM) is a glycopeptide antibiotic agent that inhibits the synthesis of peptidolgycan in the production of bacterial cell walls. It is active in most Gram-positive bacteria and acts by binding to the D-Ala-D-Ala C-terminus of pentapeptide, thus blocking the addition of subunits to the peptidoglycan backbone. The resistance to VCM was first described among enterococci in the late 1980s in Europe. Subsequently, vancomycin-resistant enteococci (VRE) have been observed in many countries. The genes encoding resistance are carried on plasmids that can tranfer themselves from cell to cell and on tranponsons that can jump from plamids to the chromosome. Up to now, at least six resistance operon types are characterized, namely VanA, VanB, VanC, VanD, VanE and VanG. The VanA and VanB operons are detected on plasmids or in the chromosome, whereas the rest are found in the chromosome. In addition, Streptomyces, which is non-pathogenic, non-glycopeptide-producing actinomycete, also shows high-level resistance to VCM. In Streptomyces, novel genes (vanK and vanJ) are characterized with no counterparts found in the resistance clusters of pathogenic bacteria.\n",
      "            The expression of vancomycin resistance pathway is induced by the VanS-VanR two component system, and D-Ala-D-Lac or D-Ala-D-Ser dipsipeptides are formed in place of D-Ala-D-Ala dipeptide. Thus, VCM cannot bind to pentadepsipeptide[D-Lac] or pentadepsipeptide[D-Ser]. The D-Ala-D-Lac variation shows high levels of resistance to VCM, especially in VanA type, wheras the D-Ala-D-Ser variation shows low levels of resistance. The modified pepidoglycan can interact with cross-linking enzymes (PBPs), thus cell wall is successfully formed.\n",
      "CLASS       Human Diseases; Drug resistance: antimicrobial\n",
      "PATHWAY_MAP pae01502  Vancomycin resistance\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "REFERENCE   PMID:16323116\n",
      "  AUTHORS   Courvalin P\n",
      "  TITLE     Vancomycin resistance in gram-positive cocci.\n",
      "  JOURNAL   Clin Infect Dis 42 Suppl 1:S25-34 (2006)\n",
      "            DOI:10.1086/491711\n",
      "REFERENCE   PMID:22913355\n",
      "  AUTHORS   Chancey ST, Zahner D, Stephens DS\n",
      "  TITLE     Acquired inducible antimicrobial resistance in Gram-positive bacteria.\n",
      "  JOURNAL   Future Microbiol 7:959-78 (2012)\n",
      "            DOI:10.2217/fmb.12.63\n",
      "REFERENCE   PMID:8899966\n",
      "  AUTHORS   Arthur M, Reynolds P, Courvalin P\n",
      "  TITLE     Glycopeptide resistance in enterococci.\n",
      "  JOURNAL   Trends Microbiol 4:401-7 (1996)\n",
      "            DOI:10.1016/0966-842X(96)10063-9\n",
      "REFERENCE   PMID:11023964\n",
      "  AUTHORS   Cetinkaya Y, Falk P, Mayhall CG\n",
      "  TITLE     Vancomycin-resistant enterococci.\n",
      "  JOURNAL   Clin Microbiol Rev 13:686-707 (2000)\n",
      "            DOI:10.1128/CMR.13.4.686-707.2000\n",
      "REFERENCE   PMID:7854121\n",
      "  AUTHORS   Reynolds PE, Depardieu F, Dutka-Malen S, Arthur M, Courvalin P\n",
      "  TITLE     Glycopeptide resistance mediated by enterococcal transposon Tn1546 requires production of VanX for hydrolysis of D-alanyl-D-alanine.\n",
      "  JOURNAL   Mol Microbiol 13:1065-70 (1994)\n",
      "            DOI:10.1111/j.1365-2958.1994.tb00497.x\n",
      "REFERENCE   PMID:8631706\n",
      "  AUTHORS   Evers S, Courvalin P\n",
      "  TITLE     Regulation of VanB-type vancomycin resistance gene expression by the VanS(B)-VanR (B) two-component regulatory system in Enterococcus faecalis V583.\n",
      "  JOURNAL   J Bacteriol 178:1302-9 (1996)\n",
      "            DOI:10.1128/JB.178.5.1302-1309.1996\n",
      "REFERENCE   PMID:10564477\n",
      "  AUTHORS   Reynolds PE, Arias CA, Courvalin P\n",
      "  TITLE     Gene vanXYC encodes D,D -dipeptidase (VanX) and D,D-carboxypeptidase (VanY) activities in vancomycin-resistant Enterococcus gallinarum BM4174.\n",
      "  JOURNAL   Mol Microbiol 34:341-9 (1999)\n",
      "            DOI:10.1046/j.1365-2958.1999.01604.x\n",
      "REFERENCE   PMID:12047383\n",
      "  AUTHORS   Podmore AH, Reynolds PE\n",
      "  TITLE     Purification and characterization of VanXY(C), a D,D-dipeptidase/D,D-carboxypeptidase in vancomycin-resistant Enterococcus gallinarum BM4174.\n",
      "  JOURNAL   Eur J Biochem 269:2740-6 (2002)\n",
      "            DOI:10.1046/j.1432-1033.2002.02946.x\n",
      "REFERENCE   PMID:15616270\n",
      "  AUTHORS   Reynolds PE, Courvalin P\n",
      "  TITLE     Vancomycin resistance in enterococci due to synthesis of precursors terminating in D-alanyl-D-serine.\n",
      "  JOURNAL   Antimicrob Agents Chemother 49:21-5 (2005)\n",
      "            DOI:10.1128/AAC.49.1.21-25.2005\n",
      "REFERENCE   PMID:10878136\n",
      "  AUTHORS   Arias CA, Weisner J, Blackburn JM, Reynolds PE\n",
      "  TITLE     Serine and alanine racemase activities of VanT: a protein necessary for vancomycin resistance in Enterococcus gallinarum BM4174.\n",
      "  JOURNAL   Microbiology 146 ( Pt 7):1727-34 (2000)\n",
      "            DOI:10.1099/00221287-146-7-1727\n",
      "REFERENCE   PMID:18792691\n",
      "  AUTHORS   Hong HJ, Hutchings MI, Buttner MJ\n",
      "  TITLE     Vancomycin resistance VanS/VanR two-component systems.\n",
      "  JOURNAL   Adv Exp Med Biol 631:200-13 (2008)\n",
      "            DOI:10.1007/978-0-387-78885-2_14\n",
      "REFERENCE   PMID:14617152\n",
      "  AUTHORS   Depardieu F, Bonora MG, Reynolds PE, Courvalin P\n",
      "  TITLE     The vanG glycopeptide resistance operon from Enterococcus faecalis revisited.\n",
      "  JOURNAL   Mol Microbiol 50:931-48 (2003)\n",
      "            DOI:10.1046/j.1365-2958.2003.03737.x\n",
      "REFERENCE   PMID:11036060\n",
      "  AUTHORS   McKessar SJ, Berry AM, Bell JM, Turnidge JD, Paton JC\n",
      "  TITLE     Genetic characterization of vanG, a novel vancomycin resistance locus of Enterococcus faecalis.\n",
      "  JOURNAL   Antimicrob Agents Chemother 44:3224-8 (2000)\n",
      "            DOI:10.1128/AAC.44.11.3224-3228.2000\n",
      "REFERENCE   PMID:23836175\n",
      "  AUTHORS   Kwun MJ, Novotna G, Hesketh AR, Hill L, Hong HJ\n",
      "  TITLE     In vivo studies suggest that induction of VanS-dependent vancomycin resistance requires binding of the drug to D-Ala-D-Ala termini in the peptidoglycan cell wall.\n",
      "  JOURNAL   Antimicrob Agents Chemother 57:4470-80 (2013)\n",
      "            DOI:10.1128/AAC.00523-13\n",
      "REFERENCE   PMID:15130128\n",
      "  AUTHORS   Hong HJ, Hutchings MI, Neu JM, Wright GD, Paget MS, Buttner MJ\n",
      "  TITLE     Characterization of an inducible vancomycin resistance system in Streptomyces coelicolor reveals a novel gene (vanK) required for drug resistance.\n",
      "  JOURNAL   Mol Microbiol 52:1107-21 (2004)\n",
      "            DOI:10.1111/j.1365-2958.2004.04032.x\n",
      "REFERENCE   PMID:15632111\n",
      "  AUTHORS   Hong HJ, Hutchings MI, Hill LM, Buttner MJ\n",
      "  TITLE     The role of the novel Fem protein VanK in vancomycin resistance in Streptomyces coelicolor.\n",
      "  JOURNAL   J Biol Chem 280:13055-61 (2005)\n",
      "            DOI:10.1074/jbc.M413801200\n",
      "REFERENCE   PMID:16420361\n",
      "  AUTHORS   Hutchings MI, Hong HJ, Buttner MJ\n",
      "  TITLE     The vancomycin resistance VanRS two-component signal transduction system of Streptomyces coelicolor.\n",
      "  JOURNAL   Mol Microbiol 59:923-35 (2006)\n",
      "            DOI:10.1111/j.1365-2958.2005.04953.x\n",
      "REFERENCE   PMID:21569315\n",
      "  AUTHORS   Hesketh A, Hill C, Mokhtar J, Novotna G, Tran N, Bibb M, Hong HJ\n",
      "  TITLE     Genome-wide dynamics of a bacterial response to antibiotics that target the cell envelope.\n",
      "  JOURNAL   BMC Genomics 12:226 (2011)\n",
      "            DOI:10.1186/1471-2164-12-226\n",
      "REFERENCE   PMID:20383152\n",
      "  AUTHORS   Koteva K, Hong HJ, Wang XD, Nazi I, Hughes D, Naldrett MJ, Buttner MJ, Wright GD\n",
      "  TITLE     A vancomycin photoprobe identifies the histidine kinase VanSsc as a vancomycin receptor.\n",
      "  JOURNAL   Nat Chem Biol 6:327-9 (2010)\n",
      "            DOI:10.1038/nchembio.350\n",
      "REL_PATHWAY pae00550  Peptidoglycan biosynthesis\n",
      "KO_PATHWAY  ko01502\n",
      "///\n",
      "path:pae01503\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae01503                    Pathway\n",
      "NAME        Cationic antimicrobial peptide (CAMP) resistance - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Cationic antimicrobial peptides (CAMPs) play an important role in host defense against microbial infection and are key components of the innate immune response. These are found among all classes of life ranging from prokaryotes to humans. In addition to the natural peptides, thousands of synthetic variants have been produced. CAMPs weaken the integrity of the bacterial inner and outer membranes and subsequently kill bacterial cells. On the other hand, bacteria have developed a number of mechanisms against CAMPs. These resistance mechanisms include decreased affinity to CAMPs by substitution of anionic cell surface constituents with cationic molecules; biosynthesis and crosslinking of cell envelope components; external trapping mechanisms that bind or neutralize the CAMPs by direct secretion of proteins, or the release of CAMPs binding molecules from the host cell surface; membrane efflux pumps; and production of peptidases.\n",
      "CLASS       Human Diseases; Drug resistance: antimicrobial\n",
      "PATHWAY_MAP pae01503  Cationic antimicrobial peptide (CAMP) resistance\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "REFERENCE   PMID:16450883\n",
      "  AUTHORS   Nizet V\n",
      "  TITLE     Antimicrobial peptide resistance mechanisms of human bacterial pathogens.\n",
      "  JOURNAL   Curr Issues Mol Biol 8:11-26 (2006)\n",
      "REFERENCE   PMID:22339775\n",
      "  AUTHORS   Gruenheid S, Le Moual H\n",
      "  TITLE     Resistance to antimicrobial peptides in Gram-negative bacteria.\n",
      "  JOURNAL   FEMS Microbiol Lett 330:81-9 (2012)\n",
      "            DOI:10.1111/j.1574-6968.2012.02528.x\n",
      "REFERENCE   PMID:25419466\n",
      "  AUTHORS   Nawrocki KL, Crispell EK, McBride SM\n",
      "  TITLE     Antimicrobial Peptide Resistance Mechanisms of Gram-Positive Bacteria.\n",
      "  JOURNAL   Antibiotics (Basel) 3:461-492 (2014)\n",
      "            DOI:10.3390/antibiotics3040461\n",
      "REFERENCE   PMID:22799636\n",
      "  AUTHORS   Anaya-Lopez JL, Lopez-Meza JE, Ochoa-Zarzosa A\n",
      "  TITLE     Bacterial resistance to cationic antimicrobial peptides.\n",
      "  JOURNAL   Crit Rev Microbiol 39:180-95 (2013)\n",
      "            DOI:10.3109/1040841X.2012.699025\n",
      "REFERENCE   PMID:23748343\n",
      "  AUTHORS   Needham BD, Trent MS\n",
      "  TITLE     Fortifying the barrier: the impact of lipid A remodelling on bacterial pathogenesis.\n",
      "  JOURNAL   Nat Rev Microbiol 11:467-81 (2013)\n",
      "            DOI:10.1038/nrmicro3047\n",
      "REFERENCE   PMID:17928292\n",
      "  AUTHORS   Yan A, Guan Z, Raetz CR\n",
      "  TITLE     An undecaprenyl phosphate-aminoarabinose flippase required for polymyxin resistance in Escherichia coli.\n",
      "  JOURNAL   J Biol Chem 282:36077-89 (2007)\n",
      "            DOI:10.1074/jbc.M706172200\n",
      "REFERENCE   PMID:15695810\n",
      "  AUTHORS   Breazeale SD, Ribeiro AA, McClerren AL, Raetz CR.\n",
      "  TITLE     A formyltransferase required for polymyxin resistance in Escherichia coli and the modification of lipid A with 4-Amino-4-deoxy-L-arabinose. Identification and function oF UDP-4-deoxy-4-formamido-L-arabinose.\n",
      "  JOURNAL   J Biol Chem 280:14154-67 (2005)\n",
      "            DOI:10.1074/jbc.M414265200\n",
      "REFERENCE   PMID:25505462\n",
      "  AUTHORS   Olaitan AO, Morand S, Rolain JM\n",
      "  TITLE     Mechanisms of polymyxin resistance: acquired and intrinsic resistance in bacteria.\n",
      "  JOURNAL   Front Microbiol 5:643 (2014)\n",
      "            DOI:10.3389/fmicb.2014.00643\n",
      "REFERENCE   PMID:21645016\n",
      "  AUTHORS   Martinez JL, Rojo F\n",
      "  TITLE     Metabolic regulation of antibiotic resistance.\n",
      "  JOURNAL   FEMS Microbiol Rev 35:768-89 (2011)\n",
      "            DOI:10.1111/j.1574-6976.2011.00282.x\n",
      "REFERENCE   PMID:20843473\n",
      "  AUTHORS   Falagas ME, Rafailidis PI, Matthaiou DK\n",
      "  TITLE     Resistance to polymyxins: Mechanisms, frequency and treatment options.\n",
      "  JOURNAL   Drug Resist Updat 13:132-8 (2010)\n",
      "            DOI:10.1016/j.drup.2010.05.002\n",
      "REFERENCE   PMID:19243444\n",
      "  AUTHORS   Gooderham WJ, Hancock RE\n",
      "  TITLE     Regulation of virulence and antibiotic resistance by two-component regulatory systems in Pseudomonas aeruginosa.\n",
      "  JOURNAL   FEMS Microbiol Rev 33:279-94 (2009)\n",
      "            DOI:10.1111/j.1574-6976.2008.00135.x\n",
      "REFERENCE   PMID:15205413\n",
      "  AUTHORS   Lee H, Hsu FF, Turk J, Groisman EA\n",
      "  TITLE     The PmrA-regulated pmrC gene mediates phosphoethanolamine modification of lipid A and polymyxin resistance in Salmonella enterica.\n",
      "  JOURNAL   J Bacteriol 186:4124-33 (2004)\n",
      "            DOI:10.1128/JB.186.13.4124-4133.2004\n",
      "REFERENCE   PMID:24034668\n",
      "  AUTHORS   Wang D, Seeve C, Pierson LS 3rd, Pierson EA\n",
      "  TITLE     Transcriptome profiling reveals links between ParS/ParR, MexEF-OprN, and quorum sensing in the regulation of adaptation and virulence in Pseudomonas aeruginosa.\n",
      "  JOURNAL   BMC Genomics 14:618 (2013)\n",
      "            DOI:10.1186/1471-2164-14-618\n",
      "REFERENCE   PMID:22123691\n",
      "  AUTHORS   Falord M, Karimova G, Hiron A, Msadek T\n",
      "  TITLE     GraXSR proteins interact with the VraFG ABC transporter to form a five-component system required for cationic antimicrobial peptide sensing and resistance in Staphylococcus aureus.\n",
      "  JOURNAL   Antimicrob Agents Chemother 56:1047-58 (2012)\n",
      "            DOI:10.1128/AAC.05054-11\n",
      "REFERENCE   PMID:16715050\n",
      "  AUTHORS   Rowley G, Spector M, Kormanec J, Roberts M\n",
      "  TITLE     Pushing the envelope: extracytoplasmic stress responses in bacterial pathogens.\n",
      "  JOURNAL   Nat Rev Microbiol 4:383-94 (2006)\n",
      "            DOI:10.1038/nrmicro1394\n",
      "REFERENCE   PMID:23267440\n",
      "  AUTHORS   Piek S, Kahler CM\n",
      "  TITLE     A comparison of the endotoxin biosynthesis and protein oxidation pathways in the biogenesis of the outer membrane of Escherichia coli and Neisseria meningitidis.\n",
      "  JOURNAL   Front Cell Infect Microbiol 2:162 (2012)\n",
      "            DOI:10.3389/fcimb.2012.00162\n",
      "REFERENCE   PMID:25294881\n",
      "  AUTHORS   Weatherspoon-Griffin N, Yang D, Kong W, Hua Z, Shi Y\n",
      "  TITLE     The CpxR/CpxA two-component regulatory system up-regulates the multidrug resistance cascade to facilitate Escherichia coli resistance to a model antimicrobial peptide.\n",
      "  JOURNAL   J Biol Chem 289:32571-82 (2014)\n",
      "            DOI:10.1074/jbc.M114.565762\n",
      "REFERENCE   PMID:22092948\n",
      "  AUTHORS   Vogt SL, Raivio TL\n",
      "  TITLE     Just scratching the surface: an expanding view of the Cpx envelope stress response.\n",
      "  JOURNAL   FEMS Microbiol Lett 326:2-11 (2012)\n",
      "            DOI:10.1111/j.1574-6968.2011.02406.x\n",
      "REFERENCE   PMID:22072973\n",
      "  AUTHORS   Shelton CL, Raffel FK, Beatty WL, Johnson SM, Mason KM\n",
      "  TITLE     Sap transporter mediated import and subsequent degradation of antimicrobial peptides in Haemophilus.\n",
      "  JOURNAL   PLoS Pathog 7:e1002360 (2011)\n",
      "            DOI:10.1371/journal.ppat.1002360\n",
      "REFERENCE   PMID:19609257\n",
      "  AUTHORS   Otto M\n",
      "  TITLE     Staphylococcus epidermidis--the 'accidental' pathogen.\n",
      "  JOURNAL   Nat Rev Microbiol 7:555-67 (2009)\n",
      "            DOI:10.1038/nrmicro2182\n",
      "REFERENCE   PMID:21986630\n",
      "  AUTHORS   Yang SJ, Bayer AS, Mishra NN, Meehl M, Ledala N, Yeaman MR, Xiong YQ, Cheung AL\n",
      "  TITLE     The Staphylococcus aureus two-component regulatory system, GraRS, senses and confers resistance to selected cationic antimicrobial peptides.\n",
      "  JOURNAL   Infect Immun 80:74-81 (2012)\n",
      "            DOI:10.1128/IAI.05669-11\n",
      "REFERENCE   PMID:12366839\n",
      "  AUTHORS   Schmidtchen A, Frick IM, Andersson E, Tapper H, Bjorck L\n",
      "  TITLE     Proteinases of common pathogenic bacteria degrade and inactivate the antibacterial peptide LL-37.\n",
      "  JOURNAL   Mol Microbiol 46:157-68 (2002)\n",
      "            DOI:10.1046/j.1365-2958.2002.03146.x\n",
      "REFERENCE   PMID:23554414\n",
      "  AUTHORS   Beceiro A, Tomas M, Bou G\n",
      "  TITLE     Antimicrobial resistance and virulence: a successful or deleterious association in the bacterial world?\n",
      "  JOURNAL   Clin Microbiol Rev 26:185-230 (2013)\n",
      "            DOI:10.1128/CMR.00059-12\n",
      "REL_PATHWAY pae00520  Amino sugar and nucleotide sugar metabolism\n",
      "            pae00540  Lipopolysaccharide biosynthesis\n",
      "            pae02020  Two-component system\n",
      "KO_PATHWAY  ko01503\n",
      "///\n",
      "path:pae02010\n",
      "ENTRY       pae02010                    Pathway\n",
      "NAME        ABC transporters - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION The ATP-binding cassette (ABC) transporters form one of the largest known protein families, and are widespread in bacteria, archaea, and eukaryotes. They couple ATP hydrolysis to active transport of a wide variety of substrates such as ions, sugars, lipids, sterols, peptides, proteins, and drugs. The structure of a prokaryotic ABC transporter usually consists of three components; typically two integral membrane proteins each having six transmembrane segments, two peripheral proteins that bind and hydrolyze ATP, and a periplasmic (or lipoprotein) substrate-binding protein. Many of the genes for the three components form operons as in fact observed in many bacterial and archaeal genomes. On the other hand, in a typical eukaryotic ABC transporter, the membrane spanning protein and the ATP-binding protein are fused, forming a multi-domain protein with the membrane-spanning domain (MSD) and the nucleotide-binding domain (NBD).\n",
      "CLASS       Environmental Information Processing; Membrane transport\n",
      "PATHWAY_MAP pae02010  ABC transporters\n",
      "DBLINKS     GO: 0043190 0043191 0043192\n",
      "            TC: 3.A.1\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1493  cysP; sulfate ABC transporter substrate-binding protein [KO:K02048]\n",
      "line in gene PA1493  cysP; sulfate ABC transporter substrate-binding protein [KO:K02048]\n",
      "            PA0283  sbp; sulfate-binding protein [KO:K23163]\n",
      "line in gene PA0283  sbp; sulfate-binding protein [KO:K23163]\n",
      "            PA0282  cysT; sulfate transporter CysT [KO:K02046]\n",
      "line in gene PA0282  cysT; sulfate transporter CysT [KO:K02046]\n",
      "            PA0281  cysW; sulfate transporter CysW [KO:K02047]\n",
      "line in gene PA0281  cysW; sulfate transporter CysW [KO:K02047]\n",
      "            PA0280  cysA; sulfate.thiosulfate ABC transporter ATP-binding protein CysA [KO:K02045] [EC:7.3.2.3]\n",
      "line in gene PA0280  cysA; sulfate.thiosulfate ABC transporter ATP-binding protein CysA [KO:K02045] [EC:7.3.2.3]\n",
      "            PA3938  taurine-binding protein [KO:K15551]\n",
      "line in gene PA3938  taurine-binding protein [KO:K15551]\n",
      "            PA3936  taurine ABC transporter permease [KO:K15552]\n",
      "line in gene PA3936  taurine ABC transporter permease [KO:K15552]\n",
      "            PA3937  taurine ABC transporter ATP-binding protein [KO:K10831] [EC:7.6.2.7]\n",
      "line in gene PA3937  taurine ABC transporter ATP-binding protein [KO:K10831] [EC:7.6.2.7]\n",
      "            PA3449  hypothetical protein [KO:K15553]\n",
      "line in gene PA3449  hypothetical protein [KO:K15553]\n",
      "            PA2595  hypothetical protein [KO:K15553]\n",
      "line in gene PA2595  hypothetical protein [KO:K15553]\n",
      "            PA2594  hypothetical protein [KO:K15553]\n",
      "line in gene PA2594  hypothetical protein [KO:K15553]\n",
      "            PA3445  hypothetical protein [KO:K15553]\n",
      "line in gene PA3445  hypothetical protein [KO:K15553]\n",
      "            PA0186  ABC transporter [KO:K15553]\n",
      "line in gene PA0186  ABC transporter [KO:K15553]\n",
      "            PA2596  hypothetical protein [KO:K15553]\n",
      "line in gene PA2596  hypothetical protein [KO:K15553]\n",
      "            PA2599  hypothetical protein [KO:K15553]\n",
      "line in gene PA2599  hypothetical protein [KO:K15553]\n",
      "            PA2309  hypothetical protein [KO:K15553]\n",
      "line in gene PA2309  hypothetical protein [KO:K15553]\n",
      "            PA2296  hypothetical protein [KO:K15553]\n",
      "line in gene PA2296  hypothetical protein [KO:K15553]\n",
      "            PA3443  ABC transporter permease [KO:K15554]\n",
      "line in gene PA3443  ABC transporter permease [KO:K15554]\n",
      "            PA3448  ABC transporter permease [KO:K15554]\n",
      "line in gene PA3448  ABC transporter permease [KO:K15554]\n",
      "            PA0185  ABC transporter permease [KO:K15554]\n",
      "line in gene PA0185  ABC transporter permease [KO:K15554]\n",
      "            PA2307  ABC transporter permease [KO:K15554]\n",
      "line in gene PA2307  ABC transporter permease [KO:K15554]\n",
      "            PA2295  ABC transporter permease [KO:K15554]\n",
      "line in gene PA2295  ABC transporter permease [KO:K15554]\n",
      "            PA3442  aliphatic sulfonates ABC transporter ATP-binding subunit [KO:K15555] [EC:7.6.2.14]\n",
      "line in gene PA3442  aliphatic sulfonates ABC transporter ATP-binding subunit [KO:K15555] [EC:7.6.2.14]\n",
      "            PA0184  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "line in gene PA0184  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "            PA3447  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "line in gene PA3447  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "            PA2308  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "line in gene PA2308  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "            PA2294  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "line in gene PA2294  ABC transporter ATP-binding protein [KO:K15555] [EC:7.6.2.14]\n",
      "            PA1863  modA; molybdenum ABC transporter substrate-binding protein ModA [KO:K02020]\n",
      "line in gene PA1863  modA; molybdenum ABC transporter substrate-binding protein ModA [KO:K02020]\n",
      "            PA1862  modB; molybdenum ABC transporter permease ModB [KO:K02018]\n",
      "line in gene PA1862  modB; molybdenum ABC transporter permease ModB [KO:K02018]\n",
      "            PA1861  modC; molybdenum ABC transporter ATP-binding protein ModC [KO:K02017] [EC:7.3.2.5]\n",
      "line in gene PA1861  modC; molybdenum ABC transporter ATP-binding protein ModC [KO:K02017] [EC:7.3.2.5]\n",
      "            PA5217  iron ABC transporter substrate-binding protein [KO:K02012]\n",
      "line in gene PA5217  iron ABC transporter substrate-binding protein [KO:K02012]\n",
      "            PA4687  hitA; ferric iron-binding periplasmic protein HitA [KO:K02012]\n",
      "line in gene PA4687  hitA; ferric iron-binding periplasmic protein HitA [KO:K02012]\n",
      "            PA5216  iron ABC transporter substrate-binding protein [KO:K02011]\n",
      "line in gene PA5216  iron ABC transporter substrate-binding protein [KO:K02011]\n",
      "            PA4688  hitB; iron (III)-transporter permease HitB [KO:K02011]\n",
      "line in gene PA4688  hitB; iron (III)-transporter permease HitB [KO:K02011]\n",
      "            PA3538  ABC transporter ATP-binding protein [KO:K02010] [EC:7.2.2.7]\n",
      "line in gene PA3538  ABC transporter ATP-binding protein [KO:K02010] [EC:7.2.2.7]\n",
      "            PA3610  potD; polyamine ABC transporter substrate-binding protein PotD [KO:K11069]\n",
      "line in gene PA3610  potD; polyamine ABC transporter substrate-binding protein PotD [KO:K11069]\n",
      "            PA0323  ABC transporter [KO:K11069]\n",
      "line in gene PA0323  ABC transporter [KO:K11069]\n",
      "            PA0203  ABC transporter [KO:K11069]\n",
      "line in gene PA0203  ABC transporter [KO:K11069]\n",
      "            PA3609  potC; polyamine ABC transporter permease PotC [KO:K11070]\n",
      "line in gene PA3609  potC; polyamine ABC transporter permease PotC [KO:K11070]\n",
      "            PA0324  ABC transporter permease [KO:K11070]\n",
      "line in gene PA0324  ABC transporter permease [KO:K11070]\n",
      "            PA0204  ABC transporter permease [KO:K11070]\n",
      "line in gene PA0204  ABC transporter permease [KO:K11070]\n",
      "            PA3608  potB; polyamine ABC transporter permease PotB [KO:K11071]\n",
      "line in gene PA3608  potB; polyamine ABC transporter permease PotB [KO:K11071]\n",
      "            PA0325  ABC transporter permease [KO:K11071]\n",
      "line in gene PA0325  ABC transporter permease [KO:K11071]\n",
      "            PA0205  ABC transporter permease [KO:K11071]\n",
      "line in gene PA0205  ABC transporter permease [KO:K11071]\n",
      "            PA3607  potA; polyamine transporter ATP-binding protein PotA [KO:K11072] [EC:7.6.2.11]\n",
      "line in gene PA3607  potA; polyamine transporter ATP-binding protein PotA [KO:K11072] [EC:7.6.2.11]\n",
      "            PA0206  spermidine/putrescine ABC transporter ATP-binding protein [KO:K11072] [EC:7.6.2.11]\n",
      "line in gene PA0206  spermidine/putrescine ABC transporter ATP-binding protein [KO:K11072] [EC:7.6.2.11]\n",
      "            PA0326  ABC transporter ATP-binding protein [KO:K11072] [EC:7.6.2.11]\n",
      "line in gene PA0326  ABC transporter ATP-binding protein [KO:K11072] [EC:7.6.2.11]\n",
      "            PA0300  spuD; putrescine ABC transporter substrate-binding protein SpuD [KO:K11073]\n",
      "line in gene PA0300  spuD; putrescine ABC transporter substrate-binding protein SpuD [KO:K11073]\n",
      "            PA0301  spuE; spermidine ABC transporter substrate-binding protein SpuE [KO:K11073]\n",
      "line in gene PA0301  spuE; spermidine ABC transporter substrate-binding protein SpuE [KO:K11073]\n",
      "            PA1410  spermidine/putrescine-binding protein [KO:K11073]\n",
      "line in gene PA1410  spermidine/putrescine-binding protein [KO:K11073]\n",
      "            PA2711  spermidine/putrescine-binding protein [KO:K11073]\n",
      "line in gene PA2711  spermidine/putrescine-binding protein [KO:K11073]\n",
      "            PA2592  spermidine/putrescine-binding protein [KO:K11073]\n",
      "line in gene PA2592  spermidine/putrescine-binding protein [KO:K11073]\n",
      "            PA0295  polyamine binding protein [KO:K11073]\n",
      "line in gene PA0295  polyamine binding protein [KO:K11073]\n",
      "            PA0304  spuH; polyamine transporter PotI [KO:K11074]\n",
      "line in gene PA0304  spuH; polyamine transporter PotI [KO:K11074]\n",
      "            PA0303  spuG; polyamine transporter PotH [KO:K11075]\n",
      "line in gene PA0303  spuG; polyamine transporter PotH [KO:K11075]\n",
      "            PA0302  spuF; polyamine transporter PotG [KO:K11076] [EC:7.6.2.16]\n",
      "line in gene PA0302  spuF; polyamine transporter PotG [KO:K11076] [EC:7.6.2.16]\n",
      "            PA5096  ABC transporter [KO:K02002]\n",
      "line in gene PA5096  ABC transporter [KO:K02002]\n",
      "            PA0030  hypothetical protein [KO:K02002]\n",
      "line in gene PA0030  hypothetical protein [KO:K02002]\n",
      "            PA5103  hypothetical protein [KO:K02002]\n",
      "line in gene PA5103  hypothetical protein [KO:K02002]\n",
      "            PA5378  hypothetical protein [KO:K02002]\n",
      "line in gene PA5378  hypothetical protein [KO:K02002]\n",
      "            PA5388  hypothetical protein [KO:K02002]\n",
      "line in gene PA5388  hypothetical protein [KO:K02002]\n",
      "            PA3236  glycine betaine-binding protein [KO:K02002]\n",
      "line in gene PA3236  glycine betaine-binding protein [KO:K02002]\n",
      "            PA5095  ABC transporter permease [KO:K02001]\n",
      "line in gene PA5095  ABC transporter permease [KO:K02001]\n",
      "            PA5377  ABC transporter permease [KO:K02001]\n",
      "line in gene PA5377  ABC transporter permease [KO:K02001]\n",
      "            PA5094  ABC transporter ATP-binding protein [KO:K02000] [EC:7.6.2.9]\n",
      "line in gene PA5094  ABC transporter ATP-binding protein [KO:K02000] [EC:7.6.2.9]\n",
      "            PA5376  ABC transporter ATP-binding protein [KO:K02000] [EC:7.6.2.9]\n",
      "line in gene PA5376  ABC transporter ATP-binding protein [KO:K02000] [EC:7.6.2.9]\n",
      "            PA3889  ABC transporter [KO:K05845]\n",
      "line in gene PA3889  ABC transporter [KO:K05845]\n",
      "            PA3890  ABC transporter permease [KO:K05846]\n",
      "line in gene PA3890  ABC transporter permease [KO:K05846]\n",
      "            PA3888  ABC transporter permease [KO:K05846]\n",
      "line in gene PA3888  ABC transporter permease [KO:K05846]\n",
      "            PA3891  ABC transporter ATP-binding protein [KO:K05847] [EC:7.6.2.9]\n",
      "line in gene PA3891  ABC transporter ATP-binding protein [KO:K05847] [EC:7.6.2.9]\n",
      "            PA2341  maltose ABC transporter ATP-binding protein [KO:K10111] [EC:7.5.2.-]\n",
      "line in gene PA2341  maltose ABC transporter ATP-binding protein [KO:K10111] [EC:7.5.2.-]\n",
      "            PA3187  ABC transporter ATP-binding protein [KO:K10112] [EC:7.5.2.-]\n",
      "line in gene PA3187  ABC transporter ATP-binding protein [KO:K10112] [EC:7.5.2.-]\n",
      "            PA2338  maltose/mannitol ABC transporter substrate-binding protein [KO:K10227]\n",
      "line in gene PA2338  maltose/mannitol ABC transporter substrate-binding protein [KO:K10227]\n",
      "            PA2339  binding-protein-dependent maltose/mannitol transport protein [KO:K10228]\n",
      "line in gene PA2339  binding-protein-dependent maltose/mannitol transport protein [KO:K10228]\n",
      "            PA2340  binding-protein-dependent maltose/mannitol transport protein [KO:K10229]\n",
      "line in gene PA2340  binding-protein-dependent maltose/mannitol transport protein [KO:K10229]\n",
      "            PA4453  hypothetical protein [KO:K07323]\n",
      "line in gene PA4453  hypothetical protein [KO:K07323]\n",
      "            PA4454  hypothetical protein [KO:K02067]\n",
      "line in gene PA4454  hypothetical protein [KO:K02067]\n",
      "            PA3213  hypothetical protein [KO:K02067]\n",
      "line in gene PA3213  hypothetical protein [KO:K02067]\n",
      "            PA4455  ABC transporter permease [KO:K02066]\n",
      "line in gene PA4455  ABC transporter permease [KO:K02066]\n",
      "            PA3211  ABC transporter permease [KO:K02066]\n",
      "line in gene PA3211  ABC transporter permease [KO:K02066]\n",
      "            PA4452  hypothetical protein [KO:K07122]\n",
      "line in gene PA4452  hypothetical protein [KO:K07122]\n",
      "            PA4456  ABC transporter ATP-binding protein [KO:K02065]\n",
      "line in gene PA4456  ABC transporter ATP-binding protein [KO:K02065]\n",
      "            PA3212  ABC transporter ATP-binding protein [KO:K02065]\n",
      "line in gene PA3212  ABC transporter ATP-binding protein [KO:K02065]\n",
      "            PA0146  hypothetical protein [KO:K07335]\n",
      "line in gene PA0146  hypothetical protein [KO:K07335]\n",
      "            PA0137  ABC transporter permease [KO:K23535]\n",
      "line in gene PA0137  ABC transporter permease [KO:K23535]\n",
      "            PA0138  ABC transporter permease [KO:K23536]\n",
      "line in gene PA0138  ABC transporter permease [KO:K23536]\n",
      "            PA0136  ABC transporter ATP-binding protein [KO:K23537]\n",
      "line in gene PA0136  ABC transporter ATP-binding protein [KO:K23537]\n",
      "            PA3190  sugar ABC transporter substrate-binding protein [KO:K17315]\n",
      "line in gene PA3190  sugar ABC transporter substrate-binding protein [KO:K17315]\n",
      "            PA3189  sugar ABC transporter permease [KO:K17316]\n",
      "line in gene PA3189  sugar ABC transporter permease [KO:K17316]\n",
      "            PA3188  sugar ABC transporter permease [KO:K17317]\n",
      "line in gene PA3188  sugar ABC transporter permease [KO:K17317]\n",
      "            PA1946  rbsB; ribose ABC transporter substrate-binding protein [KO:K10439]\n",
      "line in gene PA1946  rbsB; ribose ABC transporter substrate-binding protein [KO:K10439]\n",
      "            PA1948  rbsC; ribose ABC transporter permease [KO:K10440]\n",
      "line in gene PA1948  rbsC; ribose ABC transporter permease [KO:K10440]\n",
      "            PA1947  rbsA; ribose transporter RbsA [KO:K10441] [EC:7.5.2.7]\n",
      "line in gene PA1947  rbsA; ribose transporter RbsA [KO:K10441] [EC:7.5.2.7]\n",
      "            PA2548  hypothetical protein [KO:K02040]\n",
      "line in gene PA2548  hypothetical protein [KO:K02040]\n",
      "            PA5369  pstS; phosphate ABC transporter substrate-binding protein [KO:K02040]\n",
      "line in gene PA5369  pstS; phosphate ABC transporter substrate-binding protein [KO:K02040]\n",
      "            PA5368  pstC; phosphate ABC transporter permease [KO:K02037]\n",
      "line in gene PA5368  pstC; phosphate ABC transporter permease [KO:K02037]\n",
      "            PA5367  pstA; phosphate ABC transporter permease [KO:K02038]\n",
      "line in gene PA5367  pstA; phosphate ABC transporter permease [KO:K02038]\n",
      "            PA5366  pstB; phosphate ABC transporter ATP-binding protein [KO:K02036] [EC:7.3.2.1]\n",
      "line in gene PA5366  pstB; phosphate ABC transporter ATP-binding protein [KO:K02036] [EC:7.3.2.1]\n",
      "            PA3313  hypothetical protein [KO:K02044]\n",
      "line in gene PA3313  hypothetical protein [KO:K02044]\n",
      "            PA3383  phosphonate ABC transporter substrate-binding protein [KO:K02044]\n",
      "line in gene PA3383  phosphonate ABC transporter substrate-binding protein [KO:K02044]\n",
      "            PA3315  phosphonates ABC transporter permease [KO:K02042]\n",
      "line in gene PA3315  phosphonates ABC transporter permease [KO:K02042]\n",
      "            PA3316  ABC transporter permease [KO:K02042]\n",
      "line in gene PA3316  ABC transporter permease [KO:K02042]\n",
      "            PA3382  phnE; phosphonate transporter PhnE [KO:K02042]\n",
      "line in gene PA3382  phnE; phosphonate transporter PhnE [KO:K02042]\n",
      "            PA3314  phosphonates ABC transporter ATP-binding protein [KO:K02041] [EC:7.3.2.2]\n",
      "line in gene PA3314  phosphonates ABC transporter ATP-binding protein [KO:K02041] [EC:7.3.2.2]\n",
      "            PA3384  phnC; phosphonate ABC transporter ATP-binding protein [KO:K02041] [EC:7.3.2.2]\n",
      "line in gene PA3384  phnC; phosphonate ABC transporter ATP-binding protein [KO:K02041] [EC:7.3.2.2]\n",
      "            PA2925  hisM; histidine ABC transporter permease HisM [KO:K10015]\n",
      "line in gene PA2925  hisM; histidine ABC transporter permease HisM [KO:K10015]\n",
      "            PA2924  hisQ; histidine ABC transporter permease HisQ [KO:K10016]\n",
      "line in gene PA2924  hisQ; histidine ABC transporter permease HisQ [KO:K10016]\n",
      "            PA2926  hisP; histidine ABC transporter ATP-binding protein HisP [KO:K10017] [EC:7.4.2.1]\n",
      "line in gene PA2926  hisP; histidine ABC transporter ATP-binding protein HisP [KO:K10017] [EC:7.4.2.1]\n",
      "            PA2923  hisJ; histidine ABC transporter substrate-binding protein HisJ [KO:K10014]\n",
      "line in gene PA2923  hisJ; histidine ABC transporter substrate-binding protein HisJ [KO:K10014]\n",
      "            PA1342  ABC transporter [KO:K10001]\n",
      "line in gene PA1342  ABC transporter [KO:K10001]\n",
      "            PA5082  ABC transporter [KO:K10001]\n",
      "line in gene PA5082  ABC transporter [KO:K10001]\n",
      "            PA1340  amino acid ABC transporter permease [KO:K10002]\n",
      "line in gene PA1340  amino acid ABC transporter permease [KO:K10002]\n",
      "            PA1341  amino acid ABC transporter permease [KO:K10003]\n",
      "line in gene PA1341  amino acid ABC transporter permease [KO:K10003]\n",
      "            PA1339  amino acid ABC transporter ATP binding protein [KO:K10004] [EC:7.4.2.1]\n",
      "line in gene PA1339  amino acid ABC transporter ATP binding protein [KO:K10004] [EC:7.4.2.1]\n",
      "            PA3858  amino acid-binding protein [KO:K09969]\n",
      "line in gene PA3858  amino acid-binding protein [KO:K09969]\n",
      "            PA0314  L-cysteine ABC transporter protein FliY [KO:K02424]\n",
      "line in gene PA0314  L-cysteine ABC transporter protein FliY [KO:K02424]\n",
      "            PA0313  L-cysteine ABC transporter protein YecS [KO:K10009]\n",
      "line in gene PA0313  L-cysteine ABC transporter protein YecS [KO:K10009]\n",
      "            PA0888  aotJ; arginine/ornithine ABC transporter substrate-binding protein AotJ [KO:K10022]\n",
      "line in gene PA0888  aotJ; arginine/ornithine ABC transporter substrate-binding protein AotJ [KO:K10022]\n",
      "            PA3865  amino acid binding protein [KO:K10022]\n",
      "line in gene PA3865  amino acid binding protein [KO:K10022]\n",
      "            PA0890  aotM; arginine/ornithine ABC transporter permease AotM [KO:K10023]\n",
      "line in gene PA0890  aotM; arginine/ornithine ABC transporter permease AotM [KO:K10023]\n",
      "            PA0889  aotQ; arginine/ornithine ABC transporter permease AotQ [KO:K10024]\n",
      "line in gene PA0889  aotQ; arginine/ornithine ABC transporter permease AotQ [KO:K10024]\n",
      "            PA0892  aotP; arginine/ornithine ABC transporter ATP-binding protein AotP [KO:K10025] [EC:7.4.2.1]\n",
      "line in gene PA0892  aotP; arginine/ornithine ABC transporter ATP-binding protein AotP [KO:K10025] [EC:7.4.2.1]\n",
      "            PA5153  amino acid ABC transporter substrate-binding protein [KO:K23055]\n",
      "line in gene PA5153  amino acid ABC transporter substrate-binding protein [KO:K23055]\n",
      "            PA5154  ABC transporter permease [KO:K23056]\n",
      "line in gene PA5154  ABC transporter permease [KO:K23056]\n",
      "            PA5155  amino acid ABC transporter permease [KO:K23057]\n",
      "line in gene PA5155  amino acid ABC transporter permease [KO:K23057]\n",
      "            PA5152  ABC transporter ATP-binding protein [KO:K23058] [EC:7.4.2.1]\n",
      "line in gene PA5152  ABC transporter ATP-binding protein [KO:K23058] [EC:7.4.2.1]\n",
      "            PA1260  amino acid ABC transporter substrate-binding protein [KO:K23061]\n",
      "line in gene PA1260  amino acid ABC transporter substrate-binding protein [KO:K23061]\n",
      "            PA1258  ABC transporter permease [KO:K23062]\n",
      "line in gene PA1258  ABC transporter permease [KO:K23062]\n",
      "            PA1257  amino acid ABC transporter permease [KO:K23063]\n",
      "line in gene PA1257  amino acid ABC transporter permease [KO:K23063]\n",
      "            PA1256  amino acid ABC transporter ATP binding protein [KO:K23064] [EC:7.4.2.1]\n",
      "line in gene PA1256  amino acid ABC transporter ATP binding protein [KO:K23064] [EC:7.4.2.1]\n",
      "            PA1074  braC; branched-chain amino acid ABC transporter substrate-binding protein BraC [KO:K01999]\n",
      "line in gene PA1074  braC; branched-chain amino acid ABC transporter substrate-binding protein BraC [KO:K01999]\n",
      "            PA4913  ABC transporter [KO:K01999]\n",
      "line in gene PA4913  ABC transporter [KO:K01999]\n",
      "            PA3364  amiC; aliphatic amidase expression-regulating protein [KO:K01999]\n",
      "line in gene PA3364  amiC; aliphatic amidase expression-regulating protein [KO:K01999]\n",
      "            PA1073  braD; branched-chain amino acid ABC transporter permease BraD [KO:K01997]\n",
      "line in gene PA1073  braD; branched-chain amino acid ABC transporter permease BraD [KO:K01997]\n",
      "            PA4912  branched-chain amino acid ABC transporter [KO:K01997]\n",
      "line in gene PA4912  branched-chain amino acid ABC transporter [KO:K01997]\n",
      "            PA1072  braE; branched-chain amino acid ABC transporter permease BraE [KO:K01998]\n",
      "line in gene PA1072  braE; branched-chain amino acid ABC transporter permease BraE [KO:K01998]\n",
      "            PA4911  branched-chain amino acid ABC transporter permease [KO:K01998]\n",
      "line in gene PA4911  branched-chain amino acid ABC transporter permease [KO:K01998]\n",
      "            PA1071  braF; ABC transporter ATP-binding protein [KO:K01995]\n",
      "line in gene PA1071  braF; ABC transporter ATP-binding protein [KO:K01995]\n",
      "            PA4910  ABC transporter ATP-binding protein [KO:K01995]\n",
      "line in gene PA4910  ABC transporter ATP-binding protein [KO:K01995]\n",
      "            PA1070  braG; ABC transporter ATP-binding protein [KO:K01996]\n",
      "line in gene PA1070  braG; ABC transporter ATP-binding protein [KO:K01996]\n",
      "            PA4909  ABC transporter ATP-binding protein [KO:K01996]\n",
      "line in gene PA4909  ABC transporter ATP-binding protein [KO:K01996]\n",
      "            PA5505  TonB-dependent receptor [KO:K02073]\n",
      "line in gene PA5505  TonB-dependent receptor [KO:K02073]\n",
      "            PA3931  hypothetical protein [KO:K02073]\n",
      "line in gene PA3931  hypothetical protein [KO:K02073]\n",
      "            PA2349  hypothetical protein [KO:K02073]\n",
      "line in gene PA2349  hypothetical protein [KO:K02073]\n",
      "            PA2351  ABC transporter permease [KO:K02072]\n",
      "line in gene PA2351  ABC transporter permease [KO:K02072]\n",
      "            PA5504  D-methionine ABC transporter [KO:K02072]\n",
      "line in gene PA5504  D-methionine ABC transporter [KO:K02072]\n",
      "            PA5503  ABC transporter ATP-binding protein [KO:K02071]\n",
      "line in gene PA5503  ABC transporter ATP-binding protein [KO:K02071]\n",
      "            PA2350  methionine ABC transporter ATP-binding protein [KO:K02071]\n",
      "line in gene PA2350  methionine ABC transporter ATP-binding protein [KO:K02071]\n",
      "            PA4858  hypothetical protein [KO:K11959]\n",
      "line in gene PA4858  hypothetical protein [KO:K11959]\n",
      "            PA4859  ABC transporter permease [KO:K11960]\n",
      "line in gene PA4859  ABC transporter permease [KO:K11960]\n",
      "            PA4860  ABC transporter permease [KO:K11961]\n",
      "line in gene PA4860  ABC transporter permease [KO:K11961]\n",
      "            PA4861  ABC transporter ATP-binding protein [KO:K11962]\n",
      "line in gene PA4861  ABC transporter ATP-binding protein [KO:K11962]\n",
      "            PA4862  ABC transporter ATP-binding protein [KO:K11963]\n",
      "line in gene PA4862  ABC transporter ATP-binding protein [KO:K11963]\n",
      "            PA4496  ABC transporter [KO:K12368]\n",
      "line in gene PA4496  ABC transporter [KO:K12368]\n",
      "            PA4500  ABC transporter [KO:K12368]\n",
      "line in gene PA4500  ABC transporter [KO:K12368]\n",
      "            PA4502  ABC transporter [KO:K12368]\n",
      "line in gene PA4502  ABC transporter [KO:K12368]\n",
      "            PA5317  dipeptide ABC transporter substrate-binding protein [KO:K12368]\n",
      "line in gene PA5317  dipeptide ABC transporter substrate-binding protein [KO:K12368]\n",
      "            PA4497  ABC transporter [KO:K12368]\n",
      "line in gene PA4497  ABC transporter [KO:K12368]\n",
      "            PA4503  ABC transporter permease [KO:K12369]\n",
      "line in gene PA4503  ABC transporter permease [KO:K12369]\n",
      "            PA4504  ABC transporter permease [KO:K12370]\n",
      "line in gene PA4504  ABC transporter permease [KO:K12370]\n",
      "            PA4505  ABC transporter ATP-binding protein [KO:K12371]\n",
      "line in gene PA4505  ABC transporter ATP-binding protein [KO:K12371]\n",
      "            PA4506  peptide ABC transporter ATP-binding protein [KO:K12372]\n",
      "line in gene PA4506  peptide ABC transporter ATP-binding protein [KO:K12372]\n",
      "            PA2058  ABC transporter [KO:K13893]\n",
      "line in gene PA2058  ABC transporter [KO:K13893]\n",
      "            PA1810  ABC transporter [KO:K13893]\n",
      "line in gene PA1810  ABC transporter [KO:K13893]\n",
      "            PA1811  solute-binding protein [KO:K13893]\n",
      "line in gene PA1811  solute-binding protein [KO:K13893]\n",
      "            PA2059  ABC transporter permease [KO:K13894]\n",
      "line in gene PA2059  ABC transporter permease [KO:K13894]\n",
      "            PA1809  ABC transporter permease [KO:K13894]\n",
      "line in gene PA1809  ABC transporter permease [KO:K13894]\n",
      "            PA1808  ABC transporter permease [KO:K13895]\n",
      "line in gene PA1808  ABC transporter permease [KO:K13895]\n",
      "            PA2060  ABC transporter permease [KO:K13895]\n",
      "line in gene PA2060  ABC transporter permease [KO:K13895]\n",
      "            PA2061  ABC transporter ATP-binding protein [KO:K13896]\n",
      "line in gene PA2061  ABC transporter ATP-binding protein [KO:K13896]\n",
      "            PA1807  ABC transporter ATP-binding protein [KO:K13896]\n",
      "line in gene PA1807  ABC transporter ATP-binding protein [KO:K13896]\n",
      "            PA4159  fepB; iron-enterobactin transporter periplasmic binding protein [KO:K23185]\n",
      "line in gene PA4159  fepB; iron-enterobactin transporter periplasmic binding protein [KO:K23185]\n",
      "            PA4160  fepD; ferric enterobactin transporter FepD [KO:K23186]\n",
      "line in gene PA4160  fepD; ferric enterobactin transporter FepD [KO:K23186]\n",
      "            PA4161  fepG; ferric enterobactin transporter FepG [KO:K23187]\n",
      "line in gene PA4161  fepG; ferric enterobactin transporter FepG [KO:K23187]\n",
      "            PA4158  fepC; ferric enterobactin transporter FepC [KO:K23188] [EC:7.2.2.17 7.2.2.-]\n",
      "line in gene PA4158  fepC; ferric enterobactin transporter FepC [KO:K23188] [EC:7.2.2.17 7.2.2.-]\n",
      "            PA4045  hypothetical protein [KO:K06858]\n",
      "line in gene PA4045  hypothetical protein [KO:K06858]\n",
      "            PA2407  adhesion protein [KO:K09815]\n",
      "line in gene PA2407  adhesion protein [KO:K09815]\n",
      "            PA5498  adhesin [KO:K09815]\n",
      "line in gene PA5498  adhesin [KO:K09815]\n",
      "            PA2409  ABC transporter permease [KO:K09816]\n",
      "line in gene PA2409  ABC transporter permease [KO:K09816]\n",
      "            PA5501  znuB; zinc ABC transporter permease [KO:K09816]\n",
      "line in gene PA5501  znuB; zinc ABC transporter permease [KO:K09816]\n",
      "            PA2408  ABC transporter ATP-binding protein [KO:K09817] [EC:7.2.2.20]\n",
      "line in gene PA2408  ABC transporter ATP-binding protein [KO:K09817] [EC:7.2.2.20]\n",
      "            PA5500  znuC; zinc transporter ZnuC [KO:K09817] [EC:7.2.2.20]\n",
      "line in gene PA5500  znuC; zinc transporter ZnuC [KO:K09817] [EC:7.2.2.20]\n",
      "            PA5451  wzm; LPS efflux transporter membrane protein [KO:K09690]\n",
      "line in gene PA5451  wzm; LPS efflux transporter membrane protein [KO:K09690]\n",
      "            PA2678  ABC transporter permease [KO:K09690]\n",
      "line in gene PA2678  ABC transporter permease [KO:K09690]\n",
      "            PA5450  wzt; ABC transporter [KO:K09691]\n",
      "line in gene PA5450  wzt; ABC transporter [KO:K09691]\n",
      "            PA1386  ABC transporter ATP-binding protein [KO:K09691]\n",
      "line in gene PA1386  ABC transporter ATP-binding protein [KO:K09691]\n",
      "            PA2986  hypothetical protein [KO:K09808]\n",
      "line in gene PA2986  hypothetical protein [KO:K09808]\n",
      "            PA2988  hypothetical protein [KO:K09808]\n",
      "line in gene PA2988  hypothetical protein [KO:K09808]\n",
      "            PA2987  lipoprotein-releasing system ABC transporter ATP-binding protein [KO:K09810] [EC:7.6.2.-]\n",
      "line in gene PA2987  lipoprotein-releasing system ABC transporter ATP-binding protein [KO:K09810] [EC:7.6.2.-]\n",
      "            PA1478  heme exporter protein CcmD [KO:K02196]\n",
      "line in gene PA1478  heme exporter protein CcmD [KO:K02196]\n",
      "            PA1477  ccmC; heme exporter protein CcmC [KO:K02195]\n",
      "line in gene PA1477  ccmC; heme exporter protein CcmC [KO:K02195]\n",
      "            PA1476  ccmB; heme exporter protein CcmB [KO:K02194]\n",
      "line in gene PA1476  ccmB; heme exporter protein CcmB [KO:K02194]\n",
      "            PA1475  ccmA; cytochrome c biogenesis ATP-binding export protein CcmA [KO:K02193] [EC:7.6.2.5]\n",
      "line in gene PA1475  ccmA; cytochrome c biogenesis ATP-binding export protein CcmA [KO:K02193] [EC:7.6.2.5]\n",
      "            PA3828  hypothetical protein [KO:K07091]\n",
      "line in gene PA3828  hypothetical protein [KO:K07091]\n",
      "            PA3827  hypothetical protein [KO:K11720]\n",
      "line in gene PA3827  hypothetical protein [KO:K11720]\n",
      "            PA4461  ABC transporter ATP-binding protein [KO:K06861] [EC:7.5.2.5]\n",
      "line in gene PA4461  ABC transporter ATP-binding protein [KO:K06861] [EC:7.5.2.5]\n",
      "            PA3395  nosY; membrane protein NosY [KO:K19341]\n",
      "line in gene PA3395  nosY; membrane protein NosY [KO:K19341]\n",
      "            PA3394  nosF; copper ABC transporter ATP-binding protein [KO:K19340]\n",
      "line in gene PA3394  nosF; copper ABC transporter ATP-binding protein [KO:K19340]\n",
      "            PA0375  ftsX; cell division protein FtsX [KO:K09811]\n",
      "line in gene PA0375  ftsX; cell division protein FtsX [KO:K09811]\n",
      "            PA0374  ftsE; cell division protein FtsE [KO:K09812]\n",
      "line in gene PA0374  ftsE; cell division protein FtsE [KO:K09812]\n",
      "            PA4997  msbA; transporter MsbA [KO:K11085] [EC:7.5.2.6]\n",
      "line in gene PA4997  msbA; transporter MsbA [KO:K11085] [EC:7.5.2.6]\n",
      "            PA4143  toxin transporter [KO:K13409]\n",
      "line in gene PA4143  toxin transporter [KO:K13409]\n",
      "            PA3228  ABC transporter ATP-binding protein/permease [KO:K18893]\n",
      "line in gene PA3228  ABC transporter ATP-binding protein/permease [KO:K18893]\n",
      "            PA1876  ABC transporter ATP-binding protein/permease [KO:K12541]\n",
      "line in gene PA1876  ABC transporter ATP-binding protein/permease [KO:K12541]\n",
      "            PA3406  hasD; transporter HasD [KO:K12536]\n",
      "line in gene PA3406  hasD; transporter HasD [KO:K12536]\n",
      "            PA1246  aprD; alkaline protease secretion ATP-binding protein AprD [KO:K12536]\n",
      "line in gene PA1246  aprD; alkaline protease secretion ATP-binding protein AprD [KO:K12536]\n",
      "            PA2390  pvdT; pyoverdine biosynthesis protein PvdT [KO:K05685] [EC:7.6.2.-]\n",
      "line in gene PA2390  pvdT; pyoverdine biosynthesis protein PvdT [KO:K05685] [EC:7.6.2.-]\n",
      "            PA2397  pvdE; pyoverdine biosynthesis protein PvdE [KO:K06160]\n",
      "line in gene PA2397  pvdE; pyoverdine biosynthesis protein PvdE [KO:K06160]\n",
      "COMPOUND    C00009  Orthophosphate\n",
      "            C00025  L-Glutamate\n",
      "            C00031  D-Glucose\n",
      "            C00032  Heme\n",
      "            C00034  Manganese\n",
      "            C00037  Glycine\n",
      "            C00038  Zinc cation\n",
      "            C00041  L-Alanine\n",
      "            C00047  L-Lysine\n",
      "            C00049  L-Aspartate\n",
      "            C00051  Glutathione\n",
      "            C00059  Sulfate\n",
      "            C00062  L-Arginine\n",
      "            C00064  L-Glutamine\n",
      "            C00065  L-Serine\n",
      "            C00070  Copper\n",
      "            C00077  L-Ornithine\n",
      "            C00079  L-Phenylalanine\n",
      "            C00086  Urea\n",
      "            C00088  Nitrite\n",
      "            C00089  Sucrose\n",
      "            C00093  sn-Glycerol 3-phosphate\n",
      "            C00095  D-Fructose\n",
      "            C00098  Oligopeptide\n",
      "            C00107  Dipeptide\n",
      "            C00114  Choline\n",
      "            C00116  Glycerol\n",
      "            C00120  Biotin\n",
      "            C00121  D-Ribose\n",
      "            C00123  L-Leucine\n",
      "            C00134  Putrescine\n",
      "            C00135  L-Histidine\n",
      "            C00137  myo-Inositol\n",
      "            C00140  N-Acetyl-D-glucosamine\n",
      "            C00148  L-Proline\n",
      "            C00151  L-Amino acid\n",
      "            C00159  D-Mannose\n",
      "            C00175  Cobalt ion\n",
      "            C00181  D-Xylose\n",
      "            C00183  L-Valine\n",
      "            C00185  Cellobiose\n",
      "            C00188  L-Threonine\n",
      "            C00208  Maltose\n",
      "            C00212  Adenosine\n",
      "            C00243  Lactose\n",
      "            C00244  Nitrate\n",
      "            C00245  Taurine\n",
      "            C00255  Riboflavin\n",
      "            C00259  L-Arabinose\n",
      "            C00288  HCO3-\n",
      "            C00291  Nickel\n",
      "            C00294  Inosine\n",
      "            C00299  Uridine\n",
      "            C00315  Spermidine\n",
      "            C00320  Thiosulfate\n",
      "            C00330  Deoxyguanosine\n",
      "            C00333  D-Galacturonate\n",
      "            C00338  Lipopolysaccharide\n",
      "            C00378  Thiamine\n",
      "            C00379  Xylitol\n",
      "            C00387  Guanosine\n",
      "            C00392  Mannitol\n",
      "            C00407  L-Isoleucine\n",
      "            C00430  5-Aminolevulinate\n",
      "            C00470  Pectate\n",
      "            C00475  Cytidine\n",
      "            C00487  Carnitine\n",
      "            C00491  L-Cystine\n",
      "            C00492  Raffinose\n",
      "            C00503  Erythritol\n",
      "            C00526  Deoxyuridine\n",
      "            C00559  Deoxyadenosine\n",
      "            C00719  Betaine\n",
      "            C00794  D-Sorbitol\n",
      "            C00855  D-Methionine\n",
      "            C00865  Phospholipid\n",
      "            C00881  Deoxycytidine\n",
      "            C00919  Choline sulfate\n",
      "            C00973  Cyclomaltodextrin\n",
      "            C01083  alpha,alpha-Trehalose\n",
      "            C01153  Orthophosphoric monoester\n",
      "            C01157  Hydroxyproline\n",
      "            C01177  Inositol 1-phosphate\n",
      "            C01181  4-Trimethylammoniobutanoate\n",
      "            C01279  4-Amino-5-hydroxymethyl-2-methylpyrimidine\n",
      "            C01330  Sodium cation\n",
      "            C01417  Cyanate\n",
      "            C01487  D-Allose\n",
      "            C01606  Phthalate\n",
      "            C01630  Xylobiose\n",
      "            C01667  Bacitracin\n",
      "            C01674  Chitobiose\n",
      "            C01682  Nopaline\n",
      "            C01684  D-Rhamnose\n",
      "            C01762  Xanthosine\n",
      "            C01834  Lipoprotein\n",
      "            C01835  Maltotriose\n",
      "            C01935  Maltodextrin\n",
      "            C01946  Oleandomycin\n",
      "            C02160  Isomaltotriose\n",
      "            C02273  Digalacturonate\n",
      "            C03557  2-Aminoethylphosphonate\n",
      "            C03611  Galactose oligosaccharide\n",
      "            C03619  Methyl beta-D-galactoside\n",
      "            C04114  (E)-4-(Trimethylammonio)but-2-enoate\n",
      "            C04137  Octopine\n",
      "            C05349  Ciprofloxacin\n",
      "            C05402  Melibiose\n",
      "            C05512  Deoxyinosine\n",
      "            C05776  Vitamin B12\n",
      "            C06227  Fe(III)hydroxamate\n",
      "            C06229  Fe(III)dicitrate\n",
      "            C06230  Fe-enterobactin\n",
      "            C06232  Molybdate\n",
      "            C06687  Norfloxacin\n",
      "            C06705  Capsular polysaccharide\n",
      "            C06706  Lipo-oligosaccharide\n",
      "            C06707  Teichoic acid\n",
      "            C06767  Hemine\n",
      "            C07662  Sparfloxacin\n",
      "            C07663  Moxifloxacin\n",
      "            C11612  Polymyxin B\n",
      "            C13768  Colistin sulfate\n",
      "            C14818  Fe2+\n",
      "            C14819  Fe3+\n",
      "            C15521  Alkanesulfonate\n",
      "            C15719  Coelichelin\n",
      "            C16421  AI-2\n",
      "            C16692  Mannopine\n",
      "            C19609  Nickel(2+)\n",
      "            C19872  N-Formyl-4-amino-5-aminomethyl-2-methylpyrimidine\n",
      "            C20570  alpha-1,5-L-Arabinobiose\n",
      "            C20571  alpha-1,5-L-Arabinotriose\n",
      "            C20572  alpha-1,5-L-Arabinotetraose\n",
      "            C20573  Aldotetraouronic acid\n",
      "            C20679  Tungstate\n",
      "            C21066  D-Galactofuranose\n",
      "            C22040  S-Methyl-L-cysteine\n",
      "            G00457  Maltotetraose\n",
      "REFERENCE   PMID:9799792\n",
      "  AUTHORS   Tomii K, Kanehisa M.\n",
      "  TITLE     A comparative analysis of ABC transporters in complete microbial genomes.\n",
      "  JOURNAL   Genome Res 8:1048-59 (1998)\n",
      "            DOI:10.1101/gr.8.10.1048\n",
      "REFERENCE   PMID:18424520\n",
      "  AUTHORS   Sperandeo P, Lau FK, Carpentieri A, De Castro C, Molinaro A, Deho G, Silhavy TJ, Polissi A\n",
      "  TITLE     Functional analysis of the protein machinery required for transport of lipopolysaccharide to the outer membrane of Escherichia coli.\n",
      "  JOURNAL   J Bacteriol 190:4460-9 (2008)\n",
      "            DOI:10.1128/JB.00270-08\n",
      "REFERENCE   PMID:18375759\n",
      "  AUTHORS   Ruiz N, Gronenberg LS, Kahne D, Silhavy TJ\n",
      "  TITLE     Identification of two inner-membrane proteins required for the transport of lipopolysaccharide to the outer membrane of Escherichia coli.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 105:5537-42 (2008)\n",
      "            DOI:10.1073/pnas.0801196105\n",
      "REFERENCE   PMID:10557362\n",
      "  AUTHORS   Omata T, Price GD, Badger MR, Okamura M, Gohta S, Ogawa T\n",
      "  TITLE     Identification of an ATP-binding cassette transporter involved in bicarbonate uptake in the cyanobacterium Synechococcus sp. strain PCC 7942.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 96:13571-6 (1999)\n",
      "            DOI:10.1073/pnas.96.23.13571\n",
      "REFERENCE   PMID:11731493\n",
      "  AUTHORS   Quintero MJ, Montesinos ML, Herrero A, Flores E\n",
      "  TITLE     Identification of genes encoding amino acid permeases by inactivation of selected ORFs from the Synechocystis genomic sequence.\n",
      "  JOURNAL   Genome Res 11:2034-40 (2001)\n",
      "            DOI:10.1101/gr.196301\n",
      "REFERENCE   PMID:16135226\n",
      "  AUTHORS   Picossi S, Montesinos ML, Pernil R, Lichtle C, Herrero A, Flores E\n",
      "  TITLE     ABC-type neutral amino acid permease N-I is required for optimal diazotrophic growth and is repressed in the heterocysts of Anabaena sp. strain PCC 7120.\n",
      "  JOURNAL   Mol Microbiol 57:1582-92 (2005)\n",
      "            DOI:10.1111/j.1365-2958.2005.04791.x\n",
      "REFERENCE   PMID:11929526\n",
      "  AUTHORS   Valladares A, Montesinos ML, Herrero A, Flores E.\n",
      "  TITLE     An ABC-type, high-affinity urea permease identified in cyanobacteria.\n",
      "  JOURNAL   Mol Microbiol 43:703-15 (2002)\n",
      "            DOI:10.1046/j.1365-2958.2002.02778.x\n",
      "REFERENCE   PMID:18326572\n",
      "  AUTHORS   Richard-Fogal CL, Frawley ER, Kranz RG\n",
      "  TITLE     Topology and function of CcmD in cytochrome c maturation.\n",
      "  JOURNAL   J Bacteriol 190:3489-93 (2008)\n",
      "            DOI:10.1128/JB.00146-08\n",
      "REFERENCE   PMID:16109926\n",
      "  AUTHORS   Suzuki H, Koyanagi T, Izuka S, Onishi A, Kumagai H\n",
      "  TITLE     The yliA, -B, -C, and -D genes of Escherichia coli K-12 encode a novel glutathione importer with an ATP-binding cassette.\n",
      "  JOURNAL   J Bacteriol 187:5861-7 (2005)\n",
      "            DOI:10.1128/JB.187.17.5861-5867.2005\n",
      "REFERENCE   PMID:17873039\n",
      "  AUTHORS   Novikova M, Metlitskaya A, Datsenko K, Kazakov T, Kazakov A, Wanner B, Severinov K\n",
      "  TITLE     The Escherichia coli Yej transporter is required for the uptake of translation inhibitor microcin C.\n",
      "  JOURNAL   J Bacteriol 189:8361-5 (2007)\n",
      "            DOI:10.1128/JB.01028-07\n",
      "REFERENCE   PMID:17214741\n",
      "  AUTHORS   Tikhonova EB, Devroy VK, Lau SY, Zgurskaya HI\n",
      "  TITLE     Reconstitution of the Escherichia coli macrolide transporter: the periplasmic membrane fusion protein MacA stimulates the ATPase activity of MacB.\n",
      "  JOURNAL   Mol Microbiol 63:895-910 (2007)\n",
      "            DOI:10.1111/j.1365-2958.2006.05549.x\n",
      "REFERENCE   PMID:18805970\n",
      "  AUTHORS   Yamanaka H, Kobayashi H, Takahashi E, Okamoto K\n",
      "  TITLE     MacAB is involved in the secretion of Escherichia coli heat-stable enterotoxin II.\n",
      "  JOURNAL   J Bacteriol 190:7693-8 (2008)\n",
      "            DOI:10.1128/JB.00853-08\n",
      "KO_PATHWAY  ko02010\n",
      "///\n",
      "path:pae02020\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae02020                    Pathway\n",
      "NAME        Two-component system - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Two-component signal transduction systems enable bacteria to sense, respond, and adapt to changes in their environment or in their intracellular state. Each two-component system consists of a sensor protein-histidine kinase (HK) and a response regulator (RR). In the prototypical two-component pathway, the sensor HK phosphorylates its own conserved His residue in response to a signal(s) in the environment. Subsequently, the phosphoryl group of HK is transferred onto a specific Asp residue on the RR. The activated RR can then effect changes in cellular physiology, often by regulating gene expression. Two-component pathways thus often enable cells to sense and respond to stimuli by inducing changes in transcription.\n",
      "CLASS       Environmental Information Processing; Signal transduction\n",
      "PATHWAY_MAP pae02020  Two-component system\n",
      "DBLINKS     GO: 0000160\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA5361  phoR; two-component sensor PhoR [KO:K07636] [EC:2.7.13.3]\n",
      "line in gene PA5361  phoR; two-component sensor PhoR [KO:K07636] [EC:2.7.13.3]\n",
      "            PA5360  phoB; two-component response regulator PhoB [KO:K07657]\n",
      "line in gene PA5360  phoB; two-component response regulator PhoB [KO:K07657]\n",
      "            PA3296  phoA; alkaline phosphatase [KO:K01077] [EC:3.1.3.1]\n",
      "line in gene PA3296  phoA; alkaline phosphatase [KO:K01077] [EC:3.1.3.1]\n",
      "            PA3910  eddA; extracelullar DNA degradation protein EddA [KO:K01113] [EC:3.1.3.1]\n",
      "line in gene PA3910  eddA; extracelullar DNA degradation protein EddA [KO:K01113] [EC:3.1.3.1]\n",
      "            PA2548  hypothetical protein [KO:K02040]\n",
      "line in gene PA2548  hypothetical protein [KO:K02040]\n",
      "            PA5369  pstS; phosphate ABC transporter substrate-binding protein [KO:K02040]\n",
      "line in gene PA5369  pstS; phosphate ABC transporter substrate-binding protein [KO:K02040]\n",
      "            PA1180  phoQ; two-component sensor PhoQ [KO:K07637] [EC:2.7.13.3]\n",
      "line in gene PA1180  phoQ; two-component sensor PhoQ [KO:K07637] [EC:2.7.13.3]\n",
      "            PA1179  phoP; two-component response regulator PhoP [KO:K07660]\n",
      "line in gene PA1179  phoP; two-component response regulator PhoP [KO:K07660]\n",
      "            PA2101  hypothetical protein [KO:K07790]\n",
      "line in gene PA2101  hypothetical protein [KO:K07790]\n",
      "            PA0190  acid phosphatase [KO:K09474] [EC:3.1.3.2]\n",
      "line in gene PA0190  acid phosphatase [KO:K09474] [EC:3.1.3.2]\n",
      "            PA5199  amgS; protein AmgS [KO:K07638] [EC:2.7.13.3]\n",
      "line in gene PA5199  amgS; protein AmgS [KO:K07638] [EC:2.7.13.3]\n",
      "            PA5200  amgR; osmolarity response regulator [KO:K07659]\n",
      "line in gene PA5200  amgR; osmolarity response regulator [KO:K07659]\n",
      "            PA1158  two-component sensor [KO:K07639] [EC:2.7.13.3]\n",
      "line in gene PA1158  two-component sensor [KO:K07639] [EC:2.7.13.3]\n",
      "            PA1157  two-component response regulator [KO:K07661]\n",
      "line in gene PA1157  two-component response regulator [KO:K07661]\n",
      "            PA3204  two-component response regulator [KO:K07662]\n",
      "line in gene PA3204  two-component response regulator [KO:K07662]\n",
      "            PA0766  mucD; serine protease MucD [KO:K04771] [EC:3.4.21.107]\n",
      "line in gene PA0766  mucD; serine protease MucD [KO:K04771] [EC:3.4.21.107]\n",
      "            PA0464  creC; sensory histidine kinase CreC [KO:K07641] [EC:2.7.13.3]\n",
      "line in gene PA0464  creC; sensory histidine kinase CreC [KO:K07641] [EC:2.7.13.3]\n",
      "            PA0463  creB; DNA-binding response regulator CreB [KO:K07663]\n",
      "line in gene PA0463  creB; DNA-binding response regulator CreB [KO:K07663]\n",
      "            PA2528  resistance-nodulation-cell division (RND) efflux membrane fusion protein [KO:K07799]\n",
      "line in gene PA2528  resistance-nodulation-cell division (RND) efflux membrane fusion protein [KO:K07799]\n",
      "            PA2527  resistance-nodulation-cell division (RND) efflux transporter [KO:K07788]\n",
      "line in gene PA2527  resistance-nodulation-cell division (RND) efflux transporter [KO:K07788]\n",
      "            PA2526  resistance-nodulation-cell division (RND) efflux transporter [KO:K07789]\n",
      "line in gene PA2526  resistance-nodulation-cell division (RND) efflux transporter [KO:K07789]\n",
      "            PA3552  arnB; UDP-4-amino-4-deoxy-L-arabinose--oxoglutarate aminotransferase [KO:K07806] [EC:2.6.1.87]\n",
      "line in gene PA3552  arnB; UDP-4-amino-4-deoxy-L-arabinose--oxoglutarate aminotransferase [KO:K07806] [EC:2.6.1.87]\n",
      "            PA1438  two-component sensor [KO:K07644] [EC:2.7.13.3]\n",
      "line in gene PA1438  two-component sensor [KO:K07644] [EC:2.7.13.3]\n",
      "            PA2810  copS; two-component sensor CopS [KO:K07644] [EC:2.7.13.3]\n",
      "line in gene PA2810  copS; two-component sensor CopS [KO:K07644] [EC:2.7.13.3]\n",
      "            PA4886  two-component sensor [KO:K07644] [EC:2.7.13.3]\n",
      "line in gene PA4886  two-component sensor [KO:K07644] [EC:2.7.13.3]\n",
      "            PA2524  two-component sensor [KO:K07644] [EC:2.7.13.3]\n",
      "line in gene PA2524  two-component sensor [KO:K07644] [EC:2.7.13.3]\n",
      "            PA1437  two-component response regulator [KO:K07665]\n",
      "line in gene PA1437  two-component response regulator [KO:K07665]\n",
      "            PA2523  two-component response regulator [KO:K07665]\n",
      "line in gene PA2523  two-component response regulator [KO:K07665]\n",
      "            PA4885  irlR; two-component response regulator [KO:K07665]\n",
      "line in gene PA4885  irlR; two-component response regulator [KO:K07665]\n",
      "            PA2809  copR; two-component response regulator CopR [KO:K07665]\n",
      "line in gene PA2809  copR; two-component response regulator CopR [KO:K07665]\n",
      "            PA4777  pmrB; two-component regulator system signal sensor kinase PmrB [KO:K07645] [EC:2.7.13.3]\n",
      "line in gene PA4777  pmrB; two-component regulator system signal sensor kinase PmrB [KO:K07645] [EC:2.7.13.3]\n",
      "            PA2480  two-component sensor [KO:K07645] [EC:2.7.13.3]\n",
      "line in gene PA2480  two-component sensor [KO:K07645] [EC:2.7.13.3]\n",
      "            PA4776  pmrA; two-component regulator system response regulator PmrA [KO:K07666]\n",
      "line in gene PA4776  pmrA; two-component regulator system response regulator PmrA [KO:K07666]\n",
      "            PA2479  two-component response regulator [KO:K07666]\n",
      "line in gene PA2479  two-component response regulator [KO:K07666]\n",
      "            PA1092  fliC; B-type flagellin [KO:K02406]\n",
      "line in gene PA1092  fliC; B-type flagellin [KO:K02406]\n",
      "            PA1455  fliA; flagellar biosynthesis sigma factor FliA [KO:K02405]\n",
      "line in gene PA1455  fliA; flagellar biosynthesis sigma factor FliA [KO:K02405]\n",
      "            PA1460  motC; flagellar motor protein [KO:K02556]\n",
      "line in gene PA1460  motC; flagellar motor protein [KO:K02556]\n",
      "            PA4954  motA; flagellar motor protein MotA [KO:K02556]\n",
      "line in gene PA4954  motA; flagellar motor protein MotA [KO:K02556]\n",
      "            PA1636  kdpD; two-component sensor KdpD [KO:K07646] [EC:2.7.13.3]\n",
      "line in gene PA1636  kdpD; two-component sensor KdpD [KO:K07646] [EC:2.7.13.3]\n",
      "            PA1637  kdpE; two-component response regulator KdpE [KO:K07667]\n",
      "line in gene PA1637  kdpE; two-component response regulator KdpE [KO:K07667]\n",
      "            PA1633  kdpA; potassium-transporting ATPase subunit A [KO:K01546]\n",
      "line in gene PA1633  kdpA; potassium-transporting ATPase subunit A [KO:K01546]\n",
      "            PA1634  kdpB; potassium-transporting ATPase subunit B [KO:K01547] [EC:7.2.2.6]\n",
      "line in gene PA1634  kdpB; potassium-transporting ATPase subunit B [KO:K01547] [EC:7.2.2.6]\n",
      "            PA1635  kdpC; potassium-transporting ATPase subunit C [KO:K01548]\n",
      "line in gene PA1635  kdpC; potassium-transporting ATPase subunit C [KO:K01548]\n",
      "            PA1632  kdpF; potassium-transporting ATPase subunit F [KO:K01545]\n",
      "line in gene PA1632  kdpF; potassium-transporting ATPase subunit F [KO:K01545]\n",
      "            PA0757  two-component sensor [KO:K07649] [EC:2.7.13.3]\n",
      "line in gene PA0757  two-component sensor [KO:K07649] [EC:2.7.13.3]\n",
      "            PA0756  two-component response regulator [KO:K07774]\n",
      "line in gene PA0756  two-component response regulator [KO:K07774]\n",
      "            PA0752  hypothetical protein [KO:K07793]\n",
      "line in gene PA0752  hypothetical protein [KO:K07793]\n",
      "            PA2207  hypothetical protein [KO:K07793]\n",
      "line in gene PA2207  hypothetical protein [KO:K07793]\n",
      "            PA0753  hypothetical protein [KO:K07794]\n",
      "line in gene PA0753  hypothetical protein [KO:K07794]\n",
      "            PA2208  hypothetical protein [KO:K07794]\n",
      "line in gene PA2208  hypothetical protein [KO:K07794]\n",
      "            PA0754  hypothetical protein [KO:K07795]\n",
      "line in gene PA0754  hypothetical protein [KO:K07795]\n",
      "            PA2209  hypothetical protein [KO:K07795]\n",
      "line in gene PA2209  hypothetical protein [KO:K07795]\n",
      "            PA0112  hypothetical protein [KO:K02259] [EC:1.17.99.9]\n",
      "line in gene PA0112  hypothetical protein [KO:K02259] [EC:1.17.99.9]\n",
      "            PA0001  dnaA; chromosome replication initiator DnaA [KO:K02313]\n",
      "line in gene PA0001  dnaA; chromosome replication initiator DnaA [KO:K02313]\n",
      "            PA0920  hypothetical protein [KO:K14205] [EC:2.3.2.3]\n",
      "line in gene PA0920  hypothetical protein [KO:K14205] [EC:2.3.2.3]\n",
      "            PA1798  parS; two-component sensor ParS [KO:K18072] [EC:2.7.13.3]\n",
      "line in gene PA1798  parS; two-component sensor ParS [KO:K18072] [EC:2.7.13.3]\n",
      "            PA1799  parR; two-component response regulator ParR [KO:K18073]\n",
      "line in gene PA1799  parR; two-component response regulator ParR [KO:K18073]\n",
      "            PA0958  oprD; porin D [KO:K18093] [EC:3.4.21.-]\n",
      "line in gene PA0958  oprD; porin D [KO:K18093] [EC:3.4.21.-]\n",
      "            PA0189  porin [KO:K18093] [EC:3.4.21.-]\n",
      "line in gene PA0189  porin [KO:K18093] [EC:3.4.21.-]\n",
      "            PA2700  opdB; OpdB proline porin [KO:K18093] [EC:3.4.21.-]\n",
      "line in gene PA2700  opdB; OpdB proline porin [KO:K18093] [EC:3.4.21.-]\n",
      "            PA2019  multidrug efflux lipoprotein [KO:K18094]\n",
      "line in gene PA2019  multidrug efflux lipoprotein [KO:K18094]\n",
      "            PA2018  multidrug efflux protein [KO:K18095]\n",
      "line in gene PA2018  multidrug efflux protein [KO:K18095]\n",
      "            PA0749  hypothetical protein [KO:K18346]\n",
      "line in gene PA0749  hypothetical protein [KO:K18346]\n",
      "            PA2687  pfeS; two-component sensor histidine kinase PfeS [KO:K19609] [EC:2.7.13.3]\n",
      "line in gene PA2687  pfeS; two-component sensor histidine kinase PfeS [KO:K19609] [EC:2.7.13.3]\n",
      "            PA0930  two-component sensor [KO:K19609] [EC:2.7.13.3]\n",
      "line in gene PA0930  two-component sensor [KO:K19609] [EC:2.7.13.3]\n",
      "            PA0929  two-component response regulator [KO:K19610]\n",
      "line in gene PA0929  two-component response regulator [KO:K19610]\n",
      "            PA2686  pfeR; two-component response regulator PfeR [KO:K19610]\n",
      "line in gene PA2686  pfeR; two-component response regulator PfeR [KO:K19610]\n",
      "            PA0931  pirA; outer membrane receptor FepA [KO:K19611]\n",
      "line in gene PA0931  pirA; outer membrane receptor FepA [KO:K19611]\n",
      "            PA2688  pfeA; ferric enterobactin receptor [KO:K19611]\n",
      "line in gene PA2688  pfeA; ferric enterobactin receptor [KO:K19611]\n",
      "            PA3471  NAD-dependent malic enzyme [KO:K00027] [EC:1.1.1.38]\n",
      "line in gene PA3471  NAD-dependent malic enzyme [KO:K00027] [EC:1.1.1.38]\n",
      "            PA5262  algZ; alginate biosynthesis protein AlgZ/FimS [KO:K08082] [EC:2.7.13.3]\n",
      "line in gene PA5262  algZ; alginate biosynthesis protein AlgZ/FimS [KO:K08082] [EC:2.7.13.3]\n",
      "            PA5261  algR; alginate biosynthesis regulatory protein AlgR [KO:K08083]\n",
      "line in gene PA5261  algR; alginate biosynthesis regulatory protein AlgR [KO:K08083]\n",
      "            PA3540  algD; GDP-mannose 6-dehydrogenase AlgD [KO:K00066] [EC:1.1.1.132]\n",
      "line in gene PA3540  algD; GDP-mannose 6-dehydrogenase AlgD [KO:K00066] [EC:1.1.1.132]\n",
      "            PA3878  narX; two-component sensor NarX [KO:K07673] [EC:2.7.13.3]\n",
      "line in gene PA3878  narX; two-component sensor NarX [KO:K07673] [EC:2.7.13.3]\n",
      "            PA3879  narL; transcriptional regulator NarL [KO:K07684]\n",
      "line in gene PA3879  narL; transcriptional regulator NarL [KO:K07684]\n",
      "            PA2376  transcriptional regulator [KO:K07684]\n",
      "line in gene PA2376  transcriptional regulator [KO:K07684]\n",
      "            PA3875  narG; respiratory nitrate reductase subunit alpha [KO:K00370] [EC:1.7.5.1 1.7.99.-]\n",
      "line in gene PA3875  narG; respiratory nitrate reductase subunit alpha [KO:K00370] [EC:1.7.5.1 1.7.99.-]\n",
      "            PA3874  narH; respiratory nitrate reductase subunit beta [KO:K00371] [EC:1.7.5.1 1.7.99.-]\n",
      "line in gene PA3874  narH; respiratory nitrate reductase subunit beta [KO:K00371] [EC:1.7.5.1 1.7.99.-]\n",
      "            PA3872  narI; respiratory nitrate reductase subunit gamma [KO:K00374] [EC:1.7.5.1 1.7.99.-]\n",
      "line in gene PA3872  narI; respiratory nitrate reductase subunit gamma [KO:K00374] [EC:1.7.5.1 1.7.99.-]\n",
      "            PA3873  narJ; respiratory nitrate reductase subunit delta [KO:K00373]\n",
      "line in gene PA3873  narJ; respiratory nitrate reductase subunit delta [KO:K00373]\n",
      "            PA1980  eraR; response regulator EraR [KO:K07686]\n",
      "line in gene PA1980  eraR; response regulator EraR [KO:K07686]\n",
      "            PA4080  response regulator [KO:K07687]\n",
      "line in gene PA4080  response regulator [KO:K07687]\n",
      "            PA3714  two-component response regulator [KO:K07687]\n",
      "line in gene PA3714  two-component response regulator [KO:K07687]\n",
      "            PA2899  transcriptional regulator [KO:K07687]\n",
      "line in gene PA2899  transcriptional regulator [KO:K07687]\n",
      "            PA0928  gacS; sensor/response regulator hybrid protein [KO:K07678] [EC:2.7.13.3]\n",
      "line in gene PA0928  gacS; sensor/response regulator hybrid protein [KO:K07678] [EC:2.7.13.3]\n",
      "            PA2586  gacA; response regulator GacA [KO:K07689]\n",
      "line in gene PA2586  gacA; response regulator GacA [KO:K07689]\n",
      "            PA0905  rsmA; carbon storage regulator [KO:K03563]\n",
      "line in gene PA0905  rsmA; carbon storage regulator [KO:K03563]\n",
      "            PA3946  rocS1; two-component sensor [KO:K07679] [EC:2.7.13.3]\n",
      "line in gene PA3946  rocS1; two-component sensor [KO:K07679] [EC:2.7.13.3]\n",
      "            PA2583  sensor/response regulator hybrid protein [KO:K07679] [EC:2.7.13.3]\n",
      "line in gene PA2583  sensor/response regulator hybrid protein [KO:K07679] [EC:2.7.13.3]\n",
      "            PA3948  rocA1; two-component response regulator RocA1 [KO:K07690]\n",
      "line in gene PA3948  rocA1; two-component response regulator RocA1 [KO:K07690]\n",
      "            PA3045  two-component response regulator [KO:K07690]\n",
      "line in gene PA3045  two-component response regulator [KO:K07690]\n",
      "            PA0034  two-component response regulator [KO:K07690]\n",
      "line in gene PA0034  two-component response regulator [KO:K07690]\n",
      "            PA3658  glnD; bifunctional uridylyltransferase/uridylyl-removing protein [KO:K00990] [EC:2.7.7.59]\n",
      "line in gene PA3658  glnD; bifunctional uridylyltransferase/uridylyl-removing protein [KO:K00990] [EC:2.7.7.59]\n",
      "            PA5124  ntrB; two-component sensor NtrB [KO:K07708] [EC:2.7.13.3]\n",
      "line in gene PA5124  ntrB; two-component sensor NtrB [KO:K07708] [EC:2.7.13.3]\n",
      "            PA5125  ntrC; two-component response regulator NtrC [KO:K07712]\n",
      "line in gene PA5125  ntrC; two-component response regulator NtrC [KO:K07712]\n",
      "            PA0296  spuI; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA0296  spuI; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA0298  spuB; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA0298  spuB; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA2040  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA2040  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5119  glnA; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5119  glnA; glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5522  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5522  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA1566  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA1566  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "            PA3356  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA3356  hypothetical protein [KO:K01915] [EC:6.3.1.2]\n",
      "            PA5508  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "line in gene PA5508  glutamine synthetase [KO:K01915] [EC:6.3.1.2]\n",
      "            PA4726  cbrB; two-component response regulator CbrB [KO:K07713]\n",
      "line in gene PA4726  cbrB; two-component response regulator CbrB [KO:K07713]\n",
      "            PA2002  hypothetical protein [KO:K02106]\n",
      "line in gene PA2002  hypothetical protein [KO:K02106]\n",
      "            PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2001  atoB; acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA2553  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3589  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA3925  acyl-CoA thiolase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA1736  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "line in gene PA4785  acetyl-CoA acetyltransferase [KO:K00626] [EC:2.3.1.9]\n",
      "            PA5165  two-component sensor [KO:K10125] [EC:2.7.13.3]\n",
      "line in gene PA5165  two-component sensor [KO:K10125] [EC:2.7.13.3]\n",
      "            PA5512  mifS; sensor histidine kinase MifS [KO:K10125] [EC:2.7.13.3]\n",
      "line in gene PA5512  mifS; sensor histidine kinase MifS [KO:K10125] [EC:2.7.13.3]\n",
      "            PA5166  two-component response regulator [KO:K10126]\n",
      "line in gene PA5166  two-component response regulator [KO:K10126]\n",
      "            PA5511  mifR; DNA-binding response regulator MifR [KO:K10126]\n",
      "line in gene PA5511  mifR; DNA-binding response regulator MifR [KO:K10126]\n",
      "            PA0119  C4-dicarboxylate transporter DctA [KO:K11103]\n",
      "line in gene PA0119  C4-dicarboxylate transporter DctA [KO:K11103]\n",
      "            PA1183  dctA; C4-dicarboxylate transport protein [KO:K11103]\n",
      "line in gene PA1183  dctA; C4-dicarboxylate transport protein [KO:K11103]\n",
      "            PA4546  pilS; two-component sensor PilS [KO:K02668] [EC:2.7.13.3]\n",
      "line in gene PA4546  pilS; two-component sensor PilS [KO:K02668] [EC:2.7.13.3]\n",
      "            PA4547  pilR; two-component response regulator PilR [KO:K02667]\n",
      "line in gene PA4547  pilR; two-component response regulator PilR [KO:K02667]\n",
      "            PA4462  rpoN; RNA polymerase factor sigma-54 [KO:K03092]\n",
      "line in gene PA4462  rpoN; RNA polymerase factor sigma-54 [KO:K03092]\n",
      "            PA4525  pilA; type 4 fimbrial protein PilA [KO:K02650]\n",
      "line in gene PA4525  pilA; type 4 fimbrial protein PilA [KO:K02650]\n",
      "            PA5484  two-component sensor [KO:K11383] [EC:2.7.13.3]\n",
      "line in gene PA5484  two-component sensor [KO:K11383] [EC:2.7.13.3]\n",
      "            PA5483  algB; two-component response regulator AlgB [KO:K11384]\n",
      "line in gene PA5483  algB; two-component response regulator AlgB [KO:K11384]\n",
      "            PA0175  chemotaxis protein methyltransferase [KO:K00575] [EC:2.1.1.80]\n",
      "line in gene PA0175  chemotaxis protein methyltransferase [KO:K00575] [EC:2.1.1.80]\n",
      "            PA3348  chemotaxis protein methyltransferase [KO:K00575] [EC:2.1.1.80]\n",
      "line in gene PA3348  chemotaxis protein methyltransferase [KO:K00575] [EC:2.1.1.80]\n",
      "            PA0180  cttP; trichloroethylene chemotactic transducer CttP [KO:K03406]\n",
      "line in gene PA0180  cttP; trichloroethylene chemotactic transducer CttP [KO:K03406]\n",
      "            PA0176  aer2; aerotaxis transducer Aer2 [KO:K03406]\n",
      "line in gene PA0176  aer2; aerotaxis transducer Aer2 [KO:K03406]\n",
      "            PA4310  pctB; chemotactic transducer PctB [KO:K03406]\n",
      "line in gene PA4310  pctB; chemotactic transducer PctB [KO:K03406]\n",
      "            PA4633  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA4633  chemotaxis transducer [KO:K03406]\n",
      "            PA2573  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA2573  chemotaxis transducer [KO:K03406]\n",
      "            PA1930  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA1930  chemotaxis transducer [KO:K03406]\n",
      "            PA1608  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA1608  chemotaxis transducer [KO:K03406]\n",
      "            PA2652  methyl-accepting chemotaxis protein [KO:K03406]\n",
      "line in gene PA2652  methyl-accepting chemotaxis protein [KO:K03406]\n",
      "            PA5072  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA5072  chemotaxis transducer [KO:K03406]\n",
      "            PA1646  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA1646  chemotaxis transducer [KO:K03406]\n",
      "            PA4290  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA4290  chemotaxis transducer [KO:K03406]\n",
      "            PA4915  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA4915  chemotaxis transducer [KO:K03406]\n",
      "            PA1423  bdlA; biofilm dispersion protein [KO:K03406]\n",
      "line in gene PA1423  bdlA; biofilm dispersion protein [KO:K03406]\n",
      "            PA4309  pctA; chemotactic transducer PctA [KO:K03406]\n",
      "line in gene PA4309  pctA; chemotactic transducer PctA [KO:K03406]\n",
      "            PA4307  pctC; chemotactic transducer PctC [KO:K03406]\n",
      "line in gene PA4307  pctC; chemotactic transducer PctC [KO:K03406]\n",
      "            PA2561  methyl-accepting chemotaxis protein CtpH [KO:K03406]\n",
      "line in gene PA2561  methyl-accepting chemotaxis protein CtpH [KO:K03406]\n",
      "            PA2654  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA2654  chemotaxis transducer [KO:K03406]\n",
      "            PA2920  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA2920  chemotaxis transducer [KO:K03406]\n",
      "            PA4520  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA4520  chemotaxis transducer [KO:K03406]\n",
      "            PA1251  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA1251  chemotaxis transducer [KO:K03406]\n",
      "            PA1561  aer; aerotaxis receptor Aer [KO:K03776]\n",
      "line in gene PA1561  aer; aerotaxis receptor Aer [KO:K03776]\n",
      "            PA0177  purine-binding chemotaxis protein [KO:K03408]\n",
      "line in gene PA0177  purine-binding chemotaxis protein [KO:K03408]\n",
      "            PA1464  purine-binding chemotaxis protein [KO:K03408]\n",
      "line in gene PA1464  purine-binding chemotaxis protein [KO:K03408]\n",
      "            PA1463  hypothetical protein [KO:K03408]\n",
      "line in gene PA1463  hypothetical protein [KO:K03408]\n",
      "            PA0178  two-component sensor [KO:K03407] [EC:2.7.13.3]\n",
      "line in gene PA0178  two-component sensor [KO:K03407] [EC:2.7.13.3]\n",
      "            PA1458  two-component sensor [KO:K03407] [EC:2.7.13.3]\n",
      "line in gene PA1458  two-component sensor [KO:K03407] [EC:2.7.13.3]\n",
      "            PA0179  two-component response regulator [KO:K03413]\n",
      "line in gene PA0179  two-component response regulator [KO:K03413]\n",
      "            PA1456  cheY; chemotaxis protein CheY [KO:K03413]\n",
      "line in gene PA1456  cheY; chemotaxis protein CheY [KO:K03413]\n",
      "            PA0173  chemotaxis response regulator protein-glutamate methylesterase [KO:K03412] [EC:3.1.1.61 3.5.1.44]\n",
      "line in gene PA0173  chemotaxis response regulator protein-glutamate methylesterase [KO:K03412] [EC:3.1.1.61 3.5.1.44]\n",
      "            PA1459  chemotaxis-specific methylesterase [KO:K03412] [EC:3.1.1.61 3.5.1.44]\n",
      "line in gene PA1459  chemotaxis-specific methylesterase [KO:K03412] [EC:3.1.1.61 3.5.1.44]\n",
      "            PA3349  chemotaxis protein [KO:K03415]\n",
      "line in gene PA3349  chemotaxis protein [KO:K03415]\n",
      "            PA0412  pilK; methyltransferase PilK [KO:K02661]\n",
      "line in gene PA0412  pilK; methyltransferase PilK [KO:K02661]\n",
      "            PA0411  pilJ; twitching motility protein PilJ [KO:K02660]\n",
      "line in gene PA0411  pilJ; twitching motility protein PilJ [KO:K02660]\n",
      "            PA0410  pilI; twitching motility protein PilI [KO:K02659]\n",
      "line in gene PA0410  pilI; twitching motility protein PilI [KO:K02659]\n",
      "            PA0413  chpA; chemotactic signal transduction system protein [KO:K06596]\n",
      "line in gene PA0413  chpA; chemotactic signal transduction system protein [KO:K06596]\n",
      "            PA0408  pilG; pilus biosynthesis/twitching motility protein PilG [KO:K02657]\n",
      "line in gene PA0408  pilG; pilus biosynthesis/twitching motility protein PilG [KO:K02657]\n",
      "            PA0409  pilH; twitching motility protein PilH [KO:K02658]\n",
      "line in gene PA0409  pilH; twitching motility protein PilH [KO:K02658]\n",
      "            PA0414  chpB; methylesterase [KO:K06597]\n",
      "line in gene PA0414  chpB; methylesterase [KO:K06597]\n",
      "            PA0415  chpC; chemotaxis protein [KO:K06598]\n",
      "line in gene PA0415  chpC; chemotaxis protein [KO:K06598]\n",
      "            PA3706  wspC; biofilm formation methyltransferase WspC [KO:K13486]\n",
      "line in gene PA3706  wspC; biofilm formation methyltransferase WspC [KO:K13486]\n",
      "            PA3708  wspA; chemotaxis transducer [KO:K13487]\n",
      "line in gene PA3708  wspA; chemotaxis transducer [KO:K13487]\n",
      "            PA3707  hypothetical protein [KO:K13488]\n",
      "line in gene PA3707  hypothetical protein [KO:K13488]\n",
      "            PA3705  wspD; hypothetical protein [KO:K13489]\n",
      "line in gene PA3705  wspD; hypothetical protein [KO:K13489]\n",
      "            PA3704  wspE; chemotaxis sensor/effector fusion protein [KO:K13490]\n",
      "line in gene PA3704  wspE; chemotaxis sensor/effector fusion protein [KO:K13490]\n",
      "            PA3703  wspF; chemotaxis-specific methylesterase [KO:K13491] [EC:3.1.1.61]\n",
      "line in gene PA3703  wspF; chemotaxis-specific methylesterase [KO:K13491] [EC:3.1.1.61]\n",
      "            PA3702  wspR; two-component response regulator [KO:K11444] [EC:2.7.7.65]\n",
      "line in gene PA3702  wspR; two-component response regulator [KO:K11444] [EC:2.7.7.65]\n",
      "            PA5167  C4-dicarboxylate-binding protein [KO:K11688]\n",
      "line in gene PA5167  C4-dicarboxylate-binding protein [KO:K11688]\n",
      "            PA4616  C4-dicarboxylate-binding protein [KO:K11688]\n",
      "line in gene PA4616  C4-dicarboxylate-binding protein [KO:K11688]\n",
      "            PA0884  C4-dicarboxylate-binding protein [KO:K11688]\n",
      "line in gene PA0884  C4-dicarboxylate-binding protein [KO:K11688]\n",
      "            PA0885  C4-dicarboxylate transporter [KO:K11689]\n",
      "line in gene PA0885  C4-dicarboxylate transporter [KO:K11689]\n",
      "            PA5168  dicarboxylate transporter [KO:K11689]\n",
      "line in gene PA5168  dicarboxylate transporter [KO:K11689]\n",
      "            PA5169  C4-dicarboxylate transporter [KO:K11690]\n",
      "line in gene PA5169  C4-dicarboxylate transporter [KO:K11690]\n",
      "            PA0886  C4-dicarboxylate transporter [KO:K11690]\n",
      "line in gene PA0886  C4-dicarboxylate transporter [KO:K11690]\n",
      "            PA1097  fleQ; transcriptional regulator FleQ [KO:K10941]\n",
      "line in gene PA1097  fleQ; transcriptional regulator FleQ [KO:K10941]\n",
      "            PA1098  fleS; two-component sensor [KO:K10942] [EC:2.7.13.3]\n",
      "line in gene PA1098  fleS; two-component sensor [KO:K10942] [EC:2.7.13.3]\n",
      "            PA1099  fleR; two-component response regulator [KO:K10943]\n",
      "line in gene PA1099  fleR; two-component response regulator [KO:K10943]\n",
      "            PA1336  two-component sensor [KO:K17060] [EC:2.7.13.3]\n",
      "line in gene PA1336  two-component sensor [KO:K17060] [EC:2.7.13.3]\n",
      "            PA1335  two-component response regulator [KO:K17061]\n",
      "line in gene PA1335  two-component response regulator [KO:K17061]\n",
      "            PA1339  amino acid ABC transporter ATP binding protein [KO:K10004] [EC:7.4.2.1]\n",
      "line in gene PA1339  amino acid ABC transporter ATP binding protein [KO:K10004] [EC:7.4.2.1]\n",
      "            PA1340  amino acid ABC transporter permease [KO:K10002]\n",
      "line in gene PA1340  amino acid ABC transporter permease [KO:K10002]\n",
      "            PA1341  amino acid ABC transporter permease [KO:K10003]\n",
      "line in gene PA1341  amino acid ABC transporter permease [KO:K10003]\n",
      "            PA1342  ABC transporter [KO:K10001]\n",
      "line in gene PA1342  ABC transporter [KO:K10001]\n",
      "            PA5082  ABC transporter [KO:K10001]\n",
      "line in gene PA5082  ABC transporter [KO:K10001]\n",
      "            PA1337  ansB; glutaminase-asparaginase [KO:K05597] [EC:3.5.1.38]\n",
      "line in gene PA1337  ansB; glutaminase-asparaginase [KO:K05597] [EC:3.5.1.38]\n",
      "            PA0652  vfr; cAMP-regulatory protein [KO:K10914]\n",
      "line in gene PA0652  vfr; cAMP-regulatory protein [KO:K10914]\n",
      "            PA0275  transcriptional regulator [KO:K10914]\n",
      "line in gene PA0275  transcriptional regulator [KO:K10914]\n",
      "            PA1638  glutaminase [KO:K01425] [EC:3.5.1.2]\n",
      "line in gene PA1638  glutaminase [KO:K01425] [EC:3.5.1.2]\n",
      "            PA4494  roxS; sensor histidine kinase RoxS [KO:K15011] [EC:2.7.13.3]\n",
      "line in gene PA4494  roxS; sensor histidine kinase RoxS [KO:K15011] [EC:2.7.13.3]\n",
      "            PA4493  roxR; DNA-binding response regulator RoxR [KO:K15012]\n",
      "line in gene PA4493  roxR; DNA-binding response regulator RoxR [KO:K15012]\n",
      "            PA4431  iron-sulfur protein [KO:K00411] [EC:7.1.1.8]\n",
      "line in gene PA4431  iron-sulfur protein [KO:K00411] [EC:7.1.1.8]\n",
      "            PA4430  cytochrome b [KO:K00412]\n",
      "line in gene PA4430  cytochrome b [KO:K00412]\n",
      "            PA4429  cytochrome C1 [KO:K00413]\n",
      "line in gene PA4429  cytochrome C1 [KO:K00413]\n",
      "            PA0518  nirM; cytochrome C-551 [KO:K08738]\n",
      "line in gene PA0518  nirM; cytochrome C-551 [KO:K08738]\n",
      "            PA4133  cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "line in gene PA4133  cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "            PA1554  ccoN1; cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "line in gene PA1554  ccoN1; cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "            PA1557  ccoN2; cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "line in gene PA1557  ccoN2; cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "            PA1856  cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "line in gene PA1856  cbb3-type cytochrome C oxidase subunit I [KO:K00404] [EC:7.1.1.9]\n",
      "            PA1553  ccoO1; cbb3-type cytochrome C oxidase subunit II [KO:K00405]\n",
      "line in gene PA1553  ccoO1; cbb3-type cytochrome C oxidase subunit II [KO:K00405]\n",
      "            PA1556  ccoO2; cbb3-type cytochrome C oxidase subunit II [KO:K00405]\n",
      "line in gene PA1556  ccoO2; cbb3-type cytochrome C oxidase subunit II [KO:K00405]\n",
      "            PA1552.1  ccoQ1; cytochrome C oxidase cbb3-type subunit CcoQ [KO:K00407]\n",
      "line in gene PA1552.1  ccoQ1; cytochrome C oxidase cbb3-type subunit CcoQ [KO:K00407]\n",
      "            PA1555.1  ccoQ2; cytochrome C oxidase cbb3-type subunit CcoQ [KO:K00407]\n",
      "line in gene PA1555.1  ccoQ2; cytochrome C oxidase cbb3-type subunit CcoQ [KO:K00407]\n",
      "            PA1552  ccoP1; cytochrome C oxidase cbb3-type subunit CcoP [KO:K00406]\n",
      "line in gene PA1552  ccoP1; cytochrome C oxidase cbb3-type subunit CcoP [KO:K00406]\n",
      "            PA1555  ccoP2; cytochrome C oxidase cbb3-type subunit CcoP [KO:K00406]\n",
      "line in gene PA1555  ccoP2; cytochrome C oxidase cbb3-type subunit CcoP [KO:K00406]\n",
      "            PA3930  cioA; cyanide insensitive terminal oxidase [KO:K00425] [EC:7.1.1.7]\n",
      "line in gene PA3930  cioA; cyanide insensitive terminal oxidase [KO:K00425] [EC:7.1.1.7]\n",
      "            PA3929  cioB; cyanide insensitive terminal oxidase [KO:K00426] [EC:7.1.1.7]\n",
      "line in gene PA3929  cioB; cyanide insensitive terminal oxidase [KO:K00426] [EC:7.1.1.7]\n",
      "            PA4974  hypothetical protein [KO:K12340]\n",
      "line in gene PA4974  hypothetical protein [KO:K12340]\n",
      "            PA4144  hypothetical protein [KO:K12340]\n",
      "line in gene PA4144  hypothetical protein [KO:K12340]\n",
      "            PA4592  hypothetical protein [KO:K12340]\n",
      "line in gene PA4592  hypothetical protein [KO:K12340]\n",
      "            PA2978  ptpA; phosphotyrosine protein phosphatase [KO:K25307] [EC:3.1.3.48]\n",
      "line in gene PA2978  ptpA; phosphotyrosine protein phosphatase [KO:K25307] [EC:3.1.3.48]\n",
      "            PA3476  rhlI; acyl-homoserine-lactone synthase [KO:K13061] [EC:2.3.1.184]\n",
      "line in gene PA3476  rhlI; acyl-homoserine-lactone synthase [KO:K13061] [EC:2.3.1.184]\n",
      "            PA3899  RNA polymerase sigma factor [KO:K23514]\n",
      "line in gene PA3899  RNA polymerase sigma factor [KO:K23514]\n",
      "            PA2050  RNA polymerase sigma factor [KO:K23514]\n",
      "line in gene PA2050  RNA polymerase sigma factor [KO:K23514]\n",
      "            PA0472  RNA polymerase sigma factor [KO:K23514]\n",
      "line in gene PA0472  RNA polymerase sigma factor [KO:K23514]\n",
      "            PA3900  transmembrane sensor [KO:K07165]\n",
      "line in gene PA3900  transmembrane sensor [KO:K07165]\n",
      "            PA3409  transmembrane sensor [KO:K07165]\n",
      "line in gene PA3409  transmembrane sensor [KO:K07165]\n",
      "            PA2051  transmembrane sensor [KO:K07165]\n",
      "line in gene PA2051  transmembrane sensor [KO:K07165]\n",
      "            PA1911  femR; sigma factor regulator FemR [KO:K07165]\n",
      "line in gene PA1911  femR; sigma factor regulator FemR [KO:K07165]\n",
      "            PA2094  transmembrane sensor [KO:K07165]\n",
      "line in gene PA2094  transmembrane sensor [KO:K07165]\n",
      "            PA2388  fpvR; protein FpvR [KO:K07165]\n",
      "line in gene PA2388  fpvR; protein FpvR [KO:K07165]\n",
      "            PA2467  foxR; anti-sigma factor FoxR [KO:K07165]\n",
      "line in gene PA2467  foxR; anti-sigma factor FoxR [KO:K07165]\n",
      "            PA0150  transmembrane sensor [KO:K07165]\n",
      "line in gene PA0150  transmembrane sensor [KO:K07165]\n",
      "            PA0471  transmembrane sensor [KO:K07165]\n",
      "line in gene PA0471  transmembrane sensor [KO:K07165]\n",
      "            PA1301  transmembrane sensor [KO:K07165]\n",
      "line in gene PA1301  transmembrane sensor [KO:K07165]\n",
      "            PA2824  sensor/response regulator hybrid protein [KO:K20973] [EC:2.7.13.3]\n",
      "line in gene PA2824  sensor/response regulator hybrid protein [KO:K20973] [EC:2.7.13.3]\n",
      "            PA1611  sensor/response regulator hybrid protein [KO:K20974] [EC:2.7.13.3]\n",
      "line in gene PA1611  sensor/response regulator hybrid protein [KO:K20974] [EC:2.7.13.3]\n",
      "            PA1976  ercS'; sensor histidine kinase [KO:K20975] [EC:2.7.13.3]\n",
      "line in gene PA1976  ercS'; sensor histidine kinase [KO:K20975] [EC:2.7.13.3]\n",
      "            PA3345  hypothetical protein [KO:K20976]\n",
      "line in gene PA3345  hypothetical protein [KO:K20976]\n",
      "            PA3346  two-component response regulator [KO:K20977]\n",
      "line in gene PA3346  two-component response regulator [KO:K20977]\n",
      "            PA3347  hypothetical protein [KO:K20978]\n",
      "line in gene PA3347  hypothetical protein [KO:K20978]\n",
      "            PA3351  flgM; protein FlgM [KO:K02398]\n",
      "line in gene PA3351  flgM; protein FlgM [KO:K02398]\n",
      "COMPOUND    C00007  Oxygen\n",
      "            C00009  Orthophosphate\n",
      "            C00025  L-Glutamate\n",
      "            C00031  D-Glucose\n",
      "            C00034  Manganese\n",
      "            C00036  Oxaloacetate\n",
      "            C00038  Zinc cation\n",
      "            C00042  Succinate\n",
      "            C00049  L-Aspartate\n",
      "            C00064  L-Glutamine\n",
      "            C00070  Copper\n",
      "            C00088  Nitrite\n",
      "            C00092  D-Glucose 6-phosphate\n",
      "            C00122  Fumarate\n",
      "            C00149  (S)-Malate\n",
      "            C00158  Citrate\n",
      "            C00164  Acetoacetate\n",
      "            C00238  Potassium cation\n",
      "            C00244  Nitrate\n",
      "            C00282  Hydrogen\n",
      "            C00291  Nickel\n",
      "            C00305  Magnesium cation\n",
      "            C00399  Ubiquinone\n",
      "            C00547  L-Noradrenaline\n",
      "            C00697  Nitrogen\n",
      "            C00788  L-Adrenaline\n",
      "            C00828  Menaquinone\n",
      "            C01019  6-Deoxy-L-galactose\n",
      "            C01104  Trimethylamine N-oxide\n",
      "            C01667  Bacitracin\n",
      "            C02084  Tetrathionate\n",
      "            C05757  (R)-3-Hydroxydodecanoyl-[acp]\n",
      "            C05818  2-Demethylmenaquinone\n",
      "            C06230  Fe-enterobactin\n",
      "            C06689  Vancomycin\n",
      "            C06696  Lead\n",
      "            C11612  Polymyxin B\n",
      "            C11842  N-(3-(S)-Hydroxybutyryl)homoserine lactone\n",
      "            C11849  3-Hydroxy-palmitic acid methyl ester\n",
      "            C12008  Ramoplanin A2\n",
      "            C13768  Colistin sulfate\n",
      "            C14819  Fe3+\n",
      "            C16421  AI-2\n",
      "            C16463  3',5'-Cyclic diGMP\n",
      "            C16640  CAI-1\n",
      "            C16665  12-Methyltetradecanoic acid\n",
      "            C18206  cis-11-Methyl-2-dodecenoic acid\n",
      "            C21220  AIP-1\n",
      "            C21221  CSP\n",
      "            C21241  AIP-2\n",
      "            C21242  AIP-3\n",
      "            C21243  AIP-4\n",
      "            C21245  AgrD1 peptide\n",
      "            D07746  Colistin (INN)\n",
      "            D08401  Polymyxin B (INN)\n",
      "REFERENCE   PMID:15522865\n",
      "  AUTHORS   Yamamoto K, Hirao K, Oshima T, Aiba H, Utsumi R, Ishihama A.\n",
      "  TITLE     Functional characterization in vitro of all two-component signal transduction systems from Escherichia coli.\n",
      "  JOURNAL   J Biol Chem 280:1448-56 (2005)\n",
      "            DOI:10.1074/jbc.M410104200\n",
      "REFERENCE   PMID:16030210\n",
      "  AUTHORS   Abdel-Fattah WR, Chen Y, Eldakak A, Hulett FM.\n",
      "  TITLE     Bacillus subtilis phosphorylated PhoP: direct activation of the E(sigma)A- and repression of the E(sigma)E-responsive phoB-PS+V promoters during pho response.\n",
      "  JOURNAL   J Bacteriol 187:5166-78 (2005)\n",
      "            DOI:10.1128/JB.187.15.5166-5178.2005\n",
      "REFERENCE   PMID:15813739\n",
      "  AUTHORS   Navarre WW, Halsey TA, Walthers D, Frye J, McClelland M, Potter JL, Kenney LJ, Gunn JS, Fang FC, Libby SJ.\n",
      "  TITLE     Co-regulation of Salmonella enterica genes required for virulence and resistance to antimicrobial peptides by SlyA and PhoP/PhoQ.\n",
      "  JOURNAL   Mol Microbiol 56:492-508 (2005)\n",
      "            DOI:10.1111/j.1365-2958.2005.04553.x\n",
      "REFERENCE   PMID:11973328\n",
      "  AUTHORS   Cai SJ, Inouye M.\n",
      "  TITLE     EnvZ-OmpR interaction and osmoregulation in Escherichia coli.\n",
      "  JOURNAL   J Biol Chem 277:24155-61 (2002)\n",
      "            DOI:10.1074/jbc.M110715200\n",
      "REFERENCE   PMID:16077119\n",
      "  AUTHORS   Batchelor E, Walthers D, Kenney LJ, Goulian M.\n",
      "  TITLE     The Escherichia coli CpxA-CpxR envelope stress response system regulates expression of the porins ompF and ompC.\n",
      "  JOURNAL   J Bacteriol 187:5723-31 (2005)\n",
      "            DOI:10.1128/JB.187.16.5723-5731.2005\n",
      "REFERENCE   PMID:12107133\n",
      "  AUTHORS   Nagakubo S, Nishino K, Hirata T, Yamaguchi A.\n",
      "  TITLE     The putative response regulator BaeR stimulates multidrug resistance of Escherichia coli via a novel multidrug exporter system, MdtABC.\n",
      "  JOURNAL   J Bacteriol 184:4161-7 (2002)\n",
      "            DOI:10.1128/JB.184.15.4161-4167.2002\n",
      "REFERENCE   PMID:15322361\n",
      "  AUTHORS   Hagiwara D, Yamashino T, Mizuno T.\n",
      "  TITLE     A Genome-wide view of the Escherichia coli BasS-BasR two-component system implicated in iron-responses.\n",
      "  JOURNAL   Biosci Biotechnol Biochem 68:1758-67 (2004)\n",
      "            DOI:10.1271/bbb.68.1758\n",
      "REFERENCE   PMID:11004187\n",
      "  AUTHORS   Munson GP, Lam DL, Outten FW, O'Halloran TV.\n",
      "  TITLE     Identification of a copper-responsive two-component system on the chromosome of Escherichia coli K-12.\n",
      "  JOURNAL   J Bacteriol 182:5864-71 (2000)\n",
      "            DOI:10.1128/JB.182.20.5864-5871.2000\n",
      "REFERENCE   PMID:11929534\n",
      "  AUTHORS   Sperandio V, Torres AG, Kaper JB.\n",
      "  TITLE     Quorum sensing Escherichia coli regulators B and C (QseBC): a novel two-component regulatory system involved in the regulation of flagella and motility by quorum sensing in E. coli.\n",
      "  JOURNAL   Mol Microbiol 43:809-21 (2002)\n",
      "            DOI:10.1046/j.1365-2958.2002.02803.x\n",
      "REFERENCE   PMID:14604015\n",
      "  AUTHORS   Epstein W.\n",
      "  TITLE     The roles and regulation of potassium in bacteria.\n",
      "  JOURNAL   Prog Nucleic Acid Res Mol Biol 75:293-320 (2003)\n",
      "            DOI:10.1016/S0079-6603(03)75008-9\n",
      "REFERENCE   PMID:10648521\n",
      "  AUTHORS   Ansaldi M, Simon G, Lepelletier M, Mejean V.\n",
      "  TITLE     The TorR high-affinity binding site plays a key role in both torR autoregulation and torCAD operon expression in Escherichia coli.\n",
      "  JOURNAL   J Bacteriol 182:961-6 (2000)\n",
      "            DOI:10.1128/JB.182.4.961-966.2000\n",
      "REFERENCE   PMID:15326287\n",
      "  AUTHORS   Malpica R, Franco B, Rodriguez C, Kwon O, Georgellis D.\n",
      "  TITLE     Identification of a quinone-sensitive redox switch in the ArcB sensor kinase.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 101:13318-23 (2004)\n",
      "            DOI:10.1073/pnas.0403064101\n",
      "REFERENCE   PMID:2666399\n",
      "  AUTHORS   Widenhorn KA, Somers JM, Kay WW.\n",
      "  TITLE     Genetic regulation of the tricarboxylate transport operon (tctI) of Salmonella typhimurium.\n",
      "  JOURNAL   J Bacteriol 171:4436-41 (1989)\n",
      "            DOI:10.1128/JB.171.8.4436-4441.1989\n",
      "REFERENCE   PMID:11555295\n",
      "  AUTHORS   Hyyrylainen HL, Bolhuis A, Darmon E, Muukkonen L, Koski P, Vitikainen M, Sarvas M, Pragai Z, Bron S, van Dijl JM, Kontinen VP.\n",
      "  TITLE     A novel two-component regulatory system in Bacillus subtilis for the survival of severe secretion stress.\n",
      "  JOURNAL   Mol Microbiol 41:1159-72 (2001)\n",
      "            DOI:10.1046/j.1365-2958.2001.02576.x\n",
      "REFERENCE   PMID:10972837\n",
      "  AUTHORS   Zhang X, Hulett FM.\n",
      "  TITLE     ResD signal transduction regulator of aerobic respiration in Bacillus subtilis: ctaA promoter regulation.\n",
      "  JOURNAL   Mol Microbiol 37:1208-19 (2000)\n",
      "            DOI:10.1046/j.1365-2958.2000.02076.x\n",
      "REFERENCE   PMID:15937169\n",
      "  AUTHORS   Senadheera MD, Guggenheim B, Spatafora GA, Huang YC, Choi J, Hung DC, Treglown JS, Goodman SD, Ellen RP, Cvitkovitch DG.\n",
      "  TITLE     A VicRK signal transduction system in Streptococcus mutans affects gtfBCD, gbpB, and ftf expression, biofilm formation, and genetic competence development.\n",
      "  JOURNAL   J Bacteriol 187:4064-76 (2005)\n",
      "            DOI:10.1128/JB.187.12.4064-4076.2005\n",
      "REFERENCE   PMID:15601704\n",
      "  AUTHORS   He H, Zahrt TC.\n",
      "  TITLE     Identification and characterization of a regulatory sequence recognized by Mycobacterium tuberculosis persistence regulator MprA.\n",
      "  JOURNAL   J Bacteriol 187:202-12 (2005)\n",
      "            DOI:10.1128/JB.187.1.202-212.2005\n",
      "REFERENCE   PMID:17448786\n",
      "  AUTHORS   Hutchings MI\n",
      "  TITLE     Unusual two-component signal transduction pathways in the actinobacteria.\n",
      "  JOURNAL   Adv Appl Microbiol 61:1-26 (2007)\n",
      "            DOI:10.1016/S0065-2164(06)61001-0\n",
      "REFERENCE   PMID:19308662\n",
      "  AUTHORS   Kramer R\n",
      "  TITLE     Osmosensing and osmosignaling in Corynebacterium glutamicum.\n",
      "  JOURNAL   Amino Acids 37:487-97 (2009)\n",
      "            DOI:10.1007/s00726-009-0271-6\n",
      "REFERENCE   PMID:7751278\n",
      "  AUTHORS   Eraso JM, Kaplan S.\n",
      "  TITLE     Oxygen-insensitive synthesis of the photosynthetic membranes of Rhodobacter sphaeroides: a mutant histidine kinase.\n",
      "  JOURNAL   J Bacteriol 177:2695-706 (1995)\n",
      "            DOI:10.1128/JB.177.10.2695-2706.1995\n",
      "REFERENCE   PMID:10089160\n",
      "  AUTHORS   Haydel SE, Dunlap NE, Benjamin WH Jr.\n",
      "  TITLE     In vitro evidence of two-component system phosphorylation between the Mycobacterium tuberculosis TrcR/TrcS proteins.\n",
      "  JOURNAL   Microb Pathog 26:195-206 (1999)\n",
      "            DOI:10.1006/mpat.1998.0265\n",
      "REFERENCE   PMID:11673427\n",
      "  AUTHORS   Quisel JD, Burkholder WF, Grossman AD.\n",
      "  TITLE     In vivo effects of sporulation kinases on mutant Spo0A proteins in Bacillus subtilis.\n",
      "  JOURNAL   J Bacteriol 183:6573-8 (2001)\n",
      "            DOI:10.1128/JB.183.22.6573-6578.2001\n",
      "REFERENCE   PMID:11889485\n",
      "  AUTHORS   Kaspar S, Bott M.\n",
      "  TITLE     The sensor kinase CitA (DpiB) of Escherichia coli functions as a high-affinity citrate receptor.\n",
      "  JOURNAL   Arch Microbiol 177:313-21 (2002)\n",
      "            DOI:10.1007/s00203-001-0393-z\n",
      "REFERENCE   PMID:12167640\n",
      "  AUTHORS   Janausch IG, Garcia-Moreno I, Unden G.\n",
      "  TITLE     Function of DcuS from Escherichia coli as a fumarate-stimulated histidine protein kinase in vitro.\n",
      "  JOURNAL   J Biol Chem 277:39809-14 (2002)\n",
      "            DOI:10.1074/jbc.M204482200\n",
      "REFERENCE   PMID:8550490\n",
      "  AUTHORS   Brunskill EW, Bayles KW.\n",
      "  TITLE     Identification and molecular characterization of a putative regulatory locus that affects autolysis in Staphylococcus aureus.\n",
      "  JOURNAL   J Bacteriol 178:611-8 (1996)\n",
      "            DOI:10.1128/JB.178.3.611-618.1996\n",
      "REFERENCE   PMID:10940023\n",
      "  AUTHORS   Fujimoto DF, Brunskill EW, Bayles KW.\n",
      "  TITLE     Analysis of genetic elements controlling Staphylococcus aureus lrgAB expression: potential role of DNA topology in SarA regulation.\n",
      "  JOURNAL   J Bacteriol 182:4822-8 (2000)\n",
      "            DOI:10.1128/JB.182.17.4822-4828.2000\n",
      "REFERENCE   PMID:11807079\n",
      "  AUTHORS   Dufour P, Jarraud S, Vandenesch F, Greenland T, Novick RP, Bes M, Etienne J, Lina G.\n",
      "  TITLE     High genetic variability of the agr locus in Staphylococcus species.\n",
      "  JOURNAL   J Bacteriol 184:1180-6 (2002)\n",
      "            DOI:10.1128/JB.184.4.1180-1186.2002\n",
      "REFERENCE   PMID:8501030\n",
      "  AUTHORS   Rabin RS, Stewart V.\n",
      "  TITLE     Dual response regulators (NarL and NarP) interact with dual sensors (NarX and NarQ) to control nitrate- and nitrite-regulated gene expression in Escherichia coli K-12.\n",
      "  JOURNAL   J Bacteriol 175:3259-68 (1993)\n",
      "            DOI:10.1128/JB.175.11.3259-3268.1993\n",
      "REFERENCE   PMID:9220011\n",
      "  AUTHORS   Chiang RC, Cavicchioli R, Gunsalus RP.\n",
      "  TITLE     'Locked-on' and 'locked-off' signal transduction mutations in the periplasmic domain of the Escherichia coli NarQ and NarX sensors affect nitrate- and nitrite-dependent regulation by NarL and NarP.\n",
      "  JOURNAL   Mol Microbiol 24:1049-60 (1997)\n",
      "            DOI:10.1046/j.1365-2958.1997.4131779.x\n",
      "REFERENCE   PMID:12107138\n",
      "  AUTHORS   Verhamme DT, Postma PW, Crielaard W, Hellingwerf KJ.\n",
      "  TITLE     Cooperativity in signal transfer through the Uhp system of Escherichia coli.\n",
      "  JOURNAL   J Bacteriol 184:4205-10 (2002)\n",
      "            DOI:10.1128/JB.184.15.4205-4210.2002\n",
      "REFERENCE   PMID:13129944\n",
      "  AUTHORS   Hagiwara D, Sugiura M, Oshima T, Mori H, Aiba H, Yamashino T, Mizuno T.\n",
      "  TITLE     Genome-wide analyses revealing a signaling network of the RcsC-YojN-RcsB phosphorelay system in Escherichia coli.\n",
      "  JOURNAL   J Bacteriol 185:5735-46 (2003)\n",
      "            DOI:10.1128/JB.185.19.5735-5746.2003\n",
      "REFERENCE   PMID:12951338\n",
      "  AUTHORS   Hirakawa H, Nishino K, Yamada J, Hirata T, Yamaguchi A.\n",
      "  TITLE     Beta-lactam resistance modulated by the overexpression of response regulators of two-component signal transduction systems in Escherichia coli.\n",
      "  JOURNAL   J Antimicrob Chemother 52:576-82 (2003)\n",
      "            DOI:10.1093/jac/dkg406\n",
      "REFERENCE   PMID:12193630\n",
      "  AUTHORS   Suzuki K, Wang X, Weilbacher T, Pernestig AK, Melefors O, Georgellis D, Babitzke P, Romeo T.\n",
      "  TITLE     Regulatory circuitry of the CsrA/CsrB and BarA/UvrY systems of Escherichia coli.\n",
      "  JOURNAL   J Bacteriol 184:5130-40 (2002)\n",
      "            DOI:10.1128/JB.184.18.5130-5140.2002\n",
      "REFERENCE   PMID:10923791\n",
      "  AUTHORS   Kato A, Ohnishi H, Yamamoto K, Furuta E, Tanabe H, Utsumi R.\n",
      "  TITLE     Transcription of emrKY is regulated by the EvgA-EvgS two-component system in Escherichia coli K-12.\n",
      "  JOURNAL   Biosci Biotechnol Biochem 64:1203-9 (2000)\n",
      "            DOI:10.1271/bbb.64.1203\n",
      "REFERENCE   PMID:14679219\n",
      "  AUTHORS   Ansaldi M, Dubnau D.\n",
      "  TITLE     Diversifying selection at the Bacillus quorum-sensing locus and determinants of modification specificity during synthesis of the ComX pheromone.\n",
      "  JOURNAL   J Bacteriol 186:15-21 (2004)\n",
      "            DOI:10.1128/JB.186.1.15-21.2004\n",
      "REFERENCE   PMID:16035234\n",
      "  AUTHORS   Gabdrakhmanova L, Vishniakov I, Sharipova M, Balaban N, Kostrov S, Leshchinskaya I.\n",
      "  TITLE     Salt stress induction of glutamyl endopeptidase biosynthesis in Bacillus intermedius.\n",
      "  JOURNAL   Microbiol Res 160:233-42 (2005)\n",
      "            DOI:10.1016/j.micres.2004.05.005\n",
      "REFERENCE   PMID:11285232\n",
      "  AUTHORS   Aguilar PS, Hernandez-Arriaga AM, Cybulski LE, Erazo AC, de Mendoza D.\n",
      "  TITLE     Molecular basis of thermosensing: a two-component signal transduction thermometer in Bacillus subtilis.\n",
      "  JOURNAL   EMBO J 20:1681-91 (2001)\n",
      "            DOI:10.1093/emboj/20.7.1681\n",
      "REFERENCE   PMID:12864861\n",
      "  AUTHORS   Kuroda M, Kuroda H, Oshima T, Takeuchi F, Mori H, Hiramatsu K.\n",
      "  TITLE     Two-component system VraSR positively modulates the regulation of cell-wall biosynthesis pathway in Staphylococcus aureus.\n",
      "  JOURNAL   Mol Microbiol 49:807-21 (2003)\n",
      "            DOI:10.1046/j.1365-2958.2003.03599.x\n",
      "REFERENCE   PMID:15073296\n",
      "  AUTHORS   Saini DK, Malhotra V, Dey D, Pant N, Das TK, Tyagi JS.\n",
      "  TITLE     DevR-DevS is a bona fide two-component system of Mycobacterium tuberculosis that is hypoxia-responsive in the absence of the DNA-binding domain of DevR.\n",
      "  JOURNAL   Microbiology 150:865-75 (2004)\n",
      "            DOI:10.1099/mic.0.26218-0\n",
      "REFERENCE   PMID:12426351\n",
      "  AUTHORS   Fedtke I, Kamps A, Krismer B, Gotz F.\n",
      "  TITLE     The nitrate reductase and nitrite reductase operons and the narT gene of Staphylococcus carnosus are positively controlled by the novel two-component system NreBC.\n",
      "  JOURNAL   J Bacteriol 184:6624-34 (2002)\n",
      "            DOI:10.1128/JB.184.23.6624-6634.2002\n",
      "REFERENCE   PMID:8416895\n",
      "  AUTHORS   Shiau SP, Chen P, Reitzer LJ.\n",
      "  TITLE     Effects of insertions and deletions in glnG (ntrC) of Escherichia coli on nitrogen regulator I-dependent DNA binding and transcriptional activation.\n",
      "  JOURNAL   J Bacteriol 175:190-9 (1993)\n",
      "            DOI:10.1128/JB.175.1.190-199.1993\n",
      "REFERENCE   PMID:11243806\n",
      "  AUTHORS   Leonhartsberger S, Huber A, Lottspeich F, Bock A.\n",
      "  TITLE     The hydH/G Genes from Escherichia coli code for a zinc and lead responsive two-component regulatory system.\n",
      "  JOURNAL   J Mol Biol 307:93-105 (2001)\n",
      "            DOI:10.1006/jmbi.2000.4451\n",
      "REFERENCE   PMID:15200682\n",
      "  AUTHORS   Lioliou EE, Kyriakidis DA.\n",
      "  TITLE     The role of bacterial antizyme: From an inhibitory protein to AtoC transcriptional regulator.\n",
      "  JOURNAL   Microb Cell Fact 3:8 (2004)\n",
      "            DOI:10.1186/1475-2859-3-8\n",
      "REFERENCE   PMID:15773991\n",
      "  AUTHORS   Yamamoto K, Ishihama A\n",
      "  TITLE     Transcriptional response of Escherichia coli to external copper.\n",
      "  JOURNAL   Mol Microbiol 56:215-27 (2005)\n",
      "            DOI:10.1111/j.1365-2958.2005.04532.x\n",
      "REFERENCE   PMID:10585546\n",
      "  AUTHORS   Ashby MK, Mullineaux CW\n",
      "  TITLE     Cyanobacterial ycf27 gene products regulate energy transfer from phycobilisomes to photosystems I and II.\n",
      "  JOURNAL   FEMS Microbiol Lett 181:253-60 (1999)\n",
      "            DOI:10.1111/j.1574-6968.1999.tb08852.x\n",
      "REFERENCE   PMID:11849552\n",
      "  AUTHORS   Lopez-Maury L, Garcia-Dominguez M, Florencio FJ, Reyes JC\n",
      "  TITLE     A two-component signal transduction system involved in nickel sensing in the cyanobacterium Synechocystis sp. PCC 6803.\n",
      "  JOURNAL   Mol Microbiol 43:247-56 (2002)\n",
      "            DOI:10.1046/j.1365-2958.2002.02741.x\n",
      "REFERENCE   PMID:17526710\n",
      "  AUTHORS   Glover RT, Kriakov J, Garforth SJ, Baughn AD, Jacobs WR Jr\n",
      "  TITLE     The two-component regulatory system senX3-regX3 regulates phosphate-dependent gene expression in Mycobacterium smegmatis.\n",
      "  JOURNAL   J Bacteriol 189:5495-503 (2007)\n",
      "            DOI:10.1128/JB.00190-07\n",
      "REFERENCE   PMID:15175302\n",
      "  AUTHORS   Hsiao HY, He Q, Van Waasbergen LG, Grossman AR\n",
      "  TITLE     Control of photosynthetic and high-light-responsive genes by the histidine kinase DspA: negative and positive regulation and interactions between signal transduction pathways.\n",
      "  JOURNAL   J Bacteriol 186:3882-8 (2004)\n",
      "            DOI:10.1128/JB.186.12.3882-3888.2004\n",
      "REFERENCE   PMID:12626498\n",
      "  AUTHORS   Mutsuda M, Michel KP, Zhang X, Montgomery BL, Golden SS\n",
      "  TITLE     Biochemical properties of CikA, an unusual phytochrome-like histidine protein kinase that resets the circadian clock in Synechococcus elongatus PCC 7942.\n",
      "  JOURNAL   J Biol Chem 278:19102-10 (2003)\n",
      "            DOI:10.1074/jbc.M213255200\n",
      "REFERENCE   PMID:9278513\n",
      "  AUTHORS   Yeh KC, Wu SH, Murphy JT, Lagarias JC\n",
      "  TITLE     A cyanobacterial phytochrome two-component light sensory system.\n",
      "  JOURNAL   Science 277:1505-8 (1997)\n",
      "            DOI:10.1126/science.277.5331.1505\n",
      "REFERENCE   PMID:7635815\n",
      "  AUTHORS   Niu S, Jiang SQ, Hong J\n",
      "  TITLE     Salmonella typhimurium pgtB mutants conferring constitutive expression of phosphoglycerate transporter pgtP independent of pgtC.\n",
      "  JOURNAL   J Bacteriol 177:4297-302 (1995)\n",
      "            DOI:10.1128/JB.177.15.4297-4302.1995\n",
      "REFERENCE   PMID:16882723\n",
      "  AUTHORS   Takai N, Nakajima M, Oyama T, Kito R, Sugita C, Sugita M, Kondo T, Iwasaki H\n",
      "  TITLE     A KaiC-associating SasA-RpaA two-component regulatory system as a major circadian timing mediator in cyanobacteria.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 103:12109-14 (2006)\n",
      "            DOI:10.1073/pnas.0602955103\n",
      "REFERENCE   PMID:11803016\n",
      "  AUTHORS   Janausch IG, Zientz E, Tran QH, Kroger A, Unden G\n",
      "  TITLE     C4-dicarboxylate carriers and sensors in bacteria.\n",
      "  JOURNAL   Biochim Biophys Acta 1553:39-56 (2002)\n",
      "            DOI:10.1016/S0005-2728(01)00233-X\n",
      "REFERENCE   PMID:18725229\n",
      "  AUTHORS   Zhou YF, Nan B, Nan J, Ma Q, Panjikar S, Liang YH, Wang Y, Su XD\n",
      "  TITLE     C4-dicarboxylates sensing mechanism revealed by the crystal structures of DctB sensor domain.\n",
      "  JOURNAL   J Mol Biol 383:49-61 (2008)\n",
      "            DOI:10.1016/j.jmb.2008.08.010\n",
      "REFERENCE   PMID:14523104\n",
      "  AUTHORS   Novick RP, Jiang D\n",
      "  TITLE     The staphylococcal saeRS system coordinates environmental signals with agr quorum sensing.\n",
      "  JOURNAL   Microbiology 149:2709-17 (2003)\n",
      "            DOI:10.1099/mic.0.26575-0\n",
      "REFERENCE   PMID:17981963\n",
      "  AUTHORS   Leech AJ, Sprinkle A, Wood L, Wozniak DJ, Ohman DE\n",
      "  TITLE     The NtrC family regulator AlgB, which controls alginate biosynthesis in mucoid Pseudomonas aeruginosa, binds directly to the algD promoter.\n",
      "  JOURNAL   J Bacteriol 190:581-9 (2008)\n",
      "            DOI:10.1128/JB.01307-07\n",
      "REFERENCE   PMID:18455986\n",
      "  AUTHORS   Paul R, Jaeger T, Abel S, Wiederkehr I, Folcher M, Biondi EG, Laub MT, Jenal U\n",
      "  TITLE     Allosteric regulation of histidine kinases by their cognate response regulator determines cell fate.\n",
      "  JOURNAL   Cell 133:452-61 (2008)\n",
      "            DOI:10.1016/j.cell.2008.02.045\n",
      "REFERENCE   PMID:15101991\n",
      "  AUTHORS   Whitchurch CB, Leech AJ, Young MD, Kennedy D, Sargent JL, Bertrand JJ, Semmler AB, Mellick AS, Martin PR, Alm RA, Hobbs M, Beatson SA, Huang B, Nguyen L, Commolli JC, Engel JN, Darzins A, Mattick JS.\n",
      "  TITLE     Characterization of a complex chemosensory signal transduction system which controls twitching motility in Pseudomonas aeruginosa.\n",
      "  JOURNAL   Mol Microbiol 52:873-93 (2004)\n",
      "            DOI:10.1111/j.1365-2958.2004.04026.x\n",
      "REFERENCE   PMID:15170479\n",
      "  AUTHORS   Yoshihara S, Ikeuchi M.\n",
      "  TITLE     Phototactic motility in the unicellular cyanobacterium Synechocystis sp. PCC 6803.\n",
      "  JOURNAL   Photochem Photobiol Sci 3:512-8 (2004)\n",
      "            DOI:10.1039/b402320j\n",
      "REFERENCE   PMID:12039966\n",
      "  AUTHORS   Ogawa T, Bao DH, Katoh H, Shibata M, Pakrasi HB, Bhattacharyya-Pakrasi M\n",
      "  TITLE     A two-component signal transduction pathway regulates manganese homeostasis in Synechocystis 6803, a photosynthetic organism.\n",
      "  JOURNAL   J Biol Chem 277:28981-6 (2002)\n",
      "            DOI:10.1074/jbc.M204175200\n",
      "REFERENCE   PMID:12417709\n",
      "  AUTHORS   Yamaguchi K, Suzuki I, Yamamoto H, Lyukevich A, Bodrova I, Los DA, Piven I, Zinchenko V, Kanehisa M, Murata N\n",
      "  TITLE     A two-component Mn2+-sensing system negatively regulates expression of the mntCAB operon in Synechocystis.\n",
      "  JOURNAL   Plant Cell 14:2901-13 (2002)\n",
      "            DOI:10.1105/tpc.006262\n",
      "REFERENCE   PMID:15273097\n",
      "  AUTHORS   Mascher T, Zimmer SL, Smith TA, Helmann JD\n",
      "  TITLE     Antibiotic-inducible promoter regulated by the cell envelope stress-sensing two-component system LiaRS of Bacillus subtilis.\n",
      "  JOURNAL   Antimicrob Agents Chemother 48:2888-96 (2004)\n",
      "            DOI:10.1128/AAC.48.8.2888-2896.2004\n",
      "REFERENCE   PMID:15856219\n",
      "  AUTHORS   Hyyrylainen HL, Sarvas M, Kontinen VP\n",
      "  TITLE     Transcriptome analysis of the secretion stress response of Bacillus subtilis.\n",
      "  JOURNAL   Appl Microbiol Biotechnol 67:389-96 (2005)\n",
      "            DOI:10.1007/s00253-005-1898-1\n",
      "REFERENCE   PMID:15941986\n",
      "  AUTHORS   Serizawa M, Sekiguchi J\n",
      "  TITLE     The Bacillus subtilis YdfHI two-component system regulates the transcription of ydfJ, a member of the RND superfamily.\n",
      "  JOURNAL   Microbiology 151:1769-78 (2005)\n",
      "            DOI:10.1099/mic.0.27619-0\n",
      "REFERENCE   PMID:12890034\n",
      "  AUTHORS   Ohki R, Giyanto, Tateno K, Masuyama W, Moriya S, Kobayashi K, Ogasawara N\n",
      "  TITLE     The BceRS two-component regulatory system induces expression of the bacitracin transporter, BceAB, in Bacillus subtilis.\n",
      "  JOURNAL   Mol Microbiol 49:1135-44 (2003)\n",
      "            DOI:10.1046/j.1365-2958.2003.03653.x\n",
      "REFERENCE   PMID:17322186\n",
      "  AUTHORS   Ogura M, Tsukahara K, Hayashi K, Tanaka T\n",
      "  TITLE     The Bacillus subtilis NatK-NatR two-component system regulates expression of the natAB operon encoding an ABC transporter for sodium ion extrusion.\n",
      "  JOURNAL   Microbiology 153:667-75 (2007)\n",
      "            DOI:10.1099/mic.0.2006/003673-0\n",
      "REFERENCE   PMID:10972810\n",
      "  AUTHORS   Yamamoto H, Murata M, Sekiguchi J\n",
      "  TITLE     The CitST two-component system regulates the expression of the Mg-citrate transporter in Bacillus subtilis.\n",
      "  JOURNAL   Mol Microbiol 37:898-912 (2000)\n",
      "            DOI:10.1046/j.1365-2958.2000.02055.x\n",
      "REFERENCE   PMID:15995196\n",
      "  AUTHORS   Satomura T, Shimura D, Asai K, Sadaie Y, Hirooka K, Fujita Y\n",
      "  TITLE     Enhancement of glutamine utilization in Bacillus subtilis through the GlnK-GlnL two-component regulatory system.\n",
      "  JOURNAL   J Bacteriol 187:4813-21 (2005)\n",
      "            DOI:10.1128/JB.187.14.4813-4821.2005\n",
      "REFERENCE   PMID:12949160\n",
      "  AUTHORS   Doan T, Servant P, Tojo S, Yamaguchi H, Lerondel G, Yoshida K, Fujita Y, Aymerich S\n",
      "  TITLE     The Bacillus subtilis ywkA gene encodes a malic enzyme and its transcription is activated by the YufL/YufM two-component system in response to malate.\n",
      "  JOURNAL   Microbiology 149:2331-43 (2003)\n",
      "            DOI:10.1099/mic.0.26256-0\n",
      "REFERENCE   PMID:12949159\n",
      "  AUTHORS   Tanaka K, Kobayashi K, Ogasawara N\n",
      "  TITLE     The Bacillus subtilis YufLM two-component system regulates the expression of the malate transporters MaeN (YufR) and YflS, and is essential for utilization of malate in minimal medium.\n",
      "  JOURNAL   Microbiology 149:2317-29 (2003)\n",
      "            DOI:10.1099/mic.0.26257-0\n",
      "REFERENCE   PMID:10708364\n",
      "  AUTHORS   Asai K, Baik SH, Kasahara Y, Moriya S, Ogasawara N\n",
      "  TITLE     Regulation of the transport system for C4-dicarboxylic acids in Bacillus subtilis.\n",
      "  JOURNAL   Microbiology 146 ( Pt 2):263-71 (2000)\n",
      "            DOI:10.1099/00221287-146-2-263\n",
      "REFERENCE   PMID:8455557\n",
      "  AUTHORS   Hamblin MJ, Shaw JG, Kelly DJ\n",
      "  TITLE     Sequence analysis and interposon mutagenesis of a sensor-kinase (DctS) and response-regulator (DctR) controlling synthesis of the high-affinity C4-dicarboxylate transport system in Rhodobacter capsulatus.\n",
      "  JOURNAL   Mol Gen Genet 237:215-24 (1993)\n",
      "            DOI:10.1007/BF00282803\n",
      "REFERENCE   PMID:9287004\n",
      "  AUTHORS   Forward JA, Behrendt MC, Wyborn NR, Cross R, Kelly DJ\n",
      "  TITLE     TRAP transporters: a new family of periplasmic solute transport systems encoded by the dctPQM genes of Rhodobacter capsulatus and by homologs in diverse gram-negative bacteria.\n",
      "  JOURNAL   J Bacteriol 179:5482-93 (1997)\n",
      "            DOI:10.1128/JB.179.17.5482-5493.1997\n",
      "REFERENCE   PMID:16820295\n",
      "  AUTHORS   Martin B, Quentin Y, Fichant G, Claverys JP\n",
      "  TITLE     Independent evolution of competence regulatory cascades in streptococci?\n",
      "  JOURNAL   Trends Microbiol 14:339-45 (2006)\n",
      "            DOI:10.1016/j.tim.2006.06.007\n",
      "REFERENCE   PMID:10231485\n",
      "  AUTHORS   Hensel M, Hinsley AP, Nikolaus T, Sawers G, Berks BC\n",
      "  TITLE     The genetic basis of tetrathionate respiration in Salmonella typhimurium.\n",
      "  JOURNAL   Mol Microbiol 32:275-87 (1999)\n",
      "            DOI:10.1046/j.1365-2958.1999.01345.x\n",
      "REFERENCE   PMID:18028314\n",
      "  AUTHORS   Guvener ZT, Harwood CS\n",
      "  TITLE     Subcellular location characteristics of the Pseudomonas aeruginosa GGDEF protein, WspR, indicate that it produces cyclic-di-GMP in response to growth on surfaces.\n",
      "  JOURNAL   Mol Microbiol 66:1459-73 (2007)\n",
      "            DOI:10.1111/j.1365-2958.2007.06008.x\n",
      "REFERENCE   PMID:17339222\n",
      "  AUTHORS   Bantinaki E, Kassen R, Knight CG, Robinson Z, Spiers AJ, Rainey PB\n",
      "  TITLE     Adaptive divergence in experimental populations of Pseudomonas fluorescens. III. Mutational origins of wrinkly spreader diversity.\n",
      "  JOURNAL   Genetics 176:441-53 (2007)\n",
      "            DOI:10.1534/genetics.106.069906\n",
      "REFERENCE   PMID:11069677\n",
      "  AUTHORS   Jiang M, Shao W, Perego M, Hoch JA\n",
      "  TITLE     Multiple histidine kinases regulate entry into stationary phase and sporulation in Bacillus subtilis.\n",
      "  JOURNAL   Mol Microbiol 38:535-42 (2000)\n",
      "            DOI:10.1046/j.1365-2958.2000.02148.x\n",
      "REFERENCE   PMID:19781638\n",
      "  AUTHORS   Jenal U\n",
      "  TITLE     The role of proteolysis in the Caulobacter crescentus cell cycle and development.\n",
      "  JOURNAL   Res Microbiol 160:687-95 (2009)\n",
      "            DOI:10.1016/j.resmic.2009.09.006\n",
      "REFERENCE   PMID:19246239\n",
      "  AUTHORS   Jenal U, Galperin MY\n",
      "  TITLE     Single domain response regulators: molecular switches with emerging roles in cell organization and dynamics.\n",
      "  JOURNAL   Curr Opin Microbiol 12:152-60 (2009)\n",
      "            DOI:10.1016/j.mib.2009.01.010\n",
      "REFERENCE   PMID:16640581\n",
      "  AUTHORS   Drepper T, Wiethaus J, Giaourakis D, Gross S, Schubert B, Vogt M, Wiencek Y, McEwan AG, Masepohl B\n",
      "  TITLE     Cross-talk towards the response regulator NtrC controlling nitrogen metabolism in Rhodobacter capsulatus.\n",
      "  JOURNAL   FEMS Microbiol Lett 258:250-6 (2006)\n",
      "            DOI:10.1111/j.1574-6968.2006.00228.x\n",
      "REFERENCE   PMID:1661370\n",
      "  AUTHORS   Pawlowski K, Klosse U, de Bruijn FJ\n",
      "  TITLE     Characterization of a novel Azorhizobium caulinodans ORS571 two-component regulatory system, NtrY/NtrX, involved in nitrogen fixation and metabolism.\n",
      "  JOURNAL   Mol Gen Genet 231:124-38 (1991)\n",
      "            DOI:10.1007/BF00293830\n",
      "REFERENCE   PMID:18422428\n",
      "  AUTHORS   Chatterjee S, Almeida RP, Lindow S\n",
      "  TITLE     Living in two worlds: the plant and insect lifestyles of Xylella fastidiosa.\n",
      "  JOURNAL   Annu Rev Phytopathol 46:243-71 (2008)\n",
      "            DOI:10.1146/annurev.phyto.45.062806.094342\n",
      "REFERENCE   PMID:18506032\n",
      "  AUTHORS   Dow M\n",
      "  TITLE     Diversification of the function of cell-to-cell signaling in regulation of virulence within plant pathogenic xanthomonads.\n",
      "  JOURNAL   Sci Signal 1:pe23 (2008)\n",
      "            DOI:10.1126/stke.121pe23\n",
      "REFERENCE   PMID:17902112\n",
      "  AUTHORS   Colnaghi Simionato AV, da Silva DS, Lambais MR, Carrilho E\n",
      "  TITLE     Characterization of a putative Xylella fastidiosa diffusible signal factor by HRGC-EI-MS.\n",
      "  JOURNAL   J Mass Spectrom 42:1375-81 (2007)\n",
      "            DOI:10.1002/jms.1325\n",
      "REFERENCE   PMID:19203648\n",
      "  AUTHORS   Miller LD, Russell MH, Alexandre G\n",
      "  TITLE     Diversity in bacterial chemotactic responses and niche adaptation.\n",
      "  JOURNAL   Adv Appl Microbiol 66:53-75 (2009)\n",
      "            DOI:10.1016/S0065-2164(08)00803-4\n",
      "REFERENCE   PMID:8760916\n",
      "  AUTHORS   Eder S, Shi L, Jensen K, Yamane K, Hulett FM\n",
      "  TITLE     A Bacillus subtilis secreted phosphodiesterase/alkaline phosphatase is the product of a Pho regulon gene, phoD.\n",
      "  JOURNAL   Microbiology 142 ( Pt 8):2041-7 (1996)\n",
      "            DOI:10.1099/13500872-142-8-2041\n",
      "REL_PATHWAY pae00190  Oxidative phosphorylation\n",
      "            pae00550  Peptidoglycan biosynthesis\n",
      "            pae00910  Nitrogen metabolism\n",
      "            pae02030  Bacterial chemotaxis\n",
      "            pae02040  Flagellar assembly\n",
      "            pae03070  Bacterial secretion system\n",
      "KO_PATHWAY  ko02020\n",
      "///\n",
      "path:pae02024\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae02024                    Pathway\n",
      "NAME        Quorum sensing - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Quorum sensing (QS) is a regulatory system that allows bacteria to share information about cell density and adjust gene expression accordingly. All QS bacteria produce and release chemical signal molecules called autoinducers (AIs) that increase in concentration as a function of cell density. The most commonly studied AIs belong to one of the following three categories: acylated homoserine lactones, also referred to as AI-1, used by Gram-negative bacteria; peptide signals, used by Gram-positive bacteria; and AI-2, used by both Gram-negative and Gram-positive bacteria. QS can be divided into at least 4 steps: production of AIs by the bacterial cell; release of AIs, either actively or passively, into the surrounding environment; recognition of AIs by specific receptors; and leading to changes in gene regulation once they exceed a threshold concentration. The processes controlled by QS include virulence, competence, conjugation, antibiotic production, motility, sporulation, and biofilm formation.\n",
      "CLASS       Cellular Processes; Cellular community - prokaryotes\n",
      "PATHWAY_MAP pae02024  Quorum sensing\n",
      "DBLINKS     GO: 0009372\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4944  hfq; RNA-binding protein Hfq [KO:K03666]\n",
      "line in gene PA4944  hfq; RNA-binding protein Hfq [KO:K03666]\n",
      "            PA1432  lasI; acyl-homoserine-lactone synthase [KO:K13060] [EC:2.3.1.184]\n",
      "line in gene PA1432  lasI; acyl-homoserine-lactone synthase [KO:K13060] [EC:2.3.1.184]\n",
      "            PA1430  lasR; transcriptional regulator LasR [KO:K18304]\n",
      "line in gene PA1430  lasR; transcriptional regulator LasR [KO:K18304]\n",
      "            PA0609  trpE; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "line in gene PA0609  trpE; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "            PA1001  phnA; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "line in gene PA1001  phnA; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "            PA0649  trpG; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "line in gene PA0649  trpG; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "            PA1002  phnB; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "line in gene PA1002  phnB; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "            PA0996  pqsA; anthranilate--CoA ligase [KO:K18000] [EC:6.2.1.32]\n",
      "line in gene PA0996  pqsA; anthranilate--CoA ligase [KO:K18000] [EC:6.2.1.32]\n",
      "            PA0997  pqsB; hypothetical protein [KO:K18001] [EC:2.3.1.230]\n",
      "line in gene PA0997  pqsB; hypothetical protein [KO:K18001] [EC:2.3.1.230]\n",
      "            PA0998  pqsC; hypothetical protein [KO:K18002] [EC:2.3.1.230]\n",
      "line in gene PA0998  pqsC; hypothetical protein [KO:K18002] [EC:2.3.1.230]\n",
      "            PA0999  pqsD; 3-oxoacyl-ACP synthase [KO:K18003] [EC:2.3.1.262]\n",
      "line in gene PA0999  pqsD; 3-oxoacyl-ACP synthase [KO:K18003] [EC:2.3.1.262]\n",
      "            PA1000  pqsE; thioesterase PqsE [KO:K20257] [EC:3.1.2.32]\n",
      "line in gene PA1000  pqsE; thioesterase PqsE [KO:K20257] [EC:3.1.2.32]\n",
      "            PA2587  pqsH; 2-heptyl-3-hydroxy-4(1H)-quinolone synthase [KO:K17940] [EC:1.14.13.182]\n",
      "line in gene PA2587  pqsH; 2-heptyl-3-hydroxy-4(1H)-quinolone synthase [KO:K17940] [EC:1.14.13.182]\n",
      "            PA1003  mvfR; transcriptional regulator MvfR [KO:K19735]\n",
      "line in gene PA1003  mvfR; transcriptional regulator MvfR [KO:K19735]\n",
      "            PA3476  rhlI; acyl-homoserine-lactone synthase [KO:K13061] [EC:2.3.1.184]\n",
      "line in gene PA3476  rhlI; acyl-homoserine-lactone synthase [KO:K13061] [EC:2.3.1.184]\n",
      "            PA3477  rhlR; transcriptional regulator RhlR [KO:K18099]\n",
      "line in gene PA3477  rhlR; transcriptional regulator RhlR [KO:K18099]\n",
      "            PA3479  rhlA; rhamnosyltransferase subunit A [KO:K18100] [EC:2.4.1.-]\n",
      "line in gene PA3479  rhlA; rhamnosyltransferase subunit A [KO:K18100] [EC:2.4.1.-]\n",
      "            PA3478  rhlB; rhamnosyltransferase subunit B [KO:K18101] [EC:2.4.1.-]\n",
      "line in gene PA3478  rhlB; rhamnosyltransferase subunit B [KO:K18101] [EC:2.4.1.-]\n",
      "            PA1130  rhlC; rhamnosyltransferase [KO:K12990] [EC:2.4.1.-]\n",
      "line in gene PA1130  rhlC; rhamnosyltransferase [KO:K12990] [EC:2.4.1.-]\n",
      "            PA1871  lasA; protease LasA [KO:K08642] [EC:3.4.24.-]\n",
      "line in gene PA1871  lasA; protease LasA [KO:K08642] [EC:3.4.24.-]\n",
      "            PA3724  lasB; elastase LasB [KO:K01399] [EC:3.4.24.26]\n",
      "line in gene PA3724  lasB; elastase LasB [KO:K01399] [EC:3.4.24.26]\n",
      "            PA2570  lecA; PA-I galactophilic lectin [KO:K20258]\n",
      "line in gene PA2570  lecA; PA-I galactophilic lectin [KO:K20258]\n",
      "            PA2569  hypothetical protein [KO:K20259]\n",
      "line in gene PA2569  hypothetical protein [KO:K20259]\n",
      "            PA4210  phzA1; phenazine biosynthesis protein [KO:K20260]\n",
      "line in gene PA4210  phzA1; phenazine biosynthesis protein [KO:K20260]\n",
      "            PA1899  phzA2; phenazine biosynthesis protein PhzA [KO:K20260]\n",
      "line in gene PA1899  phzA2; phenazine biosynthesis protein PhzA [KO:K20260]\n",
      "            PA4211  phzB1; phenazine biosynthesis protein [KO:K20260]\n",
      "line in gene PA4211  phzB1; phenazine biosynthesis protein [KO:K20260]\n",
      "            PA1900  phzB2; phenazine biosynthesis protein PhzB [KO:K20260]\n",
      "line in gene PA1900  phzB2; phenazine biosynthesis protein PhzB [KO:K20260]\n",
      "            PA1750  phospho-2-dehydro-3-deoxyheptonate aldolase [KO:K01626] [EC:2.5.1.54]\n",
      "line in gene PA1750  phospho-2-dehydro-3-deoxyheptonate aldolase [KO:K01626] [EC:2.5.1.54]\n",
      "            PA1901  phzC2; phenazine biosynthesis protein PhzC [KO:K01626] [EC:2.5.1.54]\n",
      "line in gene PA1901  phzC2; phenazine biosynthesis protein PhzC [KO:K01626] [EC:2.5.1.54]\n",
      "            PA2843  aldolase [KO:K01626] [EC:2.5.1.54]\n",
      "line in gene PA2843  aldolase [KO:K01626] [EC:2.5.1.54]\n",
      "            PA2943  phospho-2-dehydro-3-deoxyheptonate aldolase [KO:K01626] [EC:2.5.1.54]\n",
      "line in gene PA2943  phospho-2-dehydro-3-deoxyheptonate aldolase [KO:K01626] [EC:2.5.1.54]\n",
      "            PA4212  phzC1; phenazine biosynthesis protein PhzC [KO:K01626] [EC:2.5.1.54]\n",
      "line in gene PA4212  phzC1; phenazine biosynthesis protein PhzC [KO:K01626] [EC:2.5.1.54]\n",
      "            PA1902  phzD2; phenazine biosynthesis protein PhzD [KO:K20261] [EC:3.3.2.15]\n",
      "line in gene PA1902  phzD2; phenazine biosynthesis protein PhzD [KO:K20261] [EC:3.3.2.15]\n",
      "            PA4213  phzD1; phenazine biosynthesis protein PhzD [KO:K20261] [EC:3.3.2.15]\n",
      "line in gene PA4213  phzD1; phenazine biosynthesis protein PhzD [KO:K20261] [EC:3.3.2.15]\n",
      "            PA1903  phzE2; phenazine biosynthesis protein PhzE [KO:K13063] [EC:2.6.1.86]\n",
      "line in gene PA1903  phzE2; phenazine biosynthesis protein PhzE [KO:K13063] [EC:2.6.1.86]\n",
      "            PA4214  phzE1; phenazine biosynthesis protein PhzE [KO:K13063] [EC:2.6.1.86]\n",
      "line in gene PA4214  phzE1; phenazine biosynthesis protein PhzE [KO:K13063] [EC:2.6.1.86]\n",
      "            PA1904  phzF2; trans-2,3-dihydro-3-hydroxyanthranilate isomerase [KO:K06998] [EC:5.3.3.17]\n",
      "line in gene PA1904  phzF2; trans-2,3-dihydro-3-hydroxyanthranilate isomerase [KO:K06998] [EC:5.3.3.17]\n",
      "            PA4215  phzF1; trans-2,3-dihydro-3-hydroxyanthranilate isomerase [KO:K06998] [EC:5.3.3.17]\n",
      "line in gene PA4215  phzF1; trans-2,3-dihydro-3-hydroxyanthranilate isomerase [KO:K06998] [EC:5.3.3.17]\n",
      "            PA1905  phzG2; pyridoxamine 5'-phosphate oxidase [KO:K20262] [EC:1.10.3.16]\n",
      "line in gene PA1905  phzG2; pyridoxamine 5'-phosphate oxidase [KO:K20262] [EC:1.10.3.16]\n",
      "            PA4216  phzG1; pyridoxamine 5'-phosphate oxidase [KO:K20262] [EC:1.10.3.16]\n",
      "line in gene PA4216  phzG1; pyridoxamine 5'-phosphate oxidase [KO:K20262] [EC:1.10.3.16]\n",
      "            PA4777  pmrB; two-component regulator system signal sensor kinase PmrB [KO:K07645] [EC:2.7.13.3]\n",
      "line in gene PA4777  pmrB; two-component regulator system signal sensor kinase PmrB [KO:K07645] [EC:2.7.13.3]\n",
      "            PA2480  two-component sensor [KO:K07645] [EC:2.7.13.3]\n",
      "line in gene PA2480  two-component sensor [KO:K07645] [EC:2.7.13.3]\n",
      "            PA4776  pmrA; two-component regulator system response regulator PmrA [KO:K07666]\n",
      "line in gene PA4776  pmrA; two-component regulator system response regulator PmrA [KO:K07666]\n",
      "            PA2479  two-component response regulator [KO:K07666]\n",
      "line in gene PA2479  two-component response regulator [KO:K07666]\n",
      "            PA1637  kdpE; two-component response regulator KdpE [KO:K07667]\n",
      "line in gene PA1637  kdpE; two-component response regulator KdpE [KO:K07667]\n",
      "            PA4804  amino acid permease [KO:K20265]\n",
      "line in gene PA4804  amino acid permease [KO:K20265]\n",
      "            PA0709  hypothetical protein [KO:K11530] [EC:5.3.1.32]\n",
      "line in gene PA0709  hypothetical protein [KO:K11530] [EC:5.3.1.32]\n",
      "            PA1136  transcriptional regulator [KO:K18098]\n",
      "line in gene PA1136  transcriptional regulator [KO:K18098]\n",
      "            PA0602  ABC transporter [KO:K02055]\n",
      "line in gene PA0602  ABC transporter [KO:K02055]\n",
      "            PA0604  ABC transporter [KO:K02055]\n",
      "line in gene PA0604  ABC transporter [KO:K02055]\n",
      "            PA0222  hypothetical protein [KO:K02055]\n",
      "line in gene PA0222  hypothetical protein [KO:K02055]\n",
      "            PA3250  hypothetical protein [KO:K02055]\n",
      "line in gene PA3250  hypothetical protein [KO:K02055]\n",
      "            PA4985  hypothetical protein [KO:K02055]\n",
      "line in gene PA4985  hypothetical protein [KO:K02055]\n",
      "            PA0606  ABC transporter permease [KO:K02053]\n",
      "line in gene PA0606  ABC transporter permease [KO:K02053]\n",
      "            PA3253  ABC transporter permease [KO:K02053]\n",
      "line in gene PA3253  ABC transporter permease [KO:K02053]\n",
      "            PA0605  ABC transporter permease [KO:K02054]\n",
      "line in gene PA0605  ABC transporter permease [KO:K02054]\n",
      "            PA3252  ABC transporter permease [KO:K02054]\n",
      "line in gene PA3252  ABC transporter permease [KO:K02054]\n",
      "            PA0603  ABC transporter ATP-binding protein [KO:K02052]\n",
      "line in gene PA0603  ABC transporter ATP-binding protein [KO:K02052]\n",
      "            PA3254  ABC transporter ATP-binding protein [KO:K02052]\n",
      "line in gene PA3254  ABC transporter ATP-binding protein [KO:K02052]\n",
      "            PA1074  braC; branched-chain amino acid ABC transporter substrate-binding protein BraC [KO:K01999]\n",
      "line in gene PA1074  braC; branched-chain amino acid ABC transporter substrate-binding protein BraC [KO:K01999]\n",
      "            PA4913  ABC transporter [KO:K01999]\n",
      "line in gene PA4913  ABC transporter [KO:K01999]\n",
      "            PA3364  amiC; aliphatic amidase expression-regulating protein [KO:K01999]\n",
      "line in gene PA3364  amiC; aliphatic amidase expression-regulating protein [KO:K01999]\n",
      "            PA1073  braD; branched-chain amino acid ABC transporter permease BraD [KO:K01997]\n",
      "line in gene PA1073  braD; branched-chain amino acid ABC transporter permease BraD [KO:K01997]\n",
      "            PA4912  branched-chain amino acid ABC transporter [KO:K01997]\n",
      "line in gene PA4912  branched-chain amino acid ABC transporter [KO:K01997]\n",
      "            PA1072  braE; branched-chain amino acid ABC transporter permease BraE [KO:K01998]\n",
      "line in gene PA1072  braE; branched-chain amino acid ABC transporter permease BraE [KO:K01998]\n",
      "            PA4911  branched-chain amino acid ABC transporter permease [KO:K01998]\n",
      "line in gene PA4911  branched-chain amino acid ABC transporter permease [KO:K01998]\n",
      "            PA1071  braF; ABC transporter ATP-binding protein [KO:K01995]\n",
      "line in gene PA1071  braF; ABC transporter ATP-binding protein [KO:K01995]\n",
      "            PA4910  ABC transporter ATP-binding protein [KO:K01995]\n",
      "line in gene PA4910  ABC transporter ATP-binding protein [KO:K01995]\n",
      "            PA1070  braG; ABC transporter ATP-binding protein [KO:K01996]\n",
      "line in gene PA1070  braG; ABC transporter ATP-binding protein [KO:K01996]\n",
      "            PA4909  ABC transporter ATP-binding protein [KO:K01996]\n",
      "line in gene PA4909  ABC transporter ATP-binding protein [KO:K01996]\n",
      "            PA1874  hypothetical protein [KO:K20276]\n",
      "line in gene PA1874  hypothetical protein [KO:K20276]\n",
      "            PA3361  lecB; fucose-binding lectin PA-IIL [KO:K20277]\n",
      "line in gene PA3361  lecB; fucose-binding lectin PA-IIL [KO:K20277]\n",
      "            PA3299  fadD1; long-chain-fatty-acid--CoA ligase [KO:K01897] [EC:6.2.1.3]\n",
      "line in gene PA3299  fadD1; long-chain-fatty-acid--CoA ligase [KO:K01897] [EC:6.2.1.3]\n",
      "            PA3300  fadD2; long-chain-fatty-acid--CoA ligase [KO:K01897] [EC:6.2.1.3]\n",
      "line in gene PA3300  fadD2; long-chain-fatty-acid--CoA ligase [KO:K01897] [EC:6.2.1.3]\n",
      "            PA1617  AMP-binding protein [KO:K01897] [EC:6.2.1.3]\n",
      "line in gene PA1617  AMP-binding protein [KO:K01897] [EC:6.2.1.3]\n",
      "            PA0652  vfr; cAMP-regulatory protein [KO:K10914]\n",
      "line in gene PA0652  vfr; cAMP-regulatory protein [KO:K10914]\n",
      "            PA0275  transcriptional regulator [KO:K10914]\n",
      "line in gene PA0275  transcriptional regulator [KO:K10914]\n",
      "            PA5499  np20; transcriptional regulator [KO:K09823]\n",
      "line in gene PA5499  np20; transcriptional regulator [KO:K09823]\n",
      "            PA4047  ribA; GTP cyclohydrolase II [KO:K01497] [EC:3.5.4.25]\n",
      "line in gene PA4047  ribA; GTP cyclohydrolase II [KO:K01497] [EC:3.5.4.25]\n",
      "            PA4056  ribD; riboflavin-specific deaminase/reductase [KO:K11752] [EC:3.5.4.26 1.1.1.193]\n",
      "line in gene PA4056  ribD; riboflavin-specific deaminase/reductase [KO:K11752] [EC:3.5.4.26 1.1.1.193]\n",
      "            PA4815  hypothetical protein [KO:K09936]\n",
      "line in gene PA4815  hypothetical protein [KO:K09936]\n",
      "            PA3320  hypothetical protein [KO:K09936]\n",
      "line in gene PA3320  hypothetical protein [KO:K09936]\n",
      "            PA4206  mexH; resistance-nodulation-cell division (RND) efflux membrane fusion protein [KO:K18306]\n",
      "line in gene PA4206  mexH; resistance-nodulation-cell division (RND) efflux membrane fusion protein [KO:K18306]\n",
      "            PA4207  mexI; resistance-nodulation-cell division (RND) efflux transporter [KO:K18307]\n",
      "line in gene PA4207  mexI; resistance-nodulation-cell division (RND) efflux transporter [KO:K18307]\n",
      "            PA0427  oprM; outer membrane protein OprM [KO:K18139]\n",
      "line in gene PA0427  oprM; outer membrane protein OprM [KO:K18139]\n",
      "            PA5568  inner membrane protein translocase subunit YidC [KO:K03217]\n",
      "line in gene PA5568  inner membrane protein translocase subunit YidC [KO:K03217]\n",
      "            PA4276  secE; preprotein translocase subunit SecE [KO:K03073]\n",
      "line in gene PA4276  secE; preprotein translocase subunit SecE [KO:K03073]\n",
      "            PA4747  secG; preprotein translocase subunit SecG [KO:K03075]\n",
      "line in gene PA4747  secG; preprotein translocase subunit SecG [KO:K03075]\n",
      "            PA4243  secY; preprotein translocase subunit SecY [KO:K03076]\n",
      "line in gene PA4243  secY; preprotein translocase subunit SecY [KO:K03076]\n",
      "            PA3822  preprotein translocase subunit YajC [KO:K03210]\n",
      "line in gene PA3822  preprotein translocase subunit YajC [KO:K03210]\n",
      "            PA4403  secA; preprotein translocase subunit SecA [KO:K03070] [EC:7.4.2.8]\n",
      "line in gene PA4403  secA; preprotein translocase subunit SecA [KO:K03070] [EC:7.4.2.8]\n",
      "            PA0373  ftsY; signal recognition particle receptor FtsY [KO:K03110]\n",
      "line in gene PA0373  ftsY; signal recognition particle receptor FtsY [KO:K03110]\n",
      "            PA5128  secB; preprotein translocase subunit SecB [KO:K03071]\n",
      "line in gene PA5128  secB; preprotein translocase subunit SecB [KO:K03071]\n",
      "            PA3746  ffh; signal recognition particle protein Ffh [KO:K03106] [EC:3.6.5.4]\n",
      "line in gene PA3746  ffh; signal recognition particle protein Ffh [KO:K03106] [EC:3.6.5.4]\n",
      "            PA0844  plcH; hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "line in gene PA0844  plcH; hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "            PA3319  plcN; non-hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "line in gene PA3319  plcN; non-hemolytic phospholipase C [KO:K01114] [EC:3.1.4.3]\n",
      "COMPOUND    C00007  Oxygen\n",
      "            C00009  Orthophosphate\n",
      "            C00059  Sulfate\n",
      "            C00232  Succinate semialdehyde\n",
      "            C00334  4-Aminobutanoate\n",
      "            C00547  L-Noradrenaline\n",
      "            C00788  L-Adrenaline\n",
      "            C01019  6-Deoxy-L-galactose\n",
      "            C11837  N-Butyryl-L-homoserine lactone\n",
      "            C11841  N-(3-Oxooctanoyl)homoserine lactone\n",
      "            C11844  N-Heptanoylhomoserine lactone\n",
      "            C11845  N-(3-Hydroxy-7-cis-tetradecenoyl)homoserine lactone\n",
      "            C11848  2-Heptyl-3-hydroxy-quinolone\n",
      "            C16421  AI-2\n",
      "            C16463  3',5'-Cyclic diGMP\n",
      "            C16640  CAI-1\n",
      "            C18049  N-Acyl-L-homoserine lactone\n",
      "            C18206  cis-11-Methyl-2-dodecenoic acid\n",
      "            C20643  2-Heptyl-4(1H)-quinolone\n",
      "            C20677  N-(4-Coumaroyl)-L-homoserine lactone\n",
      "            C20959  (4S)-4-Hydroxy-5-phosphooxypentane-2,3-dione\n",
      "            C21195  N-(3-Hydroxybutanoyl)-L-homoserine lactone\n",
      "            C21197  N-Hexanoyl-L-homoserine lactone\n",
      "            C21198  N-(3-Oxohexanoyl)-L-homoserine lactone\n",
      "            C21199  N-Octanoyl-L-homoserine lactone\n",
      "            C21200  N-3-Hydroxyoctanoyl-L-homoserine lactone\n",
      "            C21201  N-3-Oxo-dodecanoyl-L-homoserine lactone\n",
      "            C21202  cis-2-Dodecenoic acid\n",
      "            C21220  AIP-1\n",
      "            C21221  CSP\n",
      "            C21222  CSP\n",
      "            C21223  Blp\n",
      "            C21224  SHP2\n",
      "            C21225  SHP3\n",
      "            C21226  cCF10\n",
      "            C21227  iCF10\n",
      "            C21228  cAD1\n",
      "            C21229  iAD1\n",
      "            C21230  ComX pheromone\n",
      "            C21231  PhrA pentapeptide\n",
      "            C21232  PhrC pentapeptide\n",
      "            C21233  PhrE pentapeptide\n",
      "            C21234  PhrF pentapeptide\n",
      "            C21235  PhrG pentapeptide\n",
      "            C21236  PhrH pentapeptide\n",
      "            C21237  PhrK pentapeptide\n",
      "            C21238  NprX peptide\n",
      "            C21239  PapR peptide\n",
      "            C21241  AIP-2\n",
      "            C21242  AIP-3\n",
      "            C21243  AIP-4\n",
      "            C21244  AgrD peptide\n",
      "            C21245  AgrD1 peptide\n",
      "            C21246  GBAP\n",
      "            C21382  R-THMF\n",
      "REFERENCE   PMID:22712853\n",
      "  AUTHORS   Pereira CS, Thompson JA, Xavier KB\n",
      "  TITLE     AI-2-mediated signalling in bacteria.\n",
      "  JOURNAL   FEMS Microbiol Rev 37:156-81 (2013)\n",
      "            DOI:10.1111/j.1574-6976.2012.00345.x\n",
      "REFERENCE   PMID:23965960\n",
      "  AUTHORS   Verma SC, Miyashiro T\n",
      "  TITLE     Quorum sensing in the squid-Vibrio symbiosis.\n",
      "  JOURNAL   Int J Mol Sci 14:16386-401 (2013)\n",
      "            DOI:10.3390/ijms140816386\n",
      "REFERENCE   PMID:24348467\n",
      "  AUTHORS   Norsworthy AN, Visick KL\n",
      "  TITLE     Gimme shelter: how Vibrio fischeri successfully navigates an animal's multiple environments.\n",
      "  JOURNAL   Front Microbiol 4:356 (2013)\n",
      "            DOI:10.3389/fmicb.2013.00356\n",
      "REFERENCE   PMID:24065108\n",
      "  AUTHORS   Papaioannou E, Utari PD, Quax WJ\n",
      "  TITLE     Choosing an appropriate infection model to study quorum sensing inhibition in Pseudomonas infections.\n",
      "  JOURNAL   Int J Mol Sci 14:19309-40 (2013)\n",
      "            DOI:10.3390/ijms140919309\n",
      "REFERENCE   PMID:20738404\n",
      "  AUTHORS   Heeb S, Fletcher MP, Chhabra SR, Diggle SP, Williams P, Camara M\n",
      "  TITLE     Quinolones: from antibiotics to autoinducers.\n",
      "  JOURNAL   FEMS Microbiol Rev 35:247-74 (2011)\n",
      "            DOI:10.1111/j.1574-6976.2010.00247.x\n",
      "REFERENCE   PMID:25866808\n",
      "  AUTHORS   Rasamiravaka T, Labtani Q, Duez P, El Jaziri M\n",
      "  TITLE     The formation of biofilms by Pseudomonas aeruginosa: a review of the natural and synthetic compounds interfering with control mechanisms.\n",
      "  JOURNAL   Biomed Res Int 2015:759348 (2015)\n",
      "            DOI:10.1155/2015/759348\n",
      "REFERENCE   PMID:11544214\n",
      "  AUTHORS   Whiteley M, Greenberg EP\n",
      "  TITLE     Promoter specificity elements in Pseudomonas aeruginosa quorum-sensing-controlled genes.\n",
      "  JOURNAL   J Bacteriol 183:5529-34 (2001)\n",
      "            DOI:10.1128/JB.183.19.5529-5534.2001\n",
      "REFERENCE   PMID:11497469\n",
      "  AUTHORS   Chin-A-Woeng TF, van den Broek D, de Voer G, van der Drift KM, Tuinman S, Thomas-Oates JE, Lugtenberg BJ, Bloemberg GV\n",
      "  TITLE     Phenazine-1-carboxamide production in the biocontrol strain Pseudomonas chlororaphis PCL1391 is regulated by multiple factors secreted into the growth medium.\n",
      "  JOURNAL   Mol Plant Microbe Interact 14:969-79 (2001)\n",
      "            DOI:10.1094/MPMI.2001.14.8.969\n",
      "REFERENCE   PMID:19696934\n",
      "  AUTHORS   Hughes DT, Clarke MB, Yamamoto K, Rasko DA, Sperandio V\n",
      "  TITLE     The QseC adrenergic signaling cascade in Enterohemorrhagic E. coli (EHEC).\n",
      "  JOURNAL   PLoS Pathog 5:e1000553 (2009)\n",
      "            DOI:10.1371/journal.ppat.1000553\n",
      "REFERENCE   PMID:22144490\n",
      "  AUTHORS   Njoroge J, Sperandio V\n",
      "  TITLE     Enterohemorrhagic Escherichia coli virulence regulation by two bacterial adrenergic kinases, QseC and QseE.\n",
      "  JOURNAL   Infect Immun 80:688-703 (2012)\n",
      "            DOI:10.1128/IAI.05921-11\n",
      "REFERENCE   PMID:23160491\n",
      "  AUTHORS   Pacheco AR, Curtis MM, Ritchie JM, Munera D, Waldor MK, Moreira CG, Sperandio V\n",
      "  TITLE     Fucose sensing regulates bacterial intestinal colonization.\n",
      "  JOURNAL   Nature 492:113-7 (2012)\n",
      "            DOI:10.1038/nature11623\n",
      "REFERENCE   PMID:22778415\n",
      "  AUTHORS   Hirakawa H, Harwood CS, Pechter KB, Schaefer AL, Greenberg EP\n",
      "  TITLE     Antisense RNA that affects Rhodopseudomonas palustris quorum-sensing signal receptor expression.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 109:12141-6 (2012)\n",
      "            DOI:10.1073/pnas.1200243109\n",
      "REFERENCE   PMID:14617175\n",
      "  AUTHORS   Danino VE, Wilkinson A, Edwards A, Downie JA\n",
      "  TITLE     Recipient-induced transfer of the symbiotic plasmid pRL1JI in Rhizobium leguminosarum bv. viciae is regulated by a quorum-sensing relay.\n",
      "  JOURNAL   Mol Microbiol 50:511-25 (2003)\n",
      "            DOI:10.1046/j.1365-2958.2003.03699.x\n",
      "REFERENCE   PMID:12448738\n",
      "  AUTHORS   Wisniewski-Dye F, Downie JA\n",
      "  TITLE     Quorum-sensing in Rhizobium.\n",
      "  JOURNAL   Antonie Van Leeuwenhoek 81:397-407 (2002)\n",
      "            DOI:10.1023/A:1020501104051\n",
      "REFERENCE   PMID:10368158\n",
      "  AUTHORS   Rodelas B, Lithgow JK, Wisniewski-Dye F, Hardman A, Wilkinson A, Economou A, Williams P, Downie JA\n",
      "  TITLE     Analysis of quorum-sensing-dependent control of rhizosphere-expressed (rhi) genes in Rhizobium leguminosarum bv. viciae.\n",
      "  JOURNAL   J Bacteriol 181:3816-23 (1999)\n",
      "            DOI:10.1128/JB.181.12.3816-3823.1999\n",
      "REFERENCE   PMID:24550924\n",
      "  AUTHORS   Lang J, Faure D\n",
      "  TITLE     Functions and regulation of quorum-sensing in Agrobacterium tumefaciens.\n",
      "  JOURNAL   Front Plant Sci 5:14 (2014)\n",
      "            DOI:10.3389/fpls.2014.00014\n",
      "REFERENCE   PMID:23710277\n",
      "  AUTHORS   Planamente S, Morera S, Faure D\n",
      "  TITLE     In planta fitness-cost of the Atu4232-regulon encoding for a selective GABA-binding sensor in Agrobacterium.\n",
      "  JOURNAL   Commun Integr Biol 6:e23692 (2013)\n",
      "            DOI:10.4161/cib.23692\n",
      "REFERENCE   PMID:16943991\n",
      "  AUTHORS   Barnard AM, Salmond GP\n",
      "  TITLE     Quorum sensing in Erwinia species.\n",
      "  JOURNAL   Anal Bioanal Chem 387:415-23 (2007)\n",
      "            DOI:10.1007/s00216-006-0701-1\n",
      "REFERENCE   PMID:17360277\n",
      "  AUTHORS   Barnard AM, Bowden SD, Burr T, Coulthurst SJ, Monson RE, Salmond GP\n",
      "  TITLE     Quorum sensing, virulence and secondary metabolite production in plant soft-rotting bacteria.\n",
      "  JOURNAL   Philos Trans R Soc Lond B Biol Sci 362:1165-83 (2007)\n",
      "            DOI:10.1098/rstb.2007.2042\n",
      "REFERENCE   PMID:19843225\n",
      "  AUTHORS   Carlier A, Burbank L, von Bodman SB\n",
      "  TITLE     Identification and characterization of three novel EsaI/EsaR quorum-sensing controlled stewartan exopolysaccharide biosynthetic genes in Pantoea stewartii ssp. stewartii.\n",
      "  JOURNAL   Mol Microbiol 74:903-13 (2009)\n",
      "            DOI:10.1111/j.1365-2958.2009.06906.x\n",
      "REFERENCE   PMID:7665477\n",
      "  AUTHORS   Beck von Bodman S, Farrand SK\n",
      "  TITLE     Capsular polysaccharide biosynthesis and pathogenicity in Erwinia stewartii require induction by an N-acylhomoserine lactone autoinducer.\n",
      "  JOURNAL   J Bacteriol 177:5000-8 (1995)\n",
      "            DOI:10.1128/JB.177.17.5000-5008.1995\n",
      "REFERENCE   PMID:19761612\n",
      "  AUTHORS   O'Grady EP, Viteri DF, Malott RJ, Sokol PA\n",
      "  TITLE     Reciprocal regulation by the CepIR and CciIR quorum sensing systems in Burkholderia cenocepacia.\n",
      "  JOURNAL   BMC Genomics 10:441 (2009)\n",
      "            DOI:10.1186/1471-2164-10-441\n",
      "REFERENCE   PMID:16041013\n",
      "  AUTHORS   Malott RJ, Baldwin A, Mahenthiralingam E, Sokol PA\n",
      "  TITLE     Characterization of the cciIR quorum-sensing system in Burkholderia cenocepacia.\n",
      "  JOURNAL   Infect Immun 73:4982-92 (2005)\n",
      "            DOI:10.1128/IAI.73.8.4982-4992.2005\n",
      "REFERENCE   PMID:22949660\n",
      "  AUTHORS   Deng Y, Schmid N, Wang C, Wang J, Pessi G, Wu D, Lee J, Aguilar C, Ahrens CH, Chang C, Song H, Eberl L, Zhang LH\n",
      "  TITLE     Cis-2-dodecenoic acid receptor RpfR links quorum-sensing signal perception with regulation of virulence through cyclic dimeric guanosine monophosphate turnover.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 109:15479-84 (2012)\n",
      "            DOI:10.1073/pnas.1205037109\n",
      "REFERENCE   PMID:23799665\n",
      "  AUTHORS   Suppiger A, Schmid N, Aguilar C, Pessi G, Eberl L\n",
      "  TITLE     Two quorum sensing systems control biofilm formation and virulence in members of the Burkholderia cepacia complex.\n",
      "  JOURNAL   Virulence 4:400-9 (2013)\n",
      "            DOI:10.4161/viru.25338\n",
      "REFERENCE   PMID:21166386\n",
      "  AUTHORS   Deng Y, Wu J, Tao F, Zhang LH\n",
      "  TITLE     Listening to a new language: DSF-based quorum sensing in Gram-negative bacteria.\n",
      "  JOURNAL   Chem Rev 111:160-73 (2011)\n",
      "            DOI:10.1021/cr100354f\n",
      "REFERENCE   PMID:21227698\n",
      "  AUTHORS   Ryan RP, Dow JM\n",
      "  TITLE     Communication with a growing family: diffusible signal factor (DSF) signaling in bacteria.\n",
      "  JOURNAL   Trends Microbiol 19:145-52 (2011)\n",
      "            DOI:10.1016/j.tim.2010.12.003\n",
      "REFERENCE   PMID:15522077\n",
      "  AUTHORS   Kim J, Kim JG, Kang Y, Jang JY, Jog GJ, Lim JY, Kim S, Suga H, Nagamatsu T, Hwang I\n",
      "  TITLE     Quorum sensing and the LysR-type transcriptional activator ToxR regulate toxoflavin biosynthesis and transport in Burkholderia glumae.\n",
      "  JOURNAL   Mol Microbiol 54:921-34 (2004)\n",
      "            DOI:10.1111/j.1365-2958.2004.04338.x\n",
      "REFERENCE   PMID:17376080\n",
      "  AUTHORS   Kim J, Kang Y, Choi O, Jeong Y, Jeong JE, Lim JY, Kim M, Moon JS, Suga H, Hwang I\n",
      "  TITLE     Regulation of polar flagellum genes is mediated by quorum sensing and FlhDC in Burkholderia glumae.\n",
      "  JOURNAL   Mol Microbiol 64:165-79 (2007)\n",
      "            DOI:10.1111/j.1365-2958.2007.05646.x\n",
      "REFERENCE   PMID:21622734\n",
      "  AUTHORS   Stauff DL, Bassler BL\n",
      "  TITLE     Quorum sensing in Chromobacterium violaceum: DNA recognition and gene regulation by the CviR receptor.\n",
      "  JOURNAL   J Bacteriol 193:3871-8 (2011)\n",
      "            DOI:10.1128/JB.05125-11\n",
      "REFERENCE   PMID:21504831\n",
      "  AUTHORS   Chen G, Swem LR, Swem DL, Stauff DL, O'Loughlin CT, Jeffrey PD, Bassler BL, Hughson FM\n",
      "  TITLE     A strategy for antagonizing quorum sensing.\n",
      "  JOURNAL   Mol Cell 42:199-209 (2011)\n",
      "            DOI:10.1016/j.molcel.2011.04.003\n",
      "REFERENCE   PMID:16513747\n",
      "  AUTHORS   Hao G, Burr TJ\n",
      "  TITLE     Regulation of long-chain N-acyl-homoserine lactones in Agrobacterium vitis.\n",
      "  JOURNAL   J Bacteriol 188:2173-83 (2006)\n",
      "            DOI:10.1128/JB.188.6.2173-2183.2006\n",
      "REFERENCE   PMID:20514483\n",
      "  AUTHORS   Savka MA, Le PT, Burr TJ\n",
      "  TITLE     LasR receptor for detection of long-chain quorum-sensing signals: identification of N-acyl-homoserine lactones encoded by the avsI locus of Agrobacterium vitis.\n",
      "  JOURNAL   Curr Microbiol 62:101-10 (2011)\n",
      "            DOI:10.1007/s00284-010-9679-1\n",
      "REFERENCE   PMID:26579084\n",
      "  AUTHORS   Le KY, Otto M\n",
      "  TITLE     Quorum-sensing regulation in staphylococci-an overview.\n",
      "  JOURNAL   Front Microbiol 6:1174 (2015)\n",
      "            DOI:10.3389/fmicb.2015.01174\n",
      "REFERENCE   PMID:18851841\n",
      "  AUTHORS   Queck SY, Jameson-Lee M, Villaruz AE, Bach TH, Khan BA, Sturdevant DE, Ricklefs SM, Li M, Otto M\n",
      "  TITLE     RNAIII-independent target gene control by the agr quorum-sensing system: insight into the evolution of virulence regulation in Staphylococcus aureus.\n",
      "  JOURNAL   Mol Cell 32:150-8 (2008)\n",
      "            DOI:10.1016/j.molcel.2008.08.005\n",
      "REFERENCE   PMID:25714717\n",
      "  AUTHORS   Darkoh C, DuPont HL, Norris SJ, Kaplan HB\n",
      "  TITLE     Toxin synthesis by Clostridium difficile is regulated through quorum signaling.\n",
      "  JOURNAL   MBio 6:e02569 (2015)\n",
      "            DOI:10.1128/mBio.02569-14\n",
      "REFERENCE   PMID:25824838\n",
      "  AUTHORS   Vidal JE, Shak JR, Canizalez-Roman A\n",
      "  TITLE     The CpAL quorum sensing system regulates production of hemolysins CPA and PFO to build Clostridium perfringens biofilms.\n",
      "  JOURNAL   Infect Immun 83:2430-42 (2015)\n",
      "            DOI:10.1128/IAI.00240-15\n",
      "REFERENCE   PMID:19363118\n",
      "  AUTHORS   Ohtani K, Yuan Y, Hassan S, Wang R, Wang Y, Shimizu T\n",
      "  TITLE     Virulence gene regulation by the agr system in Clostridium perfringens.\n",
      "  JOURNAL   J Bacteriol 191:3919-27 (2009)\n",
      "            DOI:10.1128/JB.01455-08\n",
      "REFERENCE   PMID:22736963\n",
      "  AUTHORS   Li YH, Tian X\n",
      "  TITLE     Quorum sensing and bacterial social interactions in biofilms.\n",
      "  JOURNAL   Sensors (Basel) 12:2519-38 (2012)\n",
      "            DOI:10.3390/s120302519\n",
      "REFERENCE   PMID:22407687\n",
      "  AUTHORS   Berg KH, Biornstad TJ, Johnsborg O, Havarstein LS\n",
      "  TITLE     Properties and biological role of streptococcal fratricins.\n",
      "  JOURNAL   Appl Environ Microbiol 78:3515-22 (2012)\n",
      "            DOI:10.1128/AEM.00098-12\n",
      "REFERENCE   PMID:25475081\n",
      "  AUTHORS   Conrads G, de Soet JJ, Song L, Henne K, Sztajer H, Wagner-Dobler I, Zeng AP\n",
      "  TITLE     Comparing the cariogenic species Streptococcus sobrinus and S. mutans on whole genome level.\n",
      "  JOURNAL   J Oral Microbiol 6:26189 (2014)\n",
      "            DOI:10.3402/jom.v6.26189\n",
      "REFERENCE   PMID:25309879\n",
      "  AUTHORS   Jimenez JC, Federle MJ\n",
      "  TITLE     Quorum sensing in group A Streptococcus.\n",
      "  JOURNAL   Front Cell Infect Microbiol 4:127 (2014)\n",
      "            DOI:10.3389/fcimb.2014.00127\n",
      "REFERENCE   PMID:24118108\n",
      "  AUTHORS   Cook LC, Federle MJ\n",
      "  TITLE     Peptide pheromone signaling in Streptococcus and Enterococcus.\n",
      "  JOURNAL   FEMS Microbiol Rev 38:473-92 (2014)\n",
      "            DOI:10.1111/1574-6976.12046\n",
      "REFERENCE   PMID:9680220\n",
      "  AUTHORS   Podbielski A, Leonard BA\n",
      "  TITLE     The group A streptococcal dipeptide permease (Dpp) is involved in the uptake of essential amino acids and affects the expression of cysteine protease.\n",
      "  JOURNAL   Mol Microbiol 28:1323-34 (1998)\n",
      "            DOI:10.1046/j.1365-2958.1998.00898.x\n",
      "REFERENCE   PMID:8885277\n",
      "  AUTHORS   Podbielski A, Pohl B, Woischnik M, Korner C, Schmidt KH, Rozdzinski E, Leonard BA\n",
      "  TITLE     Molecular characterization of group A streptococcal (GAS) oligopeptide permease (opp) and its effect on cysteine protease production.\n",
      "  JOURNAL   Mol Microbiol 21:1087-99 (1996)\n",
      "            DOI:10.1046/j.1365-2958.1996.661421.x\n",
      "REFERENCE   PMID:22636768\n",
      "  AUTHORS   Anbalagan S, Dmitriev A, McShan WM, Dunman PM, Chaussee MS\n",
      "  TITLE     Growth phase-dependent modulation of Rgg binding specificity in Streptococcus pyogenes.\n",
      "  JOURNAL   J Bacteriol 194:3961-71 (2012)\n",
      "            DOI:10.1128/JB.06709-11\n",
      "REFERENCE   PMID:22016844\n",
      "  AUTHORS   Clewell DB\n",
      "  TITLE     Tales of conjugation and sex pheromones: A plasmid and enterococcal odyssey.\n",
      "  JOURNAL   Mob Genet Elements 1:38-54 (2011)\n",
      "            DOI:10.4161/mge.1.1.15409\n",
      "REFERENCE   PMID:20734751\n",
      "  AUTHORS   Wardal E, Sadowy E, Hryniewicz W\n",
      "  TITLE     Complex nature of enterococcal pheromone-responsive plasmids.\n",
      "  JOURNAL   Pol J Microbiol 59:79-87 (2010)\n",
      "REFERENCE   PMID:19683678\n",
      "  AUTHORS   Roux A, Payne SM, Gilmore MS\n",
      "  TITLE     Microbial telesensing: probing the environment for friends, foes, and food.\n",
      "  JOURNAL   Cell Host Microbe 6:115-24 (2009)\n",
      "            DOI:10.1016/j.chom.2009.07.004\n",
      "REFERENCE   PMID:25089349\n",
      "  AUTHORS   Slamti L, Perchat S, Huillet E, Lereclus D\n",
      "  TITLE     Quorum sensing in Bacillus thuringiensis is required for completion of a full infectious cycle in the insect.\n",
      "  JOURNAL   Toxins (Basel) 6:2239-55 (2014)\n",
      "            DOI:10.3390/toxins6082239\n",
      "REFERENCE   PMID:20030732\n",
      "  AUTHORS   Lopez D, Kolter R\n",
      "  TITLE     Extracellular signals that define distinct and coexisting cell fates in Bacillus subtilis.\n",
      "  JOURNAL   FEMS Microbiol Rev 34:134-49 (2010)\n",
      "            DOI:10.1111/j.1574-6976.2009.00199.x\n",
      "REFERENCE   PMID:12897006\n",
      "  AUTHORS   Stephenson S, Mueller C, Jiang M, Perego M\n",
      "  TITLE     Molecular analysis of Phr peptide processing in Bacillus subtilis.\n",
      "  JOURNAL   J Bacteriol 185:4861-71 (2003)\n",
      "            DOI:10.1128/JB.185.16.4861-4871.2003\n",
      "REFERENCE   PMID:17666034\n",
      "  AUTHORS   Lanigan-Gerdes S, Dooley AN, Faull KF, Lazazzera BA\n",
      "  TITLE     Identification of subtilisin, Epr and Vpr as enzymes that produce CSF, an extracellular signalling peptide of Bacillus subtilis.\n",
      "  JOURNAL   Mol Microbiol 65:1321-33 (2007)\n",
      "            DOI:10.1111/j.1365-2958.2007.05869.x\n",
      "REFERENCE   PMID:15374644\n",
      "  AUTHORS   Kleerebezem M\n",
      "  TITLE     Quorum sensing control of lantibiotic production; nisin and subtilin autoregulate their own biosynthesis.\n",
      "  JOURNAL   Peptides 25:1405-14 (2004)\n",
      "            DOI:10.1016/j.peptides.2003.10.021\n",
      "REFERENCE   PMID:23125205\n",
      "  AUTHORS   Rutherford ST, Bassler BL\n",
      "  TITLE     Bacterial quorum sensing: its role in virulence and possibilities for its control.\n",
      "  JOURNAL   Cold Spring Harb Perspect Med 2:a012427 (2012)\n",
      "            DOI:10.1101/cshperspect.a012427\n",
      "REFERENCE   PMID:23471618\n",
      "  AUTHORS   LaSarre B, Federle MJ\n",
      "  TITLE     Exploiting quorum sensing to confuse bacterial pathogens.\n",
      "  JOURNAL   Microbiol Mol Biol Rev 77:73-111 (2013)\n",
      "            DOI:10.1128/MMBR.00046-12\n",
      "REFERENCE   PMID:18781869\n",
      "  AUTHORS   Sifri CD\n",
      "  TITLE     Healthcare epidemiology: quorum sensing: bacteria talk sense.\n",
      "  JOURNAL   Clin Infect Dis 47:1070-6 (2008)\n",
      "            DOI:10.1086/592072\n",
      "REL_PATHWAY pae00405  Phenazine biosynthesis\n",
      "            pae02040  Flagellar assembly\n",
      "KO_PATHWAY  ko02024\n",
      "///\n",
      "path:pae02025\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae02025                    Pathway\n",
      "NAME        Biofilm formation - Pseudomonas aeruginosa - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Surface colonization and subsequent biofilm formation and development provide numerous advantages to microorganisms. The main key actors relevant in the regulation of biofilm formation by Pseudomonas aeruginosa include cAMP/Vfr signaling, quorum sensing (QS) systems, Gac/Rsm pathway, and c-di-GMP signaling. cAMP/Vfr signaling regulates the transcription of hundreds of genes encoding diverse virulence factors, including the type 2 secretion system and type 3 secretion system and their associated toxins, type IV pili, and flagella. It is demonstrated that the accumulation of cAMP inhibits the attachment phase of biofilm formation. The QS systems are arranged hierarchically with the las system positively regulating both the rhl and PQS systems. These three QS systems are involved in the regulation of virulence factor production, biofilm maturation, and motility phenotypes. GacS/GacA two-component system is promoting the expression of two small regulatory RNAs, RsmY and RsmZ, which sequester the translational repressor RsmA. Titration of RsmA induces the production of sessile and biofilm determinants, whereas free RsmA leads to a planktonic and more virulent lifestyle. c-di-GMP is a ubiquitous bacterial second messenger. Synthesis and degradation of c-di-GMP is facilitated by diguanylate cyclases (DGC) and phosphodiesterases (PDE). High intracellular levels of c-di-GMP evoke the processes leading to the production of biofilm matrix, while a decrease in its level causes an increase in cell motility and transition into the planktonic forms of growth.\n",
      "CLASS       Cellular Processes; Cellular community - prokaryotes\n",
      "PATHWAY_MAP pae02025  Biofilm formation - Pseudomonas aeruginosa\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0411  pilJ; twitching motility protein PilJ [KO:K02660]\n",
      "line in gene PA0411  pilJ; twitching motility protein PilJ [KO:K02660]\n",
      "            PA0410  pilI; twitching motility protein PilI [KO:K02659]\n",
      "line in gene PA0410  pilI; twitching motility protein PilI [KO:K02659]\n",
      "            PA0413  chpA; chemotactic signal transduction system protein [KO:K06596]\n",
      "line in gene PA0413  chpA; chemotactic signal transduction system protein [KO:K06596]\n",
      "            PA0415  chpC; chemotaxis protein [KO:K06598]\n",
      "line in gene PA0415  chpC; chemotaxis protein [KO:K06598]\n",
      "            PA0409  pilH; twitching motility protein PilH [KO:K02658]\n",
      "line in gene PA0409  pilH; twitching motility protein PilH [KO:K02658]\n",
      "            PA0408  pilG; pilus biosynthesis/twitching motility protein PilG [KO:K02657]\n",
      "line in gene PA0408  pilG; pilus biosynthesis/twitching motility protein PilG [KO:K02657]\n",
      "            PA3217  cyaB; protein CyaB [KO:K01768] [EC:4.6.1.1]\n",
      "line in gene PA3217  cyaB; protein CyaB [KO:K01768] [EC:4.6.1.1]\n",
      "            PA4969  cpdA; cAMP phosphodiesterase [KO:K03651] [EC:3.1.4.53]\n",
      "line in gene PA4969  cpdA; cAMP phosphodiesterase [KO:K03651] [EC:3.1.4.53]\n",
      "            PA0652  vfr; cAMP-regulatory protein [KO:K10914]\n",
      "line in gene PA0652  vfr; cAMP-regulatory protein [KO:K10914]\n",
      "            PA0275  transcriptional regulator [KO:K10914]\n",
      "line in gene PA0275  transcriptional regulator [KO:K10914]\n",
      "            PA1097  fleQ; transcriptional regulator FleQ [KO:K10941]\n",
      "line in gene PA1097  fleQ; transcriptional regulator FleQ [KO:K10941]\n",
      "            PA1713  exsA; exoenzyme S transcriptional regulator ExsA [KO:K20968]\n",
      "line in gene PA1713  exsA; exoenzyme S transcriptional regulator ExsA [KO:K20968]\n",
      "            PA1432  lasI; acyl-homoserine-lactone synthase [KO:K13060] [EC:2.3.1.184]\n",
      "line in gene PA1432  lasI; acyl-homoserine-lactone synthase [KO:K13060] [EC:2.3.1.184]\n",
      "            PA1430  lasR; transcriptional regulator LasR [KO:K18304]\n",
      "line in gene PA1430  lasR; transcriptional regulator LasR [KO:K18304]\n",
      "            PA3476  rhlI; acyl-homoserine-lactone synthase [KO:K13061] [EC:2.3.1.184]\n",
      "line in gene PA3476  rhlI; acyl-homoserine-lactone synthase [KO:K13061] [EC:2.3.1.184]\n",
      "            PA3477  rhlR; transcriptional regulator RhlR [KO:K18099]\n",
      "line in gene PA3477  rhlR; transcriptional regulator RhlR [KO:K18099]\n",
      "            PA3479  rhlA; rhamnosyltransferase subunit A [KO:K18100] [EC:2.4.1.-]\n",
      "line in gene PA3479  rhlA; rhamnosyltransferase subunit A [KO:K18100] [EC:2.4.1.-]\n",
      "            PA3478  rhlB; rhamnosyltransferase subunit B [KO:K18101] [EC:2.4.1.-]\n",
      "line in gene PA3478  rhlB; rhamnosyltransferase subunit B [KO:K18101] [EC:2.4.1.-]\n",
      "            PA1130  rhlC; rhamnosyltransferase [KO:K12990] [EC:2.4.1.-]\n",
      "line in gene PA1130  rhlC; rhamnosyltransferase [KO:K12990] [EC:2.4.1.-]\n",
      "            PA2570  lecA; PA-I galactophilic lectin [KO:K20258]\n",
      "line in gene PA2570  lecA; PA-I galactophilic lectin [KO:K20258]\n",
      "            PA2569  hypothetical protein [KO:K20259]\n",
      "line in gene PA2569  hypothetical protein [KO:K20259]\n",
      "            PA0609  trpE; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "line in gene PA0609  trpE; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "            PA1001  phnA; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "line in gene PA1001  phnA; anthranilate synthase component I [KO:K01657] [EC:4.1.3.27]\n",
      "            PA0649  trpG; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "line in gene PA0649  trpG; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "            PA1002  phnB; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "line in gene PA1002  phnB; anthranilate synthase component II [KO:K01658] [EC:4.1.3.27]\n",
      "            PA0996  pqsA; anthranilate--CoA ligase [KO:K18000] [EC:6.2.1.32]\n",
      "line in gene PA0996  pqsA; anthranilate--CoA ligase [KO:K18000] [EC:6.2.1.32]\n",
      "            PA0997  pqsB; hypothetical protein [KO:K18001] [EC:2.3.1.230]\n",
      "line in gene PA0997  pqsB; hypothetical protein [KO:K18001] [EC:2.3.1.230]\n",
      "            PA0998  pqsC; hypothetical protein [KO:K18002] [EC:2.3.1.230]\n",
      "line in gene PA0998  pqsC; hypothetical protein [KO:K18002] [EC:2.3.1.230]\n",
      "            PA0999  pqsD; 3-oxoacyl-ACP synthase [KO:K18003] [EC:2.3.1.262]\n",
      "line in gene PA0999  pqsD; 3-oxoacyl-ACP synthase [KO:K18003] [EC:2.3.1.262]\n",
      "            PA1000  pqsE; thioesterase PqsE [KO:K20257] [EC:3.1.2.32]\n",
      "line in gene PA1000  pqsE; thioesterase PqsE [KO:K20257] [EC:3.1.2.32]\n",
      "            PA2587  pqsH; 2-heptyl-3-hydroxy-4(1H)-quinolone synthase [KO:K17940] [EC:1.14.13.182]\n",
      "line in gene PA2587  pqsH; 2-heptyl-3-hydroxy-4(1H)-quinolone synthase [KO:K17940] [EC:1.14.13.182]\n",
      "            PA1003  mvfR; transcriptional regulator MvfR [KO:K19735]\n",
      "line in gene PA1003  mvfR; transcriptional regulator MvfR [KO:K19735]\n",
      "            PA0928  gacS; sensor/response regulator hybrid protein [KO:K07678] [EC:2.7.13.3]\n",
      "line in gene PA0928  gacS; sensor/response regulator hybrid protein [KO:K07678] [EC:2.7.13.3]\n",
      "            PA3974  ladS; lost adherence sensor LadS [KO:K20971]\n",
      "line in gene PA3974  ladS; lost adherence sensor LadS [KO:K20971]\n",
      "            PA4856  retS; sensor histidine kinase MifS [KO:K20972]\n",
      "line in gene PA4856  retS; sensor histidine kinase MifS [KO:K20972]\n",
      "            PA2586  gacA; response regulator GacA [KO:K07689]\n",
      "line in gene PA2586  gacA; response regulator GacA [KO:K07689]\n",
      "            PA0527.1  rsmY; regulatory RNA RsmY [KO:K20969]\n",
      "line in gene PA0527.1  rsmY; regulatory RNA RsmY [KO:K20969]\n",
      "            PA3621.1  rsmZ; regulatory RNA RsmZ [KO:K20970]\n",
      "line in gene PA3621.1  rsmZ; regulatory RNA RsmZ [KO:K20970]\n",
      "            PA0905  rsmA; carbon storage regulator [KO:K03563]\n",
      "line in gene PA0905  rsmA; carbon storage regulator [KO:K03563]\n",
      "            PA2824  sensor/response regulator hybrid protein [KO:K20973] [EC:2.7.13.3]\n",
      "line in gene PA2824  sensor/response regulator hybrid protein [KO:K20973] [EC:2.7.13.3]\n",
      "            PA1611  sensor/response regulator hybrid protein [KO:K20974] [EC:2.7.13.3]\n",
      "line in gene PA1611  sensor/response regulator hybrid protein [KO:K20974] [EC:2.7.13.3]\n",
      "            PA1976  ercS'; sensor histidine kinase [KO:K20975] [EC:2.7.13.3]\n",
      "line in gene PA1976  ercS'; sensor histidine kinase [KO:K20975] [EC:2.7.13.3]\n",
      "            PA3345  hypothetical protein [KO:K20976]\n",
      "line in gene PA3345  hypothetical protein [KO:K20976]\n",
      "            PA3346  two-component response regulator [KO:K20977]\n",
      "line in gene PA3346  two-component response regulator [KO:K20977]\n",
      "            PA3347  hypothetical protein [KO:K20978]\n",
      "line in gene PA3347  hypothetical protein [KO:K20978]\n",
      "            PA3351  flgM; protein FlgM [KO:K02398]\n",
      "line in gene PA3351  flgM; protein FlgM [KO:K02398]\n",
      "            PA1455  fliA; flagellar biosynthesis sigma factor FliA [KO:K02405]\n",
      "line in gene PA1455  fliA; flagellar biosynthesis sigma factor FliA [KO:K02405]\n",
      "            PA0074  ppkA; serine/threonine protein kinase PpkA [KO:K11912] [EC:2.7.11.1]\n",
      "line in gene PA0074  ppkA; serine/threonine protein kinase PpkA [KO:K11912] [EC:2.7.11.1]\n",
      "            PA1670  stp1; serine/threonine phosphoprotein phosphatase Stp1 [KO:K11915] [EC:3.1.3.16]\n",
      "line in gene PA1670  stp1; serine/threonine phosphoprotein phosphatase Stp1 [KO:K11915] [EC:3.1.3.16]\n",
      "            PA0075  pppA; serine/threonine phosphatase [KO:K11915] [EC:3.1.3.16]\n",
      "line in gene PA0075  pppA; serine/threonine phosphatase [KO:K11915] [EC:3.1.3.16]\n",
      "            PA0076  hypothetical protein [KO:K11890]\n",
      "line in gene PA0076  hypothetical protein [KO:K11890]\n",
      "            PA1669  hypothetical protein [KO:K11891]\n",
      "line in gene PA1669  hypothetical protein [KO:K11891]\n",
      "            PA2361  hypothetical protein [KO:K11891]\n",
      "line in gene PA2361  hypothetical protein [KO:K11891]\n",
      "            PA0077  icmF1; type VI secretion protein IcmF [KO:K11891]\n",
      "line in gene PA0077  icmF1; type VI secretion protein IcmF [KO:K11891]\n",
      "            PA1667  hypothetical protein [KO:K11893]\n",
      "line in gene PA1667  hypothetical protein [KO:K11893]\n",
      "            PA2363  hypothetical protein [KO:K11893]\n",
      "line in gene PA2363  hypothetical protein [KO:K11893]\n",
      "            PA0079  hypothetical protein [KO:K11893]\n",
      "line in gene PA0079  hypothetical protein [KO:K11893]\n",
      "            PA0081  fha1; Fha domain-containing protein [KO:K11913]\n",
      "line in gene PA0081  fha1; Fha domain-containing protein [KO:K11913]\n",
      "            PA0082  hypothetical protein [KO:K11902]\n",
      "line in gene PA0082  hypothetical protein [KO:K11902]\n",
      "            PA2360  hypothetical protein [KO:K11902]\n",
      "line in gene PA2360  hypothetical protein [KO:K11902]\n",
      "            PA1657  hypothetical protein [KO:K11901]\n",
      "line in gene PA1657  hypothetical protein [KO:K11901]\n",
      "            PA0083  hypothetical protein [KO:K11901]\n",
      "line in gene PA0083  hypothetical protein [KO:K11901]\n",
      "            PA2365  hypothetical protein [KO:K11901]\n",
      "line in gene PA2365  hypothetical protein [KO:K11901]\n",
      "            PA0084  hypothetical protein [KO:K11900]\n",
      "line in gene PA0084  hypothetical protein [KO:K11900]\n",
      "            PA2366  uricase [KO:K11900]\n",
      "line in gene PA2366  uricase [KO:K11900]\n",
      "            PA1658  hypothetical protein [KO:K11900]\n",
      "line in gene PA1658  hypothetical protein [KO:K11900]\n",
      "            PA1512  hcpA; secreted protein Hcp [KO:K11903]\n",
      "line in gene PA1512  hcpA; secreted protein Hcp [KO:K11903]\n",
      "            PA2367  hypothetical protein [KO:K11903]\n",
      "line in gene PA2367  hypothetical protein [KO:K11903]\n",
      "            PA0263  hcpC; secreted protein Hcp [KO:K11903]\n",
      "line in gene PA0263  hcpC; secreted protein Hcp [KO:K11903]\n",
      "            PA5267  hcpB; secreted protein Hcp [KO:K11903]\n",
      "line in gene PA5267  hcpB; secreted protein Hcp [KO:K11903]\n",
      "            PA0085  hcp1; protein secretion apparatus assembly protein [KO:K11903]\n",
      "line in gene PA0085  hcp1; protein secretion apparatus assembly protein [KO:K11903]\n",
      "            PA2370  hypothetical protein [KO:K11895]\n",
      "line in gene PA2370  hypothetical protein [KO:K11895]\n",
      "            PA1661  hypothetical protein [KO:K11895]\n",
      "line in gene PA1661  hypothetical protein [KO:K11895]\n",
      "            PA0089  hypothetical protein [KO:K11895]\n",
      "line in gene PA0089  hypothetical protein [KO:K11895]\n",
      "            PA1662  ClpA/B-type protease [KO:K11907]\n",
      "line in gene PA1662  ClpA/B-type protease [KO:K11907]\n",
      "            PA0090  clpV1; secretion protein ClpV1 [KO:K11907]\n",
      "line in gene PA0090  clpV1; secretion protein ClpV1 [KO:K11907]\n",
      "            PA2371  ClpA/B-type protease [KO:K11907]\n",
      "line in gene PA2371  ClpA/B-type protease [KO:K11907]\n",
      "            PA3708  wspA; chemotaxis transducer [KO:K13487]\n",
      "line in gene PA3708  wspA; chemotaxis transducer [KO:K13487]\n",
      "            PA3704  wspE; chemotaxis sensor/effector fusion protein [KO:K13490]\n",
      "line in gene PA3704  wspE; chemotaxis sensor/effector fusion protein [KO:K13490]\n",
      "            PA3707  hypothetical protein [KO:K13488]\n",
      "line in gene PA3707  hypothetical protein [KO:K13488]\n",
      "            PA3705  wspD; hypothetical protein [KO:K13489]\n",
      "line in gene PA3705  wspD; hypothetical protein [KO:K13489]\n",
      "            PA3703  wspF; chemotaxis-specific methylesterase [KO:K13491] [EC:3.1.1.61]\n",
      "line in gene PA3703  wspF; chemotaxis-specific methylesterase [KO:K13491] [EC:3.1.1.61]\n",
      "            PA3702  wspR; two-component response regulator [KO:K11444] [EC:2.7.7.65]\n",
      "line in gene PA3702  wspR; two-component response regulator [KO:K11444] [EC:2.7.7.65]\n",
      "            PA4332  hypothetical protein [KO:K21019] [EC:2.7.7.65]\n",
      "line in gene PA4332  hypothetical protein [KO:K21019] [EC:2.7.7.65]\n",
      "            PA0169  hypothetical protein [KO:K21020] [EC:2.7.7.65]\n",
      "line in gene PA0169  hypothetical protein [KO:K21020] [EC:2.7.7.65]\n",
      "            PA1120  tpbB; diguanylate cyclase [KO:K21021] [EC:2.7.7.65]\n",
      "line in gene PA1120  tpbB; diguanylate cyclase [KO:K21021] [EC:2.7.7.65]\n",
      "            PA1107  hypothetical protein [KO:K21022] [EC:2.7.7.65]\n",
      "line in gene PA1107  hypothetical protein [KO:K21022] [EC:2.7.7.65]\n",
      "            PA1727  mucR; signaling protein [KO:K21023] [EC:2.7.7.65]\n",
      "line in gene PA1727  mucR; signaling protein [KO:K21023] [EC:2.7.7.65]\n",
      "            PA4367  bifA; protein BifA [KO:K21024] [EC:3.1.4.52]\n",
      "line in gene PA4367  bifA; protein BifA [KO:K21024] [EC:3.1.4.52]\n",
      "            PA2231  pslA; biofilm formation protein PslA [KO:K20997]\n",
      "line in gene PA2231  pslA; biofilm formation protein PslA [KO:K20997]\n",
      "            PA3551  algA; bifunctional mannose-1-phosphate guanylyltransferase/mannose-6-phosphate isomerase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "line in gene PA3551  algA; bifunctional mannose-1-phosphate guanylyltransferase/mannose-6-phosphate isomerase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "            PA2232  pslB; biofilm formation protein PslB [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "line in gene PA2232  pslB; biofilm formation protein PslB [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "            PA5452  wbpW; phosphomannose isomerase/mannose-1-phosphate guanylyl transferase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "line in gene PA5452  wbpW; phosphomannose isomerase/mannose-1-phosphate guanylyl transferase [KO:K16011] [EC:2.7.7.13 5.3.1.8]\n",
      "            PA2233  pslC; biofilm formation protein PslC [KO:K25205]\n",
      "line in gene PA2233  pslC; biofilm formation protein PslC [KO:K25205]\n",
      "            PA2234  pslD; biofilm formation protein PslD [KO:K20987]\n",
      "line in gene PA2234  pslD; biofilm formation protein PslD [KO:K20987]\n",
      "            PA2235  pslE; biofilm formation protein PslE [KO:K20998]\n",
      "line in gene PA2235  pslE; biofilm formation protein PslE [KO:K20998]\n",
      "            PA2236  pslF; biofilm formation protein PslF [KO:K20999]\n",
      "line in gene PA2236  pslF; biofilm formation protein PslF [KO:K20999]\n",
      "            PA2237  pslG; biofilm formation protein PslG [KO:K21000]\n",
      "line in gene PA2237  pslG; biofilm formation protein PslG [KO:K21000]\n",
      "            PA2238  pslH; biofilm formation protein PslH [KO:K21001]\n",
      "line in gene PA2238  pslH; biofilm formation protein PslH [KO:K21001]\n",
      "            PA2239  pslI; biofilm formation protein PslI [KO:K21002]\n",
      "line in gene PA2239  pslI; biofilm formation protein PslI [KO:K21002]\n",
      "            PA2240  pslJ; biofilm formation protein PslJ [KO:K21003]\n",
      "line in gene PA2240  pslJ; biofilm formation protein PslJ [KO:K21003]\n",
      "            PA2241  pslK; biofilm formation protein PslL [KO:K21004]\n",
      "line in gene PA2241  pslK; biofilm formation protein PslL [KO:K21004]\n",
      "            PA2242  pslL; hypothetical protein [KO:K21005]\n",
      "line in gene PA2242  pslL; hypothetical protein [KO:K21005]\n",
      "            PA3064  pelA; hypothetical protein [KO:K21006]\n",
      "line in gene PA3064  pelA; hypothetical protein [KO:K21006]\n",
      "            PA3063  pelB; pellicle/biofilm biosynthesis protein PelB [KO:K21007]\n",
      "line in gene PA3063  pelB; pellicle/biofilm biosynthesis protein PelB [KO:K21007]\n",
      "            PA3062  pelC; pellicle/biofilm biosynthesis outer membrane protein PelC [KO:K21008]\n",
      "line in gene PA3062  pelC; pellicle/biofilm biosynthesis outer membrane protein PelC [KO:K21008]\n",
      "            PA3061  pelD; pellicle/biofilm biosynthesis protein PelD [KO:K21009]\n",
      "line in gene PA3061  pelD; pellicle/biofilm biosynthesis protein PelD [KO:K21009]\n",
      "            PA3060  pelE; pellicle/biofilm biosynthesis protein PelE [KO:K21010]\n",
      "line in gene PA3060  pelE; pellicle/biofilm biosynthesis protein PelE [KO:K21010]\n",
      "            PA3059  pelF; pellicle/biofilm biosynthesis glycosyltransferase PelF [KO:K21011]\n",
      "line in gene PA3059  pelF; pellicle/biofilm biosynthesis glycosyltransferase PelF [KO:K21011]\n",
      "            PA3058  pelG; pellicle/biofilm biosynthesis Wzx-like polysaccharide transporter PelG [KO:K21012]\n",
      "line in gene PA3058  pelG; pellicle/biofilm biosynthesis Wzx-like polysaccharide transporter PelG [KO:K21012]\n",
      "            PA3542  alg44; alginate biosynthesis protein Alg44 [KO:K19291] [EC:2.4.1.33]\n",
      "line in gene PA3542  alg44; alginate biosynthesis protein Alg44 [KO:K19291] [EC:2.4.1.33]\n",
      "            PA4959  fimX; protein FimX [KO:K21025]\n",
      "line in gene PA4959  fimX; protein FimX [KO:K21025]\n",
      "            PA4958  fimW; cyclic-di-GMP receptor FimW [KO:K23127]\n",
      "line in gene PA4958  fimW; cyclic-di-GMP receptor FimW [KO:K23127]\n",
      "COMPOUND    C00044  GTP\n",
      "            C00575  3',5'-Cyclic AMP\n",
      "            C11837  N-Butyryl-L-homoserine lactone\n",
      "            C11848  2-Heptyl-3-hydroxy-quinolone\n",
      "            C16463  3',5'-Cyclic diGMP\n",
      "            C18076  5'-Phosphoguanylyl(3'->5')guanosine\n",
      "            C20643  2-Heptyl-4(1H)-quinolone\n",
      "            C21201  N-3-Oxo-dodecanoyl-L-homoserine lactone\n",
      "REFERENCE   PMID:22354680\n",
      "  AUTHORS   Coggan KA, Wolfgang MC\n",
      "  TITLE     Global regulatory pathways and cross-talk control pseudomonas aeruginosa environmental lifestyle and virulence phenotype.\n",
      "  JOURNAL   Curr Issues Mol Biol 14:47-70 (2012)\n",
      "REFERENCE   PMID:25866808\n",
      "  AUTHORS   Rasamiravaka T, Labtani Q, Duez P, El Jaziri M\n",
      "  TITLE     The formation of biofilms by Pseudomonas aeruginosa: a review of the natural and synthetic compounds interfering with control mechanisms.\n",
      "  JOURNAL   Biomed Res Int 2015:759348 (2015)\n",
      "            DOI:10.1155/2015/759348\n",
      "REFERENCE   PMID:27129226\n",
      "  AUTHORS   Valentini M, Filloux A\n",
      "  TITLE     Biofilms and Cyclic di-GMP (c-di-GMP) Signaling: Lessons from Pseudomonas aeruginosa and Other Bacteria.\n",
      "  JOURNAL   J Biol Chem 291:12547-55 (2016)\n",
      "            DOI:10.1074/jbc.R115.711507\n",
      "REFERENCE   PMID:19287449\n",
      "  AUTHORS   Hengge R\n",
      "  TITLE     Principles of c-di-GMP signalling in bacteria.\n",
      "  JOURNAL   Nat Rev Microbiol 7:263-73 (2009)\n",
      "            DOI:10.1038/nrmicro2109\n",
      "REFERENCE   PMID:25666534\n",
      "  AUTHORS   Liang ZX\n",
      "  TITLE     The expanding roles of c-di-GMP in the biosynthesis of exopolysaccharides and secondary metabolites.\n",
      "  JOURNAL   Nat Prod Rep 32:663-83 (2015)\n",
      "            DOI:10.1039/c4np00086b\n",
      "REFERENCE   PMID:20345659\n",
      "  AUTHORS   Fulcher NB, Holliday PM, Klem E, Cann MJ, Wolfgang MC\n",
      "  TITLE     The Pseudomonas aeruginosa Chp chemosensory system regulates intracellular cAMP levels by modulating adenylate cyclase activity.\n",
      "  JOURNAL   Mol Microbiol 76:889-904 (2010)\n",
      "            DOI:10.1111/j.1365-2958.2010.07135.x\n",
      "REFERENCE   PMID:26929300\n",
      "  AUTHORS   Marsden AE, Intile PJ, Schulmeyer KH, Simmons-Patterson ER, Urbanowski ML, Wolfgang MC, Yahr TL\n",
      "  TITLE     Vfr Directly Activates exsA Transcription To Regulate Expression of the Pseudomonas aeruginosa Type III Secretion System.\n",
      "  JOURNAL   J Bacteriol 198:1442-50 (2016)\n",
      "            DOI:10.1128/JB.00049-16\n",
      "REFERENCE   PMID:19426209\n",
      "  AUTHORS   Brencic A, Lory S\n",
      "  TITLE     Determination of the regulon and identification of novel mRNA targets of Pseudomonas aeruginosa RsmA.\n",
      "  JOURNAL   Mol Microbiol 72:612-32 (2009)\n",
      "            DOI:10.1111/j.1365-2958.2009.06670.x\n",
      "REFERENCE   PMID:18256026\n",
      "  AUTHORS   Hsu JL, Chen HC, Peng HL, Chang HY\n",
      "  TITLE     Characterization of the histidine-containing phosphotransfer protein B-mediated multistep phosphorelay system in Pseudomonas aeruginosa PAO1.\n",
      "  JOURNAL   J Biol Chem 283:9933-44 (2008)\n",
      "            DOI:10.1074/jbc.M708836200\n",
      "REFERENCE   PMID:18028314\n",
      "  AUTHORS   Guvener ZT, Harwood CS\n",
      "  TITLE     Subcellular location characteristics of the Pseudomonas aeruginosa GGDEF protein, WspR, indicate that it produces cyclic-di-GMP in response to growth on surfaces.\n",
      "  JOURNAL   Mol Microbiol 66:1459-73 (2007)\n",
      "            DOI:10.1111/j.1365-2958.2007.06008.x\n",
      "REFERENCE   PMID:20735777\n",
      "  AUTHORS   Irie Y, Starkey M, Edwards AN, Wozniak DJ, Romeo T, Parsek MR\n",
      "  TITLE     Pseudomonas aeruginosa biofilm matrix polysaccharide Psl is regulated transcriptionally by RpoS and post-transcriptionally by RsmA.\n",
      "  JOURNAL   Mol Microbiol 78:158-72 (2010)\n",
      "            DOI:10.1111/j.1365-2958.2010.07320.x\n",
      "REL_PATHWAY pae02024  Quorum sensing\n",
      "            pae02040  Flagellar assembly\n",
      "            pae03070  Bacterial secretion system\n",
      "KO_PATHWAY  ko02025\n",
      "///\n",
      "path:pae02030\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae02030                    Pathway\n",
      "NAME        Bacterial chemotaxis - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Chemotaxis is the process by which cells sense chemical gradients in their environment and then move towards more favorable conditions. In chemotaxis, events at the receptors control autophosphorylation of the CheA histidine kinase, and the phosphohistidine is the substrate for the response regulator CheY, which catalyzes the transfer of the phosphoryl group to a conserved aspartate. The resulting CheY-P can interact with the switch mechanism in the motor. This interaction causes a change in behavior, such as in direction or speed of rotation of flagella.\n",
      "CLASS       Cellular Processes; Cell motility\n",
      "PATHWAY_MAP pae02030  Bacterial chemotaxis\n",
      "DBLINKS     GO: 0006935\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0180  cttP; trichloroethylene chemotactic transducer CttP [KO:K03406]\n",
      "line in gene PA0180  cttP; trichloroethylene chemotactic transducer CttP [KO:K03406]\n",
      "            PA0176  aer2; aerotaxis transducer Aer2 [KO:K03406]\n",
      "line in gene PA0176  aer2; aerotaxis transducer Aer2 [KO:K03406]\n",
      "            PA4310  pctB; chemotactic transducer PctB [KO:K03406]\n",
      "line in gene PA4310  pctB; chemotactic transducer PctB [KO:K03406]\n",
      "            PA4633  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA4633  chemotaxis transducer [KO:K03406]\n",
      "            PA2573  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA2573  chemotaxis transducer [KO:K03406]\n",
      "            PA1930  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA1930  chemotaxis transducer [KO:K03406]\n",
      "            PA1608  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA1608  chemotaxis transducer [KO:K03406]\n",
      "            PA2652  methyl-accepting chemotaxis protein [KO:K03406]\n",
      "line in gene PA2652  methyl-accepting chemotaxis protein [KO:K03406]\n",
      "            PA5072  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA5072  chemotaxis transducer [KO:K03406]\n",
      "            PA1646  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA1646  chemotaxis transducer [KO:K03406]\n",
      "            PA4290  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA4290  chemotaxis transducer [KO:K03406]\n",
      "            PA4915  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA4915  chemotaxis transducer [KO:K03406]\n",
      "            PA1423  bdlA; biofilm dispersion protein [KO:K03406]\n",
      "line in gene PA1423  bdlA; biofilm dispersion protein [KO:K03406]\n",
      "            PA4309  pctA; chemotactic transducer PctA [KO:K03406]\n",
      "line in gene PA4309  pctA; chemotactic transducer PctA [KO:K03406]\n",
      "            PA4307  pctC; chemotactic transducer PctC [KO:K03406]\n",
      "line in gene PA4307  pctC; chemotactic transducer PctC [KO:K03406]\n",
      "            PA2561  methyl-accepting chemotaxis protein CtpH [KO:K03406]\n",
      "line in gene PA2561  methyl-accepting chemotaxis protein CtpH [KO:K03406]\n",
      "            PA2654  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA2654  chemotaxis transducer [KO:K03406]\n",
      "            PA2920  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA2920  chemotaxis transducer [KO:K03406]\n",
      "            PA4520  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA4520  chemotaxis transducer [KO:K03406]\n",
      "            PA1251  chemotaxis transducer [KO:K03406]\n",
      "line in gene PA1251  chemotaxis transducer [KO:K03406]\n",
      "            PA1561  aer; aerotaxis receptor Aer [KO:K03776]\n",
      "line in gene PA1561  aer; aerotaxis receptor Aer [KO:K03776]\n",
      "            PA1946  rbsB; ribose ABC transporter substrate-binding protein [KO:K10439]\n",
      "line in gene PA1946  rbsB; ribose ABC transporter substrate-binding protein [KO:K10439]\n",
      "            PA4496  ABC transporter [KO:K12368]\n",
      "line in gene PA4496  ABC transporter [KO:K12368]\n",
      "            PA4500  ABC transporter [KO:K12368]\n",
      "line in gene PA4500  ABC transporter [KO:K12368]\n",
      "            PA4502  ABC transporter [KO:K12368]\n",
      "line in gene PA4502  ABC transporter [KO:K12368]\n",
      "            PA5317  dipeptide ABC transporter substrate-binding protein [KO:K12368]\n",
      "line in gene PA5317  dipeptide ABC transporter substrate-binding protein [KO:K12368]\n",
      "            PA4497  ABC transporter [KO:K12368]\n",
      "line in gene PA4497  ABC transporter [KO:K12368]\n",
      "            PA0178  two-component sensor [KO:K03407] [EC:2.7.13.3]\n",
      "line in gene PA0178  two-component sensor [KO:K03407] [EC:2.7.13.3]\n",
      "            PA1458  two-component sensor [KO:K03407] [EC:2.7.13.3]\n",
      "line in gene PA1458  two-component sensor [KO:K03407] [EC:2.7.13.3]\n",
      "            PA0177  purine-binding chemotaxis protein [KO:K03408]\n",
      "line in gene PA0177  purine-binding chemotaxis protein [KO:K03408]\n",
      "            PA1464  purine-binding chemotaxis protein [KO:K03408]\n",
      "line in gene PA1464  purine-binding chemotaxis protein [KO:K03408]\n",
      "            PA1463  hypothetical protein [KO:K03408]\n",
      "line in gene PA1463  hypothetical protein [KO:K03408]\n",
      "            PA0179  two-component response regulator [KO:K03413]\n",
      "line in gene PA0179  two-component response regulator [KO:K03413]\n",
      "            PA1456  cheY; chemotaxis protein CheY [KO:K03413]\n",
      "line in gene PA1456  cheY; chemotaxis protein CheY [KO:K03413]\n",
      "            PA1457  cheZ; protein phosphatase CheZ [KO:K03414]\n",
      "line in gene PA1457  cheZ; protein phosphatase CheZ [KO:K03414]\n",
      "            PA0173  chemotaxis response regulator protein-glutamate methylesterase [KO:K03412] [EC:3.1.1.61 3.5.1.44]\n",
      "line in gene PA0173  chemotaxis response regulator protein-glutamate methylesterase [KO:K03412] [EC:3.1.1.61 3.5.1.44]\n",
      "            PA1459  chemotaxis-specific methylesterase [KO:K03412] [EC:3.1.1.61 3.5.1.44]\n",
      "line in gene PA1459  chemotaxis-specific methylesterase [KO:K03412] [EC:3.1.1.61 3.5.1.44]\n",
      "            PA3349  chemotaxis protein [KO:K03415]\n",
      "line in gene PA3349  chemotaxis protein [KO:K03415]\n",
      "            PA0174  hypothetical protein [KO:K03411] [EC:3.5.1.44]\n",
      "line in gene PA0174  hypothetical protein [KO:K03411] [EC:3.5.1.44]\n",
      "            PA0175  chemotaxis protein methyltransferase [KO:K00575] [EC:2.1.1.80]\n",
      "line in gene PA0175  chemotaxis protein methyltransferase [KO:K00575] [EC:2.1.1.80]\n",
      "            PA3348  chemotaxis protein methyltransferase [KO:K00575] [EC:2.1.1.80]\n",
      "line in gene PA3348  chemotaxis protein methyltransferase [KO:K00575] [EC:2.1.1.80]\n",
      "            PA1102  fliG; flagellar motor switch protein FliG [KO:K02410]\n",
      "line in gene PA1102  fliG; flagellar motor switch protein FliG [KO:K02410]\n",
      "            PA1443  fliM; flagellar motor switch protein FliM [KO:K02416]\n",
      "line in gene PA1443  fliM; flagellar motor switch protein FliM [KO:K02416]\n",
      "            PA1444  fliN; flagellar motor switch protein FliN [KO:K02417]\n",
      "line in gene PA1444  fliN; flagellar motor switch protein FliN [KO:K02417]\n",
      "            PA1460  motC; flagellar motor protein [KO:K02556]\n",
      "line in gene PA1460  motC; flagellar motor protein [KO:K02556]\n",
      "            PA4954  motA; flagellar motor protein MotA [KO:K02556]\n",
      "line in gene PA4954  motA; flagellar motor protein MotA [KO:K02556]\n",
      "            PA1461  motD; flagellar motor protein MotD [KO:K02557]\n",
      "line in gene PA1461  motD; flagellar motor protein MotD [KO:K02557]\n",
      "            PA4953  motB; flagellar motor protein MotB [KO:K02557]\n",
      "line in gene PA4953  motB; flagellar motor protein MotB [KO:K02557]\n",
      "COMPOUND    C00049  L-Aspartate\n",
      "            C00107  Dipeptide\n",
      "            C00121  D-Ribose\n",
      "            C00124  D-Galactose\n",
      "            C00208  Maltose\n",
      "            C00716  Serine\n",
      "REFERENCE   PMID:12011078\n",
      "  AUTHORS   Kristich CJ, Ordal GW.\n",
      "  TITLE     Bacillus subtilis CheD is a chemoreceptor modification enzyme required for chemotaxis.\n",
      "  JOURNAL   J Biol Chem 277:25356-62 (2002)\n",
      "            DOI:10.1074/jbc.M201334200\n",
      "REFERENCE   PMID:11722727\n",
      "  AUTHORS   Kirby JR, Kristich CJ, Saulmon MM, Zimmer MA, Garrity LF, Zhulin IB, Ordal GW.\n",
      "  TITLE     CheC is related to the family of flagellar switch proteins and acts independently from CheD to control chemotaxis in Bacillus subtilis.\n",
      "  JOURNAL   Mol Microbiol 42:573-85 (2001)\n",
      "            DOI:10.1046/j.1365-2958.2001.02581.x\n",
      "REFERENCE   PMID:12920116\n",
      "  AUTHORS   Szurmant H, Bunn MW, Cannistraro VJ, Ordal GW.\n",
      "  TITLE     Bacillus subtilis hydrolyzes CheY-P at the location of its action, the flagellar switch.\n",
      "  JOURNAL   J Biol Chem 278:48611-6 (2003)\n",
      "            DOI:10.1074/jbc.M306180200\n",
      "REFERENCE   PMID:14749334\n",
      "  AUTHORS   Szurmant H, Muff TJ, Ordal GW.\n",
      "  TITLE     Bacillus subtilis CheC and FliY are members of a novel class of CheY-P-hydrolyzing proteins in the chemotactic signal transduction cascade.\n",
      "  JOURNAL   J Biol Chem 279:21787-92 (2004)\n",
      "            DOI:10.1074/jbc.M311497200\n",
      "REFERENCE   PMID:11158445\n",
      "  AUTHORS   Yoshihara S, Geng X, Okamoto S, Yura K, Murata T, Go M, Ohmori M, Ikeuchi M.\n",
      "  TITLE     Mutational analysis of genes involved in pilus structure, motility and transformation competency in the unicellular motile cyanobacterium Synechocystis sp. PCC 6803.\n",
      "  JOURNAL   Plant Cell Physiol 42:63-73 (2001)\n",
      "            DOI:10.1093/pcp/pce007\n",
      "REFERENCE   PMID:8104179\n",
      "  AUTHORS   Darzins A.\n",
      "  TITLE     The pilG gene product, required for Pseudomonas aeruginosa pilus production and twitching motility, is homologous to the enteric, single-domain response regulator CheY.\n",
      "  JOURNAL   J Bacteriol 175:5934-44 (1993)\n",
      "            DOI:10.1128/JB.175.18.5934-5944.1993\n",
      "REFERENCE   PMID:15187186\n",
      "  AUTHORS   Szurmant H, Ordal GW\n",
      "  TITLE     Diversity in chemotaxis mechanisms among the bacteria and archaea.\n",
      "  JOURNAL   Microbiol Mol Biol Rev 68:301-19 (2004)\n",
      "            DOI:10.1128/MMBR.68.2.301-319.2004\n",
      "REFERENCE   PMID:18774298\n",
      "  AUTHORS   Rao CV, Glekas GD, Ordal GW\n",
      "  TITLE     The three adaptation systems of Bacillus subtilis chemotaxis.\n",
      "  JOURNAL   Trends Microbiol 16:480-7 (2008)\n",
      "            DOI:10.1016/j.tim.2008.07.003\n",
      "REFERENCE   PMID:18990184\n",
      "  AUTHORS   Muff TJ, Ordal GW\n",
      "  TITLE     The diverse CheC-type phosphatases: chemotaxis and beyond.\n",
      "  JOURNAL   Mol Microbiol 70:1054-61 (2008)\n",
      "            DOI:10.1111/j.1365-2958.2008.06482.x\n",
      "REFERENCE   PMID:19203648\n",
      "  AUTHORS   Miller LD, Russell MH, Alexandre G\n",
      "  TITLE     Diversity in bacterial chemotactic responses and niche adaptation.\n",
      "  JOURNAL   Adv Appl Microbiol 66:53-75 (2009)\n",
      "            DOI:10.1016/S0065-2164(08)00803-4\n",
      "REL_PATHWAY pae02040  Flagellar assembly\n",
      "KO_PATHWAY  ko02030\n",
      "///\n",
      "path:pae02040\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae02040                    Pathway\n",
      "NAME        Flagellar assembly - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Cellular Processes; Cell motility\n",
      "PATHWAY_MAP pae02040  Flagellar assembly\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1100  fliE; flagellar hook-basal body complex protein FliE [KO:K02408]\n",
      "line in gene PA1100  fliE; flagellar hook-basal body complex protein FliE [KO:K02408]\n",
      "            PA1101  fliF; flagellar MS-ring protein [KO:K02409]\n",
      "line in gene PA1101  fliF; flagellar MS-ring protein [KO:K02409]\n",
      "            PA1102  fliG; flagellar motor switch protein FliG [KO:K02410]\n",
      "line in gene PA1102  fliG; flagellar motor switch protein FliG [KO:K02410]\n",
      "            PA1103  flagellar assembly protein FliH [KO:K02411]\n",
      "line in gene PA1103  flagellar assembly protein FliH [KO:K02411]\n",
      "            PA1104  fliI; flagellum-specific ATP synthase [KO:K02412] [EC:7.4.2.8]\n",
      "line in gene PA1104  fliI; flagellum-specific ATP synthase [KO:K02412] [EC:7.4.2.8]\n",
      "            PA1105  fliJ; flagellar biosynthesis chaperone [KO:K02413]\n",
      "line in gene PA1105  fliJ; flagellar biosynthesis chaperone [KO:K02413]\n",
      "            PA1441  hypothetical protein [KO:K02414]\n",
      "line in gene PA1441  hypothetical protein [KO:K02414]\n",
      "            PA1442  flagellar basal body protein FliL [KO:K02415]\n",
      "line in gene PA1442  flagellar basal body protein FliL [KO:K02415]\n",
      "            PA5233  flagellar basal body protein FliL [KO:K02415]\n",
      "line in gene PA5233  flagellar basal body protein FliL [KO:K02415]\n",
      "            PA1443  fliM; flagellar motor switch protein FliM [KO:K02416]\n",
      "line in gene PA1443  fliM; flagellar motor switch protein FliM [KO:K02416]\n",
      "            PA1444  fliN; flagellar motor switch protein FliN [KO:K02417]\n",
      "line in gene PA1444  fliN; flagellar motor switch protein FliN [KO:K02417]\n",
      "            PA1445  fliO; flagellar protein FliO [KO:K02418]\n",
      "line in gene PA1445  fliO; flagellar protein FliO [KO:K02418]\n",
      "            PA1446  fliP; flagellar biosynthesis protein FliP [KO:K02419]\n",
      "line in gene PA1446  fliP; flagellar biosynthesis protein FliP [KO:K02419]\n",
      "            PA1447  fliQ; flagellar biosynthesis protein FliQ [KO:K02420]\n",
      "line in gene PA1447  fliQ; flagellar biosynthesis protein FliQ [KO:K02420]\n",
      "            PA1448  fliR; flagellar biosynthesis protein FliR [KO:K02421]\n",
      "line in gene PA1448  fliR; flagellar biosynthesis protein FliR [KO:K02421]\n",
      "            PA1452  flhA; flagellar biosynthesis protein FlhA [KO:K02400]\n",
      "line in gene PA1452  flhA; flagellar biosynthesis protein FlhA [KO:K02400]\n",
      "            PA1449  flhB; flagellar biosynthesis protein FlhB [KO:K02401]\n",
      "line in gene PA1449  flhB; flagellar biosynthesis protein FlhB [KO:K02401]\n",
      "            PA3350  flagellar basal body P-ring biosynthesis protein FlgA [KO:K02386]\n",
      "line in gene PA3350  flagellar basal body P-ring biosynthesis protein FlgA [KO:K02386]\n",
      "            PA1077  flgB; flagellar basal-body rod protein FlgB [KO:K02387]\n",
      "line in gene PA1077  flgB; flagellar basal-body rod protein FlgB [KO:K02387]\n",
      "            PA1078  flgC; flagellar basal body rod protein FlgC [KO:K02388]\n",
      "line in gene PA1078  flgC; flagellar basal body rod protein FlgC [KO:K02388]\n",
      "            PA1079  flgD; flagellar basal body rod modification protein [KO:K02389]\n",
      "line in gene PA1079  flgD; flagellar basal body rod modification protein [KO:K02389]\n",
      "            PA1080  flgE; flagellar hook protein FlgE [KO:K02390]\n",
      "line in gene PA1080  flgE; flagellar hook protein FlgE [KO:K02390]\n",
      "            PA1081  flgF; flagellar basal body rod protein FlgF [KO:K02391]\n",
      "line in gene PA1081  flgF; flagellar basal body rod protein FlgF [KO:K02391]\n",
      "            PA1082  flgG; flagellar basal body rod protein FlgG [KO:K02392]\n",
      "line in gene PA1082  flgG; flagellar basal body rod protein FlgG [KO:K02392]\n",
      "            PA1083  flgH; flagellar basal body L-ring protein [KO:K02393]\n",
      "line in gene PA1083  flgH; flagellar basal body L-ring protein [KO:K02393]\n",
      "            PA1084  flgI; flagellar basal body P-ring protein [KO:K02394]\n",
      "line in gene PA1084  flgI; flagellar basal body P-ring protein [KO:K02394]\n",
      "            PA1085  flgJ; peptidoglycan hydrolase FlgJ [KO:K02395]\n",
      "line in gene PA1085  flgJ; peptidoglycan hydrolase FlgJ [KO:K02395]\n",
      "            PA1086  flgK; flagellar hook-associated protein FlgK [KO:K02396]\n",
      "line in gene PA1086  flgK; flagellar hook-associated protein FlgK [KO:K02396]\n",
      "            PA1087  flgL; flagellar hook-associated protein FlgL [KO:K02397]\n",
      "line in gene PA1087  flgL; flagellar hook-associated protein FlgL [KO:K02397]\n",
      "            PA3526  hypothetical protein [KO:K21218]\n",
      "line in gene PA3526  hypothetical protein [KO:K21218]\n",
      "            PA1092  fliC; B-type flagellin [KO:K02406]\n",
      "line in gene PA1092  fliC; B-type flagellin [KO:K02406]\n",
      "            PA1094  fliD; B-type flagellar hook-associated protein [KO:K02407]\n",
      "line in gene PA1094  fliD; B-type flagellar hook-associated protein [KO:K02407]\n",
      "            PA1095  B-type flagellar protein FliS [KO:K02422]\n",
      "line in gene PA1095  B-type flagellar protein FliS [KO:K02422]\n",
      "            PA1096  hypothetical protein [KO:K02423]\n",
      "line in gene PA1096  hypothetical protein [KO:K02423]\n",
      "            PA1460  motC; flagellar motor protein [KO:K02556]\n",
      "line in gene PA1460  motC; flagellar motor protein [KO:K02556]\n",
      "            PA4954  motA; flagellar motor protein MotA [KO:K02556]\n",
      "line in gene PA4954  motA; flagellar motor protein MotA [KO:K02556]\n",
      "            PA1461  motD; flagellar motor protein MotD [KO:K02557]\n",
      "line in gene PA1461  motD; flagellar motor protein MotD [KO:K02557]\n",
      "            PA4953  motB; flagellar motor protein MotB [KO:K02557]\n",
      "line in gene PA4953  motB; flagellar motor protein MotB [KO:K02557]\n",
      "            PA3351  flgM; protein FlgM [KO:K02398]\n",
      "line in gene PA3351  flgM; protein FlgM [KO:K02398]\n",
      "            PA3352  hypothetical protein [KO:K02399]\n",
      "line in gene PA3352  hypothetical protein [KO:K02399]\n",
      "            PA0314  L-cysteine ABC transporter protein FliY [KO:K02424]\n",
      "line in gene PA0314  L-cysteine ABC transporter protein FliY [KO:K02424]\n",
      "            PA0576  rpoD; RNA polymerase sigma factor RpoD [KO:K03086]\n",
      "line in gene PA0576  rpoD; RNA polymerase sigma factor RpoD [KO:K03086]\n",
      "            PA1097  fleQ; transcriptional regulator FleQ [KO:K10941]\n",
      "line in gene PA1097  fleQ; transcriptional regulator FleQ [KO:K10941]\n",
      "            PA4462  rpoN; RNA polymerase factor sigma-54 [KO:K03092]\n",
      "line in gene PA4462  rpoN; RNA polymerase factor sigma-54 [KO:K03092]\n",
      "            PA1099  fleR; two-component response regulator [KO:K10943]\n",
      "line in gene PA1099  fleR; two-component response regulator [KO:K10943]\n",
      "            PA1455  fliA; flagellar biosynthesis sigma factor FliA [KO:K02405]\n",
      "line in gene PA1455  fliA; flagellar biosynthesis sigma factor FliA [KO:K02405]\n",
      "REFERENCE   PMID:24970213\n",
      "  AUTHORS   Morimoto YV, Minamino T\n",
      "  TITLE     Structure and function of the bi-directional bacterial flagellar motor.\n",
      "  JOURNAL   Biomolecules 4:217-34 (2014)\n",
      "            DOI:10.3390/biom4010217\n",
      "REFERENCE   PMID:11520608\n",
      "  AUTHORS   Aizawa SI.\n",
      "  TITLE     Bacterial flagella and type III secretion systems.\n",
      "  JOURNAL   FEMS Microbiol Lett 202:157-64 (2001)\n",
      "            DOI:10.1111/j.1574-6968.2001.tb10797.x\n",
      "REFERENCE   PMID:11934612\n",
      "  AUTHORS   Aldridge P, Hughes KT.\n",
      "  TITLE     Regulation of flagellar assembly.\n",
      "  JOURNAL   Curr Opin Microbiol 5:160-5 (2002)\n",
      "            DOI:10.1016/S1369-5274(02)00302-8\n",
      "REFERENCE   PMID:32781026\n",
      "  AUTHORS   Takekawa N, Imada K, Homma M\n",
      "  TITLE     Structure and Energy-Conversion Mechanism of the Bacterial Na(+)-Driven Flagellar Motor.\n",
      "  JOURNAL   Trends Microbiol 28:719-731 (2020)\n",
      "            DOI:10.1016/j.tim.2020.03.010\n",
      "REFERENCE   PMID:17644605\n",
      "  AUTHORS   Liu R, Ochman H\n",
      "  TITLE     Origins of flagellar gene operons and secondary flagellar systems.\n",
      "  JOURNAL   J Bacteriol 189:7098-104 (2007)\n",
      "            DOI:10.1128/JB.00643-07\n",
      "REFERENCE   PMID:26615435\n",
      "  AUTHORS   Osterman IA, Dikhtyar YY, Bogdanov AA, Dontsova OA, Sergiev PV\n",
      "  TITLE     Regulation of Flagellar Gene Expression in Bacteria.\n",
      "  JOURNAL   Biochemistry (Mosc) 80:1447-56 (2015)\n",
      "            DOI:10.1134/S000629791511005X\n",
      "REFERENCE   PMID:32816086\n",
      "  AUTHORS   Khan F, Tabassum N, Anand R, Kim YM\n",
      "  TITLE     Motility of Vibrio spp.: regulation and controlling strategies.\n",
      "  JOURNAL   Appl Microbiol Biotechnol 104:8187-8208 (2020)\n",
      "            DOI:10.1007/s00253-020-10794-7\n",
      "REL_PATHWAY pae02030  Bacterial chemotaxis\n",
      "            pae03070  Bacterial secretion system\n",
      "KO_PATHWAY  ko02040\n",
      "///\n",
      "path:pae02060\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae02060                    Pathway\n",
      "NAME        Phosphotransferase system (PTS) - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION The phosphoenolpyruvate (PEP)-dependent phosphotransferase system (PTS) is a major mechanism used by bacteria for uptake of carbohydrates, particularly hexoses, hexitols, and disaccharides, where the source of energy is from PEP. The PTS consists of two general components, enzyme I (EI) and histidine phosphocarrier protein (HPr), and of membrane-bound sugar specific permeases (enzymes II). Each enzyme II (EII) complex consists of one or two hydrophobic integral membrane domains (domains C and D) and two hydrophilic domains (domains A and B). EII complexes may exist as distinct proteins or as a single multidomain protein. The PTS catalyzes the uptake of carbohydrates and their conversion into their respective phosphoesters during transport. There are four successive phosphoryl transfers in the PTS. Initial autophosphorylation of EI, using PEP as a substrate, is followed by transfer of the phosphoryl group from EI to HPr. EIIA catalyzes the self-phosphoryl transfer from HPr after which the phosphoryl group is transferred to histidine or cysteine residues of EIIB. The sugar is transported through the membrane-bound EIIC and is phosphorylated by the appropriate sugar-specific EIIB.\n",
      "CLASS       Environmental Information Processing; Membrane transport\n",
      "PATHWAY_MAP pae02060  Phosphotransferase system (PTS)\n",
      "DBLINKS     GO: 0009401\n",
      "            TC: 4.A\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4466  phosphoryl carrier protein [KO:K02784]\n",
      "line in gene PA4466  phosphoryl carrier protein [KO:K02784]\n",
      "            PA3761  N-acetyl-D-glucosamine phosphotransferase system transporter [KO:K02804] [EC:2.7.1.193]\n",
      "line in gene PA3761  N-acetyl-D-glucosamine phosphotransferase system transporter [KO:K02804] [EC:2.7.1.193]\n",
      "            PA3562  fruI; PTS system fructose-specific transporter subunit FruI [KO:K23993] [EC:2.7.3.9 2.7.1.202]\n",
      "line in gene PA3562  fruI; PTS system fructose-specific transporter subunit FruI [KO:K23993] [EC:2.7.3.9 2.7.1.202]\n",
      "            PA3760  N-acetyl-D-glucosamine phosphotransferase system transporter [KO:K23993] [EC:2.7.3.9 2.7.1.202]\n",
      "line in gene PA3760  N-acetyl-D-glucosamine phosphotransferase system transporter [KO:K23993] [EC:2.7.3.9 2.7.1.202]\n",
      "            PA3560  fruA; PTS system fructose-specific transporter subunit IIBC [KO:K02770] [EC:2.7.1.202]\n",
      "line in gene PA3560  fruA; PTS system fructose-specific transporter subunit IIBC [KO:K02770] [EC:2.7.1.202]\n",
      "            PA3561  fruK; 1-phosphofructokinase [KO:K00882] [EC:2.7.1.56]\n",
      "line in gene PA3561  fruK; 1-phosphofructokinase [KO:K00882] [EC:2.7.1.56]\n",
      "            PA0337  ptsP; phosphoenolpyruvate-protein phosphotransferase PtsP [KO:K08484] [EC:2.7.3.9]\n",
      "line in gene PA0337  ptsP; phosphoenolpyruvate-protein phosphotransferase PtsP [KO:K08484] [EC:2.7.3.9]\n",
      "            PA4464  ptsN; nitrogen regulatory IIA protein [KO:K02806] [EC:2.7.1.-]\n",
      "line in gene PA4464  ptsN; nitrogen regulatory IIA protein [KO:K02806] [EC:2.7.1.-]\n",
      "COMPOUND    C00022  Pyruvate\n",
      "            C00031  D-Glucose\n",
      "            C00072  Ascorbate\n",
      "            C00074  Phosphoenolpyruvate\n",
      "            C00089  Sucrose\n",
      "            C00092  D-Glucose 6-phosphate\n",
      "            C00095  D-Fructose\n",
      "            C00124  D-Galactose\n",
      "            C00140  N-Acetyl-D-glucosamine\n",
      "            C00159  D-Mannose\n",
      "            C00185  Cellobiose\n",
      "            C00208  Maltose\n",
      "            C00243  Lactose\n",
      "            C00275  D-Mannose 6-phosphate\n",
      "            C00329  D-Glucosamine\n",
      "            C00352  D-Glucosamine 6-phosphate\n",
      "            C00357  N-Acetyl-D-glucosamine 6-phosphate\n",
      "            C00392  Mannitol\n",
      "            C00644  D-Mannitol 1-phosphate\n",
      "            C00689  alpha,alpha'-Trehalose 6-phosphate\n",
      "            C00697  Nitrogen\n",
      "            C00794  D-Sorbitol\n",
      "            C01083  alpha,alpha-Trehalose\n",
      "            C01094  D-Fructose 1-phosphate\n",
      "            C01096  Sorbitol 6-phosphate\n",
      "            C01113  D-Galactose 6-phosphate\n",
      "            C01132  N-Acetyl-D-galactosamine\n",
      "            C01451  Salicin\n",
      "            C01452  Sorbose\n",
      "            C01674  Chitobiose\n",
      "            C01697  Galactitol\n",
      "            C01798  D-Glucoside\n",
      "            C02262  D-Galactosamine\n",
      "            C02713  N-Acetylmuramate\n",
      "            C02888  Sorbose 1-phosphate\n",
      "            C02995  Maltose 6'-phosphate\n",
      "            C03752  2-Amino-2-deoxy-D-gluconate\n",
      "            C04534  6-Phospho-beta-D-glucosyl-(1,4)-D-glucose\n",
      "            C05378  beta-D-Fructose 1,6-bisphosphate\n",
      "            C05396  Lactose 6'-phosphate\n",
      "            C06186  Arbutin\n",
      "            C06187  Arbutin 6-phosphate\n",
      "            C06188  Salicin 6-phosphate\n",
      "            C06311  Galactitol 1-phosphate\n",
      "            C06376  N-Acetyl-D-galactosamine 6-phosphate\n",
      "            C06377  D-Galactosamine 6-phosphate\n",
      "            C11544  2-O-(alpha-D-Mannosyl)-D-glycerate\n",
      "            C16186  L-Ascorbate 6-phosphate\n",
      "            C16488  Fructoselysine\n",
      "            C16489  Fructoselysine 6-phosphate\n",
      "            C16688  Sucrose 6-phosphate\n",
      "            C16698  N-Acetylmuramic acid 6-phosphate\n",
      "            C16699  2-O-(6-Phospho-alpha-mannosyl)-D-glycerate\n",
      "            C20589  D-Glucosaminate-6-phosphate\n",
      "            C20978  Glucoselysine\n",
      "            C20979  Glucoselysine-6-phosphate\n",
      "            C21152  N,N'-Diacetylchitobiose 6'-phosphate\n",
      "REFERENCE   PMID:11361076\n",
      "  AUTHORS   Darbon E, Galinier A, Le Coq D, Deutscher J.\n",
      "  TITLE     Phosphotransfer functions mutated Bacillus subtilis HPr-like protein Crh carrying a histidine in the active site.\n",
      "  JOURNAL   J Mol Microbiol Biotechnol 3:439-44 (2001)\n",
      "REFERENCE   PMID:15060041\n",
      "  AUTHORS   Dahl U, Jaeger T, Nguyen BT, Sattler JM, Mayer C.\n",
      "  TITLE     Identification of a phosphotransferase system of Escherichia coli required for growth on N-acetylmuramic acid.\n",
      "  JOURNAL   J Bacteriol 186:2385-92 (2004)\n",
      "            DOI:10.1128/JB.186.8.2385-2392.2004\n",
      "REFERENCE   PMID:2117666\n",
      "  AUTHORS   Martin-Verstraete I, Debarbouille M, Klier A, Rapoport G\n",
      "  TITLE     Levanase operon of Bacillus subtilis includes a fructose-specific phosphotransferase system regulating the expression of the operon.\n",
      "  JOURNAL   J Mol Biol 214:657-71 (1990)\n",
      "            DOI:10.1016/0022-2836(90)90284-S\n",
      "REFERENCE   PMID:12603740\n",
      "  AUTHORS   Kristich CJ, Glekas GD, Ordal GW.\n",
      "  TITLE     The conserved cytoplasmic module of the transmembrane chemoreceptor McpC mediates carbohydrate chemotaxis in Bacillus subtilis.\n",
      "  JOURNAL   Mol Microbiol 47:1353-66 (2003)\n",
      "            DOI:10.1046/j.1365-2958.2003.03375.x\n",
      "REFERENCE   PMID:14645248\n",
      "  AUTHORS   Sampaio MM, Chevance F, Dippel R, Eppler T, Schlegel A, Boos W, Lu YJ, Rock CO.\n",
      "  TITLE     Phosphotransferase-mediated transport of the osmolyte 2-O-alpha-mannosyl-D-glycerate in Escherichia coli occurs by the product of the mngA (hrsA) gene and is regulated by the mngR (farR) gene product acting as repressor.\n",
      "  JOURNAL   J Biol Chem 279:5537-48 (2004)\n",
      "            DOI:10.1074/jbc.M310980200\n",
      "REFERENCE   PMID:15608210\n",
      "  AUTHORS   Keseler IM, Collado-Vides J, Gama-Castro S, Ingraham J, Paley S, Paulsen IT, Peralta-Gil M, Karp PD.\n",
      "  TITLE     EcoCyc: a comprehensive database resource for Escherichia coli.\n",
      "  JOURNAL   Nucleic Acids Res 33:D334-7 (2005)\n",
      "            DOI:10.1093/nar/gki108\n",
      "REFERENCE   PMID:18156259\n",
      "  AUTHORS   Mukherjee A, Mammel MK, LeClerc JE, Cebula TA.\n",
      "  TITLE     Altered utilization of N-acetyl-D-galactosamine by Escherichia coli O157:H7 from the 2006 spinach outbreak.\n",
      "  JOURNAL   J Bacteriol 190:1710-7 (2008)\n",
      "            DOI:10.1128/JB.01737-07\n",
      "REFERENCE   PMID:10931310\n",
      "  AUTHORS   Brinkkotter A, Kloss H, Alpert C, Lengeler JW.\n",
      "  TITLE     Pathways for the utilization of N-acetyl-galactosamine and galactosamine in Escherichia coli.\n",
      "  JOURNAL   Mol Microbiol 37:125-35 (2000)\n",
      "            DOI:10.1046/j.1365-2958.2000.01969.x\n",
      "REFERENCE   PMID:11361063\n",
      "  AUTHORS   Tchieu JH, Norris V, Edwards JS, Saier MH Jr.\n",
      "  TITLE     The complete phosphotranferase system in Escherichia coli.\n",
      "  JOURNAL   J Mol Microbiol Biotechnol 3:329-46 (2001)\n",
      "REFERENCE   PMID:16233138\n",
      "  AUTHORS   Kotrba P, Inui M, Yukawa H.\n",
      "  TITLE     Bacterial phosphotransferase system (PTS) in carbohydrate uptake and control of carbon metabolism.\n",
      "  JOURNAL   J Biosci Bioeng 92:502-17 (2001)\n",
      "            DOI:10.1016/S1389-1723(01)80308-X\n",
      "REL_PATHWAY pae00051  Fructose and mannose metabolism\n",
      "KO_PATHWAY  ko02060\n",
      "///\n",
      "path:pae03010\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae03010                    Pathway\n",
      "NAME        Ribosome - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Genetic Information Processing; Translation\n",
      "PATHWAY_MAP pae03010  Ribosome\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3162  rpsA; 30S ribosomal protein S1 [KO:K02945]\n",
      "line in gene PA3162  rpsA; 30S ribosomal protein S1 [KO:K02945]\n",
      "            PA3656  rpsB; 30S ribosomal protein S2 [KO:K02967]\n",
      "line in gene PA3656  rpsB; 30S ribosomal protein S2 [KO:K02967]\n",
      "            PA4257  rpsC; 30S ribosomal protein S3 [KO:K02982]\n",
      "line in gene PA4257  rpsC; 30S ribosomal protein S3 [KO:K02982]\n",
      "            PA4239  rpsD; 30S ribosomal protein S4 [KO:K02986]\n",
      "line in gene PA4239  rpsD; 30S ribosomal protein S4 [KO:K02986]\n",
      "            PA4246  rpsE; 30S ribosomal protein S5 [KO:K02988]\n",
      "line in gene PA4246  rpsE; 30S ribosomal protein S5 [KO:K02988]\n",
      "            PA4935  rpsF; 30S ribosomal protein S6 [KO:K02990]\n",
      "line in gene PA4935  rpsF; 30S ribosomal protein S6 [KO:K02990]\n",
      "            PA4267  rpsG; 30S ribosomal protein S7 [KO:K02992]\n",
      "line in gene PA4267  rpsG; 30S ribosomal protein S7 [KO:K02992]\n",
      "            PA4249  rpsH; 30S ribosomal protein S8 [KO:K02994]\n",
      "line in gene PA4249  rpsH; 30S ribosomal protein S8 [KO:K02994]\n",
      "            PA4432  rpsI; 30S ribosomal protein S9 [KO:K02996]\n",
      "line in gene PA4432  rpsI; 30S ribosomal protein S9 [KO:K02996]\n",
      "            PA4264  rpsJ; 30S ribosomal protein S10 [KO:K02946]\n",
      "line in gene PA4264  rpsJ; 30S ribosomal protein S10 [KO:K02946]\n",
      "            PA4240  rpsK; 30S ribosomal protein S11 [KO:K02948]\n",
      "line in gene PA4240  rpsK; 30S ribosomal protein S11 [KO:K02948]\n",
      "            PA4268  rpsL; 30S ribosomal protein S12 [KO:K02950]\n",
      "line in gene PA4268  rpsL; 30S ribosomal protein S12 [KO:K02950]\n",
      "            PA4241  rpsM; 30S ribosomal protein S13 [KO:K02952]\n",
      "line in gene PA4241  rpsM; 30S ribosomal protein S13 [KO:K02952]\n",
      "            PA4250  rpsN; 30S ribosomal protein S14 [KO:K02954]\n",
      "line in gene PA4250  rpsN; 30S ribosomal protein S14 [KO:K02954]\n",
      "            PA4741  rpsO; 30S ribosomal protein S15 [KO:K02956]\n",
      "line in gene PA4741  rpsO; 30S ribosomal protein S15 [KO:K02956]\n",
      "            PA3745  rpsP; 30S ribosomal protein S16 [KO:K02959]\n",
      "line in gene PA3745  rpsP; 30S ribosomal protein S16 [KO:K02959]\n",
      "            PA4254  rpsQ; 30S ribosomal protein S17 [KO:K02961]\n",
      "line in gene PA4254  rpsQ; 30S ribosomal protein S17 [KO:K02961]\n",
      "            PA4934  rpsR; 30S ribosomal protein S18 [KO:K02963]\n",
      "line in gene PA4934  rpsR; 30S ribosomal protein S18 [KO:K02963]\n",
      "            PA4259  rpsS; 30S ribosomal protein S19 [KO:K02965]\n",
      "line in gene PA4259  rpsS; 30S ribosomal protein S19 [KO:K02965]\n",
      "            PA4563  rpsT; 30S ribosomal protein S20 [KO:K02968]\n",
      "line in gene PA4563  rpsT; 30S ribosomal protein S20 [KO:K02968]\n",
      "            PA0579  rpsU; 30S ribosomal protein S21 [KO:K02970]\n",
      "line in gene PA0579  rpsU; 30S ribosomal protein S21 [KO:K02970]\n",
      "            PA4273  rplA; 50S ribosomal protein L1 [KO:K02863]\n",
      "line in gene PA4273  rplA; 50S ribosomal protein L1 [KO:K02863]\n",
      "            PA4260  rplB; 50S ribosomal protein L2 [KO:K02886]\n",
      "line in gene PA4260  rplB; 50S ribosomal protein L2 [KO:K02886]\n",
      "            PA4263  rplC; 50S ribosomal protein L3 [KO:K02906]\n",
      "line in gene PA4263  rplC; 50S ribosomal protein L3 [KO:K02906]\n",
      "            PA4262  rplD; 50S ribosomal protein L4 [KO:K02926]\n",
      "line in gene PA4262  rplD; 50S ribosomal protein L4 [KO:K02926]\n",
      "            PA4251  rplE; 50S ribosomal protein L5 [KO:K02931]\n",
      "line in gene PA4251  rplE; 50S ribosomal protein L5 [KO:K02931]\n",
      "            PA4248  rplF; 50S ribosomal protein L6 [KO:K02933]\n",
      "line in gene PA4248  rplF; 50S ribosomal protein L6 [KO:K02933]\n",
      "            PA4271  rplL; 50S ribosomal protein L7/L12 [KO:K02935]\n",
      "line in gene PA4271  rplL; 50S ribosomal protein L7/L12 [KO:K02935]\n",
      "            PA4932  rplI; 50S ribosomal protein L9 [KO:K02939]\n",
      "line in gene PA4932  rplI; 50S ribosomal protein L9 [KO:K02939]\n",
      "            PA4272  rplJ; 50S ribosomal protein L10 [KO:K02864]\n",
      "line in gene PA4272  rplJ; 50S ribosomal protein L10 [KO:K02864]\n",
      "            PA4274  rplK; 50S ribosomal protein L11 [KO:K02867]\n",
      "line in gene PA4274  rplK; 50S ribosomal protein L11 [KO:K02867]\n",
      "            PA4433  rplM; 50S ribosomal protein L13 [KO:K02871]\n",
      "line in gene PA4433  rplM; 50S ribosomal protein L13 [KO:K02871]\n",
      "            PA4253  rplN; 50S ribosomal protein L14 [KO:K02874]\n",
      "line in gene PA4253  rplN; 50S ribosomal protein L14 [KO:K02874]\n",
      "            PA4244  rplO; 50S ribosomal protein L15 [KO:K02876]\n",
      "line in gene PA4244  rplO; 50S ribosomal protein L15 [KO:K02876]\n",
      "            PA4256  rplP; 50S ribosomal protein L16 [KO:K02878]\n",
      "line in gene PA4256  rplP; 50S ribosomal protein L16 [KO:K02878]\n",
      "            PA4237  rplQ; 50S ribosomal protein L17 [KO:K02879]\n",
      "line in gene PA4237  rplQ; 50S ribosomal protein L17 [KO:K02879]\n",
      "            PA4247  rplR; 50S ribosomal protein L18 [KO:K02881]\n",
      "line in gene PA4247  rplR; 50S ribosomal protein L18 [KO:K02881]\n",
      "            PA3742  rplS; 50S ribosomal protein L19 [KO:K02884]\n",
      "line in gene PA3742  rplS; 50S ribosomal protein L19 [KO:K02884]\n",
      "            PA2741  rplT; 50S ribosomal protein L20 [KO:K02887]\n",
      "line in gene PA2741  rplT; 50S ribosomal protein L20 [KO:K02887]\n",
      "            PA4568  rplU; 50S ribosomal protein L21 [KO:K02888]\n",
      "line in gene PA4568  rplU; 50S ribosomal protein L21 [KO:K02888]\n",
      "            PA4258  rplV; 50S ribosomal protein L22 [KO:K02890]\n",
      "line in gene PA4258  rplV; 50S ribosomal protein L22 [KO:K02890]\n",
      "            PA4261  rplW; 50S ribosomal protein L23 [KO:K02892]\n",
      "line in gene PA4261  rplW; 50S ribosomal protein L23 [KO:K02892]\n",
      "            PA4252  rplX; 50S ribosomal protein L24 [KO:K02895]\n",
      "line in gene PA4252  rplX; 50S ribosomal protein L24 [KO:K02895]\n",
      "            PA4671  50S ribosomal protein L25/general stress protein Ctc [KO:K02897]\n",
      "line in gene PA4671  50S ribosomal protein L25/general stress protein Ctc [KO:K02897]\n",
      "            PA4567  rpmA; 50S ribosomal protein L27 [KO:K02899]\n",
      "line in gene PA4567  rpmA; 50S ribosomal protein L27 [KO:K02899]\n",
      "            PA5316  rpmB; 50S ribosomal protein L28 [KO:K02902]\n",
      "line in gene PA5316  rpmB; 50S ribosomal protein L28 [KO:K02902]\n",
      "            PA4255  rpmC; 50S ribosomal protein L29 [KO:K02904]\n",
      "line in gene PA4255  rpmC; 50S ribosomal protein L29 [KO:K02904]\n",
      "            PA4245  rpmD; 50S ribosomal protein L30 [KO:K02907]\n",
      "line in gene PA4245  rpmD; 50S ribosomal protein L30 [KO:K02907]\n",
      "            PA3601  50S ribosomal protein L31 type B [KO:K02909]\n",
      "line in gene PA3601  50S ribosomal protein L31 type B [KO:K02909]\n",
      "            PA5049  rpmE; 50S ribosomal protein L31 [KO:K02909]\n",
      "line in gene PA5049  rpmE; 50S ribosomal protein L31 [KO:K02909]\n",
      "            PA2970  rpmF; 50S ribosomal protein L32 [KO:K02911]\n",
      "line in gene PA2970  rpmF; 50S ribosomal protein L32 [KO:K02911]\n",
      "            PA5315  rpmG; 50S ribosomal protein L33 [KO:K02913]\n",
      "line in gene PA5315  rpmG; 50S ribosomal protein L33 [KO:K02913]\n",
      "            PA5570  rpmH; 50S ribosomal protein L34 [KO:K02914]\n",
      "line in gene PA5570  rpmH; 50S ribosomal protein L34 [KO:K02914]\n",
      "            PA2742  rpmI; 50S ribosomal protein L35 [KO:K02916]\n",
      "line in gene PA2742  rpmI; 50S ribosomal protein L35 [KO:K02916]\n",
      "            PA3600  50S ribosomal protein L36 [KO:K02919]\n",
      "line in gene PA3600  50S ribosomal protein L36 [KO:K02919]\n",
      "            PA4242  rpmJ; 50S ribosomal protein L36 [KO:K02919]\n",
      "line in gene PA4242  rpmJ; 50S ribosomal protein L36 [KO:K02919]\n",
      "            PA0668.1  16S ribosomal RNA [KO:K01977]\n",
      "line in gene PA0668.1  16S ribosomal RNA [KO:K01977]\n",
      "            PA4280.5  16S ribosomal RNA [KO:K01977]\n",
      "line in gene PA4280.5  16S ribosomal RNA [KO:K01977]\n",
      "            PA4690.5  16S ribosomal RNA [KO:K01977]\n",
      "line in gene PA4690.5  16S ribosomal RNA [KO:K01977]\n",
      "            PA5369.5  16S ribosomal RNA [KO:K01977]\n",
      "line in gene PA5369.5  16S ribosomal RNA [KO:K01977]\n",
      "            PA4690.2  23S ribosomal RNA [KO:K01980]\n",
      "line in gene PA4690.2  23S ribosomal RNA [KO:K01980]\n",
      "            PA5369.2  23S ribosomal RNA [KO:K01980]\n",
      "line in gene PA5369.2  23S ribosomal RNA [KO:K01980]\n",
      "            PA0668.4  23S ribosomal RNA [KO:K01980]\n",
      "line in gene PA0668.4  23S ribosomal RNA [KO:K01980]\n",
      "            PA4280.2  23S ribosomal RNA [KO:K01980]\n",
      "line in gene PA4280.2  23S ribosomal RNA [KO:K01980]\n",
      "            PA0668.5  5S ribosomal RNA [KO:K01985]\n",
      "line in gene PA0668.5  5S ribosomal RNA [KO:K01985]\n",
      "            PA5369.1  5S ribosomal RNA [KO:K01985]\n",
      "line in gene PA5369.1  5S ribosomal RNA [KO:K01985]\n",
      "            PA4280.1  5S ribosomal RNA [KO:K01985]\n",
      "line in gene PA4280.1  5S ribosomal RNA [KO:K01985]\n",
      "            PA4690.1  5S ribosomal RNA [KO:K01985]\n",
      "line in gene PA4690.1  5S ribosomal RNA [KO:K01985]\n",
      "REFERENCE   PMID:10476961\n",
      "  AUTHORS   Ban N, Nissen P, Hansen J, Capel M, Moore PB, Steitz TA.\n",
      "  TITLE     Placement of protein and RNA structures into a 5 A-resolution map of the 50S ribosomal subunit.\n",
      "  JOURNAL   Nature 400:841-847 (1999)\n",
      "            DOI:10.1038/23641\n",
      "REFERENCE   PMID:10476960\n",
      "  AUTHORS   Clemons WM Jr, May JL, Wimberly BT, McCutcheon JP, Capel MS, Ramakrishnan V.\n",
      "  TITLE     Structure of a bacterial 30S ribosomal subunit at 5.5 A resolution.\n",
      "  JOURNAL   Nature 400:833-840 (1999)\n",
      "            DOI:10.1038/23631\n",
      "REFERENCE   PMID:10497122\n",
      "  AUTHORS   Cate JH, Yusupov MM, Yusupova GZ, Earnest TN, Noller HF.\n",
      "  TITLE     X-ray crystal structures of 70S ribosome functional complexes.\n",
      "  JOURNAL   Science 285:2095-2104 (1999)\n",
      "            DOI:10.1126/science.285.5436.2095\n",
      "REFERENCE   PMID:10937989\n",
      "  AUTHORS   Ban N, Nissen P, Hansen J, Moore PB, Steitz TA.\n",
      "  TITLE     The complete atomic structure of the large ribosomal subunit at 2.4 A resolution.\n",
      "  JOURNAL   Science 289:905-920 (2000)\n",
      "            DOI:10.1126/science.289.5481.905\n",
      "REFERENCE   PMID:10937990\n",
      "  AUTHORS   Nissen P, Hansen J, Ban N, Moore PB, Steitz TA.\n",
      "  TITLE     The structural basis of ribosome activity in peptide bond synthesis.\n",
      "  JOURNAL   Science 289:920-930 (2000)\n",
      "            DOI:10.1126/science.289.5481.920\n",
      "REFERENCE   PMID:11014182\n",
      "  AUTHORS   Wimberly BT, Brodersen DE, Clemons WM Jr, Morgan-Warren RJ, Carter AP, Vonrhein C, Hartsch T, Ramakrishnan V.\n",
      "  TITLE     Structure of the 30S ribosomal subunit.\n",
      "  JOURNAL   Nature 407:327-339 (2000)\n",
      "            DOI:10.1038/35030006\n",
      "REFERENCE   PMID:11014183\n",
      "  AUTHORS   Carter AP, Clemons WM, Brodersen DE, Morgan-Warren RJ, Wimberly BT, Ramakrishnan V.\n",
      "  TITLE     Functional insights from the structure of the 30S ribosomal subunit and its interactions with antibiotics.\n",
      "  JOURNAL   Nature 407:340-348 (2000)\n",
      "            DOI:10.1038/35030019\n",
      "KO_PATHWAY  ko03010\n",
      "///\n",
      "path:pae03018\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae03018                    Pathway\n",
      "NAME        RNA degradation - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION The correct processing, quality control and turnover of cellular RNA molecules are critical to many aspects in the expression of genetic information. In eukaryotes, two major pathways of mRNA decay exist and both pathways are initiated by poly(A) shortening of the mRNA. In the 5' to 3' pathway, this is followed by decapping which then permits the 5' to 3' exonucleolytic degradation of transcripts. In the 3' to 5' pathway, the exosome, a large multisubunit complex, plays a key role. The exosome exists in archaeal cells, too. In bacteria, endoribonuclease E, a key enzyme involved in RNA decay and processing, organizes a protein complex called degradosome. RNase E or R interacts with the phosphate-dependent exoribonuclease polynucleotide phosphorylase, DEAD-box helicases, and additional factors in the RNA-degrading complex.\n",
      "CLASS       Genetic Information Processing; Folding, sorting and degradation\n",
      "PATHWAY_MAP pae03018  RNA degradation\n",
      "DBLINKS     GO: 0006401\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0336  ygdP; RNA pyrophosphohydrolase [KO:K08311] [EC:3.6.1.-]\n",
      "line in gene PA0336  ygdP; RNA pyrophosphohydrolase [KO:K08311] [EC:3.6.1.-]\n",
      "            PA2976  rne; ribonuclease E [KO:K08300] [EC:3.1.26.12]\n",
      "line in gene PA2976  rne; ribonuclease E [KO:K08300] [EC:3.1.26.12]\n",
      "            PA3861  rhl; ATP-dependent RNA helicase RhlB [KO:K03732] [EC:3.6.4.13]\n",
      "line in gene PA3861  rhl; ATP-dependent RNA helicase RhlB [KO:K03732] [EC:3.6.4.13]\n",
      "            PA3635  eno; enolase [KO:K01689] [EC:4.2.1.11]\n",
      "line in gene PA3635  eno; enolase [KO:K01689] [EC:4.2.1.11]\n",
      "            PA4740  pnp; polynucleotide phosphorylase [KO:K00962] [EC:2.7.7.8]\n",
      "line in gene PA4740  pnp; polynucleotide phosphorylase [KO:K00962] [EC:2.7.7.8]\n",
      "            PA0428  ATP-dependent RNA helicase [KO:K11927] [EC:3.6.4.13]\n",
      "line in gene PA0428  ATP-dependent RNA helicase [KO:K11927] [EC:3.6.4.13]\n",
      "            PA3950  ATP-dependent RNA helicase [KO:K11927] [EC:3.6.4.13]\n",
      "line in gene PA3950  ATP-dependent RNA helicase [KO:K11927] [EC:3.6.4.13]\n",
      "            PA4937  rnr; exoribonuclease R [KO:K12573] [EC:3.1.13.1]\n",
      "line in gene PA4937  rnr; exoribonuclease R [KO:K12573] [EC:3.1.13.1]\n",
      "            PA2840  ATP-dependent RNA helicase [KO:K05592] [EC:3.6.4.13]\n",
      "line in gene PA2840  ATP-dependent RNA helicase [KO:K05592] [EC:3.6.4.13]\n",
      "            PA3344  recQ; ATP-dependent DNA helicase RecQ [KO:K03654] [EC:5.6.2.4]\n",
      "line in gene PA3344  recQ; ATP-dependent DNA helicase RecQ [KO:K03654] [EC:5.6.2.4]\n",
      "            PA5239  rho; transcription termination factor Rho [KO:K03628]\n",
      "line in gene PA5239  rho; transcription termination factor Rho [KO:K03628]\n",
      "            PA4761  dnaK; molecular chaperone DnaK [KO:K04043]\n",
      "line in gene PA4761  dnaK; molecular chaperone DnaK [KO:K04043]\n",
      "            PA4385  groEL; molecular chaperone GroEL [KO:K04077]\n",
      "line in gene PA4385  groEL; molecular chaperone GroEL [KO:K04077]\n",
      "            PA4944  hfq; RNA-binding protein Hfq [KO:K03666]\n",
      "line in gene PA4944  hfq; RNA-binding protein Hfq [KO:K03666]\n",
      "            PA5242  ppk; polyphosphate kinase [KO:K00937] [EC:2.7.4.1]\n",
      "line in gene PA5242  ppk; polyphosphate kinase [KO:K00937] [EC:2.7.4.1]\n",
      "            PA2428  hypothetical protein [KO:K22468] [EC:2.7.4.34]\n",
      "line in gene PA2428  hypothetical protein [KO:K22468] [EC:2.7.4.34]\n",
      "            PA0141  hypothetical protein [KO:K22468] [EC:2.7.4.34]\n",
      "line in gene PA0141  hypothetical protein [KO:K22468] [EC:2.7.4.34]\n",
      "            PA4727  pcnB; poly(A) polymerase [KO:K00970] [EC:2.7.7.19]\n",
      "line in gene PA4727  pcnB; poly(A) polymerase [KO:K00970] [EC:2.7.7.19]\n",
      "REFERENCE   PMID:19239894\n",
      "  AUTHORS   Houseley J, Tollervey D\n",
      "  TITLE     The many pathways of RNA degradation.\n",
      "  JOURNAL   Cell 136:763-76 (2009)\n",
      "            DOI:10.1016/j.cell.2009.01.019\n",
      "REFERENCE   PMID:16968219\n",
      "  AUTHORS   Buttner K, Wenig K, Hopfner KP\n",
      "  TITLE     The exosome: a macromolecular cage for controlled RNA degradation.\n",
      "  JOURNAL   Mol Microbiol 61:1372-9 (2006)\n",
      "            DOI:10.1111/j.1365-2958.2006.05331.x\n",
      "REFERENCE   PMID:16939780\n",
      "  AUTHORS   Schilders G, van Dijk E, Raijmakers R, Pruijn GJ\n",
      "  TITLE     Cell and molecular biology of the exosome: how to make or break an RNA.\n",
      "  JOURNAL   Int Rev Cytol 251:159-208 (2006)\n",
      "            DOI:10.1016/S0074-7696(06)51005-8\n",
      "REFERENCE   PMID:17471261\n",
      "  AUTHORS   Hartung S, Hopfner KP\n",
      "  TITLE     The exosome, plugged.\n",
      "  JOURNAL   EMBO Rep 8:456-7 (2007)\n",
      "            DOI:10.1038/sj.embor.7400961\n",
      "REFERENCE   PMID:17447862\n",
      "  AUTHORS   Carpousis AJ\n",
      "  TITLE     The RNA degradosome of Escherichia coli: an mRNA-degrading machine assembled on RNase E.\n",
      "  JOURNAL   Annu Rev Microbiol 61:71-87 (2007)\n",
      "            DOI:10.1146/annurev.micro.61.080706.093440\n",
      "REFERENCE   PMID:16766188\n",
      "  AUTHORS   Marcaida MJ, DePristo MA, Chandran V, Carpousis AJ, Luisi BF\n",
      "  TITLE     The RNA degradosome: life in the fast lane of adaptive molecular evolution.\n",
      "  JOURNAL   Trends Biochem Sci 31:359-65 (2006)\n",
      "            DOI:10.1016/j.tibs.2006.05.005\n",
      "REFERENCE   PMID:17560162\n",
      "  AUTHORS   Condon C\n",
      "  TITLE     Maturation and degradation of RNA in bacteria.\n",
      "  JOURNAL   Curr Opin Microbiol 10:271-8 (2007)\n",
      "            DOI:10.1016/j.mib.2007.05.008\n",
      "KO_PATHWAY  ko03018\n",
      "///\n",
      "path:pae03020\n",
      "ENTRY       pae03020                    Pathway\n",
      "NAME        RNA polymerase - Pseudomonas aeruginosa PAO1\n",
      "CLASS       Genetic Information Processing; Transcription\n",
      "PATHWAY_MAP pae03020  RNA polymerase\n",
      "DBLINKS     GO: 0000428\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4270  rpoB; DNA-directed RNA polymerase subunit beta [KO:K03043] [EC:2.7.7.6]\n",
      "line in gene PA4270  rpoB; DNA-directed RNA polymerase subunit beta [KO:K03043] [EC:2.7.7.6]\n",
      "            PA4269  rpoC; DNA-directed RNA polymerase subunit beta' [KO:K03046] [EC:2.7.7.6]\n",
      "line in gene PA4269  rpoC; DNA-directed RNA polymerase subunit beta' [KO:K03046] [EC:2.7.7.6]\n",
      "            PA4238  rpoA; DNA-directed RNA polymerase subunit alpha [KO:K03040] [EC:2.7.7.6]\n",
      "line in gene PA4238  rpoA; DNA-directed RNA polymerase subunit alpha [KO:K03040] [EC:2.7.7.6]\n",
      "            PA5337  rpoZ; DNA-directed RNA polymerase subunit omega [KO:K03060] [EC:2.7.7.6]\n",
      "line in gene PA5337  rpoZ; DNA-directed RNA polymerase subunit omega [KO:K03060] [EC:2.7.7.6]\n",
      "REFERENCE   PMID:10499798\n",
      "  AUTHORS   Zhang G, Campbell EA, Minakhin L, Richter C, Severinov K, Darst SA.\n",
      "  TITLE     Crystal structure of Thermus aquaticus core RNA polymerase at 3.3 A resolution.\n",
      "  JOURNAL   Cell 98:811-824 (1999)\n",
      "            DOI:10.1016/S0092-8674(00)81515-9\n",
      "REFERENCE   PMID:10784442\n",
      "  AUTHORS   Cramer P, Bushnell DA, Fu J, Gnatt AL, Maier-Davis B, Thompson NE, Burgess RR, Edwards AM, David PR, Kornberg RD.\n",
      "  TITLE     Architecture of RNA polymerase II and implications for the transcription mechanism.\n",
      "  JOURNAL   Science 288:640-649 (2000)\n",
      "            DOI:10.1126/science.288.5466.640\n",
      "REFERENCE   PMID:19880312\n",
      "  AUTHORS   Hirata A, Murakami KS\n",
      "  TITLE     Archaeal RNA polymerase.\n",
      "  JOURNAL   Curr Opin Struct Biol 19:724-31 (2009)\n",
      "            DOI:10.1016/j.sbi.2009.10.006\n",
      "REFERENCE   PMID:18573085\n",
      "  AUTHORS   Cramer P, Armache KJ, Baumli S, Benkert S, Brueckner F, Buchen C, Damsma GE, Dengl S, Geiger SR, Jasiak AJ, Jawhari A, Jennebach S, Kamenski T, Kettenberger H, Kuhn CD, Lehmann E, Leike K, Sydow JF, Vannini A\n",
      "  TITLE     Structure of eukaryotic RNA polymerases.\n",
      "  JOURNAL   Annu Rev Biophys 37:337-52 (2008)\n",
      "            DOI:10.1146/annurev.biophys.37.032807.130008\n",
      "KO_PATHWAY  ko03020\n",
      "///\n",
      "path:pae03030\n",
      "ENTRY       pae03030                    Pathway\n",
      "NAME        DNA replication - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION A complex network of interacting proteins and enzymes is required for DNA replication. Generally, DNA replication follows a multistep enzymatic pathway. At the DNA replication fork, a DNA helicase (DnaB or MCM complex) precedes the DNA synthetic machinery and unwinds the duplex parental DNA in cooperation with the SSB or RPA. On the leading strand, replication occurs continuously in a 5 to 3 direction, whereas on the lagging strand, DNA replication occurs discontinuously by synthesis and joining of short Okazaki fragments. In prokaryotes, the leading strand replication apparatus consists of a DNA polymerase (pol III core), a sliding clamp (beta), and a clamp loader (gamma delta complex). The DNA primase (DnaG) is needed to form RNA primers. Normally, during replication of the lagging-strand DNA template, an RNA primer is removed either by an RNase H or by the 5 to 3 exonuclease activity of DNA pol I, and the DNA ligase joins the Okazaki fragments. In eukaryotes, three DNA polymerases (alpha, delta, and epsilon) have been identified. DNA primase forms a permanent complex with DNA polymerase alpha. PCNA and RFC function as a clamp and a clamp loader. FEN 1 and RNase H1 remove the RNA from the Okazaki fragments and DNA ligase I joins the DNA.\n",
      "CLASS       Genetic Information Processing; Replication and repair\n",
      "PATHWAY_MAP pae03030  DNA replication\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA1816  dnaQ; DNA polymerase III subunit epsilon [KO:K02342] [EC:2.7.7.7]\n",
      "line in gene PA1816  dnaQ; DNA polymerase III subunit epsilon [KO:K02342] [EC:2.7.7.7]\n",
      "            PA3232  DNA polymerase III subunit epsilon [KO:K02342] [EC:2.7.7.7]\n",
      "line in gene PA3232  DNA polymerase III subunit epsilon [KO:K02342] [EC:2.7.7.7]\n",
      "            PA3640  dnaE; DNA polymerase III subunit alpha [KO:K02337] [EC:2.7.7.7]\n",
      "line in gene PA3640  dnaE; DNA polymerase III subunit alpha [KO:K02337] [EC:2.7.7.7]\n",
      "            PA1532  dnaX; DNA polymerase III subunits gamma/tau [KO:K02343] [EC:2.7.7.7]\n",
      "line in gene PA1532  dnaX; DNA polymerase III subunits gamma/tau [KO:K02343] [EC:2.7.7.7]\n",
      "            PA2961  holB; DNA polymerase III subunit delta' [KO:K02341] [EC:2.7.7.7]\n",
      "line in gene PA2961  holB; DNA polymerase III subunit delta' [KO:K02341] [EC:2.7.7.7]\n",
      "            PA3989  holA; DNA polymerase III subunit delta [KO:K02340] [EC:2.7.7.7]\n",
      "line in gene PA3989  holA; DNA polymerase III subunit delta [KO:K02340] [EC:2.7.7.7]\n",
      "            PA3832  holC; DNA polymerase III subunit chi [KO:K02339] [EC:2.7.7.7]\n",
      "line in gene PA3832  holC; DNA polymerase III subunit chi [KO:K02339] [EC:2.7.7.7]\n",
      "            PA0002  dnaN; DNA polymerase III subunit beta [KO:K02338] [EC:2.7.7.7]\n",
      "line in gene PA0002  dnaN; DNA polymerase III subunit beta [KO:K02338] [EC:2.7.7.7]\n",
      "            PA4931  dnaB; replicative DNA helicase [KO:K02314] [EC:5.6.2.3]\n",
      "line in gene PA4931  dnaB; replicative DNA helicase [KO:K02314] [EC:5.6.2.3]\n",
      "            PA0577  dnaG; DNA primase [KO:K02316] [EC:2.7.7.101]\n",
      "line in gene PA0577  dnaG; DNA primase [KO:K02316] [EC:2.7.7.101]\n",
      "            PA4232  ssb; single-stranded DNA-binding protein [KO:K03111]\n",
      "line in gene PA4232  ssb; single-stranded DNA-binding protein [KO:K03111]\n",
      "            PA1815  rnhA; ribonuclease HI [KO:K03469] [EC:3.1.26.4]\n",
      "line in gene PA1815  rnhA; ribonuclease HI [KO:K03469] [EC:3.1.26.4]\n",
      "            PA3642  rnhB; ribonuclease HII [KO:K03470] [EC:3.1.26.4]\n",
      "line in gene PA3642  rnhB; ribonuclease HII [KO:K03470] [EC:3.1.26.4]\n",
      "            PA5493  polA; DNA polymerase I [KO:K02335] [EC:2.7.7.7]\n",
      "line in gene PA5493  polA; DNA polymerase I [KO:K02335] [EC:2.7.7.7]\n",
      "            PA1529  lig; NAD-dependent DNA ligase LigA [KO:K01972] [EC:6.5.1.2]\n",
      "line in gene PA1529  lig; NAD-dependent DNA ligase LigA [KO:K01972] [EC:6.5.1.2]\n",
      "REFERENCE   PMID:8087839\n",
      "  AUTHORS   Stillman B.\n",
      "  TITLE     Smart machines at the DNA replication fork.\n",
      "  JOURNAL   Cell 78:725-8 (1994)\n",
      "            DOI:10.1016/S0092-8674(94)90362-X\n",
      "REFERENCE   PMID:9759502\n",
      "  AUTHORS   Waga S, Stillman B.\n",
      "  TITLE     The DNA replication fork in eukaryotic cells.\n",
      "  JOURNAL   Annu Rev Biochem 67:721-51 (1998)\n",
      "            DOI:10.1146/annurev.biochem.67.1.721\n",
      "KO_PATHWAY  ko03030\n",
      "///\n",
      "path:pae03060\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae03060                    Pathway\n",
      "NAME        Protein export - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION The protein export is the active transport of proteins from the cytoplasm to the exterior of the cell, or to the periplasmic compartment in Gram-negative bacteria. The sec dependent pathway is the general protein export system that transports newly synthesized proteins into or across the cell membrane. The translocation channel is formed from a conserved trimeric membrane protein complex, called the Sec61/SecY complex. The twin-arginine translocation (Tat) pathway is another protein transport system that transports folded proteins in bacteria, archaea, and chloroplasts. Many Tat systems comprise three functionally different membrane proteins, TatA, TatB, and TatC, but TatA and TatE seem to have overlapping functions, with TatA having by far the more important role.\n",
      "CLASS       Genetic Information Processing; Folding, sorting and degradation\n",
      "PATHWAY_MAP pae03060  Protein export\n",
      "DBLINKS     GO: 0043952 0043953\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4243  secY; preprotein translocase subunit SecY [KO:K03076]\n",
      "line in gene PA4243  secY; preprotein translocase subunit SecY [KO:K03076]\n",
      "            PA4276  secE; preprotein translocase subunit SecE [KO:K03073]\n",
      "line in gene PA4276  secE; preprotein translocase subunit SecE [KO:K03073]\n",
      "            PA4747  secG; preprotein translocase subunit SecG [KO:K03075]\n",
      "line in gene PA4747  secG; preprotein translocase subunit SecG [KO:K03075]\n",
      "            PA3821  secD; preprotein translocase subunit SecD [KO:K03072]\n",
      "line in gene PA3821  secD; preprotein translocase subunit SecD [KO:K03072]\n",
      "            PA3820  secF; preprotein translocase subunit SecF [KO:K03074]\n",
      "line in gene PA3820  secF; preprotein translocase subunit SecF [KO:K03074]\n",
      "            PA3822  preprotein translocase subunit YajC [KO:K03210]\n",
      "line in gene PA3822  preprotein translocase subunit YajC [KO:K03210]\n",
      "            PA5568  inner membrane protein translocase subunit YidC [KO:K03217]\n",
      "line in gene PA5568  inner membrane protein translocase subunit YidC [KO:K03217]\n",
      "            PA4403  secA; preprotein translocase subunit SecA [KO:K03070] [EC:7.4.2.8]\n",
      "line in gene PA4403  secA; preprotein translocase subunit SecA [KO:K03070] [EC:7.4.2.8]\n",
      "            PA5128  secB; preprotein translocase subunit SecB [KO:K03071]\n",
      "line in gene PA5128  secB; preprotein translocase subunit SecB [KO:K03071]\n",
      "            PA3746  ffh; signal recognition particle protein Ffh [KO:K03106] [EC:3.6.5.4]\n",
      "line in gene PA3746  ffh; signal recognition particle protein Ffh [KO:K03106] [EC:3.6.5.4]\n",
      "            PA1530.1  ffs; 4.5S ribosomal RNA [KO:K01983]\n",
      "line in gene PA1530.1  ffs; 4.5S ribosomal RNA [KO:K01983]\n",
      "            PA0373  ftsY; signal recognition particle receptor FtsY [KO:K03110]\n",
      "line in gene PA0373  ftsY; signal recognition particle receptor FtsY [KO:K03110]\n",
      "            PA5068  tatA; twin-arginine translocation protein TatA [KO:K03116]\n",
      "line in gene PA5068  tatA; twin-arginine translocation protein TatA [KO:K03116]\n",
      "            PA5069  tatB; sec-independent translocase [KO:K03117]\n",
      "line in gene PA5069  tatB; sec-independent translocase [KO:K03117]\n",
      "            PA5070  tatC; transporter TatC [KO:K03118]\n",
      "line in gene PA5070  tatC; transporter TatC [KO:K03118]\n",
      "            PA0768  lepB; signal peptidase I [KO:K03100] [EC:3.4.21.89]\n",
      "line in gene PA0768  lepB; signal peptidase I [KO:K03100] [EC:3.4.21.89]\n",
      "            PA1303  signal peptidase [KO:K03100] [EC:3.4.21.89]\n",
      "line in gene PA1303  signal peptidase [KO:K03100] [EC:3.4.21.89]\n",
      "            PA4559  lspA; lipoprotein signal peptidase [KO:K03101] [EC:3.4.23.36]\n",
      "line in gene PA4559  lspA; lipoprotein signal peptidase [KO:K03101] [EC:3.4.23.36]\n",
      "REFERENCE   PMID:10087917\n",
      "  AUTHORS   Dalbey RE, Robinson C.\n",
      "  TITLE     Protein translocation into and across the bacterial plasma membrane and the plant thylakoid membrane.\n",
      "  JOURNAL   Trends Biochem Sci 24:17-22 (1999)\n",
      "            DOI:10.1016/S0968-0004(98)01333-4\n",
      "REFERENCE   PMID:10652088\n",
      "  AUTHORS   Berks BC, Sargent F, Palmer T.\n",
      "  TITLE     The Tat protein export pathway.\n",
      "  JOURNAL   Mol Microbiol 35:260-74 (2000)\n",
      "            DOI:10.1046/j.1365-2958.2000.01719.x\n",
      "REFERENCE   PMID:10747959\n",
      "  AUTHORS   Wexler M, Sargent F, Jack RL, Stanley NR, Bogsch EG, Robinson C, Berks BC, Palmer T.\n",
      "  TITLE     TatD is a cytoplasmic protein with DNase activity. No requirement for TatD family proteins in sec-independent protein export.\n",
      "  JOURNAL   J Biol Chem 275:16717-22 (2000)\n",
      "            DOI:10.1074/jbc.M000800200\n",
      "REFERENCE   PMID:11007775\n",
      "  AUTHORS   Jongbloed JD, Martin U, Antelmann H, Hecker M, Tjalsma H, Venema G, Bron S, van Dijl JM, Muller J.\n",
      "  TITLE     TatC is a specificity determinant for protein secretion via the twin-arginine translocation pathway.\n",
      "  JOURNAL   J Biol Chem 275:41350-7 (2000)\n",
      "            DOI:10.1074/jbc.M004887200\n",
      "REFERENCE   PMID:18046402\n",
      "  AUTHORS   Rapoport TA\n",
      "  TITLE     Protein translocation across the eukaryotic endoplasmic reticulum and bacterial plasma membranes.\n",
      "  JOURNAL   Nature 450:663-9 (2007)\n",
      "            DOI:10.1038/nature06384\n",
      "REFERENCE   PMID:17368028\n",
      "  AUTHORS   Flower AM\n",
      "  TITLE     The SecY translocation complex: convergence of genetics and structure.\n",
      "  JOURNAL   Trends Microbiol 15:203-10 (2007)\n",
      "            DOI:10.1016/j.tim.2007.03.001\n",
      "REFERENCE   PMID:16212506\n",
      "  AUTHORS   Osborne AR, Rapoport TA, van den Berg B\n",
      "  TITLE     Protein translocation by the Sec61/SecY channel.\n",
      "  JOURNAL   Annu Rev Cell Dev Biol 21:529-50 (2005)\n",
      "            DOI:10.1146/annurev.cellbio.21.012704.133214\n",
      "REFERENCE   PMID:12475201\n",
      "  AUTHORS   Paetzel M, Karla A, Strynadka NC, Dalbey RE\n",
      "  TITLE     Signal peptidases.\n",
      "  JOURNAL   Chem Rev 102:4549-80 (2002)\n",
      "            DOI:10.1021/cr010166y\n",
      "REFERENCE   PMID:10924740\n",
      "  AUTHORS   Paetzel M, Dalbey RE, Strynadka NC\n",
      "  TITLE     The structure and mechanism of bacterial type I signal peptidases. A novel antibiotic target.\n",
      "  JOURNAL   Pharmacol Ther 87:27-49 (2000)\n",
      "            DOI:10.1016/S0163-7258(00)00064-4\n",
      "REFERENCE   PMID:17259602\n",
      "  AUTHORS   Ng SY, Chaban B, VanDyke DJ, Jarrell KF\n",
      "  TITLE     Archaeal signal peptidases.\n",
      "  JOURNAL   Microbiology 153:305-14 (2007)\n",
      "            DOI:10.1099/mic.0.2006/003087-0\n",
      "REFERENCE   PMID:19299134\n",
      "  AUTHORS   Desvaux M, Hebraud M, Talon R, Henderson IR\n",
      "  TITLE     Secretion and subcellular localizations of bacterial proteins: a semantic awareness issue.\n",
      "  JOURNAL   Trends Microbiol 17:139-45 (2009)\n",
      "            DOI:10.1016/j.tim.2009.01.004\n",
      "REFERENCE   PMID:15063851\n",
      "  AUTHORS   Nakatogawa H, Murakami A, Ito K\n",
      "  TITLE     Control of SecA and SecM translation by protein secretion.\n",
      "  JOURNAL   Curr Opin Microbiol 7:145-50 (2004)\n",
      "            DOI:10.1016/j.mib.2004.01.001\n",
      "KO_PATHWAY  ko03060\n",
      "///\n",
      "path:pae03070\n",
      "ENTRY       pae03070                    Pathway\n",
      "NAME        Bacterial secretion system - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Gram-negative bacteria secrete a wide range of proteins whose functions include biogenesis of organelles, such as pilli and flagella, nutrient acquisition, virulence, and efflux of drugs and other toxins. Six distinct secretion systems have been shown to mediate protein export through the inner and outer membranes of Gram-negative bacteria. These pathways are highly conserved throughout the Gram-negative bacterial species. In Gram-positive bacteria, secreted proteins are commonly translocated across the single membrane by the Sec pathway or the two-arginine (Tat) pathway.\n",
      "CLASS       Environmental Information Processing; Membrane transport\n",
      "PATHWAY_MAP pae03070  Bacterial secretion system\n",
      "DBLINKS     GO: 0030253 0015628 0030254 0030255 0046819 0033103 0044098 0043952 0043953\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA4974  hypothetical protein [KO:K12340]\n",
      "line in gene PA4974  hypothetical protein [KO:K12340]\n",
      "            PA4144  hypothetical protein [KO:K12340]\n",
      "line in gene PA4144  hypothetical protein [KO:K12340]\n",
      "            PA4592  hypothetical protein [KO:K12340]\n",
      "line in gene PA4592  hypothetical protein [KO:K12340]\n",
      "            PA1719  pscF; type III export protein PscF [KO:K03221]\n",
      "line in gene PA1719  pscF; type III export protein PscF [KO:K03221]\n",
      "            PA1696  pscO; translocation protein in type III secretion [KO:K04056]\n",
      "line in gene PA1696  pscO; translocation protein in type III secretion [KO:K04056]\n",
      "            PA1695  pscP; translocation protein in type III secretion [KO:K04057]\n",
      "line in gene PA1695  pscP; translocation protein in type III secretion [KO:K04057]\n",
      "            PA1701  hypothetical protein [KO:K04059]\n",
      "line in gene PA1701  hypothetical protein [KO:K04059]\n",
      "            PA1716  pscC; type III secretion outer membrane protein PscC [KO:K03219]\n",
      "line in gene PA1716  pscC; type III secretion outer membrane protein PscC [KO:K03219]\n",
      "            PA1698  popN; type III secretion outer membrane protein PopN [KO:K04058]\n",
      "line in gene PA1698  popN; type III secretion outer membrane protein PopN [KO:K04058]\n",
      "            PA1723  pscJ; type III export protein PscJ [KO:K03222]\n",
      "line in gene PA1723  pscJ; type III export protein PscJ [KO:K03222]\n",
      "            PA1693  pscR; type III secretion system protein [KO:K03226]\n",
      "line in gene PA1693  pscR; type III secretion system protein [KO:K03226]\n",
      "            PA1692  translocation protein in type III secretion [KO:K03227]\n",
      "line in gene PA1692  translocation protein in type III secretion [KO:K03227]\n",
      "            PA1691  pscT; translocation protein in type III secretion [KO:K03228]\n",
      "line in gene PA1691  pscT; translocation protein in type III secretion [KO:K03228]\n",
      "            PA1690  pscU; translocation protein in type III secretion [KO:K03229]\n",
      "line in gene PA1690  pscU; translocation protein in type III secretion [KO:K03229]\n",
      "            PA1703  pcrD; type III secretory apparatus protein PcrD [KO:K03230]\n",
      "line in gene PA1703  pcrD; type III secretory apparatus protein PcrD [KO:K03230]\n",
      "            PA1697  type III secretion system ATPase [KO:K03224] [EC:7.4.2.8]\n",
      "line in gene PA1697  type III secretion system ATPase [KO:K03224] [EC:7.4.2.8]\n",
      "            PA1694  pscQ; type III secretion system protein [KO:K03225]\n",
      "line in gene PA1694  pscQ; type III secretion system protein [KO:K03225]\n",
      "            PA1725  pscL; type III secretion system protein [KO:K03223]\n",
      "line in gene PA1725  pscL; type III secretion system protein [KO:K03223]\n",
      "            PA1868  xqhA; secretion protein XqhA [KO:K02453]\n",
      "line in gene PA1868  xqhA; secretion protein XqhA [KO:K02453]\n",
      "            PA3105  xcpQ; type II secretion system protein D [KO:K02453]\n",
      "line in gene PA3105  xcpQ; type II secretion system protein D [KO:K02453]\n",
      "            PA1382  type II secretion system protein [KO:K02453]\n",
      "line in gene PA1382  type II secretion system protein [KO:K02453]\n",
      "            PA0685  type II secretion system protein [KO:K02453]\n",
      "line in gene PA0685  type II secretion system protein [KO:K02453]\n",
      "            PA1867  xphA; type VI pilus biosynthesis protein [KO:K02452]\n",
      "line in gene PA1867  xphA; type VI pilus biosynthesis protein [KO:K02452]\n",
      "            PA3104  xcpP; type II secretion system protein N [KO:K02452]\n",
      "line in gene PA3104  xcpP; type II secretion system protein N [KO:K02452]\n",
      "            PA0679  hypothetical protein [KO:K02452]\n",
      "line in gene PA0679  hypothetical protein [KO:K02452]\n",
      "            PA0687  type II secretion system protein [KO:K02455]\n",
      "line in gene PA0687  type II secretion system protein [KO:K02455]\n",
      "            PA2676  type II secretion system protein [KO:K02455]\n",
      "line in gene PA2676  type II secretion system protein [KO:K02455]\n",
      "            PA3102  xcpS; type II secretion system protein F [KO:K02455]\n",
      "line in gene PA3102  xcpS; type II secretion system protein F [KO:K02455]\n",
      "            PA0681  HxcT pseudopilin [KO:K02456]\n",
      "line in gene PA0681  HxcT pseudopilin [KO:K02456]\n",
      "            PA2675  type II secretion system protein [KO:K02456]\n",
      "line in gene PA2675  type II secretion system protein [KO:K02456]\n",
      "            PA3101  xcpT; type II secretion system protein G [KO:K02456]\n",
      "line in gene PA3101  xcpT; type II secretion system protein G [KO:K02456]\n",
      "            PA2674  type II secretion system protein [KO:K02456]\n",
      "line in gene PA2674  type II secretion system protein [KO:K02456]\n",
      "            PA3100  xcpU; type II secretion system protein H [KO:K02457]\n",
      "line in gene PA3100  xcpU; type II secretion system protein H [KO:K02457]\n",
      "            PA0678  HxcU pseudopilin [KO:K02457]\n",
      "line in gene PA0678  HxcU pseudopilin [KO:K02457]\n",
      "            PA0680  HxcV pseudopilin [KO:K02458]\n",
      "line in gene PA0680  HxcV pseudopilin [KO:K02458]\n",
      "            PA3099  xcpV; type II secretion system protein I [KO:K02458]\n",
      "line in gene PA3099  xcpV; type II secretion system protein I [KO:K02458]\n",
      "            PA2673  type II secretion system protein [KO:K02458]\n",
      "line in gene PA2673  type II secretion system protein [KO:K02458]\n",
      "            PA3098  xcpW; type II secretion system protein J [KO:K02459]\n",
      "line in gene PA3098  xcpW; type II secretion system protein J [KO:K02459]\n",
      "            PA0677  HxcW pseudopilin [KO:K02459]\n",
      "line in gene PA0677  HxcW pseudopilin [KO:K02459]\n",
      "            PA2672  type II secretion system protein [KO:K02459]\n",
      "line in gene PA2672  type II secretion system protein [KO:K02459]\n",
      "            PA0682  HxcX atypical pseudopilin [KO:K02460]\n",
      "line in gene PA0682  HxcX atypical pseudopilin [KO:K02460]\n",
      "            PA3097  xcpX; type II secretion system protein K [KO:K02460]\n",
      "line in gene PA3097  xcpX; type II secretion system protein K [KO:K02460]\n",
      "            PA0683  type II secretion system protein [KO:K02461]\n",
      "line in gene PA0683  type II secretion system protein [KO:K02461]\n",
      "            PA3096  xcpY; type II secretion system protein L [KO:K02461]\n",
      "line in gene PA3096  xcpY; type II secretion system protein L [KO:K02461]\n",
      "            PA0684  type II secretion system protein [KO:K02462]\n",
      "line in gene PA0684  type II secretion system protein [KO:K02462]\n",
      "            PA3095  xcpZ; type II secretion system protein M [KO:K02462]\n",
      "line in gene PA3095  xcpZ; type II secretion system protein M [KO:K02462]\n",
      "            PA2677  type II secretion protein [KO:K02454] [EC:7.4.2.8]\n",
      "line in gene PA2677  type II secretion protein [KO:K02454] [EC:7.4.2.8]\n",
      "            PA3103  xcpR; type II secretion system protein E [KO:K02454] [EC:7.4.2.8]\n",
      "line in gene PA3103  xcpR; type II secretion system protein E [KO:K02454] [EC:7.4.2.8]\n",
      "            PA5210  secretion pathway ATPase [KO:K02454] [EC:7.4.2.8]\n",
      "line in gene PA5210  secretion pathway ATPase [KO:K02454] [EC:7.4.2.8]\n",
      "            PA0686  type II secretion system protein HxcR [KO:K02454] [EC:7.4.2.8]\n",
      "line in gene PA0686  type II secretion system protein HxcR [KO:K02454] [EC:7.4.2.8]\n",
      "            PA3821  secD; preprotein translocase subunit SecD [KO:K03072]\n",
      "line in gene PA3821  secD; preprotein translocase subunit SecD [KO:K03072]\n",
      "            PA3820  secF; preprotein translocase subunit SecF [KO:K03074]\n",
      "line in gene PA3820  secF; preprotein translocase subunit SecF [KO:K03074]\n",
      "            PA4276  secE; preprotein translocase subunit SecE [KO:K03073]\n",
      "line in gene PA4276  secE; preprotein translocase subunit SecE [KO:K03073]\n",
      "            PA4747  secG; preprotein translocase subunit SecG [KO:K03075]\n",
      "line in gene PA4747  secG; preprotein translocase subunit SecG [KO:K03075]\n",
      "            PA4243  secY; preprotein translocase subunit SecY [KO:K03076]\n",
      "line in gene PA4243  secY; preprotein translocase subunit SecY [KO:K03076]\n",
      "            PA3822  preprotein translocase subunit YajC [KO:K03210]\n",
      "line in gene PA3822  preprotein translocase subunit YajC [KO:K03210]\n",
      "            PA5568  inner membrane protein translocase subunit YidC [KO:K03217]\n",
      "line in gene PA5568  inner membrane protein translocase subunit YidC [KO:K03217]\n",
      "            PA4403  secA; preprotein translocase subunit SecA [KO:K03070] [EC:7.4.2.8]\n",
      "line in gene PA4403  secA; preprotein translocase subunit SecA [KO:K03070] [EC:7.4.2.8]\n",
      "            PA0373  ftsY; signal recognition particle receptor FtsY [KO:K03110]\n",
      "line in gene PA0373  ftsY; signal recognition particle receptor FtsY [KO:K03110]\n",
      "            PA5128  secB; preprotein translocase subunit SecB [KO:K03071]\n",
      "line in gene PA5128  secB; preprotein translocase subunit SecB [KO:K03071]\n",
      "            PA3746  ffh; signal recognition particle protein Ffh [KO:K03106] [EC:3.6.5.4]\n",
      "line in gene PA3746  ffh; signal recognition particle protein Ffh [KO:K03106] [EC:3.6.5.4]\n",
      "            PA5068  tatA; twin-arginine translocation protein TatA [KO:K03116]\n",
      "line in gene PA5068  tatA; twin-arginine translocation protein TatA [KO:K03116]\n",
      "            PA5069  tatB; sec-independent translocase [KO:K03117]\n",
      "line in gene PA5069  tatB; sec-independent translocase [KO:K03117]\n",
      "            PA5070  tatC; transporter TatC [KO:K03118]\n",
      "line in gene PA5070  tatC; transporter TatC [KO:K03118]\n",
      "            PA0262  hypothetical protein [KO:K11904]\n",
      "line in gene PA0262  hypothetical protein [KO:K11904]\n",
      "            PA3294  hypothetical protein [KO:K11904]\n",
      "line in gene PA3294  hypothetical protein [KO:K11904]\n",
      "            PA3486  hypothetical protein [KO:K11904]\n",
      "line in gene PA3486  hypothetical protein [KO:K11904]\n",
      "            PA1511  hypothetical protein [KO:K11904]\n",
      "line in gene PA1511  hypothetical protein [KO:K11904]\n",
      "            PA5266  hypothetical protein [KO:K11904]\n",
      "line in gene PA5266  hypothetical protein [KO:K11904]\n",
      "            PA2685  hypothetical protein [KO:K11904]\n",
      "line in gene PA2685  hypothetical protein [KO:K11904]\n",
      "            PA0095  hypothetical protein [KO:K11904]\n",
      "line in gene PA0095  hypothetical protein [KO:K11904]\n",
      "            PA0091  vgrG1; type VI secretion system protein VgrG [KO:K11904]\n",
      "line in gene PA0091  vgrG1; type VI secretion system protein VgrG [KO:K11904]\n",
      "            PA2373  hypothetical protein [KO:K11904]\n",
      "line in gene PA2373  hypothetical protein [KO:K11904]\n",
      "            PA5090  hypothetical protein [KO:K11904]\n",
      "line in gene PA5090  hypothetical protein [KO:K11904]\n",
      "            PA1512  hcpA; secreted protein Hcp [KO:K11903]\n",
      "line in gene PA1512  hcpA; secreted protein Hcp [KO:K11903]\n",
      "            PA2367  hypothetical protein [KO:K11903]\n",
      "line in gene PA2367  hypothetical protein [KO:K11903]\n",
      "            PA0263  hcpC; secreted protein Hcp [KO:K11903]\n",
      "line in gene PA0263  hcpC; secreted protein Hcp [KO:K11903]\n",
      "            PA5267  hcpB; secreted protein Hcp [KO:K11903]\n",
      "line in gene PA5267  hcpB; secreted protein Hcp [KO:K11903]\n",
      "            PA0085  hcp1; protein secretion apparatus assembly protein [KO:K11903]\n",
      "line in gene PA0085  hcp1; protein secretion apparatus assembly protein [KO:K11903]\n",
      "            PA0080  hypothetical protein [KO:K11906]\n",
      "line in gene PA0080  hypothetical protein [KO:K11906]\n",
      "            PA1666  hypothetical protein [KO:K11906]\n",
      "line in gene PA1666  hypothetical protein [KO:K11906]\n",
      "            PA1669  hypothetical protein [KO:K11891]\n",
      "line in gene PA1669  hypothetical protein [KO:K11891]\n",
      "            PA2361  hypothetical protein [KO:K11891]\n",
      "line in gene PA2361  hypothetical protein [KO:K11891]\n",
      "            PA0077  icmF1; type VI secretion protein IcmF [KO:K11891]\n",
      "line in gene PA0077  icmF1; type VI secretion protein IcmF [KO:K11891]\n",
      "            PA1668  hypothetical protein [KO:K11892]\n",
      "line in gene PA1668  hypothetical protein [KO:K11892]\n",
      "            PA2362  hypothetical protein [KO:K11892]\n",
      "line in gene PA2362  hypothetical protein [KO:K11892]\n",
      "            PA0078  hypothetical protein [KO:K11892]\n",
      "line in gene PA0078  hypothetical protein [KO:K11892]\n",
      "            PA1662  ClpA/B-type protease [KO:K11907]\n",
      "line in gene PA1662  ClpA/B-type protease [KO:K11907]\n",
      "            PA0090  clpV1; secretion protein ClpV1 [KO:K11907]\n",
      "line in gene PA0090  clpV1; secretion protein ClpV1 [KO:K11907]\n",
      "            PA2371  ClpA/B-type protease [KO:K11907]\n",
      "line in gene PA2371  ClpA/B-type protease [KO:K11907]\n",
      "            PA0074  ppkA; serine/threonine protein kinase PpkA [KO:K11912] [EC:2.7.11.1]\n",
      "line in gene PA0074  ppkA; serine/threonine protein kinase PpkA [KO:K11912] [EC:2.7.11.1]\n",
      "            PA0081  fha1; Fha domain-containing protein [KO:K11913]\n",
      "line in gene PA0081  fha1; Fha domain-containing protein [KO:K11913]\n",
      "            PA1670  stp1; serine/threonine phosphoprotein phosphatase Stp1 [KO:K11915] [EC:3.1.3.16]\n",
      "line in gene PA1670  stp1; serine/threonine phosphoprotein phosphatase Stp1 [KO:K11915] [EC:3.1.3.16]\n",
      "            PA0075  pppA; serine/threonine phosphatase [KO:K11915] [EC:3.1.3.16]\n",
      "line in gene PA0075  pppA; serine/threonine phosphatase [KO:K11915] [EC:3.1.3.16]\n",
      "COMPOUND    C00002  ATP\n",
      "REFERENCE   PMID:11173102\n",
      "  AUTHORS   Kim JF.\n",
      "  TITLE     Revisiting the chlamydial type III protein secretion system: clues to the origin of type III protein secretion.\n",
      "  JOURNAL   Trends Genet 17:65-9 (2001)\n",
      "            DOI:10.1016/S0168-9525(00)02175-2\n",
      "REFERENCE   PMID:11309112\n",
      "  AUTHORS   Plano GV, Day JB, Ferracci F.\n",
      "  TITLE     Type III export: new uses for an old pathway.\n",
      "  JOURNAL   Mol Microbiol 40:284-93 (2001)\n",
      "            DOI:10.1046/j.1365-2958.2001.02354.x\n",
      "REFERENCE   PMID:11413015\n",
      "  AUTHORS   Gauthier A, Finlay BB.\n",
      "  TITLE     Bacterial pathogenesis: the answer to virulence is in the pore.\n",
      "  JOURNAL   Curr Biol 11:R264-7 (2001)\n",
      "            DOI:10.1016/S0960-9822(01)00134-8\n",
      "REFERENCE   PMID:10047550\n",
      "  AUTHORS   Burns DL.\n",
      "  TITLE     Biochemistry of type IV secretion.\n",
      "  JOURNAL   Curr Opin Microbiol 2:25-9 (1999)\n",
      "            DOI:10.1016/S1369-5274(99)80004-6\n",
      "REFERENCE   PMID:10920394\n",
      "  AUTHORS   Christie PJ, Vogel JP.\n",
      "  TITLE     Bacterial type IV secretion: conjugation systems adapted to deliver effector molecules to host cells.\n",
      "  JOURNAL   Trends Microbiol 8:354-60 (2000)\n",
      "            DOI:10.1016/S0966-842X(00)01792-3\n",
      "REFERENCE   PMID:11309113\n",
      "  AUTHORS   Christie PJ.\n",
      "  TITLE     Type IV secretion: intercellular transfer of macromolecules by systems ancestrally related to conjugation machines.\n",
      "  JOURNAL   Mol Microbiol 40:294-305 (2001)\n",
      "            DOI:10.1046/j.1365-2958.2001.02302.x\n",
      "REFERENCE   PMID:10481178\n",
      "  AUTHORS   Nunn D.\n",
      "  TITLE     Bacterial type II protein export and pilus biogenesis: more than just homologies?\n",
      "  JOURNAL   Trends Cell Biol 9:402-8 (1999)\n",
      "            DOI:10.1016/S0962-8924(99)01634-7\n",
      "REFERENCE   PMID:11309111\n",
      "  AUTHORS   Sandkvist M.\n",
      "  TITLE     Biology of type II secretion.\n",
      "  JOURNAL   Mol Microbiol 40:271-83 (2001)\n",
      "            DOI:10.1046/j.1365-2958.2001.02403.x\n",
      "REFERENCE   PMID:11349009\n",
      "  AUTHORS   Sandkvist M.\n",
      "  TITLE     Type II secretion and pathogenesis.\n",
      "  JOURNAL   Infect Immun 69:3523-35 (2001)\n",
      "            DOI:10.1128/IAI.69.6.3523-3535.2001\n",
      "REFERENCE   PMID:15546664\n",
      "  AUTHORS   Delepelaire P\n",
      "  TITLE     Type I secretion in gram-negative bacteria.\n",
      "  JOURNAL   Biochim Biophys Acta 1694:149-61 (2004)\n",
      "            DOI:10.1016/j.bbamcr.2004.05.001\n",
      "REFERENCE   PMID:18406342\n",
      "  AUTHORS   Hazes B, Frost L\n",
      "  TITLE     Towards a systems biology approach to study type II/IV secretion systems.\n",
      "  JOURNAL   Biochim Biophys Acta 1778:1839-50 (2008)\n",
      "            DOI:10.1016/j.bbamem.2008.03.011\n",
      "REFERENCE   PMID:16448494\n",
      "  AUTHORS   Johnson TL, Abendroth J, Hol WG, Sandkvist M\n",
      "  TITLE     Type II secretion: from structure to function.\n",
      "  JOURNAL   FEMS Microbiol Lett 255:175-86 (2006)\n",
      "            DOI:10.1111/j.1574-6968.2006.00102.x\n",
      "REFERENCE   PMID:15546665\n",
      "  AUTHORS   Filloux A\n",
      "  TITLE     The underlying mechanisms of type II protein secretion.\n",
      "  JOURNAL   Biochim Biophys Acta 1694:163-79 (2004)\n",
      "            DOI:10.1016/j.bbamcr.2004.05.003\n",
      "REFERENCE   PMID:19396170\n",
      "  AUTHORS   Spreter T, Yip CK, Sanowar S, Andre I, Kimbrough TG, Vuckovic M, Pfuetzner RA, Deng W, Yu AC, Finlay BB, Baker D, Miller SI, Strynadka NC\n",
      "  TITLE     A conserved structural motif mediates formation of the periplasmic rings in the type III secretion system.\n",
      "  JOURNAL   Nat Struct Mol Biol 16:468-76 (2009)\n",
      "            DOI:10.1038/nsmb.1603\n",
      "REFERENCE   PMID:15590783\n",
      "  AUTHORS   Ghosh P\n",
      "  TITLE     Process of protein transport by the type III secretion system.\n",
      "  JOURNAL   Microbiol Mol Biol Rev 68:771-95 (2004)\n",
      "            DOI:10.1128/MMBR.68.4.771-795.2004\n",
      "REFERENCE   PMID:12360191\n",
      "  AUTHORS   Cornelis GR\n",
      "  TITLE     The Yersinia Ysc-Yop 'type III' weaponry.\n",
      "  JOURNAL   Nat Rev Mol Cell Biol 3:742-52 (2002)\n",
      "            DOI:10.1038/nrm932\n",
      "REFERENCE   PMID:17215876\n",
      "  AUTHORS   Baron C\n",
      "  TITLE     VirB8: a conserved type IV secretion system assembly factor and drug target.\n",
      "  JOURNAL   Biochem Cell Biol 84:890-9 (2006)\n",
      "            DOI:10.1139/o06-148\n",
      "REFERENCE   PMID:17943123\n",
      "  AUTHORS   Kwok T, Zabler D, Urman S, Rohde M, Hartig R, Wessler S, Misselwitz R, Berger J, Sewald N, Konig W, Backert S\n",
      "  TITLE     Helicobacter exploits integrin for type IV secretion and kinase activation.\n",
      "  JOURNAL   Nature 449:862-6 (2007)\n",
      "            DOI:10.1038/nature06187\n",
      "REFERENCE   PMID:16092524\n",
      "  AUTHORS   Christie PJ, Cascales E\n",
      "  TITLE     Structural and dynamic properties of bacterial type IV secretion systems (review).\n",
      "  JOURNAL   Mol Membr Biol 22:51-61 (2005)\n",
      "            DOI:10.1080/09687860500063316\n",
      "REFERENCE   PMID:15546668\n",
      "  AUTHORS   Christie PJ\n",
      "  TITLE     Type IV secretion: the Agrobacterium VirB/D4 and related conjugation systems.\n",
      "  JOURNAL   Biochim Biophys Acta 1694:219-34 (2004)\n",
      "            DOI:10.1016/j.bbamcr.2004.02.013\n",
      "REFERENCE   PMID:15590781\n",
      "  AUTHORS   Henderson IR, Navarro-Garcia F, Desvaux M, Fernandez RC, Ala'Aldeen D\n",
      "  TITLE     Type V protein secretion pathway: the autotransporter story.\n",
      "  JOURNAL   Microbiol Mol Biol Rev 68:692-744 (2004)\n",
      "            DOI:10.1128/MMBR.68.4.692-744.2004\n",
      "REFERENCE   PMID:9778731\n",
      "  AUTHORS   Henderson IR, Navarro-Garcia F, Nataro JP\n",
      "  TITLE     The great escape: structure and function of the autotransporter proteins.\n",
      "  JOURNAL   Trends Microbiol 6:370-8 (1998)\n",
      "            DOI:10.1016/S0966-842X(98)01318-3\n",
      "REFERENCE   PMID:18524912\n",
      "  AUTHORS   Filloux A, Hachani A, Bleves S\n",
      "  TITLE     The bacterial type VI secretion machine: yet another player for protein transport across membranes.\n",
      "  JOURNAL   Microbiology 154:1570-83 (2008)\n",
      "            DOI:10.1099/mic.0.2008/016840-0\n",
      "REFERENCE   PMID:19251641\n",
      "  AUTHORS   Leiman PG, Basler M, Ramagopal UA, Bonanno JB, Sauder JM, Pukatzki S, Burley SK, Almo SC, Mekalanos JJ\n",
      "  TITLE     Type VI secretion apparatus and phage tail-associated protein complexes share a common evolutionary origin.\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 106:4154-9 (2009)\n",
      "            DOI:10.1073/pnas.0813360106\n",
      "REFERENCE   PMID:17603507\n",
      "  AUTHORS   Kulasekara HD, Miller SI\n",
      "  TITLE     Threonine phosphorylation times bacterial secretion.\n",
      "  JOURNAL   Nat Cell Biol 9:734-6 (2007)\n",
      "            DOI:10.1038/ncb0707-734\n",
      "REFERENCE   PMID:18078384\n",
      "  AUTHORS   Driessen AJ, Nouwen N\n",
      "  TITLE     Protein translocation across the bacterial cytoplasmic membrane.\n",
      "  JOURNAL   Annu Rev Biochem 77:643-67 (2008)\n",
      "            DOI:10.1146/annurev.biochem.77.061606.160747\n",
      "REFERENCE   PMID:14641032\n",
      "  AUTHORS   Pool MR\n",
      "  TITLE     Getting to the membrane: how is co-translational protein targeting to the endoplasmic reticulum regulated?\n",
      "  JOURNAL   Biochem Soc Trans 31:1232-7 (2003)\n",
      "            DOI:10.1042/bst0311232\n",
      "REFERENCE   PMID:19299134\n",
      "  AUTHORS   Desvaux M, Hebraud M, Talon R, Henderson IR\n",
      "  TITLE     Secretion and subcellular localizations of bacterial proteins: a semantic awareness issue.\n",
      "  JOURNAL   Trends Microbiol 17:139-45 (2009)\n",
      "            DOI:10.1016/j.tim.2009.01.004\n",
      "REFERENCE   PMID:15063851\n",
      "  AUTHORS   Nakatogawa H, Murakami A, Ito K\n",
      "  TITLE     Control of SecA and SecM translation by protein secretion.\n",
      "  JOURNAL   Curr Opin Microbiol 7:145-50 (2004)\n",
      "            DOI:10.1016/j.mib.2004.01.001\n",
      "KO_PATHWAY  ko03070\n",
      "///\n",
      "path:pae03410\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae03410                    Pathway\n",
      "NAME        Base excision repair - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Base excision repair (BER) is the predominant DNA damage repair pathway for the processing of small base lesions, derived from oxidation and alkylation damages. BER is normally defined as DNA repair initiated by lesion-specific DNA glycosylases and completed by either of the two sub-pathways: short-patch BER where only one nucleotide is replaced and long-patch BER where 2-13 nucleotides are replaced. Each sub-pathway of BER relies on the formation of protein complexes that assemble at the site of the DNA lesion and facilitate repair in a coordinated fashion. This process of complex formation appears to provide an increase in specificity and efficiency to the BER pathway, thereby facilitating the maintenance of genome integrity by preventing the accumulation of highly toxic repair intermediates.\n",
      "CLASS       Genetic Information Processing; Replication and repair\n",
      "PATHWAY_MAP pae03410  Base excision repair\n",
      "DBLINKS     GO: 0006284 0006285 0006286 0006287 0006288\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA0357  mutM; formamidopyrimidine-DNA glycosylase [KO:K10563] [EC:3.2.2.23 4.2.99.18]\n",
      "line in gene PA0357  mutM; formamidopyrimidine-DNA glycosylase [KO:K10563] [EC:3.2.2.23 4.2.99.18]\n",
      "            PA3495  nth; endonuclease III [KO:K10773] [EC:4.2.99.18]\n",
      "line in gene PA3495  nth; endonuclease III [KO:K10773] [EC:4.2.99.18]\n",
      "            PA1686  alkA; DNA-3-methyladenine glycosidase II [KO:K01247] [EC:3.2.2.21]\n",
      "line in gene PA1686  alkA; DNA-3-methyladenine glycosidase II [KO:K01247] [EC:3.2.2.21]\n",
      "            PA0010  tag; DNA-3-methyladenine glycosidase I [KO:K01246] [EC:3.2.2.20]\n",
      "line in gene PA0010  tag; DNA-3-methyladenine glycosidase I [KO:K01246] [EC:3.2.2.20]\n",
      "            PA0750  ung; uracil-DNA glycosylase [KO:K03648] [EC:3.2.2.27]\n",
      "line in gene PA0750  ung; uracil-DNA glycosylase [KO:K03648] [EC:3.2.2.27]\n",
      "            PA5147  mutY; A/G-specific adenine glycosylase [KO:K03575] [EC:3.2.2.31]\n",
      "line in gene PA5147  mutY; A/G-specific adenine glycosylase [KO:K03575] [EC:3.2.2.31]\n",
      "            PA4010  3-methyladenine DNA glycosylase [KO:K03652] [EC:3.2.2.21]\n",
      "line in gene PA4010  3-methyladenine DNA glycosylase [KO:K03652] [EC:3.2.2.21]\n",
      "            PA2545  xthA; exonuclease III [KO:K01142] [EC:3.1.11.2]\n",
      "line in gene PA2545  xthA; exonuclease III [KO:K01142] [EC:3.1.11.2]\n",
      "            PA5332  crc; catabolite repression control protein [KO:K01142] [EC:3.1.11.2]\n",
      "line in gene PA5332  crc; catabolite repression control protein [KO:K01142] [EC:3.1.11.2]\n",
      "            PA4172  nuclease [KO:K01142] [EC:3.1.11.2]\n",
      "line in gene PA4172  nuclease [KO:K01142] [EC:3.1.11.2]\n",
      "            PA5493  polA; DNA polymerase I [KO:K02335] [EC:2.7.7.7]\n",
      "line in gene PA5493  polA; DNA polymerase I [KO:K02335] [EC:2.7.7.7]\n",
      "            PA3725  recJ; single-stranded-DNA-specific exonuclease RecJ [KO:K07462] [EC:3.1.-.-]\n",
      "line in gene PA3725  recJ; single-stranded-DNA-specific exonuclease RecJ [KO:K07462] [EC:3.1.-.-]\n",
      "            PA1529  lig; NAD-dependent DNA ligase LigA [KO:K01972] [EC:6.5.1.2]\n",
      "line in gene PA1529  lig; NAD-dependent DNA ligase LigA [KO:K01972] [EC:6.5.1.2]\n",
      "REFERENCE   PMID:17893748\n",
      "  AUTHORS   Krwawicz J, Arczewska KD, Speina E, Maciejewska A, Grzesiuk E.\n",
      "  TITLE     Bacterial DNA repair genes and their eukaryotic homologues: 1. Mutations in genes involved in base excision repair (BER) and DNA-end processors and their implication in mutagenesis and human disease.\n",
      "  JOURNAL   Acta Biochim Pol 54:413-34 (2007)\n",
      "REFERENCE   PMID:17337257\n",
      "  AUTHORS   Almeida KH, Sobol RW.\n",
      "  TITLE     A unified view of base excision repair: lesion-dependent protein complexes regulated by post-translational modification.\n",
      "  JOURNAL   DNA Repair (Amst) 6:695-711 (2007)\n",
      "            DOI:10.1016/j.dnarep.2007.01.009\n",
      "REFERENCE   PMID:21665970\n",
      "  AUTHORS   Moen MN, Knaevelsrud I, Haugland GT, Grosvik K, Birkeland NK, Klungland A, Bjelland S\n",
      "  TITLE     Uracil-DNA glycosylase of Thermoplasma acidophilum directs long-patch base excision repair, which is promoted by deoxynucleoside triphosphates and ATP/ADP, into short-patch repair.\n",
      "  JOURNAL   J Bacteriol 193:4495-508 (2011)\n",
      "            DOI:10.1128/JB.00233-11\n",
      "REFERENCE   PMID:11436317\n",
      "  AUTHORS   Ikeda S, Seki S.\n",
      "  TITLE     [Base excision repair: DNA glycosylase and AP endonuclease]\n",
      "  JOURNAL   Tanpakushitsu Kakusan Koso 46:916-23 (2001)\n",
      "KO_PATHWAY  ko03410\n",
      "///\n",
      "path:pae03420\n",
      "ENTRY       pae03420                    Pathway\n",
      "NAME        Nucleotide excision repair - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Nucleotide excision repair (NER) is a mechanism to recognize and repair bulky DNA damage caused by compounds, environmental carcinogens, and exposure to UV-light. In humans hereditary defects in the NER pathway are linked to at least three diseases: xeroderma pigmentosum (XP), Cockayne syndrome (CS), and trichothiodystrophy (TTD). The repair of damaged DNA involves at least 30 polypeptides within two different sub-pathways of NER known as transcription-coupled repair (TCR-NER) and global genome repair (GGR-NER). TCR refers to the expedited repair of lesions located in the actively transcribed strand of genes by RNA polymerase II (RNAP II). In GGR-NER the first step of damage recognition involves XPC-hHR23B complex together with XPE complex (in prokaryotes, uvrAB complex). The following steps of GGR-NER and TCR-NER are similar.\n",
      "CLASS       Genetic Information Processing; Replication and repair\n",
      "PATHWAY_MAP pae03420  Nucleotide excision repair\n",
      "DBLINKS     GO: 0006289\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3002  mfd; transcription-repair coupling factor [KO:K03723] [EC:3.6.4.-]\n",
      "line in gene PA3002  mfd; transcription-repair coupling factor [KO:K03723] [EC:3.6.4.-]\n",
      "            PA4234  uvrA; excinuclease ABC subunit A [KO:K03701]\n",
      "line in gene PA4234  uvrA; excinuclease ABC subunit A [KO:K03701]\n",
      "            PA3138  uvrB; excinuclease ABC subunit B [KO:K03702]\n",
      "line in gene PA3138  uvrB; excinuclease ABC subunit B [KO:K03702]\n",
      "            PA2585  uvrC; excinuclease ABC subunit C [KO:K03703]\n",
      "line in gene PA2585  uvrC; excinuclease ABC subunit C [KO:K03703]\n",
      "            PA5443  uvrD; DNA-dependent helicase II [KO:K03657] [EC:5.6.2.4]\n",
      "line in gene PA5443  uvrD; DNA-dependent helicase II [KO:K03657] [EC:5.6.2.4]\n",
      "            PA5493  polA; DNA polymerase I [KO:K02335] [EC:2.7.7.7]\n",
      "line in gene PA5493  polA; DNA polymerase I [KO:K02335] [EC:2.7.7.7]\n",
      "            PA1529  lig; NAD-dependent DNA ligase LigA [KO:K01972] [EC:6.5.1.2]\n",
      "line in gene PA1529  lig; NAD-dependent DNA ligase LigA [KO:K01972] [EC:6.5.1.2]\n",
      "            PA1866  hypothetical protein [KO:K10844] [EC:5.6.2.3]\n",
      "line in gene PA1866  hypothetical protein [KO:K10844] [EC:5.6.2.3]\n",
      "REFERENCE   PMID:11436314\n",
      "  AUTHORS   Sugasawa K.\n",
      "  TITLE     [Molecular mechanism of mammalian nucleotide excision repair]\n",
      "  JOURNAL   Tanpakushitsu Kakusan Koso 46:893-901 (2001)\n",
      "REFERENCE   PMID:11436316\n",
      "  AUTHORS   Nakatsu Y.\n",
      "  TITLE     [Transcription-coupled repair and Cockayne syndrome]\n",
      "  JOURNAL   Tanpakushitsu Kakusan Koso 46:908-15 (2001)\n",
      "REFERENCE   PMID:11436323\n",
      "  AUTHORS   Nakagawa N, Masui R, Kuramitsu S.\n",
      "  TITLE     [Structure and function of DNA repair enzyme UvrB from Thermus thermophilus HB8]\n",
      "  JOURNAL   Tanpakushitsu Kakusan Koso 46:968-75 (2001)\n",
      "REFERENCE   PMID:17276014\n",
      "  AUTHORS   Kraemer KH, Patronas NJ, Schiffmann R, Brooks BP, Tamura D, DiGiovanna JJ.\n",
      "  TITLE     Xeroderma pigmentosum, trichothiodystrophy and Cockayne syndrome: a complex genotype-phenotype relationship.\n",
      "  JOURNAL   Neuroscience 145:1388-96 (2007)\n",
      "            DOI:10.1016/j.neuroscience.2006.12.020\n",
      "REFERENCE   PMID:17884153\n",
      "  AUTHORS   Saldivar JS, Wu X, Follen M, Gershenson D.\n",
      "  TITLE     Nucleotide excision repair pathway review I: implications in ovarian cancer and platinum sensitivity.\n",
      "  JOURNAL   Gynecol Oncol 107:S56-71 (2007)\n",
      "            DOI:10.1016/j.ygyno.2007.07.043\n",
      "REFERENCE   PMID:16922398\n",
      "  AUTHORS   Sugasawa K.\n",
      "  TITLE     [DNA repair pathways involving Cul4A ubiquitin ligases]\n",
      "  JOURNAL   Tanpakushitsu Kakusan Koso 51:1339-44 (2006)\n",
      "REFERENCE   PMID:18018633\n",
      "  AUTHORS   Ito S, Ando D, Tanaka K.\n",
      "  TITLE     [Cross-talk between DNA repair and transcription: molecular mechanism and disorders]\n",
      "  JOURNAL   Tanpakushitsu Kakusan Koso 52:1823-31 (2007)\n",
      "REL_PATHWAY pae03020  RNA polymerase\n",
      "KO_PATHWAY  ko03420\n",
      "///\n",
      "path:pae03430\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae03430                    Pathway\n",
      "NAME        Mismatch repair - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION DNA mismatch repair (MMR) is a highly conserved biological pathway that plays a key role in maintaining genomic stability. MMR corrects DNA mismatches generated during DNA replication, thereby preventing mutations from becoming permanent in dividing cells. MMR also suppresses homologous recombination and was recently shown to play a role in DNA damage signaling. Defects in MMR are associated with genome-wide instability, predisposition to certain types of cancer including HNPCC, resistance to certain chemotherapeutic agents, and abnormalities in meiosis and sterility in mammalian systems.\n",
      "            The Escherichia coli MMR pathway has been extensively studied and is well characterized. In E. coli, the mismatch-activated MutS-MutL-ATP complex licenses MutH to incise the nearest unmethylated GATC sequence. UvrD and an exonuclease generate a gap. This gap is filled by pol III and DNA ligase. The GATC sites are then methylated by Dam. Several human MMR proteins have been identified based on their homology to E. coli MMR proteins. These include human homologs of MutS and MutL. Although E. coli MutS and MutL proteins are homodimers, human MutS and MutL homologs are heterodimers. The role of hemimethylated dGATC sites as a signal for strand discrimination is not conserved from E. coli to human. Human MMR is presumed to be nick-directed in vivo, and is thought to discriminate daughter and template strands using a strand-specific nick.\n",
      "CLASS       Genetic Information Processing; Replication and repair\n",
      "PATHWAY_MAP pae03430  Mismatch repair\n",
      "DBLINKS     GO: 0006298\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3620  mutS; DNA mismatch repair protein MutS [KO:K03555]\n",
      "line in gene PA3620  mutS; DNA mismatch repair protein MutS [KO:K03555]\n",
      "            PA4946  mutL; DNA mismatch repair protein [KO:K03572]\n",
      "line in gene PA4946  mutL; DNA mismatch repair protein [KO:K03572]\n",
      "            PA5443  uvrD; DNA-dependent helicase II [KO:K03657] [EC:5.6.2.4]\n",
      "line in gene PA5443  uvrD; DNA-dependent helicase II [KO:K03657] [EC:5.6.2.4]\n",
      "            PA4316  sbcB; exonuclease I [KO:K01141] [EC:3.1.11.1]\n",
      "line in gene PA4316  sbcB; exonuclease I [KO:K01141] [EC:3.1.11.1]\n",
      "            PA3777  xseA; exodeoxyribonuclease VII large subunit [KO:K03601] [EC:3.1.11.6]\n",
      "line in gene PA3777  xseA; exodeoxyribonuclease VII large subunit [KO:K03601] [EC:3.1.11.6]\n",
      "            PA4042  xseB; exodeoxyribonuclease VII small subunit [KO:K03602] [EC:3.1.11.6]\n",
      "line in gene PA4042  xseB; exodeoxyribonuclease VII small subunit [KO:K03602] [EC:3.1.11.6]\n",
      "            PA3725  recJ; single-stranded-DNA-specific exonuclease RecJ [KO:K07462] [EC:3.1.-.-]\n",
      "line in gene PA3725  recJ; single-stranded-DNA-specific exonuclease RecJ [KO:K07462] [EC:3.1.-.-]\n",
      "            PA4232  ssb; single-stranded DNA-binding protein [KO:K03111]\n",
      "line in gene PA4232  ssb; single-stranded DNA-binding protein [KO:K03111]\n",
      "            PA3640  dnaE; DNA polymerase III subunit alpha [KO:K02337] [EC:2.7.7.7]\n",
      "line in gene PA3640  dnaE; DNA polymerase III subunit alpha [KO:K02337] [EC:2.7.7.7]\n",
      "            PA0002  dnaN; DNA polymerase III subunit beta [KO:K02338] [EC:2.7.7.7]\n",
      "line in gene PA0002  dnaN; DNA polymerase III subunit beta [KO:K02338] [EC:2.7.7.7]\n",
      "            PA1532  dnaX; DNA polymerase III subunits gamma/tau [KO:K02343] [EC:2.7.7.7]\n",
      "line in gene PA1532  dnaX; DNA polymerase III subunits gamma/tau [KO:K02343] [EC:2.7.7.7]\n",
      "            PA3989  holA; DNA polymerase III subunit delta [KO:K02340] [EC:2.7.7.7]\n",
      "line in gene PA3989  holA; DNA polymerase III subunit delta [KO:K02340] [EC:2.7.7.7]\n",
      "            PA2961  holB; DNA polymerase III subunit delta' [KO:K02341] [EC:2.7.7.7]\n",
      "line in gene PA2961  holB; DNA polymerase III subunit delta' [KO:K02341] [EC:2.7.7.7]\n",
      "            PA1816  dnaQ; DNA polymerase III subunit epsilon [KO:K02342] [EC:2.7.7.7]\n",
      "line in gene PA1816  dnaQ; DNA polymerase III subunit epsilon [KO:K02342] [EC:2.7.7.7]\n",
      "            PA3232  DNA polymerase III subunit epsilon [KO:K02342] [EC:2.7.7.7]\n",
      "line in gene PA3232  DNA polymerase III subunit epsilon [KO:K02342] [EC:2.7.7.7]\n",
      "            PA3832  holC; DNA polymerase III subunit chi [KO:K02339] [EC:2.7.7.7]\n",
      "line in gene PA3832  holC; DNA polymerase III subunit chi [KO:K02339] [EC:2.7.7.7]\n",
      "            PA1529  lig; NAD-dependent DNA ligase LigA [KO:K01972] [EC:6.5.1.2]\n",
      "line in gene PA1529  lig; NAD-dependent DNA ligase LigA [KO:K01972] [EC:6.5.1.2]\n",
      "REFERENCE   PMID:16612326\n",
      "  AUTHORS   Jiricny J.\n",
      "  TITLE     The multifaceted mismatch-repair system.\n",
      "  JOURNAL   Nat Rev Mol Cell Biol 7:335-46 (2006)\n",
      "            DOI:10.1038/nrm1907\n",
      "REFERENCE   PMID:18157157\n",
      "  AUTHORS   Li GM.\n",
      "  TITLE     Mechanisms and functions of DNA mismatch repair.\n",
      "  JOURNAL   Cell Res 18:85-98 (2008)\n",
      "            DOI:10.1038/cr.2007.115\n",
      "REFERENCE   PMID:11920679\n",
      "  AUTHORS   Marti TM, Kunz C, Fleck O.\n",
      "  TITLE     DNA mismatch repair and mutation avoidance pathways.\n",
      "  JOURNAL   J Cell Physiol 191:28-41 (2002)\n",
      "            DOI:10.1002/jcp.10077\n",
      "REFERENCE   PMID:11436301\n",
      "  AUTHORS   Ikejima M, Shimada T.\n",
      "  TITLE     [Molecular mechanism of mismatch repair]\n",
      "  JOURNAL   Tanpakushitsu Kakusan Koso 46:1124-9 (2001)\n",
      "KO_PATHWAY  ko03430\n",
      "///\n",
      "path:pae03440\n",
      "ENTRY       pae03440                    Pathway\n",
      "NAME        Homologous recombination - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Homologous recombination (HR) is essential for the accurate repair of DNA double-strand breaks (DSBs), potentially lethal lesions. HR takes place in the late S-G2 phase of the cell cycle and involves the generation of a single-stranded region of DNA, followed by strand invasion, formation of a Holliday junction, DNA synthesis using the intact strand as a template, branch migration and resolution. It is investigated that RecA/Rad51 family proteins play a central role. The breast cancer susceptibility protein Brca2 and the RecQ helicase BLM (Bloom syndrome mutated) are tumor suppressors that maintain genome integrity, at least in part, through HR.\n",
      "CLASS       Genetic Information Processing; Replication and repair\n",
      "PATHWAY_MAP pae03440  Homologous recombination\n",
      "DBLINKS     GO: 0000724\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3725  recJ; single-stranded-DNA-specific exonuclease RecJ [KO:K07462] [EC:3.1.-.-]\n",
      "line in gene PA3725  recJ; single-stranded-DNA-specific exonuclease RecJ [KO:K07462] [EC:3.1.-.-]\n",
      "            PA4232  ssb; single-stranded DNA-binding protein [KO:K03111]\n",
      "line in gene PA4232  ssb; single-stranded DNA-binding protein [KO:K03111]\n",
      "            PA0003  recF; DNA replication and repair protein RecF [KO:K03629]\n",
      "line in gene PA0003  recF; DNA replication and repair protein RecF [KO:K03629]\n",
      "            PA0772  recO; DNA repair protein RecO [KO:K03584]\n",
      "line in gene PA0772  recO; DNA repair protein RecO [KO:K03584]\n",
      "            PA1534  recR; recombination protein RecR [KO:K06187]\n",
      "line in gene PA1534  recR; recombination protein RecR [KO:K06187]\n",
      "            PA3617  recA; recombinase A [KO:K03553]\n",
      "line in gene PA3617  recA; recombinase A [KO:K03553]\n",
      "            PA5493  polA; DNA polymerase I [KO:K02335] [EC:2.7.7.7]\n",
      "line in gene PA5493  polA; DNA polymerase I [KO:K02335] [EC:2.7.7.7]\n",
      "            PA0966  ruvA; Holliday junction ATP-dependent DNA helicase RuvA [KO:K03550] [EC:5.6.2.4]\n",
      "line in gene PA0966  ruvA; Holliday junction ATP-dependent DNA helicase RuvA [KO:K03550] [EC:5.6.2.4]\n",
      "            PA0967  ruvB; Holliday junction ATP-dependent DNA helicase RuvB [KO:K03551] [EC:5.6.2.4]\n",
      "line in gene PA0967  ruvB; Holliday junction ATP-dependent DNA helicase RuvB [KO:K03551] [EC:5.6.2.4]\n",
      "            PA0965  ruvC; crossover junction endodeoxyribonuclease RuvC [KO:K01159] [EC:3.1.21.10]\n",
      "line in gene PA0965  ruvC; crossover junction endodeoxyribonuclease RuvC [KO:K01159] [EC:3.1.21.10]\n",
      "            PA5345  recG; ATP-dependent DNA helicase RecG [KO:K03655] [EC:5.6.2.4]\n",
      "line in gene PA5345  recG; ATP-dependent DNA helicase RecG [KO:K03655] [EC:5.6.2.4]\n",
      "            PA4284  recB; exodeoxyribonuclease V subunit beta [KO:K03582] [EC:3.1.11.5]\n",
      "line in gene PA4284  recB; exodeoxyribonuclease V subunit beta [KO:K03582] [EC:3.1.11.5]\n",
      "            PA4285  recC; exodeoxyribonuclease V subunit gamma [KO:K03583] [EC:3.1.11.5]\n",
      "line in gene PA4285  recC; exodeoxyribonuclease V subunit gamma [KO:K03583] [EC:3.1.11.5]\n",
      "            PA4283  recD; exodeoxyribonuclease V subunit alpha [KO:K03581] [EC:3.1.11.5]\n",
      "line in gene PA4283  recD; exodeoxyribonuclease V subunit alpha [KO:K03581] [EC:3.1.11.5]\n",
      "            PA3640  dnaE; DNA polymerase III subunit alpha [KO:K02337] [EC:2.7.7.7]\n",
      "line in gene PA3640  dnaE; DNA polymerase III subunit alpha [KO:K02337] [EC:2.7.7.7]\n",
      "            PA0002  dnaN; DNA polymerase III subunit beta [KO:K02338] [EC:2.7.7.7]\n",
      "line in gene PA0002  dnaN; DNA polymerase III subunit beta [KO:K02338] [EC:2.7.7.7]\n",
      "            PA1532  dnaX; DNA polymerase III subunits gamma/tau [KO:K02343] [EC:2.7.7.7]\n",
      "line in gene PA1532  dnaX; DNA polymerase III subunits gamma/tau [KO:K02343] [EC:2.7.7.7]\n",
      "            PA3989  holA; DNA polymerase III subunit delta [KO:K02340] [EC:2.7.7.7]\n",
      "line in gene PA3989  holA; DNA polymerase III subunit delta [KO:K02340] [EC:2.7.7.7]\n",
      "            PA2961  holB; DNA polymerase III subunit delta' [KO:K02341] [EC:2.7.7.7]\n",
      "line in gene PA2961  holB; DNA polymerase III subunit delta' [KO:K02341] [EC:2.7.7.7]\n",
      "            PA1816  dnaQ; DNA polymerase III subunit epsilon [KO:K02342] [EC:2.7.7.7]\n",
      "line in gene PA1816  dnaQ; DNA polymerase III subunit epsilon [KO:K02342] [EC:2.7.7.7]\n",
      "            PA3232  DNA polymerase III subunit epsilon [KO:K02342] [EC:2.7.7.7]\n",
      "line in gene PA3232  DNA polymerase III subunit epsilon [KO:K02342] [EC:2.7.7.7]\n",
      "            PA3832  holC; DNA polymerase III subunit chi [KO:K02339] [EC:2.7.7.7]\n",
      "line in gene PA3832  holC; DNA polymerase III subunit chi [KO:K02339] [EC:2.7.7.7]\n",
      "            PA5050  priA; primosome assembly protein PriA [KO:K04066] [EC:3.6.4.-]\n",
      "line in gene PA5050  priA; primosome assembly protein PriA [KO:K04066] [EC:3.6.4.-]\n",
      "REFERENCE   PMID:18086882\n",
      "  AUTHORS   Maloisel L, Fabre F, Gangloff S.\n",
      "  TITLE     DNA polymerase delta is preferentially recruited during homologous recombination to promote heteroduplex DNA extension.\n",
      "  JOURNAL   Mol Cell Biol 28:1373-82 (2008)\n",
      "            DOI:10.1128/MCB.01651-07\n",
      "REFERENCE   PMID:17571595\n",
      "  AUTHORS   Haruta-Takahashi N, Iwasaki H.\n",
      "  TITLE     [Recombination mediators]\n",
      "  JOURNAL   Seikagaku 79:449-53 (2007)\n",
      "REFERENCE   PMID:18430459\n",
      "  AUTHORS   Ouyang KJ, Woo LL, Ellis NA.\n",
      "  TITLE     Homologous recombination and maintenance of genome integrity: Cancer and aging through the prism of human RecQ helicases.\n",
      "  JOURNAL   Mech Ageing Dev 129:425-40 (2008)\n",
      "            DOI:10.1016/j.mad.2008.03.003\n",
      "REFERENCE   PMID:15902993\n",
      "  AUTHORS   Kawabata M, Kawabata T, Nishibori M.\n",
      "  TITLE     Role of recA/RAD51 family proteins in mammals.\n",
      "  JOURNAL   Acta Med Okayama 59:1-9 (2005)\n",
      "            DOI:10.18926/AMO/31987\n",
      "REFERENCE   PMID:12589470\n",
      "  AUTHORS   Prado F, Cortes-Ledesma F, Huertas P, Aguilera A.\n",
      "  TITLE     Mitotic recombination in Saccharomyces cerevisiae.\n",
      "  JOURNAL   Curr Genet 42:185-98 (2003)\n",
      "            DOI:10.1007/s00294-002-0346-3\n",
      "REFERENCE   PMID:16935872\n",
      "  AUTHORS   Heyer WD, Li X, Rolfsmeier M, Zhang XP.\n",
      "  TITLE     Rad54: the Swiss Army knife of homologous recombination?\n",
      "  JOURNAL   Nucleic Acids Res 34:4115-25 (2006)\n",
      "            DOI:10.1093/nar/gkl481\n",
      "REFERENCE   PMID:17395553\n",
      "  AUTHORS   Michel B, Boubakri H, Baharoglu Z, LeMasson M, Lestini R.\n",
      "  TITLE     Recombination proteins and rescue of arrested replication forks.\n",
      "  JOURNAL   DNA Repair (Amst) 6:967-80 (2007)\n",
      "            DOI:10.1016/j.dnarep.2007.02.016\n",
      "KO_PATHWAY  ko03440\n",
      "///\n",
      "path:pae03450\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ENTRY       pae03450                    Pathway\n",
      "NAME        Non-homologous end-joining - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Nonhomologous end joining (NHEJ) eliminates DNA double-strand breaks (DSBs) by direct ligation. NHEJ involves binding of the KU heterodimer to double-stranded DNA ends, recruitment of DNA-PKcs (MRX complex in yeast), processing of ends, and recruitment of the DNA ligase IV (LIG4)-XRCC4 complex, which brings about ligation. A recent study shows that bacteria accomplish NHEJ using just two proteins (Ku and DNA ligase), whereas eukaryotes require many factors. NHEJ repairs DSBs at all stages of the cell cycle, bringing about the ligation of two DNA DSBs without the need for sequence homology, and so is error-prone.\n",
      "CLASS       Genetic Information Processing; Replication and repair\n",
      "PATHWAY_MAP pae03450  Non-homologous end-joining\n",
      "DBLINKS     GO: 0006303\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA2150  non-homologous end joining protein Ku [KO:K10979]\n",
      "line in gene PA2150  non-homologous end joining protein Ku [KO:K10979]\n",
      "            PA2138  multifunctional non-homologous end joining protein LigD [KO:K01971] [EC:6.5.1.1]\n",
      "line in gene PA2138  multifunctional non-homologous end joining protein LigD [KO:K01971] [EC:6.5.1.1]\n",
      "REFERENCE   PMID:16971555\n",
      "  AUTHORS   Gennery AR.\n",
      "  TITLE     Primary immunodeficiency syndromes associated with defective DNA double-strand break repair.\n",
      "  JOURNAL   Br Med Bull 77-78:71-85 (2006)\n",
      "            DOI:10.1093/bmb/ldl006\n",
      "REFERENCE   PMID:17589524\n",
      "  AUTHORS   Zhang Y, Hefferin ML, Chen L, Shim EY, Tseng HM, Kwon Y, Sung P, Lee SE, Tomkinson AE.\n",
      "  TITLE     Role of Dnl4-Lif1 in nonhomologous end-joining repair complex assembly and suppression of homologous recombination.\n",
      "  JOURNAL   Nat Struct Mol Biol 14:639-46 (2007)\n",
      "            DOI:10.1038/nsmb1261\n",
      "REFERENCE   PMID:16895466\n",
      "  AUTHORS   Wyman C, Kanaar R.\n",
      "  TITLE     DNA double-strand break repair: all's well that ends well.\n",
      "  JOURNAL   Annu Rev Genet 40:363-83 (2006)\n",
      "            DOI:10.1146/annurev.genet.40.110405.090451\n",
      "REFERENCE   PMID:15578053\n",
      "  AUTHORS   Weller GR, Brandt VL, Roth DB.\n",
      "  TITLE     Doing more with less in bacterial DNA repair.\n",
      "  JOURNAL   Nat Struct Mol Biol 11:1158-9 (2004)\n",
      "            DOI:10.1038/nsmb1204-1158\n",
      "REFERENCE   PMID:12215643\n",
      "  AUTHORS   Weller GR, Kysela B, Roy R, Tonkin LM, Scanlan E, Della M, Devine SK, Day JP, Wilkinson A, d'Adda di Fagagna F, Devine KM, Bowater RP, Jeggo PA, Jackson SP, Doherty AJ.\n",
      "  TITLE     Identification of a DNA nonhomologous end-joining complex in bacteria.\n",
      "  JOURNAL   Science 297:1686-9 (2002)\n",
      "            DOI:10.1126/science.1074584\n",
      "REFERENCE   PMID:17506672\n",
      "  AUTHORS   Pitcher RS, Brissett NC, Doherty AJ.\n",
      "  TITLE     Nonhomologous end-joining in bacteria: a microbial perspective.\n",
      "  JOURNAL   Annu Rev Microbiol 61:259-82 (2007)\n",
      "            DOI:10.1146/annurev.micro.61.080706.093354\n",
      "KO_PATHWAY  ko03450\n",
      "///\n",
      "path:pae04122\n",
      "ENTRY       pae04122                    Pathway\n",
      "NAME        Sulfur relay system - Pseudomonas aeruginosa PAO1\n",
      "DESCRIPTION Ubiquitin and ubiquitin-like proteins (Ubls) are signalling messengers that control many cellular functions, such as cell proliferation, apoptosis, and DNA repair. It is suggested that Ub-protein modification evolved from prokaryotic sulfurtransfer systems. Molybdenum cofactor (Moco) and thiamin are sulfur-containing cofactors whose biosynthesis includes a key sulfur transfer step that uses unique sulfur carrier proteins, MoaD and ThiS. Ubiquitin, MoaD, and ThiS are all structurally related proteins whose C-termini are activated through adenylation by homologous E1-like enzymes. s2T biosynthesis may share similar chemistry with Moco and thiamin synthesis. In Saccharomyces cerevisiae, Urm1 and Uba4 function as part of a ubl protein conjugation system, though they have sequence homology to bacterial sulfur-transfer enzymes and the ability to function in sulfur transfer.\n",
      "CLASS       Genetic Information Processing; Folding, sorting and degradation\n",
      "PATHWAY_MAP pae04122  Sulfur relay system\n",
      "ORGANISM    Pseudomonas aeruginosa PAO1 [GN:pae]\n",
      "GENE        PA3814  iscS; cysteine desulfurase [KO:K04487] [EC:2.8.1.7]\n",
      "line in gene PA3814  iscS; cysteine desulfurase [KO:K04487] [EC:2.8.1.7]\n",
      "            PA2062  pyridoxal-phosphate dependent protein [KO:K04487] [EC:2.8.1.7]\n",
      "line in gene PA2062  pyridoxal-phosphate dependent protein [KO:K04487] [EC:2.8.1.7]\n",
      "            PA1292  3-mercaptopyruvate sulfurtransferase [KO:K01011] [EC:2.8.1.1 2.8.1.2]\n",
      "line in gene PA1292  3-mercaptopyruvate sulfurtransferase [KO:K01011] [EC:2.8.1.1 2.8.1.2]\n",
      "            PA4956  rhdA; thiosulfate:cyanide sulfurtransferase [KO:K01011] [EC:2.8.1.1 2.8.1.2]\n",
      "line in gene PA4956  rhdA; thiosulfate:cyanide sulfurtransferase [KO:K01011] [EC:2.8.1.1 2.8.1.2]\n",
      "            PA1564  sulfurtransferase TusA [KO:K04085] [EC:2.8.1.-]\n",
      "line in gene PA1564  sulfurtransferase TusA [KO:K04085] [EC:2.8.1.-]\n",
      "            PA1006  hypothetical protein [KO:K04085] [EC:2.8.1.-]\n",
      "line in gene PA1006  hypothetical protein [KO:K04085] [EC:2.8.1.-]\n",
      "            PA2607  hypothetical protein [KO:K07237]\n",
      "line in gene PA2607  hypothetical protein [KO:K07237]\n",
      "            PA2606  sulfur relay protein TusC [KO:K07236]\n",
      "line in gene PA2606  sulfur relay protein TusC [KO:K07236]\n",
      "            PA2605  sulfurtransferase TusD [KO:K07235] [EC:2.8.1.-]\n",
      "line in gene PA2605  sulfurtransferase TusD [KO:K07235] [EC:2.8.1.-]\n",
      "            PA2608  hypothetical protein [KO:K11179] [EC:2.8.1.-]\n",
      "line in gene PA2608  hypothetical protein [KO:K11179] [EC:2.8.1.-]\n",
      "            PA2626  trmU; tRNA-specific 2-thiouridylase MnmA [KO:K00566] [EC:2.8.1.13]\n",
      "line in gene PA2626  trmU; tRNA-specific 2-thiouridylase MnmA [KO:K00566] [EC:2.8.1.13]\n",
      "            PA3916  moaE; molybdopterin converting factor large subunit [KO:K03635] [EC:2.8.1.12]\n",
      "line in gene PA3916  moaE; molybdopterin converting factor large subunit [KO:K03635] [EC:2.8.1.12]\n",
      "            PA1505  moaA2; cyclic pyranopterin monophosphate synthase [KO:K03639] [EC:4.1.99.22]\n",
      "line in gene PA1505  moaA2; cyclic pyranopterin monophosphate synthase [KO:K03639] [EC:4.1.99.22]\n",
      "            PA3870  moaA1; molybdenum cofactor biosynthesis protein A [KO:K03639] [EC:4.1.99.22]\n",
      "line in gene PA3870  moaA1; molybdenum cofactor biosynthesis protein A [KO:K03639] [EC:4.1.99.22]\n",
      "            PA3917  moaD; molybdopterin converting factor small subunit [KO:K03636]\n",
      "line in gene PA3917  moaD; molybdopterin converting factor small subunit [KO:K03636]\n",
      "            PA1355  hypothetical protein [KO:K03636]\n",
      "line in gene PA1355  hypothetical protein [KO:K03636]\n",
      "            PA4663  moeB; molybdopterin biosynthesis protein MoeB [KO:K21029] [EC:2.7.7.80]\n",
      "line in gene PA4663  moeB; molybdopterin biosynthesis protein MoeB [KO:K21029] [EC:2.7.7.80]\n",
      "            PA2103  molybdopterin biosynthesis protein MoeB [KO:K21147] [EC:2.7.7.80 2.7.7.- 2.8.1.11 2.8.1.-]\n",
      "line in gene PA2103  molybdopterin biosynthesis protein MoeB [KO:K21147] [EC:2.7.7.80 2.7.7.- 2.8.1.11 2.8.1.-]\n",
      "            PA3918  moaC; molybdenum cofactor biosynthesis protein MoaC [KO:K03637] [EC:4.6.1.17]\n",
      "line in gene PA3918  moaC; molybdenum cofactor biosynthesis protein MoaC [KO:K03637] [EC:4.6.1.17]\n",
      "            PA3029  moaB2; molybdopterin biosynthesis protein B [KO:K03638] [EC:2.7.7.75]\n",
      "line in gene PA3029  moaB2; molybdopterin biosynthesis protein B [KO:K03638] [EC:2.7.7.75]\n",
      "            PA3915  moaB1; molybdopterin biosynthesis protein B [KO:K03638] [EC:2.7.7.75]\n",
      "line in gene PA3915  moaB1; molybdopterin biosynthesis protein B [KO:K03638] [EC:2.7.7.75]\n",
      "            PA5118  thiI; thiamine biosynthesis protein ThiI [KO:K03151] [EC:2.8.1.4]\n",
      "line in gene PA5118  thiI; thiamine biosynthesis protein ThiI [KO:K03151] [EC:2.8.1.4]\n",
      "            PA0380  sulfur carrier protein ThiS [KO:K03154]\n",
      "line in gene PA0380  sulfur carrier protein ThiS [KO:K03154]\n",
      "COMPOUND    C00019  S-Adenosyl-L-methionine\n",
      "            C00041  L-Alanine\n",
      "            C00044  GTP\n",
      "            C00097  L-Cysteine\n",
      "            C00378  Thiamine\n",
      "            C00868  tRNA uridine\n",
      "            C00979  O-Acetyl-L-serine\n",
      "            C05924  Molybdopterin\n",
      "            C17322  tRNA containing 2-thiouridine\n",
      "            C18237  Molybdoenzyme molybdenum cofactor\n",
      "            C18239  Precursor Z\n",
      "REFERENCE   PMID:19151091\n",
      "  AUTHORS   Noma A, Sakaguchi Y, Suzuki T\n",
      "  TITLE     Mechanistic characterization of the sulfur-relay system for eukaryotic 2-thiouridine biogenesis at tRNA wobble positions.\n",
      "  JOURNAL   Nucleic Acids Res 37:1335-52 (2009)\n",
      "            DOI:10.1093/nar/gkn1023\n",
      "REFERENCE   PMID:16859499\n",
      "  AUTHORS   Iyer LM, Burroughs AM, Aravind L\n",
      "  TITLE     The prokaryotic antecedents of the ubiquitin-signaling system and the early evolution of ubiquitin-like beta-grasp domains.\n",
      "  JOURNAL   Genome Biol 7:R60 (2006)\n",
      "            DOI:10.1186/gb-2006-7-7-r60\n",
      "REFERENCE   PMID:19675644\n",
      "  AUTHORS   Schwarz G, Mendel RR, Ribbe MW\n",
      "  TITLE     Molybdenum cofactors, enzymes and pathways.\n",
      "  JOURNAL   Nature 460:839-47 (2009)\n",
      "            DOI:10.1038/nature08302\n",
      "REL_PATHWAY pae00270  Cysteine and methionine metabolism\n",
      "            pae00730  Thiamine metabolism\n",
      "            pae00790  Folate biosynthesis\n",
      "KO_PATHWAY  ko04122\n",
      "///\n"
     ]
    }
   ],
   "source": [
    "# Get the genes for pathways and add them to a list\n",
    "pathway_id_to_genes = {}\n",
    "pathway_id_to_pathway_name = {}\n",
    "\n",
    "for pathway in pao1_pathways_clean:\n",
    "    print(pathway)\n",
    "    pathway_file = REST.kegg_get(pathway).read()  # query and read each pathway\n",
    "\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "\n",
    "    # Initiate empty list to store genes\n",
    "    pathway_gene_ids = []\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        print(line)\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"NAME\":\n",
    "            pathway_id_to_pathway_name[pathway] = line\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            print(\"line in gene\", line[12:])\n",
    "            if line.count(\";\") == 1:\n",
    "                gene_identifiers, gene_description = line[12:].split(\"; \")\n",
    "                gene_id, gene_symbol = gene_identifiers.split()\n",
    "            elif line.count(\";\") == 0:\n",
    "                gene_identifiers = line[12:].split(\" \")\n",
    "                gene_id = gene_identifiers[0]\n",
    "            else:\n",
    "                # in the case of 2 semicolons\n",
    "                gene_identifiers, gene_description, protein_name = line[12:].split(\"; \")\n",
    "                gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if gene_id not in pathway_gene_ids:\n",
    "                pathway_gene_ids.append(gene_id)\n",
    "\n",
    "    pathway_id_to_genes[pathway] = pathway_gene_ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4e5e0644",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathway_id_to_genes[\"path:pae01110\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9a656be4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# import pandas as pd\n",
    "# pathway_gene_df = pd.DataFrame.from_dict(pathway_id_to_genes)\n",
    "# Need a two column pandas df with pathway - list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "215cbfbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# \"PA2624\" in pathway_id_to_genes.values()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b6b677a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import requests as r\n",
    "# r.requests.get(\"http://rest.kegg.jp/list/pathway/pae\")"
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "formats": "ipynb,py"
  },
  "kernelspec": {
   "display_name": "Python [conda env:core_acc] *",
   "language": "python",
   "name": "conda-env-core_acc-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
