{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c31a0c93",
   "metadata": {},
   "source": [
    "# KEGG enrichment of stable genes\n",
    "\n",
    "This notebooks looks at the group of most and least stable genes and performs a KEGG enrichment analysis to determine if there are any KEGG pathways that are significantly over-represented in our most or least stable gene sets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7d4fd6b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "%matplotlib inline\n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "from scripts import paths, utils, annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3256ca93",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load KEGG pathway data\n",
    "pao1_pathway_filename = \"https://raw.githubusercontent.com/greenelab/adage/7a4eda39d360b224268921dc1f2c14b32788ab16/Node_interpretation/pseudomonas_KEGG_terms.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "57366d1f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KEGG-Pathway-pae00072: Synthesis and degradation of ketone bodies</th>\n",
       "      <td>10</td>\n",
       "      <td>{PA3925, PA2003, PA2553, PA2001, PA3589, PA478...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KEGG-Pathway-pae00071: Fatty acid degradation</th>\n",
       "      <td>32</td>\n",
       "      <td>{PA3454, PA0879, PA4994, PA1748, PA2574, PA118...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KEGG-Pathway-pae00903: Limonene and pinene degradation</th>\n",
       "      <td>9</td>\n",
       "      <td>{PA3331, PA1748, PA1027, PA3014, PA4899, PA182...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KEGG-Pathway-pae00380: Tryptophan metabolism</th>\n",
       "      <td>27</td>\n",
       "      <td>{PA2080, PA1748, PA3589, PA3426, PA1736, PA208...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KEGG-Pathway-pae00900: Terpenoid backbone biosynthesis</th>\n",
       "      <td>16</td>\n",
       "      <td>{PA4557, PA3650, PA4044, PA4669, PA3652, PA392...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     1  \\\n",
       "0                                                        \n",
       "KEGG-Pathway-pae00072: Synthesis and degradatio...  10   \n",
       "KEGG-Pathway-pae00071: Fatty acid degradation       32   \n",
       "KEGG-Pathway-pae00903: Limonene and pinene degr...   9   \n",
       "KEGG-Pathway-pae00380: Tryptophan metabolism        27   \n",
       "KEGG-Pathway-pae00900: Terpenoid backbone biosy...  16   \n",
       "\n",
       "                                                                                                    2  \n",
       "0                                                                                                      \n",
       "KEGG-Pathway-pae00072: Synthesis and degradatio...  {PA3925, PA2003, PA2553, PA2001, PA3589, PA478...  \n",
       "KEGG-Pathway-pae00071: Fatty acid degradation       {PA3454, PA0879, PA4994, PA1748, PA2574, PA118...  \n",
       "KEGG-Pathway-pae00903: Limonene and pinene degr...  {PA3331, PA1748, PA1027, PA3014, PA4899, PA182...  \n",
       "KEGG-Pathway-pae00380: Tryptophan metabolism        {PA2080, PA1748, PA3589, PA3426, PA1736, PA208...  \n",
       "KEGG-Pathway-pae00900: Terpenoid backbone biosy...  {PA4557, PA3650, PA4044, PA4669, PA3652, PA392...  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pao1_pathways = annotations.load_format_KEGG(pao1_pathway_filename)\n",
    "print(pao1_pathways.shape)\n",
    "pao1_pathways.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e6ca29b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load transcriptional similarity df\n",
    "# These are the subset of genes that we will consider\n",
    "pao1_similarity_scores_filename = \"pao1_similarity_scores.tsv\"\n",
    "\n",
    "pao1_similarity_scores = pd.read_csv(\n",
    "    pao1_similarity_scores_filename, sep=\"\\t\", header=0, index_col=0\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d04c5f06",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PA14 homolog id</th>\n",
       "      <th>Transcriptional similarity across strains</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Name</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAO1 id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PA1950</th>\n",
       "      <td>PA14_39280</td>\n",
       "      <td>0.353718</td>\n",
       "      <td>1.962739e-157</td>\n",
       "      <td>rbsK</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PA4872</th>\n",
       "      <td>PA14_64440</td>\n",
       "      <td>0.381711</td>\n",
       "      <td>4.085891e-185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PA3680</th>\n",
       "      <td>PA14_16770</td>\n",
       "      <td>0.381384</td>\n",
       "      <td>8.943350e-185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PA0586</th>\n",
       "      <td>PA14_07650</td>\n",
       "      <td>0.542184</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>most stable</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PA1468</th>\n",
       "      <td>PA14_45450</td>\n",
       "      <td>0.365443</td>\n",
       "      <td>1.080005e-168</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        PA14 homolog id  Transcriptional similarity across strains  \\\n",
       "PAO1 id                                                              \n",
       "PA1950       PA14_39280                                   0.353718   \n",
       "PA4872       PA14_64440                                   0.381711   \n",
       "PA3680       PA14_16770                                   0.381384   \n",
       "PA0586       PA14_07650                                   0.542184   \n",
       "PA1468       PA14_45450                                   0.365443   \n",
       "\n",
       "               P-value  Name        label  \n",
       "PAO1 id                                    \n",
       "PA1950   1.962739e-157  rbsK          NaN  \n",
       "PA4872   4.085891e-185   NaN          NaN  \n",
       "PA3680   8.943350e-185   NaN          NaN  \n",
       "PA0586    0.000000e+00   NaN  most stable  \n",
       "PA1468   1.080005e-168   NaN          NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pao1_similarity_scores.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "fba64fd2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get most and least stable core genes based on label\n",
    "pao1_most_stable_genes = list(\n",
    "    pao1_similarity_scores[pao1_similarity_scores[\"label\"] == \"most stable\"].index\n",
    ")\n",
    "pao1_least_stable_genes = list(\n",
    "    pao1_similarity_scores[pao1_similarity_scores[\"label\"] == \"least stable\"].index\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "51ab5e82",
   "metadata": {},
   "outputs": [],
   "source": [
    "# For each KEGG pathway, perform stat test, save p-values to get corrected p-values, report stats per pathway\n",
    "def KEGG_enrichment_of_stable_genes(similarity_score_df, gene_list, kegg_df):\n",
    "    \"\"\"\n",
    "    This function performs a KEGG enrichment using most or least stable genes,\n",
    "    provided in `gene_list`\n",
    "    \"\"\"\n",
    "\n",
    "    all_genes = set(similarity_score_df.index)\n",
    "    module_genes = set(gene_list)\n",
    "    not_module_genes = all_genes.difference(module_genes)\n",
    "\n",
    "    rows = []\n",
    "    # Find the KEGG pathway with significant over-representation\n",
    "    for kegg_name in kegg_df.index:\n",
    "        num_kegg_genes = kegg_df.loc[kegg_name, 1]\n",
    "        kegg_genes = set(kegg_df.loc[kegg_name, 2])\n",
    "        not_kegg_genes = all_genes.difference(kegg_genes)\n",
    "\n",
    "        # Make contingency table\n",
    "        # ---------------------| most stable  | not most stable\n",
    "        # in KEGG pathway      | # genes      | # genes\n",
    "        # not in KEGG pathway  | # genes     | # genes\n",
    "        module_kegg_genes = module_genes.intersection(kegg_genes)\n",
    "        not_module_kegg_genes = not_module_genes.intersection(kegg_genes)\n",
    "        module_not_kegg_genes = module_genes.intersection(not_kegg_genes)\n",
    "        not_module_not_kegg_genes = not_module_genes.intersection(not_kegg_genes)\n",
    "\n",
    "        observed_contingency_table = np.array(\n",
    "            [\n",
    "                [len(module_kegg_genes), len(not_module_kegg_genes)],\n",
    "                [len(module_not_kegg_genes), len(not_module_not_kegg_genes)],\n",
    "            ]\n",
    "        )\n",
    "        # Fisher's exact test\n",
    "        oddsr, pval = scipy.stats.fisher_exact(\n",
    "            observed_contingency_table, alternative=\"greater\"\n",
    "        )\n",
    "        # chi2 test will not accept 0 counts for the contingency table\n",
    "        # chi2, pval, dof, expected_counts = scipy.stats.chi2_contingency(\n",
    "        #    observed_contingency_table\n",
    "        # )\n",
    "        # print(oddsr, pval)\n",
    "\n",
    "        rows.append(\n",
    "            {\n",
    "                \"enriched KEGG pathway\": kegg_name,\n",
    "                \"odds ratio\": oddsr,\n",
    "                \"p-value\": pval,\n",
    "                \"num shared genes\": len(module_kegg_genes),\n",
    "                \"size gene set\": len(module_genes),\n",
    "                \"size KEGG pathway\": num_kegg_genes,\n",
    "            }\n",
    "        )\n",
    "\n",
    "    enrichment_df = pd.DataFrame(rows)\n",
    "\n",
    "    # Get corrected pvalues\n",
    "    (\n",
    "        reject_,\n",
    "        pvals_corrected_,\n",
    "        alphacSidak,\n",
    "        alphacBonf,\n",
    "    ) = statsmodels.stats.multitest.multipletests(\n",
    "        enrichment_df[\"p-value\"].values,\n",
    "        alpha=0.05,\n",
    "        method=\"fdr_bh\",\n",
    "        is_sorted=False,\n",
    "    )\n",
    "\n",
    "    enrichment_df[\"corrected p-value\"] = pvals_corrected_\n",
    "\n",
    "    return enrichment_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "68639ce9",
   "metadata": {},
   "outputs": [],
   "source": [
    "pao1_most_stable_enrichment = KEGG_enrichment_of_stable_genes(\n",
    "    pao1_similarity_scores, pao1_most_stable_genes, pao1_pathways\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ab49683c",
   "metadata": {},
   "outputs": [],
   "source": [
    "pao1_least_stable_enrichment = KEGG_enrichment_of_stable_genes(\n",
    "    pao1_similarity_scores, pao1_least_stable_genes, pao1_pathways\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "d4d6967d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 7)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>enriched KEGG pathway</th>\n",
       "      <th>odds ratio</th>\n",
       "      <th>p-value</th>\n",
       "      <th>num shared genes</th>\n",
       "      <th>size gene set</th>\n",
       "      <th>size KEGG pathway</th>\n",
       "      <th>corrected p-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>KEGG-Module-M00178: Ribosome, bacteria</td>\n",
       "      <td>24.968492</td>\n",
       "      <td>3.834788e-28</td>\n",
       "      <td>38</td>\n",
       "      <td>473</td>\n",
       "      <td>56</td>\n",
       "      <td>3.240396e-26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>KEGG-Pathway-pae03010: Ribosome</td>\n",
       "      <td>24.968492</td>\n",
       "      <td>3.834788e-28</td>\n",
       "      <td>38</td>\n",
       "      <td>473</td>\n",
       "      <td>68</td>\n",
       "      <td>3.240396e-26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>KEGG-Pathway-pae03070: Bacterial secretion system</td>\n",
       "      <td>4.859193</td>\n",
       "      <td>2.662838e-09</td>\n",
       "      <td>27</td>\n",
       "      <td>473</td>\n",
       "      <td>92</td>\n",
       "      <td>1.500066e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>KEGG-Module-M00332: Type III secretion system</td>\n",
       "      <td>16.561224</td>\n",
       "      <td>4.148334e-08</td>\n",
       "      <td>11</td>\n",
       "      <td>473</td>\n",
       "      <td>18</td>\n",
       "      <td>1.752671e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>KEGG-Module-M00334: Type VI secretion system</td>\n",
       "      <td>6.793523</td>\n",
       "      <td>1.398506e-07</td>\n",
       "      <td>16</td>\n",
       "      <td>473</td>\n",
       "      <td>45</td>\n",
       "      <td>4.726952e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 enriched KEGG pathway  odds ratio  \\\n",
       "168             KEGG-Module-M00178: Ribosome, bacteria   24.968492   \n",
       "50                     KEGG-Pathway-pae03010: Ribosome   24.968492   \n",
       "32   KEGG-Pathway-pae03070: Bacterial secretion system    4.859193   \n",
       "114      KEGG-Module-M00332: Type III secretion system   16.561224   \n",
       "112       KEGG-Module-M00334: Type VI secretion system    6.793523   \n",
       "\n",
       "          p-value  num shared genes  size gene set  size KEGG pathway  \\\n",
       "168  3.834788e-28                38            473                 56   \n",
       "50   3.834788e-28                38            473                 68   \n",
       "32   2.662838e-09                27            473                 92   \n",
       "114  4.148334e-08                11            473                 18   \n",
       "112  1.398506e-07                16            473                 45   \n",
       "\n",
       "     corrected p-value  \n",
       "168       3.240396e-26  \n",
       "50        3.240396e-26  \n",
       "32        1.500066e-07  \n",
       "114       1.752671e-06  \n",
       "112       4.726952e-06  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(pao1_most_stable_enrichment.shape)\n",
    "pao1_most_stable_enrichment.sort_values(by=\"corrected p-value\").head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "df5cf0aa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 7)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>enriched KEGG pathway</th>\n",
       "      <th>odds ratio</th>\n",
       "      <th>p-value</th>\n",
       "      <th>num shared genes</th>\n",
       "      <th>size gene set</th>\n",
       "      <th>size KEGG pathway</th>\n",
       "      <th>corrected p-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KEGG-Pathway-pae00072: Synthesis and degradati...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>201</td>\n",
       "      <td>10</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>KEGG-Module-M00299: Spermidine/putrescine tran...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>201</td>\n",
       "      <td>12</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>KEGG-Module-M00222: Phosphate transport system</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>201</td>\n",
       "      <td>5</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>KEGG-Module-M00053: Pyrimidine deoxyribonuleot...</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>201</td>\n",
       "      <td>9</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>KEGG-Module-M00050: Guanine ribonucleotide bio...</td>\n",
       "      <td>5.143</td>\n",
       "      <td>0.205401</td>\n",
       "      <td>1</td>\n",
       "      <td>201</td>\n",
       "      <td>6</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 enriched KEGG pathway  odds ratio   p-value  \\\n",
       "0    KEGG-Pathway-pae00072: Synthesis and degradati...       0.000  1.000000   \n",
       "107  KEGG-Module-M00299: Spermidine/putrescine tran...       0.000  1.000000   \n",
       "108     KEGG-Module-M00222: Phosphate transport system       0.000  1.000000   \n",
       "109  KEGG-Module-M00053: Pyrimidine deoxyribonuleot...       0.000  1.000000   \n",
       "110  KEGG-Module-M00050: Guanine ribonucleotide bio...       5.143  0.205401   \n",
       "\n",
       "     num shared genes  size gene set  size KEGG pathway  corrected p-value  \n",
       "0                   0            201                 10                1.0  \n",
       "107                 0            201                 12                1.0  \n",
       "108                 0            201                  5                1.0  \n",
       "109                 0            201                  9                1.0  \n",
       "110                 1            201                  6                1.0  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(pao1_least_stable_enrichment.shape)\n",
    "pao1_least_stable_enrichment.sort_values(by=\"corrected p-value\").head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "cc42c147",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save\n",
    "pao1_most_stable_enrichment.to_csv(\"pao1_most_stable_enrichment.tsv\", sep=\"\\t\")\n",
    "pao1_least_stable_enrichment.to_csv(\"pao1_least_stable_enrichment.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "00cb6dbc",
   "metadata": {},
   "source": [
    "**Takeaway:**\n",
    "* There does not appear to be any enriched KEGG pathways in the least stable genes.\n",
    "    * What does this mean about the role of these least stable core genes? Maybe they are spread across multiple pathways?\n",
    "    * Based on the dataframe created in the [previous notebook](2_find_KEGG_associations.ipynb) like many least stable core genes are not found in any KEGG pathway, but there are some that are found in many KEGG pathways: https://docs.google.com/spreadsheets/d/1SqEyBvutfbsOTo4afg9GiEzP32ZKplkN1a6MpAQBvZI/edit#gid=1943176121\n",
    "* The most stable core genes are significantly enriched KEGG pathways include Ribosome (commonly enriched in humans), secretion system, metabolism/Krebs cycle\n",
    "    * These KEGG pathways represent some of the essential functions for Pa, so it makes sense that they are enriched amongst the set of stable core genes whose transcriptional relationships don’t vary across strains.\n",
    "    * Only some metabolisms and not others, is that interesting?"
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "encoding": "# -*- coding: utf-8 -*-",
   "formats": "ipynb,py"
  },
  "kernelspec": {
   "display_name": "Python [conda env:core_acc] *",
   "language": "python",
   "name": "conda-env-core_acc-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
