{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c31a0c93",
   "metadata": {},
   "source": [
    "# KEGG enrichment of stable genes\n",
    "\n",
    "This notebooks looks at the group of most and least stable genes and performs a KEGG enrichment analysis to determine if there are any KEGG pathways that are significantly over-represented in our most or least stable gene sets."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7d4fd6b3",
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "%matplotlib inline\n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats\n",
    "import statsmodels.stats.multitest\n",
    "from scripts import paths, utils, annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3256ca93",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load KEGG pathway data\n",
    "pao1_pathway_filename = \"https://raw.githubusercontent.com/greenelab/adage/7a4eda39d360b224268921dc1f2c14b32788ab16/Node_interpretation/pseudomonas_KEGG_terms.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "57366d1f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>KEGG-Pathway-pae00072: Synthesis and degradation of ketone bodies</th>\n",
       "      <td>10</td>\n",
       "      <td>{PA4785, PA2003, PA1999, PA3589, PA2000, PA200...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KEGG-Pathway-pae00071: Fatty acid degradation</th>\n",
       "      <td>32</td>\n",
       "      <td>{PA4785, PA2815, PA4994, PA5427, PA1525, PA534...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KEGG-Pathway-pae00903: Limonene and pinene degradation</th>\n",
       "      <td>9</td>\n",
       "      <td>{PA1737, PA1748, PA4899, PA2475, PA3014, PA333...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KEGG-Pathway-pae00380: Tryptophan metabolism</th>\n",
       "      <td>27</td>\n",
       "      <td>{PA4785, PA2081, PA4163, PA3589, PA2001, PA102...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KEGG-Pathway-pae00900: Terpenoid backbone biosynthesis</th>\n",
       "      <td>16</td>\n",
       "      <td>{PA4557, PA4785, PA3803, PA3627, PA4669, PA456...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     1  \\\n",
       "0                                                        \n",
       "KEGG-Pathway-pae00072: Synthesis and degradatio...  10   \n",
       "KEGG-Pathway-pae00071: Fatty acid degradation       32   \n",
       "KEGG-Pathway-pae00903: Limonene and pinene degr...   9   \n",
       "KEGG-Pathway-pae00380: Tryptophan metabolism        27   \n",
       "KEGG-Pathway-pae00900: Terpenoid backbone biosy...  16   \n",
       "\n",
       "                                                                                                    2  \n",
       "0                                                                                                      \n",
       "KEGG-Pathway-pae00072: Synthesis and degradatio...  {PA4785, PA2003, PA1999, PA3589, PA2000, PA200...  \n",
       "KEGG-Pathway-pae00071: Fatty acid degradation       {PA4785, PA2815, PA4994, PA5427, PA1525, PA534...  \n",
       "KEGG-Pathway-pae00903: Limonene and pinene degr...  {PA1737, PA1748, PA4899, PA2475, PA3014, PA333...  \n",
       "KEGG-Pathway-pae00380: Tryptophan metabolism        {PA4785, PA2081, PA4163, PA3589, PA2001, PA102...  \n",
       "KEGG-Pathway-pae00900: Terpenoid backbone biosy...  {PA4557, PA4785, PA3803, PA3627, PA4669, PA456...  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pao1_pathways = annotations.load_format_KEGG(pao1_pathway_filename)\n",
    "print(pao1_pathways.shape)\n",
    "pao1_pathways.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e6ca29b6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load transcriptional similarity df\n",
    "# These are the subset of genes that we will consider\n",
    "pao1_similarity_scores_filename = \"pao1_similarity_scores.tsv\"\n",
    "\n",
    "pao1_similarity_scores = pd.read_csv(\n",
    "    pao1_similarity_scores_filename, sep=\"\\t\", header=0, index_col=0\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d04c5f06",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PA14 homolog id</th>\n",
       "      <th>Transcriptional similarity across strains</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Name</th>\n",
       "      <th>label</th>\n",
       "      <th>comparison</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAO1 id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PA0984</th>\n",
       "      <td>PA14_59230</td>\n",
       "      <td>0.469691</td>\n",
       "      <td>9.403422e-292</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PA0680</th>\n",
       "      <td>PA14_55500</td>\n",
       "      <td>0.672592</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>hxcV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PA0847</th>\n",
       "      <td>PA14_53310</td>\n",
       "      <td>0.767508</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PA0164</th>\n",
       "      <td>PA14_02050</td>\n",
       "      <td>0.703763</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PA1160</th>\n",
       "      <td>PA14_49400</td>\n",
       "      <td>0.228437</td>\n",
       "      <td>2.819471e-64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        PA14 homolog id  Transcriptional similarity across strains  \\\n",
       "PAO1 id                                                              \n",
       "PA0984       PA14_59230                                   0.469691   \n",
       "PA0680       PA14_55500                                   0.672592   \n",
       "PA0847       PA14_53310                                   0.767508   \n",
       "PA0164       PA14_02050                                   0.703763   \n",
       "PA1160       PA14_49400                                   0.228437   \n",
       "\n",
       "               P-value  Name label comparison  \n",
       "PAO1 id                                        \n",
       "PA0984   9.403422e-292   NaN   NaN        NaN  \n",
       "PA0680    0.000000e+00  hxcV   NaN        NaN  \n",
       "PA0847    0.000000e+00   NaN   NaN        NaN  \n",
       "PA0164    0.000000e+00   NaN   NaN        NaN  \n",
       "PA1160    2.819471e-64   NaN   NaN        NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pao1_similarity_scores.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "fba64fd2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get most and least stable core genes based on label\n",
    "pao1_most_stable_genes = list(\n",
    "    pao1_similarity_scores[pao1_similarity_scores[\"label\"] == \"most stable\"].index\n",
    ")\n",
    "pao1_least_stable_genes = list(\n",
    "    pao1_similarity_scores[pao1_similarity_scores[\"label\"] == \"least stable\"].index\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "51ab5e82",
   "metadata": {},
   "outputs": [],
   "source": [
    "# For each KEGG pathway, perform stat test, save p-values to get corrected p-values, report stats per pathway\n",
    "def KEGG_enrichment_of_stable_genes(similarity_score_df, gene_list, kegg_df):\n",
    "    \"\"\"\n",
    "    This function performs a KEGG enrichment using most or least stable genes,\n",
    "    provided in `gene_list`\n",
    "    \"\"\"\n",
    "\n",
    "    all_genes = set(similarity_score_df.index)\n",
    "    module_genes = set(gene_list)\n",
    "    not_module_genes = all_genes.difference(module_genes)\n",
    "\n",
    "    rows = []\n",
    "    # Find the KEGG pathway with significant over-representation\n",
    "    for kegg_name in kegg_df.index:\n",
    "        num_kegg_genes = kegg_df.loc[kegg_name, 1]\n",
    "        kegg_genes = set(kegg_df.loc[kegg_name, 2])\n",
    "        not_kegg_genes = all_genes.difference(kegg_genes)\n",
    "\n",
    "        # Make contingency table\n",
    "        # ---------------------| most stable  | not most stable\n",
    "        # in KEGG pathway      | # genes      | # genes\n",
    "        # not in KEGG pathway  | # genes     | # genes\n",
    "        module_kegg_genes = module_genes.intersection(kegg_genes)\n",
    "        not_module_kegg_genes = not_module_genes.intersection(kegg_genes)\n",
    "        module_not_kegg_genes = module_genes.intersection(not_kegg_genes)\n",
    "        not_module_not_kegg_genes = not_module_genes.intersection(not_kegg_genes)\n",
    "\n",
    "        observed_contingency_table = np.array(\n",
    "            [\n",
    "                [len(module_kegg_genes), len(not_module_kegg_genes)],\n",
    "                [len(module_not_kegg_genes), len(not_module_not_kegg_genes)],\n",
    "            ]\n",
    "        )\n",
    "        # Fisher's exact test\n",
    "        oddsr, pval = scipy.stats.fisher_exact(\n",
    "            observed_contingency_table, alternative=\"greater\"\n",
    "        )\n",
    "        # chi2 test will not accept 0 counts for the contingency table\n",
    "        # chi2, pval, dof, expected_counts = scipy.stats.chi2_contingency(\n",
    "        #    observed_contingency_table\n",
    "        # )\n",
    "        # print(oddsr, pval)\n",
    "\n",
    "        rows.append(\n",
    "            {\n",
    "                \"enriched KEGG pathway\": kegg_name,\n",
    "                \"odds ratio\": oddsr,\n",
    "                \"p-value\": pval,\n",
    "                \"num shared genes\": len(module_kegg_genes),\n",
    "                \"size gene set\": len(module_genes),\n",
    "                \"size KEGG pathway\": num_kegg_genes,\n",
    "            }\n",
    "        )\n",
    "\n",
    "    enrichment_df = pd.DataFrame(rows)\n",
    "\n",
    "    # Get corrected pvalues\n",
    "    (\n",
    "        reject_,\n",
    "        pvals_corrected_,\n",
    "        alphacSidak,\n",
    "        alphacBonf,\n",
    "    ) = statsmodels.stats.multitest.multipletests(\n",
    "        enrichment_df[\"p-value\"].values,\n",
    "        alpha=0.05,\n",
    "        method=\"fdr_bh\",\n",
    "        is_sorted=False,\n",
    "    )\n",
    "\n",
    "    enrichment_df[\"corrected p-value\"] = pvals_corrected_\n",
    "\n",
    "    return enrichment_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "68639ce9",
   "metadata": {},
   "outputs": [],
   "source": [
    "pao1_most_stable_enrichment = KEGG_enrichment_of_stable_genes(\n",
    "    pao1_similarity_scores, pao1_most_stable_genes, pao1_pathways\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ab49683c",
   "metadata": {},
   "outputs": [],
   "source": [
    "pao1_least_stable_enrichment = KEGG_enrichment_of_stable_genes(\n",
    "    pao1_similarity_scores, pao1_least_stable_genes, pao1_pathways\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "d4d6967d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 7)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>enriched KEGG pathway</th>\n",
       "      <th>odds ratio</th>\n",
       "      <th>p-value</th>\n",
       "      <th>num shared genes</th>\n",
       "      <th>size gene set</th>\n",
       "      <th>size KEGG pathway</th>\n",
       "      <th>corrected p-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>KEGG-Module-M00178: Ribosome, bacteria</td>\n",
       "      <td>11.948434</td>\n",
       "      <td>8.019412e-17</td>\n",
       "      <td>29</td>\n",
       "      <td>481</td>\n",
       "      <td>56</td>\n",
       "      <td>6.776403e-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>KEGG-Pathway-pae03010: Ribosome</td>\n",
       "      <td>11.948434</td>\n",
       "      <td>8.019412e-17</td>\n",
       "      <td>29</td>\n",
       "      <td>481</td>\n",
       "      <td>68</td>\n",
       "      <td>6.776403e-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>KEGG-Pathway-pae00970: Aminoacyl-tRNA biosynth...</td>\n",
       "      <td>13.025751</td>\n",
       "      <td>1.037838e-09</td>\n",
       "      <td>15</td>\n",
       "      <td>481</td>\n",
       "      <td>90</td>\n",
       "      <td>5.846488e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>KEGG-Module-M00360: Aminoacyl-tRNA biosynthesi...</td>\n",
       "      <td>14.996914</td>\n",
       "      <td>5.020267e-09</td>\n",
       "      <td>13</td>\n",
       "      <td>481</td>\n",
       "      <td>22</td>\n",
       "      <td>2.121063e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>KEGG-Pathway-pae03060: Protein export</td>\n",
       "      <td>11.583686</td>\n",
       "      <td>4.504319e-06</td>\n",
       "      <td>9</td>\n",
       "      <td>481</td>\n",
       "      <td>18</td>\n",
       "      <td>1.522460e-04</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 enriched KEGG pathway  odds ratio  \\\n",
       "168             KEGG-Module-M00178: Ribosome, bacteria   11.948434   \n",
       "50                     KEGG-Pathway-pae03010: Ribosome   11.948434   \n",
       "61   KEGG-Pathway-pae00970: Aminoacyl-tRNA biosynth...   13.025751   \n",
       "162  KEGG-Module-M00360: Aminoacyl-tRNA biosynthesi...   14.996914   \n",
       "38               KEGG-Pathway-pae03060: Protein export   11.583686   \n",
       "\n",
       "          p-value  num shared genes  size gene set  size KEGG pathway  \\\n",
       "168  8.019412e-17                29            481                 56   \n",
       "50   8.019412e-17                29            481                 68   \n",
       "61   1.037838e-09                15            481                 90   \n",
       "162  5.020267e-09                13            481                 22   \n",
       "38   4.504319e-06                 9            481                 18   \n",
       "\n",
       "     corrected p-value  \n",
       "168       6.776403e-15  \n",
       "50        6.776403e-15  \n",
       "61        5.846488e-08  \n",
       "162       2.121063e-07  \n",
       "38        1.522460e-04  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(pao1_most_stable_enrichment.shape)\n",
    "pao1_most_stable_enrichment.sort_values(by=\"corrected p-value\").head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "df5cf0aa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 7)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>enriched KEGG pathway</th>\n",
       "      <th>odds ratio</th>\n",
       "      <th>p-value</th>\n",
       "      <th>num shared genes</th>\n",
       "      <th>size gene set</th>\n",
       "      <th>size KEGG pathway</th>\n",
       "      <th>corrected p-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>KEGG-Module-M00452: CusS-CusR (copper toleranc...</td>\n",
       "      <td>14.587678</td>\n",
       "      <td>0.003046</td>\n",
       "      <td>3</td>\n",
       "      <td>214</td>\n",
       "      <td>8</td>\n",
       "      <td>0.514792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>KEGG-Module-M00193: Putative spermidine/putres...</td>\n",
       "      <td>8.097946</td>\n",
       "      <td>0.010628</td>\n",
       "      <td>3</td>\n",
       "      <td>214</td>\n",
       "      <td>12</td>\n",
       "      <td>0.898051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>KEGG-Module-M00222: Phosphate transport system</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>214</td>\n",
       "      <td>5</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>KEGG-Module-M00053: Pyrimidine deoxyribonuleot...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>214</td>\n",
       "      <td>9</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>KEGG-Module-M00050: Guanine ribonucleotide bio...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>214</td>\n",
       "      <td>6</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 enriched KEGG pathway  odds ratio   p-value  \\\n",
       "106  KEGG-Module-M00452: CusS-CusR (copper toleranc...   14.587678  0.003046   \n",
       "92   KEGG-Module-M00193: Putative spermidine/putres...    8.097946  0.010628   \n",
       "108     KEGG-Module-M00222: Phosphate transport system    0.000000  1.000000   \n",
       "109  KEGG-Module-M00053: Pyrimidine deoxyribonuleot...    0.000000  1.000000   \n",
       "110  KEGG-Module-M00050: Guanine ribonucleotide bio...    0.000000  1.000000   \n",
       "\n",
       "     num shared genes  size gene set  size KEGG pathway  corrected p-value  \n",
       "106                 3            214                  8           0.514792  \n",
       "92                  3            214                 12           0.898051  \n",
       "108                 0            214                  5           1.000000  \n",
       "109                 0            214                  9           1.000000  \n",
       "110                 0            214                  6           1.000000  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(pao1_least_stable_enrichment.shape)\n",
    "pao1_least_stable_enrichment.sort_values(by=\"corrected p-value\").head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "cc42c147",
   "metadata": {},
   "outputs": [],
   "source": [
    "# TO DO: Remove 'compare' when we decide which input to use\n",
    "# Save\n",
    "pao1_most_stable_enrichment.to_csv(\"pao1_most_stable_enrichment_compare.tsv\", sep=\"\\t\")\n",
    "pao1_least_stable_enrichment.to_csv(\n",
    "    \"pao1_least_stable_enrichment_compare.tsv\", sep=\"\\t\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "00cb6dbc",
   "metadata": {},
   "source": [
    "**Takeaway:**\n",
    "* There does not appear to be any enriched KEGG pathways in the least stable genes.\n",
    "    * What does this mean about the role of these least stable core genes? Maybe they are spread across multiple pathways?\n",
    "    * Based on the dataframe created in the [previous notebook](2_find_KEGG_associations.ipynb) like many least stable core genes are not found in any KEGG pathway, but there are some that are found in many KEGG pathways: https://docs.google.com/spreadsheets/d/1SqEyBvutfbsOTo4afg9GiEzP32ZKplkN1a6MpAQBvZI/edit#gid=1943176121\n",
    "* The most stable core genes are significantly enriched KEGG pathways include Ribosome (commonly enriched in humans), secretion system, metabolism/Krebs cycle\n",
    "    * These KEGG pathways represent some of the essential functions for Pa, so it makes sense that they are enriched amongst the set of stable core genes whose transcriptional relationships donâ€™t vary across strains.\n",
    "    * Only some metabolisms and not others, is that interesting?"
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "encoding": "# -*- coding: utf-8 -*-",
   "formats": "ipynb,py"
  },
  "kernelspec": {
   "display_name": "Python [conda env:core_acc] *",
   "language": "python",
   "name": "conda-env-core_acc-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
