{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Align and quantify samples\n",
    "\n",
    "This notebook aligns test samples against the phage and PAO1 reference genomes. Our goal is to test our phage reference genome alignment before we port it to the Discovery (Dartmouth computing cluster). We want to check that we are getting more expression of phage genes in a phage sample compared to a non-pseudomonas samples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The autoreload extension is already loaded. To reload it, use:\n",
      "  %reload_ext autoreload\n"
     ]
    }
   ],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "import os\n",
    "import shutil\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from core_acc_modules import paths\n",
    "\n",
    "np.random.seed(123)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Download SRA data\n",
    "\n",
    "Note: Need to delete `sra` folder between runs otherwise `fastq-dump` will be called on all files in `sra` folder which can include more than your sra accessions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "shutil.rmtree(paths.SRA_DIR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2020-12-16T20:11:58 prefetch.2.8.2: 1) Downloading 'SRR12922100'...\n",
      "2020-12-16T20:11:58 prefetch.2.8.2:  Downloading via https...\n",
      "2020-12-16T20:13:38 prefetch.2.8.2: 1) 'SRR12922100' was downloaded successfully\n",
      "2020-12-16T20:13:38 prefetch.2.8.2: 'SRR12922100' has 0 unresolved dependencies\n",
      "\n",
      "2020-12-16T20:13:38 prefetch.2.8.2: 2) Downloading 'SRR13196071'...\n",
      "2020-12-16T20:13:38 prefetch.2.8.2:  Downloading via https...\n",
      "2020-12-16T20:13:55 prefetch.2.8.2: 2) 'SRR13196071' was downloaded successfully\n",
      "2020-12-16T20:13:55 prefetch.2.8.2: 'SRR13196071' has 0 unresolved dependencies\n",
      "\n",
      "2020-12-16T20:13:56 prefetch.2.8.2: 3) Downloading 'SRR13196068'...\n",
      "2020-12-16T20:13:56 prefetch.2.8.2:  Downloading via https...\n",
      "2020-12-16T20:13:57 prefetch.2.8.2: 3) 'SRR13196068' was downloaded successfully\n",
      "2020-12-16T20:13:57 prefetch.2.8.2: 'SRR13196068' has 0 unresolved dependencies\n",
      "\n",
      "2020-12-16T20:13:58 prefetch.2.8.2: 4) Downloading 'SRR11809598'...\n",
      "2020-12-16T20:13:58 prefetch.2.8.2:  Downloading via https...\n",
      "2020-12-16T20:15:29 prefetch.2.8.2: 4) 'SRR11809598' was downloaded successfully\n",
      "2020-12-16T20:15:29 prefetch.2.8.2: 'SRR11809598' has 0 unresolved dependencies\n"
     ]
    }
   ],
   "source": [
    "# Download sra data files\n",
    "! prefetch --option-file $paths.SRA_ACC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get FASTQ files associated with SRA downloads\n",
    "\n",
    "The fastq files store the RNA-seq results, including: sequencing and quality scores for each base call.\n",
    "\n",
    "Here is a nice blog to explain how to read fastq files: https://thesequencingcenter.com/knowledge-base/fastq-files/\n",
    "\n",
    "The fastq files gives the sequence of a read at a given location. Our goal is to map these reads to a reference genome so that we can quantify the number of reads that are at a given location, to determine the level of expression."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/home/alexandra/ncbi/public/fastq_phage’: File exists\r\n"
     ]
    }
   ],
   "source": [
    "if not os.path.exists(paths.FASTQ_DIR):\n",
    "    os.makedirs(paths.FASTQ_DIR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read 2101445 spots for /home/alexandra/ncbi/public/sra/SRR11809598.sra\n",
      "Written 2101445 spots for /home/alexandra/ncbi/public/sra/SRR11809598.sra\n",
      "Read 3686954 spots for /home/alexandra/ncbi/public/sra/SRR12922100.sra\n",
      "Written 3686954 spots for /home/alexandra/ncbi/public/sra/SRR12922100.sra\n",
      "Read 1089559 spots for /home/alexandra/ncbi/public/sra/SRR13196068.sra\n",
      "Written 1089559 spots for /home/alexandra/ncbi/public/sra/SRR13196068.sra\n",
      "Read 629579 spots for /home/alexandra/ncbi/public/sra/SRR13196071.sra\n",
      "Written 629579 spots for /home/alexandra/ncbi/public/sra/SRR13196071.sra\n",
      "Read 7507537 spots total\n",
      "Written 7507537 spots total\n"
     ]
    }
   ],
   "source": [
    "!fastq-dump $paths.SRA_DIR/* --split-files --outdir $paths.FASTQ_DIR/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Copied from https://github.com/hoganlab-dartmouth/sraProcessingPipeline/blob/5974e040c85724a8d385e53153b7707ae7c9c255/DiscoveryScripts/quantifier.py#L83\n",
    "\n",
    "#!fastq-dump $paths_phage.SRA_DIR/* --skip-technical --readids --split-3 --clip --outdir $paths_phage.FASTQ_DIR/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Quantify gene expression\n",
    "Now that we have our index built and all of our data downloaded, we’re ready to quantify our samples\n",
    "\n",
    "**Input:**\n",
    "* Index of reference transcriptome\n",
    "* FASTQ of experimental samples\n",
    "\n",
    "**Output:**\n",
    "\n",
    "After the salmon commands finish running, you should have a directory named quants, which will have a sub-directory for each sample. These sub-directories contain the quantification results of salmon, as well as a lot of other information salmon records about the sample and the run. \n",
    "\n",
    "The main output file (called `quant.sf`). Inside the quantification file for sample DRR016125 in quants/DRR016125/quant.sf, you’ll see a TSV format file listing the name (`Name`) of each transcript, its length (`Length`), effective length (`EffectiveLength`), and its abundance in terms of Transcripts Per Million (`TPM`) and estimated number of reads (`NumReads`) originating from this transcript.\n",
    "\n",
    "**For each sample we have read counts per gene (where the genes are based on the reference gene file provided above).** "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get quants using PAO1 reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.exists(paths.PAO1_QUANT):\n",
    "    os.makedirs(paths.PAO1_QUANT)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing sample SRR11809598\n",
      "Processing sample SRR11809598\n",
      "Processing sample SRR12922100\n",
      "Processing sample SRR12922100\n",
      "Processing sample SRR13196068\n",
      "Processing sample SRR13196068\n",
      "Processing sample SRR13196071\n",
      "Processing sample SRR13196071\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/pao1_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR11809598_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR11809598_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_pao1/SRR11809598_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_pao1/SRR11809598_quant/logs\n",
      "[2020-12-16 15:20:11.957] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:20:11.957] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:20:11.957] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:20:11.957] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:20:11.957] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:20:11.957] [jointLog] [info] There is 1 library.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 766.05 us\n",
      "-----------------------------------------\n",
      "size = 7452\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 29.439 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 17.703 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 471.09 us\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "Number of ones: 7451\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 15\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 7.7025 ms\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 425.97 us\n",
      "-----------------------------------------\n",
      "size = 5417921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 4.3568 ms\n",
      "-----------------------------------------\n",
      "size = 5626187\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 434.23 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 29.952 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:20:12.001] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:20:12.001] [jointLog] [info] Loading dense pufferfish index.\n",
      "[2020-12-16 15:20:12.015] [jointLog] [info] done\n",
      "[2020-12-16 15:20:12.015] [jointLog] [info] Index contained 5,685 targets\n",
      "[2020-12-16 15:20:12.017] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[2020-12-16 15:20:12.490] [jointLog] [info] Automatically detected most likely library type as IU\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 266,245, hits per frag:  0.591594\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 537,949, hits per frag:  0.560214\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,500,001 \u001b[32mfragments\u001b[0m\n",
      "hits: 805,881, hits per frag:  0.539319\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 2,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,074,524, hits per frag:  0.547974\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[2020-12-16 15:20:18.357] [jointLog] [info] Computed 11,820 rich equivalence classes for further processing\n",
      "[2020-12-16 15:20:18.357] [jointLog] [info] Counted 1,101,988 total reads in the equivalence classes \n",
      "[2020-12-16 15:20:18.359] [jointLog] [info] Number of mappings discarded because of alignment score : 1,248,509\n",
      "[2020-12-16 15:20:18.359] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 759,833\n",
      "[2020-12-16 15:20:18.359] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 0\n",
      "[2020-12-16 15:20:18.359] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 685,362\n",
      "[2020-12-16 15:20:18.360] [jointLog] [warning] Only 1101988 fragments were mapped, but the number of burn-in fragments was set to 5000000.\n",
      "The effective lengths have been computed using the observed mappings.\n",
      "\n",
      "[2020-12-16 15:20:18.360] [jointLog] [info] Mapping rate = 52.4395%\n",
      "\n",
      "[2020-12-16 15:20:18.360] [jointLog] [info] finished quantifyLibrary()\n",
      "[2020-12-16 15:20:18.360] [jointLog] [info] Starting optimizer\n",
      "[2020-12-16 15:20:18.363] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "[2020-12-16 15:20:18.363] [jointLog] [info] iteration = 0 | max rel diff. = 188.591\n",
      "[2020-12-16 15:20:18.410] [jointLog] [info] iteration = 100 | max rel diff. = 0.00193643\n",
      "[2020-12-16 15:20:18.410] [jointLog] [info] Finished optimizer\n",
      "[2020-12-16 15:20:18.410] [jointLog] [info] writing output \n",
      "\n",
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/pao1_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR11809598_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR11809598_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_pao1/SRR11809598_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_pao1/SRR11809598_quant/logs\n",
      "[2020-12-16 15:20:18.773] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:20:18.773] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:20:18.773] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:20:18.773] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:20:18.773] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:20:18.773] [jointLog] [info] There is 1 library.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 758.26 us\n",
      "-----------------------------------------\n",
      "size = 7452\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 32.033 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 17.387 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 470.11 us\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "Number of ones: 7451\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 15\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 7.9108 ms\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 427.39 us\n",
      "-----------------------------------------\n",
      "size = 5417921\n",
      "[2020-12-16 15:20:18.821] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:20:18.821] [jointLog] [info] Loading dense pufferfish index.\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 4.4167 ms\n",
      "-----------------------------------------\n",
      "size = 5626187\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 426.99 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 29.693 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:20:18.835] [jointLog] [info] done\n",
      "[2020-12-16 15:20:18.835] [jointLog] [info] Index contained 5,685 targets\n",
      "[2020-12-16 15:20:18.837] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[2020-12-16 15:20:19.343] [jointLog] [info] Automatically detected most likely library type as IU\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 266,264, hits per frag:  0.585739\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 537,937, hits per frag:  0.544106\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 805,961, hits per frag:  0.538671\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 2,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,074,581, hits per frag:  0.539212\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[2020-12-16 15:20:25.992] [jointLog] [info] Computed 11,820 rich equivalence classes for further processing\n",
      "[2020-12-16 15:20:25.992] [jointLog] [info] Counted 1,101,988 total reads in the equivalence classes \n",
      "[2020-12-16 15:20:25.994] [jointLog] [info] Number of mappings discarded because of alignment score : 1,248,509\n",
      "[2020-12-16 15:20:25.994] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 759,833\n",
      "[2020-12-16 15:20:25.994] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 0\n",
      "[2020-12-16 15:20:25.994] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 685,362\n",
      "[2020-12-16 15:20:25.995] [jointLog] [warning] Only 1101988 fragments were mapped, but the number of burn-in fragments was set to 5000000.\n",
      "The effective lengths have been computed using the observed mappings.\n",
      "\n",
      "[2020-12-16 15:20:25.995] [jointLog] [info] Mapping rate = 52.4395%\n",
      "\n",
      "[2020-12-16 15:20:25.995] [jointLog] [info] finished quantifyLibrary()\n",
      "[2020-12-16 15:20:25.995] [jointLog] [info] Starting optimizer\n",
      "[2020-12-16 15:20:25.998] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "[2020-12-16 15:20:25.998] [jointLog] [info] iteration = 0 | max rel diff. = 188.591\n",
      "[2020-12-16 15:20:26.048] [jointLog] [info] iteration = 100 | max rel diff. = 0.00192924\n",
      "[2020-12-16 15:20:26.048] [jointLog] [info] Finished optimizer\n",
      "[2020-12-16 15:20:26.048] [jointLog] [info] writing output \n",
      "\n",
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/pao1_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR12922100_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR12922100_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_pao1/SRR12922100_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_pao1/SRR12922100_quant/logs\n",
      "[2020-12-16 15:20:26.552] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:20:26.552] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:20:26.552] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:20:26.552] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:20:26.552] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:20:26.552] [jointLog] [info] There is 1 library.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 744.9 us\n",
      "-----------------------------------------\n",
      "size = 7452\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 29.716 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 18.211 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 498.58 us\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "Number of ones: 7451\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 15\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 7.7304 ms\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 434.03 us\n",
      "-----------------------------------------\n",
      "size = 5417921\n",
      "[2020-12-16 15:20:26.598] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:20:26.598] [jointLog] [info] Loading dense pufferfish index.\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 4.5553 ms\n",
      "-----------------------------------------\n",
      "size = 5626187\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 441 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 29.822 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:20:26.613] [jointLog] [info] done\n",
      "[2020-12-16 15:20:26.613] [jointLog] [info] Index contained 5,685 targets\n",
      "[2020-12-16 15:20:26.614] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 428, hits per frag:  0.000895598\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 911, hits per frag:  0.0009372\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,366, hits per frag:  0.000929127\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 2,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,779, hits per frag:  0.000902312\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 2,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 2,247, hits per frag:  0.000903189\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 3,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 2,660, hits per frag:  0.000890486\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 3,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 3,041, hits per frag:  0.000875772\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[2020-12-16 15:20:42.783] [jointLog] [info] Computed 181 rich equivalence classes for further processing\n",
      "[2020-12-16 15:20:42.783] [jointLog] [info] Counted 2,339 total reads in the equivalence classes \n",
      "[2020-12-16 15:20:42.784] [jointLog] [info] Number of mappings discarded because of alignment score : 29,617\n",
      "[2020-12-16 15:20:42.784] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 16,307\n",
      "[2020-12-16 15:20:42.784] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 0\n",
      "[2020-12-16 15:20:42.784] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 2,412\n",
      "[2020-12-16 15:20:42.785] [jointLog] [warning] Only 2339 fragments were mapped, but the number of burn-in fragments was set to 5000000.\n",
      "The effective lengths have been computed using the observed mappings.\n",
      "\n",
      "[2020-12-16 15:20:42.785] [jointLog] [info] Mapping rate = 0.0634399%\n",
      "\n",
      "[2020-12-16 15:20:42.785] [jointLog] [info] finished quantifyLibrary()\n",
      "[2020-12-16 15:20:42.785] [jointLog] [info] Starting optimizer\n",
      "[2020-12-16 15:20:42.787] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "[2020-12-16 15:20:42.787] [jointLog] [info] iteration = 0 | max rel diff. = 13.531\n",
      "[2020-12-16 15:20:42.821] [jointLog] [info] iteration = 100 | max rel diff. = 0.00449927\n",
      "[2020-12-16 15:20:42.822] [jointLog] [info] Finished optimizer\n",
      "[2020-12-16 15:20:42.822] [jointLog] [info] writing output \n",
      "\n",
      "[2020-12-16 15:20:42.827] [jointLog] [warning] NOTE: Read Lib [[ /home/alexandra/ncbi/public/fastq_phage/SRR12922100_1.fastq, /home/alexandra/ncbi/public/fastq_phage/SRR12922100_2.fastq]] :\n",
      "\n",
      "Detected a *potential* strand bias > 1% in an unstranded protocol check the file: /home/alexandra/ncbi/public/quants_pao1/SRR12922100_quant/lib_format_counts.json for details\n",
      "\n",
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/pao1_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR12922100_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR12922100_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_pao1/SRR12922100_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_pao1/SRR12922100_quant/logs\n",
      "[2020-12-16 15:20:43.344] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:20:43.344] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:20:43.344] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:20:43.344] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:20:43.344] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:20:43.344] [jointLog] [info] There is 1 library.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 760.27 us\n",
      "-----------------------------------------\n",
      "size = 7452\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 30.632 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 16.023 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 454.61 us\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "Number of ones: 7451\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 15\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 7.9487 ms\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 429.13 us\n",
      "-----------------------------------------\n",
      "size = 5417921\n",
      "[2020-12-16 15:20:43.393] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:20:43.393] [jointLog] [info] Loading dense pufferfish index.\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 4.418 ms\n",
      "-----------------------------------------\n",
      "size = 5626187\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 421.18 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 28.995 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:20:43.408] [jointLog] [info] done\n",
      "[2020-12-16 15:20:43.408] [jointLog] [info] Index contained 5,685 targets\n",
      "[2020-12-16 15:20:43.409] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 427, hits per frag:  0.000876421\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 915, hits per frag:  0.00093256\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,363, hits per frag:  0.00091849\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 2,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,779, hits per frag:  0.000890188\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 2,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 2,248, hits per frag:  0.000906703\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 3,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 2,660, hits per frag:  0.00088683\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 3,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 3,043, hits per frag:  0.000872662\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[2020-12-16 15:20:59.081] [jointLog] [info] Computed 181 rich equivalence classes for further processing\n",
      "[2020-12-16 15:20:59.081] [jointLog] [info] Counted 2,339 total reads in the equivalence classes \n",
      "[2020-12-16 15:20:59.083] [jointLog] [info] Number of mappings discarded because of alignment score : 29,617\n",
      "[2020-12-16 15:20:59.083] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 16,307\n",
      "[2020-12-16 15:20:59.083] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 0\n",
      "[2020-12-16 15:20:59.083] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 2,412\n",
      "[2020-12-16 15:20:59.083] [jointLog] [warning] Only 2339 fragments were mapped, but the number of burn-in fragments was set to 5000000.\n",
      "The effective lengths have been computed using the observed mappings.\n",
      "\n",
      "[2020-12-16 15:20:59.083] [jointLog] [info] Mapping rate = 0.0634399%\n",
      "\n",
      "[2020-12-16 15:20:59.083] [jointLog] [info] finished quantifyLibrary()\n",
      "[2020-12-16 15:20:59.083] [jointLog] [info] Starting optimizer\n",
      "[2020-12-16 15:20:59.085] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "[2020-12-16 15:20:59.086] [jointLog] [info] iteration = 0 | max rel diff. = 13.5282\n",
      "[2020-12-16 15:20:59.120] [jointLog] [info] iteration = 100 | max rel diff. = 0.00449935\n",
      "[2020-12-16 15:20:59.120] [jointLog] [info] Finished optimizer\n",
      "[2020-12-16 15:20:59.120] [jointLog] [info] writing output \n",
      "\n",
      "[2020-12-16 15:20:59.126] [jointLog] [warning] NOTE: Read Lib [[ /home/alexandra/ncbi/public/fastq_phage/SRR12922100_1.fastq, /home/alexandra/ncbi/public/fastq_phage/SRR12922100_2.fastq]] :\n",
      "\n",
      "Detected a *potential* strand bias > 1% in an unstranded protocol check the file: /home/alexandra/ncbi/public/quants_pao1/SRR12922100_quant/lib_format_counts.json for details\n",
      "\n",
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/pao1_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196068_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196068_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_pao1/SRR13196068_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_pao1/SRR13196068_quant/logs\n",
      "[2020-12-16 15:20:59.605] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:20:59.605] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:20:59.605] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:20:59.605] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:20:59.605] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:20:59.605] [jointLog] [info] There is 1 library.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 775.37 us\n",
      "-----------------------------------------\n",
      "size = 7452\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 30.586 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 16.051 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 466.43 us\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "Number of ones: 7451\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 15\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 7.6536 ms\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 433.99 us\n",
      "-----------------------------------------\n",
      "size = 5417921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 4.5127 ms\n",
      "-----------------------------------------\n",
      "size = 5626187\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 431.83 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 29.645 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:20:59.650] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:20:59.650] [jointLog] [info] Loading dense pufferfish index.\n",
      "[2020-12-16 15:20:59.665] [jointLog] [info] done\n",
      "[2020-12-16 15:20:59.665] [jointLog] [info] Index contained 5,685 targets\n",
      "[2020-12-16 15:20:59.666] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 0, hits per frag:  0\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 0, hits per frag:  0Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/pao1_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196068_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196068_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_pao1/SRR13196068_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_pao1/SRR13196068_quant/logs\n",
      "[2020-12-16 15:21:00.354] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:21:00.354] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:21:00.354] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:21:00.354] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:21:00.354] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:21:00.354] [jointLog] [info] There is 1 library.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 763.73 us\n",
      "-----------------------------------------\n",
      "size = 7452\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 29.86 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 17.202 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 466.14 us\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "Number of ones: 7451\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 15\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 7.6624 ms\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 426.51 us\n",
      "-----------------------------------------\n",
      "size = 5417921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 4.4607 ms\n",
      "-----------------------------------------\n",
      "size = 5626187\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 439.27 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 31.047 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:21:00.399] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:21:00.399] [jointLog] [info] Loading dense pufferfish index.\n",
      "[2020-12-16 15:21:00.413] [jointLog] [info] done\n",
      "[2020-12-16 15:21:00.413] [jointLog] [info] Index contained 5,685 targets\n",
      "[2020-12-16 15:21:00.415] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 0, hits per frag:  0\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 0, hits per frag:  0Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/pao1_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196071_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196071_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_pao1/SRR13196071_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_pao1/SRR13196071_quant/logs\n",
      "[2020-12-16 15:21:01.122] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:21:01.122] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:21:01.122] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:21:01.122] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:21:01.122] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:21:01.122] [jointLog] [info] There is 1 library.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 752.08 us\n",
      "-----------------------------------------\n",
      "size = 7452\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 29.846 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 17.468 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 509.6 us\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "Number of ones: 7451\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 15\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 8.0199 ms\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 442.02 us\n",
      "-----------------------------------------\n",
      "size = 5417921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 4.6358 ms\n",
      "-----------------------------------------\n",
      "size = 5626187\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 438.11 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 29.844 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:21:01.167] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:21:01.167] [jointLog] [info] Loading dense pufferfish index.\n",
      "[2020-12-16 15:21:01.182] [jointLog] [info] done\n",
      "[2020-12-16 15:21:01.182] [jointLog] [info] Index contained 5,685 targets\n",
      "[2020-12-16 15:21:01.184] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 0, hits per frag:  0Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/pao1_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196071_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196071_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_pao1/SRR13196071_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_pao1/SRR13196071_quant/logs\n",
      "[2020-12-16 15:21:01.889] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:21:01.889] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:21:01.889] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:21:01.889] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:21:01.889] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:21:01.889] [jointLog] [info] There is 1 library.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 748.13 us\n",
      "-----------------------------------------\n",
      "size = 7452\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 30.256 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 17.766 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 473.9 us\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "Number of ones: 7451\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 15\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 7.7442 ms\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 442.2 us\n",
      "-----------------------------------------\n",
      "size = 5417921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 4.5057 ms\n",
      "-----------------------------------------\n",
      "size = 5626187\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 439.12 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 29.373 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:21:01.934] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:21:01.934] [jointLog] [info] Loading dense pufferfish index.\n",
      "[2020-12-16 15:21:01.949] [jointLog] [info] done\n",
      "[2020-12-16 15:21:01.949] [jointLog] [info] Index contained 5,685 targets\n",
      "[2020-12-16 15:21:01.950] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 0, hits per frag:  0"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'\\nfor FILE_PATH in $2/*;\\ndo\\n\\n# get file name\\nsample_name=`basename ${FILE_PATH}`\\n\\n# remove extension from file name\\nsample_name=\"${sample_name%_*}\"\\n\\n# get base path\\nbase_name=${FILE_PATH%/*}\\n\\necho \"Processing sample ${sample_name}\"\\n\\nsalmon quant -i $3 -l A \\\\\\n            -1 ${base_name}/${sample_name}_1.fastq \\\\\\n            -2 ${base_name}/${sample_name}_2.fastq \\\\\\n            -p 8 --validateMappings -o $1/${sample_name}_quant\\ndone\\n'' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-13-7050c133f320>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'-s $paths.PAO1_QUANT $paths.FASTQ_DIR $paths.PAO1_INDEX'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'\\nfor FILE_PATH in $2/*;\\ndo\\n\\n# get file name\\nsample_name=`basename ${FILE_PATH}`\\n\\n# remove extension from file name\\nsample_name=\"${sample_name%_*}\"\\n\\n# get base path\\nbase_name=${FILE_PATH%/*}\\n\\necho \"Processing sample ${sample_name}\"\\n\\nsalmon quant -i $3 -l A \\\\\\n            -1 ${base_name}/${sample_name}_1.fastq \\\\\\n            -2 ${base_name}/${sample_name}_2.fastq \\\\\\n            -p 8 --validateMappings -o $1/${sample_name}_quant\\ndone\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/envs/core_acc/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2369\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2370\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2371\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2372\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2373\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/core_acc/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<decorator-gen-110>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/core_acc/lib/python3.7/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/core_acc/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'\\nfor FILE_PATH in $2/*;\\ndo\\n\\n# get file name\\nsample_name=`basename ${FILE_PATH}`\\n\\n# remove extension from file name\\nsample_name=\"${sample_name%_*}\"\\n\\n# get base path\\nbase_name=${FILE_PATH%/*}\\n\\necho \"Processing sample ${sample_name}\"\\n\\nsalmon quant -i $3 -l A \\\\\\n            -1 ${base_name}/${sample_name}_1.fastq \\\\\\n            -2 ${base_name}/${sample_name}_2.fastq \\\\\\n            -p 8 --validateMappings -o $1/${sample_name}_quant\\ndone\\n'' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%bash -s $paths.PAO1_QUANT $paths.FASTQ_DIR $paths.PAO1_INDEX\n",
    "\n",
    "for FILE_PATH in $2/*;\n",
    "do\n",
    "\n",
    "# get file name\n",
    "sample_name=`basename ${FILE_PATH}`\n",
    "\n",
    "# remove extension from file name\n",
    "sample_name=\"${sample_name%_*}\"\n",
    "\n",
    "# get base path\n",
    "base_name=${FILE_PATH%/*}\n",
    "\n",
    "echo \"Processing sample ${sample_name}\"\n",
    "\n",
    "salmon quant -i $3 -l A \\\n",
    "            -1 ${base_name}/${sample_name}_1.fastq \\\n",
    "            -2 ${base_name}/${sample_name}_2.fastq \\\n",
    "            -p 8 --validateMappings -o $1/${sample_name}_quant\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get quants using phage reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.exists(paths.PHAGE_QUANT):\n",
    "    os.makedirs(paths.PHAGE_QUANT)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing sample SRR11809598\n",
      "Processing sample SRR11809598\n",
      "Processing sample SRR12922100\n",
      "Processing sample SRR12922100\n",
      "Processing sample SRR13196068\n",
      "Processing sample SRR13196068\n",
      "Processing sample SRR13196071\n",
      "Processing sample SRR13196071\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/home/alexandra/ncbi/public/quants_phage’: File exists\n",
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/phage_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR11809598_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR11809598_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_phage/SRR11809598_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_phage/SRR11809598_quant/logs\n",
      "[2020-12-16 15:18:11.030] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:18:11.030] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:18:11.030] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:18:11.030] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:18:11.030] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:18:11.030] [jointLog] [info] There is 1 library.\n",
      "[2020-12-16 15:18:11.074] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:18:11.074] [jointLog] [info] Loading dense pufferfish index.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 109.85 ms\n",
      "-----------------------------------------\n",
      "size = 279402\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 134 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 12.967 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 3.9184 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "Number of ones: 279401\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 546\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 47.774 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 2.3683 ms\n",
      "-----------------------------------------\n",
      "size = 21980921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 19.424 ms\n",
      "-----------------------------------------\n",
      "size = 49770970\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 3.5396 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 21.584 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:18:11.261] [jointLog] [info] done\n",
      "[2020-12-16 15:18:11.261] [jointLog] [info] Index contained 1,128 targets\n",
      "[2020-12-16 15:18:11.262] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[2020-12-16 15:18:11.690] [jointLog] [info] Automatically detected most likely library type as IU\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 406,651, hits per frag:  0.825523\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 817,677, hits per frag:  0.826107\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,225,160, hits per frag:  0.819217\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 2,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,634,299, hits per frag:  0.82256\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[2020-12-16 15:18:21.128] [jointLog] [info] Computed 2,071 rich equivalence classes for further processing\n",
      "[2020-12-16 15:18:21.128] [jointLog] [info] Counted 1,376,590 total reads in the equivalence classes \n",
      "[2020-12-16 15:18:21.130] [jointLog] [info] Number of mappings discarded because of alignment score : 1,272,267\n",
      "[2020-12-16 15:18:21.130] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 591,753\n",
      "[2020-12-16 15:18:21.130] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 0\n",
      "[2020-12-16 15:18:21.130] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 719,875\n",
      "[2020-12-16 15:18:21.131] [jointLog] [warning] Only 1376590 fragments were mapped, but the number of burn-in fragments was set to 5000000.\n",
      "The effective lengths have been computed using the observed mappings.\n",
      "\n",
      "[2020-12-16 15:18:21.131] [jointLog] [info] Mapping rate = 65.5068%\n",
      "\n",
      "[2020-12-16 15:18:21.131] [jointLog] [info] finished quantifyLibrary()\n",
      "[2020-12-16 15:18:21.131] [jointLog] [info] Starting optimizer\n",
      "[2020-12-16 15:18:21.132] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "[2020-12-16 15:18:21.132] [jointLog] [info] iteration = 0 | max rel diff. = 98440.8\n",
      "[2020-12-16 15:18:21.185] [jointLog] [info] iteration = 100 | max rel diff. = 0.01931\n",
      "[2020-12-16 15:18:21.238] [jointLog] [info] iteration = 200 | max rel diff. = 2.26075\n",
      "[2020-12-16 15:18:21.291] [jointLog] [info] iteration = 300 | max rel diff. = 0.790033\n",
      "[2020-12-16 15:18:21.292] [jointLog] [info] iteration = 303 | max rel diff. = 0.00726311\n",
      "[2020-12-16 15:18:21.292] [jointLog] [info] Finished optimizer\n",
      "[2020-12-16 15:18:21.292] [jointLog] [info] writing output \n",
      "\n",
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/phage_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR11809598_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR11809598_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_phage/SRR11809598_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_phage/SRR11809598_quant/logs\n",
      "[2020-12-16 15:18:22.022] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:18:22.022] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:18:22.022] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:18:22.022] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:18:22.022] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:18:22.022] [jointLog] [info] There is 1 library.\n",
      "[2020-12-16 15:18:22.068] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:18:22.068] [jointLog] [info] Loading dense pufferfish index.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 110.13 ms\n",
      "-----------------------------------------\n",
      "size = 279402\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 144.46 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 10.014 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 4.1537 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "Number of ones: 279401\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 546\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 48.407 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 2.1859 ms\n",
      "-----------------------------------------\n",
      "size = 21980921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 20.929 ms\n",
      "-----------------------------------------\n",
      "size = 49770970\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 3.8674 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 17.458 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:18:22.258] [jointLog] [info] done\n",
      "[2020-12-16 15:18:22.258] [jointLog] [info] Index contained 1,128 targets\n",
      "[2020-12-16 15:18:22.259] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[2020-12-16 15:18:22.739] [jointLog] [info] Automatically detected most likely library type as IU\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,001 \u001b[32mfragments\u001b[0m\n",
      "hits: 406,663, hits per frag:  0.864387\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 817,651, hits per frag:  0.830237\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,224,304, hits per frag:  0.817969\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 2,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,634,175, hits per frag:  0.826613\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[2020-12-16 15:18:32.865] [jointLog] [info] Computed 2,071 rich equivalence classes for further processing\n",
      "[2020-12-16 15:18:32.865] [jointLog] [info] Counted 1,376,590 total reads in the equivalence classes \n",
      "[2020-12-16 15:18:32.868] [jointLog] [info] Number of mappings discarded because of alignment score : 1,272,267\n",
      "[2020-12-16 15:18:32.868] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 591,753\n",
      "[2020-12-16 15:18:32.868] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 0\n",
      "[2020-12-16 15:18:32.868] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 719,875\n",
      "[2020-12-16 15:18:32.868] [jointLog] [warning] Only 1376590 fragments were mapped, but the number of burn-in fragments was set to 5000000.\n",
      "The effective lengths have been computed using the observed mappings.\n",
      "\n",
      "[2020-12-16 15:18:32.868] [jointLog] [info] Mapping rate = 65.5068%\n",
      "\n",
      "[2020-12-16 15:18:32.868] [jointLog] [info] finished quantifyLibrary()\n",
      "[2020-12-16 15:18:32.868] [jointLog] [info] Starting optimizer\n",
      "[2020-12-16 15:18:32.870] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "[2020-12-16 15:18:32.870] [jointLog] [info] iteration = 0 | max rel diff. = 98441.2\n",
      "[2020-12-16 15:18:32.930] [jointLog] [info] iteration = 100 | max rel diff. = 0.01931\n",
      "[2020-12-16 15:18:32.991] [jointLog] [info] iteration = 200 | max rel diff. = 2.25708\n",
      "[2020-12-16 15:18:33.052] [jointLog] [info] iteration = 300 | max rel diff. = 0.789114\n",
      "[2020-12-16 15:18:33.053] [jointLog] [info] iteration = 303 | max rel diff. = 0.00726307\n",
      "[2020-12-16 15:18:33.053] [jointLog] [info] Finished optimizer\n",
      "[2020-12-16 15:18:33.053] [jointLog] [info] writing output \n",
      "\n",
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/phage_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR12922100_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR12922100_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_phage/SRR12922100_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_phage/SRR12922100_quant/logs\n",
      "[2020-12-16 15:18:33.358] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:18:33.358] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:18:33.358] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:18:33.358] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:18:33.358] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:18:33.358] [jointLog] [info] There is 1 library.\n",
      "[2020-12-16 15:18:33.406] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:18:33.406] [jointLog] [info] Loading dense pufferfish index.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 109.72 ms\n",
      "-----------------------------------------\n",
      "size = 279402\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 128.75 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 8.18 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 3.9265 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "Number of ones: 279401\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 546\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 47.675 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 2.1857 ms\n",
      "-----------------------------------------\n",
      "size = 21980921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 19.45 ms\n",
      "-----------------------------------------\n",
      "size = 49770970\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 3.5237 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 13.573 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:18:33.593] [jointLog] [info] done\n",
      "[2020-12-16 15:18:33.593] [jointLog] [info] Index contained 1,128 targets\n",
      "[2020-12-16 15:18:33.593] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 349, hits per frag:  0.000737287\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 742, hits per frag:  0.00074552\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,094, hits per frag:  0.000737142\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 2,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,427, hits per frag:  0.000713892\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 2,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,802, hits per frag:  0.000737726\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 3,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 2,127, hits per frag:  0.000715324\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 3,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 2,449, hits per frag:  0.000700576\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[2020-12-16 15:18:59.033] [jointLog] [info] Computed 5 rich equivalence classes for further processing\n",
      "[2020-12-16 15:18:59.033] [jointLog] [info] Counted 2,533 total reads in the equivalence classes \n",
      "[2020-12-16 15:18:59.035] [jointLog] [info] Number of mappings discarded because of alignment score : 41,723\n",
      "[2020-12-16 15:18:59.035] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 19,010\n",
      "[2020-12-16 15:18:59.035] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 0\n",
      "[2020-12-16 15:18:59.035] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 6,545\n",
      "[2020-12-16 15:18:59.035] [jointLog] [warning] Only 2533 fragments were mapped, but the number of burn-in fragments was set to 5000000.\n",
      "The effective lengths have been computed using the observed mappings.\n",
      "\n",
      "[2020-12-16 15:18:59.035] [jointLog] [info] Mapping rate = 0.0687017%\n",
      "\n",
      "[2020-12-16 15:18:59.035] [jointLog] [info] finished quantifyLibrary()\n",
      "[2020-12-16 15:18:59.035] [jointLog] [info] Starting optimizer\n",
      "[2020-12-16 15:18:59.036] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "[2020-12-16 15:18:59.036] [jointLog] [info] iteration = 0 | max rel diff. = 7.80902\n",
      "[2020-12-16 15:18:59.044] [jointLog] [info] iteration = 100 | max rel diff. = 0\n",
      "[2020-12-16 15:18:59.044] [jointLog] [info] Finished optimizer\n",
      "[2020-12-16 15:18:59.044] [jointLog] [info] writing output \n",
      "\n",
      "[2020-12-16 15:18:59.045] [jointLog] [warning] NOTE: Read Lib [[ /home/alexandra/ncbi/public/fastq_phage/SRR12922100_1.fastq, /home/alexandra/ncbi/public/fastq_phage/SRR12922100_2.fastq]] :\n",
      "\n",
      "Detected a *potential* strand bias > 1% in an unstranded protocol check the file: /home/alexandra/ncbi/public/quants_phage/SRR12922100_quant/lib_format_counts.json for details\n",
      "\n",
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/phage_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR12922100_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR12922100_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_phage/SRR12922100_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_phage/SRR12922100_quant/logs\n",
      "[2020-12-16 15:18:59.833] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:18:59.833] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:18:59.833] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:18:59.833] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:18:59.833] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:18:59.833] [jointLog] [info] There is 1 library.\n",
      "[2020-12-16 15:18:59.879] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:18:59.879] [jointLog] [info] Loading dense pufferfish index.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 110.07 ms\n",
      "-----------------------------------------\n",
      "size = 279402\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 149.59 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 13.025 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 4.8816 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "Number of ones: 279401\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 546\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 47.495 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 3.8052 ms\n",
      "-----------------------------------------\n",
      "size = 21980921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 21.935 ms\n",
      "-----------------------------------------\n",
      "size = 49770970\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 3.5634 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 14.815 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:19:00.071] [jointLog] [info] done\n",
      "[2020-12-16 15:19:00.071] [jointLog] [info] Index contained 1,128 targets\n",
      "[2020-12-16 15:19:00.072] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 349, hits per frag:  0.000709644\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 741, hits per frag:  0.000748542\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,093, hits per frag:  0.000729928\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 2,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,424, hits per frag:  0.00072049\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 2,500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 1,803, hits per frag:  0.00072779\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 3,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 2,127, hits per frag:  0.000709433\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 3,500,001 \u001b[32mfragments\u001b[0m\n",
      "hits: 2,449, hits per frag:  0.00070356\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "[2020-12-16 15:19:25.174] [jointLog] [info] Computed 5 rich equivalence classes for further processing\n",
      "[2020-12-16 15:19:25.174] [jointLog] [info] Counted 2,533 total reads in the equivalence classes \n",
      "[2020-12-16 15:19:25.175] [jointLog] [info] Number of mappings discarded because of alignment score : 41,723\n",
      "[2020-12-16 15:19:25.175] [jointLog] [info] Number of fragments entirely discarded because of alignment score : 19,010\n",
      "[2020-12-16 15:19:25.175] [jointLog] [info] Number of fragments discarded because they are best-mapped to decoys : 0\n",
      "[2020-12-16 15:19:25.175] [jointLog] [info] Number of fragments discarded because they have only dovetail (discordant) mappings to valid targets : 6,545\n",
      "[2020-12-16 15:19:25.176] [jointLog] [warning] Only 2533 fragments were mapped, but the number of burn-in fragments was set to 5000000.\n",
      "The effective lengths have been computed using the observed mappings.\n",
      "\n",
      "[2020-12-16 15:19:25.176] [jointLog] [info] Mapping rate = 0.0687017%\n",
      "\n",
      "[2020-12-16 15:19:25.176] [jointLog] [info] finished quantifyLibrary()\n",
      "[2020-12-16 15:19:25.176] [jointLog] [info] Starting optimizer\n",
      "[2020-12-16 15:19:25.176] [jointLog] [info] Marked 0 weighted equivalence classes as degenerate\n",
      "[2020-12-16 15:19:25.177] [jointLog] [info] iteration = 0 | max rel diff. = 7.80902\n",
      "[2020-12-16 15:19:25.184] [jointLog] [info] iteration = 100 | max rel diff. = 0\n",
      "[2020-12-16 15:19:25.184] [jointLog] [info] Finished optimizer\n",
      "[2020-12-16 15:19:25.184] [jointLog] [info] writing output \n",
      "\n",
      "[2020-12-16 15:19:25.185] [jointLog] [warning] NOTE: Read Lib [[ /home/alexandra/ncbi/public/fastq_phage/SRR12922100_1.fastq, /home/alexandra/ncbi/public/fastq_phage/SRR12922100_2.fastq]] :\n",
      "\n",
      "Detected a *potential* strand bias > 1% in an unstranded protocol check the file: /home/alexandra/ncbi/public/quants_phage/SRR12922100_quant/lib_format_counts.json for details\n",
      "\n",
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/phage_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196068_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196068_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_phage/SRR13196068_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_phage/SRR13196068_quant/logs\n",
      "[2020-12-16 15:19:25.613] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:19:25.613] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:19:25.613] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:19:25.613] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:19:25.613] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:19:25.613] [jointLog] [info] There is 1 library.\n",
      "[2020-12-16 15:19:25.659] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:19:25.659] [jointLog] [info] Loading dense pufferfish index.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 110.7 ms\n",
      "-----------------------------------------\n",
      "size = 279402\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 127.79 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 11.834 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 3.966 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "Number of ones: 279401\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 546\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 48.081 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 2.2118 ms\n",
      "-----------------------------------------\n",
      "size = 21980921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 20.334 ms\n",
      "-----------------------------------------\n",
      "size = 49770970\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 3.6279 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 18.036 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:19:25.848] [jointLog] [info] done\n",
      "[2020-12-16 15:19:25.848] [jointLog] [info] Index contained 1,128 targets\n",
      "[2020-12-16 15:19:25.849] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 0, hits per frag:  0\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 0, hits per frag:  0[2020-12-16 15:19:26.303] [jointLog] [warning] salmon was only able to assign 0 fragments to transcripts in the index, but the minimum number of required assigned fragments (--minAssignedFrags) was 10. This could be indicative of a mismatch between the reference and sample, or a very bad sample.  You can change the --minAssignedFrags parameter to force salmon to quantify with fewer assigned fragments (must have at least 1).\n",
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/phage_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196068_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196068_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_phage/SRR13196068_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_phage/SRR13196068_quant/logs\n",
      "[2020-12-16 15:19:26.864] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:19:26.864] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:19:26.864] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:19:26.864] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:19:26.864] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:19:26.864] [jointLog] [info] There is 1 library.\n",
      "[2020-12-16 15:19:26.908] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:19:26.908] [jointLog] [info] Loading dense pufferfish index.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 109.94 ms\n",
      "-----------------------------------------\n",
      "size = 279402\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 127.78 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 10.327 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 3.9539 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "Number of ones: 279401\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 546\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 48.201 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 2.1999 ms\n",
      "-----------------------------------------\n",
      "size = 21980921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 19.801 ms\n",
      "-----------------------------------------\n",
      "size = 49770970\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 3.9705 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 25.532 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:19:27.097] [jointLog] [info] done\n",
      "[2020-12-16 15:19:27.097] [jointLog] [info] Index contained 1,128 targets\n",
      "[2020-12-16 15:19:27.097] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 0, hits per frag:  0\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 1,000,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 0, hits per frag:  0Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/phage_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196071_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196071_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_phage/SRR13196071_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_phage/SRR13196071_quant/logs\n",
      "[2020-12-16 15:19:27.631] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:19:27.631] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:19:27.631] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:19:27.631] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:19:27.631] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:19:27.631] [jointLog] [info] There is 1 library.\n",
      "[2020-12-16 15:19:27.675] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:19:27.675] [jointLog] [info] Loading dense pufferfish index.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 109.72 ms\n",
      "-----------------------------------------\n",
      "size = 279402\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 122.6 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 9.502 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 4.0447 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "Number of ones: 279401\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 546\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 47.59 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 2.2165 ms\n",
      "-----------------------------------------\n",
      "size = 21980921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 21.831 ms\n",
      "-----------------------------------------\n",
      "size = 49770970\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 3.5959 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 14.35 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:19:27.865] [jointLog] [info] done\n",
      "[2020-12-16 15:19:27.865] [jointLog] [info] Index contained 1,128 targets\n",
      "[2020-12-16 15:19:27.865] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 0, hits per frag:  0Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "### salmon (selective-alignment-based) v1.3.0\n",
      "### [ program ] => salmon \n",
      "### [ command ] => quant \n",
      "### [ index ] => { /home/alexandra/Documents/Data/Core_accessory/phage_index }\n",
      "### [ libType ] => { A }\n",
      "### [ mates1 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196071_1.fastq }\n",
      "### [ mates2 ] => { /home/alexandra/ncbi/public/fastq_phage/SRR13196071_2.fastq }\n",
      "### [ threads ] => { 8 }\n",
      "### [ validateMappings ] => { }\n",
      "### [ output ] => { /home/alexandra/ncbi/public/quants_phage/SRR13196071_quant }\n",
      "Logs will be written to /home/alexandra/ncbi/public/quants_phage/SRR13196071_quant/logs\n",
      "[2020-12-16 15:19:28.391] [jointLog] [info] setting maxHashResizeThreads to 8\n",
      "[2020-12-16 15:19:28.391] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.\n",
      "[2020-12-16 15:19:28.391] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65\n",
      "[2020-12-16 15:19:28.391] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.35.\n",
      "[2020-12-16 15:19:28.391] [jointLog] [info] parsing read library format\n",
      "[2020-12-16 15:19:28.391] [jointLog] [info] There is 1 library.\n",
      "[2020-12-16 15:19:28.444] [jointLog] [info] Loading pufferfish index\n",
      "[2020-12-16 15:19:28.444] [jointLog] [info] Loading dense pufferfish index.\n",
      "-----------------------------------------\n",
      "| Loading contig table | Time = 110.26 ms\n",
      "-----------------------------------------\n",
      "size = 279402\n",
      "-----------------------------------------\n",
      "| Loading contig offsets | Time = 128.43 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference lengths | Time = 10.281 us\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading mphf table | Time = 4.0309 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "Number of ones: 279401\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 546\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 47.83 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "-----------------------------------------\n",
      "| Loading sequence | Time = 2.2781 ms\n",
      "-----------------------------------------\n",
      "size = 21980921\n",
      "-----------------------------------------\n",
      "| Loading positions | Time = 19.756 ms\n",
      "-----------------------------------------\n",
      "size = 49770970\n",
      "-----------------------------------------\n",
      "| Loading reference sequence | Time = 3.5388 ms\n",
      "-----------------------------------------\n",
      "-----------------------------------------\n",
      "| Loading reference accumulative lengths | Time = 15.372 us\n",
      "-----------------------------------------\n",
      "[2020-12-16 15:19:28.632] [jointLog] [info] done\n",
      "[2020-12-16 15:19:28.632] [jointLog] [info] Index contained 1,128 targets\n",
      "[2020-12-16 15:19:28.632] [jointLog] [info] Number of decoys : 0\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\u001b[A\r",
      "\r",
      "\u001b[32mprocessed\u001b[31m 500,000 \u001b[32mfragments\u001b[0m\n",
      "hits: 0, hits per frag:  0"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'mkdir $1\\n\\nfor FILE_PATH in $2/*;\\ndo\\n\\n# get file name\\nsample_name=`basename ${FILE_PATH}`\\n\\n# remove extension from file name\\nsample_name=\"${sample_name%_*}\"\\n\\n# get base path\\nbase_name=${FILE_PATH%/*}\\n\\necho \"Processing sample ${sample_name}\"\\n\\nsalmon quant -i $3 -l A \\\\\\n            -1 ${base_name}/${sample_name}_1.fastq \\\\\\n            -2 ${base_name}/${sample_name}_2.fastq \\\\\\n            -p 8 --validateMappings -o $1/${sample_name}_quant\\ndone\\n'' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-12-3345055502ef>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'-s $paths.PHAGE_QUANT $paths.FASTQ_DIR $paths.PHAGE_INDEX'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'mkdir $1\\n\\nfor FILE_PATH in $2/*;\\ndo\\n\\n# get file name\\nsample_name=`basename ${FILE_PATH}`\\n\\n# remove extension from file name\\nsample_name=\"${sample_name%_*}\"\\n\\n# get base path\\nbase_name=${FILE_PATH%/*}\\n\\necho \"Processing sample ${sample_name}\"\\n\\nsalmon quant -i $3 -l A \\\\\\n            -1 ${base_name}/${sample_name}_1.fastq \\\\\\n            -2 ${base_name}/${sample_name}_2.fastq \\\\\\n            -p 8 --validateMappings -o $1/${sample_name}_quant\\ndone\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/envs/core_acc/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2369\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2370\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2371\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2372\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2373\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/core_acc/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<decorator-gen-110>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/core_acc/lib/python3.7/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/core_acc/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'mkdir $1\\n\\nfor FILE_PATH in $2/*;\\ndo\\n\\n# get file name\\nsample_name=`basename ${FILE_PATH}`\\n\\n# remove extension from file name\\nsample_name=\"${sample_name%_*}\"\\n\\n# get base path\\nbase_name=${FILE_PATH%/*}\\n\\necho \"Processing sample ${sample_name}\"\\n\\nsalmon quant -i $3 -l A \\\\\\n            -1 ${base_name}/${sample_name}_1.fastq \\\\\\n            -2 ${base_name}/${sample_name}_2.fastq \\\\\\n            -p 8 --validateMappings -o $1/${sample_name}_quant\\ndone\\n'' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%bash -s $paths.PHAGE_QUANT $paths.FASTQ_DIR $paths.PHAGE_INDEX\n",
    "\n",
    "for FILE_PATH in $2/*;\n",
    "do\n",
    "\n",
    "# get file name\n",
    "sample_name=`basename ${FILE_PATH}`\n",
    "\n",
    "# remove extension from file name\n",
    "sample_name=\"${sample_name%_*}\"\n",
    "\n",
    "# get base path\n",
    "base_name=${FILE_PATH%/*}\n",
    "\n",
    "echo \"Processing sample ${sample_name}\"\n",
    "\n",
    "salmon quant -i $3 -l A \\\n",
    "            -1 ${base_name}/${sample_name}_1.fastq \\\n",
    "            -2 ${base_name}/${sample_name}_2.fastq \\\n",
    "            -p 8 --validateMappings -o $1/${sample_name}_quant\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Consolidate sample quantification to gene expression dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>PGD134012</th>\n",
       "      <th>PGD134018</th>\n",
       "      <th>PGD134020</th>\n",
       "      <th>PGD134022</th>\n",
       "      <th>PGD134024</th>\n",
       "      <th>PGD134014</th>\n",
       "      <th>PGD134016</th>\n",
       "      <th>PGD134026</th>\n",
       "      <th>PGD134030</th>\n",
       "      <th>PGD134032</th>\n",
       "      <th>...</th>\n",
       "      <th>PGD133904</th>\n",
       "      <th>PGD133906</th>\n",
       "      <th>PGD133902</th>\n",
       "      <th>PGD133898</th>\n",
       "      <th>PGD133900</th>\n",
       "      <th>PGD133894</th>\n",
       "      <th>PGD133896</th>\n",
       "      <th>PGD133892</th>\n",
       "      <th>PGD133884</th>\n",
       "      <th>PGD133886</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SRR7886564</th>\n",
       "      <td>231.716766</td>\n",
       "      <td>190.742049</td>\n",
       "      <td>251.434363</td>\n",
       "      <td>244.575451</td>\n",
       "      <td>232.667420</td>\n",
       "      <td>338.778285</td>\n",
       "      <td>154.619466</td>\n",
       "      <td>135.309196</td>\n",
       "      <td>197.297614</td>\n",
       "      <td>208.656767</td>\n",
       "      <td>...</td>\n",
       "      <td>384.006646</td>\n",
       "      <td>160.423009</td>\n",
       "      <td>217.461069</td>\n",
       "      <td>156.885980</td>\n",
       "      <td>207.875748</td>\n",
       "      <td>401.447266</td>\n",
       "      <td>243.877369</td>\n",
       "      <td>242.992502</td>\n",
       "      <td>246.314856</td>\n",
       "      <td>2505.558825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR7886563</th>\n",
       "      <td>221.861101</td>\n",
       "      <td>206.228200</td>\n",
       "      <td>262.510896</td>\n",
       "      <td>236.227489</td>\n",
       "      <td>218.772353</td>\n",
       "      <td>277.169311</td>\n",
       "      <td>172.770017</td>\n",
       "      <td>131.655150</td>\n",
       "      <td>203.906846</td>\n",
       "      <td>195.419829</td>\n",
       "      <td>...</td>\n",
       "      <td>422.431994</td>\n",
       "      <td>186.070549</td>\n",
       "      <td>221.798564</td>\n",
       "      <td>169.894611</td>\n",
       "      <td>206.047320</td>\n",
       "      <td>364.537806</td>\n",
       "      <td>220.715998</td>\n",
       "      <td>222.137392</td>\n",
       "      <td>290.767265</td>\n",
       "      <td>1822.432926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR13196071</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR7886556</th>\n",
       "      <td>239.035540</td>\n",
       "      <td>214.717657</td>\n",
       "      <td>274.756361</td>\n",
       "      <td>234.900934</td>\n",
       "      <td>211.425667</td>\n",
       "      <td>331.008247</td>\n",
       "      <td>182.534175</td>\n",
       "      <td>151.789441</td>\n",
       "      <td>177.389040</td>\n",
       "      <td>214.459340</td>\n",
       "      <td>...</td>\n",
       "      <td>313.613041</td>\n",
       "      <td>203.419543</td>\n",
       "      <td>219.460975</td>\n",
       "      <td>122.112693</td>\n",
       "      <td>216.618620</td>\n",
       "      <td>356.472812</td>\n",
       "      <td>204.935818</td>\n",
       "      <td>236.919636</td>\n",
       "      <td>344.232315</td>\n",
       "      <td>840.672326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR7886554</th>\n",
       "      <td>207.540231</td>\n",
       "      <td>225.539932</td>\n",
       "      <td>225.249527</td>\n",
       "      <td>219.545459</td>\n",
       "      <td>187.796107</td>\n",
       "      <td>288.503175</td>\n",
       "      <td>170.652752</td>\n",
       "      <td>152.999445</td>\n",
       "      <td>178.938258</td>\n",
       "      <td>189.161895</td>\n",
       "      <td>...</td>\n",
       "      <td>310.622609</td>\n",
       "      <td>162.000331</td>\n",
       "      <td>226.673993</td>\n",
       "      <td>151.294458</td>\n",
       "      <td>188.400366</td>\n",
       "      <td>344.227676</td>\n",
       "      <td>230.639268</td>\n",
       "      <td>215.712007</td>\n",
       "      <td>231.559415</td>\n",
       "      <td>2818.206826</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 5685 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name          PGD134012   PGD134018   PGD134020   PGD134022   PGD134024  \\\n",
       "SRR7886564   231.716766  190.742049  251.434363  244.575451  232.667420   \n",
       "SRR7886563   221.861101  206.228200  262.510896  236.227489  218.772353   \n",
       "SRR13196071    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "SRR7886556   239.035540  214.717657  274.756361  234.900934  211.425667   \n",
       "SRR7886554   207.540231  225.539932  225.249527  219.545459  187.796107   \n",
       "\n",
       "Name          PGD134014   PGD134016   PGD134026   PGD134030   PGD134032  ...  \\\n",
       "SRR7886564   338.778285  154.619466  135.309196  197.297614  208.656767  ...   \n",
       "SRR7886563   277.169311  172.770017  131.655150  203.906846  195.419829  ...   \n",
       "SRR13196071    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "SRR7886556   331.008247  182.534175  151.789441  177.389040  214.459340  ...   \n",
       "SRR7886554   288.503175  170.652752  152.999445  178.938258  189.161895  ...   \n",
       "\n",
       "Name          PGD133904   PGD133906   PGD133902   PGD133898   PGD133900  \\\n",
       "SRR7886564   384.006646  160.423009  217.461069  156.885980  207.875748   \n",
       "SRR7886563   422.431994  186.070549  221.798564  169.894611  206.047320   \n",
       "SRR13196071    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "SRR7886556   313.613041  203.419543  219.460975  122.112693  216.618620   \n",
       "SRR7886554   310.622609  162.000331  226.673993  151.294458  188.400366   \n",
       "\n",
       "Name          PGD133894   PGD133896   PGD133892   PGD133884    PGD133886  \n",
       "SRR7886564   401.447266  243.877369  242.992502  246.314856  2505.558825  \n",
       "SRR7886563   364.537806  220.715998  222.137392  290.767265  1822.432926  \n",
       "SRR13196071    0.000000    0.000000    0.000000    0.000000     0.000000  \n",
       "SRR7886556   356.472812  204.935818  236.919636  344.232315   840.672326  \n",
       "SRR7886554   344.227676  230.639268  215.712007  231.559415  2818.206826  \n",
       "\n",
       "[5 rows x 5685 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read through all sample subdirectories in quant/\n",
    "# Within each sample subdirectory, get quant.sf file\n",
    "data_dir = paths.PAO1_QUANT\n",
    "\n",
    "expression_pao1_df = pd.DataFrame(\n",
    "    pd.read_csv(file, sep=\"\\t\", index_col=0)[\"TPM\"].\n",
    "    rename(file.parent.name.split(\"_\")[0]) \n",
    "    for file in data_dir.rglob(\"*/quant.sf\"))    \n",
    "\n",
    "expression_pao1_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>NC_028999.1</th>\n",
       "      <th>MT133560.1</th>\n",
       "      <th>MK599315.1</th>\n",
       "      <th>MH725810.1</th>\n",
       "      <th>MF974178.1</th>\n",
       "      <th>NC_016765.1</th>\n",
       "      <th>NC_031063.1</th>\n",
       "      <th>NC_027375.1</th>\n",
       "      <th>NC_011810.1</th>\n",
       "      <th>MT108726.1</th>\n",
       "      <th>...</th>\n",
       "      <th>DI373497.1</th>\n",
       "      <th>DI373496.1</th>\n",
       "      <th>DI373495.1</th>\n",
       "      <th>DI373494.1</th>\n",
       "      <th>DI373493.1</th>\n",
       "      <th>DI373492.1</th>\n",
       "      <th>DI373491.1</th>\n",
       "      <th>DI373490.1</th>\n",
       "      <th>DI373489.1</th>\n",
       "      <th>DI373488.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SRR13196071</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR13196068</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR12922100</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR11809598</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 1128 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name         NC_028999.1  MT133560.1  MK599315.1  MH725810.1  MF974178.1  \\\n",
       "SRR13196071          0.0         0.0         0.0         0.0         0.0   \n",
       "SRR13196068          0.0         0.0         0.0         0.0         0.0   \n",
       "SRR12922100          0.0         0.0         0.0         0.0         0.0   \n",
       "SRR11809598          0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "Name         NC_016765.1  NC_031063.1  NC_027375.1  NC_011810.1  MT108726.1  \\\n",
       "SRR13196071          0.0          0.0          0.0          0.0         0.0   \n",
       "SRR13196068          0.0          0.0          0.0          0.0         0.0   \n",
       "SRR12922100          0.0          0.0          0.0          0.0         0.0   \n",
       "SRR11809598          0.0          0.0          0.0          0.0         0.0   \n",
       "\n",
       "Name         ...  DI373497.1  DI373496.1  DI373495.1  DI373494.1  DI373493.1  \\\n",
       "SRR13196071  ...         0.0         0.0         0.0         0.0         0.0   \n",
       "SRR13196068  ...         0.0         0.0         0.0         0.0         0.0   \n",
       "SRR12922100  ...         0.0         0.0         0.0         0.0         0.0   \n",
       "SRR11809598  ...         0.0         0.0         0.0         0.0         0.0   \n",
       "\n",
       "Name         DI373492.1  DI373491.1  DI373490.1  DI373489.1  DI373488.1  \n",
       "SRR13196071         0.0         0.0         0.0         0.0         0.0  \n",
       "SRR13196068         0.0         0.0         0.0         0.0         0.0  \n",
       "SRR12922100         0.0         0.0         0.0         0.0         0.0  \n",
       "SRR11809598         0.0         0.0         0.0         0.0         0.0  \n",
       "\n",
       "[4 rows x 1128 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read through all sample subdirectories in quant/\n",
    "# Within each sample subdirectory, get quant.sf file\n",
    "data_dir = paths.PHAGE_QUANT\n",
    "\n",
    "expression_phage_df = pd.DataFrame(\n",
    "    pd.read_csv(file, sep=\"\\t\", index_col=0)[\"TPM\"].\n",
    "    rename(file.parent.name.split(\"_\")[0]) \n",
    "    for file in data_dir.rglob(\"*/quant.sf\"))    \n",
    "\n",
    "expression_phage_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save gene expression data\n",
    "expression_pao1_df.to_csv(paths.PAO1_GE, sep='\\t')\n",
    "expression_phage_df.to_csv(paths.PHAGE_GE, sep='\\t')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:core_acc] *",
   "language": "python",
   "name": "conda-env-core_acc-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
