{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Build reference transcriptome\n",
    "\n",
    "Here we are using [Salmon](https://combine-lab.github.io/salmon/)\n",
    "\n",
    "**Input:**\n",
    "* Target transcriptome\n",
    "* This transcriptome is given to Salmon in the form of a (possibly compressed) multi-FASTA file, with each entry providing the sequence of a transcript\n",
    "* We downloaded the phage GENOMES from NCBI GenBank\n",
    "\n",
    "**Note:** For prokaryotes, transcripts and genes have a more 1-1 mapping so we're using genes for our reference transcriptome and so we don't need to use tximport to map transcript quants to genes. \n",
    "\n",
    "**Output:**\n",
    "* The index is a structure that salmon uses to quasi-map RNA-seq reads during quantification\n",
    "* [Quasi-map](https://academic.oup.com/bioinformatics/article/32/12/i192/2288985) is a way to map sequenced fragments (single or paired-end reads) to a target transcriptome. Quasi-mapping produces what we refer to as fragment mapping information. In particular, it provides, for each query (fragment), the reference sequences (transcripts), strand and position from which the query may have likely originated. In many cases, this mapping information is sufficient for downstream analysis like quantification.\n",
    "\n",
    "*Algorithm:*\n",
    "\n",
    "For a query read r through repeated application of: \n",
    "1. Determining the next hash table k-mer that starts past the current query position\n",
    "2. Computing the maximum mappable prefix (MMP) of the query beginning with this k-mer\n",
    "3. Determining the next informative position (NIP) by performing a longest common prefix (LCP) query on two specifically chosen suffixes in the SA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "from core_acc_modules import paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "[2020-12-22 10:50:46.910] [jLog] [warning] The salmon index is being built without any decoy sequences.  It is recommended that decoy sequence (either computed auxiliary decoy sequence or the genome of the organism) be provided during indexing. Further details can be found at https://salmon.readthedocs.io/en/latest/salmon.html#preparing-transcriptome-indices-mapping-based-mode.\n",
      "[2020-12-22 10:50:46.910] [jLog] [info] building index\n",
      "out : /home/alexandra/Documents/Data/Core_accessory/pao1_index\n",
      "\u001b[00m[2020-12-22 10:50:46.911] [puff::index::jointLog] [info] Running fixFasta\n",
      "\u001b[00m\n",
      "[Step 1 of 4] : counting k-mers\n",
      "\n",
      "\u001b[35m[2020-12-22 10:50:47.079] [puff::index::jointLog] [warning] Removed 34 transcripts that were sequence duplicates of indexed transcripts.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:47.079] [puff::index::jointLog] [warning] If you wish to retain duplicate transcripts, please use the `--keepDuplicates` flag\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:47.079] [puff::index::jointLog] [info] Replaced 0 non-ATCG nucleotides\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:47.079] [puff::index::jointLog] [info] Clipped poly-A tails from 0 transcripts\n",
      "\u001b[00mwrote 5685 cleaned references\n",
      "\u001b[00m[2020-12-22 10:50:47.097] [puff::index::jointLog] [info] Filter size not provided; estimating from number of distinct k-mers\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:47.140] [puff::index::jointLog] [info] ntHll estimated 5373563 distinct k-mers, setting filter size to 2^27\n",
      "\u001b[00mThreads = 2\n",
      "Vertex length = 31\n",
      "Hash functions = 5\n",
      "Filter size = 134217728\n",
      "Capacity = 2\n",
      "Files: \n",
      "/home/alexandra/Documents/Data/Core_accessory/pao1_index/ref_k31_fixed.fa\n",
      "--------------------------------------------------------------------------------\n",
      "Round 0, 0:134217728\n",
      "Pass\tFilling\tFiltering\n",
      "1\t0\t2\t\n",
      "2\t0\t0\n",
      "True junctions count = 1229\n",
      "False junctions count = 20823\n",
      "Hash table size = 22052\n",
      "Candidate marks count = 23789\n",
      "--------------------------------------------------------------------------------\n",
      "Reallocating bifurcations time: 0\n",
      "True marks count: 14005\n",
      "Edges construction time: 0\n",
      "--------------------------------------------------------------------------------\n",
      "Distinct junctions = 1229\n",
      "\n",
      "allowedIn: 12\n",
      "Max Junction ID: 12490\n",
      "seen.size():99929 kmerInfo.size():12491\n",
      "approximateContigTotalLength: 5240272\n",
      "counters for complex kmers:\n",
      "(prec>1 & succ>1)=50 | (succ>1 & isStart)=0 | (prec>1 & isEnd)=0 | (isStart & isEnd)=0\n",
      "contig count: 7452 element count: 5641451 complex nodes: 50\n",
      "# of ones in rank vector: 7451\n",
      "\u001b[00m[2020-12-22 10:50:49.618] [puff::index::jointLog] [info] Starting the Pufferfish indexing by reading the GFA binary file.\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.622] [puff::index::jointLog] [info] Setting the index/BinaryGfa directory /home/alexandra/Documents/Data/Core_accessory/pao1_index\n",
      "\u001b[00msize = 5641451\n",
      "-----------------------------------------\n",
      "| Loading contigs | Time = 544.03 us\n",
      "-----------------------------------------\n",
      "size = 5641451\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 255.32 us\n",
      "-----------------------------------------\n",
      "Number of ones: 7451\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 15\n",
      "7451\n",
      "\u001b[00m[2020-12-22 10:50:49.633] [puff::index::jointLog] [info] Done wrapping the rank vector with a rank9sel structure.\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.633] [puff::index::jointLog] [info] contig count for validation: 7,451\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.635] [puff::index::jointLog] [info] Total # of Contigs : 7,451\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.635] [puff::index::jointLog] [info] Total # of numerical Contigs : 7,451\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.635] [puff::index::jointLog] [info] Total # of contig vec entries: 8,421\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.635] [puff::index::jointLog] [info] bits per offset entry 14\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.635] [puff::index::jointLog] [info] Done constructing the contig vector. 7452\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.648] [puff::index::jointLog] [info] # segments = 7,451\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.648] [puff::index::jointLog] [info] total length = 5,641,451\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.651] [puff::index::jointLog] [info] Reading the reference files ...\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.682] [puff::index::jointLog] [info] positional integer width = 23\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.682] [puff::index::jointLog] [info] seqSize = 5,641,451\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.682] [puff::index::jointLog] [info] rankSize = 5,641,451\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.682] [puff::index::jointLog] [info] edgeVecSize = 0\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.682] [puff::index::jointLog] [info] num keys = 5,417,921\n",
      "\u001b[00mfor info, total work write each  : 2.331    total work inram from level 3 : 4.322  total work raw : 25.000 \n",
      "[Building BooPHF]  100  %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec\n",
      "Bitarray        28393984  bits (100.00 %)   (array + ranks )\n",
      "final hash             0  bits (0.00 %) (nb in final hash 0)\n",
      "\u001b[00m[2020-12-22 10:50:49.891] [puff::index::jointLog] [info] mphf size = 3.38483 MB\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.896] [puff::index::jointLog] [info] chunk size = 2,820,726\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.901] [puff::index::jointLog] [info] chunk 0 = [0, 2,820,726)\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:49.901] [puff::index::jointLog] [info] chunk 1 = [2,820,726, 5,641,421)\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:50.239] [puff::index::jointLog] [info] finished populating pos vector\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:50.239] [puff::index::jointLog] [info] writing index components\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:50.277] [puff::index::jointLog] [info] finished writing dense pufferfish index\n",
      "\u001b[00m[2020-12-22 10:50:50.282] [jLog] [info] done building index\n"
     ]
    }
   ],
   "source": [
    "# Get PAO1 index\n",
    "! salmon index -t $paths.PAO1_REF -i $paths.PAO1_INDEX"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "[2020-12-22 10:50:50.568] [jLog] [warning] The salmon index is being built without any decoy sequences.  It is recommended that decoy sequence (either computed auxiliary decoy sequence or the genome of the organism) be provided during indexing. Further details can be found at https://salmon.readthedocs.io/en/latest/salmon.html#preparing-transcriptome-indices-mapping-based-mode.\n",
      "[2020-12-22 10:50:50.568] [jLog] [info] building index\n",
      "out : /home/alexandra/Documents/Data/Core_accessory/pa14_index\n",
      "\u001b[00m[2020-12-22 10:50:50.568] [puff::index::jointLog] [info] Running fixFasta\n",
      "\u001b[00m\n",
      "[Step 1 of 4] : counting k-mers\n",
      "\n",
      "\u001b[35m[2020-12-22 10:50:50.714] [puff::index::jointLog] [warning] Removed 37 transcripts that were sequence duplicates of indexed transcripts.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:50.714] [puff::index::jointLog] [warning] If you wish to retain duplicate transcripts, please use the `--keepDuplicates` flag\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:50.714] [puff::index::jointLog] [info] Replaced 3 non-ATCG nucleotides\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:50.714] [puff::index::jointLog] [info] Clipped poly-A tails from 0 transcripts\n",
      "\u001b[00mwrote 5959 cleaned references\n",
      "\u001b[00m[2020-12-22 10:50:50.752] [puff::index::jointLog] [info] Filter size not provided; estimating from number of distinct k-mers\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:50.809] [puff::index::jointLog] [info] ntHll estimated 5599628 distinct k-mers, setting filter size to 2^27\n",
      "\u001b[00mThreads = 2\n",
      "Vertex length = 31\n",
      "Hash functions = 5\n",
      "Filter size = 134217728\n",
      "Capacity = 2\n",
      "Files: \n",
      "/home/alexandra/Documents/Data/Core_accessory/pa14_index/ref_k31_fixed.fa\n",
      "--------------------------------------------------------------------------------\n",
      "Round 0, 0:134217728\n",
      "Pass\tFilling\tFiltering\n",
      "1\t1\t1\t\n",
      "2\t0\t0\n",
      "True junctions count = 1467\n",
      "False junctions count = 23416\n",
      "Hash table size = 24883\n",
      "Candidate marks count = 26949\n",
      "--------------------------------------------------------------------------------\n",
      "Reallocating bifurcations time: 0\n",
      "True marks count: 15045\n",
      "Edges construction time: 0\n",
      "--------------------------------------------------------------------------------\n",
      "Distinct junctions = 1467\n",
      "\n",
      "allowedIn: 11\n",
      "Max Junction ID: 13266\n",
      "seen.size():106137 kmerInfo.size():13267\n",
      "approximateContigTotalLength: 5435809\n",
      "counters for complex kmers:\n",
      "(prec>1 & succ>1)=63 | (succ>1 & isStart)=0 | (prec>1 & isEnd)=0 | (isStart & isEnd)=0\n",
      "contig count: 8094 element count: 5890652 complex nodes: 63\n",
      "# of ones in rank vector: 8093\n",
      "\u001b[00m[2020-12-22 10:50:53.232] [puff::index::jointLog] [info] Starting the Pufferfish indexing by reading the GFA binary file.\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.232] [puff::index::jointLog] [info] Setting the index/BinaryGfa directory /home/alexandra/Documents/Data/Core_accessory/pa14_index\n",
      "\u001b[00msize = 5890652\n",
      "-----------------------------------------\n",
      "| Loading contigs | Time = 164.91 us\n",
      "-----------------------------------------\n",
      "size = 5890652\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 89.142 us\n",
      "-----------------------------------------\n",
      "Number of ones: 8093\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 16\n",
      "8093\n",
      "\u001b[00m[2020-12-22 10:50:53.240] [puff::index::jointLog] [info] Done wrapping the rank vector with a rank9sel structure.\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.240] [puff::index::jointLog] [info] contig count for validation: 8,093\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.242] [puff::index::jointLog] [info] Total # of Contigs : 8,093\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.242] [puff::index::jointLog] [info] Total # of numerical Contigs : 8,093\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.242] [puff::index::jointLog] [info] Total # of contig vec entries: 9,227\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.242] [puff::index::jointLog] [info] bits per offset entry 14\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.242] [puff::index::jointLog] [info] Done constructing the contig vector. 8094\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.243] [puff::index::jointLog] [info] # segments = 8,093\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.243] [puff::index::jointLog] [info] total length = 5,890,652\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.244] [puff::index::jointLog] [info] Reading the reference files ...\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.257] [puff::index::jointLog] [info] positional integer width = 23\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.257] [puff::index::jointLog] [info] seqSize = 5,890,652\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.257] [puff::index::jointLog] [info] rankSize = 5,890,652\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.257] [puff::index::jointLog] [info] edgeVecSize = 0\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.257] [puff::index::jointLog] [info] num keys = 5,647,862\n",
      "\u001b[00mfor info, total work write each  : 2.331    total work inram from level 3 : 4.322  total work raw : 25.000 \n",
      "[Building BooPHF]  99.9 %   elapsed:   0 min 0  sec   remaining:   0 min 0  sec\n",
      "Bitarray        29598784  bits (100.00 %)   (array + ranks )\n",
      "final hash             0  bits (0.00 %) (nb in final hash 0)\n",
      "\u001b[00m[2020-12-22 10:50:53.472] [puff::index::jointLog] [info] mphf size = 3.52845 MB\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.477] [puff::index::jointLog] [info] chunk size = 2,945,326\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.478] [puff::index::jointLog] [info] chunk 0 = [0, 2,945,326)\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.478] [puff::index::jointLog] [info] chunk 1 = [2,945,326, 5,890,622)\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.857] [puff::index::jointLog] [info] finished populating pos vector\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.857] [puff::index::jointLog] [info] writing index components\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:53.869] [puff::index::jointLog] [info] finished writing dense pufferfish index\n",
      "\u001b[00m[2020-12-22 10:50:53.986] [jLog] [info] done building index\n"
     ]
    }
   ],
   "source": [
    "# Get PA14 index\n",
    "! salmon index -t $paths.PA14_REF -i $paths.PA14_INDEX"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Version Info: ### PLEASE UPGRADE SALMON ###\n",
      "### A newer version of salmon with important bug fixes and improvements is available. ####\n",
      "###\n",
      "The newest version, available at https://github.com/COMBINE-lab/salmon/releases\n",
      "contains new features, improvements, and bug fixes; please upgrade at your\n",
      "earliest convenience.\n",
      "###\n",
      "Sign up for the salmon mailing list to hear about new versions, features and updates at:\n",
      "https://oceangenomics.com/subscribe\n",
      "###\n",
      "[2020-12-22 10:50:54.175] [jLog] [warning] The salmon index is being built without any decoy sequences.  It is recommended that decoy sequence (either computed auxiliary decoy sequence or the genome of the organism) be provided during indexing. Further details can be found at https://salmon.readthedocs.io/en/latest/salmon.html#preparing-transcriptome-indices-mapping-based-mode.\n",
      "[2020-12-22 10:50:54.176] [jLog] [info] building index\n",
      "out : /home/alexandra/Documents/Data/Core_accessory/phage_index\n",
      "\u001b[00m[2020-12-22 10:50:54.176] [puff::index::jointLog] [info] Running fixFasta\n",
      "\u001b[00m\n",
      "[Step 1 of 4] : counting k-mers\n",
      "\u001b[35m[2020-12-22 10:50:54.270] [puff::index::jointLog] [warning] Entry with header [NC_028999.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.276] [puff::index::jointLog] [warning] Entry with header [HQ630627.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.282] [puff::index::jointLog] [warning] Entry with header [MT133560.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.289] [puff::index::jointLog] [warning] Entry with header [MK599315.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.545] [puff::index::jointLog] [warning] Entry with header [CP029707.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.597] [puff::index::jointLog] [warning] Entry with header [LR588166.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.604] [puff::index::jointLog] [warning] Entry with header [AP019418.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.618] [puff::index::jointLog] [warning] Entry with header [NC_007623.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.659] [puff::index::jointLog] [warning] Entry with header [AJ697969.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.742] [puff::index::jointLog] [warning] Entry with header [NC_010821.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.788] [puff::index::jointLog] [warning] Entry with header [MK797984.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.804] [puff::index::jointLog] [warning] Entry with header [EU197055.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.844] [puff::index::jointLog] [warning] Entry with header [NC_017972.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.869] [puff::index::jointLog] [warning] Entry with header [JQ768459.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.885] [puff::index::jointLog] [warning] Entry with header [NC_022096.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.890] [puff::index::jointLog] [warning] Entry with header [KF147891.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.919] [puff::index::jointLog] [warning] Entry with header [NC_016571.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:54.968] [puff::index::jointLog] [warning] Entry with header [JN627160.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.022] [puff::index::jointLog] [warning] Entry with header [NC_041994.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.042] [puff::index::jointLog] [warning] Entry with header [MF063068.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.125] [puff::index::jointLog] [warning] Entry with header [MK552327.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.161] [puff::index::jointLog] [warning] Entry with header [MF042360.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.187] [puff::index::jointLog] [warning] Entry with header [KU521356.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.222] [puff::index::jointLog] [warning] Entry with header [MW057919.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.235] [puff::index::jointLog] [warning] Entry with header [NC_042081.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.404] [puff::index::jointLog] [warning] Entry with header [NZ_CP029707.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.418] [puff::index::jointLog] [warning] Entry with header [MN103543.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.462] [puff::index::jointLog] [warning] Entry with header [MF805716.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.594] [puff::index::jointLog] [warning] Entry with header [NC_004629.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.620] [puff::index::jointLog] [warning] Entry with header [MN871489.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.635] [puff::index::jointLog] [warning] Entry with header [MN871480.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.669] [puff::index::jointLog] [warning] Entry with header [NC_042060.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.718] [puff::index::jointLog] [warning] Entry with header [AF399011.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.723] [puff::index::jointLog] [warning] Entry with header [JX233784.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.802] [puff::index::jointLog] [warning] Entry with header [DI373487.1] was longer than 200000 nucleotides.  This is probably a chromosome instead of a transcript.\n",
      "\u001b[00m\n",
      "\u001b[35m[2020-12-22 10:50:55.802] [puff::index::jointLog] [warning] Removed 391 transcripts that were sequence duplicates of indexed transcripts.\n",
      "\u001b[00m\u001b[35m[2020-12-22 10:50:55.802] [puff::index::jointLog] [warning] If you wish to retain duplicate transcripts, please use the `--keepDuplicates` flag\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:55.802] [puff::index::jointLog] [info] Replaced 6,326 non-ATCG nucleotides\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:55.802] [puff::index::jointLog] [info] Clipped poly-A tails from 0 transcripts\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[00mwrote 1128 cleaned references\n",
      "\u001b[00m[2020-12-22 10:50:55.856] [puff::index::jointLog] [info] Filter size not provided; estimating from number of distinct k-mers\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:50:56.178] [puff::index::jointLog] [info] ntHll estimated 22009247 distinct k-mers, setting filter size to 2^29\n",
      "\u001b[00mThreads = 2\n",
      "Vertex length = 31\n",
      "Hash functions = 5\n",
      "Filter size = 536870912\n",
      "Capacity = 2\n",
      "Files: \n",
      "/home/alexandra/Documents/Data/Core_accessory/phage_index/ref_k31_fixed.fa\n",
      "--------------------------------------------------------------------------------\n",
      "Round 0, 0:536870912\n",
      "Pass\tFilling\tFiltering\n",
      "1\t5\t14\t\n",
      "2\t1\t0\n",
      "True junctions count = 183497\n",
      "False junctions count = 225552\n",
      "Hash table size = 409049\n",
      "Candidate marks count = 3344391\n",
      "--------------------------------------------------------------------------------\n",
      "Reallocating bifurcations time: 0\n",
      "True marks count: 2238898\n",
      "Edges construction time: 2\n",
      "--------------------------------------------------------------------------------\n",
      "Distinct junctions = 183497\n",
      "\n",
      "allowedIn: 11\n",
      "Max Junction ID: 184101\n",
      "seen.size():1472817 kmerInfo.size():184102\n",
      "approximateContigTotalLength: 10966072\n",
      "counters for complex kmers:\n",
      "(prec>1 & succ>1)=3091 | (succ>1 & isStart)=23 | (prec>1 & isEnd)=10 | (isStart & isEnd)=2\n",
      "contig count: 279402 element count: 30362951 complex nodes: 3126\n",
      "# of ones in rank vector: 279401\n",
      "\u001b[00m[2020-12-22 10:51:22.250] [puff::index::jointLog] [info] Starting the Pufferfish indexing by reading the GFA binary file.\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.250] [puff::index::jointLog] [info] Setting the index/BinaryGfa directory /home/alexandra/Documents/Data/Core_accessory/phage_index\n",
      "\u001b[00msize = 30362951\n",
      "-----------------------------------------\n",
      "| Loading contigs | Time = 2.4102 ms\n",
      "-----------------------------------------\n",
      "size = 30362951\n",
      "-----------------------------------------\n",
      "| Loading contig boundaries | Time = 1.213 ms\n",
      "-----------------------------------------\n",
      "Number of ones: 279401\n",
      "Number of ones per inventory item: 512\n",
      "Inventory entries filled: 546\n",
      "279401\n",
      "\u001b[00m[2020-12-22 10:51:22.300] [puff::index::jointLog] [info] Done wrapping the rank vector with a rank9sel structure.\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.306] [puff::index::jointLog] [info] contig count for validation: 279,401\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.402] [puff::index::jointLog] [info] Total # of Contigs : 279,401\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.402] [puff::index::jointLog] [info] Total # of numerical Contigs : 279,401\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.408] [puff::index::jointLog] [info] Total # of contig vec entries: 2,264,464\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.408] [puff::index::jointLog] [info] bits per offset entry 22\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.431] [puff::index::jointLog] [info] Done constructing the contig vector. 279402\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.685] [puff::index::jointLog] [info] # segments = 279,401\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.685] [puff::index::jointLog] [info] total length = 30,362,951\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.703] [puff::index::jointLog] [info] Reading the reference files ...\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.889] [puff::index::jointLog] [info] positional integer width = 25\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.889] [puff::index::jointLog] [info] seqSize = 30,362,951\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.889] [puff::index::jointLog] [info] rankSize = 30,362,951\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.889] [puff::index::jointLog] [info] edgeVecSize = 0\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:22.889] [puff::index::jointLog] [info] num keys = 21,980,921\n",
      "\u001b[00mfor info, total work write each  : 2.331    total work inram from level 3 : 4.322  total work raw : 25.000 \n",
      "[Building BooPHF]  100  %   elapsed:   0 min 1  sec   remaining:   0 min 0  sec\n",
      "Bitarray       115178752  bits (100.00 %)   (array + ranks )\n",
      "final hash             0  bits (0.00 %) (nb in final hash 0)\n",
      "\u001b[00m[2020-12-22 10:51:24.061] [puff::index::jointLog] [info] mphf size = 13.7304 MB\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:24.082] [puff::index::jointLog] [info] chunk size = 15,181,476\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:24.082] [puff::index::jointLog] [info] chunk 0 = [0, 15,181,477)\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:24.082] [puff::index::jointLog] [info] chunk 1 = [15,181,477, 30,362,921)\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:26.055] [puff::index::jointLog] [info] finished populating pos vector\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:26.055] [puff::index::jointLog] [info] writing index components\n",
      "\u001b[00m\u001b[00m[2020-12-22 10:51:26.215] [puff::index::jointLog] [info] finished writing dense pufferfish index\n",
      "\u001b[00m[2020-12-22 10:51:26.229] [jLog] [info] done building index\n"
     ]
    }
   ],
   "source": [
    "# Get phage index\n",
    "! salmon index -t $paths.PHAGE_REF -i $paths.PHAGE_INDEX"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:core_acc] *",
   "language": "python",
   "name": "conda-env-core_acc-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
